# compomics-utilities #

  * [What is compomics-utilities?](#what-is-compomics-utilities?)
  * [Projects Using compomics-utilities](#projects-using-compomics-utilities)
  * [Using compomics-utilities](#using-compomics-utilities)
  * [Download](#download)
  * [Maven Dependency](#maven-dependency)
  * [Screenshots](#screenshots)

  * [Bioinformatics for Proteomics Tutorial](http://compomics.com/bioinformatics-for-proteomics/)

---

**compomics-utilities publication:**
  * [Barsnes et al: BMC Bioinformatics. 2011 Mar 8;12(1):70](http://www.ncbi.nlm.nih.gov/pubmed/21385435).
  * If you use **compomics-utilities** as part of a paper, please include the reference above.

---

## What is compomics-utilities? ##

The [Computational Omics and Systems Biology Group](http://www.compomics.com) develops various bioinformatics tools for analyzing omics data.

**compomics-utilities** is a library containing code shared by many of our research projects, amongst others containing panels for visualizing spectra and chromatograms and objects for representing peptides and proteins etc. We believe that this library can be of use to other research groups doing computational proteomics, and have therefore made it available as open source.

[Go to top of page](#compomics-utilities)

---

## Projects using compomics-utilities ##

| **Project** | **Description** | **Publication** |
|:------------|:----------------|:----------------|
| [PeptideShaker](http://compomics.github.io/projects/peptide-shaker.html) | _interpretation of proteomics identification results_|_[Vaudel et al: Nature Biotechnol. 2015 Jan;33(1):22–24.](http://www.nature.com/nbt/journal/v33/n1/full/nbt.3109.html)_|
| [MetaProteomeAnalyzer](http://code.google.com/p/meta-proteome-analyzer) | _analyzing meta-proteomics data_|_[Muth et al: J Proteome Res. 2015 (in press).](http://www.ncbi.nlm.nih.gov/pubmed/25660940)_|
| [DeNovoGUI](http://compomics.github.io/projects/denovogui.html) | _de novo sequencing of tandem mass spectra_|_[Muth at al: J Proteome Res. 2014 Feb 7;13(2):1143-6.](http://www.ncbi.nlm.nih.gov/pubmed/24295440)_|
| [pride-asap](http://compomics.github.io/projects/pride-asa-pipeline.html) | _uniform annotation of identified spectra stored in PRIDE_|_[Hulstaert et al: J Proteomics. 2013 Apr 17](http://www.ncbi.nlm.nih.gov/pubmed/23603108)_|
| [ProteoCloud](http://proteocloud.googlecode.com) | _proteomics cloud computing pipeline_|_[Muth et al: J Proteomics. 2013 Aug 2;88:104-8](http://www.ncbi.nlm.nih.gov/pubmed/23305951)_|
| [thermo-msf-parser](http://compomics.github.io/projects/thermo-msf-parser.html) | _parser and viewer for thermo msf files_|_[Colaert et al: J Proteome Res. 2011;10(8):3840-3](http://www.ncbi.nlm.nih.gov/pubmed/21714566)_|
| [SearchGUI](http://compomics.github.io/projects/searchgui.html) | _graphical user interface for proteomics identification search engines_|_[Vaudel et al: Proteomics 2011;11(5):996-9](http://www.ncbi.nlm.nih.gov/pubmed/21337703)_|
| [XTandem Parser](http://compomics.github.io/projects/xtandem-parser.html) |_java parser for X!Tandem output xml files_|_[Muth et al: Proteomics 2010;10(7):1522-4](http://www.ncbi.nlm.nih.gov/pubmed/20140905)_|
| [jmzML](http://jmzML.googlecode.com) |_java implementation of the PSI-MS mzML specification_|_[Côté et al: Proteomics 2010;10(7):1332-5](http://www.ncbi.nlm.nih.gov/pubmed/20127693)_|
| [Fragmentation Analyzer](http://compomics.github.io/projects/fragmentation-analyzer.html) | _analyzing MS/MS fragmentation data_|_[Barsnes et al: Proteomics 2010;10(5):1087-90](http://www.ncbi.nlm.nih.gov/pubmed/20049869)_|
| [ms\_lims](http://compomics.github.io/projects/ms-lims.html) | _mass spectrometry based proteomics information management system_|_[Helsens et al: Proteomics 2010;10(6):1261-4](http://www.ncbi.nlm.nih.gov/pubmed/20058248)_|
| [Rover](https://github.com/compomics/compomics-rover) | _visualize and validate quantitative proteomics data_|_[Colaert et al: Proteomics 2010;10(6):1226-9](http://www.ncbi.nlm.nih.gov/pubmed/20058247)_|
| [IceLogo](http://compomics.github.io/projects/icelogo.html) | _next-generation visualization of protein consensus sequences_|_[Colaert et al: Nature Methods 2009;6(11):786-7](http://www.ncbi.nlm.nih.gov/pubmed/19876014)_|
| [OMSSA Parser](http://compomics.github.io/projects/omssa-parser.html) | _java parser for OMSSA omx files_|_[Barsnes et al: Proteomics 2009;9(14):3772-4](http://www3.interscience.wiley.com/journal/122524294/abstract)_|
| [Peptizer](https://github.com/compomics/peptizer) | _automating manual validation of MS/MS search results_|_[Helsens et al: MCP 2008;7(12):2363-72](http://www.ncbi.nlm.nih.gov/pubmed/18667410)_|
| [MascotDatfile](http://compomics.github.io/projects/mascotdatfile.html) | _java API for MS/MS search results by Mascot_|_[Helsens et al: Proteomics 2007;7(3):364-6](http://www.ncbi.nlm.nih.gov/pubmed/17203510)_|
| [DBToolkit](https://github.com/compomics/dbtoolkit) | _manipulating fasta sequence databases_|_[Martens et al: Bioinformatics 2005;21(17):3584-5](http://www.ncbi.nlm.nih.gov/pubmed/16030071)_|

[Go to top of page](#compomics-utilities)

---

## Using compomics-utilities ##

For example code and demos showing how the **compomics-utilities** library can be used in your project see [Examples](https://github.com/compomics/compomics-utilities/wiki/Examples). Also check the [JavaDoc](http://genesis.ugent.be/maven2/com/compomics/utilities/javadoc/).

[Go to top of page](#compomics-utilities)

---

## Download ##

We strongly recommend [Maven](http://maven.apache.org) when using compomics-utilities, as this makes sure that one gets all the dependencies right. See [Maven Dependency](#Maven_Dependency.md) below.

All compomics-utilities builds are available [here](http://genesis.ugent.be/maven2/com/compomics/utilities/).

See also the [JavaDoc](http://genesis.ugent.be/maven2/com/compomics/utilities/javadoc/).

If you for some reason cannot use Maven and need a complete build of compomics-utilities, please let us know and we will send you the latest build as a zip file.

[Go to top of page](#compomics-utilities)

---

## Maven Dependency ##

**compomics-utilities** is available for use in [Maven](http://maven.apache.org) projects:

```
<dependency>
    <groupId>com.compomics</groupId>
    <artifactId>utilities</artifactId>
    <version>X.Y.Z</version>
</dependency>
```

```
<repository>
    <id>genesis-maven2-repository</id>
    <name>Genesis maven2 repository</name>
    <url>http://genesis.UGent.be/maven2</url>
</repository>
```

Update the version number to latest released version, see the [Maven repository](http://genesis.ugent.be/maven2/com/compomics/utilities/).

[Go to top of page](#compomics-utilities)

---

## Screenshots ##

(Click on figure to see the full size version)

[![](https://github.com/compomics/compomics-utilities/wiki/images/WellcomePanel_small.png)](https://github.com/compomics/compomics-utilities/wiki/images/WellcomePanel.png)
[![](https://github.com/compomics/compomics-utilities/wiki/images/SpectrumPanel_small.png)](https://github.com/compomics/compomics-utilities/wiki/images/SpectrumPanel.png)
[![](https://github.com/compomics/compomics-utilities/wiki/images/ChromatogramPanel_small.png)](https://github.com/compomics/compomics-utilities/wiki/images/ChromatogramPanel.png)
[![](https://github.com/compomics/compomics-utilities/wiki/images/IsotopicDistributionPanel_small.png)](https://github.com/compomics/compomics-utilities/wiki/images/IsotopicDistributionPanel.png)
[![](https://github.com/compomics/compomics-utilities/wiki/images/InSilicoDigestionPanel_small.png)](https://github.com/compomics/compomics-utilities/wiki/images/InSilicoDigestionPanel.png)

[Go to top of page](#compomics-utilities)
