<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="de">
<head>
<!-- Generated by javadoc -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>FMIndex (computational omics utilities 5.0.1-SNAPSHOT API)</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="FMIndex (computational omics utilities 5.0.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10,"i20":10,"i21":10,"i22":10,"i23":10,"i24":10,"i25":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FMIndex.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/AccessionMetaData.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/MassIndexMap.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" target="_top">Frames</a></li>
<li><a href="FMIndex.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.compomics.util.experiment.identification.protein_inference.fm_index</div>
<h2 title="Class FMIndex" class="title">Class FMIndex</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>org.zoodb.api.impl.ZooPC</li>
<li>
<ul class="inheritance">
<li><a href="../../../../../../../com/compomics/util/db/object/DbObject.html" title="class in com.compomics.util.db.object">com.compomics.util.db.object.DbObject</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../../../com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">com.compomics.util.experiment.personalization.ExperimentObject</a></li>
<li>
<ul class="inheritance">
<li>com.compomics.util.experiment.identification.protein_inference.fm_index.FMIndex</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/FastaMapper.html" title="interface in com.compomics.util.experiment.identification.protein_inference">FastaMapper</a>, <a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a>, <a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></dd>
</dl>
<hr>
<br>
<pre>public class <span class="typeNameLabel">FMIndex</span>
extends <a href="../../../../../../../com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a>
implements <a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/FastaMapper.html" title="interface in com.compomics.util.experiment.identification.protein_inference">FastaMapper</a>, <a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a>, <a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></pre>
<div class="block">The FM index.</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Dominik Kopczynski, Marc Vaudel</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#indexStringLengths">indexStringLengths</a></span></code>
<div class="block">Length of the indexed string (all concatenated protein sequences).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;int[]&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#lessTablesPrimary">lessTablesPrimary</a></span></code>
<div class="block">Less table for doing an update step according to the LF step.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;int[]&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#lessTablesReversed">lessTablesReversed</a></span></code>
<div class="block">Less table for doing an update step according to the LF step reversed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#occurrenceTablesPrimary">occurrenceTablesPrimary</a></span></code>
<div class="block">Wavelet tree for storing the burrows wheeler transform.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#occurrenceTablesReversed">occurrenceTablesReversed</a></span></code>
<div class="block">Wavelet tree for storing the burrows wheeler transform reversed.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="fields.inherited.from.class.com.compomics.util.experiment.personalization.ExperimentObject">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;com.compomics.util.experiment.personalization.<a href="../../../../../../../com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></h3>
<code><a href="../../../../../../../com/compomics/util/experiment/personalization/ExperimentObject.html#NO_KEY">NO_KEY</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#FMIndex-java.io.File-com.compomics.util.experiment.io.biology.protein.FastaParameters-com.compomics.util.waiting.WaitingHandler-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters-">FMIndex</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;fastaFile,
       <a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a>&nbsp;fastaParameters,
       <a href="../../../../../../../com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a>&nbsp;waitingHandler,
       boolean&nbsp;displayProgress,
       <a href="../../../../../../../com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a>&nbsp;modificationSettings,
       <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a>&nbsp;peptideVariantsPreferences)</code>
<div class="block">Constructor.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#FMIndex-java.io.File-com.compomics.util.experiment.io.biology.protein.FastaParameters-com.compomics.util.waiting.WaitingHandler-boolean-com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters-com.compomics.util.parameters.identification.search.SearchParameters-">FMIndex</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;fastaFile,
       <a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a>&nbsp;fastaParameters,
       <a href="../../../../../../../com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a>&nbsp;waitingHandler,
       boolean&nbsp;displayProgress,
       <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a>&nbsp;peptideVariantsPreferences,
       <a href="../../../../../../../com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a>&nbsp;searchParameters)</code>
<div class="block">Constructor.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#addModificationPattern-com.compomics.util.experiment.biology.modifications.Modification-">addModificationPattern</a></span>(<a href="../../../../../../../com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a>&nbsp;modification)</code>
<div class="block">Adds a modification pattern for bitwise pattern search.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#checkModificationPattern-com.compomics.util.experiment.identification.protein_inference.PeptideProteinMapping-">checkModificationPattern</a></span>(<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&nbsp;peptideProteinMapping)</code>
<div class="block">Checking if peptide-protein should be discarded due to pattern
 modification conflict.</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#computeInverseMassValue-double-double-">computeInverseMassValue</a></span>(double&nbsp;currentMass,
                       double&nbsp;refMass)</code>
<div class="block">Compute the inverse mass value.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>int[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#computeMappingRanges-double-">computeMappingRanges</a></span>(double&nbsp;mass)</code>
<div class="block">Compute mapping ranges.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#computeMassValue-double-double-">computeMassValue</a></span>(double&nbsp;currentMass,
                double&nbsp;refMass)</code>
<div class="block">Compute the mass value.</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getAccessions--">getAccessions</a></span>()</code>
<div class="block">Returns all accessions loaded in the provider.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getAllocatedBytes--">getAllocatedBytes</a></span>()</code>
<div class="block">Computes the number of allocated bytes.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html?is-external=true" title="class or interface in java.util">HashSet</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getDecoyAccessions--">getDecoyAccessions</a></span>()</code>
<div class="block">Returns the decoy accessions.</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getDescription-java.lang.String-">getDescription</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>
<div class="block">Returns the description of the protein with the given accession.</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getGeneName-java.lang.String-">getGeneName</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>
<div class="block">Returns the gene name for the given protein.</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getHeader-java.lang.String-">getHeader</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;proteinAccession)</code>
<div class="block">Returns the FASTA header of the protein as found in the FASTA file.</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html" title="enum in com.compomics.util.experiment.io.biology.protein">ProteinDatabase</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinDatabase-java.lang.String-">getProteinDatabase</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>
<div class="block">Returns the the protein database for the given protein.</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinEvidence-java.lang.String-">getProteinEvidence</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>
<div class="block">Returns an integer representing the protein evidence level as indexed by uniprot.</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMapping-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getProteinMapping</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;peptide,
                 <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingParameters)</code>
<div class="block">Returns the protein mapping in the FASTA file loaded in the sequence
 factory for the given peptide sequence in a map: peptide sequence found
 in the FASTA file | protein accession | list of indexes of the peptide
 sequence on the protein sequence. 0 is the first amino acid.</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMapping-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getProteinMapping</a></span>(<a href="../../../../../../../com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a>&nbsp;tag,
                 <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingPreferences)</code>
<div class="block">Returns the protein mappings for the given peptide sequence.</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMappingWithoutVariants-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">getProteinMappingWithoutVariants</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;peptide,
                                <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                int&nbsp;indexPart)</code>
<div class="block">Exact mapping peptides against the proteome.</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMappingWithoutVariants-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">getProteinMappingWithoutVariants</a></span>(<a href="../../../../../../../com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a>&nbsp;tag,
                                <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingPreferences,
                                int&nbsp;indexPart)</code>
<div class="block">Mapping tags against proteome without variants.</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMappingWithVariants-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">getProteinMappingWithVariants</a></span>(<a href="../../../../../../../com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a>&nbsp;tag,
                             <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingPreferences,
                             int&nbsp;indexPart)</code>
<div class="block">Mapping tags against proteome with variants.</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMappingWithVariantsGeneric-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">getProteinMappingWithVariantsGeneric</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;peptide,
                                    <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                    int&nbsp;indexPart)</code>
<div class="block">Variant tolerant mapping peptides against the proteome.</div>
</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMappingWithVariantsSpecific-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">getProteinMappingWithVariantsSpecific</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;peptide,
                                     <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                     int&nbsp;indexPart)</code>
<div class="block">Variant tolerant mapping peptides against the proteome.</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getSequence-java.lang.String-">getSequence</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;proteinAccession)</code>
<div class="block">Returns the protein sequence for the given accession.</div>
</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getSimpleDescription-java.lang.String-">getSimpleDescription</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>
<div class="block">Returns the simple description of the protein with the given accession.</div>
</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getSubsequence-java.lang.String-int-int-">getSubsequence</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession,
              int&nbsp;start,
              int&nbsp;end)</code>
<div class="block">Returns the subsequence of the sequence of a given protein.</div>
</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getTaxonomy-java.lang.String-">getTaxonomy</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>
<div class="block">Returns the taxonomy for the given protein.</div>
</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#pepMass-java.lang.String-">pepMass</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;peptide)</code>
<div class="block">Computing the mass of a peptide.</div>
</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#withinMassTolerance-double-int-">withinMassTolerance</a></span>(double&nbsp;mass,
                   int&nbsp;numX)</code>
<div class="block">Lookup, if mass can be described a combination of numX different amino
 acids</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.com.compomics.util.experiment.personalization.ExperimentObject">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.compomics.util.experiment.personalization.<a href="../../../../../../../com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></h3>
<code><a href="../../../../../../../com/compomics/util/experiment/personalization/ExperimentObject.html#addUrParam-com.compomics.util.experiment.personalization.UrParameter-">addUrParam</a>, <a href="../../../../../../../com/compomics/util/experiment/personalization/ExperimentObject.html#asLong-java.lang.String-">asLong</a>, <a href="../../../../../../../com/compomics/util/experiment/personalization/ExperimentObject.html#clearParametersMap--">clearParametersMap</a>, <a href="../../../../../../../com/compomics/util/experiment/personalization/ExperimentObject.html#getUrParam-com.compomics.util.experiment.personalization.UrParameter-">getUrParam</a>, <a href="../../../../../../../com/compomics/util/experiment/personalization/ExperimentObject.html#getUrParams--">getUrParams</a>, <a href="../../../../../../../com/compomics/util/experiment/personalization/ExperimentObject.html#removeUrParam-long-">removeUrParam</a>, <a href="../../../../../../../com/compomics/util/experiment/personalization/ExperimentObject.html#setUrParams-java.util.HashMap-">setUrParams</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.com.compomics.util.db.object.DbObject">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.compomics.util.db.object.<a href="../../../../../../../com/compomics/util/db/object/DbObject.html" title="class in com.compomics.util.db.object">DbObject</a></h3>
<code><a href="../../../../../../../com/compomics/util/db/object/DbObject.html#getFirstLevel--">getFirstLevel</a>, <a href="../../../../../../../com/compomics/util/db/object/DbObject.html#getId--">getId</a>, <a href="../../../../../../../com/compomics/util/db/object/DbObject.html#readDBMode--">readDBMode</a>, <a href="../../../../../../../com/compomics/util/db/object/DbObject.html#setFirstLevel-boolean-">setFirstLevel</a>, <a href="../../../../../../../com/compomics/util/db/object/DbObject.html#setId-long-">setId</a>, <a href="../../../../../../../com/compomics/util/db/object/DbObject.html#writeDBMode--">writeDBMode</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.org.zoodb.api.impl.ZooPC">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.zoodb.api.impl.ZooPC</h3>
<code>jdoZooEvict, jdoZooGetBackup, jdoZooGetClassDef, jdoZooGetContext, jdoZooGetNode, jdoZooGetOid, jdoZooGetTimestamp, jdoZooHasState, jdoZooInit, jdoZooIsDeleted, jdoZooIsDetached, jdoZooIsDirty, jdoZooIsNew, jdoZooIsPersistent, jdoZooIsStateHollow, jdoZooIsTransactional, jdoZooMarkClean, jdoZooMarkDeleted, jdoZooMarkDetached, jdoZooMarkDirty, jdoZooMarkHollow, jdoZooMarkTransient, jdoZooSetOid, jdoZooSetTimestamp, toString, zooActivateRead, zooActivateWrite, zooActivateWrite</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#clone--" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#equals-java.lang.Object-" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#finalize--" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#getClass--" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#hashCode--" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notify--" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notifyAll--" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait--" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait-long-" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait-long-int-" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="occurrenceTablesPrimary">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>occurrenceTablesPrimary</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a>&gt; occurrenceTablesPrimary</pre>
<div class="block">Wavelet tree for storing the burrows wheeler transform.</div>
</li>
</ul>
<a name="occurrenceTablesReversed">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>occurrenceTablesReversed</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a>&gt; occurrenceTablesReversed</pre>
<div class="block">Wavelet tree for storing the burrows wheeler transform reversed.</div>
</li>
</ul>
<a name="lessTablesPrimary">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>lessTablesPrimary</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;int[]&gt; lessTablesPrimary</pre>
<div class="block">Less table for doing an update step according to the LF step.</div>
</li>
</ul>
<a name="lessTablesReversed">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>lessTablesReversed</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;int[]&gt; lessTablesReversed</pre>
<div class="block">Less table for doing an update step according to the LF step reversed.</div>
</li>
</ul>
<a name="indexStringLengths">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>indexStringLengths</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt; indexStringLengths</pre>
<div class="block">Length of the indexed string (all concatenated protein sequences).</div>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="FMIndex-java.io.File-com.compomics.util.experiment.io.biology.protein.FastaParameters-com.compomics.util.waiting.WaitingHandler-boolean-com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters-com.compomics.util.parameters.identification.search.SearchParameters-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>FMIndex</h4>
<pre>public&nbsp;FMIndex(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;fastaFile,
               <a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a>&nbsp;fastaParameters,
               <a href="../../../../../../../com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a>&nbsp;waitingHandler,
               boolean&nbsp;displayProgress,
               <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a>&nbsp;peptideVariantsPreferences,
               <a href="../../../../../../../com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a>&nbsp;searchParameters)
        throws <a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Constructor. If modification settings are provided the index will contain
 modification information, ignored if null.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fastaFile</code> - the fasta file to index</dd>
<dd><code>fastaParameters</code> - the parameters for the fasta file parsing</dd>
<dd><code>waitingHandler</code> - the waiting handler</dd>
<dd><code>displayProgress</code> - if true, the progress is displayed</dd>
<dd><code>peptideVariantsPreferences</code> - contains all parameters for variants</dd>
<dd><code>searchParameters</code> - the search parameters</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - exception thrown if an error occurs while iterating
 the fasta file.</dd>
</dl>
</li>
</ul>
<a name="FMIndex-java.io.File-com.compomics.util.experiment.io.biology.protein.FastaParameters-com.compomics.util.waiting.WaitingHandler-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>FMIndex</h4>
<pre>public&nbsp;FMIndex(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;fastaFile,
               <a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a>&nbsp;fastaParameters,
               <a href="../../../../../../../com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a>&nbsp;waitingHandler,
               boolean&nbsp;displayProgress,
               <a href="../../../../../../../com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a>&nbsp;modificationSettings,
               <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a>&nbsp;peptideVariantsPreferences)
        throws <a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Constructor. If modification settings are provided the index will contain
 modification information, ignored if null.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fastaFile</code> - the fasta file to index</dd>
<dd><code>fastaParameters</code> - the parameters for the fasta file parsing</dd>
<dd><code>waitingHandler</code> - the waiting handler</dd>
<dd><code>displayProgress</code> - if true, the progress is displayed</dd>
<dd><code>modificationSettings</code> - contains modification parameters for
 identification</dd>
<dd><code>peptideVariantsPreferences</code> - contains all parameters for variants</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - exception thrown if an error occurs while iterating
 the fasta file.</dd>
</dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="computeMassValue-double-double-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>computeMassValue</h4>
<pre>public&nbsp;double&nbsp;computeMassValue(double&nbsp;currentMass,
                               double&nbsp;refMass)</pre>
<div class="block">Compute the mass value.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>currentMass</code> - the current mass</dd>
<dd><code>refMass</code> - the reference mass</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the mass value</dd>
</dl>
</li>
</ul>
<a name="computeInverseMassValue-double-double-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>computeInverseMassValue</h4>
<pre>public&nbsp;double&nbsp;computeInverseMassValue(double&nbsp;currentMass,
                                      double&nbsp;refMass)</pre>
<div class="block">Compute the inverse mass value.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>currentMass</code> - the current mass</dd>
<dd><code>refMass</code> - the reference mass</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the inverse mass value</dd>
</dl>
</li>
</ul>
<a name="addModificationPattern-com.compomics.util.experiment.biology.modifications.Modification-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addModificationPattern</h4>
<pre>public&nbsp;void&nbsp;addModificationPattern(<a href="../../../../../../../com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a>&nbsp;modification)</pre>
<div class="block">Adds a modification pattern for bitwise pattern search.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>modification</code> - modification object</dd>
</dl>
</li>
</ul>
<a name="checkModificationPattern-com.compomics.util.experiment.identification.protein_inference.PeptideProteinMapping-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>checkModificationPattern</h4>
<pre>public&nbsp;boolean&nbsp;checkModificationPattern(<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&nbsp;peptideProteinMapping)</pre>
<div class="block">Checking if peptide-protein should be discarded due to pattern
 modification conflict.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptideProteinMapping</code> - the peptide protein mapping</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>either yes or no</dd>
</dl>
</li>
</ul>
<a name="computeMappingRanges-double-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>computeMappingRanges</h4>
<pre>public&nbsp;int[]&nbsp;computeMappingRanges(double&nbsp;mass)</pre>
<div class="block">Compute mapping ranges.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>mass</code> - the mass</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the mapping ranges</dd>
</dl>
</li>
</ul>
<a name="getAllocatedBytes--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllocatedBytes</h4>
<pre>public&nbsp;long&nbsp;getAllocatedBytes()</pre>
<div class="block">Computes the number of allocated bytes.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>allocated bytes</dd>
</dl>
</li>
</ul>
<a name="getProteinMapping-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMapping</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMapping(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;peptide,
                                                          <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingParameters)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/FastaMapper.html#getProteinMapping-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">FastaMapper</a></code></span></div>
<div class="block">Returns the protein mapping in the FASTA file loaded in the sequence
 factory for the given peptide sequence in a map: peptide sequence found
 in the FASTA file | protein accession | list of indexes of the peptide
 sequence on the protein sequence. 0 is the first amino acid.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/FastaMapper.html#getProteinMapping-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getProteinMapping</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/FastaMapper.html" title="interface in com.compomics.util.experiment.identification.protein_inference">FastaMapper</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptide</code> - the peptide sequence</dd>
<dd><code>sequenceMatchingParameters</code> - the sequence matching preferences</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the peptide to protein mapping: peptide sequence &gt; protein
 accession &gt; index in the protein An empty map if not</dd>
</dl>
</li>
</ul>
<a name="getProteinMappingWithoutVariants-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMappingWithoutVariants</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMappingWithoutVariants(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;peptide,
                                                                         <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                                                         int&nbsp;indexPart)</pre>
<div class="block">Exact mapping peptides against the proteome.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptide</code> - the peptide</dd>
<dd><code>seqMatchPref</code> - the sequence matching preferences</dd>
<dd><code>indexPart</code> - the index part</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the mapping</dd>
</dl>
</li>
</ul>
<a name="getProteinMappingWithVariantsGeneric-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMappingWithVariantsGeneric</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMappingWithVariantsGeneric(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;peptide,
                                                                             <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                                                             int&nbsp;indexPart)</pre>
<div class="block">Variant tolerant mapping peptides against the proteome.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptide</code> - the peptide</dd>
<dd><code>seqMatchPref</code> - the sequence match preferences</dd>
<dd><code>indexPart</code> - the index part</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the mapping</dd>
</dl>
</li>
</ul>
<a name="getProteinMappingWithVariantsSpecific-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMappingWithVariantsSpecific</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMappingWithVariantsSpecific(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;peptide,
                                                                              <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                                                              int&nbsp;indexPart)</pre>
<div class="block">Variant tolerant mapping peptides against the proteome.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptide</code> - the peptide</dd>
<dd><code>seqMatchPref</code> - the sequence matching preferences</dd>
<dd><code>indexPart</code> - the index part</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the mapping</dd>
</dl>
</li>
</ul>
<a name="pepMass-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>pepMass</h4>
<pre>public&nbsp;double&nbsp;pepMass(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;peptide)</pre>
<div class="block">Computing the mass of a peptide.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptide</code> - the peptide</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the peptide mass</dd>
</dl>
</li>
</ul>
<a name="withinMassTolerance-double-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>withinMassTolerance</h4>
<pre>public&nbsp;boolean&nbsp;withinMassTolerance(double&nbsp;mass,
                                   int&nbsp;numX)</pre>
<div class="block">Lookup, if mass can be described a combination of numX different amino
 acids</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>mass</code> - to be described</dd>
<dd><code>numX</code> - number of Xs</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>decision</dd>
</dl>
</li>
</ul>
<a name="getProteinMapping-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMapping</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMapping(<a href="../../../../../../../com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a>&nbsp;tag,
                                                          <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingPreferences)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/FastaMapper.html#getProteinMapping-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">FastaMapper</a></code></span></div>
<div class="block">Returns the protein mappings for the given peptide sequence. Peptide
 sequence | Protein accession | Index in the protein. An empty map if not
 found.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/FastaMapper.html#getProteinMapping-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getProteinMapping</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/FastaMapper.html" title="interface in com.compomics.util.experiment.identification.protein_inference">FastaMapper</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>tag</code> - the tag to look for in the tree. Must contain a consecutive
 amino acid sequence of longer or equal size than the initialTagSize of
 the tree</dd>
<dd><code>sequenceMatchingPreferences</code> - the sequence matching preferences</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the protein mapping for the given peptide sequence</dd>
</dl>
</li>
</ul>
<a name="getProteinMappingWithoutVariants-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMappingWithoutVariants</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMappingWithoutVariants(<a href="../../../../../../../com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a>&nbsp;tag,
                                                                         <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingPreferences,
                                                                         int&nbsp;indexPart)</pre>
<div class="block">Mapping tags against proteome without variants.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>tag</code> - the tag</dd>
<dd><code>sequenceMatchingPreferences</code> - the sequence matching preferences</dd>
<dd><code>indexPart</code> - the index part</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the protein mapping</dd>
</dl>
</li>
</ul>
<a name="getProteinMappingWithVariants-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMappingWithVariants</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMappingWithVariants(<a href="../../../../../../../com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a>&nbsp;tag,
                                                                      <a href="../../../../../../../com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingPreferences,
                                                                      int&nbsp;indexPart)</pre>
<div class="block">Mapping tags against proteome with variants.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>tag</code> - the tag</dd>
<dd><code>sequenceMatchingPreferences</code> - the sequence matching preferences</dd>
<dd><code>indexPart</code> - the index part</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the protein mapping</dd>
</dl>
</li>
</ul>
<a name="getSequence-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequence</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSequence(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;proteinAccession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getSequence-java.lang.String-">SequenceProvider</a></code></span></div>
<div class="block">Returns the protein sequence for the given accession.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getSequence-java.lang.String-">getSequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>proteinAccession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the sequence of the protein</dd>
</dl>
</li>
</ul>
<a name="getSubsequence-java.lang.String-int-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSubsequence</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSubsequence(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession,
                             int&nbsp;start,
                             int&nbsp;end)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getSubsequence-java.lang.String-int-int-">SequenceProvider</a></code></span></div>
<div class="block">Returns the subsequence of the sequence of a given protein. Indexes are
 0-based like for strings and no exception is thrown if indexes are out of
 bounds, the substring is trimmed.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getSubsequence-java.lang.String-int-int-">getSubsequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession of the protein</dd>
<dd><code>start</code> - the start index</dd>
<dd><code>end</code> - the end index</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the subsequence as string</dd>
</dl>
</li>
</ul>
<a name="getAccessions--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAccessions</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;getAccessions()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getAccessions--">SequenceProvider</a></code></span></div>
<div class="block">Returns all accessions loaded in the provider.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getAccessions--">getAccessions</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>all accessions loaded in the provider</dd>
</dl>
</li>
</ul>
<a name="getDecoyAccessions--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDecoyAccessions</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html?is-external=true" title="class or interface in java.util">HashSet</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;getDecoyAccessions()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getDecoyAccessions--">SequenceProvider</a></code></span></div>
<div class="block">Returns the decoy accessions.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getDecoyAccessions--">getDecoyAccessions</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the decoy accessions</dd>
</dl>
</li>
</ul>
<a name="getHeader-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getHeader</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getHeader(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;proteinAccession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getHeader-java.lang.String-">SequenceProvider</a></code></span></div>
<div class="block">Returns the FASTA header of the protein as found in the FASTA file.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getHeader-java.lang.String-">getHeader</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>proteinAccession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the FASTA header of the protein as found in the FASTA file</dd>
</dl>
</li>
</ul>
<a name="getDescription-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDescription</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getDescription(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getDescription-java.lang.String-">ProteinDetailsProvider</a></code></span></div>
<div class="block">Returns the description of the protein with the given accession.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getDescription-java.lang.String-">getDescription</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the description of the protein with the given accession</dd>
</dl>
</li>
</ul>
<a name="getSimpleDescription-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSimpleDescription</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSimpleDescription(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getSimpleDescription-java.lang.String-">ProteinDetailsProvider</a></code></span></div>
<div class="block">Returns the simple description of the protein with the given accession.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getSimpleDescription-java.lang.String-">getSimpleDescription</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the description of the protein with the given accession</dd>
</dl>
</li>
</ul>
<a name="getProteinDatabase-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinDatabase</h4>
<pre>public&nbsp;<a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html" title="enum in com.compomics.util.experiment.io.biology.protein">ProteinDatabase</a>&nbsp;getProteinDatabase(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getProteinDatabase-java.lang.String-">ProteinDetailsProvider</a></code></span></div>
<div class="block">Returns the the protein database for the given protein.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getProteinDatabase-java.lang.String-">getProteinDatabase</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the name of the protein database</dd>
</dl>
</li>
</ul>
<a name="getGeneName-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneName</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getGeneName(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getGeneName-java.lang.String-">ProteinDetailsProvider</a></code></span></div>
<div class="block">Returns the gene name for the given protein.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getGeneName-java.lang.String-">getGeneName</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the gene name for the given protein</dd>
</dl>
</li>
</ul>
<a name="getTaxonomy-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTaxonomy</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getTaxonomy(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getTaxonomy-java.lang.String-">ProteinDetailsProvider</a></code></span></div>
<div class="block">Returns the taxonomy for the given protein.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getTaxonomy-java.lang.String-">getTaxonomy</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the taxonomy for the given protein</dd>
</dl>
</li>
</ul>
<a name="getProteinEvidence-java.lang.String-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getProteinEvidence</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;getProteinEvidence(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getProteinEvidence-java.lang.String-">ProteinDetailsProvider</a></code></span></div>
<div class="block">Returns an integer representing the protein evidence level as indexed by uniprot.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getProteinEvidence-java.lang.String-">getProteinEvidence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../../com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the protein accession</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>an integer representing the protein evidence level as indexed by uniprot</dd>
</dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FMIndex.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/AccessionMetaData.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/MassIndexMap.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" target="_top">Frames</a></li>
<li><a href="FMIndex.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2018. All rights reserved.</small></p>
</body>
</html>
