<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc -->
<title>FMIndex (computational omics utilities 5.0.0-beta API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../../jquery/jquery-3.3.1.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jquery-migrate-3.0.1.js"></script>
<script type="text/javascript" src="../../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="FMIndex (computational omics utilities 5.0.0-beta API)";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":9,"i10":10,"i11":10,"i12":10,"i13":10,"i14":10,"i15":10,"i16":10,"i17":10,"i18":10,"i19":10,"i20":10,"i21":10,"i22":10,"i23":10,"i24":10,"i25":10,"i26":10,"i27":10,"i28":10,"i29":10,"i30":10,"i31":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FMIndex.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">com.compomics.util.experiment.identification.protein_inference.fm_index</a></div>
<h2 title="Class FMIndex" class="title">Class FMIndex</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">com.compomics.util.experiment.personalization.ExperimentObject</a></li>
<li>
<ul class="inheritance">
<li>com.compomics.util.experiment.identification.protein_inference.fm_index.FMIndex</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><code><a href="../FastaMapper.html" title="interface in com.compomics.util.experiment.identification.protein_inference">FastaMapper</a></code>, <code><a href="../../../io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code>, <code><a href="../../../io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></code>, <code><a href="http://docs.oracle.com/javase/7/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io" class="externalLink">Serializable</a></code></dd>
</dl>
<hr>
<pre>public class <span class="typeNameLabel">FMIndex</span>
extends <a href="../../../personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a>
implements <a href="../FastaMapper.html" title="interface in com.compomics.util.experiment.identification.protein_inference">FastaMapper</a>, <a href="../../../io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a>, <a href="../../../io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></pre>
<div class="block">The FM index.</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Dominik Kopczynski, Marc Vaudel</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../../../../../serialized-form.html#com.compomics.util.experiment.identification.protein_inference.fm_index.FMIndex">Serialized Form</a></dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static char</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#DELIMITER">DELIMITER</a></span></code></th>
<td class="colLast">
<div class="block">Character defining as delimiter between protein sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#fmiFastaFile">fmiFastaFile</a></span></code></th>
<td class="colLast">
<div class="block">The fmi fasta file</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#indexStringLengths">indexStringLengths</a></span></code></th>
<td class="colLast">
<div class="block">Length of the indexed string (all concatenated protein sequences).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;int[]&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#lessTablesPrimary">lessTablesPrimary</a></span></code></th>
<td class="colLast">
<div class="block">Less table for doing an update step according to the LF step.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;int[]&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#lessTablesReversed">lessTablesReversed</a></span></code></th>
<td class="colLast">
<div class="block">Less table for doing an update step according to the LF step reversed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#maxPTMsPerPeptide">maxPTMsPerPeptide</a></span></code></th>
<td class="colLast">
<div class="block">Maximal number of PTMs per peptide.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#occurrenceTablesPrimary">occurrenceTablesPrimary</a></span></code></th>
<td class="colLast">
<div class="block">Wavelet tree for storing the burrows wheeler transform.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#occurrenceTablesReversed">occurrenceTablesReversed</a></span></code></th>
<td class="colLast">
<div class="block">Wavelet tree for storing the burrows wheeler transform reversed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#onlyTrypticPeptides">onlyTrypticPeptides</a></span></code></th>
<td class="colLast">
<div class="block">Flag for only considering tryptic digested peptides.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static char</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#SENTINEL">SENTINEL</a></span></code></th>
<td class="colLast">
<div class="block">Sentinel character necessary for computation of the suffix array.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="fields.inherited.from.class.com.compomics.util.experiment.personalization.ExperimentObject">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;com.compomics.util.experiment.personalization.<a href="../../../personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></h3>
<code><a href="../../../personalization/ExperimentObject.html#NO_KEY">NO_KEY</a></code></li>
</ul>
</li>
</ul>
</section>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E()">FMIndex</a></span>()</code></th>
<td class="colLast">
<div class="block">Empty default constructor.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.io.File,com.compomics.util.experiment.io.biology.protein.FastaParameters,com.compomics.util.waiting.WaitingHandler,boolean,com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters,com.compomics.util.parameters.identification.search.SearchParameters)">FMIndex</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;fastaFile,
       <a href="../../../io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a>&nbsp;fastaParameters,
       <a href="../../../../waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a>&nbsp;waitingHandler,
       boolean&nbsp;displayProgress,
       <a href="../../../../parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a>&nbsp;peptideVariantsPreferences,
       <a href="../../../../parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a>&nbsp;searchParameters)</code></th>
<td class="colLast">
<div class="block">Constructor.</div>
</td>
</tr>
<tr class="altColor">
<th class="colConstructorName" scope="row"><code><span class="memberNameLink"><a href="#%3Cinit%3E(java.io.File,com.compomics.util.experiment.io.biology.protein.FastaParameters,com.compomics.util.waiting.WaitingHandler,boolean,com.compomics.util.parameters.identification.IdentificationParameters)">FMIndex</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;fastaFile,
       <a href="../../../io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a>&nbsp;fastaParameters,
       <a href="../../../../waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a>&nbsp;waitingHandler,
       boolean&nbsp;displayProgress,
       <a href="../../../../parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a>&nbsp;identificationParameters)</code></th>
<td class="colLast">
<div class="block">Constructor.</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#addModificationPattern(com.compomics.util.experiment.biology.modifications.Modification)">addModificationPattern</a></span>&#8203;(<a href="../../../biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a>&nbsp;modification)</code></th>
<td class="colLast">
<div class="block">Adds a modification pattern for bitwise pattern search.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#checkModificationPattern(com.compomics.util.experiment.identification.protein_inference.PeptideProteinMapping)">checkModificationPattern</a></span>&#8203;(<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&nbsp;peptideProteinMapping)</code></th>
<td class="colLast">
<div class="block">Checking if peptide-protein should be discarded due to pattern
 modification conflict.</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>int[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#computeMappingRanges(double)">computeMappingRanges</a></span>&#8203;(double&nbsp;mass)</code></th>
<td class="colLast">
<div class="block">Compute mapping ranges.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#computeMassTolerance(double,double)">computeMassTolerance</a></span>&#8203;(double&nbsp;currentTollerance,
                    double&nbsp;refMass)</code></th>
<td class="colLast">
<div class="block">Compute the inverse mass value.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#computeMassValue(double,double)">computeMassValue</a></span>&#8203;(double&nbsp;currentMass,
                double&nbsp;refMass)</code></th>
<td class="colLast">
<div class="block">Compute the mass value.</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAccessions()">getAccessions</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns all accessions loaded in the provider.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>long</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getAllocatedBytes()">getAllocatedBytes</a></span>()</code></th>
<td class="colLast">
<div class="block">Computes the number of allocated bytes.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html?is-external=true" title="class or interface in java.util" class="externalLink">HashSet</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDecoyAccessions()">getDecoyAccessions</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the decoy accessions.</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getDescription(java.lang.String)">getDescription</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession)</code></th>
<td class="colLast">
<div class="block">Returns the description of the protein with the given accession.</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getFileExtension(java.io.File)">getFileExtension</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getGeneName(java.lang.String)">getGeneName</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession)</code></th>
<td class="colLast">
<div class="block">Returns the gene name for the given protein.</div>
</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getHeader(java.lang.String)">getHeader</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;proteinAccession)</code></th>
<td class="colLast">
<div class="block">Returns the FASTA header of the protein as found in the FASTA file.</div>
</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code><a href="../../../io/biology/protein/ProteinDatabase.html" title="enum in com.compomics.util.experiment.io.biology.protein">ProteinDatabase</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteinDatabase(java.lang.String)">getProteinDatabase</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession)</code></th>
<td class="colLast">
<div class="block">Returns the the protein database for the given protein.</div>
</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteinEvidence(java.lang.String)">getProteinEvidence</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession)</code></th>
<td class="colLast">
<div class="block">Returns an integer representing the protein evidence level as indexed by
 UniProt.</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteinMapping(com.compomics.util.experiment.identification.amino_acid_tags.Tag,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters)">getProteinMapping</a></span>&#8203;(<a href="../../amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a>&nbsp;tag,
                 <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingPreferences)</code></th>
<td class="colLast">
<div class="block">Returns the protein mappings for the given peptide sequence.</div>
</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteinMapping(java.lang.String,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters)">getProteinMapping</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;peptide,
                 <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingParameters)</code></th>
<td class="colLast">
<div class="block">Returns the protein mapping in the FASTA file loaded in the sequence
 factory for the given peptide sequence in a map: peptide sequence found
 in the FASTA file | protein accession | list of indexes of the peptide
 sequence on the protein sequence.</div>
</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteinMappingWithoutVariants(com.compomics.util.experiment.identification.amino_acid_tags.Tag,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters,int)">getProteinMappingWithoutVariants</a></span>&#8203;(<a href="../../amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a>&nbsp;tag,
                                <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingPreferences,
                                int&nbsp;indexPart)</code></th>
<td class="colLast">
<div class="block">Mapping tags against proteome without variants.</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteinMappingWithoutVariants(java.lang.String,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters,int)">getProteinMappingWithoutVariants</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;peptide,
                                <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                int&nbsp;indexPart)</code></th>
<td class="colLast">
<div class="block">Exact mapping peptides against the proteome.</div>
</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteinMappingWithVariants(com.compomics.util.experiment.identification.amino_acid_tags.Tag,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters,int)">getProteinMappingWithVariants</a></span>&#8203;(<a href="../../amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a>&nbsp;tag,
                             <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingPreferences,
                             int&nbsp;indexPart)</code></th>
<td class="colLast">
<div class="block">Mapping tags against proteome with variants.</div>
</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteinMappingWithVariantsFixed(java.lang.String,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters,int)">getProteinMappingWithVariantsFixed</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;peptide,
                                  <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                  int&nbsp;indexPart)</code></th>
<td class="colLast">
<div class="block">Variant tolerant mapping peptides against the proteome.</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteinMappingWithVariantsGeneric(java.lang.String,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters,int)">getProteinMappingWithVariantsGeneric</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;peptide,
                                    <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                    int&nbsp;indexPart)</code></th>
<td class="colLast">
<div class="block">Variant tolerant mapping peptides against the proteome.</div>
</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getProteinMappingWithVariantsSpecific(java.lang.String,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters,int)">getProteinMappingWithVariantsSpecific</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;peptide,
                                     <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                     int&nbsp;indexPart)</code></th>
<td class="colLast">
<div class="block">Variant tolerant mapping peptides against the proteome</div>
</td>
</tr>
<tr id="i22" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSequence(java.lang.String)">getSequence</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;proteinAccession)</code></th>
<td class="colLast">
<div class="block">Returns the protein sequence for the given accession.</div>
</td>
</tr>
<tr id="i23" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSimpleDescription(java.lang.String)">getSimpleDescription</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession)</code></th>
<td class="colLast">
<div class="block">Returns the simple description of the protein with the given accession.</div>
</td>
</tr>
<tr id="i24" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSubsequence(java.lang.String,int,int)">getSubsequence</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession,
              int&nbsp;start,
              int&nbsp;end)</code></th>
<td class="colLast">
<div class="block">Returns the subsequence of the sequence of a given protein.</div>
</td>
</tr>
<tr id="i25" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTaxonomy(java.lang.String)">getTaxonomy</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession)</code></th>
<td class="colLast">
<div class="block">Returns the taxonomy for the given protein.</div>
</td>
</tr>
<tr id="i26" class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#mapTagToProteinTermini(com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent,double,boolean,java.util.LinkedList%5B%5D,int,int,int)">mapTagToProteinTermini</a></span>&#8203;(<a href="MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a>&nbsp;cell,
                      double&nbsp;combinationMass,
                      boolean&nbsp;CTermDirection,
                      <a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedList.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedList</a>&lt;<a href="MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a>&gt;[]&nbsp;matrix,
                      int&nbsp;k,
                      int&nbsp;leftIndex,
                      int&nbsp;rightIndex)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr id="i27" class="rowColor">
<td class="colFirst"><code>double</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#pepMass(java.lang.String)">pepMass</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;peptide)</code></th>
<td class="colLast">
<div class="block">Computing the mass of a peptide.</div>
</td>
</tr>
<tr id="i28" class="altColor">
<td class="colFirst"><code>char</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#prefixCharacter(java.lang.String,int)">prefixCharacter</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;proteinAccession,
               int&nbsp;index)</code></th>
<td class="colLast">
<div class="block">Backward propagation of the BWT to get the previous character.</div>
</td>
</tr>
<tr id="i29" class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#reconstructFasta(java.io.File)">reconstructFasta</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file)</code></th>
<td class="colLast">
<div class="block">Reconstructs the fasta file stored in the index</div>
</td>
</tr>
<tr id="i30" class="altColor">
<td class="colFirst"><code>char</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#suffixCharacter(java.lang.String,int,int)">suffixCharacter</a></span>&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;proteinAccession,
               int&nbsp;index,
               int&nbsp;length)</code></th>
<td class="colLast">
<div class="block">Forward propagation of the BWT to get the n'th consecutive character</div>
</td>
</tr>
<tr id="i31" class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#withinMassTolerance(double,int)">withinMassTolerance</a></span>&#8203;(double&nbsp;mass,
                   int&nbsp;numX)</code></th>
<td class="colLast">
<div class="block">Lookup, if mass can be described a combination of numX different amino
 acids</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.com.compomics.util.experiment.personalization.ExperimentObject">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;com.compomics.util.experiment.personalization.<a href="../../../personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></h3>
<code><a href="../../../personalization/ExperimentObject.html#addUrParam(com.compomics.util.experiment.personalization.UrParameter)">addUrParam</a>, <a href="../../../personalization/ExperimentObject.html#asLong(java.lang.String)">asLong</a>, <a href="../../../personalization/ExperimentObject.html#clearParametersMap()">clearParametersMap</a>, <a href="../../../personalization/ExperimentObject.html#getId()">getId</a>, <a href="../../../personalization/ExperimentObject.html#getUrParam(com.compomics.util.experiment.personalization.UrParameter)">getUrParam</a>, <a href="../../../personalization/ExperimentObject.html#getUrParams()">getUrParams</a>, <a href="../../../personalization/ExperimentObject.html#removeUrParam(long)">removeUrParam</a>, <a href="../../../personalization/ExperimentObject.html#setId(long)">setId</a>, <a href="../../../personalization/ExperimentObject.html#setUrParams(java.util.HashMap)">setUrParams</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a id="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang" class="externalLink">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang" class="externalLink">clone</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang" class="externalLink">equals</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang" class="externalLink">finalize</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang" class="externalLink">getClass</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang" class="externalLink">hashCode</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang" class="externalLink">notify</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang" class="externalLink">notifyAll</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang" class="externalLink">toString</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang" class="externalLink">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang" class="externalLink">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long,int)" title="class or interface in java.lang" class="externalLink">wait</a></code></li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="maxPTMsPerPeptide">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>maxPTMsPerPeptide</h4>
<pre>public&nbsp;int maxPTMsPerPeptide</pre>
<div class="block">Maximal number of PTMs per peptide.</div>
</li>
</ul>
<a id="onlyTrypticPeptides">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>onlyTrypticPeptides</h4>
<pre>public&nbsp;boolean onlyTrypticPeptides</pre>
<div class="block">Flag for only considering tryptic digested peptides.</div>
</li>
</ul>
<a id="DELIMITER">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DELIMITER</h4>
<pre>public static&nbsp;char DELIMITER</pre>
<div class="block">Character defining as delimiter between protein sequences.</div>
</li>
</ul>
<a id="SENTINEL">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SENTINEL</h4>
<pre>public static&nbsp;char SENTINEL</pre>
<div class="block">Sentinel character necessary for computation of the suffix array.</div>
</li>
</ul>
<a id="occurrenceTablesPrimary">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>occurrenceTablesPrimary</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a>&gt; occurrenceTablesPrimary</pre>
<div class="block">Wavelet tree for storing the burrows wheeler transform.</div>
</li>
</ul>
<a id="occurrenceTablesReversed">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>occurrenceTablesReversed</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a>&gt; occurrenceTablesReversed</pre>
<div class="block">Wavelet tree for storing the burrows wheeler transform reversed.</div>
</li>
</ul>
<a id="lessTablesPrimary">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>lessTablesPrimary</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;int[]&gt; lessTablesPrimary</pre>
<div class="block">Less table for doing an update step according to the LF step.</div>
</li>
</ul>
<a id="lessTablesReversed">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>lessTablesReversed</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;int[]&gt; lessTablesReversed</pre>
<div class="block">Less table for doing an update step according to the LF step reversed.</div>
</li>
</ul>
<a id="indexStringLengths">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>indexStringLengths</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&gt; indexStringLengths</pre>
<div class="block">Length of the indexed string (all concatenated protein sequences).</div>
</li>
</ul>
<a id="fmiFastaFile">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>fmiFastaFile</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a> fmiFastaFile</pre>
<div class="block">The fmi fasta file</div>
</li>
</ul>
</li>
</ul>
</section>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a id="&lt;init&gt;()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>FMIndex</h4>
<pre>public&nbsp;FMIndex()</pre>
<div class="block">Empty default constructor.</div>
</li>
</ul>
<a id="&lt;init&gt;(java.io.File,com.compomics.util.experiment.io.biology.protein.FastaParameters,com.compomics.util.waiting.WaitingHandler,boolean,com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters,com.compomics.util.parameters.identification.search.SearchParameters)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>FMIndex</h4>
<pre>public&nbsp;FMIndex&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;fastaFile,
               <a href="../../../io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a>&nbsp;fastaParameters,
               <a href="../../../../waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a>&nbsp;waitingHandler,
               boolean&nbsp;displayProgress,
               <a href="../../../../parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a>&nbsp;peptideVariantsPreferences,
               <a href="../../../../parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a>&nbsp;searchParameters)
        throws <a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io" class="externalLink">IOException</a>,
               <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/OutOfMemoryError.html?is-external=true" title="class or interface in java.lang" class="externalLink">OutOfMemoryError</a>,
               <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang" class="externalLink">RuntimeException</a>,
               <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang" class="externalLink">IllegalArgumentException</a></pre>
<div class="block">Constructor. If modification settings are provided the index will contain
 modification information, ignored if null.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fastaFile</code> - the FASTA file to index</dd>
<dd><code>fastaParameters</code> - the parameters for the FASTA file parsing</dd>
<dd><code>waitingHandler</code> - the waiting handler</dd>
<dd><code>displayProgress</code> - if true, the progress is displayed</dd>
<dd><code>peptideVariantsPreferences</code> - contains all parameters for variants</dd>
<dd><code>searchParameters</code> - the search parameters</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io" class="externalLink">IOException</a></code> - exception thrown if an error occurs while iterating
 the FASTA file.</dd>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/OutOfMemoryError.html?is-external=true" title="class or interface in java.lang" class="externalLink">OutOfMemoryError</a></code></dd>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang" class="externalLink">RuntimeException</a></code></dd>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang" class="externalLink">IllegalArgumentException</a></code></dd>
</dl>
</li>
</ul>
<a id="&lt;init&gt;(java.io.File,com.compomics.util.experiment.io.biology.protein.FastaParameters,com.compomics.util.waiting.WaitingHandler,boolean,com.compomics.util.parameters.identification.IdentificationParameters)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>FMIndex</h4>
<pre>public&nbsp;FMIndex&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;fastaFile,
               <a href="../../../io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a>&nbsp;fastaParameters,
               <a href="../../../../waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a>&nbsp;waitingHandler,
               boolean&nbsp;displayProgress,
               <a href="../../../../parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a>&nbsp;identificationParameters)
        throws <a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io" class="externalLink">IOException</a>,
               <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/OutOfMemoryError.html?is-external=true" title="class or interface in java.lang" class="externalLink">OutOfMemoryError</a>,
               <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang" class="externalLink">RuntimeException</a>,
               <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang" class="externalLink">IllegalArgumentException</a></pre>
<div class="block">Constructor. If modification settings are provided the index will contain
 modification information, ignored if null.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>fastaFile</code> - the FASTA file to index</dd>
<dd><code>fastaParameters</code> - the parameters for the FASTA file parsing</dd>
<dd><code>waitingHandler</code> - the waiting handler</dd>
<dd><code>displayProgress</code> - if true, the progress is displayed</dd>
<dd><code>identificationParameters</code> - contains all identification parameters</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io" class="externalLink">IOException</a></code> - exception thrown if an error occurs while iterating
 the FASTA file</dd>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/OutOfMemoryError.html?is-external=true" title="class or interface in java.lang" class="externalLink">OutOfMemoryError</a></code></dd>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/RuntimeException.html?is-external=true" title="class or interface in java.lang" class="externalLink">RuntimeException</a></code></dd>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang" class="externalLink">IllegalArgumentException</a></code></dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="computeMassValue(double,double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>computeMassValue</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;computeMassValue&#8203;(double&nbsp;currentMass,
                               double&nbsp;refMass)</pre>
<div class="block">Compute the mass value.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>currentMass</code> - the current mass</dd>
<dd><code>refMass</code> - the reference mass</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the mass value</dd>
</dl>
</li>
</ul>
<a id="computeMassTolerance(double,double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>computeMassTolerance</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;computeMassTolerance&#8203;(double&nbsp;currentTollerance,
                                   double&nbsp;refMass)</pre>
<div class="block">Compute the inverse mass value.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>currentTollerance</code> - the current mass</dd>
<dd><code>refMass</code> - the reference mass</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the inverse mass value</dd>
</dl>
</li>
</ul>
<a id="addModificationPattern(com.compomics.util.experiment.biology.modifications.Modification)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addModificationPattern</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;addModificationPattern&#8203;(<a href="../../../biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a>&nbsp;modification)</pre>
<div class="block">Adds a modification pattern for bitwise pattern search.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>modification</code> - modification object</dd>
</dl>
</li>
</ul>
<a id="checkModificationPattern(com.compomics.util.experiment.identification.protein_inference.PeptideProteinMapping)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>checkModificationPattern</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;checkModificationPattern&#8203;(<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&nbsp;peptideProteinMapping)</pre>
<div class="block">Checking if peptide-protein should be discarded due to pattern
 modification conflict.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptideProteinMapping</code> - the peptide protein mapping</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>either yes or no</dd>
</dl>
</li>
</ul>
<a id="computeMappingRanges(double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>computeMappingRanges</h4>
<pre class="methodSignature">public&nbsp;int[]&nbsp;computeMappingRanges&#8203;(double&nbsp;mass)</pre>
<div class="block">Compute mapping ranges.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>mass</code> - the mass</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the mapping ranges</dd>
</dl>
</li>
</ul>
<a id="getAllocatedBytes()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllocatedBytes</h4>
<pre class="methodSignature">public&nbsp;long&nbsp;getAllocatedBytes()</pre>
<div class="block">Computes the number of allocated bytes.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>allocated bytes</dd>
</dl>
</li>
</ul>
<a id="getProteinMapping(java.lang.String,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMapping</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMapping&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;peptide,
                                                          <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingParameters)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../FastaMapper.html#getProteinMapping(java.lang.String,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters)">FastaMapper</a></code></span></div>
<div class="block">Returns the protein mapping in the FASTA file loaded in the sequence
 factory for the given peptide sequence in a map: peptide sequence found
 in the FASTA file | protein accession | list of indexes of the peptide
 sequence on the protein sequence. 0 is the first amino acid.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../FastaMapper.html#getProteinMapping(java.lang.String,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters)">getProteinMapping</a></code>&nbsp;in interface&nbsp;<code><a href="../FastaMapper.html" title="interface in com.compomics.util.experiment.identification.protein_inference">FastaMapper</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptide</code> - the peptide sequence</dd>
<dd><code>sequenceMatchingParameters</code> - the sequence matching preferences</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the peptide to protein mapping: peptide sequence &gt; protein
 accession &gt; index in the protein An empty map if not</dd>
</dl>
</li>
</ul>
<a id="getProteinMappingWithoutVariants(java.lang.String,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMappingWithoutVariants</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMappingWithoutVariants&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;peptide,
                                                                         <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                                                         int&nbsp;indexPart)</pre>
<div class="block">Exact mapping peptides against the proteome.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptide</code> - the peptide</dd>
<dd><code>seqMatchPref</code> - the sequence matching preferences</dd>
<dd><code>indexPart</code> - the index part</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the mapping</dd>
</dl>
</li>
</ul>
<a id="getProteinMappingWithVariantsFixed(java.lang.String,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMappingWithVariantsFixed</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMappingWithVariantsFixed&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;peptide,
                                                                           <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                                                           int&nbsp;indexPart)</pre>
<div class="block">Variant tolerant mapping peptides against the proteome.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptide</code> - the peptide</dd>
<dd><code>seqMatchPref</code> - the sequence match preferences</dd>
<dd><code>indexPart</code> - the index part</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the mapping</dd>
</dl>
</li>
</ul>
<a id="getProteinMappingWithVariantsGeneric(java.lang.String,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMappingWithVariantsGeneric</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMappingWithVariantsGeneric&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;peptide,
                                                                             <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                                                             int&nbsp;indexPart)</pre>
<div class="block">Variant tolerant mapping peptides against the proteome.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptide</code> - the peptide</dd>
<dd><code>seqMatchPref</code> - the sequence match preferences</dd>
<dd><code>indexPart</code> - the index part</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the mapping</dd>
</dl>
</li>
</ul>
<a id="getProteinMappingWithVariantsSpecific(java.lang.String,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMappingWithVariantsSpecific</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMappingWithVariantsSpecific&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;peptide,
                                                                              <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;seqMatchPref,
                                                                              int&nbsp;indexPart)</pre>
<div class="block">Variant tolerant mapping peptides against the proteome</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptide</code> - the peptide</dd>
<dd><code>seqMatchPref</code> - the sequence matching preferences</dd>
<dd><code>indexPart</code> - the index part</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the mapping</dd>
</dl>
</li>
</ul>
<a id="pepMass(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>pepMass</h4>
<pre class="methodSignature">public&nbsp;double&nbsp;pepMass&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;peptide)</pre>
<div class="block">Computing the mass of a peptide.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>peptide</code> - the peptide</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the peptide mass</dd>
</dl>
</li>
</ul>
<a id="withinMassTolerance(double,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>withinMassTolerance</h4>
<pre class="methodSignature">public&nbsp;boolean&nbsp;withinMassTolerance&#8203;(double&nbsp;mass,
                                   int&nbsp;numX)</pre>
<div class="block">Lookup, if mass can be described a combination of numX different amino
 acids</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>mass</code> - to be described</dd>
<dd><code>numX</code> - number of Xs</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>decision</dd>
</dl>
</li>
</ul>
<a id="mapTagToProteinTermini(com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent,double,boolean,java.util.LinkedList[],int,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mapTagToProteinTermini</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;mapTagToProteinTermini&#8203;(<a href="MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a>&nbsp;cell,
                                   double&nbsp;combinationMass,
                                   boolean&nbsp;CTermDirection,
                                   <a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedList.html?is-external=true" title="class or interface in java.util" class="externalLink">LinkedList</a>&lt;<a href="MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a>&gt;[]&nbsp;matrix,
                                   int&nbsp;k,
                                   int&nbsp;leftIndex,
                                   int&nbsp;rightIndex)</pre>
</li>
</ul>
<a id="getProteinMapping(com.compomics.util.experiment.identification.amino_acid_tags.Tag,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMapping</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMapping&#8203;(<a href="../../amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a>&nbsp;tag,
                                                          <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingPreferences)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../FastaMapper.html#getProteinMapping(com.compomics.util.experiment.identification.amino_acid_tags.Tag,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters)">FastaMapper</a></code></span></div>
<div class="block">Returns the protein mappings for the given peptide sequence. Peptide
 sequence | Protein accession | Index in the protein. An empty map if not
 found.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../FastaMapper.html#getProteinMapping(com.compomics.util.experiment.identification.amino_acid_tags.Tag,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters)">getProteinMapping</a></code>&nbsp;in interface&nbsp;<code><a href="../FastaMapper.html" title="interface in com.compomics.util.experiment.identification.protein_inference">FastaMapper</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>tag</code> - the tag to look for in the tree. Must contain a consecutive
 amino acid sequence of longer or equal size than the initialTagSize of
 the tree</dd>
<dd><code>sequenceMatchingPreferences</code> - the sequence matching preferences</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the protein mapping for the given peptide sequence</dd>
</dl>
</li>
</ul>
<a id="getProteinMappingWithoutVariants(com.compomics.util.experiment.identification.amino_acid_tags.Tag,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMappingWithoutVariants</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMappingWithoutVariants&#8203;(<a href="../../amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a>&nbsp;tag,
                                                                         <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingPreferences,
                                                                         int&nbsp;indexPart)</pre>
<div class="block">Mapping tags against proteome without variants.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>tag</code> - the tag</dd>
<dd><code>sequenceMatchingPreferences</code> - the sequence matching preferences</dd>
<dd><code>indexPart</code> - the index part</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the protein mapping</dd>
</dl>
</li>
</ul>
<a id="getProteinMappingWithVariants(com.compomics.util.experiment.identification.amino_acid_tags.Tag,com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinMappingWithVariants</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util" class="externalLink">ArrayList</a>&lt;<a href="../PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a>&gt;&nbsp;getProteinMappingWithVariants&#8203;(<a href="../../amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a>&nbsp;tag,
                                                                      <a href="../../../../parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a>&nbsp;sequenceMatchingPreferences,
                                                                      int&nbsp;indexPart)</pre>
<div class="block">Mapping tags against proteome with variants.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>tag</code> - the tag</dd>
<dd><code>sequenceMatchingPreferences</code> - the sequence matching preferences</dd>
<dd><code>indexPart</code> - the index part</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the protein mapping</dd>
</dl>
</li>
</ul>
<a id="getFileExtension(java.io.File)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFileExtension</h4>
<pre class="methodSignature">public static&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;getFileExtension&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file)</pre>
</li>
</ul>
<a id="reconstructFasta(java.io.File)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reconstructFasta</h4>
<pre class="methodSignature">public&nbsp;void&nbsp;reconstructFasta&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;file)</pre>
<div class="block">Reconstructs the fasta file stored in the index</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>file</code> - the output fasta file object</dd>
</dl>
</li>
</ul>
<a id="prefixCharacter(java.lang.String,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>prefixCharacter</h4>
<pre class="methodSignature">public&nbsp;char&nbsp;prefixCharacter&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;proteinAccession,
                            int&nbsp;index)</pre>
<div class="block">Backward propagation of the BWT to get the previous character.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>proteinAccession</code> - the accession</dd>
<dd><code>index</code> - the index in the suffix array / BWT</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the previous character</dd>
</dl>
</li>
</ul>
<a id="suffixCharacter(java.lang.String,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>suffixCharacter</h4>
<pre class="methodSignature">public&nbsp;char&nbsp;suffixCharacter&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;proteinAccession,
                            int&nbsp;index,
                            int&nbsp;length)</pre>
<div class="block">Forward propagation of the BWT to get the n'th consecutive character</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>proteinAccession</code> - the accession</dd>
<dd><code>index</code> - the index in the suffix array / BWT</dd>
<dd><code>length</code> - number of forward steps</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the n'th next character</dd>
</dl>
</li>
</ul>
<a id="getSequence(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequence</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;getSequence&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;proteinAccession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../io/biology/protein/SequenceProvider.html#getSequence(java.lang.String)">SequenceProvider</a></code></span></div>
<div class="block">Returns the protein sequence for the given accession.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../io/biology/protein/SequenceProvider.html#getSequence(java.lang.String)">getSequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>proteinAccession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the sequence of the protein</dd>
</dl>
</li>
</ul>
<a id="getSubsequence(java.lang.String,int,int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSubsequence</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;getSubsequence&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession,
                             int&nbsp;start,
                             int&nbsp;end)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../io/biology/protein/SequenceProvider.html#getSubsequence(java.lang.String,int,int)">SequenceProvider</a></code></span></div>
<div class="block">Returns the subsequence of the sequence of a given protein. Indexes are
 0-based like for strings and no exception is thrown if indexes are out of
 bounds, the substring is trimmed.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../io/biology/protein/SequenceProvider.html#getSubsequence(java.lang.String,int,int)">getSubsequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession of the protein</dd>
<dd><code>start</code> - the start index</dd>
<dd><code>end</code> - the end index</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the subsequence as string</dd>
</dl>
</li>
</ul>
<a id="getAccessions()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAccessions</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util" class="externalLink">Collection</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;&nbsp;getAccessions()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../io/biology/protein/SequenceProvider.html#getAccessions()">SequenceProvider</a></code></span></div>
<div class="block">Returns all accessions loaded in the provider.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../io/biology/protein/SequenceProvider.html#getAccessions()">getAccessions</a></code>&nbsp;in interface&nbsp;<code><a href="../../../io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>all accessions loaded in the provider</dd>
</dl>
</li>
</ul>
<a id="getDecoyAccessions()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDecoyAccessions</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html?is-external=true" title="class or interface in java.util" class="externalLink">HashSet</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&gt;&nbsp;getDecoyAccessions()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../io/biology/protein/SequenceProvider.html#getDecoyAccessions()">SequenceProvider</a></code></span></div>
<div class="block">Returns the decoy accessions.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../io/biology/protein/SequenceProvider.html#getDecoyAccessions()">getDecoyAccessions</a></code>&nbsp;in interface&nbsp;<code><a href="../../../io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></code></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the decoy accessions</dd>
</dl>
</li>
</ul>
<a id="getHeader(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getHeader</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;getHeader&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;proteinAccession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../io/biology/protein/SequenceProvider.html#getHeader(java.lang.String)">SequenceProvider</a></code></span></div>
<div class="block">Returns the FASTA header of the protein as found in the FASTA file.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../io/biology/protein/SequenceProvider.html#getHeader(java.lang.String)">getHeader</a></code>&nbsp;in interface&nbsp;<code><a href="../../../io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>proteinAccession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the FASTA header of the protein as found in the FASTA file</dd>
</dl>
</li>
</ul>
<a id="getDescription(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDescription</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;getDescription&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../io/biology/protein/ProteinDetailsProvider.html#getDescription(java.lang.String)">ProteinDetailsProvider</a></code></span></div>
<div class="block">Returns the description of the protein with the given accession.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../io/biology/protein/ProteinDetailsProvider.html#getDescription(java.lang.String)">getDescription</a></code>&nbsp;in interface&nbsp;<code><a href="../../../io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the description of the protein with the given accession</dd>
</dl>
</li>
</ul>
<a id="getSimpleDescription(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSimpleDescription</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;getSimpleDescription&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../io/biology/protein/ProteinDetailsProvider.html#getSimpleDescription(java.lang.String)">ProteinDetailsProvider</a></code></span></div>
<div class="block">Returns the simple description of the protein with the given accession.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../io/biology/protein/ProteinDetailsProvider.html#getSimpleDescription(java.lang.String)">getSimpleDescription</a></code>&nbsp;in interface&nbsp;<code><a href="../../../io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the description of the protein with the given accession</dd>
</dl>
</li>
</ul>
<a id="getProteinDatabase(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinDatabase</h4>
<pre class="methodSignature">public&nbsp;<a href="../../../io/biology/protein/ProteinDatabase.html" title="enum in com.compomics.util.experiment.io.biology.protein">ProteinDatabase</a>&nbsp;getProteinDatabase&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../io/biology/protein/ProteinDetailsProvider.html#getProteinDatabase(java.lang.String)">ProteinDetailsProvider</a></code></span></div>
<div class="block">Returns the the protein database for the given protein.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../io/biology/protein/ProteinDetailsProvider.html#getProteinDatabase(java.lang.String)">getProteinDatabase</a></code>&nbsp;in interface&nbsp;<code><a href="../../../io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the name of the protein database</dd>
</dl>
</li>
</ul>
<a id="getGeneName(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneName</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;getGeneName&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../io/biology/protein/ProteinDetailsProvider.html#getGeneName(java.lang.String)">ProteinDetailsProvider</a></code></span></div>
<div class="block">Returns the gene name for the given protein.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../io/biology/protein/ProteinDetailsProvider.html#getGeneName(java.lang.String)">getGeneName</a></code>&nbsp;in interface&nbsp;<code><a href="../../../io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the gene name for the given protein</dd>
</dl>
</li>
</ul>
<a id="getTaxonomy(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTaxonomy</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;getTaxonomy&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../io/biology/protein/ProteinDetailsProvider.html#getTaxonomy(java.lang.String)">ProteinDetailsProvider</a></code></span></div>
<div class="block">Returns the taxonomy for the given protein.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../io/biology/protein/ProteinDetailsProvider.html#getTaxonomy(java.lang.String)">getTaxonomy</a></code>&nbsp;in interface&nbsp;<code><a href="../../../io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the accession of the protein</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the taxonomy for the given protein</dd>
</dl>
</li>
</ul>
<a id="getProteinEvidence(java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getProteinEvidence</h4>
<pre class="methodSignature">public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang" class="externalLink">Integer</a>&nbsp;getProteinEvidence&#8203;(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a>&nbsp;accession)</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../io/biology/protein/ProteinDetailsProvider.html#getProteinEvidence(java.lang.String)">ProteinDetailsProvider</a></code></span></div>
<div class="block">Returns an integer representing the protein evidence level as indexed by
 UniProt.</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../io/biology/protein/ProteinDetailsProvider.html#getProteinEvidence(java.lang.String)">getProteinEvidence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></code></dd>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - the protein accession</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>an integer representing the protein evidence level as indexed by
 UniProt</dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FMIndex.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2020. All rights reserved.</small></p>
</footer>
</body>
</html>
