<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Index (computational omics utilities 5.0.1-SNAPSHOT API)</title>
<link rel="stylesheet" type="text/css" href="stylesheet.css" title="Style">
<script type="text/javascript" src="script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Index (computational omics utilities 5.0.1-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="overview-summary.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="index.html?index-all.html" target="_top">Frames</a></li>
<li><a href="index-all.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="contentContainer"><a href="#I:Z:Z:D">$</a>&nbsp;<a href="#I:A">A</a>&nbsp;<a href="#I:B">B</a>&nbsp;<a href="#I:C">C</a>&nbsp;<a href="#I:D">D</a>&nbsp;<a href="#I:E">E</a>&nbsp;<a href="#I:F">F</a>&nbsp;<a href="#I:G">G</a>&nbsp;<a href="#I:H">H</a>&nbsp;<a href="#I:I">I</a>&nbsp;<a href="#I:J">J</a>&nbsp;<a href="#I:K">K</a>&nbsp;<a href="#I:L">L</a>&nbsp;<a href="#I:M">M</a>&nbsp;<a href="#I:N">N</a>&nbsp;<a href="#I:O">O</a>&nbsp;<a href="#I:P">P</a>&nbsp;<a href="#I:Q">Q</a>&nbsp;<a href="#I:R">R</a>&nbsp;<a href="#I:S">S</a>&nbsp;<a href="#I:T">T</a>&nbsp;<a href="#I:U">U</a>&nbsp;<a href="#I:V">V</a>&nbsp;<a href="#I:W">W</a>&nbsp;<a href="#I:X">X</a>&nbsp;<a href="#I:Y">Y</a>&nbsp;<a href="#I:Z">Z</a>&nbsp;<a name="I:Z:Z:D">
<!--   -->
</a>
<h2 class="title">$</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html#Z:Z:D:D:DgetRootComponent:D:D:D--">$$$getRootComponent$$$()</a></span> - Method in class com.compomics.util.gui.isotopic_calculator.<a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html" title="class in com.compomics.util.gui.isotopic_calculator">IsotopeDistributionGui</a></dt>
<dd>
<div class="block">Returns the root components.</div>
</dd>
</dl>
<a name="I:A">
<!--   -->
</a>
<h2 class="title">A</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#A">A</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html#a">a</a></span> - Variable in class com.compomics.util.math.statistics.linear_regression.<a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html" title="class in com.compomics.util.math.statistics.linear_regression">RegressionStatistics</a></dt>
<dd>
<div class="block">a in y = a.x + b</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#A_ION">A_ION</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Identifier for an a ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#A_ION">A_ION</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Identifier for an a ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#aa">aa</a></span> - Variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">The amino acid that can generate this ion as single letter code.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic"><span class="typeNameLink">AAIdentityFeature</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic</a></dt>
<dd>
<div class="block">Feature based on an amino acid identity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html#AAIdentityFeature--">AAIdentityFeature()</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAIdentityFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureAbsolute.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">AAIdentityFeatureAbsolute</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">An amino acid identity feature at a fixed position on the peptide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureAbsolute.html#AAIdentityFeatureAbsolute-int-char-">AAIdentityFeatureAbsolute(int, char)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureAbsolute.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAIdentityFeatureAbsolute</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureRelative.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">AAIdentityFeatureRelative</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">An amino acid identity feature at a position relative to the end of an ion on the peptide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureRelative.html#AAIdentityFeatureRelative-int-char-">AAIdentityFeatureRelative(int, char)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureRelative.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAIdentityFeatureRelative</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html#aaIndex">aaIndex</a></span> - Variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAIdentityFeature</a></dt>
<dd>
<div class="block">The index on the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html#aaIndex">aaIndex</a></span> - Variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAPropertyFeature</a></dt>
<dd>
<div class="block">The index on the sequence.</div>
</dd>
<dt><a href="com/compomics/util/gui/AaMass.html" title="class in com.compomics.util.gui"><span class="typeNameLink">AaMass</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">Displays the mass of a series of amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/AaMass.html#AaMass-java.awt.Frame-boolean-">AaMass(Frame, boolean)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/AaMass.html" title="class in com.compomics.util.gui">AaMass</a></dt>
<dd>
<div class="block">Creates a form AaMass.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/AaOccurrence.html" title="class in com.compomics.util.experiment.identification.protein_sequences"><span class="typeNameLink">AaOccurrence</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/package-summary.html">com.compomics.util.experiment.identification.protein_sequences</a></dt>
<dd>
<div class="block">This class gathers statistics on the occurrence of amino acids in a fasta
 file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/AaOccurrence.html#AaOccurrence-int:A-">AaOccurrence(int[])</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/AaOccurrence.html" title="class in com.compomics.util.experiment.identification.protein_sequences">AaOccurrence</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic"><span class="typeNameLink">AAPropertyFeature</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic</a></dt>
<dd>
<div class="block">Feature based on an amino acid property.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html#AAPropertyFeature--">AAPropertyFeature()</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAPropertyFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureAbsolute.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">AAPropertyFeatureAbsolute</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">An amino acid property feature at a fixed position on the peptide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureAbsolute.html#AAPropertyFeatureAbsolute-int-com.compomics.util.experiment.biology.aminoacids.AminoAcid.Property-">AAPropertyFeatureAbsolute(int, AminoAcid.Property)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureAbsolute.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyFeatureAbsolute</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureRelative.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">AAPropertyFeatureRelative</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">An amino acid property feature at a position relative to the end of an ion on the peptide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureRelative.html#AAPropertyFeatureRelative-int-com.compomics.util.experiment.biology.aminoacids.AminoAcid.Property-">AAPropertyFeatureRelative(int, AminoAcid.Property)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureRelative.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyFeatureRelative</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">AAPropertyRelationshipFeature</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">Feature based on the relationship between two amino acid features.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html#AAPropertyRelationshipFeature-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.AAPropertyRelationshipFeature.Relationship-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.AAPropertyFeature-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.AAPropertyFeature-">AAPropertyRelationshipFeature(AAPropertyRelationshipFeature.Relationship, AAPropertyFeature, AAPropertyFeature)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.Relationship.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">AAPropertyRelationshipFeature.Relationship</span></a> - Enum in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">The different relationships implemented.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic"><span class="typeNameLink">AASequenceFeature</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic</a></dt>
<dd>
<div class="block">Feature based on the amino acid properties of a sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html#AASequenceFeature--">AASequenceFeature()</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AASequenceFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.Function.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic"><span class="typeNameLink">AASequenceFeature.Function</span></a> - Enum in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic</a></dt>
<dd>
<div class="block">The different functions implemented to compare the amino acid features.</div>
</dd>
<dt><a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein"><span class="typeNameLink">AASequenceImpl</span></a> - Class in <a href="com/compomics/util/protein/package-summary.html">com.compomics.util.protein</a></dt>
<dd>
<div class="block">This class represents a sequence for a protein or peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#AASequenceImpl-java.lang.String-">AASequenceImpl(String)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">Constructor that allows the initialization of the sequence
 with a String variable representing that sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#AASequenceImpl-java.lang.String-java.util.Vector-">AASequenceImpl(String, Vector)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">Constructor that allows the initialization of the sequence
 with a String variable representing that sequence and a
 Vector of Modification instances.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants"><span class="typeNameLink">AaSubstitutionMatrix</span></a> - Class in <a href="com/compomics/util/experiment/biology/variants/package-summary.html">com.compomics.util.experiment.biology.variants</a></dt>
<dd>
<div class="block">Matrix of amino acid substitutions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#AaSubstitutionMatrix-java.lang.String-java.lang.String-">AaSubstitutionMatrix(String, String)</a></span> - Constructor for class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html" title="class in com.compomics.util.gui.variants.aa_substitutions"><span class="typeNameLink">AaSubstitutionMatrixTableModel</span></a> - Class in <a href="com/compomics/util/gui/variants/aa_substitutions/package-summary.html">com.compomics.util.gui.variants.aa_substitutions</a></dt>
<dd>
<div class="block">Table model to display the content of an amino acid substitution matrix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html#AaSubstitutionMatrixTableModel-com.compomics.util.experiment.biology.variants.AaSubstitutionMatrix-boolean-">AaSubstitutionMatrixTableModel(AaSubstitutionMatrix, boolean)</a></span> - Constructor for class com.compomics.util.gui.variants.aa_substitutions.<a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html" title="class in com.compomics.util.gui.variants.aa_substitutions">AaSubstitutionMatrixTableModel</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/StandardUnit.html#ABBREVIATION">ABBREVIATION</a></span> - Variable in enum com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/StandardUnit.html" title="enum in com.compomics.util.experiment.units">StandardUnit</a></dt>
<dd>
<div class="block">The abbreviated name of the unit.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#abort--">abort()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Send Abort command.</div>
</dd>
<dt><a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html" title="class in com.compomics.cli.identification_parameters"><span class="typeNameLink">AbstractIdentificationParametersCli</span></a> - Class in <a href="com/compomics/cli/identification_parameters/package-summary.html">com.compomics.cli.identification_parameters</a></dt>
<dd>
<div class="block">This class serves as a base for the implementation of
 IdentificationParametersCLI in specific tools.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html#AbstractIdentificationParametersCli--">AbstractIdentificationParametersCli()</a></span> - Constructor for class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html" title="class in com.compomics.cli.identification_parameters">AbstractIdentificationParametersCli</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/JarFileFilter.html#accept-java.io.File-java.lang.String-">accept(File, String)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/JarFileFilter.html" title="class in com.compomics.software.autoupdater">JarFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/DatFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/DatFileFilter.html" title="class in com.compomics.util.io.file.filefilters">DatFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.dat files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/DtaFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/DtaFileFilter.html" title="class in com.compomics.util.io.file.filefilters">DtaFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.dta files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FastaFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FastaFileFilter.html" title="class in com.compomics.util.io.file.filefilters">FastaFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.fasta files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/JpegFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/JpegFileFilter.html" title="class in com.compomics.util.io.file.filefilters">JpegFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.jpeg files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/MgfFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/MgfFileFilter.html" title="class in com.compomics.util.io.file.filefilters">MgfFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.mgf files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/Ms2FileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/Ms2FileFilter.html" title="class in com.compomics.util.io.file.filefilters">Ms2FileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.ms2 files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/MzDataFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/MzDataFileFilter.html" title="class in com.compomics.util.io.file.filefilters">MzDataFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, and *.mzData files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/MzMlFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/MzMlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">MzMlFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, and *.mzML files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/MzXmlFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/MzXmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">MzXmlFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, and *.mzXML files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/OmxFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/OmxFileFilter.html" title="class in com.compomics.util.io.file.filefilters">OmxFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.omx files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/OutFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/OutFileFilter.html" title="class in com.compomics.util.io.file.filefilters">OutFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.out files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PdfFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PdfFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PdfFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.pdf files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PeffFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PeffFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PeffFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.peff files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PepXmlFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PepXmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PepXmlFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.pep.xml, *.pepxml and *.xml files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PklFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PklFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PklFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.pkl files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PklSpoFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PklSpoFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PklSpoFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.pkl.spo files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PkxFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PkxFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PkxFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.pkx files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PngFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PngFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PngFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.png files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/ProtXmlFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/ProtXmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">ProtXmlFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.prot.xml, *.protxml and *.xml files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/SequestParamsFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/SequestParamsFileFilter.html" title="class in com.compomics.util.io.file.filefilters">SequestParamsFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, sequest.params files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/SvgFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/SvgFileFilter.html" title="class in com.compomics.util.io.file.filefilters">SvgFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.svg files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/TiffFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/TiffFileFilter.html" title="class in com.compomics.util.io.file.filefilters">TiffFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.tiff files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/XmlFileFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/XmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">XmlFileFilter</a></dt>
<dd>
<div class="block">Accept all directories, *.xml files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FilenameExtensionFilter.html#accept-java.io.File-java.lang.String-">accept(File, String)</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FilenameExtensionFilter.html" title="class in com.compomics.util.io.file">FilenameExtensionFilter</a></dt>
<dd>
<div class="block">Tests if a specified file should be included in a file list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FilenameExtensionFilter.html#accept-java.io.File-">accept(File)</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FilenameExtensionFilter.html" title="class in com.compomics.util.io.file">FilenameExtensionFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#ACE_K_126">ACE_K_126</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for lysine acetylation (PMID: 18338905).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#ACE_K_143">ACE_K_143</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for lysine acetylation (PMID: 18338905).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/threading/MapMutex.html#acquire-K-">acquire(K)</a></span> - Method in class com.compomics.util.threading.<a href="com/compomics/util/threading/MapMutex.html" title="class in com.compomics.util.threading">MapMutex</a></dt>
<dd>
<div class="block">Acquires a permit for the given key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/threading/ObjectMutex.html#acquire-long-">acquire(long)</a></span> - Method in class com.compomics.util.threading.<a href="com/compomics/util/threading/ObjectMutex.html" title="class in com.compomics.util.threading">ObjectMutex</a></dt>
<dd>
<div class="block">Acquire function for the given key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#add-com.compomics.util.experiment.biology.variants.AaSubstitutionMatrix-">add(AaSubstitutionMatrix)</a></span> - Method in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Adds the content of a substitution matrix in this matrix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#addAddendum-java.lang.String-">addAddendum(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method allows the addition of an addendum to the list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html#addAdditionalDataset-double:A-double:A-java.awt.Color-java.awt.Color-">addAdditionalDataset(double[], double[], Color, Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html" title="class in com.compomics.util.gui.spectrum">ChromatogramPanel</a></dt>
<dd>
<div class="block">Adds an additional profileChromatogram dataset to be displayed in the
 same Chromatogram Panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IsotopicDistributionPanel.html#addAdditionalDataset-java.lang.String-java.lang.Integer-java.awt.Color-java.awt.Color-int-">addAdditionalDataset(String, Integer, Color, Color, int)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IsotopicDistributionPanel.html" title="class in com.compomics.util.gui.spectrum">IsotopicDistributionPanel</a></dt>
<dd>
<div class="block">Adds an additional isotopic distribution dataset to be displayed in the same
 panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#addAdditionalDataset-double:A-double:A-java.awt.Color-java.awt.Color-">addAdditionalDataset(double[], double[], Color, Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Adds an additional spectrum dataset to be displayed in the same Spectrum
 Panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#addAdditionalMirroredDataset-double:A-double:A-java.awt.Color-java.awt.Color-">addAdditionalMirroredDataset(double[], double[], Color, Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Adds an additional mirrored spectrum dataset to be displayed in the same
 Spectrum Panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html#addAll-com.compomics.util.experiment.identification.modification.ModificationtableContent-">addAll(ModificationtableContent)</a></span> - Method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html" title="class in com.compomics.util.experiment.identification.modification">ModificationtableContent</a></dt>
<dd>
<div class="block">Add all.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#addAll-com.compomics.util.experiment.identification.peptide_shaker.ModificationScoring-">addAll(ModificationScoring)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Adds all scorings from another score if better.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#addAmbiguousModificationSites-int-java.util.HashMap-">addAmbiguousModificationSites(int, HashMap&lt;Integer, HashSet&lt;String&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Adds a group of modifications to the mapping of ambiguous sites.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#addAminoAcidAfter-java.lang.Character-">addAminoAcidAfter(Character)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Adds an amino acid to the list of allowed amino acids after the cleavage
 site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#addAminoAcidBefore-java.lang.Character-">addAminoAcidBefore(Character)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Adds an amino acid to the list of allowed amino acids before the cleavage
 site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#addAminoAcidSequence-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidSequence-">addAminoAcidSequence(AminoAcidSequence)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Adds a sequence of amino acids to the tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#addAutomaticDeNovoSequencing-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.identification.matches.IonMatch:A-int-int-int-boolean-boolean-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">addAutomaticDeNovoSequencing(Peptide, IonMatch[], int, int, int, boolean, boolean, boolean, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Add reference areas annotating the de novo tags, using default percent
 height of 0.9 for the forward ions and 1.0 for the reverse ions default
 alpha levels of 0.2.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#addAutomaticDeNovoSequencing-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.identification.matches.IonMatch:A-int-int-int-boolean-boolean-double-double-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">addAutomaticDeNovoSequencing(Peptide, IonMatch[], int, int, int, boolean, boolean, double, double, boolean, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Add reference areas annotating the de novo tags, using default alpha
 levels of 0.2.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#addAutomaticDeNovoSequencing-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.identification.matches.IonMatch:A-int-int-int-boolean-boolean-double-double-boolean-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">addAutomaticDeNovoSequencing(Peptide, IonMatch[], int, int, int, boolean, boolean, double, double, boolean, boolean, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Add reference areas annotating the de novo tags, using default alpha
 levels of 0.2.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#addAutomaticDeNovoSequencing-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.experiment.identification.matches.IonMatch:A-int-int-int-boolean-boolean-boolean-">addAutomaticDeNovoSequencing(Tag, IonMatch[], int, int, int, boolean, boolean, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Add reference areas annotating the de novo tags, using default percent
 height of 0.9 for the forward ions and 1.0 for the reverse ions default
 alpha levels of 0.2.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#addAutomaticDeNovoSequencing-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.experiment.identification.matches.IonMatch:A-int-int-int-boolean-boolean-double-double-boolean-">addAutomaticDeNovoSequencing(Tag, IonMatch[], int, int, int, boolean, boolean, double, double, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Add reference areas annotating the de novo tags, using default alpha
 levels of 0.2.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#addAutomaticDeNovoSequencing-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.experiment.identification.matches.IonMatch:A-int-int-int-boolean-boolean-double-double-java.util.ArrayList-boolean-boolean-">addAutomaticDeNovoSequencing(Tag, IonMatch[], int, int, int, boolean, boolean, double, double, ArrayList&lt;float[]&gt;, boolean, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Add reference areas annotating the de novo tags, using default alpha
 levels of 0.2.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#addAutomaticDeNovoSequencing-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.identification.matches.IonMatch:A-int-int-int-boolean-boolean-double-double-java.util.ArrayList-boolean-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">addAutomaticDeNovoSequencing(Peptide, IonMatch[], int, int, int, boolean, boolean, double, double, ArrayList&lt;float[]&gt;, boolean, boolean, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Add reference areas annotating the de novo tags.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#addAutomaticDeNovoSequencing-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.identification.matches.IonMatch:A-int-int-int-boolean-boolean-double-double-float-float-java.util.ArrayList-boolean-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">addAutomaticDeNovoSequencing(Peptide, IonMatch[], int, int, int, boolean, boolean, double, double, float, float, ArrayList&lt;float[]&gt;, boolean, boolean, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Add reference areas annotating the de novo tags.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#addAutomaticDeNovoSequencing-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.experiment.identification.matches.IonMatch:A-int-int-int-boolean-boolean-double-double-float-float-java.util.ArrayList-boolean-boolean-">addAutomaticDeNovoSequencing(Tag, IonMatch[], int, int, int, boolean, boolean, double, double, float, float, ArrayList&lt;float[]&gt;, boolean, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Add reference areas annotating the de novo tags.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html#addBlock-com.compomics.util.pdbfinder.pdb.PdbBlock-">addBlock(PdbBlock)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb">PdbParameter</a></dt>
<dd>
<div class="block">Add a block.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#addColumn-java.util.ArrayList-">addColumn(ArrayList&lt;Double&gt;)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#addConfidentModificationSite-java.lang.String-int-">addConfidentModificationSite(String, int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Adds a confident modification site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#addContactGroup-com.compomics.util.pride.prideobjects.ContactGroup-">addContactGroup(ContactGroup)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Adds a contact group in the PRIDE objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html#addDomain-java.lang.String-java.lang.String-">addDomain(String, String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html" title="class in com.compomics.util.experiment.biology.genes.go">GoDomains</a></dt>
<dd>
<div class="block">Adds a go domain to the mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/MolecularFormula.html#addElement-com.compomics.util.enumeration.MolecularElement-java.lang.Integer-">addElement(MolecularElement, Integer)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/MolecularFormula.html" title="class in com.compomics.util.protein">MolecularFormula</a></dt>
<dd>
<div class="block">Method to add MolecularElements to this formula</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomChainDialog.html#addElementsPanel--">addElementsPanel()</a></span> - Method in class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomChainDialog.html" title="class in com.compomics.util.gui.atoms">AtomChainDialog</a></dt>
<dd>
<div class="block">Add a new atom panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#addEnzyme-com.compomics.util.experiment.biology.enzymes.Enzyme-">addEnzyme(Enzyme)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Adds an enzyme in the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#addEnzyme-com.compomics.util.experiment.biology.enzymes.Enzyme-">addEnzyme(Enzyme)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Adds an enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#addException-java.lang.Long-">addException(Long)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Adds an exception.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#addExportFeature-java.lang.String-com.compomics.util.io.export.ExportFeature-">addExportFeature(String, ExportFeature)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Adds an export feature to the desired section.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFactory.html#addExportScheme-com.compomics.util.io.export.ExportScheme-">addExportScheme(ExportScheme)</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFactory.html" title="interface in com.compomics.util.io.export">ExportFactory</a></dt>
<dd>
<div class="block">Adds an export scheme to the map of user schemes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html#addFeature-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.Ms2pipFeature-">addFeature(Ms2pipFeature)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">FeaturesMap</a></dt>
<dd>
<div class="block">Adds a feature to the map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/ZipUtils.html#addFileToZip-java.io.File-java.util.zip.ZipOutputStream-com.compomics.util.waiting.WaitingHandler-long-">addFileToZip(File, ZipOutputStream, WaitingHandler, long)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression">ZipUtils</a></dt>
<dd>
<div class="block">Adds a new file to the zip stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/ZipUtils.html#addFileToZip-java.io.File-java.util.zip.ZipOutputStream-">addFileToZip(File, ZipOutputStream)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression">ZipUtils</a></dt>
<dd>
<div class="block">Adds a new file to the zip stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/ZipUtils.html#addFileToZip-java.lang.String-java.io.File-java.util.zip.ZipOutputStream-com.compomics.util.waiting.WaitingHandler-long-">addFileToZip(String, File, ZipOutputStream, WaitingHandler, long)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression">ZipUtils</a></dt>
<dd>
<div class="block">Adds a new file to the zip stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html#addFilter-java.lang.Integer-java.lang.String-">addFilter(Integer, String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html" title="class in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsConvertParameters</a></dt>
<dd>
<div class="block">Adds a filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#addFixedModification-com.compomics.util.experiment.biology.modifications.Modification-">addFixedModification(Modification)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Adds a fixed modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/TarUtils.html#addFolderContent-org.apache.commons.compress.archivers.ArchiveOutputStream-java.io.File-com.compomics.util.waiting.WaitingHandler-">addFolderContent(ArchiveOutputStream, File, WaitingHandler)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/TarUtils.html" title="class in com.compomics.util.io.compression">TarUtils</a></dt>
<dd>
<div class="block">Add content to the tar file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/ZipUtils.html#addFolderToZip-java.lang.String-java.util.zip.ZipOutputStream-">addFolderToZip(String, ZipOutputStream)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression">ZipUtils</a></dt>
<dd>
<div class="block">Adds a new entry to the zip file corresponding to a new folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#addFoundCharges-java.util.HashSet-">addFoundCharges(HashSet&lt;Integer&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Adds a new charge to the list of found charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#addFraction-java.lang.String-">addFraction(String)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Adds a fraction, fractions correspond to the PSM files names.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html#addFragmentIonTypeMarkers-java.util.HashMap-org.jfree.chart.JFreeChart-boolean-">addFragmentIonTypeMarkers(HashMap&lt;IonMatch, ArrayList&lt;XYZDataPoint&gt;&gt;, JFreeChart, boolean)</a></span> - Static method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorBubblePlot</a></dt>
<dd>
<div class="block">Adds interval markers for all the fragment ion types.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/genes/GeneDetailsDialog.html#addGoLink-java.lang.String-">addGoLink(String)</a></span> - Method in class com.compomics.util.gui.genes.<a href="com/compomics/util/gui/genes/GeneDetailsDialog.html" title="class in com.compomics.util.gui.genes">GeneDetailsDialog</a></dt>
<dd>
<div class="block">Returns the GO accession number as a web link to the given GO term at
 QuickGO.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#addHidingFilter-com.compomics.util.experiment.identification.filtering.MatchFilter-">addHidingFilter(MatchFilter)</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Adds a hiding filter (previous filter with same name will silently be
 overwritten).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html#addIdentificationParameters-com.compomics.util.parameters.identification.IdentificationParameters-">addIdentificationParameters(IdentificationParameters)</a></span> - Method in class com.compomics.util.experiment.identification.identification_parameters.<a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html" title="class in com.compomics.util.experiment.identification.identification_parameters">IdentificationParametersFactory</a></dt>
<dd>
<div class="block">Adds the given identification parameters to the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#addIdNameMapping-java.lang.String-java.io.File-">addIdNameMapping(String, File)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Adds an id to spectrum name in the mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#addInstrument-com.compomics.util.pride.prideobjects.Instrument-">addInstrument(Instrument)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Adds a instrument in the PRIDE objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html#addIntensity-int-java.lang.Integer-int-double-">addIntensity(int, Integer, int, double)</a></span> - Method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html" title="class in com.compomics.util.experiment.identification.modification">ModificationtableContent</a></dt>
<dd>
<div class="block">Add intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#addIonType-com.compomics.util.experiment.biology.ions.Ion.IonType-int-">addIonType(Ion.IonType, int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Adds a new ion type and subtype to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#addIonType-com.compomics.util.experiment.biology.ions.Ion.IonType-">addIonType(Ion.IonType)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Adds a new ion type to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#addIonType-com.compomics.util.experiment.biology.ions.Ion.IonType-int-">addIonType(Ion.IonType, int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Adds a new ion type and subtype to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#addIonType-com.compomics.util.experiment.biology.ions.Ion.IonType-">addIonType(Ion.IonType)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Adds a new ion type to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#addLine-java.util.ArrayList-">addLine(ArrayList&lt;Double&gt;)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#addListeners--">addListeners()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method adds the event listeners to the panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#addManualValidation-java.lang.Long-">addManualValidation(Long)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Adds a manually validated Match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#addMass-int-double-">addMass(int, double)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>
<div class="block">Add a mass for this glycan.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#addMassGap-double-">addMassGap(double)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Adds a mass gap to the tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#addMirroredSpectrum-double:A-double:A-double-java.lang.String-java.lang.String-boolean-java.awt.Color-java.awt.Color-">addMirroredSpectrum(double[], double[], double, String, String, boolean, Color, Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Add a mirrored spectrum (or chromatogram).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#addMissingPrecursorCharges-java.io.File-com.compomics.util.waiting.WaitingHandler-">addMissingPrecursorCharges(File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Add missing precursor charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#addMissingSpectrumTitles-java.io.File-com.compomics.util.waiting.WaitingHandler-">addMissingSpectrumTitles(File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Adds missing spectrum titles.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html#addMissingSpectrumTitles-java.io.File-com.compomics.util.waiting.WaitingHandler-">addMissingSpectrumTitles(File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspReader</a></dt>
<dd>
<div class="block">Adds missing spectrum titles.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#addModification-com.compomics.util.interfaces.Modification-">addModification(Modification)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method adds a modification to the list of modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#addModificationPattern-com.compomics.util.experiment.biology.modifications.Modification-">addModificationPattern(Modification)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Adds a modification pattern for bitwise pattern search.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#addModificationScoring-java.lang.String-com.compomics.util.experiment.identification.peptide_shaker.ModificationScoring-">addModificationScoring(String, ModificationScoring)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Adds a scoring result for the modification of interest.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#addModificationSite-int-java.util.ArrayList-">addModificationSite(int, ArrayList&lt;Character&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Adds a list of modifications to one of the amino acid pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/MolecularFormula.html#addMolecularFormula-com.compomics.util.protein.MolecularFormula-">addMolecularFormula(MolecularFormula)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/MolecularFormula.html" title="class in com.compomics.util.protein">MolecularFormula</a></dt>
<dd>
<div class="block">Method to add whole formulas to this formula</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#addMouseListenerToHeaderInTable-javax.swing.JTable-">addMouseListenerToHeaderInTable(JTable)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#addNeutralLoss-com.compomics.util.experiment.biology.ions.NeutralLoss-">addNeutralLoss(NeutralLoss)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Adds a neutral loss to the class static map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#addNeutralLoss-com.compomics.util.experiment.biology.ions.NeutralLoss-">addNeutralLoss(NeutralLoss)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Adds a neutral loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#addNeutralLoss-com.compomics.util.experiment.biology.ions.NeutralLoss-">addNeutralLoss(NeutralLoss)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Adds a neutral loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html#addNeutralLoss-com.compomics.util.experiment.biology.ions.NeutralLoss-int-int-">addNeutralLoss(NeutralLoss, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">NeutralLossesMap</a></dt>
<dd>
<div class="block">Adds a new neutral loss to the map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html#addNeutralLoss-java.lang.String-int-int-">addNeutralLoss(String, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">NeutralLossesMap</a></dt>
<dd>
<div class="block">Adds a new neutral loss to the map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#addNeutralLoss-com.compomics.util.experiment.biology.ions.NeutralLoss-">addNeutralLoss(NeutralLoss)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Adds a neutral loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#addObject-java.lang.Long-java.lang.Object-">addObject(Long, Object)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Adds an object to the cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#addObject-long-java.lang.Object-">addObject(long, Object)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Adds an object into the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#addObject-com.compomics.util.experiment.identification.IdentificationFeaturesCache.ObjectType-long-java.lang.Object-">addObject(IdentificationFeaturesCache.ObjectType, long, Object)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Adds an object in the cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#addObjects-java.util.HashMap-">addObjects(HashMap&lt;Long, Object&gt;)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Adds an object to the cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#addObjects-java.util.HashMap-com.compomics.util.waiting.WaitingHandler-boolean-">addObjects(HashMap&lt;Long, Object&gt;, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Adds a list of objects into the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/patient/Patient.html#addPatientInformation-com.compomics.util.experiment.patient.PatientInformation-java.lang.Comparable-">addPatientInformation(PatientInformation, Comparable)</a></span> - Method in class com.compomics.util.experiment.patient.<a href="com/compomics/util/experiment/patient/Patient.html" title="class in com.compomics.util.experiment.patient">Patient</a></dt>
<dd>
<div class="block">Adds patient information,</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#addPeptideAssumption-int-com.compomics.util.experiment.identification.spectrum_assumptions.PeptideAssumption-">addPeptideAssumption(int, PeptideAssumption)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Add a hit.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#addPeptideMatch-long-com.compomics.util.experiment.identification.matches.PeptideMatch-">addPeptideMatch(long, PeptideMatch)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Adds a peptide match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#addPeptideMatchKey-long-">addPeptideMatchKey(long)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Add a subordinated peptide key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#addPeptideMatchKeys-long:A-">addPeptideMatchKeys(long[])</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Add a subordinated peptide key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#addPeptideNormalisationFactor-java.lang.String-double-">addPeptideNormalisationFactor(String, double)</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Adds a peptide normalization factor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#addProteinMatch-long-com.compomics.util.experiment.identification.matches.ProteinMatch-">addProteinMatch(long, ProteinMatch)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Adds a protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#addProteinNormalisationFactor-java.lang.String-double-">addProteinNormalisationFactor(String, double)</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Adds a protein normalization factor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#addProtocol-com.compomics.util.pride.prideobjects.Protocol-">addProtocol(Protocol)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Adds a protocol in the PRIDE objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#addPsmNormalisationFactor-java.lang.String-double-">addPsmNormalisationFactor(String, double)</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Adds a PSM normalization factor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#addReferenceAreaXAxis-com.compomics.util.gui.spectrum.ReferenceArea-">addReferenceAreaXAxis(ReferenceArea)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Add a x-axis reference area.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#addReferenceAreaYAxis-com.compomics.util.gui.spectrum.ReferenceArea-">addReferenceAreaYAxis(ReferenceArea)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Add a y-axis reference area.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#addReferenceGroup-com.compomics.util.pride.prideobjects.ReferenceGroup-">addReferenceGroup(ReferenceGroup)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Adds a reference group in the PRIDE objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#addRefinementFixedModification-com.compomics.util.experiment.biology.modifications.Modification-">addRefinementFixedModification(Modification)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Adds a fixed refinement modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#addRefinementVariableModification-com.compomics.util.experiment.biology.modifications.Modification-">addRefinementVariableModification(Modification)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Adds a variable refinement modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#addReporterIon-com.compomics.util.experiment.biology.ions.impl.ReporterIon-">addReporterIon(ReporterIon)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Adds a reporter ion to the class static map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#addReporterIon-com.compomics.util.experiment.biology.ions.impl.ReporterIon-">addReporterIon(ReporterIon)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Adds a reporter ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#addRestrictionAfter-java.lang.Character-">addRestrictionAfter(Character)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Adds an amino acid to the list of forbidden amino acids after the
 cleavage site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#addRestrictionBefore-java.lang.Character-">addRestrictionBefore(Character)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Adds an amino acid to the list of forbidden amino acids before the
 cleavage site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#addSample-com.compomics.util.pride.prideobjects.Sample-">addSample(Sample)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Adds a sample in the PRIDE objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#addScore-java.lang.Integer-java.lang.Integer-">addScore(Integer, Integer)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Adds a score for a given algorithm to the scoring preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#addScrollListeners-javax.swing.JTable-javax.swing.JScrollPane-javax.swing.JScrollBar-">addScrollListeners(JTable, JScrollPane, JScrollBar)</a></span> - Static method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Add scroll bar and mouse wheel listeners.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#addSelectedCharge-int-">addSelectedCharge(int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Add a charge to take into account when annotating the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#addSeparator--">addSeparator()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Adds a separator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#addSeparator-com.compomics.util.io.export.WorkbookStyle-">addSeparator(WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Adds a separator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#addSeparator-com.compomics.util.io.export.WorkbookStyle-">addSeparator(WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/TextWriter.html#addSeparator-com.compomics.util.io.export.WorkbookStyle-">addSeparator(WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/TextWriter.html" title="class in com.compomics.util.io.export.writers">TextWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/FileDAO.html#addShortcutAtDeskTop-com.compomics.software.autoupdater.MavenJarFile-">addShortcutAtDeskTop(MavenJarFile)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/FileDAO.html" title="class in com.compomics.software.autoupdater">FileDAO</a></dt>
<dd>
<div class="block">Add desktop shortcut.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/FileDAO.html#addShortcutAtDeskTop-com.compomics.software.autoupdater.MavenJarFile-java.lang.String-">addShortcutAtDeskTop(MavenJarFile, String)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/FileDAO.html" title="class in com.compomics.software.autoupdater">FileDAO</a></dt>
<dd>
<div class="block">Adds a shortcut to the desktop.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#addSortListener-javax.swing.JTable-com.compomics.util.gui.waiting.waitinghandlers.ProgressDialogX-">addSortListener(JTable, ProgressDialogX)</a></span> - Static method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Convenience method adding a row sorter listener to the given JTable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#addSpectra-java.io.File-">addSpectra(File)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Add spectra to the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#addSpectra-java.io.File-com.compomics.util.waiting.WaitingHandler-">addSpectra(File, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Add spectra to the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html#addSpectrumMatchKey-long-">addSpectrumMatchKey(long)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html" title="class in com.compomics.util.experiment.identification.matches">PeptideMatch</a></dt>
<dd>
<div class="block">Add a spectrum match key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#addSpectrumPanelListener-com.compomics.util.gui.interfaces.SpectrumPanelListener-">addSpectrumPanelListener(SpectrumPanelListener)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method registers the specified SpectrumPanelListener with this
 instance and notifies it of all future events.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#addStarringFilter-com.compomics.util.experiment.identification.filtering.MatchFilter-">addStarringFilter(MatchFilter)</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Adds a starring filter (previous filter with same name will silently be
 overwritten).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#addSubstitution-java.lang.Character-java.lang.Character-">addSubstitution(Character, Character)</a></span> - Method in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Adds a possible substitution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#addTagAssumption-int-com.compomics.util.experiment.identification.spectrum_assumptions.TagAssumption-">addTagAssumption(int, TagAssumption)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Add a hit.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/ZipUtils.html#addToZip-java.io.File-java.util.zip.ZipOutputStream-">addToZip(File, ZipOutputStream)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression">ZipUtils</a></dt>
<dd>
<div class="block">Adds a new file to the zip stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/ZipUtils.html#addToZip-java.io.File-java.util.zip.ZipOutputStream-com.compomics.util.waiting.WaitingHandler-long-">addToZip(File, ZipOutputStream, WaitingHandler, long)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression">ZipUtils</a></dt>
<dd>
<div class="block">Adds a new file to the zip stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/ZipUtils.html#addToZip-java.io.File-java.lang.String-java.util.zip.ZipOutputStream-">addToZip(File, String, ZipOutputStream)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression">ZipUtils</a></dt>
<dd>
<div class="block">Adds a new file to the zip stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/ZipUtils.html#addToZip-java.io.File-java.lang.String-java.util.zip.ZipOutputStream-com.compomics.util.waiting.WaitingHandler-long-">addToZip(File, String, ZipOutputStream, WaitingHandler, long)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression">ZipUtils</a></dt>
<dd>
<div class="block">Adds a new file to the zip stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/personalization/ExperimentObject.html#addUrParam-com.compomics.util.experiment.personalization.UrParameter-">addUrParam(UrParameter)</a></span> - Method in class com.compomics.util.experiment.personalization.<a href="com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></dt>
<dd>
<div class="block">Adds a user refinement parameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#addUserAdvocate-java.lang.String-">addUserAdvocate(String)</a></span> - Static method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Adds a user advocate and returns it.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#addUserModification-com.compomics.util.experiment.biology.modifications.Modification-">addUserModification(Modification)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Adds a new user modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#addVariableModification-com.compomics.util.experiment.identification.matches.ModificationMatch-">addVariableModification(ModificationMatch)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Adds a modification to one of the amino acid sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#addVariableModification-com.compomics.util.experiment.identification.matches.ModificationMatch-">addVariableModification(ModificationMatch)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Adds a modification match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#addVariableModification-com.compomics.util.experiment.biology.modifications.Modification-">addVariableModification(Modification)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Adds a variable modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#addWaitingActionListener-com.compomics.util.waiting.WaitingActionListener-">addWaitingActionListener(WaitingActionListener)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Adds a waiting action listener.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#addWaitingActionListener-com.compomics.util.waiting.WaitingActionListener-">addWaitingActionListener(WaitingActionListener)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Adds a waiting action listener.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html#addXYZDataSeries-java.util.HashMap-">addXYZDataSeries(HashMap&lt;IonMatch, ArrayList&lt;XYZDataPoint&gt;&gt;)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorBubblePlot</a></dt>
<dd>
<div class="block">Adds the provided data series to an XYZ data set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/TableScrollBarListener.html#adjustmentValueChanged-java.awt.event.AdjustmentEvent-">adjustmentValueChanged(AdjustmentEvent)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/TableScrollBarListener.html" title="class in com.compomics.util.gui">TableScrollBarListener</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/protein/AdvancedProteinDatabaseDialog.html" title="class in com.compomics.util.gui.protein"><span class="typeNameLink">AdvancedProteinDatabaseDialog</span></a> - Class in <a href="com/compomics/util/gui/protein/package-summary.html">com.compomics.util.gui.protein</a></dt>
<dd>
<div class="block">A dialog for advanced database settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/AdvancedProteinDatabaseDialog.html#AdvancedProteinDatabaseDialog-java.awt.Frame-">AdvancedProteinDatabaseDialog(Frame)</a></span> - Constructor for class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/AdvancedProteinDatabaseDialog.html" title="class in com.compomics.util.gui.protein">AdvancedProteinDatabaseDialog</a></dt>
<dd>
<div class="block">Creates a new AdvancedProteinDatabaseDialog.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification"><span class="typeNameLink">Advocate</span></a> - Class in <a href="com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a></dt>
<dd>
<div class="block">The advocate of a hit can be a search engine, a sequencing algorithm, a
 rescoring algorithm, etc.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#advocate">advocate</a></span> - Variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">The advocate supporting this assumption.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/Advocate.AdvocateType.html" title="enum in com.compomics.util.experiment.identification"><span class="typeNameLink">Advocate.AdvocateType</span></a> - Enum in <a href="com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a></dt>
<dd>
<div class="block">The different types of advocates.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Alanine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Alanine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html#Alanine--">Alanine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Alanine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#ALANINE">ALANINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Alanine immonium ion.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/AlgorithmParametersDialog.html" title="interface in com.compomics.util.gui.parameters.identification"><span class="typeNameLink">AlgorithmParametersDialog</span></a> - Interface in <a href="com/compomics/util/gui/parameters/identification/package-summary.html">com.compomics.util.gui.parameters.identification</a></dt>
<dd>
<div class="block">Interface for an algorithm settings dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationSiteMapping.html#align-java.util.Collection-java.util.Collection-">align(Collection&lt;Integer&gt;, Collection&lt;Integer&gt;)</a></span> - Static method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationSiteMapping.html" title="class in com.compomics.util.experiment.identification.modification">ModificationSiteMapping</a></dt>
<dd>
<div class="block">Aligns two series of integer, minimizing the distance between them and
 excluding outliers.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationSiteMapping.html#alignAll-java.util.Collection-java.util.Collection-">alignAll(Collection&lt;Integer&gt;, Collection&lt;Integer&gt;)</a></span> - Static method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationSiteMapping.html" title="class in com.compomics.util.experiment.identification.modification">ModificationSiteMapping</a></dt>
<dd>
<div class="block">Aligns two series of integer, minimizing the distance between them.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationSiteMapping.html#alignAll-java.util.HashMap-">alignAll(HashMap&lt;Integer, ArrayList&lt;Integer&gt;&gt;)</a></span> - Static method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationSiteMapping.html" title="class in com.compomics.util.experiment.identification.modification">ModificationSiteMapping</a></dt>
<dd>
<div class="block">Aligns a series of integer on possible targets maximizing the number of
 matched targets.</div>
</dd>
<dt><a href="com/compomics/util/gui/renderers/AlignedListCellRenderer.html" title="class in com.compomics.util.gui.renderers"><span class="typeNameLink">AlignedListCellRenderer</span></a> - Class in <a href="com/compomics/util/gui/renderers/package-summary.html">com.compomics.util.gui.renderers</a></dt>
<dd>
<div class="block">ListCellRenderer with alignment functionality.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/AlignedListCellRenderer.html#AlignedListCellRenderer-int-">AlignedListCellRenderer(int)</a></span> - Constructor for class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/AlignedListCellRenderer.html" title="class in com.compomics.util.gui.renderers">AlignedListCellRenderer</a></dt>
<dd>
<div class="block">Creates a new AlignedListCellRenderer</div>
</dd>
<dt><a href="com/compomics/util/gui/renderers/AlignedTableCellRenderer.html" title="class in com.compomics.util.gui.renderers"><span class="typeNameLink">AlignedTableCellRenderer</span></a> - Class in <a href="com/compomics/util/gui/renderers/package-summary.html">com.compomics.util.gui.renderers</a></dt>
<dd>
<div class="block">TableCellRenderer with alignment functionality.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/AlignedTableCellRenderer.html#AlignedTableCellRenderer-int-">AlignedTableCellRenderer(int)</a></span> - Constructor for class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/AlignedTableCellRenderer.html" title="class in com.compomics.util.gui.renderers">AlignedTableCellRenderer</a></dt>
<dd>
<div class="block">Creates a new AlignedTableCellRenderer</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/AlignedTableCellRenderer.html#AlignedTableCellRenderer-int-java.awt.Color-">AlignedTableCellRenderer(int, Color)</a></span> - Constructor for class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/AlignedTableCellRenderer.html" title="class in com.compomics.util.gui.renderers">AlignedTableCellRenderer</a></dt>
<dd>
<div class="block">Creates a new AlignedTableCellRenderer</div>
</dd>
<dt><a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html" title="class in com.compomics.util.pdbfinder.das.readers"><span class="typeNameLink">AlignmentBlock</span></a> - Class in <a href="com/compomics/util/pdbfinder/das/readers/package-summary.html">com.compomics.util.pdbfinder.das.readers</a></dt>
<dd>
<div class="block">AlignmentBlock.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html#AlignmentBlock-int-int-int-int-java.lang.String-java.lang.String-">AlignmentBlock(int, int, int, int, String, String)</a></span> - Constructor for class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html" title="class in com.compomics.util.pdbfinder.das.readers">AlignmentBlock</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#allSubstitutions">allSubstitutions</a></span> - Static variable in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Substitution matrix allowing all substitutions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#allVariants">allVariants</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Current storing of all variant operations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#allXcomponents">allXcomponents</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Information about all component where the Xs belong to.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#allXMassDiffs">allXMassDiffs</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">The all X mass differences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#alphaLevel">alphaLevel</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The opacity of the spectra.</div>
</dd>
<dt><a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util"><span class="typeNameLink">AlternateRowColoursJTable</span></a> - Class in <a href="com/compomics/util/package-summary.html">com.compomics.util</a></dt>
<dd>
<div class="block">This class provides a simple extension on a JTable, allowing even and uneven rows to
 have different colours.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#AlternateRowColoursJTable--">AlternateRowColoursJTable()</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">Default empty constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#AlternateRowColoursJTable-int-int-">AlternateRowColoursJTable(int, int)</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">Creates a new AlternateRowColoursJTable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#AlternateRowColoursJTable-javax.swing.table.TableModel-javax.swing.table.TableColumnModel-">AlternateRowColoursJTable(TableModel, TableColumnModel)</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">Creates a new AlternateRowColoursJTable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#AlternateRowColoursJTable-javax.swing.table.TableModel-">AlternateRowColoursJTable(TableModel)</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">Creates a new AlternateRowColoursJTable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#AlternateRowColoursJTable-javax.swing.table.TableModel-javax.swing.table.TableColumnModel-javax.swing.ListSelectionModel-">AlternateRowColoursJTable(TableModel, TableColumnModel, ListSelectionModel)</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">Creates a new AlternateRowColoursJTable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#AlternateRowColoursJTable-java.util.Vector-java.util.Vector-">AlternateRowColoursJTable(Vector, Vector)</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">Creates a new AlternateRowColoursJTable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#AlternateRowColoursJTable-java.lang.Object:A:A-java.lang.Object:A-">AlternateRowColoursJTable(Object[][], Object[])</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">Creates a new AlternateRowColoursJTable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#ambiguousChar">ambiguousChar</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">If original character was B, J, X, Z.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/AmbiguousSequenceIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences"><span class="typeNameLink">AmbiguousSequenceIterator</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/package-summary.html">com.compomics.util.experiment.identification.protein_sequences</a></dt>
<dd>
<div class="block">Iterator going through the possible sequences of a sequence containing
 combinations of amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/AmbiguousSequenceIterator.html#AmbiguousSequenceIterator-char:A-int-">AmbiguousSequenceIterator(char[], int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/AmbiguousSequenceIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences">AmbiguousSequenceIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/AmbiguousSequenceIterator.html#AmbiguousSequenceIterator-java.lang.String-int-">AmbiguousSequenceIterator(String, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/AmbiguousSequenceIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences">AmbiguousSequenceIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/AmbiguousSequenceIterator.html#AmbiguousSequenceIterator-java.lang.String-">AmbiguousSequenceIterator(String)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/AmbiguousSequenceIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences">AmbiguousSequenceIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids"><span class="typeNameLink">AminoAcid</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/package-summary.html">com.compomics.util.experiment.biology.aminoacids</a></dt>
<dd>
<div class="block">Class representing amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#AminoAcid--">AminoAcid()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html#aminoAcid">aminoAcid</a></span> - Variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAIdentityFeature</a></dt>
<dd>
<div class="block">The amino acid targeted represented as single letter code.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.Property.html" title="enum in com.compomics.util.experiment.biology.aminoacids"><span class="typeNameLink">AminoAcid.Property</span></a> - Enum in <a href="com/compomics/util/experiment/biology/aminoacids/package-summary.html">com.compomics.util.experiment.biology.aminoacids</a></dt>
<dd>
<div class="block">Properties of the amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#aminoAcidCombinations">aminoAcidCombinations</a></span> - Variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">The amino acid combinations.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence"><span class="typeNameLink">AminoAcidPattern</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/sequence/package-summary.html">com.compomics.util.experiment.biology.aminoacids.sequence</a></dt>
<dd>
<div class="block">An amino acid pattern is a sequence of amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#AminoAcidPattern--">AminoAcidPattern()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Creates a blank pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#AminoAcidPattern-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-">AminoAcidPattern(AminoAcidPattern)</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Creates a pattern from another pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#AminoAcidPattern-java.util.ArrayList-">AminoAcidPattern(ArrayList&lt;String&gt;)</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Convenience constructor giving a list of targeted residues as input.</div>
</dd>
<dt><a href="com/compomics/util/gui/AminoAcidPatternDialog.html" title="class in com.compomics.util.gui"><span class="typeNameLink">AminoAcidPatternDialog</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">This dialog allows the design and test of amino acid patterns.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/AminoAcidPatternDialog.html#AminoAcidPatternDialog-java.awt.Frame-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-boolean-">AminoAcidPatternDialog(Frame, AminoAcidPattern, boolean)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/AminoAcidPatternDialog.html" title="class in com.compomics.util.gui">AminoAcidPatternDialog</a></dt>
<dd>
<div class="block">Creates a new AminoAcidPatternDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html#aminoAcidProperty">aminoAcidProperty</a></span> - Variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAPropertyFeature</a></dt>
<dd>
<div class="block">The property of the amino acids to consider.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html#aminoAcidProperty">aminoAcidProperty</a></span> - Variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AASequenceFeature</a></dt>
<dd>
<div class="block">The property of the amino acids to consider.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#aminoAcids">aminoAcids</a></span> - Variable in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Amino acids that are likely to induce this loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#aminoAcidScores">aminoAcidScores</a></span> - Variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">The individual amino acid scores.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence"><span class="typeNameLink">AminoAcidSequence</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/sequence/package-summary.html">com.compomics.util.experiment.biology.aminoacids.sequence</a></dt>
<dd>
<div class="block">This class represents a series of amino acids with associated modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#AminoAcidSequence--">AminoAcidSequence()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Creates a blank sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#AminoAcidSequence-java.lang.String-">AminoAcidSequence(String)</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Constructor taking a sequence of amino acids as input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#AminoAcidSequence-java.lang.String-com.compomics.util.experiment.identification.matches.ModificationMatch:A-">AminoAcidSequence(String, ModificationMatch[])</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Constructor taking a sequence of amino acids as input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#AminoAcidSequence-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidSequence-">AminoAcidSequence(AminoAcidSequence)</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Creates a sequence from another sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#aminoAcidStrings">aminoAcidStrings</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">The amino acid one letter codes as string array.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/variants/AminoAcidSubstitution.html" title="class in com.compomics.util.experiment.biology.variants"><span class="typeNameLink">AminoAcidSubstitution</span></a> - Class in <a href="com/compomics/util/experiment/biology/variants/package-summary.html">com.compomics.util.experiment.biology.variants</a></dt>
<dd>
<div class="block">Class representing an amino acid substitution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AminoAcidSubstitution.html#AminoAcidSubstitution-char-char-">AminoAcidSubstitution(char, char)</a></span> - Constructor for class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AminoAcidSubstitution.html" title="class in com.compomics.util.experiment.biology.variants">AminoAcidSubstitution</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/Constants.html#AMU">AMU</a></span> - Static variable in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/Constants.html" title="class in com.compomics.util.experiment">Constants</a></dt>
<dd>
<div class="block">The atomic mass unit in kg.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/Constants.html#AMU_NO_EXP">AMU_NO_EXP</a></span> - Static variable in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/Constants.html" title="class in com.compomics.util.experiment">Constants</a></dt>
<dd>
<div class="block">The atomic mass unit without exponent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#andromeda">andromeda</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Andromeda search engine.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">AndromedaIdfileReader</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">This IdfileReader reads identifications from an Andromeda result file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html#AndromedaIdfileReader--">AndromedaIdfileReader()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">AndromedaIdfileReader</a></dt>
<dd>
<div class="block">Default constructor for the purpose of instantiation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html#AndromedaIdfileReader-java.io.File-">AndromedaIdfileReader(File)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">AndromedaIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an Andromeda result file reader.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">AndromedaParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">The Andromeda specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#AndromedaParameters--">AndromedaParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.AndromedaDecoyMode.html" title="enum in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">AndromedaParameters.AndromedaDecoyMode</span></a> - Enum in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">The available decoy modes.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm"><span class="typeNameLink">AndromedaParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a></dt>
<dd>
<div class="block">Dialog for the Andromeda specific settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html#AndromedaParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.AndromedaParameters-boolean-">AndromedaParametersDialog(Frame, AndromedaParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">AndromedaParametersDialog</a></dt>
<dd>
<div class="block">Creates a new TideSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html#AndromedaParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.AndromedaParameters-boolean-">AndromedaParametersDialog(Dialog, Frame, AndromedaParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">AndromedaParametersDialog</a></dt>
<dd>
<div class="block">Creates a new TideSettingsDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#annotate-com.compomics.util.gui.interfaces.SpectrumAnnotation-java.awt.Graphics-java.util.HashMap-boolean-">annotate(SpectrumAnnotation, Graphics, HashMap&lt;String, Integer&gt;, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method attempts to find the specified SpectrumAnnotation in the
 current peak list and if so, annotates it correspondingly on the screen.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#ANNOTATED_STATUS">ANNOTATED_STATUS</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This status indicates that annotation (if present) will be displayed,
 while limiting the user to zooming in/out.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#annotateHighestPeak">annotateHighestPeak</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">If more than one peak is within the accuracy range of an annotation
 setting this to true will always select the most intense of possible
 peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#annotatePeaks-java.awt.Graphics-java.util.Vector-boolean-">annotatePeaks(Graphics, Vector, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Annotate the annotated peaks.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation"><span class="typeNameLink">AnnotationParameters</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation</a></dt>
<dd>
<div class="block">This class contains the spectrum annotation parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#AnnotationParameters--">AnnotationParameters()</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#AnnotationParameters-com.compomics.util.parameters.identification.search.SearchParameters-">AnnotationParameters(SearchParameters)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Constructor setting preferences from search parameters.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.IntensityThresholdType.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation"><span class="typeNameLink">AnnotationParameters.IntensityThresholdType</span></a> - Enum in <a href="com/compomics/util/experiment/identification/spectrum_annotation/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation</a></dt>
<dd>
<div class="block">Enum of the types of intensity thresholds available.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/advanced/AnnotationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced"><span class="typeNameLink">AnnotationParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/advanced/package-summary.html">com.compomics.util.gui.parameters.identification.advanced</a></dt>
<dd>
<div class="block">A simple dialog for setting the spectrum annotation preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/AnnotationParametersDialog.html#AnnotationParametersDialog-java.awt.Frame-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-double-java.util.HashSet-java.util.ArrayList-boolean-">AnnotationParametersDialog(Frame, AnnotationParameters, double, HashSet&lt;String&gt;, ArrayList&lt;Integer&gt;, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/AnnotationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">AnnotationParametersDialog</a></dt>
<dd>
<div class="block">Creates a new AnnotationPreferencesDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/AnnotationParametersDialog.html#AnnotationParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-double-java.util.HashSet-java.util.ArrayList-boolean-">AnnotationParametersDialog(Dialog, Frame, AnnotationParameters, double, HashSet&lt;String&gt;, ArrayList&lt;Integer&gt;, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/AnnotationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">AnnotationParametersDialog</a></dt>
<dd>
<div class="block">Creates a new AnnotationPreferencesDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/mass_spectrometry/export/AplExporter.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.export"><span class="typeNameLink">AplExporter</span></a> - Class in <a href="com/compomics/util/experiment/io/mass_spectrometry/export/package-summary.html">com.compomics.util.experiment.io.mass_spectrometry.export</a></dt>
<dd>
<div class="block">This converter writes spectrum files in APL format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/export/AplExporter.html#AplExporter--">AplExporter()</a></span> - Constructor for class com.compomics.util.experiment.io.mass_spectrometry.export.<a href="com/compomics/util/experiment/io/mass_spectrometry/export/AplExporter.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.export">AplExporter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#append-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-">append(AminoAcidPattern)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Appends another pattern at the end of this pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#append-com.compomics.util.experiment.biology.atoms.AtomImpl-">append(AtomImpl)</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>
<div class="block">Appends an atom to the chain of atoms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#append-com.compomics.util.experiment.biology.atoms.AtomImpl-int-">append(AtomImpl, int)</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>
<div class="block">Appends an atom to the chain of atoms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#append-java.io.File-java.io.File-">append(File, File)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Appends the content of a file to another.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#appendAscii-java.lang.String-">appendAscii(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">APPEND (with create) to a file to the FTP server in Ascii mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#appendBinary-java.lang.String-">appendBinary(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">APPEND (with create) to a file to the FTP server in Binary mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#appendColumns-com.compomics.util.math.matrix.DoubleMatrix-">appendColumns(DoubleMatrix)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#appendCTerm-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidSequence-">appendCTerm(AminoAcidSequence)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Appends another sequence at the end of this sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#appendCTerm-java.lang.String-">appendCTerm(String)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Appends a series of unmodified amino acids to the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/converters/DecoyConverter.html#appendDecoySequences-java.io.File-java.io.File-com.compomics.util.waiting.WaitingHandler-">appendDecoySequences(File, File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.biology.protein.converters.<a href="com/compomics/util/experiment/io/biology/protein/converters/DecoyConverter.html" title="class in com.compomics.util.experiment.io.biology.protein.converters">DecoyConverter</a></dt>
<dd>
<div class="block">Appends decoy sequences to the provided FASTA file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#appendLines-com.compomics.util.math.matrix.DoubleMatrix-">appendLines(DoubleMatrix)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#appendNTerm-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidSequence-">appendNTerm(AminoAcidSequence)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Appends another sequence at the beginning of this sequence keeping the
 original order.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#appendNTerm-java.lang.String-">appendNTerm(String)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Appends a series of unmodified amino acids to the beginning sequence
 keeping the original order.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#appendReport-java.lang.String-boolean-boolean-">appendReport(String, boolean, boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#appendReport-java.lang.String-boolean-boolean-">appendReport(String, boolean, boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#appendReport-java.lang.String-boolean-boolean-">appendReport(String, boolean, boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#appendReport-java.lang.String-boolean-boolean-">appendReport(String, boolean, boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#appendReport-java.lang.String-boolean-boolean-">appendReport(String, boolean, boolean)</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Append text to the report.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#appendReportEndLine--">appendReportEndLine()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#appendReportEndLine--">appendReportEndLine()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Append a new line to the report.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#appendReportEndLine--">appendReportEndLine()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#appendReportEndLine--">appendReportEndLine()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#appendReportEndLine--">appendReportEndLine()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Append a new line to the report.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#appendReportNewLineNoDate--">appendReportNewLineNoDate()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#appendReportNewLineNoDate--">appendReportNewLineNoDate()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Append tabs to the report.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#appendReportNewLineNoDate--">appendReportNewLineNoDate()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#appendReportNewLineNoDate--">appendReportNewLineNoDate()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#appendReportNewLineNoDate--">appendReportNewLineNoDate()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Append two tabs to the report.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#appendSuffix-java.lang.String-java.lang.String-">appendSuffix(String, String)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Appends a suffix to a file name before the file extension.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#appendTaggedResidue-java.lang.StringBuilder-char-java.lang.String-java.lang.String-java.lang.String-java.lang.String-com.compomics.util.parameters.identification.search.ModificationParameters-boolean-boolean-">appendTaggedResidue(StringBuilder, char, String, String, String, String, ModificationParameters, boolean, boolean)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">Returns the single residue as a tagged string (HTML color or PTM tag).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#appendTaggedResidue-java.lang.StringBuilder-char-java.lang.String-com.compomics.util.parameters.identification.search.ModificationParameters-int-boolean-boolean-">appendTaggedResidue(StringBuilder, char, String, ModificationParameters, int, boolean, boolean)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">Appends the single residue as a tagged string (HTML color or PTM tag).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#AQUA">AQUA</a></span> - Static variable in class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">A rather pronounced AQUA colour (r=183, g=222, b=232).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#areNeutralLossesSequenceAuto--">areNeutralLossesSequenceAuto()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns whether neutral losses are considered only for amino acids of
 interest or not.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Arginine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Arginine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html#Arginine--">Arginine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Arginine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#ARGININE">ARGININE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Arginine immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#arraysMap">arraysMap</a></span> - Static variable in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">map of arrays for c-term.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#AscendingIntensityComparator">AscendingIntensityComparator</a></span> - Static variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>
<div class="block">This comparator compares two Peak instances on ascending intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#AscendingMzComparator">AscendingMzComparator</a></span> - Static variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>
<div class="block">This comparator compares two Peak instances on ascending m/z value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#ascii--">ascii()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Set transfer type to 'A' (ascii transfer).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#asGenericHeader--">asGenericHeader()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the header in generic format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/personalization/ExperimentObject.html#asLong-java.lang.String-">asLong(String)</a></span> - Static method in class com.compomics.util.experiment.personalization.<a href="com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></dt>
<dd>
<div class="block">Creating a unique 64 bit hash key from the original key of arbitrary
 length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#asMgf--">asMgf()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the peak list as an mgf bloc.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#asMgf-java.util.HashMap-">asMgf(HashMap&lt;String, String&gt;)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the peak list as an mgf bloc.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Asparagine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Asparagine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html#Asparagine--">Asparagine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Asparagine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#ASPARAGINE">ASPARAGINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Asparagine immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#ASPARTIC_ACID">ASPARTIC_ACID</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Aspartic acid immonium ion.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">AsparticAcid</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Aspartic Acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html#AsparticAcid--">AsparticAcid()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">AsparticAcid</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#asSequence-int-">asSequence(int)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the component of the amino acid pattern at the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#asSequence--">asSequence()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html#asSequence--">asSequence()</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">MassGap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#asSequence--">asSequence()</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the tag as intelligible sequence of tag components.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/TagComponent.html#asSequence--">asSequence()</a></span> - Method in interface com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/TagComponent.html" title="interface in com.compomics.util.experiment.identification.amino_acid_tags">TagComponent</a></dt>
<dd>
<div class="block">Returns the tag component as String like a peptide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#assignSample-java.lang.String-java.lang.String-">assignSample(String, String)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">Assign a sample to an ion referenced by its static index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#asStringBuilder--">asStringBuilder()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the sequence represented by this amino acid pattern in a new
 string builder.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering"><span class="typeNameLink">AssumptionFilter</span></a> - Class in <a href="com/compomics/util/experiment/identification/filtering/package-summary.html">com.compomics.util.experiment.identification.filtering</a></dt>
<dd>
<div class="block">Peptide Assumption filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html#AssumptionFilter--">AssumptionFilter()</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">AssumptionFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html#AssumptionFilter-java.lang.String-">AssumptionFilter(String)</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">AssumptionFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html#AssumptionFilter-java.lang.String-java.lang.String-">AssumptionFilter(String, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">AssumptionFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html#AssumptionFilter-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String-">AssumptionFilter(String, String, String, String, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">AssumptionFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items"><span class="typeNameLink">AssumptionFilterItem</span></a> - Enum in <a href="com/compomics/util/experiment/identification/filtering/items/package-summary.html">com.compomics.util.experiment.identification.filtering.items</a></dt>
<dd>
<div class="block">Enum of the different items an assumption filter can filter on.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms"><span class="typeNameLink">Atom</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/package-summary.html">com.compomics.util.experiment.biology.atoms</a></dt>
<dd>
<div class="block">This interface contains information about atoms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#Atom--">Atom()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms"><span class="typeNameLink">AtomChain</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/package-summary.html">com.compomics.util.experiment.biology.atoms</a></dt>
<dd>
<div class="block">A chain of atoms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#AtomChain--">AtomChain()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>
<div class="block">Creates an empty atom chain.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#atomChain">atomChain</a></span> - Variable in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">The atomic composition of the ion.</div>
</dd>
<dt><a href="com/compomics/util/gui/atoms/AtomChainDialog.html" title="class in com.compomics.util.gui.atoms"><span class="typeNameLink">AtomChainDialog</span></a> - Class in <a href="com/compomics/util/gui/atoms/package-summary.html">com.compomics.util.gui.atoms</a></dt>
<dd>
<div class="block">AtomChainDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomChainDialog.html#AtomChainDialog-javax.swing.JDialog-com.compomics.util.experiment.biology.atoms.AtomChain-com.compomics.util.experiment.biology.atoms.AtomChain-boolean-">AtomChainDialog(JDialog, AtomChain, AtomChain, boolean)</a></span> - Constructor for class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomChainDialog.html" title="class in com.compomics.util.gui.atoms">AtomChainDialog</a></dt>
<dd>
<div class="block">Creates a new dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomChainDialog.html#AtomChainDialog-java.awt.Frame-com.compomics.util.experiment.biology.atoms.AtomChain-com.compomics.util.experiment.biology.atoms.AtomChain-boolean-">AtomChainDialog(Frame, AtomChain, AtomChain, boolean)</a></span> - Constructor for class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomChainDialog.html" title="class in com.compomics.util.gui.atoms">AtomChainDialog</a></dt>
<dd>
<div class="block">Creates a new dialog.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html" title="class in com.compomics.util.experiment.biology.atoms"><span class="typeNameLink">AtomImpl</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/package-summary.html">com.compomics.util.experiment.biology.atoms</a></dt>
<dd>
<div class="block">Class for a specific atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html#AtomImpl-java.lang.String-java.lang.Integer-">AtomImpl(String, Integer)</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html" title="class in com.compomics.util.experiment.biology.atoms">AtomImpl</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html#AtomImpl-com.compomics.util.experiment.biology.atoms.Atom-java.lang.Integer-">AtomImpl(Atom, Integer)</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html" title="class in com.compomics.util.experiment.biology.atoms">AtomImpl</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/atoms/AtomPanel.html" title="class in com.compomics.util.gui.atoms"><span class="typeNameLink">AtomPanel</span></a> - Class in <a href="com/compomics/util/gui/atoms/package-summary.html">com.compomics.util.gui.atoms</a></dt>
<dd>
<div class="block">Panel for selecting atoms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomPanel.html#AtomPanel-com.compomics.util.gui.atoms.AtomChainDialog-com.compomics.util.experiment.biology.atoms.Atom-int-int-int-boolean-">AtomPanel(AtomChainDialog, Atom, int, int, int, boolean)</a></span> - Constructor for class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomPanel.html" title="class in com.compomics.util.gui.atoms">AtomPanel</a></dt>
<dd>
<div class="block">Creates a new AtomPanel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#AVERAGE">AVERAGE</a></span> - Static variable in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">Constant to indicate the position of the average mass in the mass array.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/Constants.html#Avogadro">Avogadro</a></span> - Static variable in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/Constants.html" title="class in com.compomics.util.experiment">Constants</a></dt>
<dd>
<div class="block">The Avogadro constant.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/Constants.html#AVOGADRO_NO_EXP">AVOGADRO_NO_EXP</a></span> - Static variable in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/Constants.html" title="class in com.compomics.util.experiment">Constants</a></dt>
<dd>
<div class="block">The Avogadro constant without exponent.</div>
</dd>
</dl>
<a name="I:B">
<!--   -->
</a>
<h2 class="title">B</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#B">B</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">B</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Asn or Asp: Asx (Mascot).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html#B--">B()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">B</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html#b">b</a></span> - Variable in class com.compomics.util.math.statistics.linear_regression.<a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html" title="class in com.compomics.util.math.statistics.linear_regression">RegressionStatistics</a></dt>
<dd>
<div class="block">b in y = a.x + b</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#B_ION">B_ION</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Identifier for a b ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#B_ION">B_ION</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Identifier for a b ion.</div>
</dd>
<dt><a href="com/compomics/util/examples/BareBonesBrowserLaunch.html" title="class in com.compomics.util.examples"><span class="typeNameLink">BareBonesBrowserLaunch</span></a> - Class in <a href="com/compomics/util/examples/package-summary.html">com.compomics.util.examples</a></dt>
<dd>
<div class="block">Class that makes it possible to open a URL in the default browser.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/examples/BareBonesBrowserLaunch.html#BareBonesBrowserLaunch--">BareBonesBrowserLaunch()</a></span> - Constructor for class com.compomics.util.examples.<a href="com/compomics/util/examples/BareBonesBrowserLaunch.html" title="class in com.compomics.util.examples">BareBonesBrowserLaunch</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math"><span class="typeNameLink">BasicMathFunctions</span></a> - Class in <a href="com/compomics/util/math/package-summary.html">com.compomics.util.math</a></dt>
<dd>
<div class="block">Class used to perform basic mathematical functions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#BasicMathFunctions--">BasicMathFunctions()</a></span> - Constructor for class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#BIG_ENDIAN_LABEL">BIG_ENDIAN_LABEL</a></span> - Static variable in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Defines the valid String indicating big endian byte order.</div>
</dd>
<dt><a href="com/compomics/util/math/BigFunctions.html" title="class in com.compomics.util.math"><span class="typeNameLink">BigFunctions</span></a> - Class in <a href="com/compomics/util/math/package-summary.html">com.compomics.util.math</a></dt>
<dd>
<div class="block">Functions operating with BigDecimal objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigFunctions.html#BigFunctions--">BigFunctions()</a></span> - Constructor for class com.compomics.util.math.<a href="com/compomics/util/math/BigFunctions.html" title="class in com.compomics.util.math">BigFunctions</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/math/BigMathUtils.html" title="class in com.compomics.util.math"><span class="typeNameLink">BigMathUtils</span></a> - Class in <a href="com/compomics/util/math/package-summary.html">com.compomics.util.math</a></dt>
<dd>
<div class="block">Utils for the manipulation of big numbers.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigMathUtils.html#BigMathUtils--">BigMathUtils()</a></span> - Constructor for class com.compomics.util.math.<a href="com/compomics/util/math/BigMathUtils.html" title="class in com.compomics.util.math">BigMathUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#binary--">binary()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Set transfer type to 'I' (binary transfer).</div>
</dd>
<dt><a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util"><span class="typeNameLink">BinaryArrayImpl</span></a> - Class in <a href="com/compomics/util/package-summary.html">com.compomics.util</a></dt>
<dd>
<div class="block">Persistence independent object model implementation class:</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#BinaryArrayImpl--">BinaryArrayImpl()</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Default constructor with protected access.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#BinaryArrayImpl-java.lang.String-long-java.lang.String-java.lang.String-">BinaryArrayImpl(String, long, String, String)</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Creates an instance of this BinaryArray object, setting all fields as
 per description below.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#BinaryArrayImpl-double:A-java.lang.String-">BinaryArrayImpl(double[], String)</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Constructor that allows the creation of a BinaryArray object using an
 array of double values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#BinaryArrayImpl-float:A-java.lang.String-">BinaryArrayImpl(float[], String)</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Constructor that allows the creation of a BinaryArray object using an
 array of float values.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/linear_regression/filters/BinningFilter.html" title="class in com.compomics.util.math.statistics.linear_regression.filters"><span class="typeNameLink">BinningFilter</span></a> - Class in <a href="com/compomics/util/math/statistics/linear_regression/filters/package-summary.html">com.compomics.util.math.statistics.linear_regression.filters</a></dt>
<dd>
<div class="block">Filters by binning.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/filters/BinningFilter.html#BinningFilter--">BinningFilter()</a></span> - Constructor for class com.compomics.util.math.statistics.linear_regression.filters.<a href="com/compomics/util/math/statistics/linear_regression/filters/BinningFilter.html" title="class in com.compomics.util.math.statistics.linear_regression.filters">BinningFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html" title="class in com.compomics.util.math.statistics.distributions"><span class="typeNameLink">BinomialDistribution</span></a> - Class in <a href="com/compomics/util/math/statistics/distributions/package-summary.html">com.compomics.util.math.statistics.distributions</a></dt>
<dd>
<div class="block">Implementation of a binomial distribution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html#BinomialDistribution-int-double-">BinomialDistribution(int, double)</a></span> - Constructor for class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html" title="class in com.compomics.util.math.statistics.distributions">BinomialDistribution</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#BIOMART_ENSEMBL_FILENAME">BIOMART_ENSEMBL_FILENAME</a></span> - Static variable in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">The name of the Ensembl BioMart datasets file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#BIOMART_ENSEMBL_GENOME_FILENAME">BIOMART_ENSEMBL_GENOME_FILENAME</a></span> - Static variable in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">The name of the Ensembl Genome BioMart datasets file.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/taxonomy/mappings/BiomartMapping.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings"><span class="typeNameLink">BiomartMapping</span></a> - Class in <a href="com/compomics/util/experiment/biology/taxonomy/mappings/package-summary.html">com.compomics.util.experiment.biology.taxonomy.mappings</a></dt>
<dd>
<div class="block">Mapping of the species to BioMart dataset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/BiomartMapping.html#BiomartMapping--">BiomartMapping()</a></span> - Constructor for class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/BiomartMapping.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">BiomartMapping</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/db/object/objects/BlobObject.html" title="class in com.compomics.util.db.object.objects"><span class="typeNameLink">BlobObject</span></a> - Class in <a href="com/compomics/util/db/object/objects/package-summary.html">com.compomics.util.db.object.objects</a></dt>
<dd>
<div class="block">Class used to store entire objects in the database as a blob.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/objects/BlobObject.html#BlobObject-java.lang.Object-">BlobObject(Object)</a></span> - Constructor for class com.compomics.util.db.object.objects.<a href="com/compomics/util/db/object/objects/BlobObject.html" title="class in com.compomics.util.db.object.objects">BlobObject</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/objects/BlobObject.html#BlobObject--">BlobObject()</a></span> - Constructor for class com.compomics.util.db.object.objects.<a href="com/compomics/util/db/object/objects/BlobObject.html" title="class in com.compomics.util.db.object.objects">BlobObject</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/objects/BlobObject.html#BlobObject-byte:A-">BlobObject(byte[])</a></span> - Constructor for class com.compomics.util.db.object.objects.<a href="com/compomics/util/db/object/objects/BlobObject.html" title="class in com.compomics.util.db.object.objects">BlobObject</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/general/BoxedObject.html" title="class in com.compomics.util.general"><span class="typeNameLink">BoxedObject</span></a>&lt;<a href="com/compomics/util/general/BoxedObject.html" title="type parameter in BoxedObject">K</a>&gt; - Class in <a href="com/compomics/util/general/package-summary.html">com.compomics.util.general</a></dt>
<dd>
<div class="block">Convenience class allowing the boxing of an object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/BoxedObject.html#BoxedObject--">BoxedObject()</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/BoxedObject.html" title="class in com.compomics.util.general">BoxedObject</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/BoxedObject.html#BoxedObject-K-">BoxedObject(K)</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/BoxedObject.html" title="class in com.compomics.util.general">BoxedObject</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/error_handlers/BugReport.html" title="class in com.compomics.util.gui.error_handlers"><span class="typeNameLink">BugReport</span></a> - Class in <a href="com/compomics/util/gui/error_handlers/package-summary.html">com.compomics.util.gui.error_handlers</a></dt>
<dd>
<div class="block">A simple bug report dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/BugReport.html#BugReport-javax.swing.JFrame-com.compomics.util.io.file.LastSelectedFolder-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.io.File-">BugReport(JFrame, LastSelectedFolder, String, String, String, String, String, File)</a></span> - Constructor for class com.compomics.util.gui.error_handlers.<a href="com/compomics/util/gui/error_handlers/BugReport.html" title="class in com.compomics.util.gui.error_handlers">BugReport</a></dt>
<dd>
<div class="block">Creates a new BugReport dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/BugReport.html#BugReport-javax.swing.JDialog-com.compomics.util.io.file.LastSelectedFolder-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.io.File-">BugReport(JDialog, LastSelectedFolder, String, String, String, String, String, File)</a></span> - Constructor for class com.compomics.util.gui.error_handlers.<a href="com/compomics/util/gui/error_handlers/BugReport.html" title="class in com.compomics.util.gui.error_handlers">BugReport</a></dt>
<dd>
<div class="block">Creates a new BugReport dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#builder">builder</a></span> - Variable in class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">GsonBuilder that can be used to append interfaces so the parser knows how
 to handle them.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideAndProteinBuilder.html#buildPeptidesAndProteins-com.compomics.util.experiment.identification.matches.SpectrumMatch-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-boolean-">buildPeptidesAndProteins(SpectrumMatch, SequenceMatchingParameters, SequenceProvider, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideAndProteinBuilder.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideAndProteinBuilder</a></dt>
<dd>
<div class="block">Creates the peptides and protein instances based on the given spectrum
 match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#byonic">byonic</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Byonic search engine integrated in the Byonic protein metrics
 interface.</div>
</dd>
<dt><a href="com/compomics/util/gui/renderers/ByteArrayRenderer.html" title="class in com.compomics.util.gui.renderers"><span class="typeNameLink">ByteArrayRenderer</span></a> - Class in <a href="com/compomics/util/gui/renderers/package-summary.html">com.compomics.util.gui.renderers</a></dt>
<dd>
<div class="block">ByteArrayRenderer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/ByteArrayRenderer.html#ByteArrayRenderer--">ByteArrayRenderer()</a></span> - Constructor for class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/ByteArrayRenderer.html" title="class in com.compomics.util.gui.renderers">ByteArrayRenderer</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#BYTES_TO_HOLD_DOUBLE">BYTES_TO_HOLD_DOUBLE</a></span> - Static variable in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Defines the number of bytes required in an UNENCODED byte array to hold a
 dingle double value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#BYTES_TO_HOLD_FLOAT">BYTES_TO_HOLD_FLOAT</a></span> - Static variable in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Defines the number of bytes required in an UNENCODED byte array to hold a
 single float value.</div>
</dd>
</dl>
<a name="I:C">
<!--   -->
</a>
<h2 class="title">C</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#C">C</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#C">C</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The carbon atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#C3H9N">C3H9N</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">C3H9N loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#C_ION">C_ION</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Identifier for a c ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#C_ION">C_ION</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Identifier for a c ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#C_TERM_ENZYMATIC">C_TERM_ENZYMATIC</a></span> - Static variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistribution.html#calculate--">calculate()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistribution.html" title="class in com.compomics.util.general">IsotopicDistribution</a></dt>
<dd>
<div class="block">This method will do the calculations</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html#calculate--">calculate()</a></span> - Method in class com.compomics.util.gui.isotopic_calculator.<a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html" title="class in com.compomics.util.gui.isotopic_calculator">IsotopeDistributionGui</a></dt>
<dd>
<div class="block">This method will do the calculations</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/MassCalc.html#calculateMass-java.lang.String-">calculateMass(String)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/MassCalc.html" title="class in com.compomics.util.general">MassCalc</a></dt>
<dd>
<div class="block">This method attempts to calculate the mass of a chemical formula.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLI.html#call--">call()</a></span> - Method in class com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLI.html" title="class in com.compomics.cli.enzymes">EnzymesCLI</a></dt>
<dd>
<div class="block">Calling this method will run the process.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html#call--">call()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html" title="class in com.compomics.cli.identification_parameters">AbstractIdentificationParametersCli</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLI.html#call--">call()</a></span> - Method in class com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLI.html" title="class in com.compomics.cli.modifications">ModificationsCLI</a></dt>
<dd>
<div class="block">Calling this method will run the process.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/file_handling/FileDisplayDialog.html#canceled--">canceled()</a></span> - Method in class com.compomics.util.gui.file_handling.<a href="com/compomics/util/gui/file_handling/FileDisplayDialog.html" title="class in com.compomics.util.gui.file_handling">FileDisplayDialog</a></dt>
<dd>
<div class="block">Returns if the dialog was canceled by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingActionListener.html#cancelPressed--">cancelPressed()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingActionListener.html" title="interface in com.compomics.util.waiting">WaitingActionListener</a></dt>
<dd>
<div class="block">Method called whenever the user pressed cancel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#canReverse--">canReverse()</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Indicates whether the tag can be reversed (ie if termini are mass gaps
 with mass &ge; water).</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/Carbon.html" title="class in com.compomics.util.experiment.biology.atoms.impl"><span class="typeNameLink">Carbon</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a></dt>
<dd>
<div class="block">Carbon.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/impl/Carbon.html#Carbon--">Carbon()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.impl.<a href="com/compomics/util/experiment/biology/atoms/impl/Carbon.html" title="class in com.compomics.util.experiment.biology.atoms.impl">Carbon</a></dt>
<dd>
<div class="block">Constructor</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/ExceptionHandler.html#catchException-java.lang.Exception-">catchException(Exception)</a></span> - Method in class com.compomics.util.exceptions.<a href="com/compomics/util/exceptions/ExceptionHandler.html" title="class in com.compomics.util.exceptions">ExceptionHandler</a></dt>
<dd>
<div class="block">Catches an exception and informs the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#catchException-java.lang.Exception-">catchException(Exception)</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">This method is called whenever an exception is encountered in a separate
 thread.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#cd-java.lang.String-">cd(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">CD to a specific directory on a remote FTP server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#cdup--">cdup()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Go up one directory on remote system.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#CH4OS">CH4OS</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">CH4OS loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#changeRepresentativeSite-java.lang.String-int-int-">changeRepresentativeSite(String, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Changes the representative site for a given ambiguously localized
 modification in all maps.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#character">character</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Character which was chosen.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#charAt-int-">charAt(int)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Returns the amino acid at the given index on the sequence in its single
 letter code.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/Charge.html" title="class in com.compomics.util.experiment.biology.ions"><span class="typeNameLink">Charge</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/package-summary.html">com.compomics.util.experiment.biology.ions</a></dt>
<dd>
<div class="block">This class contains convenience methods for the handling of charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Charge.html#Charge--">Charge()</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Charge.html" title="class in com.compomics.util.experiment.biology.ions">Charge</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#charge">charge</a></span> - Variable in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">The inferred charge of the ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#chargeValidated-com.compomics.util.experiment.biology.ions.Ion-int-int-">chargeValidated(Ion, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the given charge can be found on the
 given fragment ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#checkDuplicateTitles--">checkDuplicateTitles()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns if the spectra should be checked for duplicate titles or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/CompomicsWrapper.html#checkForNewDeployedVersion-java.lang.String-com.compomics.software.autoupdater.MavenJarFile-java.net.URL-java.lang.String-boolean-boolean-boolean-java.awt.Image-java.awt.Image-boolean-">checkForNewDeployedVersion(String, MavenJarFile, URL, String, boolean, boolean, boolean, Image, Image, boolean)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/CompomicsWrapper.html" title="class in com.compomics.software">CompomicsWrapper</a></dt>
<dd>
<div class="block">Check if a newer version of the tool is deployed in the Maven repository,
 and closes the tool if the user decided to update.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/CompomicsWrapper.html#checkForNewVersion-java.lang.String-java.lang.String-java.lang.String-">checkForNewVersion(String, String, String)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/CompomicsWrapper.html" title="class in com.compomics.software">CompomicsWrapper</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">use the Maven repository option instead:
 checkForNewDeployedVersion</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/CompomicsWrapper.html#checkForNewVersion-java.lang.String-java.lang.String-java.lang.String-boolean-java.lang.String-java.lang.String-">checkForNewVersion(String, String, String, boolean, String, String)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/CompomicsWrapper.html" title="class in com.compomics.software">CompomicsWrapper</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">use the Maven repository option instead:
 checkForNewDeployedVersion</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#checkIfURLExists-java.lang.String-java.lang.String-java.lang.String-">checkIfURLExists(String, String, String)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Check if a given URL exists.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/CompomicsWrapper.html#checkJavaVersion-java.lang.String-">checkJavaVersion(String)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/CompomicsWrapper.html" title="class in com.compomics.software">CompomicsWrapper</a></dt>
<dd>
<div class="block">Checks if the user is running Java 64 bit and shows a warning if not, and
 shows a dialog with a warning and a link to the JavaTroubleShooting page
 if not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#checkMgfSize--">checkMgfSize()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns if the mgf should be checked for size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#checkModel--">checkModel()</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#checkModificationPattern-com.compomics.util.experiment.identification.protein_inference.PeptideProteinMapping-">checkModificationPattern(PeptideProteinMapping)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Checking if peptide-protein should be discarded due to pattern
 modification conflict.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#checkPeakPicking--">checkPeakPicking()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns if the spectra should be checked for peak picking or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#checkProbabilityRange-double-">checkProbabilityRange(double)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Checks that a probability is between 0 and 1 and throws an
 IllegalArgumentException otherwise.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#checkProbabilityRangeInPercent-double-">checkProbabilityRangeInPercent(double)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Checks that a probability is between 0 % and 100 % and throws an
 IllegalArgumentException otherwise.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html" title="class in com.compomics.util.gui.spectrum"><span class="typeNameLink">ChromatogramPanel</span></a> - Class in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">This class provides a JPanel that can display a profileChromatogram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html#ChromatogramPanel-double:A-double:A-">ChromatogramPanel(double[], double[])</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html" title="class in com.compomics.util.gui.spectrum">ChromatogramPanel</a></dt>
<dd>
<div class="block">This constructor creates a ChromatogramPanel based on the passed
 parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html#ChromatogramPanel-double:A-double:A-java.lang.String-java.lang.String-">ChromatogramPanel(double[], double[], String, String)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html" title="class in com.compomics.util.gui.spectrum">ChromatogramPanel</a></dt>
<dd>
<div class="block">This constructor creates a ChromatogramPanel based on the passed
 parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html#ChromatogramPanel-double:A-double:A-java.lang.String-java.lang.String-java.lang.Integer-">ChromatogramPanel(double[], double[], String, String, Integer)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html" title="class in com.compomics.util.gui.spectrum">ChromatogramPanel</a></dt>
<dd>
<div class="block">This constructor creates a ChromatogramPanel based on the passed
 parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html#ChromatogramPanel-double:A-double:A-java.lang.String-java.lang.String-java.lang.Integer-boolean-">ChromatogramPanel(double[], double[], String, String, Integer, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html" title="class in com.compomics.util.gui.spectrum">ChromatogramPanel</a></dt>
<dd>
<div class="block">This constructor creates a ChromatogramPanel based on the passed
 parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#clear--">clear()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Clears the filter items.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#clear--">clear()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Clears the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#clearAllScores--">clearAllScores()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Clears all scores.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#clearCharges--">clearCharges()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Clears the selected charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#clearEnzymes--">clearEnzymes()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Clears the enzymes set including specificity and missed cleavages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#clearFactory--">clearFactory()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Clears the factory getInstance() needs to be called afterwards.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#clearFactory--">clearFactory()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Clears the factory getInstance() needs to be called afterwards.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#clearFixedModifications--">clearFixedModifications()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Clears fixed modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#clearFoundCharges--">clearFoundCharges()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Clears the found charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#clearIonTypes--">clearIonTypes()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Clears the ion types annotated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#clearIonTypes--">clearIonTypes()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Clears the ion types annotated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#clearNeutralLosses--">clearNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Clears the considered neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html#clearNeutralLosses--">clearNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">NeutralLossesMap</a></dt>
<dd>
<div class="block">Clears the mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#clearNeutralLosses--">clearNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Clears the considered neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/personalization/ExperimentObject.html#clearParametersMap--">clearParametersMap()</a></span> - Method in class com.compomics.util.experiment.personalization.<a href="com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></dt>
<dd>
<div class="block">Clears the loaded parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#clearRefinementModifications--">clearRefinementModifications()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Clears refinement modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#clearScores-java.lang.Integer-">clearScores(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Clears the score for the given algorithm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#clearSpectrumCounting--">clearSpectrumCounting()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Clears the spectrum counting data in cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#clearVariableModifications--">clearVariableModifications()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Clears the variable modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#clearVariableModifications--">clearVariableModifications()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Clears variable modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#clearVariantMatches--">clearVariantMatches()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Clears the list of imported variant matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#cleave-com.compomics.util.protein.Protein-">cleave(Protein)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">This method is the focus of the Enzyme instance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#cleave-com.compomics.util.protein.Protein-">cleave(Protein)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method is the focus of the Enzyme instance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#cleave-com.compomics.util.protein.Protein-int-int-">cleave(Protein, int, int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method is the focus of the Enzyme instance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/RegExEnzyme.html#cleave-com.compomics.util.protein.Protein-">cleave(Protein)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/RegExEnzyme.html" title="class in com.compomics.util.protein">RegExEnzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#clone--">clone()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#clone--">clone()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#clone--">clone()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/Filter.html#clone--">clone()</a></span> - Method in interface com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/Filter.html" title="interface in com.compomics.util.experiment.filtering">Filter</a></dt>
<dd>
<div class="block">Clones the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#clone--">clone()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html#clone--">clone()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches">ModificationMatch</a></dt>
<dd>
<div class="block">Clones the modification match into a new match with the same attributes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#clone--">clone()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Creates a new peptide draft with the same attributes as this one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#clone--">clone()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Clones the settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html#clone--">clone()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">NeutralLossesMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#clone--">clone()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#clone--">clone()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method provides a deep copy of the Header instance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#clone-com.compomics.util.parameters.identification.search.DigestionParameters-">clone(DigestionParameters)</a></span> - Static method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Clones the given parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#clone--">clone()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">Provides a cloned version of this DualEnzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#clone--">clone()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method returns a deep copy of the current Enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#clone--">clone()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">Override of the clone method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#clone--">clone()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">Override of the clone method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/RegExEnzyme.html#clone--">clone()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/RegExEnzyme.html" title="class in com.compomics.util.protein">RegExEnzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#close--">close()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Closes the db connection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#close-boolean-">close(boolean)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Closes the db connection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Closes the database connection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.iterators.<a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html" title="class in com.compomics.util.experiment.io.biology.protein.iterators">FastaIterator</a></dt>
<dd>
<div class="block">Closes the iterator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/iterators/HeaderIterator.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.iterators.<a href="com/compomics/util/experiment/io/biology/protein/iterators/HeaderIterator.html" title="class in com.compomics.util.experiment.io.biology.protein.iterators">HeaderIterator</a></dt>
<dd>
<div class="block">Closes the iterator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/IdfileReader.html#close--">close()</a></span> - Method in interface com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/IdfileReader.html" title="interface in com.compomics.util.experiment.io.identification">IdfileReader</a></dt>
<dd>
<div class="block">Closes the file reader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">AndromedaIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MascotIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MsAmandaIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MzIdentMLIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">NovorIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepxmlIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PNovoIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">TideIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html#close--">close()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">XTandemIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/FileLoggerImplementation.html#close--">close()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/FileLoggerImplementation.html" title="class in com.compomics.util.general">FileLoggerImplementation</a></dt>
<dd>
<div class="block">This method closes the stream to the output file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#close--">close()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Writes the content in cache and closes the connection to the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#close--">close()</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/TextWriter.html#close--">close()</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/TextWriter.html" title="class in com.compomics.util.io.export.writers">TextWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#close--">close()</a></span> - Method in class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Closes the buffered writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#closeFiles--">closeFiles()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Closes all opened files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#closeServer--">closeServer()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Issue the QUIT command to the FTP server and close the connection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#closeWhenComplete-boolean-">closeWhenComplete(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Set if the dialog is to be closed when the process is complete.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#color2Hex-java.awt.Color-">color2Hex(Color)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Converts a color to hex format for use in HTML tags.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#columnPermutation-int-int-">columnPermutation(int, int)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/db/ColumnTypeConverter.html" title="class in com.compomics.util.db"><span class="typeNameLink">ColumnTypeConverter</span></a> - Class in <a href="com/compomics/util/db/package-summary.html">com.compomics.util.db</a></dt>
<dd>
<div class="block">This class converts column types into Strings representing
 corresponding Java types.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/ColumnTypeConverter.html#ColumnTypeConverter--">ColumnTypeConverter()</a></span> - Constructor for class com.compomics.util.db.<a href="com/compomics/util/db/ColumnTypeConverter.html" title="class in com.compomics.util.db">ColumnTypeConverter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/cli/enzymes/package-summary.html">com.compomics.cli.enzymes</a> - package com.compomics.cli.enzymes</dt>
<dd>
<div class="block">Command line for the handling of enzymes.</div>
</dd>
<dt><a href="com/compomics/cli/fasta/package-summary.html">com.compomics.cli.fasta</a> - package com.compomics.cli.fasta</dt>
<dd>
<div class="block">Command line for the handling of fasta files.</div>
</dd>
<dt><a href="com/compomics/cli/identification_parameters/package-summary.html">com.compomics.cli.identification_parameters</a> - package com.compomics.cli.identification_parameters</dt>
<dd>
<div class="block">Command line for the handling of identification parameters.</div>
</dd>
<dt><a href="com/compomics/cli/modifications/package-summary.html">com.compomics.cli.modifications</a> - package com.compomics.cli.modifications</dt>
<dd>
<div class="block">Command line for the handling of modifications.</div>
</dd>
<dt><a href="com/compomics/cli/peptide_mapper/package-summary.html">com.compomics.cli.peptide_mapper</a> - package com.compomics.cli.peptide_mapper</dt>
<dd>
<div class="block">Command line for PeptideMapper.</div>
</dd>
<dt><a href="com/compomics/scripts_marc/package-summary.html">com.compomics.scripts_marc</a> - package com.compomics.scripts_marc</dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/software/package-summary.html">com.compomics.software</a> - package com.compomics.software</dt>
<dd>
<div class="block">Convenience classes for command line usage of compomics-utilities tools.</div>
</dd>
<dt><a href="com/compomics/software/autoupdater/package-summary.html">com.compomics.software.autoupdater</a> - package com.compomics.software.autoupdater</dt>
<dd>
<div class="block">Classes related to auto updating compomics-utilities tools.</div>
</dd>
<dt><a href="com/compomics/software/cli/package-summary.html">com.compomics.software.cli</a> - package com.compomics.software.cli</dt>
<dd>
<div class="block">Convenience classes for the handling of command lines.</div>
</dd>
<dt><a href="com/compomics/software/dialogs/package-summary.html">com.compomics.software.dialogs</a> - package com.compomics.software.dialogs</dt>
<dd>
<div class="block">Dialogs for starting compomics-utilities tools and setting their Java options.</div>
</dd>
<dt><a href="com/compomics/software/settings/package-summary.html">com.compomics.software.settings</a> - package com.compomics.software.settings</dt>
<dd>
<div class="block">Settings for compomics-utilities tools.</div>
</dd>
<dt><a href="com/compomics/software/settings/gui/package-summary.html">com.compomics.software.settings.gui</a> - package com.compomics.software.settings.gui</dt>
<dd>
<div class="block">GUI settings for compomics-utilities tools.</div>
</dd>
<dt><a href="com/compomics/util/package-summary.html">com.compomics.util</a> - package com.compomics.util</dt>
<dd>
<div class="block">General utilities classes.</div>
</dd>
<dt><a href="com/compomics/util/db/package-summary.html">com.compomics.util.db</a> - package com.compomics.util.db</dt>
<dd>
<div class="block">Database related classes.</div>
</dd>
<dt><a href="com/compomics/util/db/components/package-summary.html">com.compomics.util.db.components</a> - package com.compomics.util.db.components</dt>
<dd>
<div class="block">Database related components.</div>
</dd>
<dt><a href="com/compomics/util/db/interfaces/package-summary.html">com.compomics.util.db.interfaces</a> - package com.compomics.util.db.interfaces</dt>
<dd>
<div class="block">Database related interfaces.</div>
</dd>
<dt><a href="com/compomics/util/db/object/package-summary.html">com.compomics.util.db.object</a> - package com.compomics.util.db.object</dt>
<dd>
<div class="block">This package contains the classes relative to the object database.</div>
</dd>
<dt><a href="com/compomics/util/db/object/objects/package-summary.html">com.compomics.util.db.object.objects</a> - package com.compomics.util.db.object.objects</dt>
<dd>
<div class="block">This package contains classes of objects that can be stored in the database.</div>
</dd>
<dt><a href="com/compomics/util/enumeration/package-summary.html">com.compomics.util.enumeration</a> - package com.compomics.util.enumeration</dt>
<dd>
<div class="block">Enumeration used by compomics-utilities.</div>
</dd>
<dt><a href="com/compomics/util/examples/package-summary.html">com.compomics.util.examples</a> - package com.compomics.util.examples</dt>
<dd>
<div class="block">Examples of how the compomics-utilities library can
        can be used by other projects.</div>
</dd>
<dt><a href="com/compomics/util/exceptions/package-summary.html">com.compomics.util.exceptions</a> - package com.compomics.util.exceptions</dt>
<dd>
<div class="block">Classes for the handling of exceptions.</div>
</dd>
<dt><a href="com/compomics/util/exceptions/exception_handlers/package-summary.html">com.compomics.util.exceptions.exception_handlers</a> - package com.compomics.util.exceptions.exception_handlers</dt>
<dd>
<div class="block">Implementations of the ExceptionHandler interface.</div>
</dd>
<dt><a href="com/compomics/util/experiment/package-summary.html">com.compomics.util.experiment</a> - package com.compomics.util.experiment</dt>
<dd>
<div class="block">Gene factory classes.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/package-summary.html">com.compomics.util.experiment.biology.aminoacids</a> - package com.compomics.util.experiment.biology.aminoacids</dt>
<dd>
<div class="block">The amino acid classes.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a> - package com.compomics.util.experiment.biology.aminoacids.impl</dt>
<dd>
<div class="block">The amino acid implementations.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/sequence/package-summary.html">com.compomics.util.experiment.biology.aminoacids.sequence</a> - package com.compomics.util.experiment.biology.aminoacids.sequence</dt>
<dd>
<div class="block">Classes related to the handling of protein sequences.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/package-summary.html">com.compomics.util.experiment.biology.atoms</a> - package com.compomics.util.experiment.biology.atoms</dt>
<dd>
<div class="block">The atom classes.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a> - package com.compomics.util.experiment.biology.atoms.impl</dt>
<dd>
<div class="block">Atoms implementations.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/enzymes/package-summary.html">com.compomics.util.experiment.biology.enzymes</a> - package com.compomics.util.experiment.biology.enzymes</dt>
<dd>
<div class="block">Classes related to enzymes.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/genes/package-summary.html">com.compomics.util.experiment.biology.genes</a> - package com.compomics.util.experiment.biology.genes</dt>
<dd>
<div class="block">Classes related to genes.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/genes/ensembl/package-summary.html">com.compomics.util.experiment.biology.genes.ensembl</a> - package com.compomics.util.experiment.biology.genes.ensembl</dt>
<dd>
<div class="block">Classes for the handling of Ensembl data.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/genes/go/package-summary.html">com.compomics.util.experiment.biology.genes.go</a> - package com.compomics.util.experiment.biology.genes.go</dt>
<dd>
<div class="block">Classes for the handling of GO terms.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/package-summary.html">com.compomics.util.experiment.biology.ions</a> - package com.compomics.util.experiment.biology.ions</dt>
<dd>
<div class="block">Experiment classes related to ions.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/impl/package-summary.html">com.compomics.util.experiment.biology.ions.impl</a> - package com.compomics.util.experiment.biology.ions.impl</dt>
<dd>
<div class="block">Ion implementations.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/modifications/package-summary.html">com.compomics.util.experiment.biology.modifications</a> - package com.compomics.util.experiment.biology.modifications</dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/proteins/package-summary.html">com.compomics.util.experiment.biology.proteins</a> - package com.compomics.util.experiment.biology.proteins</dt>
<dd>
<div class="block">Classes related to proteins.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/taxonomy/package-summary.html">com.compomics.util.experiment.biology.taxonomy</a> - package com.compomics.util.experiment.biology.taxonomy</dt>
<dd>
<div class="block">Classes related to species and taxonomy.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/taxonomy/mappings/package-summary.html">com.compomics.util.experiment.biology.taxonomy.mappings</a> - package com.compomics.util.experiment.biology.taxonomy.mappings</dt>
<dd>
<div class="block">Classes related to species mapping.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/variants/package-summary.html">com.compomics.util.experiment.biology.variants</a> - package com.compomics.util.experiment.biology.variants</dt>
<dd>
<div class="block">Experiment classes related to variants.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/variants/amino_acids/package-summary.html">com.compomics.util.experiment.biology.variants.amino_acids</a> - package com.compomics.util.experiment.biology.variants.amino_acids</dt>
<dd>
<div class="block">Type of implemented variants.</div>
</dd>
<dt><a href="com/compomics/util/experiment/filtering/package-summary.html">com.compomics.util.experiment.filtering</a> - package com.compomics.util.experiment.filtering</dt>
<dd>
<div class="block">Filtering classes.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a> - package com.compomics.util.experiment.identification</dt>
<dd>
<div class="block">Experiment classes related to identifications.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/amino_acid_tags/package-summary.html">com.compomics.util.experiment.identification.amino_acid_tags</a> - package com.compomics.util.experiment.identification.amino_acid_tags</dt>
<dd>
<div class="block">Amino Acid Tag classes.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/package-summary.html">com.compomics.util.experiment.identification.filtering</a> - package com.compomics.util.experiment.identification.filtering</dt>
<dd>
<div class="block">Classes for the filtering of identification objects.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/items/package-summary.html">com.compomics.util.experiment.identification.filtering.items</a> - package com.compomics.util.experiment.identification.filtering.items</dt>
<dd>
<div class="block">Items used by the filters to filter on.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/identification_parameters/package-summary.html">com.compomics.util.experiment.identification.identification_parameters</a> - package com.compomics.util.experiment.identification.identification_parameters</dt>
<dd>
<div class="block">Parameters settings for the spectrum identification algorithms.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/matches/package-summary.html">com.compomics.util.experiment.identification.matches</a> - package com.compomics.util.experiment.identification.matches</dt>
<dd>
<div class="block">Experiment classes related to matches.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/matches_iterators/package-summary.html">com.compomics.util.experiment.identification.matches_iterators</a> - package com.compomics.util.experiment.identification.matches_iterators</dt>
<dd>
<div class="block">Identification match iterators.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/modification/package-summary.html">com.compomics.util.experiment.identification.modification</a> - package com.compomics.util.experiment.identification.modification</dt>
<dd>
<div class="block">Additional PSM scoring classes.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/modification/scores/package-summary.html">com.compomics.util.experiment.identification.modification.scores</a> - package com.compomics.util.experiment.identification.modification.scores</dt>
<dd>
<div class="block">PTM scores.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation</a> - package com.compomics.util.experiment.identification.peptide_fragmentation</dt>
<dd>
<div class="block">Peptide fragment intensities.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models</a> - package com.compomics.util.experiment.identification.peptide_fragmentation.models</dt>
<dd>
<div class="block">Predictors for peptide fragment intensities.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration</a> - package com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration</dt>
<dd>
<div class="block">This package contains the classes needed to configure the ms2pip features.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a> - package com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</dt>
<dd>
<div class="block">This package contains features implemented for ms2pip.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic</a> - package com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic</dt>
<dd>
<div class="block">This package contains generic features implemented for ms2pip.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_generation/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation</a> - package com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation</dt>
<dd>
<div class="block">This package contains the classes needed to generate ms2pip features.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_shaker/package-summary.html">com.compomics.util.experiment.identification.peptide_shaker</a> - package com.compomics.util.experiment.identification.peptide_shaker</dt>
<dd>
<div class="block">Experiment classes used in PeptideShaker calculations.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_inference/package-summary.html">com.compomics.util.experiment.identification.protein_inference</a> - package com.compomics.util.experiment.identification.protein_inference</dt>
<dd>
<div class="block">Classes related to protein inference.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/package-summary.html">com.compomics.util.experiment.identification.protein_inference.fm_index</a> - package com.compomics.util.experiment.identification.protein_inference.fm_index</dt>
<dd>
<div class="block">Protein sequence database indexing classes based on Burrows-Wheeler transform.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/package-summary.html">com.compomics.util.experiment.identification.protein_sequences</a> - package com.compomics.util.experiment.identification.protein_sequences</dt>
<dd>
<div class="block">Protein sequence handling related classes.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion</a> - package com.compomics.util.experiment.identification.protein_sequences.digestion</dt>
<dd>
<div class="block">This package contains classes used to digest protein sequences into peptides.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion.iterators</a> - package com.compomics.util.experiment.identification.protein_sequences.digestion.iterators</dt>
<dd>
<div class="block">This package contains implementations of the SequenceIterator interface.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/psm_scoring/package-summary.html">com.compomics.util.experiment.identification.psm_scoring</a> - package com.compomics.util.experiment.identification.psm_scoring</dt>
<dd>
<div class="block">Main PSM scoring class.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/package-summary.html">com.compomics.util.experiment.identification.psm_scoring.psm_scores</a> - package com.compomics.util.experiment.identification.psm_scoring.psm_scores</dt>
<dd>
<div class="block">Main PTM scoring classes.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation</a> - package com.compomics.util.experiment.identification.spectrum_annotation</dt>
<dd>
<div class="block">Classes used for ms2 spectrum annotation.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators</a> - package com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators</dt>
<dd>
<div class="block">This package contains implementations of the SimpleAnnotator allowing the annotation of specific sets of ions.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators</a> - package com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators</dt>
<dd>
<div class="block">This package contains implementations of the SpectrumAnnotator allowing the annotation of all ions.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_assumptions/package-summary.html">com.compomics.util.experiment.identification.spectrum_assumptions</a> - package com.compomics.util.experiment.identification.spectrum_assumptions</dt>
<dd>
<div class="block">Implementations of the SpectrumIdentificationAssumption.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/utils/package-summary.html">com.compomics.util.experiment.identification.utils</a> - package com.compomics.util.experiment.identification.utils</dt>
<dd>
<div class="block">This package contains utilities classes to handle identification objects.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/validation/package-summary.html">com.compomics.util.experiment.identification.validation</a> - package com.compomics.util.experiment.identification.validation</dt>
<dd>
<div class="block">Experiment classes related to identification matches validation.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/package-summary.html">com.compomics.util.experiment.io.biology.protein</a> - package com.compomics.util.experiment.io.biology.protein</dt>
<dd>
<div class="block">Experiment IO classes for biological resources.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/converters/package-summary.html">com.compomics.util.experiment.io.biology.protein.converters</a> - package com.compomics.util.experiment.io.biology.protein.converters</dt>
<dd>
<div class="block">Fasta file converters.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/iterators/package-summary.html">com.compomics.util.experiment.io.biology.protein.iterators</a> - package com.compomics.util.experiment.io.biology.protein.iterators</dt>
<dd>
<div class="block">Fasta file iterators.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/package-summary.html">com.compomics.util.experiment.io.identification</a> - package com.compomics.util.experiment.io.identification</dt>
<dd>
<div class="block">Experiment identification classes.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a> - package com.compomics.util.experiment.io.identification.idfilereaders</dt>
<dd>
<div class="block">Experiment classes related to reading search engine files.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/mass_spectrometry/export/package-summary.html">com.compomics.util.experiment.io.mass_spectrometry.export</a> - package com.compomics.util.experiment.io.mass_spectrometry.export</dt>
<dd>
<div class="block">Experiment classes related to exporting mass spectrometry files.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/package-summary.html">com.compomics.util.experiment.io.mass_spectrometry.mgf</a> - package com.compomics.util.experiment.io.mass_spectrometry.mgf</dt>
<dd>
<div class="block">Reading and indexing of mgf files.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/package-summary.html">com.compomics.util.experiment.io.mass_spectrometry.msp</a> - package com.compomics.util.experiment.io.mass_spectrometry.msp</dt>
<dd>
<div class="block">Reading and indexing of msp files.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/parameters/package-summary.html">com.compomics.util.experiment.io.parameters</a> - package com.compomics.util.experiment.io.parameters</dt>
<dd>
<div class="block">Experiment IO classes relative to the handling of parameters.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/package-summary.html">com.compomics.util.experiment.mass_spectrometry</a> - package com.compomics.util.experiment.mass_spectrometry</dt>
<dd>
<div class="block">Experiment classes related to mass spectrometry.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/indexes/package-summary.html">com.compomics.util.experiment.mass_spectrometry.indexes</a> - package com.compomics.util.experiment.mass_spectrometry.indexes</dt>
<dd>
<div class="block">Classes related to the indexing of spectra.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/package-summary.html">com.compomics.util.experiment.mass_spectrometry.proteowizard</a> - package com.compomics.util.experiment.mass_spectrometry.proteowizard</dt>
<dd>
<div class="block">Package containing the classes used to interact with ProteoWizard.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/gui/package-summary.html">com.compomics.util.experiment.mass_spectrometry.proteowizard.gui</a> - package com.compomics.util.experiment.mass_spectrometry.proteowizard.gui</dt>
<dd>
<div class="block">Package containing the GUI classes used to interact with ProteoWizard.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/spectra/package-summary.html">com.compomics.util.experiment.mass_spectrometry.spectra</a> - package com.compomics.util.experiment.mass_spectrometry.spectra</dt>
<dd>
<div class="block">Experiment classes related to spectra.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/utils/package-summary.html">com.compomics.util.experiment.mass_spectrometry.utils</a> - package com.compomics.util.experiment.mass_spectrometry.utils</dt>
<dd>
<div class="block">Utility classes for mass spectrometry.</div>
</dd>
<dt><a href="com/compomics/util/experiment/normalization/package-summary.html">com.compomics.util.experiment.normalization</a> - package com.compomics.util.experiment.normalization</dt>
<dd>
<div class="block">Classes related to quantification values normalization.</div>
</dd>
<dt><a href="com/compomics/util/experiment/patient/package-summary.html">com.compomics.util.experiment.patient</a> - package com.compomics.util.experiment.patient</dt>
<dd>
<div class="block">Experiment classes related to patent information.</div>
</dd>
<dt><a href="com/compomics/util/experiment/personalization/package-summary.html">com.compomics.util.experiment.personalization</a> - package com.compomics.util.experiment.personalization</dt>
<dd>
<div class="block">Experiment customization classes.</div>
</dd>
<dt><a href="com/compomics/util/experiment/quantification/package-summary.html">com.compomics.util.experiment.quantification</a> - package com.compomics.util.experiment.quantification</dt>
<dd>
<div class="block">Experiment classes related to quantification.</div>
</dd>
<dt><a href="com/compomics/util/experiment/quantification/reporterion/package-summary.html">com.compomics.util.experiment.quantification.reporterion</a> - package com.compomics.util.experiment.quantification.reporterion</dt>
<dd>
<div class="block">Experiment classes related to reporter ions.</div>
</dd>
<dt><a href="com/compomics/util/experiment/quantification/spectrumcounting/package-summary.html">com.compomics.util.experiment.quantification.spectrumcounting</a> - package com.compomics.util.experiment.quantification.spectrumcounting</dt>
<dd>
<div class="block">Experiment classes related to spectrum counting.</div>
</dd>
<dt><a href="com/compomics/util/experiment/refinement_parameters/package-summary.html">com.compomics.util.experiment.refinement_parameters</a> - package com.compomics.util.experiment.refinement_parameters</dt>
<dd>
<div class="block">Experiment classes related to specific parametrization of experiment objects.</div>
</dd>
<dt><a href="com/compomics/util/experiment/units/package-summary.html">com.compomics.util.experiment.units</a> - package com.compomics.util.experiment.units</dt>
<dd>
<div class="block">Unit related classes.</div>
</dd>
<dt><a href="com/compomics/util/general/package-summary.html">com.compomics.util.general</a> - package com.compomics.util.general</dt>
<dd>
<div class="block">General, top-level tasks that are used throughout the other packages in the util section.</div>
</dd>
<dt><a href="com/compomics/util/general/servlet/package-summary.html">com.compomics.util.general.servlet</a> - package com.compomics.util.general.servlet</dt>
<dd>
<div class="block">Servlet representation of the
        classes from the general utilities package (com.compomics.util.general)
        <br></div>
</dd>
<dt><a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a> - package com.compomics.util.gui</dt>
<dd>
<div class="block">Basic GUI classes.</div>
</dd>
<dt><a href="com/compomics/util/gui/atoms/package-summary.html">com.compomics.util.gui.atoms</a> - package com.compomics.util.gui.atoms</dt>
<dd>
<div class="block">Atoms GUI dialogs.</div>
</dd>
<dt><a href="com/compomics/util/gui/error_handlers/package-summary.html">com.compomics.util.gui.error_handlers</a> - package com.compomics.util.gui.error_handlers</dt>
<dd>
<div class="block">Help handlers.</div>
</dd>
<dt><a href="com/compomics/util/gui/error_handlers/notification/package-summary.html">com.compomics.util.gui.error_handlers.notification</a> - package com.compomics.util.gui.error_handlers.notification</dt>
<dd>
<div class="block">Notification dialogs.</div>
</dd>
<dt><a href="com/compomics/util/gui/events/package-summary.html">com.compomics.util.gui.events</a> - package com.compomics.util.gui.events</dt>
<dd>
<div class="block">GUI event classes.</div>
</dd>
<dt><a href="com/compomics/util/gui/export/graphics/package-summary.html">com.compomics.util.gui.export.graphics</a> - package com.compomics.util.gui.export.graphics</dt>
<dd>
<div class="block">Exporting of graphics.</div>
</dd>
<dt><a href="com/compomics/util/gui/export/report/package-summary.html">com.compomics.util.gui.export.report</a> - package com.compomics.util.gui.export.report</dt>
<dd>
<div class="block">Report editor GUI.</div>
</dd>
<dt><a href="com/compomics/util/gui/file_handling/package-summary.html">com.compomics.util.gui.file_handling</a> - package com.compomics.util.gui.file_handling</dt>
<dd>
<div class="block">File handling.</div>
</dd>
<dt><a href="com/compomics/util/gui/filtering/package-summary.html">com.compomics.util.gui.filtering</a> - package com.compomics.util.gui.filtering</dt>
<dd>
<div class="block">GUI filters.</div>
</dd>
<dt><a href="com/compomics/util/gui/genes/package-summary.html">com.compomics.util.gui.genes</a> - package com.compomics.util.gui.genes</dt>
<dd>
<div class="block">Dialogs for the handling of genes.</div>
</dd>
<dt><a href="com/compomics/util/gui/interfaces/package-summary.html">com.compomics.util.gui.interfaces</a> - package com.compomics.util.gui.interfaces</dt>
<dd>
<div class="block">GUI related interfaces.</div>
</dd>
<dt><a href="com/compomics/util/gui/isotopic_calculator/package-summary.html">com.compomics.util.gui.isotopic_calculator</a> - package com.compomics.util.gui.isotopic_calculator</dt>
<dd>
<div class="block">GUI class for the Isotopic Distribution calculator.</div>
</dd>
<dt><a href="com/compomics/util/gui/modification/package-summary.html">com.compomics.util.gui.modification</a> - package com.compomics.util.gui.modification</dt>
<dd>
<div class="block">PTM GUI dialogs.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/package-summary.html">com.compomics.util.gui.parameters.identification</a> - package com.compomics.util.gui.parameters.identification</dt>
<dd>
<div class="block">GUI classes for parameters, settings and preferences used for identification.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/advanced/package-summary.html">com.compomics.util.gui.parameters.identification.advanced</a> - package com.compomics.util.gui.parameters.identification.advanced</dt>
<dd>
<div class="block">Dialogs for advanced identification parameters.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a> - package com.compomics.util.gui.parameters.identification.algorithm</dt>
<dd>
<div class="block">Search algorithm settings dialogs.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/pride/package-summary.html">com.compomics.util.gui.parameters.identification.pride</a> - package com.compomics.util.gui.parameters.identification.pride</dt>
<dd>
<div class="block">Dialogs for  pride reprocessing parameters.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/search/package-summary.html">com.compomics.util.gui.parameters.identification.search</a> - package com.compomics.util.gui.parameters.identification.search</dt>
<dd>
<div class="block">Dialogs for searh parameters.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/tools/package-summary.html">com.compomics.util.gui.parameters.tools</a> - package com.compomics.util.gui.parameters.tools</dt>
<dd>
<div class="block">GUI classes for parameters, settings and preferences used in tools.</div>
</dd>
<dt><a href="com/compomics/util/gui/protein/package-summary.html">com.compomics.util.gui.protein</a> - package com.compomics.util.gui.protein</dt>
<dd>
<div class="block">GUI class related to protein sequence formatting.</div>
</dd>
<dt><a href="com/compomics/util/gui/renderers/package-summary.html">com.compomics.util.gui.renderers</a> - package com.compomics.util.gui.renderers</dt>
<dd>
<div class="block">GUI renderer classes.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a> - package com.compomics.util.gui.spectrum</dt>
<dd>
<div class="block">Spectrum and Chromatogram visualization GUI classes.</div>
</dd>
<dt><a href="com/compomics/util/gui/tablemodels/package-summary.html">com.compomics.util.gui.tablemodels</a> - package com.compomics.util.gui.tablemodels</dt>
<dd>
<div class="block">Self updating table model classes.</div>
</dd>
<dt><a href="com/compomics/util/gui/utils/package-summary.html">com.compomics.util.gui.utils</a> - package com.compomics.util.gui.utils</dt>
<dd>
<div class="block">GUI utilities classes.</div>
</dd>
<dt><a href="com/compomics/util/gui/utils/user_choice/package-summary.html">com.compomics.util.gui.utils.user_choice</a> - package com.compomics.util.gui.utils.user_choice</dt>
<dd>
<div class="block">GUI utils allowing the user to choose something.</div>
</dd>
<dt><a href="com/compomics/util/gui/utils/user_choice/list_choosers/package-summary.html">com.compomics.util.gui.utils.user_choice.list_choosers</a> - package com.compomics.util.gui.utils.user_choice.list_choosers</dt>
<dd>
<div class="block">Implementations of the ListChooser.</div>
</dd>
<dt><a href="com/compomics/util/gui/variants/aa_substitutions/package-summary.html">com.compomics.util.gui.variants.aa_substitutions</a> - package com.compomics.util.gui.variants.aa_substitutions</dt>
<dd>
<div class="block">GUI classes relative to amino acid substitution.</div>
</dd>
<dt><a href="com/compomics/util/gui/waiting/waitinghandlers/package-summary.html">com.compomics.util.gui.waiting.waitinghandlers</a> - package com.compomics.util.gui.waiting.waitinghandlers</dt>
<dd>
<div class="block">Waiting handler classes.</div>
</dd>
<dt><a href="com/compomics/util/interfaces/package-summary.html">com.compomics.util.interfaces</a> - package com.compomics.util.interfaces</dt>
<dd>
<div class="block">Interfaces for the more specific implementations in the util package.</div>
</dd>
<dt><a href="com/compomics/util/io/package-summary.html">com.compomics.util.io</a> - package com.compomics.util.io</dt>
<dd>
<div class="block">Classes related to input/output.</div>
</dd>
<dt><a href="com/compomics/util/io/compression/package-summary.html">com.compomics.util.io.compression</a> - package com.compomics.util.io.compression</dt>
<dd>
<div class="block">Classes related to file compression.</div>
</dd>
<dt><a href="com/compomics/util/io/export/package-summary.html">com.compomics.util.io.export</a> - package com.compomics.util.io.export</dt>
<dd>
<div class="block">Classes related to exporting reports.</div>
</dd>
<dt><a href="com/compomics/util/io/export/styles/package-summary.html">com.compomics.util.io.export.styles</a> - package com.compomics.util.io.export.styles</dt>
<dd>
<div class="block">Export styles.</div>
</dd>
<dt><a href="com/compomics/util/io/export/writers/package-summary.html">com.compomics.util.io.export.writers</a> - package com.compomics.util.io.export.writers</dt>
<dd>
<div class="block">Export writer.</div>
</dd>
<dt><a href="com/compomics/util/io/export/xml/package-summary.html">com.compomics.util.io.export.xml</a> - package com.compomics.util.io.export.xml</dt>
<dd>
<div class="block">Convenience classes for xml exports.</div>
</dd>
<dt><a href="com/compomics/util/io/file/package-summary.html">com.compomics.util.io.file</a> - package com.compomics.util.io.file</dt>
<dd>
<div class="block">Classes related to the handling of files.</div>
</dd>
<dt><a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a> - package com.compomics.util.io.file.filefilters</dt>
<dd>
<div class="block">Classes related to file filters.</div>
</dd>
<dt><a href="com/compomics/util/io/ftp/package-summary.html">com.compomics.util.io.ftp</a> - package com.compomics.util.io.ftp</dt>
<dd>
<div class="block">Classes related to the handling of FTP.</div>
</dd>
<dt><a href="com/compomics/util/io/json/package-summary.html">com.compomics.util.io.json</a> - package com.compomics.util.io.json</dt>
<dd>
<div class="block">Classes related to the handling of json files.</div>
</dd>
<dt><a href="com/compomics/util/io/json/adapter/package-summary.html">com.compomics.util.io.json.adapter</a> - package com.compomics.util.io.json.adapter</dt>
<dd>
<div class="block">JSON adapter classes.</div>
</dd>
<dt><a href="com/compomics/util/io/json/marshallers/package-summary.html">com.compomics.util.io.json.marshallers</a> - package com.compomics.util.io.json.marshallers</dt>
<dd>
<div class="block">JSON marshaller classes.</div>
</dd>
<dt><a href="com/compomics/util/junit/package-summary.html">com.compomics.util.junit</a> - package com.compomics.util.junit</dt>
<dd>
<div class="block">Custom classpath related test class.</div>
</dd>
<dt><a href="com/compomics/util/math/package-summary.html">com.compomics.util.math</a> - package com.compomics.util.math</dt>
<dd>
<div class="block">Basic match functions.</div>
</dd>
<dt><a href="com/compomics/util/math/clustering/package-summary.html">com.compomics.util.math.clustering</a> - package com.compomics.util.math.clustering</dt>
<dd>
<div class="block">Basic clustering functions.</div>
</dd>
<dt><a href="com/compomics/util/math/clustering/settings/package-summary.html">com.compomics.util.math.clustering.settings</a> - package com.compomics.util.math.clustering.settings</dt>
<dd>
<div class="block">Settings for the clustering.</div>
</dd>
<dt><a href="com/compomics/util/math/matrix/package-summary.html">com.compomics.util.math.matrix</a> - package com.compomics.util.math.matrix</dt>
<dd>
<div class="block">Double matrix classes.</div>
</dd>
<dt><a href="com/compomics/util/math/roc/package-summary.html">com.compomics.util.math.roc</a> - package com.compomics.util.math.roc</dt>
<dd>
<div class="block">ROC curve calculation.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/package-summary.html">com.compomics.util.math.statistics</a> - package com.compomics.util.math.statistics</dt>
<dd>
<div class="block">Math classes related to statistics.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/distributions/package-summary.html">com.compomics.util.math.statistics.distributions</a> - package com.compomics.util.math.statistics.distributions</dt>
<dd>
<div class="block">Math classes related to distributions.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/linear_regression/package-summary.html">com.compomics.util.math.statistics.linear_regression</a> - package com.compomics.util.math.statistics.linear_regression</dt>
<dd>
<div class="block">Classes to perform linear regressions.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/linear_regression/filters/package-summary.html">com.compomics.util.math.statistics.linear_regression.filters</a> - package com.compomics.util.math.statistics.linear_regression.filters</dt>
<dd>
<div class="block">Implementation of algorithms performing linear regression.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/linear_regression/regressions/package-summary.html">com.compomics.util.math.statistics.linear_regression.regressions</a> - package com.compomics.util.math.statistics.linear_regression.regressions</dt>
<dd>
<div class="block">Filters to apply prior to a linear regression.</div>
</dd>
<dt><a href="com/compomics/util/memory/package-summary.html">com.compomics.util.memory</a> - package com.compomics.util.memory</dt>
<dd>
<div class="block">Memory management.</div>
</dd>
<dt><a href="com/compomics/util/messages/package-summary.html">com.compomics.util.messages</a> - package com.compomics.util.messages</dt>
<dd>
<div class="block">Classes related to sending messages to the user.</div>
</dd>
<dt><a href="com/compomics/util/nucleotide/package-summary.html">com.compomics.util.nucleotide</a> - package com.compomics.util.nucleotide</dt>
<dd>
<div class="block">Useful objects for modeling and handling nucleotides.</div>
</dd>
<dt><a href="com/compomics/util/parameters/package-summary.html">com.compomics.util.parameters</a> - package com.compomics.util.parameters</dt>
<dd>
<div class="block">Utilities parameters.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/package-summary.html">com.compomics.util.parameters.identification</a> - package com.compomics.util.parameters.identification</dt>
<dd>
<div class="block">Identification parameters.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/advanced/package-summary.html">com.compomics.util.parameters.identification.advanced</a> - package com.compomics.util.parameters.identification.advanced</dt>
<dd>
<div class="block">Advanced Identification parameters.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/search/package-summary.html">com.compomics.util.parameters.identification.search</a> - package com.compomics.util.parameters.identification.search</dt>
<dd>
<div class="block">Identification parameters relative to the search.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a> - package com.compomics.util.parameters.identification.tool_specific</dt>
<dd>
<div class="block">Parameters settings for the search algorithms.</div>
</dd>
<dt><a href="com/compomics/util/parameters/peptide_shaker/package-summary.html">com.compomics.util.parameters.peptide_shaker</a> - package com.compomics.util.parameters.peptide_shaker</dt>
<dd>
<div class="block">PeptideShaker parameters.</div>
</dd>
<dt><a href="com/compomics/util/parameters/quantification/spectrum_counting/package-summary.html">com.compomics.util.parameters.quantification.spectrum_counting</a> - package com.compomics.util.parameters.quantification.spectrum_counting</dt>
<dd>
<div class="block">Spectrum counting parameters.</div>
</dd>
<dt><a href="com/compomics/util/parameters/searchgui/package-summary.html">com.compomics.util.parameters.searchgui</a> - package com.compomics.util.parameters.searchgui</dt>
<dd>
<div class="block">SearchGUI parameters.</div>
</dd>
<dt><a href="com/compomics/util/parameters/tools/package-summary.html">com.compomics.util.parameters.tools</a> - package com.compomics.util.parameters.tools</dt>
<dd>
<div class="block">Tool parameters.</div>
</dd>
<dt><a href="com/compomics/util/pdbfinder/package-summary.html">com.compomics.util.pdbfinder</a> - package com.compomics.util.pdbfinder</dt>
<dd>
<div class="block">Classes for mapping between UniProt accession numbers and PDB structures.</div>
</dd>
<dt><a href="com/compomics/util/pdbfinder/das/readers/package-summary.html">com.compomics.util.pdbfinder.das.readers</a> - package com.compomics.util.pdbfinder.das.readers</dt>
<dd>
<div class="block">Classes for aligning and annotating PDB files when mapped to UniProt accession numbers.</div>
</dd>
<dt><a href="com/compomics/util/pdbfinder/pdb/package-summary.html">com.compomics.util.pdbfinder.pdb</a> - package com.compomics.util.pdbfinder.pdb</dt>
<dd>
<div class="block">Classes for parsing PDB files.</div>
</dd>
<dt><a href="com/compomics/util/pride/package-summary.html">com.compomics.util.pride</a> - package com.compomics.util.pride</dt>
<dd>
<div class="block">Objects related to mapping to PRIDE.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/package-summary.html">com.compomics.util.pride.prideobjects</a> - package com.compomics.util.pride.prideobjects</dt>
<dd>
<div class="block">Objects related to mapping experiment annotation to PRIDE.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/webservice/package-summary.html">com.compomics.util.pride.prideobjects.webservice</a> - package com.compomics.util.pride.prideobjects.webservice</dt>
<dd>
<div class="block">Classes for validating PRIDE web service queries.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/webservice/file/package-summary.html">com.compomics.util.pride.prideobjects.webservice.file</a> - package com.compomics.util.pride.prideobjects.webservice.file</dt>
<dd>
<div class="block">Classes for validating PRIDE web service file types.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/webservice/query/package-summary.html">com.compomics.util.pride.prideobjects.webservice.query</a> - package com.compomics.util.pride.prideobjects.webservice.query</dt>
<dd>
<div class="block">Classes for validating PRIDE web service filters.</div>
</dd>
<dt><a href="com/compomics/util/pride/validation/package-summary.html">com.compomics.util.pride.validation</a> - package com.compomics.util.pride.validation</dt>
<dd>
<div class="block">Classes for validating PRIDE XML files.</div>
</dd>
<dt><a href="com/compomics/util/protein/package-summary.html">com.compomics.util.protein</a> - package com.compomics.util.protein</dt>
<dd>
<div class="block">Classes concerning useful Objects and their operations for modeling and handling proteins.</div>
</dd>
<dt><a href="com/compomics/util/sun/package-summary.html">com.compomics.util.sun</a> - package com.compomics.util.sun</dt>
<dd>
<div class="block">Classes related to Java's Swing library.</div>
</dd>
<dt><a href="com/compomics/util/threading/package-summary.html">com.compomics.util.threading</a> - package com.compomics.util.threading</dt>
<dd>
<div class="block">Classes related to maps.</div>
</dd>
<dt><a href="com/compomics/util/waiting/package-summary.html">com.compomics.util.waiting</a> - package com.compomics.util.waiting</dt>
<dd>
<div class="block">Waiting handler interfaces.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#comet">comet</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Comet search engine, free version of Sequest.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">CometParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">The Comet specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#CometParameters--">CometParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.CometOutputFormat.html" title="enum in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">CometParameters.CometOutputFormat</span></a> - Enum in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">The available output formats.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm"><span class="typeNameLink">CometParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a></dt>
<dd>
<div class="block">Dialog for the Comet specific settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html#CometParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.CometParameters-boolean-">CometParametersDialog(Frame, CometParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">CometParametersDialog</a></dt>
<dd>
<div class="block">Creates a new CometSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html#CometParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.CometParameters-boolean-">CometParametersDialog(Dialog, Frame, CometParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">CometParametersDialog</a></dt>
<dd>
<div class="block">Creates a new CometSettingsDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#command">command</a></span> - Variable in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">The last command issued.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFormat.html#commandLineDefaultOption">commandLineDefaultOption</a></span> - Static variable in enum com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFormat.html" title="enum in com.compomics.util.io.export">ExportFormat</a></dt>
<dd>
<div class="block">The default export format to use for command line exports.</div>
</dd>
<dt><a href="com/compomics/util/exceptions/exception_handlers/CommandLineExceptionHandler.html" title="class in com.compomics.util.exceptions.exception_handlers"><span class="typeNameLink">CommandLineExceptionHandler</span></a> - Class in <a href="com/compomics/util/exceptions/exception_handlers/package-summary.html">com.compomics.util.exceptions.exception_handlers</a></dt>
<dd>
<div class="block">Exception handler for command line processes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/exception_handlers/CommandLineExceptionHandler.html#CommandLineExceptionHandler--">CommandLineExceptionHandler()</a></span> - Constructor for class com.compomics.util.exceptions.exception_handlers.<a href="com/compomics/util/exceptions/exception_handlers/CommandLineExceptionHandler.html" title="class in com.compomics.util.exceptions.exception_handlers">CommandLineExceptionHandler</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html#commandLineOption">commandLineOption</a></span> - Variable in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsFormat</a></dt>
<dd>
<div class="block">The command line option name.</div>
</dd>
<dt><a href="com/compomics/util/general/CommandLineParser.html" title="class in com.compomics.util.general"><span class="typeNameLink">CommandLineParser</span></a> - Class in <a href="com/compomics/util/general/package-summary.html">com.compomics.util.general</a></dt>
<dd>
<div class="block">This class provides a generic interface for the parsing of command-line
 arguments, options and flags.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/CommandLineParser.html#CommandLineParser-java.lang.String:A-">CommandLineParser(String[])</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/CommandLineParser.html" title="class in com.compomics.util.general">CommandLineParser</a></dt>
<dd>
<div class="block">The constructor requires the caller to provide it with a command line
 arguments String[] that will be the basis of the parsing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/CommandLineParser.html#CommandLineParser-java.lang.String:A-java.lang.String:A-">CommandLineParser(String[], String[])</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/CommandLineParser.html" title="class in com.compomics.util.general">CommandLineParser</a></dt>
<dd>
<div class="block"><a name="./com/compomics/util/general/optionConstr"></a> This constructor requests the command-line
 String[] as well as a String[] with a list of options which in turn take
 a parameter.</div>
</dd>
<dt><a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli"><span class="typeNameLink">CommandLineUtils</span></a> - Class in <a href="com/compomics/software/cli/package-summary.html">com.compomics.software.cli</a></dt>
<dd>
<div class="block">This class groups some convenience methods for the use of compomics tools in
 command line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandLineUtils.html#CommandLineUtils--">CommandLineUtils()</a></span> - Constructor for class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli">CommandLineUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli"><span class="typeNameLink">CommandParameter</span></a> - Class in <a href="com/compomics/software/cli/package-summary.html">com.compomics.software.cli</a></dt>
<dd>
<div class="block">Convenience methods for the validation of command line parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#CommandParameter--">CommandParameter()</a></span> - Constructor for class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#comment">comment</a></span> - Static variable in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>
<div class="block">Character to start the comment lines.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#commit--">commit()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Committing all changes into the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#comparatorsMap">comparatorsMap</a></span> - Variable in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Map of the comparators to use.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/CompareVersionNumbers.html#compare-java.lang.String-java.lang.String-">compare(String, String)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/CompareVersionNumbers.html" title="class in com.compomics.software.autoupdater">CompareVersionNumbers</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/CompareVersionNumbers.html#compare-java.lang.String-java.lang.String-">compare(String, String)</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/CompareVersionNumbers.html" title="class in com.compomics.util">CompareVersionNumbers</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#compare-int-int-">compare(int, int)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#compareRowsByColumn-int-int-int-">compareRowsByColumn(int, int, int)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.HuffmanNode.html#compareTo-com.compomics.util.experiment.identification.protein_inference.fm_index.WaveletTree.HuffmanNode-">compareTo(WaveletTree.HuffmanNode)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.HuffmanNode.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree.HuffmanNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#compareTo-java.lang.Object-">compareTo(Object)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Method that compares an IsotopicDistributionSpectrum</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#compareTo-java.lang.Object-">compareTo(Object)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">Compares this object with the specified object for order.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#compareTo-java.lang.Object-">compareTo(Object)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">Compares this object with the specified object for order.</div>
</dd>
<dt><a href="com/compomics/software/autoupdater/CompareVersionNumbers.html" title="class in com.compomics.software.autoupdater"><span class="typeNameLink">CompareVersionNumbers</span></a> - Class in <a href="com/compomics/software/autoupdater/package-summary.html">com.compomics.software.autoupdater</a></dt>
<dd>
<div class="block">Class for comparing tool version numbers.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/CompareVersionNumbers.html#CompareVersionNumbers--">CompareVersionNumbers()</a></span> - Constructor for class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/CompareVersionNumbers.html" title="class in com.compomics.software.autoupdater">CompareVersionNumbers</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/CompareVersionNumbers.html" title="class in com.compomics.util"><span class="typeNameLink">CompareVersionNumbers</span></a> - Class in <a href="com/compomics/util/package-summary.html">com.compomics.util</a></dt>
<dd>
<div class="block">Comparator for tool version numbers.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/CompareVersionNumbers.html#CompareVersionNumbers--">CompareVersionNumbers()</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/CompareVersionNumbers.html" title="class in com.compomics.util">CompareVersionNumbers</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">ComplementaryIonAminoAcidFeature</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">Feature based on the amino acid properties of the complementary ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonAminoAcidFeature.html#ComplementaryIonAminoAcidFeature-com.compomics.util.experiment.biology.aminoacids.AminoAcid.Property-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.AASequenceFeature.Function-">ComplementaryIonAminoAcidFeature(AminoAcid.Property, AASequenceFeature.Function)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ComplementaryIonAminoAcidFeature</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">ComplementaryIonFeature</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">Feature based on a property of the complementary ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonFeature.html#ComplementaryIonFeature-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.IonFeature.Property-">ComplementaryIonFeature(IonFeature.Property)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ComplementaryIonFeature</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.IonSeries.html#complementarySeries">complementarySeries</a></span> - Variable in enum com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.IonSeries.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">SimplePeptideAnnotator.IonSeries</a></dt>
<dd>
<div class="block">The complementary ion index as listed in the PeptideFragmentIon
 class.</div>
</dd>
<dt><a href="com/compomics/util/enumeration/CompomicsTools.html" title="enum in com.compomics.util.enumeration"><span class="typeNameLink">CompomicsTools</span></a> - Enum in <a href="com/compomics/util/enumeration/package-summary.html">com.compomics.util.enumeration</a></dt>
<dd>
<div class="block">This enum types all computational omics tools and libraries that make use of the utilities library.</div>
</dd>
<dt><a href="com/compomics/software/CompomicsWrapper.html" title="class in com.compomics.software"><span class="typeNameLink">CompomicsWrapper</span></a> - Class in <a href="com/compomics/software/package-summary.html">com.compomics.software</a></dt>
<dd>
<div class="block">A general wrapper for compomics tools.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/CompomicsWrapper.html#CompomicsWrapper--">CompomicsWrapper()</a></span> - Constructor for class com.compomics.software.<a href="com/compomics/software/CompomicsWrapper.html" title="class in com.compomics.software">CompomicsWrapper</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#computeInverseMassValue-double-double-">computeInverseMassValue(double, double)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Compute the inverse mass value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#computeMappingRanges-double-">computeMappingRanges(double)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Compute mapping ranges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#computeMassValue-double-double-">computeMassValue(double, double)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Compute the mass value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html#computeMolecularWeight-java.lang.String-">computeMolecularWeight(String)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html" title="class in com.compomics.util.experiment.identification.utils">ProteinUtils</a></dt>
<dd>
<div class="block">Returns the protein's molecular weight.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandLineUtils.html#concatenate-java.util.ArrayList-">concatenate(ArrayList&lt;String&gt;)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli">CommandLineUtils</a></dt>
<dd>
<div class="block">Returns the list of arguments as space separated string for the command
 line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandLineUtils.html#concatenate-java.lang.String:A-">concatenate(String[])</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli">CommandLineUtils</a></dt>
<dd>
<div class="block">Returns the list of arguments as space separated string for the command
 line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#condition">condition</a></span> - Variable in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Description of the condition to meet to pass the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#CONFIDENT">CONFIDENT</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Index for a confident assignment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#configurationFileName">configurationFileName</a></span> - Static variable in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Default name for the path configuration file.</div>
</dd>
<dt><a href="com/compomics/util/interfaces/Connectable.html" title="interface in com.compomics.util.interfaces"><span class="typeNameLink">Connectable</span></a> - Interface in <a href="com/compomics/util/interfaces/package-summary.html">com.compomics.util.interfaces</a></dt>
<dd>
<div class="block">This interface describes the behaviour for classes that can accept an
 incoming DB connection.</div>
</dd>
<dt><a href="com/compomics/util/gui/utils/ConnectionDialog.html" title="class in com.compomics.util.gui.utils"><span class="typeNameLink">ConnectionDialog</span></a> - Class in <a href="com/compomics/util/gui/utils/package-summary.html">com.compomics.util.gui.utils</a></dt>
<dd>
<div class="block">This class implements a dialog to gather all information concerning a DB connection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/ConnectionDialog.html#ConnectionDialog-javax.swing.JFrame-com.compomics.util.interfaces.Connectable-java.lang.String-java.lang.String-">ConnectionDialog(JFrame, Connectable, String, String)</a></span> - Constructor for class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/ConnectionDialog.html" title="class in com.compomics.util.gui.utils">ConnectionDialog</a></dt>
<dd>
<div class="block">This constructor takes as arguments the parent JFrame and
 a title for the dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/ConnectionDialog.html#ConnectionDialog-javax.swing.JFrame-com.compomics.util.interfaces.Connectable-java.lang.String-java.util.Properties-">ConnectionDialog(JFrame, Connectable, String, Properties)</a></span> - Constructor for class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/ConnectionDialog.html" title="class in com.compomics.util.gui.utils">ConnectionDialog</a></dt>
<dd>
<div class="block">This constructor takes as arguments the parent JFrame and
 a title for the dialog.</div>
</dd>
<dt><a href="com/compomics/util/experiment/Constants.html" title="class in com.compomics.util.experiment"><span class="typeNameLink">Constants</span></a> - Class in <a href="com/compomics/util/experiment/package-summary.html">com.compomics.util.experiment</a></dt>
<dd>
<div class="block">Convenience class listing constants used throughout the library.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/Constants.html#Constants--">Constants()</a></span> - Constructor for class com.compomics.util.experiment.<a href="com/compomics/util/experiment/Constants.html" title="class in com.compomics.util.experiment">Constants</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/SwingWorker.html#construct--">construct()</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/SwingWorker.html" title="class in com.compomics.util.sun">SwingWorker</a></dt>
<dd>
<div class="block">Compute the value to be returned by the <code>get</code> method.</div>
</dd>
<dt><a href="com/compomics/util/db/components/Constructors.html" title="class in com.compomics.util.db.components"><span class="typeNameLink">Constructors</span></a> - Class in <a href="com/compomics/util/db/components/package-summary.html">com.compomics.util.db.components</a></dt>
<dd>
<div class="block">This class will generate the code for a default and full constructor for a DBAccessor, 
 defined by the metadata passed in via the constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/Constructors.html#Constructors-com.compomics.util.db.DBMetaData-">Constructors(DBMetaData)</a></span> - Constructor for class com.compomics.util.db.components.<a href="com/compomics/util/db/components/Constructors.html" title="class in com.compomics.util.db.components">Constructors</a></dt>
<dd>
<div class="block">This constructor takes care of generating the code that represents the
 constructor for a DBAccessor for the given metadata.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/Contact.html" title="class in com.compomics.util.pride.prideobjects"><span class="typeNameLink">Contact</span></a> - Class in <a href="com/compomics/util/pride/prideobjects/package-summary.html">com.compomics.util.pride.prideobjects</a></dt>
<dd>
<div class="block">An object for storing Contact details.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Contact.html#Contact-java.lang.String-java.lang.String-java.lang.String-">Contact(String, String, String)</a></span> - Constructor for class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Contact.html" title="class in com.compomics.util.pride.prideobjects">Contact</a></dt>
<dd>
<div class="block">Create a new Contact object.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/ContactGroup.html" title="class in com.compomics.util.pride.prideobjects"><span class="typeNameLink">ContactGroup</span></a> - Class in <a href="com/compomics/util/pride/prideobjects/package-summary.html">com.compomics.util.pride.prideobjects</a></dt>
<dd>
<div class="block">An object for storing ContactGroup details.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ContactGroup.html#ContactGroup-java.util.ArrayList-java.lang.String-">ContactGroup(ArrayList&lt;Contact&gt;, String)</a></span> - Constructor for class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ContactGroup.html" title="class in com.compomics.util.pride.prideobjects">ContactGroup</a></dt>
<dd>
<div class="block">Create a new ContactGroup object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#contains-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">contains(String, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether the pattern contains a subsequence of amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#contains-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">contains(AminoAcidPattern, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether the pattern contains a subsequence of amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#contains-long-">contains(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Checks if database contains a certain object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#contains-java.lang.String-">contains(String)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether a protein was found in this protein
 match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#contains-java.lang.String-">contains(String)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Indicates whether the modification is contained in the profile, looking
 into all modifications (fixed, variable and refinement)</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#contains-java.lang.String-">contains(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method reports whether a certain residu (or fixed sequence
 String) is found in the current sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#containsForbiddenCharacter-java.lang.String-">containsForbiddenCharacter(String)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Indicates whether a string contains characters forbidden in file names.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html#containsLoss-java.lang.String-">containsLoss(String)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">NeutralLossesMap</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether a loss is implemented in the
 mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#containsModification-java.lang.String-">containsModification(String)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the Modification is loaded in the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#containsModification-java.lang.String-">containsModification(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Indicates whether a modification has been already scored.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#containsSpectrum-java.lang.String-">containsSpectrum(String)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the spectrum title is implemented in
 this index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#controlSocket">controlSocket</a></span> - Variable in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Socket for communicating commands with the server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#convertBooleanToInteger-java.lang.Boolean-">convertBooleanToInteger(Boolean)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Converts a boolean value to the corresponding integer value, 0 for false
 and 1 for true.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/converters/GenericFastaConverter.html#convertFile-java.io.File-java.io.File-com.compomics.util.waiting.WaitingHandler-">convertFile(File, File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.biology.protein.converters.<a href="com/compomics/util/experiment/io/biology/protein/converters/GenericFastaConverter.html" title="class in com.compomics.util.experiment.io.biology.protein.converters">GenericFastaConverter</a></dt>
<dd>
<div class="block">Appends decoy sequences to the provided FASTA file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#convertPridePtm-java.lang.String-com.compomics.util.parameters.identification.search.ModificationParameters-java.util.ArrayList-boolean-">convertPridePtm(String, ModificationParameters, ArrayList&lt;String&gt;, boolean)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Tries to convert a PRIDE Modification to utilities Modification name, and add it to the
 modification profile.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/ColumnTypeConverter.html#convertTypes-int:A-int:A-">convertTypes(int[], int[])</a></span> - Static method in class com.compomics.util.db.<a href="com/compomics/util/db/ColumnTypeConverter.html" title="class in com.compomics.util.db">ColumnTypeConverter</a></dt>
<dd>
<div class="block">This method converts coded SQL columntypes into
 the corresponding Java objects (whose names are given
 int the String[] elements).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#copyFile-java.io.File-java.io.File-">copyFile(File, File)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Copy the content of a file to another.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#copyFile-java.io.File-java.io.File-boolean-">copyFile(File, File, boolean)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Copy the content of one file to another.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#createDefaultGeneMappingFiles-java.io.File-java.io.File-java.io.File-java.io.File-boolean-">createDefaultGeneMappingFiles(File, File, File, File, boolean)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Copies the given gene mapping files to the gene mappings folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/modification/ModificationTable.html#createDefaultTableHeader--">createDefaultTableHeader()</a></span> - Method in class com.compomics.util.gui.modification.<a href="com/compomics/util/gui/modification/ModificationTable.html" title="class in com.compomics.util.gui.modification">ModificationTable</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/FragmentIonTable.html#createDefaultTableHeader--">createDefaultTableHeader()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/FragmentIonTable.html" title="class in com.compomics.util.gui.spectrum">FragmentIonTable</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/FileDAO.html#createDesktopShortcut-com.compomics.software.autoupdater.MavenJarFile-java.lang.String-java.lang.String-boolean-">createDesktopShortcut(MavenJarFile, String, String, boolean)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/FileDAO.html" title="class in com.compomics.software.autoupdater">FileDAO</a></dt>
<dd>
<div class="block">Creates a new Desktop Shortcut to the Maven jar file, atm windows only.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/GUIFileDAO.html#createDesktopShortcut-com.compomics.software.autoupdater.MavenJarFile-java.lang.String-java.lang.String-boolean-">createDesktopShortcut(MavenJarFile, String, String, boolean)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/GUIFileDAO.html" title="class in com.compomics.software.autoupdater">GUIFileDAO</a></dt>
<dd>
<div class="block">Creates a new Desktop Shortcut to the Maven jar file, atm windows only.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/HeadlessFileDAO.html#createDesktopShortcut-com.compomics.software.autoupdater.MavenJarFile-java.lang.String-java.lang.String-boolean-">createDesktopShortcut(MavenJarFile, String, String, boolean)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/HeadlessFileDAO.html" title="class in com.compomics.software.autoupdater">HeadlessFileDAO</a></dt>
<dd>
<div class="block">Creates a new Desktop Shortcut to the Maven jar file, atm windows only.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#createImageIcon-java.lang.String-">createImageIcon(String)</a></span> - Static method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>
<div class="block">Returns an ImageIcon, or null if the path was invalid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#createIntermediateScoreMap--">createIntermediateScoreMap()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Instantiates the intermediate scores map if null.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#createLessTable--">createLessTable()</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Create the less table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIParams.html#createOptionsCLI-org.apache.commons.cli.Options-">createOptionsCLI(Options)</a></span> - Static method in enum com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIParams.html" title="enum in com.compomics.cli.enzymes">EnzymesCLIParams</a></dt>
<dd>
<div class="block">Creates the options for the command line interface based on the possible
 values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/fasta/FastaParametersCLIParams.html#createOptionsCLI-org.apache.commons.cli.Options-">createOptionsCLI(Options)</a></span> - Static method in enum com.compomics.cli.fasta.<a href="com/compomics/cli/fasta/FastaParametersCLIParams.html" title="enum in com.compomics.cli.fasta">FastaParametersCLIParams</a></dt>
<dd>
<div class="block">Creates the options for the command line interface based on the possible
 values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html#createOptionsCLI-org.apache.commons.cli.Options-">createOptionsCLI(Options)</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html" title="class in com.compomics.cli.identification_parameters">AbstractIdentificationParametersCli</a></dt>
<dd>
<div class="block">Provides the options left to the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLI.html#createOptionsCLI-org.apache.commons.cli.Options-">createOptionsCLI(Options)</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLI.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersCLI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html#createOptionsCLI-org.apache.commons.cli.Options-">createOptionsCLI(Options)</a></span> - Static method in enum com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html" title="enum in com.compomics.cli.identification_parameters">IdentificationParametersCLIParams</a></dt>
<dd>
<div class="block">Creates the options for the command line interface based on the possible
 values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIParams.html#createOptionsCLI-org.apache.commons.cli.Options-">createOptionsCLI(Options)</a></span> - Static method in enum com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIParams.html" title="enum in com.compomics.cli.modifications">ModificationsCLIParams</a></dt>
<dd>
<div class="block">Creates the options for the command line interface based on the possible
 values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html#createPeptideFilter--">createPeptideFilter()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationQCParametersDialog</a></dt>
<dd>
<div class="block">Creates a new peptide filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html#createProteinFilter--">createProteinFilter()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationQCParametersDialog</a></dt>
<dd>
<div class="block">Creates a new protein filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html#createPsmFilter--">createPsmFilter()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationQCParametersDialog</a></dt>
<dd>
<div class="block">Creates a new PSM filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#createWaveletTreeHuffman-byte:A-com.compomics.util.waiting.WaitingHandler-com.compomics.util.experiment.identification.protein_inference.fm_index.WaveletTree.HuffmanNode-int-boolean-">createWaveletTreeHuffman(byte[], WaitingHandler, WaveletTree.HuffmanNode, int, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Create wavelet tree huffman.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#CTERM">CTERM</a></span> - Static variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#cTermCorrection">cTermCorrection</a></span> - Variable in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">The mass to add to the C-terminal gap so that is corresponds to a peptide
 fragment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#cTermCorrection">cTermCorrection</a></span> - Variable in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>
<div class="block">The mass to add to the C-terminal gap so that is corresponds to a peptide
 fragment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#CTERMINAL">CTERMINAL</a></span> - Static variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">The code for C-terminal position in the resultant peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#CTERMINUS">CTERMINUS</a></span> - Static variable in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">The residue code for the C-terminus.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#CTERMTRUNC">CTERMTRUNC</a></span> - Static variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">The code for an C-terminal truncation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#CURRENT_VERSION">CURRENT_VERSION</a></span> - Static variable in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Currently supported version number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#currentGraphicsPanelType">currentGraphicsPanelType</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Sets the current GraphicsPanel type, default to centroid spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#currentPadding">currentPadding</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The current padding (distance between the axes and the border of the
 panel).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html#currentVersion">currentVersion</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">FeaturesMap</a></dt>
<dd>
<div class="block">Current version of the features.</div>
</dd>
<dt><a href="com/compomics/util/pride/CvTerm.html" title="class in com.compomics.util.pride"><span class="typeNameLink">CvTerm</span></a> - Class in <a href="com/compomics/util/pride/package-summary.html">com.compomics.util.pride</a></dt>
<dd>
<div class="block">A simple CvTerm object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/CvTerm.html#CvTerm-java.lang.String-java.lang.String-java.lang.String-java.lang.String-">CvTerm(String, String, String, String)</a></span> - Constructor for class com.compomics.util.pride.<a href="com/compomics/util/pride/CvTerm.html" title="class in com.compomics.util.pride">CvTerm</a></dt>
<dd>
<div class="block">Create a new CV term.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#cwd-java.lang.String-">cwd(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Change working directory to a specific directory on a remote FTP server.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Cysteine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Cysteine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html#Cysteine--">Cysteine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Cysteine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#CYSTEINE">CYSTEINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Cysteine immonium ion.</div>
</dd>
</dl>
<a name="I:D">
<!--   -->
</a>
<h2 class="title">D</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#D">D</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html" title="class in com.compomics.util.pdbfinder.das.readers"><span class="typeNameLink">DasAlignment</span></a> - Class in <a href="com/compomics/util/pdbfinder/das/readers/package-summary.html">com.compomics.util.pdbfinder.das.readers</a></dt>
<dd>
<div class="block">DasAlignment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html#DasAlignment-java.lang.String-">DasAlignment(String)</a></span> - Constructor for class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAlignment</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/pdbfinder/das/readers/DasAnnotationServerAlingmentReader.html" title="class in com.compomics.util.pdbfinder.das.readers"><span class="typeNameLink">DasAnnotationServerAlingmentReader</span></a> - Class in <a href="com/compomics/util/pdbfinder/das/readers/package-summary.html">com.compomics.util.pdbfinder.das.readers</a></dt>
<dd>
<div class="block">DasAnnotationServerAlingmentReader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAnnotationServerAlingmentReader.html#DasAnnotationServerAlingmentReader-java.lang.String-">DasAnnotationServerAlingmentReader(String)</a></span> - Constructor for class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAnnotationServerAlingmentReader.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAnnotationServerAlingmentReader</a></dt>
<dd>
<div class="block">Creates a new reader for a XML string.</div>
</dd>
<dt><a href="com/compomics/util/pdbfinder/das/readers/DasAnnotationServerResultReader.html" title="class in com.compomics.util.pdbfinder.das.readers"><span class="typeNameLink">DasAnnotationServerResultReader</span></a> - Class in <a href="com/compomics/util/pdbfinder/das/readers/package-summary.html">com.compomics.util.pdbfinder.das.readers</a></dt>
<dd>
<div class="block">DasAnnotationServerResultReader</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAnnotationServerResultReader.html#DasAnnotationServerResultReader-java.lang.String-">DasAnnotationServerResultReader(String)</a></span> - Constructor for class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAnnotationServerResultReader.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAnnotationServerResultReader</a></dt>
<dd>
<div class="block">Creates a new reader for an XML string.</div>
</dd>
<dt><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers"><span class="typeNameLink">DasFeature</span></a> - Class in <a href="com/compomics/util/pdbfinder/das/readers/package-summary.html">com.compomics.util.pdbfinder.das.readers</a></dt>
<dd>
<div class="block">DasFeature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#DasFeature-java.lang.String-">DasFeature(String)</a></span> - Constructor for class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#dat">dat</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#DAT">DAT</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html#databaseType">databaseType</a></span> - Variable in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaSummary</a></dt>
<dd>
<div class="block">The database type occurrence in the FASTA file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#dataMissingAtRow-int-">dataMissingAtRow(int)</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Calling this method indicates that data is missing at the given row.</div>
</dd>
<dt><a href="com/compomics/util/math/roc/DataRoc.html" title="class in com.compomics.util.math.roc"><span class="typeNameLink">DataRoc</span></a> - Class in <a href="com/compomics/util/math/roc/package-summary.html">com.compomics.util.math.roc</a></dt>
<dd>
<div class="block">This class can be used to draw roc curves from experimental data.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/roc/DataRoc.html#DataRoc-java.util.ArrayList-java.util.ArrayList-com.compomics.util.math.roc.DataRoc.RocInterpolation-">DataRoc(ArrayList&lt;Double&gt;, ArrayList&lt;Double&gt;, DataRoc.RocInterpolation)</a></span> - Constructor for class com.compomics.util.math.roc.<a href="com/compomics/util/math/roc/DataRoc.html" title="class in com.compomics.util.math.roc">DataRoc</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/math/roc/DataRoc.RocInterpolation.html" title="enum in com.compomics.util.math.roc"><span class="typeNameLink">DataRoc.RocInterpolation</span></a> - Enum in <a href="com/compomics/util/math/roc/package-summary.html">com.compomics.util.math.roc</a></dt>
<dd>
<div class="block">Enum listing the possible ways of interpolating points on the ROC.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#dataSetCounter">dataSetCounter</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The number of datasets currently displayed in the panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#dataSetCounterMirroredSpectra">dataSetCounterMirroredSpectra</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The number of mirrored datasets currently displayed in the panel.</div>
</dd>
<dt><a href="com/compomics/util/gui/utils/DateChooser.html" title="class in com.compomics.util.gui.utils"><span class="typeNameLink">DateChooser</span></a> - Class in <a href="com/compomics/util/gui/utils/package-summary.html">com.compomics.util.gui.utils</a></dt>
<dd>
<div class="block">This class provides a simple date chooser.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/DateChooser.html#DateChooser-javax.swing.JFrame-">DateChooser(JFrame)</a></span> - Constructor for class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/DateChooser.html" title="class in com.compomics.util.gui.utils">DateChooser</a></dt>
<dd>
<div class="block">Creates a new DataChooser object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/DateChooser.html#DateChooser-javax.swing.JFrame-java.util.Calendar-">DateChooser(JFrame, Calendar)</a></span> - Constructor for class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/DateChooser.html" title="class in com.compomics.util.gui.utils">DateChooser</a></dt>
<dd>
<div class="block">This constructor takes the parent JFRame for this dialog,
 as well as a reference parameter for the selected date.</div>
</dd>
<dt><a href="com/compomics/util/io/file/filefilters/DatFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">DatFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.dat files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/DatFileFilter.html#DatFileFilter--">DatFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/DatFileFilter.html" title="class in com.compomics.util.io.file.filefilters">DatFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/db/DBAccessor.html" title="class in com.compomics.util.db"><span class="typeNameLink">DBAccessor</span></a> - Class in <a href="com/compomics/util/db/package-summary.html">com.compomics.util.db</a></dt>
<dd>
<div class="block">This class represents an accessor class' sourcecode as generated by the DBAccessorGenerator
 class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBAccessor.html#DBAccessor-com.compomics.util.db.DBMetaData-java.lang.String-boolean-">DBAccessor(DBMetaData, String, boolean)</a></span> - Constructor for class com.compomics.util.db.<a href="com/compomics/util/db/DBAccessor.html" title="class in com.compomics.util.db">DBAccessor</a></dt>
<dd>
<div class="block">This constructor creates all code for the accessor,
 based on the specified metadata.</div>
</dd>
<dt><a href="com/compomics/util/db/DBAccessorGenerator.html" title="class in com.compomics.util.db"><span class="typeNameLink">DBAccessorGenerator</span></a> - Class in <a href="com/compomics/util/db/package-summary.html">com.compomics.util.db</a></dt>
<dd>
<div class="block">This class generates an Object that can be used for access to a
 given table in a JDBC accessible RDBMS.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBAccessorGenerator.html#DBAccessorGenerator--">DBAccessorGenerator()</a></span> - Constructor for class com.compomics.util.db.<a href="com/compomics/util/db/DBAccessorGenerator.html" title="class in com.compomics.util.db">DBAccessorGenerator</a></dt>
<dd>
<div class="block">Default constructor.</div>
</dd>
<dt><a href="com/compomics/util/db/interfaces/DBElement.html" title="interface in com.compomics.util.db.interfaces"><span class="typeNameLink">DBElement</span></a> - Interface in <a href="com/compomics/util/db/interfaces/package-summary.html">com.compomics.util.db.interfaces</a></dt>
<dd>
<div class="block">This interface indicates that the implementing class can be stored, manipulated and deleted in a
 permanent storage (typically a RDBMS).</div>
</dd>
<dt><a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db"><span class="typeNameLink">DBMetaData</span></a> - Class in <a href="com/compomics/util/db/package-summary.html">com.compomics.util.db</a></dt>
<dd>
<div class="block">This class wraps the metadata retrieved from a table on the DB.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#DBMetaData-java.lang.String-java.lang.String:A-int:A-int:A-">DBMetaData(String, String[], int[], int[])</a></span> - Constructor for class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">Constructor that allows the specification of table name,
 column names and column types.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#DBMetaData-java.lang.String-java.lang.String:A-int:A-int:A-java.lang.String:A-">DBMetaData(String, String[], int[], int[], String[])</a></span> - Constructor for class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">Constructor that allows the specification of table name,
 column names and column types and the primary key columns..</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#DBMetaData-java.lang.String-java.util.Vector-java.util.Vector-java.util.Vector-">DBMetaData(String, Vector, Vector, Vector)</a></span> - Constructor for class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">Constructor that allows the specification of table name,
 column names and column types.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#DBMetaData-java.lang.String-java.util.Vector-java.util.Vector-java.util.Vector-java.util.Vector-">DBMetaData(String, Vector, Vector, Vector, Vector)</a></span> - Constructor for class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">Constructor that allows the specification of table name,
 column names, column types, column sizes and the primary key columns.</div>
</dd>
<dt><a href="com/compomics/util/db/object/DbObject.html" title="class in com.compomics.util.db.object"><span class="typeNameLink">DbObject</span></a> - Class in <a href="com/compomics/util/db/object/package-summary.html">com.compomics.util.db.object</a></dt>
<dd>
<div class="block">All classes that are stored in the backend need a unique identifier,
 all further classes inherit from this class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/DbObject.html#DbObject--">DbObject()</a></span> - Constructor for class com.compomics.util.db.object.<a href="com/compomics/util/db/object/DbObject.html" title="class in com.compomics.util.db.object">DbObject</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db"><span class="typeNameLink">DBResultSet</span></a> - Class in <a href="com/compomics/util/db/package-summary.html">com.compomics.util.db</a></dt>
<dd>
<div class="block">This class wraps a DB resultset in an 'offline' object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBResultSet.html#DBResultSet--">DBResultSet()</a></span> - Constructor for class com.compomics.util.db.<a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db">DBResultSet</a></dt>
<dd>
<div class="block">Default constructor, just creates an empty resultset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBResultSet.html#DBResultSet-java.sql.ResultSet-">DBResultSet(ResultSet)</a></span> - Constructor for class com.compomics.util.db.<a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db">DBResultSet</a></dt>
<dd>
<div class="block">This constructor takes a ResultSet from which the data
 is read.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBResultSet.html#DBResultSet-java.sql.ResultSet-boolean-">DBResultSet(ResultSet, boolean)</a></span> - Constructor for class com.compomics.util.db.<a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db">DBResultSet</a></dt>
<dd>
<div class="block">This constructor takes a ResultSet from which the data
 is read.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/converters/DecoyConverter.html" title="class in com.compomics.util.experiment.io.biology.protein.converters"><span class="typeNameLink">DecoyConverter</span></a> - Class in <a href="com/compomics/util/experiment/io/biology/protein/converters/package-summary.html">com.compomics.util.experiment.io.biology.protein.converters</a></dt>
<dd>
<div class="block">This class appends decoy sequences to the given FASTA file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/converters/DecoyConverter.html#DecoyConverter--">DecoyConverter()</a></span> - Constructor for class com.compomics.util.experiment.io.biology.protein.converters.<a href="com/compomics/util/experiment/io/biology/protein/converters/DecoyConverter.html" title="class in com.compomics.util.experiment.io.biology.protein.converters">DecoyConverter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/converters/DecoyConverter.html#decoyFlag">decoyFlag</a></span> - Static variable in class com.compomics.util.experiment.io.biology.protein.converters.<a href="com/compomics/util/experiment/io/biology/protein/converters/DecoyConverter.html" title="class in com.compomics.util.experiment.io.biology.protein.converters">DecoyConverter</a></dt>
<dd>
<div class="block">The flag to append to the accessions of decoy proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#decoyFlags">decoyFlags</a></span> - Static variable in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">The decoy flags used to infer the FASTA parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#decreaseIndent--">decreaseIndent()</a></span> - Method in class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Decreases the indent counter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#decreaseRWCounter--">decreaseRWCounter()</a></span> - Static method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Function for decreasing the counter of processes accessing objects from
 the db.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#decreseDepth--">decreseDepth()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Notifies the writer that data of a lower hierarchical depth will be
 written, e.g.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#decreseDepth--">decreseDepth()</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/TextWriter.html#decreseDepth--">decreseDepth()</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/TextWriter.html" title="class in com.compomics.util.io.export.writers">TextWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html#DEFAULT_NON_VISIBLE_MARKER_ALPHA">DEFAULT_NON_VISIBLE_MARKER_ALPHA</a></span> - Static variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorBubblePlot</a></dt>
<dd>
<div class="block">The default non-visible alpha level.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html#DEFAULT_VISIBLE_MARKER_ALPHA">DEFAULT_VISIBLE_MARKER_ALPHA</a></span> - Static variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorBubblePlot</a></dt>
<dd>
<div class="block">The default visible alpha level.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#defaultModsSorted">defaultModsSorted</a></span> - Variable in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Set to true if the default mods are sorted alphabetically.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#defaultMutationMatrices">defaultMutationMatrices</a></span> - Static variable in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Returns the implemented default substitution matrices.</div>
</dd>
<dt><a href="com/compomics/util/general/DefaultOutputLoggerImplementation.html" title="class in com.compomics.util.general"><span class="typeNameLink">DefaultOutputLoggerImplementation</span></a> - Class in <a href="com/compomics/util/general/package-summary.html">com.compomics.util.general</a></dt>
<dd>
<div class="block">This class implements a default logger (for debugging purposes).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/DefaultOutputLoggerImplementation.html#DefaultOutputLoggerImplementation--">DefaultOutputLoggerImplementation()</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/DefaultOutputLoggerImplementation.html" title="class in com.compomics.util.general">DefaultOutputLoggerImplementation</a></dt>
<dd>
<div class="block">Default constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/DefaultOutputLoggerImplementation.html#DefaultOutputLoggerImplementation-java.io.PrintStream-java.io.PrintStream-">DefaultOutputLoggerImplementation(PrintStream, PrintStream)</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/DefaultOutputLoggerImplementation.html" title="class in com.compomics.util.general">DefaultOutputLoggerImplementation</a></dt>
<dd>
<div class="block">Constructor that takes two PrintStreams to output to.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#defaultPath">defaultPath</a></span> - Static variable in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Replacement for the path when not available.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#defaultPrecursorCharges">defaultPrecursorCharges</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">The precursor charges to inspect by default.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/DefaultSpectrumAnnotation.html" title="class in com.compomics.util.gui.spectrum"><span class="typeNameLink">DefaultSpectrumAnnotation</span></a> - Class in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">This class provides a default implementation of the SpectrumAnnotation interface.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/DefaultSpectrumAnnotation.html#DefaultSpectrumAnnotation-double-double-java.awt.Color-java.lang.String-">DefaultSpectrumAnnotation(double, double, Color, String)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/DefaultSpectrumAnnotation.html" title="class in com.compomics.util.gui.spectrum">DefaultSpectrumAnnotation</a></dt>
<dd>
<div class="block">Constructor creating a DefaultSpectrumAnnotation object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html#defaultStringMatching">defaultStringMatching</a></span> - Static variable in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a></dt>
<dd>
<div class="block">Default string matching.</div>
</dd>
<dt><a href="com/compomics/util/io/export/styles/DefaultStyle.html" title="class in com.compomics.util.io.export.styles"><span class="typeNameLink">DefaultStyle</span></a> - Class in <a href="com/compomics/util/io/export/styles/package-summary.html">com.compomics.util.io.export.styles</a></dt>
<dd>
<div class="block">Default style for excel sheets.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/interfaces/Deleteable.html#delete-java.sql.Connection-">delete(Connection)</a></span> - Method in interface com.compomics.util.db.interfaces.<a href="com/compomics/util/db/interfaces/Deleteable.html" title="interface in com.compomics.util.db.interfaces">Deleteable</a></dt>
<dd>
<div class="block">This method will physically delete the implemented object's data from the
 persistent store.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#delete-java.lang.String-">delete(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Delete the specified file from the ftp server file system.</div>
</dd>
<dt><a href="com/compomics/util/db/interfaces/Deleteable.html" title="interface in com.compomics.util.db.interfaces"><span class="typeNameLink">Deleteable</span></a> - Interface in <a href="com/compomics/util/db/interfaces/package-summary.html">com.compomics.util.db.interfaces</a></dt>
<dd>
<div class="block">This interface indicates that the implementing class can be deleted from permanent storage.</div>
</dd>
<dt><a href="com/compomics/util/db/components/DeleteableCode.html" title="class in com.compomics.util.db.components"><span class="typeNameLink">DeleteableCode</span></a> - Class in <a href="com/compomics/util/db/components/package-summary.html">com.compomics.util.db.components</a></dt>
<dd>
<div class="block">This class genereates the code to make a DBAccessor class an implementation of the Deleteable interface.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/DeleteableCode.html#DeleteableCode-com.compomics.util.db.DBMetaData-">DeleteableCode(DBMetaData)</a></span> - Constructor for class com.compomics.util.db.components.<a href="com/compomics/util/db/components/DeleteableCode.html" title="class in com.compomics.util.db.components">DeleteableCode</a></dt>
<dd>
<div class="block">This constructor will generate the code to make the DBAccessor an implementation
 of the Deleteable interface, based on the specified metadata.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#deleteContactGroup-com.compomics.util.pride.prideobjects.ContactGroup-">deleteContactGroup(ContactGroup)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Delete the given contact group.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#deleteDir-java.io.File-">deleteDir(File)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Deletes all files and subdirectories under dir and dir itself.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#deleteInstrument-com.compomics.util.pride.prideobjects.Instrument-">deleteInstrument(Instrument)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Delete the given instrument.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#deleteProtocol-com.compomics.util.pride.prideobjects.Protocol-">deleteProtocol(Protocol)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Delete the given protocol.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#deleteReferenceGroup-com.compomics.util.pride.prideobjects.ReferenceGroup-">deleteReferenceGroup(ReferenceGroup)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Delete the given reference group.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#deleteSample-com.compomics.util.pride.prideobjects.Sample-">deleteSample(Sample)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Delete the given sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/file_handling/TempFilesManager.html#deleteTempFolders--">deleteTempFolders()</a></span> - Static method in class com.compomics.util.gui.file_handling.<a href="com/compomics/util/gui/file_handling/TempFilesManager.html" title="class in com.compomics.util.gui.file_handling">TempFilesManager</a></dt>
<dd>
<div class="block">Deletes the temp folders created.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/variants/amino_acids/Deletion.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids"><span class="typeNameLink">Deletion</span></a> - Class in <a href="com/compomics/util/experiment/biology/variants/amino_acids/package-summary.html">com.compomics.util.experiment.biology.variants.amino_acids</a></dt>
<dd>
<div class="block">Class representing an amino acid deletion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/amino_acids/Deletion.html#Deletion-char-">Deletion(char)</a></span> - Constructor for class com.compomics.util.experiment.biology.variants.amino_acids.<a href="com/compomics/util/experiment/biology/variants/amino_acids/Deletion.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids">Deletion</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#deltaMassWindow">deltaMassWindow</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The size of the window to use when searching for matches in the known
 masses list when the user hovers over a second data point after clicking
 a previous data point.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#DescendingIntensityComparator">DescendingIntensityComparator</a></span> - Static variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>
<div class="block">This comparator compares two Peak instances on descending intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIParams.html#description">description</a></span> - Variable in enum com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIParams.html" title="enum in com.compomics.cli.enzymes">EnzymesCLIParams</a></dt>
<dd>
<div class="block">Explanation for the CLI parameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/fasta/FastaParametersCLIParams.html#description">description</a></span> - Variable in enum com.compomics.cli.fasta.<a href="com/compomics/cli/fasta/FastaParametersCLIParams.html" title="enum in com.compomics.cli.fasta">FastaParametersCLIParams</a></dt>
<dd>
<div class="block">Explanation for the CLI parameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html#description">description</a></span> - Variable in enum com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html" title="enum in com.compomics.cli.identification_parameters">IdentificationParametersCLIParams</a></dt>
<dd>
<div class="block">Explanation for the CLI parameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIParams.html#description">description</a></span> - Variable in enum com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIParams.html" title="enum in com.compomics.cli.modifications">ModificationsCLIParams</a></dt>
<dd>
<div class="block">Explanation for the CLI parameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationType.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationType.html" title="enum in com.compomics.util.experiment.biology.modifications">ModificationType</a></dt>
<dd>
<div class="block">The description of the type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItemComparator.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItemComparator.html" title="enum in com.compomics.util.experiment.filtering">FilterItemComparator</a></dt>
<dd>
<div class="block">The description to use.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">AssumptionFilterItem</a></dt>
<dd>
<div class="block">The description of the filtering item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PeptideFilterItem</a></dt>
<dd>
<div class="block">The description of the filtering item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">ProteinFilterItem</a></dt>
<dd>
<div class="block">The description of the filtering item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PsmFilterItem</a></dt>
<dd>
<div class="block">The description of the filtering item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#description">description</a></span> - Variable in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html" title="enum in com.compomics.util.experiment.identification.matches">IonMatch.MzErrorType</a></dt>
<dd>
<div class="block">The description of the error type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.Property.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">IonFeature.Property</a></dt>
<dd>
<div class="block">The description of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.Property.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ModificationFeature.Property</a></dt>
<dd>
<div class="block">The description of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.Property.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideFeature.Property</a></dt>
<dd>
<div class="block">The description of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/PeptideFragmentationModel.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.identification.peptide_fragmentation.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/PeptideFragmentationModel.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation">PeptideFragmentationModel</a></dt>
<dd>
<div class="block">The description of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html" title="enum in com.compomics.util.experiment.identification.psm_scoring">PsmScore</a></dt>
<dd>
<div class="block">Short description of the score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.IntensityThresholdType.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.IntensityThresholdType.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters.IntensityThresholdType</a></dt>
<dd>
<div class="block">The description of the threshold.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.TiesResolution.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.TiesResolution.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator.TiesResolution</a></dt>
<dd>
<div class="block">The description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html" title="enum in com.compomics.util.experiment.io.identification">MzIdentMLVersion</a></dt>
<dd>
<div class="block">The description of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsFormat</a></dt>
<dd>
<div class="block">A brief description of the format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html#description">description</a></span> - Variable in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">ProteoWizardFilter</a></dt>
<dd>
<div class="block">A brief description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFormat.html#description">description</a></span> - Variable in enum com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFormat.html" title="enum in com.compomics.util.io.export">ExportFormat</a></dt>
<dd>
<div class="block">Description of the format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html#description">description</a></span> - Variable in enum com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html" title="enum in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters.MatchingType</a></dt>
<dd>
<div class="block">The description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/peptide_shaker/ProjectType.html#description">description</a></span> - Variable in enum com.compomics.util.parameters.peptide_shaker.<a href="com/compomics/util/parameters/peptide_shaker/ProjectType.html" title="enum in com.compomics.util.parameters.peptide_shaker">ProjectType</a></dt>
<dd>
<div class="block">The description of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/searchgui/OutputParameters.html#description">description</a></span> - Variable in enum com.compomics.util.parameters.searchgui.<a href="com/compomics/util/parameters/searchgui/OutputParameters.html" title="enum in com.compomics.util.parameters.searchgui">OutputParameters</a></dt>
<dd>
<div class="block">The description of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/adapter/InterfaceAdapter.html#deserialize-com.google.gson.JsonElement-java.lang.reflect.Type-com.google.gson.JsonDeserializationContext-">deserialize(JsonElement, Type, JsonDeserializationContext)</a></span> - Method in class com.compomics.util.io.json.adapter.<a href="com/compomics/util/io/json/adapter/InterfaceAdapter.html" title="class in com.compomics.util.io.json.adapter">InterfaceAdapter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#determineColorOfPeak-java.lang.String-">determineColorOfPeak(String)</a></span> - Static method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">it is advised to use methods based on the ion type rather
 than on the peak label</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#determineDefaultFragmentIonColor-com.compomics.util.experiment.biology.ions.Ion-boolean-">determineDefaultFragmentIonColor(Ion, boolean)</a></span> - Static method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Returns the peak color to be used for the given peak label.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#determineFragmentIonColor-com.compomics.util.experiment.biology.ions.Ion-boolean-">determineFragmentIonColor(Ion, boolean)</a></span> - Static method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Returns the peak color to be used for the given peak label according to
 the color map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#determineFragmentIonColor-java.lang.String-">determineFragmentIonColor(String)</a></span> - Static method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">use the method based on the Ion class instead</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#DEUTEROACETYLATED_MASS">DEUTEROACETYLATED_MASS</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>
<div class="block">Final index for deuteroacetylated masses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#DEUTEROMETHYLATED_MASS">DEUTEROMETHYLATED_MASS</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>
<div class="block">Final index for deuteromethylated masses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#DI_METHYL_R_112">DI_METHYL_R_112</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for di-methylation of R (PMID: 16335983).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#DI_METHYL_R_115">DI_METHYL_R_115</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for di-methylation of R (PMID: 16335983).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#DI_METHYL_R_157">DI_METHYL_R_157</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for di-methylation of R (PMID: 16335983).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#digest-java.lang.String-int-java.lang.Integer-java.lang.Integer-">digest(String, int, Integer, Integer)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Digests a protein sequence in a list of expected peptide sequences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#digest-java.lang.String-int-java.lang.Double-java.lang.Double-">digest(String, int, Double, Double)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Digests a protein sequence in a list of expected peptide sequences.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search"><span class="typeNameLink">DigestionParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/search/package-summary.html">com.compomics.util.parameters.identification.search</a></dt>
<dd>
<div class="block">This class groups the parameters for the digestion of proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#DigestionParameters--">DigestionParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Constructor for empty parameters.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html" title="enum in com.compomics.util.parameters.identification.search"><span class="typeNameLink">DigestionParameters.CleavageParameter</span></a> - Enum in <a href="com/compomics/util/parameters/identification/search/package-summary.html">com.compomics.util.parameters.identification.search</a></dt>
<dd>
<div class="block">Enum for the different types of digestion.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html" title="enum in com.compomics.util.parameters.identification.search"><span class="typeNameLink">DigestionParameters.Specificity</span></a> - Enum in <a href="com/compomics/util/parameters/identification/search/package-summary.html">com.compomics.util.parameters.identification.search</a></dt>
<dd>
<div class="block">Enum for the different types of enzyme specificity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#dir--">dir()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Folder-list files on a remote FTP server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#direcTag">direcTag</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The DirecTag sequencing algorithm.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">DirecTagIdfileReader</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">An identification file reader for Direct tag results.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#DirecTagIdfileReader--">DirecTagIdfileReader()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Default constructor for the purpose of instantiation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#DirecTagIdfileReader-java.io.File-">DirecTagIdfileReader(File)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Constructors, parses a file but does not index the results.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#DirecTagIdfileReader-java.io.File-boolean-">DirecTagIdfileReader(File, boolean)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Constructors, parses a file.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">DirecTagParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">The DirecTag specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#DirecTagParameters--">DirecTagParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm"><span class="typeNameLink">DirecTagParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a></dt>
<dd>
<div class="block">Dialog for editing the DirecTag advanced settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html#DirecTagParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.DirecTagParameters-boolean-">DirecTagParametersDialog(Frame, DirecTagParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">DirecTagParametersDialog</a></dt>
<dd>
<div class="block">Creates a new DirecTagSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html#DirecTagParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.DirecTagParameters-boolean-">DirecTagParametersDialog(Dialog, Frame, DirecTagParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">DirecTagParametersDialog</a></dt>
<dd>
<div class="block">Creates a new DirecTagSettingsDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTPDownloader.html#disconnect--">disconnect()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTPDownloader.html" title="class in com.compomics.util.io.ftp">FTPDownloader</a></dt>
<dd>
<div class="block">Disconnect from the FTP server.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/Distribution.html" title="interface in com.compomics.util.math.statistics"><span class="typeNameLink">Distribution</span></a> - Interface in <a href="com/compomics/util/math/statistics/package-summary.html">com.compomics.util.math.statistics</a></dt>
<dd>
<div class="block">This class represents a statistical distribution model like a Gaussian
 distribution.</div>
</dd>
<dt><a href="com/compomics/util/math/roc/DistributionRoc.html" title="class in com.compomics.util.math.roc"><span class="typeNameLink">DistributionRoc</span></a> - Class in <a href="com/compomics/util/math/roc/package-summary.html">com.compomics.util.math.roc</a></dt>
<dd>
<div class="block">This class can be used to draw roc curves from experimental data.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/roc/DistributionRoc.html#DistributionRoc-com.compomics.util.math.statistics.Distribution-com.compomics.util.math.statistics.Distribution-">DistributionRoc(Distribution, Distribution)</a></span> - Constructor for class com.compomics.util.math.roc.<a href="com/compomics/util/math/roc/DistributionRoc.html" title="class in com.compomics.util.math.roc">DistributionRoc</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/servlet/MassCalcServlet.html#doGet-javax.servlet.http.HttpServletRequest-javax.servlet.http.HttpServletResponse-">doGet(HttpServletRequest, HttpServletResponse)</a></span> - Method in class com.compomics.util.general.servlet.<a href="com/compomics/util/general/servlet/MassCalcServlet.html" title="class in com.compomics.util.general.servlet">MassCalcServlet</a></dt>
<dd>
<div class="block">Override of parent class, implemented here as a simple forward to the
 'doPost' method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/servlet/MassCalcServlet.html#doPost-javax.servlet.http.HttpServletRequest-javax.servlet.http.HttpServletResponse-">doPost(HttpServletRequest, HttpServletResponse)</a></span> - Method in class com.compomics.util.general.servlet.<a href="com/compomics/util/general/servlet/MassCalcServlet.html" title="class in com.compomics.util.general.servlet">MassCalcServlet</a></dt>
<dd>
<div class="block">Override of parent class; this method handles post requests, and through
 forwarding also get requests.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#DOUBLE_PRECISION">DOUBLE_PRECISION</a></span> - Static variable in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Defines the valid String indicating the correct precision for encoded
 doubles.</div>
</dd>
<dt><a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix"><span class="typeNameLink">DoubleMatrix</span></a> - Class in <a href="com/compomics/util/math/matrix/package-summary.html">com.compomics.util.math.matrix</a></dt>
<dd>
<div class="block">Implementation of a matrix for double objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#DoubleMatrix--">DoubleMatrix()</a></span> - Constructor for class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#DoubleMatrix-int-">DoubleMatrix(int)</a></span> - Constructor for class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#DoubleMatrix-com.compomics.util.math.matrix.DoubleMatrix-">DoubleMatrix(DoubleMatrix)</a></span> - Constructor for class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#DOUBTFUL">DOUBTFUL</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Index for a doubtful assignment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTPDownloader.html#downloadFile-java.lang.String-java.io.File-">downloadFile(String, File)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTPDownloader.html" title="class in com.compomics.util.io.ftp">FTPDownloader</a></dt>
<dd>
<div class="block">Download the given file from the FTP server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#downloadGeneMappings-java.lang.String-java.lang.String-java.lang.String-java.lang.String-com.compomics.util.waiting.WaitingHandler-">downloadGeneMappings(String, String, String, String, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Download the gene mappings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#downloadGeneSequences-java.io.File-java.lang.String-java.lang.String-java.lang.String-com.compomics.util.waiting.WaitingHandler-">downloadGeneSequences(File, String, String, String, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Download the gene sequences mappings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#downloadGoMappings-java.lang.String-java.lang.String-java.lang.String-boolean-com.compomics.util.waiting.WaitingHandler-">downloadGoMappings(String, String, String, boolean, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Download the GO mappings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/CompomicsWrapper.html#downloadLatestVersion-java.lang.String-java.lang.String-java.lang.String-java.io.File-java.net.URL-java.lang.String-boolean-boolean-java.awt.Image-java.awt.Image-boolean-">downloadLatestVersion(String, String, String, File, URL, String, boolean, boolean, Image, Image, boolean)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/CompomicsWrapper.html" title="class in com.compomics.software">CompomicsWrapper</a></dt>
<dd>
<div class="block">Check if a newer version of the tool is deployed in the Maven repository,
 and closes the tool if the user decided to update.</div>
</dd>
<dt><a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html" title="class in com.compomics.software.autoupdater"><span class="typeNameLink">DownloadLatestZipFromRepo</span></a> - Class in <a href="com/compomics/software/autoupdater/package-summary.html">com.compomics.software.autoupdater</a></dt>
<dd>
<div class="block">Download the latest zip file from the repository.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html#DownloadLatestZipFromRepo--">DownloadLatestZipFromRepo()</a></span> - Constructor for class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html" title="class in com.compomics.software.autoupdater">DownloadLatestZipFromRepo</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html#downloadLatestZipFromRepo-java.net.URL-java.lang.String-">downloadLatestZipFromRepo(URL, String)</a></span> - Static method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html" title="class in com.compomics.software.autoupdater">DownloadLatestZipFromRepo</a></dt>
<dd>
<div class="block">Downloads the latest deploy from the genesis Maven repository of the
 artifact of the jarPath, starts it without arguments and removes the old
 jar if there was an update.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html#downloadLatestZipFromRepo-java.net.URL-java.lang.String-boolean-boolean-boolean-com.compomics.util.waiting.WaitingHandler-">downloadLatestZipFromRepo(URL, String, boolean, boolean, boolean, WaitingHandler)</a></span> - Static method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html" title="class in com.compomics.software.autoupdater">DownloadLatestZipFromRepo</a></dt>
<dd>
<div class="block">Downloads the latest deploy from the genesis Maven repository of the
 artifact and starts it without arguments.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html#downloadLatestZipFromRepo-java.net.URL-java.lang.String-boolean-java.lang.String:A-boolean-boolean-com.compomics.util.waiting.WaitingHandler-">downloadLatestZipFromRepo(URL, String, boolean, String[], boolean, boolean, WaitingHandler)</a></span> - Static method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html" title="class in com.compomics.software.autoupdater">DownloadLatestZipFromRepo</a></dt>
<dd>
<div class="block">Downloads the latest zip archive of the jar in the URL from the genesis
 Maven repository.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html#downloadLatestZipFromRepo-java.net.URL-java.lang.String-boolean-boolean-java.lang.String:A-java.net.URL-com.compomics.util.waiting.WaitingHandler-">downloadLatestZipFromRepo(URL, String, boolean, boolean, String[], URL, WaitingHandler)</a></span> - Static method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html" title="class in com.compomics.software.autoupdater">DownloadLatestZipFromRepo</a></dt>
<dd>
<div class="block">Downloads the latest zip archive of the jar in the URL from a given
 jarRepository.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html#downloadLatestZipFromRepo-java.net.URL-java.lang.String-boolean-java.lang.String-java.lang.String:A-java.net.URL-boolean-boolean-com.compomics.util.waiting.WaitingHandler-">downloadLatestZipFromRepo(URL, String, boolean, String, String[], URL, boolean, boolean, WaitingHandler)</a></span> - Static method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html" title="class in com.compomics.software.autoupdater">DownloadLatestZipFromRepo</a></dt>
<dd>
<div class="block">Retrieves the latest version of a Maven jar file from a Maven repository,
 also checks if the environment is headless or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html#downloadLatestZipFromRepo-java.net.URL-java.lang.String-boolean-java.lang.String-java.lang.String:A-java.net.URL-boolean-boolean-com.compomics.software.autoupdater.FileDAO-com.compomics.util.waiting.WaitingHandler-">downloadLatestZipFromRepo(URL, String, boolean, String, String[], URL, boolean, boolean, FileDAO, WaitingHandler)</a></span> - Static method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html" title="class in com.compomics.software.autoupdater">DownloadLatestZipFromRepo</a></dt>
<dd>
<div class="block">Retrieves the latest version of a Maven jar file from a Maven repository.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html#downloadLatestZipFromRepo-java.io.File-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String:A-java.net.URL-boolean-boolean-com.compomics.software.autoupdater.FileDAO-com.compomics.util.waiting.WaitingHandler-">downloadLatestZipFromRepo(File, String, String, String, String, String[], URL, boolean, boolean, FileDAO, WaitingHandler)</a></span> - Static method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/DownloadLatestZipFromRepo.html" title="class in com.compomics.software.autoupdater">DownloadLatestZipFromRepo</a></dt>
<dd>
<div class="block">Retrieves the latest version of a Maven jar file from a Maven repository.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html#downloadMapping-java.lang.String-">downloadMapping(String)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">UniprotTaxonomy</a></dt>
<dd>
<div class="block">Downloads the mapping for the given species name from UniProt and saves
 it to the mapping file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#downloadMappings-com.compomics.util.waiting.WaitingHandler-java.lang.Integer-">downloadMappings(WaitingHandler, Integer)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Try to download the gene and GO mappings for the currently selected
 species.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/PeptideShakerSetupDialog.html#downloadPeptideShaker--">downloadPeptideShaker()</a></span> - Method in class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/PeptideShakerSetupDialog.html" title="class in com.compomics.software.dialogs">PeptideShakerSetupDialog</a></dt>
<dd>
<div class="block">Download PeptideShaker.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/SearchGuiSetupDialog.html#downloadSearchGUI--">downloadSearchGUI()</a></span> - Method in class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/SearchGuiSetupDialog.html" title="class in com.compomics.software.dialogs">SearchGuiSetupDialog</a></dt>
<dd>
<div class="block">Download SearchGUI.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html#downloadTaxonomyFile-java.io.File-">downloadTaxonomyFile(File)</a></span> - Static method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">UniprotTaxonomy</a></dt>
<dd>
<div class="block">Downloads the UniProt taxonomy mapping to the given file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#drawAxes-java.awt.Graphics-double-double-int-double-double-">drawAxes(Graphics, double, double, int, double, double)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method draws the axes and their labels on the specified Graphics
 object, taking into account the padding.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#drawDaisyChain-java.awt.Graphics-java.util.Vector-java.util.Vector-int-int-java.util.Vector-java.util.Vector-boolean-">drawDaisyChain(Graphics, Vector, Vector, int, int, Vector, Vector, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Draw daisy chains.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#drawFilledPolygon-java.awt.Graphics-">drawFilledPolygon(Graphics)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method draws filled polygons for all of the peaks for all datasets
 in the current x-axis range on the panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#drawLabel--">drawLabel()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Returns true if the label is to be drawn, false otherwise.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#drawMeasurementLine-int-int-int-int-java.awt.Graphics-java.awt.Color-int-boolean-">drawMeasurementLine(int, int, int, int, Graphics, Color, int, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method draws a line, measuring the distance between two points in
 real mass units.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#drawMirroredPeaks-java.awt.Graphics-">drawMirroredPeaks(Graphics)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method draws all of the peaks for all datasets in the current x-axis
 range on the panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#drawOnTop--">drawOnTop()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Returns true if the area is to be drawn in front of the data, false
 otherwise.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#drawPeaks-java.awt.Graphics-">drawPeaks(Graphics)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method draws all of the peaks for all datasets in the current x-axis
 range on the panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#drawXAxisReferenceAreas-java.awt.Graphics-boolean-">drawXAxisReferenceAreas(Graphics, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Draws the x-axis reference areas if any.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#drawXTags-java.awt.Graphics-int-int-int-int-int-">drawXTags(Graphics, int, int, int, int, int)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method draws tags on the X axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#drawYAxisReferenceAreas-java.awt.Graphics-boolean-">drawYAxisReferenceAreas(Graphics, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Draws the y-axis reference areas if any.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#drawYTags-java.awt.Graphics-int-int-int-int-">drawYTags(Graphics, int, int, int, int)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method draws tags on the Y axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#dropDottedLine-int-int-int-java.awt.Graphics-boolean-">dropDottedLine(int, int, int, Graphics, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method drops a dotted line from the specified total height to the
 top of the indicated point.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#dta">dta</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#DTA">DTA</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/DtaFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">DtaFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.dta files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/DtaFileFilter.html#DtaFileFilter--">DtaFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/DtaFileFilter.html" title="class in com.compomics.util.io.file.filefilters">DtaFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein"><span class="typeNameLink">DualEnzyme</span></a> - Class in <a href="com/compomics/util/protein/package-summary.html">com.compomics.util.protein</a></dt>
<dd>
<div class="block">This class implements an enzyme with a dual specificity; the N-terminus of a
 resultant peptide will have certain residue, the C-terminus will have another,
 eg.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#DualEnzyme-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String-int-">DualEnzyme(String, String, String, String, String, int)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">This constructor allows you to specify all the information for this
 enzyme plus the number of missed cleavages that this instance will allow.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#DualEnzyme-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String-">DualEnzyme(String, String, String, String, String)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">This constructor allows you to specify all the information for this
 enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#dummy">dummy</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">An empty parameter used for instantiation.</div>
</dd>
<dt><a href="com/compomics/util/gui/DummyFrame.html" title="class in com.compomics.util.gui"><span class="typeNameLink">DummyFrame</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">A simple class used to be able to show a JDialog without a parent frame in
 the OS task bar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/DummyFrame.html#DummyFrame-java.lang.String-java.lang.String-">DummyFrame(String, String)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/DummyFrame.html" title="class in com.compomics.util.gui">DummyFrame</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/DummyFrame.html#DummyFrame-java.lang.String-java.awt.Image-">DummyFrame(String, Image)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/DummyFrame.html" title="class in com.compomics.util.gui">DummyFrame</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/parameters/DummyParameters.html" title="class in com.compomics.util.experiment.io.parameters"><span class="typeNameLink">DummyParameters</span></a> - Class in <a href="com/compomics/util/experiment/io/parameters/package-summary.html">com.compomics.util.experiment.io.parameters</a></dt>
<dd>
<div class="block">Dummy parameters used to retrieve the parameters type and version.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/parameters/DummyParameters.html#DummyParameters--">DummyParameters()</a></span> - Constructor for class com.compomics.util.experiment.io.parameters.<a href="com/compomics/util/experiment/io/parameters/DummyParameters.html" title="class in com.compomics.util.experiment.io.parameters">DummyParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#dumpToDB--">dumpToDB()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Triggers a dump of all objects within the cache into the database.</div>
</dd>
<dt><a href="com/compomics/util/waiting/Duration.html" title="class in com.compomics.util.waiting"><span class="typeNameLink">Duration</span></a> - Class in <a href="com/compomics/util/waiting/package-summary.html">com.compomics.util.waiting</a></dt>
<dd>
<div class="block">Class used to measure duration of a process.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/Duration.html#Duration--">Duration()</a></span> - Constructor for class com.compomics.util.waiting.<a href="com/compomics/util/waiting/Duration.html" title="class in com.compomics.util.waiting">Duration</a></dt>
<dd>
<div class="block">Constructor for an empty instance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/Duration.html#Duration-long-long-">Duration(long, long)</a></span> - Constructor for class com.compomics.util.waiting.<a href="com/compomics/util/waiting/Duration.html" title="class in com.compomics.util.waiting">Duration</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
</dl>
<a name="I:E">
<!--   -->
</a>
<h2 class="title">E</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#E">E</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigMathUtils.html#E">E</a></span> - Static variable in class com.compomics.util.math.<a href="com/compomics/util/math/BigMathUtils.html" title="class in com.compomics.util.math">BigMathUtils</a></dt>
<dd>
<div class="block">Big decimal value of E.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html#editFilter-com.compomics.util.experiment.filtering.Filter-">editFilter(Filter)</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationQCParametersDialog</a></dt>
<dd>
<div class="block">Edits the given filter.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/ElementaryElement.html" title="class in com.compomics.util.experiment.biology.atoms"><span class="typeNameLink">ElementaryElement</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/package-summary.html">com.compomics.util.experiment.biology.atoms</a></dt>
<dd>
<div class="block">This class can be used to retrieve elementary elements like a neutron.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl"><span class="typeNameLink">ElementaryIon</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/impl/package-summary.html">com.compomics.util.experiment.biology.ions.impl</a></dt>
<dd>
<div class="block">This class represents an elementary ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#ElementaryIon-java.lang.String-double-int-">ElementaryIon(String, double, int)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#empty">empty</a></span> - Static variable in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">Empty array for no result.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#emptyCache--">emptyCache()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Empties the cache.</div>
</dd>
<dt><a href="com/compomics/util/gui/utils/EmptyCollections.html" title="class in com.compomics.util.gui.utils"><span class="typeNameLink">EmptyCollections</span></a> - Class in <a href="com/compomics/util/gui/utils/package-summary.html">com.compomics.util.gui.utils</a></dt>
<dd>
<div class="block">This class contains empty collections that can be used for empty results without creating new objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/EmptyCollections.html#EmptyCollections--">EmptyCollections()</a></span> - Constructor for class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/EmptyCollections.html" title="class in com.compomics.util.gui.utils">EmptyCollections</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#emptyDir-java.io.File-">emptyDir(File)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Deletes all files and subdirectories under dir.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#emptyInternalCaches--">emptyInternalCaches()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">the sequence is kept in different formats internally.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/EmptyCollections.html#emptyIntSet">emptyIntSet</a></span> - Static variable in class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/EmptyCollections.html" title="class in com.compomics.util.gui.utils">EmptyCollections</a></dt>
<dd>
<div class="block">Convenience result of an empty set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/EmptyCollections.html#emptyIntTreeSet">emptyIntTreeSet</a></span> - Static variable in class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/EmptyCollections.html" title="class in com.compomics.util.gui.utils">EmptyCollections</a></dt>
<dd>
<div class="block">Convenience result of an empty set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/EmptyCollections.html#emptyStringSet">emptyStringSet</a></span> - Static variable in class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/EmptyCollections.html" title="class in com.compomics.util.gui.utils">EmptyCollections</a></dt>
<dd>
<div class="block">Convenience result of an empty set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/EmptyCollections.html#emptyStringTreeSet">emptyStringTreeSet</a></span> - Static variable in class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/EmptyCollections.html" title="class in com.compomics.util.gui.utils">EmptyCollections</a></dt>
<dd>
<div class="block">Convenience result of an empty set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#encoding">encoding</a></span> - Static variable in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>
<div class="block">Encoding for the file, cf the second rule.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/TextWriter.html#encoding">encoding</a></span> - Static variable in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/TextWriter.html" title="class in com.compomics.util.io.export.writers">TextWriter</a></dt>
<dd>
<div class="block">Encoding for the file, cf the second rule.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/Duration.html#end--">end()</a></span> - Method in class com.compomics.util.waiting.<a href="com/compomics/util/waiting/Duration.html" title="class in com.compomics.util.waiting">Duration</a></dt>
<dd>
<div class="block">Sets the end time to the system current time in milliseconds.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#ENSEMBL_GENOME_SPECIES">ENSEMBL_GENOME_SPECIES</a></span> - Static variable in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">The names of the Ensembl genome species files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#ENSEMBL_SPECIES">ENSEMBL_SPECIES</a></span> - Static variable in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">The names of the Ensembl species files.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings"><span class="typeNameLink">EnsemblGenomesSpecies</span></a> - Class in <a href="com/compomics/util/experiment/biology/taxonomy/mappings/package-summary.html">com.compomics.util.experiment.biology.taxonomy.mappings</a></dt>
<dd>
<div class="block">This class provides information about the species mapping in Ensembl Genomes
 (Bacteria, Fungi, Metazoa, Plants, Protists).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html#EnsemblGenomesSpecies--">EnsemblGenomesSpecies()</a></span> - Constructor for class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblGenomesSpecies</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.EnsemblGenomeDivision.html" title="enum in com.compomics.util.experiment.biology.taxonomy.mappings"><span class="typeNameLink">EnsemblGenomesSpecies.EnsemblGenomeDivision</span></a> - Enum in <a href="com/compomics/util/experiment/biology/taxonomy/mappings/package-summary.html">com.compomics.util.experiment.biology.taxonomy.mappings</a></dt>
<dd>
<div class="block">Enum of the different Ensembl genome divisions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.EnsemblGenomeDivision.html#ensemblName">ensemblName</a></span> - Variable in enum com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.EnsemblGenomeDivision.html" title="enum in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblGenomesSpecies.EnsemblGenomeDivision</a></dt>
<dd>
<div class="block">The name in the Ensembl mapping file.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings"><span class="typeNameLink">EnsemblSpecies</span></a> - Class in <a href="com/compomics/util/experiment/biology/taxonomy/mappings/package-summary.html">com.compomics.util.experiment.biology.taxonomy.mappings</a></dt>
<dd>
<div class="block">Mapping of the Ensembl species.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html#EnsemblSpecies--">EnsemblSpecies()</a></span> - Constructor for class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblSpecies</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.EnsemblGenomeDivision.html#ensemblType">ensemblType</a></span> - Variable in enum com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.EnsemblGenomeDivision.html" title="enum in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblGenomesSpecies.EnsemblGenomeDivision</a></dt>
<dd>
<div class="block">The schema name for XML queries.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/genes/ensembl/EnsemblVersion.html" title="class in com.compomics.util.experiment.biology.genes.ensembl"><span class="typeNameLink">EnsemblVersion</span></a> - Class in <a href="com/compomics/util/experiment/biology/genes/ensembl/package-summary.html">com.compomics.util.experiment.biology.genes.ensembl</a></dt>
<dd>
<div class="block">Class for the handling of Ensembl versions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ensembl/EnsemblVersion.html#EnsemblVersion--">EnsemblVersion()</a></span> - Constructor for class com.compomics.util.experiment.biology.genes.ensembl.<a href="com/compomics/util/experiment/biology/genes/ensembl/EnsemblVersion.html" title="class in com.compomics.util.experiment.biology.genes.ensembl">EnsemblVersion</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#ENTIRELY_NOT_ENZYMATIC">ENTIRELY_NOT_ENZYMATIC</a></span> - Static variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes"><span class="typeNameLink">Enzyme</span></a> - Class in <a href="com/compomics/util/experiment/biology/enzymes/package-summary.html">com.compomics.util.experiment.biology.enzymes</a></dt>
<dd>
<div class="block">This class models an enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#Enzyme-java.lang.String-">Enzyme(String)</a></span> - Constructor for class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Constructor for an Enzyme.</div>
</dd>
<dt><a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein"><span class="typeNameLink">Enzyme</span></a> - Class in <a href="com/compomics/util/protein/package-summary.html">com.compomics.util.protein</a></dt>
<dd>
<div class="block">This class implements the functionality for an Enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#Enzyme-java.lang.String-java.lang.String-java.lang.String-java.lang.String-">Enzyme(String, String, String, String)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This constructor requires that you specify all the information for this
 enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#Enzyme-java.lang.String-java.lang.String-java.lang.String-java.lang.String-int-">Enzyme(String, String, String, String, int)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This constructor allows you to specify all the information for this
 enzyme plus the number of missed cleavages that this instance will allow.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#Enzyme-com.compomics.util.experiment.biology.enzymes.Enzyme-int-">Enzyme(Enzyme, int)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">Creates a new Enzyme from a com.compomics.util.experiment.biology.Enzyme
 enzyme and the maximum number of missed cleavages.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes"><span class="typeNameLink">EnzymeFactory</span></a> - Class in <a href="com/compomics/util/experiment/biology/enzymes/package-summary.html">com.compomics.util.experiment.biology.enzymes</a></dt>
<dd>
<div class="block">This factory will provide the implemented enzymes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#enzymeLoaded-java.lang.String-">enzymeLoaded(String)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Indicates whether an enzyme is loaded in the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#enzymeMapping-com.compomics.util.parameters.identification.search.DigestionParameters-">enzymeMapping(DigestionParameters)</a></span> - Static method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Tries to map the utilities enzyme in the digestion preferences to the
 enzymes supported by MyriMatch.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/pride/EnzymeParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.pride"><span class="typeNameLink">EnzymeParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/pride/package-summary.html">com.compomics.util.gui.parameters.identification.pride</a></dt>
<dd>
<div class="block">Simple dialog for mapping a PRIDE enzyme to a utilities enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/pride/EnzymeParametersDialog.html#EnzymeParametersDialog-java.awt.Frame-boolean-java.lang.String-boolean-">EnzymeParametersDialog(Frame, boolean, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.pride.<a href="com/compomics/util/gui/parameters/identification/pride/EnzymeParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.pride">EnzymeParametersDialog</a></dt>
<dd>
<div class="block">Creates a new EnzymeSelectionDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/pride/EnzymeParametersDialog.html#EnzymeParametersDialog-java.awt.Dialog-java.awt.Frame-boolean-java.lang.String-boolean-">EnzymeParametersDialog(Dialog, Frame, boolean, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.pride.<a href="com/compomics/util/gui/parameters/identification/pride/EnzymeParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.pride">EnzymeParametersDialog</a></dt>
<dd>
<div class="block">Creates a new EnzymeSelectionDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/pride/EnzymeParametersDialog.html#EnzymeParametersDialog-javax.swing.JDialog-boolean-java.lang.String-">EnzymeParametersDialog(JDialog, boolean, String)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.pride.<a href="com/compomics/util/gui/parameters/identification/pride/EnzymeParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.pride">EnzymeParametersDialog</a></dt>
<dd>
<div class="block">Creates a new EnzymeSelectionDialog.</div>
</dd>
<dt><a href="com/compomics/cli/enzymes/EnzymesCLI.html" title="class in com.compomics.cli.enzymes"><span class="typeNameLink">EnzymesCLI</span></a> - Class in <a href="com/compomics/cli/enzymes/package-summary.html">com.compomics.cli.enzymes</a></dt>
<dd>
<div class="block">Command line to manage the enzymes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLI.html#EnzymesCLI-com.compomics.cli.enzymes.EnzymesCLIInputBean-">EnzymesCLI(EnzymesCLIInputBean)</a></span> - Constructor for class com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLI.html" title="class in com.compomics.cli.enzymes">EnzymesCLI</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html" title="class in com.compomics.cli.enzymes"><span class="typeNameLink">EnzymesCLIInputBean</span></a> - Class in <a href="com/compomics/cli/enzymes/package-summary.html">com.compomics.cli.enzymes</a></dt>
<dd>
<div class="block">This class parses the parameters from an EnzymeCLI.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html#EnzymesCLIInputBean-org.apache.commons.cli.CommandLine-">EnzymesCLIInputBean(CommandLine)</a></span> - Constructor for class com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html" title="class in com.compomics.cli.enzymes">EnzymesCLIInputBean</a></dt>
<dd>
<div class="block">Parses all the arguments from a command line.</div>
</dd>
<dt><a href="com/compomics/cli/enzymes/EnzymesCLIParams.html" title="enum in com.compomics.cli.enzymes"><span class="typeNameLink">EnzymesCLIParams</span></a> - Enum in <a href="com/compomics/cli/enzymes/package-summary.html">com.compomics.cli.enzymes</a></dt>
<dd>
<div class="block">Enum class specifying the EnzymesCLI parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#equals-java.lang.Object-">equals(Object)</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Performs equals methods dependent on values of instance variables and
 class of Object o.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#equals-java.lang.Object-">equals(Object)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#equals-com.compomics.util.experiment.biology.enzymes.Enzyme-">equals(Enzyme)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Returns true of the two enzymes are identical.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#equals-com.compomics.util.experiment.io.biology.protein.FastaParameters-">equals(FastaParameters)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#equals-java.lang.Object-">equals(Object)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationAlgorithmParameter.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in interface com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationAlgorithmParameter.html" title="interface in com.compomics.util.gui.parameters.identification">IdentificationAlgorithmParameter</a></dt>
<dd>
<div class="block">Indicates whether another identificationAlgorithmParameter has the same
 parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#equals-com.compomics.util.math.matrix.DoubleMatrix-">equals(DoubleMatrix)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html#equals-com.compomics.util.parameters.identification.advanced.GeneParameters-">equals(GeneParameters)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html" title="class in com.compomics.util.parameters.identification.advanced">GeneParameters</a></dt>
<dd>
<div class="block">Compares these preferences to other preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html#equals-com.compomics.util.parameters.identification.advanced.IdMatchValidationParameters-">equals(IdMatchValidationParameters)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">IdMatchValidationParameters</a></dt>
<dd>
<div class="block">Returns true if the objects have identical settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#equals-com.compomics.util.parameters.identification.advanced.ModificationLocalizationParameters-">equals(ModificationLocalizationParameters)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Returns true if the objects have identical settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#equals-com.compomics.util.parameters.identification.advanced.ProteinInferenceParameters-">equals(ProteinInferenceParameters)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Returns true if the objects have identical settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#equals-com.compomics.util.parameters.identification.advanced.PsmScoringParameters-">equals(PsmScoringParameters)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Returns true if the objects have identical settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#equals-com.compomics.util.parameters.identification.IdentificationParameters-">equals(IdentificationParameters)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns true if the identification parameter objects have identical
 settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#equals-com.compomics.util.parameters.identification.search.ModificationParameters-">equals(ModificationParameters)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Returns true of the two profiles are identical.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#equals-com.compomics.util.parameters.identification.search.SearchParameters-">equals(SearchParameters)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns true if the search parameter objects have identical settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">NovorParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#equals-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">equals(IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#equals-java.lang.Object-">equals(Object)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">Indicates whether some other object is "equal to" this one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#equals-java.lang.Object-">equals(Object)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">Indicates whether some other object is "equal to" this one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#equals-java.lang.Object-">equals(Object)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This method will check equality between this object
 and another Protein instance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html#error-org.xml.sax.SAXParseException-">error(SAXParseException)</a></span> - Method in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html" title="class in com.compomics.util.pride.validation">XMLValidationErrorHandler</a></dt>
<dd>
<div class="block">Receive notification of a recoverable error.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#establishConnection-boolean-">establishConnection(boolean)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Establishes connection to the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#estimateAACoverage-long-boolean-">estimateAACoverage(long, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns amino acid coverage of this protein by enzymatic or non-enzymatic
 peptides only in an array where the index of the best validation level of
 every peptide covering a given amino acid is given.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalKernelDensityEstimator.html#estimateDensityFunction-java.lang.Double:A-">estimateDensityFunction(Double[])</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalKernelDensityEstimator.html" title="class in com.compomics.util.math.statistics.distributions">NormalKernelDensityEstimator</a></dt>
<dd>
<div class="block">Estimate the density function.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalKernelDensityEstimator.html#estimateDensityFunction-double:A-">estimateDensityFunction(double[])</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalKernelDensityEstimator.html" title="class in com.compomics.util.math.statistics.distributions">NormalKernelDensityEstimator</a></dt>
<dd>
<div class="block">Estimate the density function.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/spectrumcounting/ScalingFactorsEstimators.html#estimateScalingFactors-com.compomics.util.experiment.identification.Identification-com.compomics.util.experiment.identification.peptide_shaker.Metrics-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.experiment.identification.IdentificationFeaturesGenerator-com.compomics.util.waiting.WaitingHandler-com.compomics.util.exceptions.ExceptionHandler-com.compomics.util.parameters.tools.ProcessingParameters-">estimateScalingFactors(Identification, Metrics, SequenceProvider, IdentificationFeaturesGenerator, WaitingHandler, ExceptionHandler, ProcessingParameters)</a></span> - Method in class com.compomics.util.experiment.quantification.spectrumcounting.<a href="com/compomics/util/experiment/quantification/spectrumcounting/ScalingFactorsEstimators.html" title="class in com.compomics.util.experiment.quantification.spectrumcounting">ScalingFactorsEstimators</a></dt>
<dd>
<div class="block">Estimates the scaling factors and stores them in the given metrics.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#estimateSpectrumCounting-com.compomics.util.experiment.identification.Identification-com.compomics.util.experiment.io.biology.protein.SequenceProvider-long-com.compomics.util.parameters.quantification.spectrum_counting.SpectrumCountingParameters-int-com.compomics.util.parameters.identification.IdentificationParameters-">estimateSpectrumCounting(Identification, SequenceProvider, long, SpectrumCountingParameters, int, IdentificationParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the spectrum counting index based on the project settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#estimateTheoreticMass-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">estimateTheoreticMass(ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Estimates the theoretic mass of the peptide.</div>
</dd>
<dt><a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers"><span class="typeNameLink">ExcelWriter</span></a> - Class in <a href="com/compomics/util/io/export/writers/package-summary.html">com.compomics.util.io.export.writers</a></dt>
<dd>
<div class="block">ExportWriter for the export to excel files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#ExcelWriter-java.io.File-">ExcelWriter(File)</a></span> - Constructor for class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/ExceptionHandler.html#exceptionCaught">exceptionCaught</a></span> - Variable in class com.compomics.util.exceptions.<a href="com/compomics/util/exceptions/ExceptionHandler.html" title="class in com.compomics.util.exceptions">ExceptionHandler</a></dt>
<dd>
<div class="block">List of caught exceptions.</div>
</dd>
<dt><a href="com/compomics/util/exceptions/ExceptionHandler.html" title="class in com.compomics.util.exceptions"><span class="typeNameLink">ExceptionHandler</span></a> - Class in <a href="com/compomics/util/exceptions/package-summary.html">com.compomics.util.exceptions</a></dt>
<dd>
<div class="block">Interface for the general handling of exceptions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/ExceptionHandler.html#ExceptionHandler--">ExceptionHandler()</a></span> - Constructor for class com.compomics.util.exceptions.<a href="com/compomics/util/exceptions/ExceptionHandler.html" title="class in com.compomics.util.exceptions">ExceptionHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#exceptions">exceptions</a></span> - Variable in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">The exceptions to the rule.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalKernelDensityEstimator.html#excludeNullValues-java.lang.Double:A-">excludeNullValues(Double[])</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalKernelDensityEstimator.html" title="class in com.compomics.util.math.statistics.distributions">NormalKernelDensityEstimator</a></dt>
<dd>
<div class="block">Exclude null values from an array of double.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigFunctions.html#exp-java.math.BigDecimal-java.math.MathContext-">exp(BigDecimal, MathContext)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BigFunctions.html" title="class in com.compomics.util.math">BigFunctions</a></dt>
<dd>
<div class="block">Returns the value of the exponential of the given BigDecimal using the
 given MathContext.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigFunctions.html#expBD-java.math.BigDecimal-java.math.MathContext-">expBD(BigDecimal, MathContext)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BigFunctions.html" title="class in com.compomics.util.math">BigFunctions</a></dt>
<dd>
<div class="block">Returns the value of the exponential of the given BigDecimal using the
 given MathContext.</div>
</dd>
<dt><a href="com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization"><span class="typeNameLink">ExperimentObject</span></a> - Class in <a href="com/compomics/util/experiment/personalization/package-summary.html">com.compomics.util.experiment.personalization</a></dt>
<dd>
<div class="block">This abstract class provides customization facilities.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/personalization/ExperimentObject.html#ExperimentObject--">ExperimentObject()</a></span> - Constructor for class com.compomics.util.experiment.personalization.<a href="com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/Export.html" title="class in com.compomics.util"><span class="typeNameLink">Export</span></a> - Class in <a href="com/compomics/util/package-summary.html">com.compomics.util</a></dt>
<dd>
<div class="block">Includes export to figure formats for Components and JFreeCharts.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Export.html#Export--">Export()</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/Export.html" title="class in com.compomics.util">Export</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Export.html#exportChart-org.jfree.chart.JFreeChart-java.awt.Rectangle-java.io.File-com.compomics.util.enumeration.ImageType-">exportChart(JFreeChart, Rectangle, File, ImageType)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Export.html" title="class in com.compomics.util">Export</a></dt>
<dd>
<div class="block">Exports the contents of a JFreeChart to an svg, png, pdf etc.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Export.html#exportComponent-java.awt.Component-java.awt.Rectangle-java.io.File-com.compomics.util.enumeration.ImageType-">exportComponent(Component, Rectangle, File, ImageType)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Export.html" title="class in com.compomics.util">Export</a></dt>
<dd>
<div class="block">Exports the contents of a Component to an svg, png, pdf etc.</div>
</dd>
<dt><a href="com/compomics/util/io/export/ExportFactory.html" title="interface in com.compomics.util.io.export"><span class="typeNameLink">ExportFactory</span></a> - Interface in <a href="com/compomics/util/io/export/package-summary.html">com.compomics.util.io.export</a></dt>
<dd>
<div class="block">An export factory manages and generates reports.</div>
</dd>
<dt><a href="com/compomics/util/io/export/ExportFeature.html" title="interface in com.compomics.util.io.export"><span class="typeNameLink">ExportFeature</span></a> - Interface in <a href="com/compomics/util/io/export/package-summary.html">com.compomics.util.io.export</a></dt>
<dd>
<div class="block">This interface represents an export feature.</div>
</dd>
<dt><a href="com/compomics/util/io/export/ExportFormat.html" title="enum in com.compomics.util.io.export"><span class="typeNameLink">ExportFormat</span></a> - Enum in <a href="com/compomics/util/io/export/package-summary.html">com.compomics.util.io.export</a></dt>
<dd>
<div class="block">Enum of the different formats available for the exports.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#exportFormat">exportFormat</a></span> - Variable in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">The format of the export.</div>
</dd>
<dt><a href="com/compomics/util/gui/ExportFormatSelectionDialog.html" title="class in com.compomics.util.gui"><span class="typeNameLink">ExportFormatSelectionDialog</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">Simple dialog for selecting the format type to export to.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/ExportFormatSelectionDialog.html#ExportFormatSelectionDialog-javax.swing.JFrame-boolean-">ExportFormatSelectionDialog(JFrame, boolean)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/ExportFormatSelectionDialog.html" title="class in com.compomics.util.gui">ExportFormatSelectionDialog</a></dt>
<dd>
<div class="block">Create and open a new ExportFormatSelectionDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/ExportFormatSelectionDialog.html#ExportFormatSelectionDialog-javax.swing.JDialog-boolean-">ExportFormatSelectionDialog(JDialog, boolean)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/ExportFormatSelectionDialog.html" title="class in com.compomics.util.gui">ExportFormatSelectionDialog</a></dt>
<dd>
<div class="block">Create and open a new ExportFormatSelectionDialog.</div>
</dd>
<dt><a href="com/compomics/util/gui/export/graphics/ExportGraphicsDialog.html" title="class in com.compomics.util.gui.export.graphics"><span class="typeNameLink">ExportGraphicsDialog</span></a> - Class in <a href="com/compomics/util/gui/export/graphics/package-summary.html">com.compomics.util.gui.export.graphics</a></dt>
<dd>
<div class="block">Simple dialog for selecting the image type to export a graphics element to.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/export/graphics/ExportGraphicsDialog.html#ExportGraphicsDialog-javax.swing.JFrame-java.awt.Image-java.awt.Image-boolean-java.awt.Component-com.compomics.util.io.file.LastSelectedFolder-">ExportGraphicsDialog(JFrame, Image, Image, boolean, Component, LastSelectedFolder)</a></span> - Constructor for class com.compomics.util.gui.export.graphics.<a href="com/compomics/util/gui/export/graphics/ExportGraphicsDialog.html" title="class in com.compomics.util.gui.export.graphics">ExportGraphicsDialog</a></dt>
<dd>
<div class="block">Create and open a new ExportGraphicsDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/export/graphics/ExportGraphicsDialog.html#ExportGraphicsDialog-javax.swing.JFrame-java.awt.Image-java.awt.Image-boolean-org.jfree.chart.ChartPanel-com.compomics.util.io.file.LastSelectedFolder-">ExportGraphicsDialog(JFrame, Image, Image, boolean, ChartPanel, LastSelectedFolder)</a></span> - Constructor for class com.compomics.util.gui.export.graphics.<a href="com/compomics/util/gui/export/graphics/ExportGraphicsDialog.html" title="class in com.compomics.util.gui.export.graphics">ExportGraphicsDialog</a></dt>
<dd>
<div class="block">Create and open a new ExportPlot dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/export/graphics/ExportGraphicsDialog.html#ExportGraphicsDialog-javax.swing.JDialog-java.awt.Image-java.awt.Image-boolean-java.awt.Component-com.compomics.util.io.file.LastSelectedFolder-">ExportGraphicsDialog(JDialog, Image, Image, boolean, Component, LastSelectedFolder)</a></span> - Constructor for class com.compomics.util.gui.export.graphics.<a href="com/compomics/util/gui/export/graphics/ExportGraphicsDialog.html" title="class in com.compomics.util.gui.export.graphics">ExportGraphicsDialog</a></dt>
<dd>
<div class="block">Create and open a new ExportGraphicsDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/export/graphics/ExportGraphicsDialog.html#ExportGraphicsDialog-javax.swing.JDialog-java.awt.Image-java.awt.Image-boolean-org.jfree.chart.ChartPanel-com.compomics.util.io.file.LastSelectedFolder-">ExportGraphicsDialog(JDialog, Image, Image, boolean, ChartPanel, LastSelectedFolder)</a></span> - Constructor for class com.compomics.util.gui.export.graphics.<a href="com/compomics/util/gui/export/graphics/ExportGraphicsDialog.html" title="class in com.compomics.util.gui.export.graphics">ExportGraphicsDialog</a></dt>
<dd>
<div class="block">Create and open a new ExportPlot dialog.</div>
</dd>
<dt><a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export"><span class="typeNameLink">ExportScheme</span></a> - Class in <a href="com/compomics/util/io/export/package-summary.html">com.compomics.util.io.export</a></dt>
<dd>
<div class="block">This class allows creating a standard output scheme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#ExportScheme-java.lang.String-boolean-java.util.HashMap-java.lang.String-boolean-boolean-int-boolean-boolean-boolean-java.lang.String-">ExportScheme(String, boolean, HashMap&lt;String, ArrayList&lt;ExportFeature&gt;&gt;, String, boolean, boolean, int, boolean, boolean, boolean, String)</a></span> - Constructor for class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#ExportScheme-java.lang.String-boolean-java.util.ArrayList-java.util.HashMap-java.lang.String-boolean-boolean-int-boolean-boolean-boolean-">ExportScheme(String, boolean, ArrayList&lt;String&gt;, HashMap&lt;String, ArrayList&lt;ExportFeature&gt;&gt;, String, boolean, boolean, int, boolean, boolean, boolean)</a></span> - Constructor for class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#ExportScheme-java.lang.String-boolean-java.util.HashMap-java.lang.String-boolean-boolean-int-boolean-boolean-boolean-">ExportScheme(String, boolean, HashMap&lt;String, ArrayList&lt;ExportFeature&gt;&gt;, String, boolean, boolean, int, boolean, boolean, boolean)</a></span> - Constructor for class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export"><span class="typeNameLink">ExportWriter</span></a> - Class in <a href="com/compomics/util/io/export/package-summary.html">com.compomics.util.io.export</a></dt>
<dd>
<div class="block">The export writer will write the output in the desired export format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#ExportWriter--">ExportWriter()</a></span> - Constructor for class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#extension">extension</a></span> - Static variable in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">The extension to use when saving objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFormat.html#extention">extention</a></span> - Variable in enum com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFormat.html" title="enum in com.compomics.util.io.export">ExportFormat</a></dt>
<dd>
<div class="block">The extention to use for this format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/TarUtils.html#extractFile-java.io.File-com.compomics.util.waiting.WaitingHandler-">extractFile(File, WaitingHandler)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/TarUtils.html" title="class in com.compomics.util.io.compression">TarUtils</a></dt>
<dd>
<div class="block">Extracts files from a tar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/TarUtils.html#extractFile-java.io.File-java.io.File-com.compomics.util.waiting.WaitingHandler-">extractFile(File, File, WaitingHandler)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/TarUtils.html" title="class in com.compomics.util.io.compression">TarUtils</a></dt>
<dd>
<div class="block">Extracts files from a tar.</div>
</dd>
<dt><a href="com/compomics/scripts_marc/ExtractFirstLines.html" title="class in com.compomics.scripts_marc"><span class="typeNameLink">ExtractFirstLines</span></a> - Class in <a href="com/compomics/scripts_marc/package-summary.html">com.compomics.scripts_marc</a></dt>
<dd>
<div class="block">Extracts the first lines of a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/scripts_marc/ExtractFirstLines.html#ExtractFirstLines--">ExtractFirstLines()</a></span> - Constructor for class com.compomics.scripts_marc.<a href="com/compomics/scripts_marc/ExtractFirstLines.html" title="class in com.compomics.scripts_marc">ExtractFirstLines</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="I:F">
<!--   -->
</a>
<h2 class="title">F</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#F">F</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#F">F</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The Fluorine atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#factorial-java.lang.Integer-">factorial(Integer)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Returns n! as a long.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#factorial-java.lang.Integer-java.lang.Integer-">factorial(Integer, Integer)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Returns n!/k!, null if it cannot fit in a long.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigFunctions.html#factorial-java.math.BigInteger-">factorial(BigInteger)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BigFunctions.html" title="class in com.compomics.util.math">BigFunctions</a></dt>
<dd>
<div class="block">Returns n! as BigInteger.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigFunctions.html#factorial-java.math.BigInteger-java.math.BigInteger-">factorial(BigInteger, BigInteger)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BigFunctions.html" title="class in com.compomics.util.math">BigFunctions</a></dt>
<dd>
<div class="block">Returns n!/k! as BigInteger.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/SwingUtils.html#fadeIn-javax.swing.JDialog-">fadeIn(JDialog)</a></span> - Static method in class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/SwingUtils.html" title="class in com.compomics.util.gui.utils">SwingUtils</a></dt>
<dd>
<div class="block">Creates an animation to fade the dialog opacity from 0 to 1.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/SwingUtils.html#fadeIn-javax.swing.JDialog-int-float-">fadeIn(JDialog, int, float)</a></span> - Static method in class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/SwingUtils.html" title="class in com.compomics.util.gui.utils">SwingUtils</a></dt>
<dd>
<div class="block">Creates an animation to fade the dialog opacity from 0 to 1.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/SwingUtils.html#fadeInAndOut-javax.swing.JDialog-">fadeInAndOut(JDialog)</a></span> - Static method in class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/SwingUtils.html" title="class in com.compomics.util.gui.utils">SwingUtils</a></dt>
<dd>
<div class="block">Creates an animation to fade the dialog opacity from 0 to 1, wait at 1
 and then fade to 0.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/SwingUtils.html#fadeInAndOut-javax.swing.JDialog-int-float-int-">fadeInAndOut(JDialog, int, float, int)</a></span> - Static method in class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/SwingUtils.html" title="class in com.compomics.util.gui.utils">SwingUtils</a></dt>
<dd>
<div class="block">Creates an animation to fade the dialog opacity from 0 to 1, wait at 1
 and then fade to 0 and dispose.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/SwingUtils.html#fadeOut-javax.swing.JDialog-">fadeOut(JDialog)</a></span> - Static method in class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/SwingUtils.html" title="class in com.compomics.util.gui.utils">SwingUtils</a></dt>
<dd>
<div class="block">Creates an animation to fade the dialog opacity from 1 to 0.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/SwingUtils.html#fadeOut-javax.swing.JDialog-int-float-">fadeOut(JDialog, int, float)</a></span> - Static method in class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/SwingUtils.html" title="class in com.compomics.util.gui.utils">SwingUtils</a></dt>
<dd>
<div class="block">Creates an animation to fade the dialog opacity from 1 to 0, and then
 dispose.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#FAS">FAS</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#fas">fas</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#FASTA">FASTA</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#fasta">fasta</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html#fastaFile">fastaFile</a></span> - Variable in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaSummary</a></dt>
<dd>
<div class="block">The file this summary represents.</div>
</dd>
<dt><a href="com/compomics/util/io/file/filefilters/FastaFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">FastaFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.fasta files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FastaFileFilter.html#FastaFileFilter--">FastaFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FastaFileFilter.html" title="class in com.compomics.util.io.file.filefilters">FastaFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html" title="class in com.compomics.util.experiment.io.biology.protein.iterators"><span class="typeNameLink">FastaIterator</span></a> - Class in <a href="com/compomics/util/experiment/io/biology/protein/iterators/package-summary.html">com.compomics.util.experiment.io.biology.protein.iterators</a></dt>
<dd>
<div class="block">Iterator for a fasta file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html#FastaIterator-java.io.File-">FastaIterator(File)</a></span> - Constructor for class com.compomics.util.experiment.io.biology.protein.iterators.<a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html" title="class in com.compomics.util.experiment.io.biology.protein.iterators">FastaIterator</a></dt>
<dd>
<div class="block">Constructor without sanity check.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html#FastaIterator-java.io.File-boolean-">FastaIterator(File, boolean)</a></span> - Constructor for class com.compomics.util.experiment.io.biology.protein.iterators.<a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html" title="class in com.compomics.util.experiment.io.biology.protein.iterators">FastaIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_inference/FastaMapper.html" title="interface in com.compomics.util.experiment.identification.protein_inference"><span class="typeNameLink">FastaMapper</span></a> - Interface in <a href="com/compomics/util/experiment/identification/protein_inference/package-summary.html">com.compomics.util.experiment.identification.protein_inference</a></dt>
<dd>
<div class="block">Interface for a class mapping peptides and tags to a protein.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein"><span class="typeNameLink">FastaParameters</span></a> - Class in <a href="com/compomics/util/experiment/io/biology/protein/package-summary.html">com.compomics.util.experiment.io.biology.protein</a></dt>
<dd>
<div class="block">The parameters used to parse a FASTA file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#FastaParameters--">FastaParameters()</a></span> - Constructor for class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/cli/fasta/FastaParametersCLIParams.html" title="enum in com.compomics.cli.fasta"><span class="typeNameLink">FastaParametersCLIParams</span></a> - Enum in <a href="com/compomics/cli/fasta/package-summary.html">com.compomics.cli.fasta</a></dt>
<dd>
<div class="block">Command line parameters for the fasta files.</div>
</dd>
<dt><a href="com/compomics/cli/fasta/FastaParametersInputBean.html" title="class in com.compomics.cli.fasta"><span class="typeNameLink">FastaParametersInputBean</span></a> - Class in <a href="com/compomics/cli/fasta/package-summary.html">com.compomics.cli.fasta</a></dt>
<dd>
<div class="block">This class gathers command line parameters for the parsing of fasta files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/fasta/FastaParametersInputBean.html#FastaParametersInputBean-org.apache.commons.cli.CommandLine-java.io.File-">FastaParametersInputBean(CommandLine, File)</a></span> - Constructor for class com.compomics.cli.fasta.<a href="com/compomics/cli/fasta/FastaParametersInputBean.html" title="class in com.compomics.cli.fasta">FastaParametersInputBean</a></dt>
<dd>
<div class="block">Parses all the arguments from a command line.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html" title="class in com.compomics.util.experiment.io.biology.protein"><span class="typeNameLink">FastaSummary</span></a> - Class in <a href="com/compomics/util/experiment/io/biology/protein/package-summary.html">com.compomics.util.experiment.io.biology.protein</a></dt>
<dd>
<div class="block">This class parses a FASTA file and gathers summary statistics.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html#FastaSummary-java.io.File-java.util.TreeMap-java.util.HashMap-int-int-long-">FastaSummary(File, TreeMap&lt;String, Integer&gt;, HashMap&lt;ProteinDatabase, Integer&gt;, int, int, long)</a></span> - Constructor for class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaSummary</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#fastaTag">fastaTag</a></span> - Static variable in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>
<div class="block">Tag for the fasta file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html#fatalError-org.xml.sax.SAXParseException-">fatalError(SAXParseException)</a></span> - Method in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html" title="class in com.compomics.util.pride.validation">XMLValidationErrorHandler</a></dt>
<dd>
<div class="block">Receive notification of a non-recoverable error.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html#fatalError-java.io.IOException-">fatalError(IOException)</a></span> - Method in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html" title="class in com.compomics.util.pride.validation">XMLValidationErrorHandler</a></dt>
<dd>
<div class="block">This is a extension method to the #fatalError(SAXParseException) method
 to allow the handling of IOExceptions.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_generation/FeaturesGenerator.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation"><span class="typeNameLink">FeaturesGenerator</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_generation/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation</a></dt>
<dd>
<div class="block">This class computes the ms2PIP features from a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_generation/FeaturesGenerator.html#FeaturesGenerator-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.FeaturesMap-">FeaturesGenerator(FeaturesMap)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_generation/FeaturesGenerator.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation">FeaturesGenerator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration"><span class="typeNameLink">FeaturesMap</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration</a></dt>
<dd>
<div class="block">This class contains a map of selected features indexed by category.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html#FeaturesMap--">FeaturesMap()</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">FeaturesMap</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMapManager.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration"><span class="typeNameLink">FeaturesMapManager</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration</a></dt>
<dd>
<div class="block">This class is used to manage the features maps.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMapManager.html#FeaturesMapManager--">FeaturesMapManager()</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMapManager.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">FeaturesMapManager</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages"><span class="typeNameLink">FeedBack</span></a> - Class in <a href="com/compomics/util/messages/package-summary.html">com.compomics.util.messages</a></dt>
<dd>
<div class="block">This class is used to report a message to the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#FeedBack-com.compomics.util.messages.FeedBack.FeedBackType-java.lang.String-java.util.ArrayList-java.lang.String-">FeedBack(FeedBack.FeedBackType, String, ArrayList&lt;String&gt;, String)</a></span> - Constructor for class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Creates a new report for the user.</div>
</dd>
<dt><a href="com/compomics/util/messages/FeedBack.FeedBackType.html" title="enum in com.compomics.util.messages"><span class="typeNameLink">FeedBack.FeedBackType</span></a> - Enum in <a href="com/compomics/util/messages/package-summary.html">com.compomics.util.messages</a></dt>
<dd>
<div class="block">The supported report types.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#FILE_GET">FILE_GET</a></span> - Static variable in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Pre-defined state.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#FILE_PUT">FILE_PUT</a></span> - Static variable in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Pre-defined state.</div>
</dd>
<dt><a href="com/compomics/util/io/json/adapter/FileAdapter.html" title="class in com.compomics.util.io.json.adapter"><span class="typeNameLink">FileAdapter</span></a> - Class in <a href="com/compomics/util/io/json/adapter/package-summary.html">com.compomics.util.io.json.adapter</a></dt>
<dd>
<div class="block">This class is intended to avoid the default behavior of the GSON parser to
 append the current path to the absolute path of a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/adapter/FileAdapter.html#FileAdapter--">FileAdapter()</a></span> - Constructor for class com.compomics.util.io.json.adapter.<a href="com/compomics/util/io/json/adapter/FileAdapter.html" title="class in com.compomics.util.io.json.adapter">FileAdapter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/FileAndFileFilter.html" title="class in com.compomics.util"><span class="typeNameLink">FileAndFileFilter</span></a> - Class in <a href="com/compomics/util/package-summary.html">com.compomics.util</a></dt>
<dd>
<div class="block">Store the selected file and the file type description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/FileAndFileFilter.html#FileAndFileFilter-java.io.File-javax.swing.filechooser.FileFilter-">FileAndFileFilter(File, FileFilter)</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/FileAndFileFilter.html" title="class in com.compomics.util">FileAndFileFilter</a></dt>
<dd>
<div class="block">Create a new FileAndFileFilter object.</div>
</dd>
<dt><a href="com/compomics/software/autoupdater/FileDAO.html" title="class in com.compomics.software.autoupdater"><span class="typeNameLink">FileDAO</span></a> - Class in <a href="com/compomics/software/autoupdater/package-summary.html">com.compomics.software.autoupdater</a></dt>
<dd>
<div class="block">FileDAO file access.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/FileDAO.html#FileDAO--">FileDAO()</a></span> - Constructor for class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/FileDAO.html" title="class in com.compomics.software.autoupdater">FileDAO</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/file_handling/FileDisplayDialog.html" title="class in com.compomics.util.gui.file_handling"><span class="typeNameLink">FileDisplayDialog</span></a> - Class in <a href="com/compomics/util/gui/file_handling/package-summary.html">com.compomics.util.gui.file_handling</a></dt>
<dd>
<div class="block">Dialog for showing a list of selected files with the option to remove files
 from the list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/file_handling/FileDisplayDialog.html#FileDisplayDialog-javax.swing.JFrame-java.util.ArrayList-boolean-">FileDisplayDialog(JFrame, ArrayList&lt;File&gt;, boolean)</a></span> - Constructor for class com.compomics.util.gui.file_handling.<a href="com/compomics/util/gui/file_handling/FileDisplayDialog.html" title="class in com.compomics.util.gui.file_handling">FileDisplayDialog</a></dt>
<dd>
<div class="block">Creates a new FileSelectionDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/file_handling/FileDisplayDialog.html#FileDisplayDialog-javax.swing.JDialog-java.util.ArrayList-boolean-">FileDisplayDialog(JDialog, ArrayList&lt;File&gt;, boolean)</a></span> - Constructor for class com.compomics.util.gui.file_handling.<a href="com/compomics/util/gui/file_handling/FileDisplayDialog.html" title="class in com.compomics.util.gui.file_handling">FileDisplayDialog</a></dt>
<dd>
<div class="block">Creates a new FileSelectionDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#fileExists-java.lang.String-java.lang.String-">fileExists(String, String)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true if the argument can be parsed as a file and the file exists.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#fileExists-java.lang.String-java.lang.String-java.util.HashSet-">fileExists(String, String, HashSet&lt;String&gt;)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true if the argument can be parsed as a file, the file exists and
 is in the specified format.</div>
</dd>
<dt><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">FileFilterUtils</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">Organizes the file filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#FileFilterUtils--">FileFilterUtils()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#fileLoaded-java.lang.String-">fileLoaded(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the spectrum file has been loaded.</div>
</dd>
<dt><a href="com/compomics/util/general/FileLoggerImplementation.html" title="class in com.compomics.util.general"><span class="typeNameLink">FileLoggerImplementation</span></a> - Class in <a href="com/compomics/util/general/package-summary.html">com.compomics.util.general</a></dt>
<dd>
<div class="block">This class implements the Logger interface for logging to a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/FileLoggerImplementation.html#FileLoggerImplementation-java.lang.String-">FileLoggerImplementation(String)</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/FileLoggerImplementation.html" title="class in com.compomics.util.general">FileLoggerImplementation</a></dt>
<dd>
<div class="block">This constructor takes the name for the log file and sets a
 default timestamp format: "dd/MM/yyyy - HH:mm:ss".</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/FileLoggerImplementation.html#FileLoggerImplementation-java.lang.String-java.lang.String-">FileLoggerImplementation(String, String)</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/FileLoggerImplementation.html" title="class in com.compomics.util.general">FileLoggerImplementation</a></dt>
<dd>
<div class="block">This constructor takes the name for the log file and the timestamp format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#fileName">fileName</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Spectrum file name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PtmToPrideMap.html#fileName">fileName</a></span> - Static variable in class com.compomics.util.pride.<a href="com/compomics/util/pride/PtmToPrideMap.html" title="class in com.compomics.util.pride">PtmToPrideMap</a></dt>
<dd>
<div class="block">The name of the file to save.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html#fileNameEnding">fileNameEnding</a></span> - Variable in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsFormat</a></dt>
<dd>
<div class="block">The file name ending of the format.</div>
</dd>
<dt><a href="com/compomics/util/io/file/FilenameExtensionFilter.html" title="class in com.compomics.util.io.file"><span class="typeNameLink">FilenameExtensionFilter</span></a> - Class in <a href="com/compomics/util/io/file/package-summary.html">com.compomics.util.io.file</a></dt>
<dd>
<div class="block">This class will implement a FilenameFilter that filters on the extension of files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FilenameExtensionFilter.html#FilenameExtensionFilter-java.lang.String-">FilenameExtensionFilter(String)</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FilenameExtensionFilter.html" title="class in com.compomics.util.io.file">FilenameExtensionFilter</a></dt>
<dd>
<div class="block">This constructor takes an extension to filter on.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FilenameExtensionFilter.html#FilenameExtensionFilter-java.lang.String-java.lang.String-">FilenameExtensionFilter(String, String)</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FilenameExtensionFilter.html" title="class in com.compomics.util.io.file">FilenameExtensionFilter</a></dt>
<dd>
<div class="block">This constructor takes an extension to filter on.</div>
</dd>
<dt><a href="com/compomics/util/gui/file_handling/FileSelectionDialog.html" title="class in com.compomics.util.gui.file_handling"><span class="typeNameLink">FileSelectionDialog</span></a> - Class in <a href="com/compomics/util/gui/file_handling/package-summary.html">com.compomics.util.gui.file_handling</a></dt>
<dd>
<div class="block">A simple dialog for selecting between different files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/file_handling/FileSelectionDialog.html#FileSelectionDialog-javax.swing.JFrame-java.util.ArrayList-java.lang.String-">FileSelectionDialog(JFrame, ArrayList&lt;File&gt;, String)</a></span> - Constructor for class com.compomics.util.gui.file_handling.<a href="com/compomics/util/gui/file_handling/FileSelectionDialog.html" title="class in com.compomics.util.gui.file_handling">FileSelectionDialog</a></dt>
<dd>
<div class="block">Creates a new FileSelection dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/file_handling/FileSelectionDialog.html#FileSelectionDialog-javax.swing.JDialog-java.util.ArrayList-java.lang.String-">FileSelectionDialog(JDialog, ArrayList&lt;File&gt;, String)</a></span> - Constructor for class com.compomics.util.gui.file_handling.<a href="com/compomics/util/gui/file_handling/FileSelectionDialog.html" title="class in com.compomics.util.gui.file_handling">FileSelectionDialog</a></dt>
<dd>
<div class="block">Creates a new FileSelection dialog.</div>
</dd>
<dt><a href="com/compomics/util/io/file/FileSystemAccessor.html" title="class in com.compomics.util.io.file"><span class="typeNameLink">FileSystemAccessor</span></a> - Class in <a href="com/compomics/util/io/file/package-summary.html">com.compomics.util.io.file</a></dt>
<dd>
<div class="block">This class serves as a fast OS independent access point to the file system.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FileSystemAccessor.html#FileSystemAccessor--">FileSystemAccessor()</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FileSystemAccessor.html" title="class in com.compomics.util.io.file">FileSystemAccessor</a></dt>
<dd>
<div class="block">Empty constructor.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/webservice/file/FileType.html" title="enum in com.compomics.util.pride.prideobjects.webservice.file"><span class="typeNameLink">FileType</span></a> - Enum in <a href="com/compomics/util/pride/prideobjects/webservice/file/package-summary.html">com.compomics.util.pride.prideobjects.webservice.file</a></dt>
<dd>
<div class="block">An enum of the possible file types for the PRIDE file type options.</div>
</dd>
<dt><a href="com/compomics/util/experiment/filtering/Filter.html" title="interface in com.compomics.util.experiment.filtering"><span class="typeNameLink">Filter</span></a> - Interface in <a href="com/compomics/util/experiment/filtering/package-summary.html">com.compomics.util.experiment.filtering</a></dt>
<dd>
<div class="block">Generic interface for a filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#filterAnnotations-java.util.Vector-java.util.HashMap-java.util.ArrayList-boolean-boolean-boolean-">filterAnnotations(Vector&lt;SpectrumAnnotation&gt;, HashMap&lt;Ion.IonType, HashSet&lt;Integer&gt;&gt;, ArrayList&lt;NeutralLoss&gt;, boolean, boolean, boolean)</a></span> - Static method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">used only in demo classes</span></div>
</div>
</dd>
<dt><a href="com/compomics/util/gui/filtering/FilterDialog.html" title="class in com.compomics.util.gui.filtering"><span class="typeNameLink">FilterDialog</span></a> - Class in <a href="com/compomics/util/gui/filtering/package-summary.html">com.compomics.util.gui.filtering</a></dt>
<dd>
<div class="block">Dialog to edit identification matches filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterDialog.html#FilterDialog-javax.swing.JDialog-com.compomics.util.experiment.identification.filtering.MatchFilter-java.util.ArrayList-">FilterDialog(JDialog, MatchFilter, ArrayList&lt;String&gt;)</a></span> - Constructor for class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterDialog.html" title="class in com.compomics.util.gui.filtering">FilterDialog</a></dt>
<dd>
<div class="block">Creates and displays a new dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#filterExists-java.lang.String-">filterExists(String)</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the name of this filter is already
 taken or not.</div>
</dd>
<dt><a href="com/compomics/util/experiment/filtering/FilterItem.html" title="interface in com.compomics.util.experiment.filtering"><span class="typeNameLink">FilterItem</span></a> - Interface in <a href="com/compomics/util/experiment/filtering/package-summary.html">com.compomics.util.experiment.filtering</a></dt>
<dd>
<div class="block">Interface for an item used for filtering.</div>
</dd>
<dt><a href="com/compomics/util/experiment/filtering/FilterItemComparator.html" title="enum in com.compomics.util.experiment.filtering"><span class="typeNameLink">FilterItemComparator</span></a> - Enum in <a href="com/compomics/util/experiment/filtering/package-summary.html">com.compomics.util.experiment.filtering</a></dt>
<dd>
<div class="block">Comparators for filter items.</div>
</dd>
<dt><a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering"><span class="typeNameLink">FilterParameters</span></a> - Class in <a href="com/compomics/util/gui/filtering/package-summary.html">com.compomics.util.gui.filtering</a></dt>
<dd>
<div class="block">This class groups the display filter preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#FilterParameters--">FilterParameters()</a></span> - Constructor for class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Constructors.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#filterTable--">filterTable()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>
<div class="block">Filters the selected values table according to the currently selected
 values and the current data filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#filterType">filterType</a></span> - Variable in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">The type of filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/FileLoggerImplementation.html#finalize--">finalize()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/FileLoggerImplementation.html" title="class in com.compomics.util.general">FileLoggerImplementation</a></dt>
<dd>
<div class="block">Called by the garbage collector on an object when garbage collection
 determines that there are no more references to the object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JTableForDB.html#finalize--">finalize()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/JTableForDB.html" title="class in com.compomics.util.gui">JTableForDB</a></dt>
<dd>
<div class="block">Called by the garbage collector on an object when garbage collection
 determines that there are no more references to the object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#findDeltaMassMatches-double-double-">findDeltaMassMatches(double, double)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method attempts to find a list of known mass deltas, corresponding
 with the specified x value in the given window.</div>
</dd>
<dt><a href="com/compomics/util/pdbfinder/FindPdbForUniprotAccessions.html" title="class in com.compomics.util.pdbfinder"><span class="typeNameLink">FindPdbForUniprotAccessions</span></a> - Class in <a href="com/compomics/util/pdbfinder/package-summary.html">com.compomics.util.pdbfinder</a></dt>
<dd>
<div class="block">Maps UniProt protein accession numbers to PDB file IDs.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/FindPdbForUniprotAccessions.html#FindPdbForUniprotAccessions-java.lang.String-com.compomics.util.waiting.WaitingHandler-">FindPdbForUniprotAccessions(String, WaitingHandler)</a></span> - Constructor for class com.compomics.util.pdbfinder.<a href="com/compomics/util/pdbfinder/FindPdbForUniprotAccessions.html" title="class in com.compomics.util.pdbfinder">FindPdbForUniprotAccessions</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/SwingWorker.html#finished--">finished()</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/SwingWorker.html" title="class in com.compomics.util.sun">SwingWorker</a></dt>
<dd>
<div class="block">Called on the event dispatching thread (not on the worker thread)
 after the <code>construct</code> method has returned.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#firstIndex-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">firstIndex(String, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the first index where the amino acid pattern is found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#firstIndex-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidSequence-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">firstIndex(AminoAcidSequence, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the first index where the amino acid pattern is found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#firstIndex-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">firstIndex(AminoAcidPattern, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the first index where the amino acid pattern is found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#firstIndex-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">firstIndex(String, SequenceMatchingParameters, int)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the first index where the amino acid pattern is found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#firstIndex-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">firstIndex(AminoAcidPattern, SequenceMatchingParameters, int)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the first index where the amino acid pattern is found in the
 given pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#firstIndex-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">firstIndex(String, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Returns the first index where the amino acid sequence is found in the
 given sequence.</div>
</dd>
<dt><a href="com/compomics/util/interfaces/Flamable.html" title="interface in com.compomics.util.interfaces"><span class="typeNameLink">Flamable</span></a> - Interface in <a href="com/compomics/util/interfaces/package-summary.html">com.compomics.util.interfaces</a></dt>
<dd>
<div class="block">This interface describes the behaviour for classes that can handle unrecoverable errors and
 exceptions, thrown by child threads etc.</div>
</dd>
<dt><a href="com/compomics/util/gui/FlamableJFrame.html" title="class in com.compomics.util.gui"><span class="typeNameLink">FlamableJFrame</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">FlamableJFrame.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/FlamableJFrame.html#FlamableJFrame--">FlamableJFrame()</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/FlamableJFrame.html" title="class in com.compomics.util.gui">FlamableJFrame</a></dt>
<dd>
<div class="block">Wrapper constructor for that of the superclass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/FlamableJFrame.html#FlamableJFrame-java.lang.String-">FlamableJFrame(String)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/FlamableJFrame.html" title="class in com.compomics.util.gui">FlamableJFrame</a></dt>
<dd>
<div class="block">Wrapper constructor for that of the superclass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/FlamableJFrame.html#FlamableJFrame-java.awt.GraphicsConfiguration-">FlamableJFrame(GraphicsConfiguration)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/FlamableJFrame.html" title="class in com.compomics.util.gui">FlamableJFrame</a></dt>
<dd>
<div class="block">Wrapper constructor for that of the superclass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/FlamableJFrame.html#FlamableJFrame-java.awt.GraphicsConfiguration-java.lang.String-">FlamableJFrame(GraphicsConfiguration, String)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/FlamableJFrame.html" title="class in com.compomics.util.gui">FlamableJFrame</a></dt>
<dd>
<div class="block">Wrapper constructor for that of the superclass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#FLOAT_PRECISION">FLOAT_PRECISION</a></span> - Static variable in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Defines the valid String indicating the correct precision for encoded
 floats.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#floorDouble-double-int-">floorDouble(double, int)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Floors a double value to the wanted number of decimal places.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/Fluorine.html" title="class in com.compomics.util.experiment.biology.atoms.impl"><span class="typeNameLink">Fluorine</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a></dt>
<dd>
<div class="block">The fluorine atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/impl/Fluorine.html#Fluorine--">Fluorine()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.impl.<a href="com/compomics/util/experiment/biology/atoms/impl/Fluorine.html" title="class in com.compomics.util.experiment.biology.atoms.impl">Fluorine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index"><span class="typeNameLink">FMIndex</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_inference/fm_index/package-summary.html">com.compomics.util.experiment.identification.protein_inference.fm_index</a></dt>
<dd>
<div class="block">The FM index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#FMIndex-java.io.File-com.compomics.util.experiment.io.biology.protein.FastaParameters-com.compomics.util.waiting.WaitingHandler-boolean-com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters-com.compomics.util.parameters.identification.search.SearchParameters-">FMIndex(File, FastaParameters, WaitingHandler, boolean, PeptideVariantsParameters, SearchParameters)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#FMIndex-java.io.File-com.compomics.util.experiment.io.biology.protein.FastaParameters-com.compomics.util.waiting.WaitingHandler-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters-">FMIndex(File, FastaParameters, WaitingHandler, boolean, ModificationParameters, PeptideVariantsParameters)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.MassIndexMap.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index"><span class="typeNameLink">FMIndex.MassIndexMap</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_inference/fm_index/package-summary.html">com.compomics.util.experiment.identification.protein_inference.fm_index</a></dt>
<dd>
<div class="block">struct for building own data structure for mass to index mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/Units.html#fmol">fmol</a></span> - Static variable in class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/Units.html" title="class in com.compomics.util.experiment.units">Units</a></dt>
<dd>
<div class="block">Femtomol.</div>
</dd>
<dt><a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file"><span class="typeNameLink">FolderMonitor</span></a> - Class in <a href="com/compomics/util/io/file/package-summary.html">com.compomics.util.io.file</a></dt>
<dd>
<div class="block">This class will monitor a specified folder for activity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#FolderMonitor-java.io.File-long-int-java.util.HashMap-">FolderMonitor(File, long, int, HashMap)</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>
<div class="block">This constructor allows the creation of a FolderMonitor that will take care
 of performing a specified action, with the specified parameters, whenever a new
 file is found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#FolderMonitor-java.io.File-long-int-java.util.HashMap-org.apache.log4j.Logger-">FolderMonitor(File, long, int, HashMap, Logger)</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>
<div class="block">This constructor allows the creation of a FolderMonitor that will take care
 of performing a specified action, with the specified parameters, whenever a new
 file is found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#FolderMonitor-java.io.File-long-java.lang.String-int-java.util.HashMap-">FolderMonitor(File, long, String, int, HashMap)</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>
<div class="block">This constructor allows the creation of a FolderMonitor for the specified filtered files that will take care
 of performing a specified action, with the specified parameters, whenever a new
 file is found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#FolderMonitor-java.io.File-long-java.lang.String-int-java.util.HashMap-org.apache.log4j.Logger-">FolderMonitor(File, long, String, int, HashMap, Logger)</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>
<div class="block">This constructor allows the creation of a FolderMonitor for the specified filtered files that will take care
 of performing a specified action, with the specified parameters, whenever a new
 file is found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html#forbiddenCharacter">forbiddenCharacter</a></span> - Static variable in class com.compomics.util.experiment.io.biology.protein.iterators.<a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html" title="class in com.compomics.util.experiment.io.biology.protein.iterators">FastaIterator</a></dt>
<dd>
<div class="block">Character forbidden in protein sequences, will be removed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#forbiddenCharacters">forbiddenCharacters</a></span> - Static variable in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Forbidden characters in file names.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/examples/UtilitiesDemo.html#formatProteinSequence--">formatProteinSequence()</a></span> - Method in class com.compomics.util.examples.<a href="com/compomics/util/examples/UtilitiesDemo.html" title="class in com.compomics.util.examples">UtilitiesDemo</a></dt>
<dd>
<div class="block">Formats the protein sequence such that both the covered parts of the
 sequence and the peptide selected in the peptide table is highlighted,
 and any modifications and variants of a PEFF file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ProteinSequencePane.html#formatProteinSequence-javax.swing.JEditorPane-java.lang.String-int:A-java.util.TreeMap-java.util.HashMap-boolean-boolean-boolean-">formatProteinSequence(JEditorPane, String, int[], TreeMap&lt;String, String&gt;, HashMap&lt;String, Color&gt;, boolean, boolean, boolean)</a></span> - Static method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ProteinSequencePane.html" title="class in com.compomics.util.gui.protein">ProteinSequencePane</a></dt>
<dd>
<div class="block">Formats the protein sequence such that the covered parts of the sequence
 is highlighted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ProteinSequencePane.html#formatProteinSequence-javax.swing.JEditorPane-java.lang.String-int:A-">formatProteinSequence(JEditorPane, String, int[])</a></span> - Static method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ProteinSequencePane.html" title="class in com.compomics.util.gui.protein">ProteinSequencePane</a></dt>
<dd>
<div class="block">Formats the protein sequence such that the covered parts of the sequence
 is highlighted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ProteinSequencePane.html#formatProteinSequence-javax.swing.JEditorPane-java.lang.String-int-int-int:A-">formatProteinSequence(JEditorPane, String, int, int, int[])</a></span> - Static method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ProteinSequencePane.html" title="class in com.compomics.util.gui.protein">ProteinSequencePane</a></dt>
<dd>
<div class="block">Formats the protein sequence such that the covered parts of the sequence
 is highlighted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ProteinSequencePane.html#formatProteinSequence-javax.swing.JEditorPane-java.lang.String-java.util.ArrayList-java.util.ArrayList-int:A-">formatProteinSequence(JEditorPane, String, ArrayList&lt;Integer&gt;, ArrayList&lt;Integer&gt;, int[])</a></span> - Static method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ProteinSequencePane.html" title="class in com.compomics.util.gui.protein">ProteinSequencePane</a></dt>
<dd>
<div class="block">Formats the protein sequence such that both the covered parts of the sequence
 and the peptide selected in the peptide table is highlighted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ProteinSequencePane.html#formatProteinSequence-javax.swing.JEditorPane-java.lang.String-int-int-int:A-java.util.TreeMap-java.util.HashMap-">formatProteinSequence(JEditorPane, String, int, int, int[], TreeMap&lt;String, String&gt;, HashMap&lt;String, Color&gt;)</a></span> - Static method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ProteinSequencePane.html" title="class in com.compomics.util.gui.protein">ProteinSequencePane</a></dt>
<dd>
<div class="block">Formats the protein sequence such that both the covered parts of the sequence
 and the peptide selected in the peptide table is highlighted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/list_choosers/ModificationChooser.html#formatTable--">formatTable()</a></span> - Method in class com.compomics.util.gui.utils.user_choice.list_choosers.<a href="com/compomics/util/gui/utils/user_choice/list_choosers/ModificationChooser.html" title="class in com.compomics.util.gui.utils.user_choice.list_choosers">ModificationChooser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/list_choosers/StringListChooser.html#formatTable--">formatTable()</a></span> - Method in class com.compomics.util.gui.utils.user_choice.list_choosers.<a href="com/compomics/util/gui/utils/user_choice/list_choosers/StringListChooser.html" title="class in com.compomics.util.gui.utils.user_choice.list_choosers">StringListChooser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/ListChooser.html#formatTable--">formatTable()</a></span> - Method in class com.compomics.util.gui.utils.user_choice.<a href="com/compomics/util/gui/utils/user_choice/ListChooser.html" title="class in com.compomics.util.gui.utils.user_choice">ListChooser</a></dt>
<dd>
<div class="block">Formats the table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandLineUtils.html#formatter">formatter</a></span> - Static variable in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli">CommandLineUtils</a></dt>
<dd>
<div class="block">The format to use for string output.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#FORMYL_K">FORMYL_K</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for formylation of K (PMID: 24895383).</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">ForwardIonAminoAcidFeature</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">Feature based on the amino acid properties of the forward ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonAminoAcidFeature.html#ForwardIonAminoAcidFeature-com.compomics.util.experiment.biology.aminoacids.AminoAcid.Property-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.AASequenceFeature.Function-">ForwardIonAminoAcidFeature(AminoAcid.Property, AASequenceFeature.Function)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ForwardIonAminoAcidFeature</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">ForwardIonFeature</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">Feature based on a property of the forward ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonFeature.html#ForwardIonFeature-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.IonFeature.Property-">ForwardIonFeature(IonFeature.Property)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ForwardIonFeature</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.IonSeries.html#forwardSeries">forwardSeries</a></span> - Variable in enum com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.IonSeries.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">SimplePeptideAnnotator.IonSeries</a></dt>
<dd>
<div class="block">The forward ion index as listed in the PeptideFragmentIon class.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html" title="class in com.compomics.util.parameters.identification.advanced"><span class="typeNameLink">FractionParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/advanced/package-summary.html">com.compomics.util.parameters.identification.advanced</a></dt>
<dd>
<div class="block">Settings for the handling of fractions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html#FractionParameters--">FractionParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html" title="class in com.compomics.util.parameters.identification.advanced">FractionParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/advanced/FractionParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced"><span class="typeNameLink">FractionParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/advanced/package-summary.html">com.compomics.util.gui.parameters.identification.advanced</a></dt>
<dd>
<div class="block">FractionSettingsDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/FractionParametersDialog.html#FractionParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.FractionParameters-boolean-">FractionParametersDialog(Frame, FractionParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/FractionParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">FractionParametersDialog</a></dt>
<dd>
<div class="block">Creates a new FractionSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/FractionParametersDialog.html#FractionParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.FractionParameters-boolean-">FractionParametersDialog(Dialog, Frame, FractionParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/FractionParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">FractionParametersDialog</a></dt>
<dd>
<div class="block">Creates a new FractionSettingsDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators"><span class="typeNameLink">FragmentAnnotator</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators</a></dt>
<dd>
<div class="block">Annotator for b and y ions without neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotator.html#FragmentAnnotator-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.SimplePeptideAnnotator.IonSeries-">FragmentAnnotator(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, SimplePeptideAnnotator.IonSeries)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">FragmentAnnotator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotator.html#FragmentAnnotator-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.SimplePeptideAnnotator.IonSeries-boolean-boolean-">FragmentAnnotator(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, SimplePeptideAnnotator.IonSeries, boolean, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">FragmentAnnotator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotatorNL.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators"><span class="typeNameLink">FragmentAnnotatorNL</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators</a></dt>
<dd>
<div class="block">A fragment annotator for peptide fragment ions with neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotatorNL.html#FragmentAnnotatorNL-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.SimplePeptideAnnotator.IonSeries-boolean-">FragmentAnnotatorNL(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, SimplePeptideAnnotator.IonSeries, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotatorNL.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">FragmentAnnotatorNL</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotatorNL.html#FragmentAnnotatorNL-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.SimplePeptideAnnotator.IonSeries-boolean-boolean-boolean-">FragmentAnnotatorNL(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, SimplePeptideAnnotator.IonSeries, boolean, boolean, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotatorNL.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">FragmentAnnotatorNL</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/FragmentationMethod.html" title="enum in com.compomics.util.experiment.mass_spectrometry"><span class="typeNameLink">FragmentationMethod</span></a> - Enum in <a href="com/compomics/util/experiment/mass_spectrometry/package-summary.html">com.compomics.util.experiment.mass_spectrometry</a></dt>
<dd>
<div class="block">Enum for the different fragmentation methods.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#fragmentFactory">fragmentFactory</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">The Fragment factory which will generate the fragment ions.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/FragmentIonTable.html" title="class in com.compomics.util.gui.spectrum"><span class="typeNameLink">FragmentIonTable</span></a> - Class in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">Creates a fragment ion table with the detected fragment ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/FragmentIonTable.html#FragmentIonTable-com.compomics.util.experiment.biology.proteins.Peptide-java.util.ArrayList-java.util.HashSet-com.compomics.util.experiment.identification.spectrum_annotation.NeutralLossesMap-boolean-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">FragmentIonTable(Peptide, ArrayList&lt;IonMatch[]&gt;, HashSet&lt;Integer&gt;, NeutralLossesMap, boolean, boolean, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/FragmentIonTable.html" title="class in com.compomics.util.gui.spectrum">FragmentIonTable</a></dt>
<dd>
<div class="block">Creates a traditional fragment ion table with the theoretical mz values
 and the detected fragment ions highlighted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/FragmentIonTable.html#FragmentIonTable-com.compomics.util.experiment.biology.proteins.Peptide-java.util.ArrayList-java.util.ArrayList-java.util.HashSet-com.compomics.util.experiment.identification.spectrum_annotation.NeutralLossesMap-boolean-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">FragmentIonTable(Peptide, ArrayList&lt;IonMatch[]&gt;, ArrayList&lt;Spectrum&gt;, HashSet&lt;Integer&gt;, NeutralLossesMap, boolean, boolean, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/FragmentIonTable.html" title="class in com.compomics.util.gui.spectrum">FragmentIonTable</a></dt>
<dd>
<div class="block">Creates a novel fragment ion table displaying bar charts with the
 intensity of each fragment ion type.</div>
</dd>
<dt><a href="com/compomics/util/gui/renderers/FragmentIonTableCellRenderer.html" title="class in com.compomics.util.gui.renderers"><span class="typeNameLink">FragmentIonTableCellRenderer</span></a> - Class in <a href="com/compomics/util/gui/renderers/package-summary.html">com.compomics.util.gui.renderers</a></dt>
<dd>
<div class="block">A cell renderer to use for the FragmentIonTable making it possible to
 highlight certain cells.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/FragmentIonTableCellRenderer.html#FragmentIonTableCellRenderer-java.util.ArrayList-java.awt.Color-java.awt.Color-">FragmentIonTableCellRenderer(ArrayList&lt;Integer&gt;, Color, Color)</a></span> - Constructor for class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/FragmentIonTableCellRenderer.html" title="class in com.compomics.util.gui.renderers">FragmentIonTableCellRenderer</a></dt>
<dd>
<div class="block">Creates a new FragmentIonTableCellRenderer.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/FragmentItensityPredictor.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation"><span class="typeNameLink">FragmentItensityPredictor</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation</a></dt>
<dd>
<div class="block">Predictor for the peptide fragment intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/FragmentItensityPredictor.html#FragmentItensityPredictor-com.compomics.util.experiment.identification.peptide_fragmentation.PeptideFragmentationModel-">FragmentItensityPredictor(PeptideFragmentationModel)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/FragmentItensityPredictor.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation">FragmentItensityPredictor</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/exceptions/exception_handlers/FrameExceptionHandler.html" title="class in com.compomics.util.exceptions.exception_handlers"><span class="typeNameLink">FrameExceptionHandler</span></a> - Class in <a href="com/compomics/util/exceptions/exception_handlers/package-summary.html">com.compomics.util.exceptions.exception_handlers</a></dt>
<dd>
<div class="block">Handles exception for a given application and displays warnings using
 dialogs.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/exception_handlers/FrameExceptionHandler.html#FrameExceptionHandler-javax.swing.JFrame-java.lang.String-">FrameExceptionHandler(JFrame, String)</a></span> - Constructor for class com.compomics.util.exceptions.exception_handlers.<a href="com/compomics/util/exceptions/exception_handlers/FrameExceptionHandler.html" title="class in com.compomics.util.exceptions.exception_handlers">FrameExceptionHandler</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#fromJson-java.lang.Class-java.lang.String-">fromJson(Class, String)</a></span> - Method in class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">Convert from JSON to object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#fromJson-java.lang.Class-java.io.File-">fromJson(Class, File)</a></span> - Method in class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">Convert from JSON to object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#fromJson-java.lang.Class-java.net.URL-">fromJson(Class, URL)</a></span> - Method in class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">Convert from JSON to object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#fromJson-java.lang.reflect.Type-java.lang.String-">fromJson(Type, String)</a></span> - Method in class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">Convert from JSON to object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#fromJson-java.lang.reflect.Type-java.io.File-">fromJson(Type, File)</a></span> - Method in class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">Convert from JSON to object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#fromJson-java.lang.reflect.Type-java.net.URL-">fromJson(Type, URL)</a></span> - Method in class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">Convert from JSON to object.</div>
</dd>
<dt><a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp"><span class="typeNameLink">FTP</span></a> - Class in <a href="com/compomics/util/io/ftp/package-summary.html">com.compomics.util.io.ftp</a></dt>
<dd>
<div class="block">This class was modified from a source found on the net (java.sun.com, search in
 developer section on 'FtpProtocolException PORT').</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#FTP-java.lang.String-">FTP(String)</a></span> - Constructor for class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Constructor for an FTP client connected to host <i>host</i>.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#FTP-java.lang.String-int-">FTP(String, int)</a></span> - Constructor for class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Constructor for an FTP client connected to host <i>host</i>
 and port <i>port</i>.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#FTP_PORT">FTP_PORT</a></span> - Static variable in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">FTP port to use for connection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#FTP_TO_SPECIFIED_DESTINATION">FTP_TO_SPECIFIED_DESTINATION</a></span> - Static variable in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/ftp/FTPClient.html" title="class in com.compomics.util.io.ftp"><span class="typeNameLink">FTPClient</span></a> - Class in <a href="com/compomics/util/io/ftp/package-summary.html">com.compomics.util.io.ftp</a></dt>
<dd>
<div class="block">This class implements a 'directory listener' to listen for new files and
 ftp them to a remote FTP server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTPClient.html#FTPClient-java.lang.String-java.lang.String-java.lang.String-">FTPClient(String, String, String)</a></span> - Constructor for class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTPClient.html" title="class in com.compomics.util.io.ftp">FTPClient</a></dt>
<dd>
<div class="block">Constructor that takes the three parameters necessary for the
 FTP connection.</div>
</dd>
<dt><a href="com/compomics/util/io/ftp/FtpConnectionException.html" title="class in com.compomics.util.io.ftp"><span class="typeNameLink">FtpConnectionException</span></a> - Exception in <a href="com/compomics/util/io/ftp/package-summary.html">com.compomics.util.io.ftp</a></dt>
<dd>
<div class="block">This class handles an FTP connection exception.</div>
</dd>
<dt><a href="com/compomics/util/io/ftp/FTPDownloader.html" title="class in com.compomics.util.io.ftp"><span class="typeNameLink">FTPDownloader</span></a> - Class in <a href="com/compomics/util/io/ftp/package-summary.html">com.compomics.util.io.ftp</a></dt>
<dd>
<div class="block">A simple FTP file downloader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTPDownloader.html#FTPDownloader-java.lang.String-">FTPDownloader(String)</a></span> - Constructor for class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTPDownloader.html" title="class in com.compomics.util.io.ftp">FTPDownloader</a></dt>
<dd>
<div class="block">Set up an anonymous FTP connection (without protocol commands printed).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTPDownloader.html#FTPDownloader-java.lang.String-boolean-">FTPDownloader(String, boolean)</a></span> - Constructor for class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTPDownloader.html" title="class in com.compomics.util.io.ftp">FTPDownloader</a></dt>
<dd>
<div class="block">Set up an anonymous FTP connection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTPDownloader.html#FTPDownloader-java.lang.String-java.lang.String-java.lang.String-boolean-">FTPDownloader(String, String, String, boolean)</a></span> - Constructor for class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTPDownloader.html" title="class in com.compomics.util.io.ftp">FTPDownloader</a></dt>
<dd>
<div class="block">Set up an FTP connection.</div>
</dd>
<dt><a href="com/compomics/util/io/ftp/FtpLoginException.html" title="class in com.compomics.util.io.ftp"><span class="typeNameLink">FtpLoginException</span></a> - Exception in <a href="com/compomics/util/io/ftp/package-summary.html">com.compomics.util.io.ftp</a></dt>
<dd>
<div class="block">This class handles an FTP login exception.</div>
</dd>
<dt><a href="com/compomics/util/io/ftp/FtpProtocolException.html" title="class in com.compomics.util.io.ftp"><span class="typeNameLink">FtpProtocolException</span></a> - Exception in <a href="com/compomics/util/io/ftp/package-summary.html">com.compomics.util.io.ftp</a></dt>
<dd>
<div class="block">This class handles an FTP protocol exception.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/StandardUnit.html#FULL_NAME">FULL_NAME</a></span> - Variable in enum com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/StandardUnit.html" title="enum in com.compomics.util.experiment.units">StandardUnit</a></dt>
<dd>
<div class="block">The full name of the unit.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html#fullName">fullName</a></span> - Variable in enum com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html" title="enum in com.compomics.util.experiment.io.biology.protein">ProteinDatabase</a></dt>
<dd>
<div class="block">The full name of the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#FULLY_ENZYMATIC">FULLY_ENZYMATIC</a></span> - Static variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html#function">function</a></span> - Variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AASequenceFeature</a></dt>
<dd>
<div class="block">The function used to compare the amino acid properties.</div>
</dd>
</dl>
<a name="I:G">
<!--   -->
</a>
<h2 class="title">G</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#G">G</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#GATHER_FILES_FOR_PICKUP">GATHER_FILES_FOR_PICKUP</a></span> - Static variable in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#GENE_MAPPING_FILE_SUFFIX">GENE_MAPPING_FILE_SUFFIX</a></span> - Static variable in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">The suffix to use for files containing gene mappings.</div>
</dd>
<dt><a href="com/compomics/util/gui/genes/GeneDetailsDialog.html" title="class in com.compomics.util.gui.genes"><span class="typeNameLink">GeneDetailsDialog</span></a> - Class in <a href="com/compomics/util/gui/genes/package-summary.html">com.compomics.util.gui.genes</a></dt>
<dd>
<div class="block">This dialog displays the gene details associated to a protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/genes/GeneDetailsDialog.html#GeneDetailsDialog-java.awt.Frame-com.compomics.util.experiment.identification.matches.ProteinMatch-com.compomics.util.experiment.biology.genes.GeneMaps-com.compomics.util.experiment.io.biology.protein.ProteinDetailsProvider-">GeneDetailsDialog(Frame, ProteinMatch, GeneMaps, ProteinDetailsProvider)</a></span> - Constructor for class com.compomics.util.gui.genes.<a href="com/compomics/util/gui/genes/GeneDetailsDialog.html" title="class in com.compomics.util.gui.genes">GeneDetailsDialog</a></dt>
<dd>
<div class="block">Creates a new GeneDetailsDialog.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html" title="class in com.compomics.util.experiment.biology.genes.ensembl"><span class="typeNameLink">GeneMapping</span></a> - Class in <a href="com/compomics/util/experiment/biology/genes/ensembl/package-summary.html">com.compomics.util.experiment.biology.genes.ensembl</a></dt>
<dd>
<div class="block">Class for the handling of gene mappings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html#GeneMapping--">GeneMapping()</a></span> - Constructor for class com.compomics.util.experiment.biology.genes.ensembl.<a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html" title="class in com.compomics.util.experiment.biology.genes.ensembl">GeneMapping</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes"><span class="typeNameLink">GeneMaps</span></a> - Class in <a href="com/compomics/util/experiment/biology/genes/package-summary.html">com.compomics.util.experiment.biology.genes</a></dt>
<dd>
<div class="block">The gene maps for a given project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#GeneMaps--">GeneMaps()</a></span> - Constructor for class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Creates new maps.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html" title="class in com.compomics.util.parameters.identification.advanced"><span class="typeNameLink">GeneParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/advanced/package-summary.html">com.compomics.util.parameters.identification.advanced</a></dt>
<dd>
<div class="block">Contains methods for downloading gene and GO mappings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html#GeneParameters--">GeneParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html" title="class in com.compomics.util.parameters.identification.advanced">GeneParameters</a></dt>
<dd>
<div class="block">Create a new GenePreferences object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html#GeneParameters-com.compomics.util.parameters.identification.advanced.GeneParameters-">GeneParameters(GeneParameters)</a></span> - Constructor for class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html" title="class in com.compomics.util.parameters.identification.advanced">GeneParameters</a></dt>
<dd>
<div class="block">Creates new gene preferences based on a GenePreferences object.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/advanced/GeneParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced"><span class="typeNameLink">GeneParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/advanced/package-summary.html">com.compomics.util.gui.parameters.identification.advanced</a></dt>
<dd>
<div class="block">Dialog for editing the Gene Mapping Preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/GeneParametersDialog.html#GeneParametersDialog-javax.swing.JFrame-com.compomics.util.parameters.identification.advanced.GeneParameters-com.compomics.util.parameters.identification.search.SearchParameters-boolean-">GeneParametersDialog(JFrame, GeneParameters, SearchParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/GeneParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">GeneParametersDialog</a></dt>
<dd>
<div class="block">Creates a new GenePreferencesDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/GeneParametersDialog.html#GeneParametersDialog-javax.swing.JDialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.GeneParameters-com.compomics.util.parameters.identification.search.SearchParameters-boolean-">GeneParametersDialog(JDialog, Frame, GeneParameters, SearchParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/GeneParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">GeneParametersDialog</a></dt>
<dd>
<div class="block">Creates a new GenePreferencesDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#generateDecoy--">generateDecoy()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns whether a decoy database shall be created and searched against.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#generateQuery--">generateQuery()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether a blast query shall be generated.</div>
</dd>
<dt><a href="com/compomics/util/db/GeneratorException.html" title="class in com.compomics.util.db"><span class="typeNameLink">GeneratorException</span></a> - Exception in <a href="com/compomics/util/db/package-summary.html">com.compomics.util.db</a></dt>
<dd>
<div class="block">This class wraps all exception that can occur while generating code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/GeneratorException.html#GeneratorException-java.lang.String-">GeneratorException(String)</a></span> - Constructor for exception com.compomics.util.db.<a href="com/compomics/util/db/GeneratorException.html" title="class in com.compomics.util.db">GeneratorException</a></dt>
<dd>
<div class="block">This constructor creates an exception with just a message.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/GeneratorException.html#GeneratorException-java.lang.String-java.lang.Exception-">GeneratorException(String, Exception)</a></span> - Constructor for exception com.compomics.util.db.<a href="com/compomics/util/db/GeneratorException.html" title="class in com.compomics.util.db">GeneratorException</a></dt>
<dd>
<div class="block">This constructor creates an exception with a message and a nested exception.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/converters/GenericFastaConverter.html" title="class in com.compomics.util.experiment.io.biology.protein.converters"><span class="typeNameLink">GenericFastaConverter</span></a> - Class in <a href="com/compomics/util/experiment/io/biology/protein/converters/package-summary.html">com.compomics.util.experiment.io.biology.protein.converters</a></dt>
<dd>
<div class="block">This converter writes a FASTA file with standardized headers.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/converters/GenericFastaConverter.html#GenericFastaConverter--">GenericFastaConverter()</a></span> - Constructor for class com.compomics.util.experiment.io.biology.protein.converters.<a href="com/compomics/util/experiment/io/biology/protein/converters/GenericFastaConverter.html" title="class in com.compomics.util.experiment.io.biology.protein.converters">GenericFastaConverter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#genericMzId">genericMzId</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Advocate type for mzId files where no software is annotated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/SwingWorker.html#get--">get()</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/SwingWorker.html" title="class in com.compomics.util.sun">SwingWorker</a></dt>
<dd>
<div class="block">Return the value created by the <code>construct</code> method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#getAaAfter-com.compomics.util.experiment.biology.proteins.Peptide-java.lang.String-int-int-com.compomics.util.experiment.io.biology.protein.SequenceProvider-">getAaAfter(Peptide, String, int, int, SequenceProvider)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns the amino acids before the given peptide as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#getAaAfter-com.compomics.util.experiment.biology.proteins.Peptide-int-com.compomics.util.experiment.io.biology.protein.SequenceProvider-">getAaAfter(Peptide, int, SequenceProvider)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns the amino acids before the given peptide as a string in a map
 based on the peptide protein mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#getAaBefore-com.compomics.util.experiment.biology.proteins.Peptide-java.lang.String-int-int-com.compomics.util.experiment.io.biology.protein.SequenceProvider-">getAaBefore(Peptide, String, int, int, SequenceProvider)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns the amino acids before the given peptide as a string in a map
 based on the peptide protein mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#getAaBefore-com.compomics.util.experiment.biology.proteins.Peptide-int-com.compomics.util.experiment.io.biology.protein.SequenceProvider-">getAaBefore(Peptide, int, SequenceProvider)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns the amino acids before the given peptide as a string in a map
 based on the peptide protein mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getAACoverage-long-">getAACoverage(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Indicates the validation level of every amino acid in the given protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html#getAaIndex--">getAaIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAIdentityFeature</a></dt>
<dd>
<div class="block">Returns the index on the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html#getAaIndex--">getAaIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAPropertyFeature</a></dt>
<dd>
<div class="block">Returns the index on the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#getAaNumber-int-">getAaNumber(int)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Returns the 1 based index of the amino acid which generated this ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#getAaSubstitutionMatrix--">getAaSubstitutionMatrix()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Returns the amino acid substitution matrix to use.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getAaTargeted--">getAaTargeted()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the map of targeted amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getAbbreviatedFASTAHeader--">getAbbreviatedFASTAHeader()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method returns an abbreviated version of the Header, suitable for
 inclusion in FASTA formatted files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getAbbreviatedFASTAHeader-java.lang.String-">getAbbreviatedFASTAHeader(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method returns an abbreviated version of the Header, suitable for
 inclusion in FASTA formatted files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getAbbreviatedFASTAHeaderWithAddenda--">getAbbreviatedFASTAHeaderWithAddenda()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method returns an abbreviated version of the Header, suitable for
 inclusion in FASTA formatted files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/UnitOfMeasurement.html#getAbbreviation--">getAbbreviation()</a></span> - Method in class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/UnitOfMeasurement.html" title="class in com.compomics.util.experiment.units">UnitOfMeasurement</a></dt>
<dd>
<div class="block">Returns the abbreviated name of the unit.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getAbsoluteError--">getAbsoluteError()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Get the absolute matching error in Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getAbsoluteError-int-int-">getAbsoluteError(int, int)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Get the absolute matching error in Da after isotope removal.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/MavenJarFile.html#getAbsoluteFilePath--">getAbsoluteFilePath()</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/MavenJarFile.html" title="class in com.compomics.software.autoupdater">MavenJarFile</a></dt>
<dd>
<div class="block">Returns the absolute file path.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Protein.html#getAccession--">getAccession()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Protein.html" title="class in com.compomics.util.experiment.biology.proteins">Protein</a></dt>
<dd>
<div class="block">Getter for the protein accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getAccession--">getAccession()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/CvTerm.html#getAccession--">getAccession()</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/CvTerm.html" title="class in com.compomics.util.pride">CvTerm</a></dt>
<dd>
<div class="block">Returns the accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getAccessionOrRest--">getAccessionOrRest()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the accession or if this is null the rest.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#getAccessions--">getAccessions()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Returns the accessions of the proteins in this match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getAccessions--">getAccessions()</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html#getAccessions--">getAccessions()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html" title="class in com.compomics.util.experiment.identification.protein_sequences">SingleProteinSequenceProvider</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getAccessions--">getAccessions()</a></span> - Method in interface com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></dt>
<dd>
<div class="block">Returns all accessions loaded in the provider.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html#getAccountedNeutralLosses--">getAccountedNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">NeutralLossesMap</a></dt>
<dd>
<div class="block">Returns an arraylist of the names of the implemented neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#getActicationType--">getActicationType()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Returns the activation type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getAddenda--">getAddenda()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method allows the caller to retrieve all addenda for the current
 header, or 'null' if there aren't any.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#getAdvocate-int-">getAdvocate(int)</a></span> - Static method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Returns the advocate corresponding to the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#getAdvocate-java.lang.String-">getAdvocate(String)</a></span> - Static method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Returns the advocate with the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#getAdvocate--">getAdvocate()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Get the used advocate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#getAdvocateColorMap--">getAdvocateColorMap()</a></span> - Static method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Returns the advocate color map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#getAdvocateFromFile-java.lang.String-">getAdvocateFromFile(String)</a></span> - Static method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">no longer used</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#getAdvocates--">getAdvocates()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Returns the advocates with a specific scoring.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#getAdvocateToolTipMap--">getAdvocateToolTipMap()</a></span> - Static method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Returns the advocate tool tip map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationAlgorithmParameter.html#getAlgorithm--">getAlgorithm()</a></span> - Method in interface com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationAlgorithmParameter.html" title="interface in com.compomics.util.gui.parameters.identification">IdentificationAlgorithmParameter</a></dt>
<dd>
<div class="block">Returns the identification algorithm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getAlgorithm--">getAlgorithm()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getAlgorithm--">getAlgorithm()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getAlgorithm--">getAlgorithm()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#getAlgorithm--">getAlgorithm()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#getAlgorithm--">getAlgorithm()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getAlgorithm--">getAlgorithm()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html#getAlgorithm--">getAlgorithm()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">NovorParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getAlgorithm--">getAlgorithm()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#getAlgorithm--">getAlgorithm()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#getAlgorithm--">getAlgorithm()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getAlgorithm--">getAlgorithm()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#getAlgorithm--">getAlgorithm()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getAlgorithmDeltaPEP--">getAlgorithmDeltaPEP()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the difference in identification algorithm level PEP with the
 next best peptide assumption with sequence difference for the given
 search engine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getAlgorithms--">getAlgorithms()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the algorithms for which specific parameters are stored.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getAlgorithmSpecificParameters--">getAlgorithmSpecificParameters()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the algorithm specific parameters in a map: algorithm as indexed
 in the Advocate class &gt; parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html#getAlignmentBlocks--">getAlignmentBlocks()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAlignment</a></dt>
<dd>
<div class="block">Returns the alignment blocks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#getAlignNonConfidentModifications--">getAlignNonConfidentModifications()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Indicates whether the non confidently localized modifications should be aligned on
 the confident sites.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAnnotationServerAlingmentReader.html#getAllAlignments--">getAllAlignments()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAnnotationServerAlingmentReader.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAnnotationServerAlingmentReader</a></dt>
<dd>
<div class="block">Get all alignment in the XML string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAnnotationServerResultReader.html#getAllFeatures--">getAllFeatures()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAnnotationServerResultReader.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAnnotationServerResultReader</a></dt>
<dd>
<div class="block">Get all features in the XML string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicElement.html#getAllIsotopicElements-java.lang.Class-org.apache.log4j.Logger-">getAllIsotopicElements(Class, Logger)</a></span> - Static method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicElement.html" title="class in com.compomics.util.general">IsotopicElement</a></dt>
<dd>
<div class="block">Static method that gives all the isotopicElements from the isotopicElement.txt file</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#getAllModifications-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getAllModifications(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">Returns a set of the names of all modifications found on a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#getAllModifications-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getAllModifications(Tag, ModificationParameters, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">Returns a set of the names of all modifications found on a tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#getAllModifications--">getAllModifications()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Returns a list of all searched modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationFactory.html#getAllModificationTitles--">getAllModificationTitles()</a></span> - Static method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationFactory.html" title="class in com.compomics.util.protein">ModificationFactory</a></dt>
<dd>
<div class="block">This method returns all modification titles known to the Factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#getAllNotFixedModifications--">getAllNotFixedModifications()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Returns a list of all searched modifications but the fixed ones.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getAllocatedBytes--">getAllocatedBytes()</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Computes the number of allocated bytes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html#getAllocatedBytes--">getAllocatedBytes()</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">Rank</a></dt>
<dd>
<div class="block">Returns the number of bytes for the allocated arrays.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#getAllocatedBytes--">getAllocatedBytes()</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Returns the number of bytes for the allocated arrays.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#getAllPeptideAssumptions-int-">getAllPeptideAssumptions(int)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Returns all peptide assumptions for the specified search engine indexed by their
 score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#getAllPeptideAssumptions--">getAllPeptideAssumptions()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Returns a stream of all peptide assumptions</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getAllPossibleSequences--">getAllPossibleSequences()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns all possible sequences which can be obtained from the targeted
 amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/interfaces/DBElement.html#getAllProperties--">getAllProperties()</a></span> - Method in interface com.compomics.util.db.interfaces.<a href="com/compomics/util/db/interfaces/DBElement.html" title="interface in com.compomics.util.db.interfaces">DBElement</a></dt>
<dd>
<div class="block">This method will return a HashMap with all the properties for this DBElement.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getAllPtmLocations--">getAllPtmLocations()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns the sites of all localized PTMs.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getAllReferenceAreasXAxis--">getAllReferenceAreasXAxis()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns all the x-axis references areas as a hashmap, with the labels as
 the keys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getAllReferenceAreasYAxis--">getAllReferenceAreasYAxis()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns all the y-axis references areas as a hashmap, with the labels as
 the keys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/IdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in interface com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/IdfileReader.html" title="interface in com.compomics.util.experiment.io.identification">IdfileReader</a></dt>
<dd>
<div class="block">Retrieves all the identifications from an identification file as a list
 of spectrum matches It is very important to close the file reader after
 creation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/IdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in interface com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/IdfileReader.html" title="interface in com.compomics.util.experiment.io.identification">IdfileReader</a></dt>
<dd>
<div class="block">Retrieves all the identifications from an identification file as a list
 of spectrum matches It is very important to close the file reader after
 creation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">AndromedaIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">AndromedaIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MascotIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MascotIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MsAmandaIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MsAmandaIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MzIdentMLIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MzIdentMLIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">NovorIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">NovorIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepxmlIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepxmlIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PNovoIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PNovoIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">TideIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">TideIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-">getAllSpectrumMatches(WaitingHandler, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">XTandemIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html#getAllSpectrumMatches-com.compomics.util.waiting.WaitingHandler-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAllSpectrumMatches(WaitingHandler, SearchParameters, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">XTandemIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VisibleTableColumnsDialogParent.html#getAllTableColumns--">getAllTableColumns()</a></span> - Method in interface com.compomics.util.gui.<a href="com/compomics/util/gui/VisibleTableColumnsDialogParent.html" title="interface in com.compomics.util.gui">VisibleTableColumnsDialogParent</a></dt>
<dd>
<div class="block">Returns all the table columns, both visible and hidden.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#getAllTableColumns--">getAllTableColumns()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#getAllTagAssumptions-int-">getAllTagAssumptions(int)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Returns all tag assumptions for the specified search engine indexed by their
 score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#getAllTagAssumptions--">getAllTagAssumptions()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Returns a stream of all tag assumptions</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#getAlpha--">getAlpha()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Get the alpha level (transparency).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getAlphaLevel--">getAlphaLevel()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the alpha level.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getAmbiguityKey--">getAmbiguityKey()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Returns the ambiguity key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#getAmbiguouslyLocalizedModifications--">getAmbiguouslyLocalizedModifications()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Returns a list of modifications presenting at least an ambiguous site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#getAmbiguousModificationsAtRepresentativeSite-int-">getAmbiguousModificationsAtRepresentativeSite(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Returns the ambiguous modification assignments registered at the given
 representative site in a map: secondary site &gt; modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getAmbiguousModificationSiteNumber-com.compomics.util.experiment.identification.IdentificationMatch-">getAmbiguousModificationSiteNumber(IdentificationMatch)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns a summary of the number of modifications present on the sequence
 ambiguously assigned to an amino acid grouped by representative site
 followed by secondary ambiguous sites.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getAmbiguousModificationSiteNumber-com.compomics.util.experiment.identification.IdentificationMatch-java.util.ArrayList-">getAmbiguousModificationSiteNumber(IdentificationMatch, ArrayList&lt;String&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns a summary of the number of modifications present on the sequence
 ambiguously assigned to an amino acid grouped by representative site
 followed by secondary ambiguous sites.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getAmbiguousModificationSites-com.compomics.util.experiment.identification.IdentificationMatch-java.lang.String-">getAmbiguousModificationSites(IdentificationMatch, String)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns a list of the modifications present on the sequence ambiguously
 assigned to an amino acid grouped by representative site followed by
 secondary ambiguous sites.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getAmbiguousModificationSites-com.compomics.util.experiment.identification.IdentificationMatch-java.lang.String-java.util.ArrayList-">getAmbiguousModificationSites(IdentificationMatch, String, ArrayList&lt;String&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns a list of the modifications present on the sequence ambiguously
 assigned to an amino acid grouped by representative site followed by
 secondary ambiguous sites.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#getAmbiguousModificationsSites-java.lang.String-">getAmbiguousModificationsSites(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Returns the ambiguous modification sites registered for the given modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getAminoAcid-java.lang.String-">getAminoAcid(String)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the amino acid corresponding to the letter given, null if not
 implemented.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getAminoAcid-char-">getAminoAcid(char)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the amino acid corresponding to the single letter code given,
 null if not implemented.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html#getAminoAcid--">getAminoAcid()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAIdentityFeature</a></dt>
<dd>
<div class="block">Returns the amino acid targeted represented as single letter code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#getAminoAcidAfter--">getAminoAcidAfter()</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Getter for the amino acids potentially following the cleavage.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#getAminoAcidAt-int-">getAminoAcidAt(int)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Returns the amino acid at the given index on the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#getAminoAcidBefore--">getAminoAcidBefore()</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Getter for the amino acids potentially preceding the cleavage.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/TagFactory.html#getAminoAcidCombinations-int-">getAminoAcidCombinations(int)</a></span> - Static method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/TagFactory.html" title="class in com.compomics.util.experiment.identification">TagFactory</a></dt>
<dd>
<div class="block">Returns all the amino acid combinations for a given tag length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html#getAminoAcidFeature1--">getAminoAcidFeature1()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature</a></dt>
<dd>
<div class="block">Returns the first amino acid feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html#getAminoAcidFeature2--">getAminoAcidFeature2()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature</a></dt>
<dd>
<div class="block">Returns the second amino acid feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getAminoAcidFromGeneticCode-java.lang.String-">getAminoAcidFromGeneticCode(String)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the amino acid from the standard genetic code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getAminoAcidPatternFromString-java.lang.String-">getAminoAcidPatternFromString(String)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Parses the amino acid pattern from the given string as created by the
 toString() method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getAminoAcidPatternFromString-java.lang.String-int-">getAminoAcidPatternFromString(String, int)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Parses the amino acid pattern from the given string as created by the
 toString() method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html#getAminoAcidProperties--">getAminoAcidProperties()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/MultipleAAPropertyFeature.html#getAminoAcidProperties--">getAminoAcidProperties()</a></span> - Method in interface com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/MultipleAAPropertyFeature.html" title="interface in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">MultipleAAPropertyFeature</a></dt>
<dd>
<div class="block">Returns the amino acid properties to consider.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html#getAminoAcidProperty--">getAminoAcidProperty()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAPropertyFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html#getAminoAcidProperty--">getAminoAcidProperty()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AASequenceFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/SingleAAPropertyFeature.html#getAminoAcidProperty--">getAminoAcidProperty()</a></span> - Method in interface com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/SingleAAPropertyFeature.html" title="interface in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">SingleAAPropertyFeature</a></dt>
<dd>
<div class="block">Returns the amino acid property to consider.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getAminoAcids--">getAminoAcids()</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Convenience method returning an array of all implemented amino acids
 represented by their singe letter code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getAminoAcidsAtTarget--">getAminoAcidsAtTarget()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the targeted amino acids at position "target".</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getAminoAcidsAtTargetSet--">getAminoAcidsAtTargetSet()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns a set containing the amino acids at target.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#getAminoAcidScores--">getAminoAcidScores()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Returns the amino acid scores.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getAminoAcidsList--">getAminoAcidsList()</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Convenience method returning an arrayList of all implemented amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getAnnotationIntensityLimit--">getAnnotationIntensityLimit()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns the intensity percentile to consider for annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getAnnotationParameters--">getAnnotationParameters()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the annotation parameters used for identification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/AnnotationParametersDialog.html#getAnnotationSettings--">getAnnotationSettings()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/AnnotationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">AnnotationParametersDialog</a></dt>
<dd>
<div class="block">Returns the annotation settings as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PropertiesManager.html#getApplicationFolder-com.compomics.util.enumeration.CompomicsTools-">getApplicationFolder(CompomicsTools)</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PropertiesManager.html" title="class in com.compomics.util.io">PropertiesManager</a></dt>
<dd>
<div class="block">Get the application folder that contains the appropriate properties.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#getAreaColor--">getAreaColor()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Get the area color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getAreaUnderCurveColors--">getAreaUnderCurveColors()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the list of colors used for the datasets.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#getArray-int-">getArray(int)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">Returns an array containing only the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/MavenJarFile.html#getArtifactId--">getArtifactId()</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/MavenJarFile.html" title="class in com.compomics.software.autoupdater">MavenJarFile</a></dt>
<dd>
<div class="block">Returns the artifact id.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#getAs--">getAs()</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Returns a histogram of the as found in the previously interpolated
 scores.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#getAscii-java.lang.String-">getAscii(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">GET a file from the FTP server in Ascii mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getAssayCount-java.lang.String-">getAssayCount(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns the assay count for a given project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getAssayDetail-java.lang.String-">getAssayDetail(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns the assay details for a given assay.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html#getAssayDetailList-java.lang.String-">getAssayDetailList(String)</a></span> - Method in class com.compomics.util.io.json.marshallers.<a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html" title="class in com.compomics.util.io.json.marshallers">PrideMarshaller</a></dt>
<dd>
<div class="block">Convert from JSON to a list of AssaytDetails.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getAssayDetails-java.lang.String-">getAssayDetails(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns the assay details for a given project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getAssayFileCount-java.lang.String-">getAssayFileCount(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a count of file details for a given assay.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getAssayFileDetails-java.lang.String-">getAssayFileDetails(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a list of file details for a given assay.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html#getAssembly-java.lang.Integer-">getAssembly(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblGenomesSpecies</a></dt>
<dd>
<div class="block">Returns the Ensembl assembly corresponding to the given NCBI taxon.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html#getAssembly-java.lang.Integer-">getAssembly(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblSpecies</a></dt>
<dd>
<div class="block">Returns the Ensembl assembly corresponding to the given NCBI taxon.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getAsStringPattern-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getAsStringPattern(SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the amino acid pattern as case insensitive pattern for String
 matching.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html#getAssumptionFilter--">getAssumptionFilter()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PsmFilter</a></dt>
<dd>
<div class="block">Returns the filter used to filter at the assumption level.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#getAtom-java.lang.String-">getAtom(String)</a></span> - Static method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">Returns the atom corresponding to the given short name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomPanel.html#getAtom--">getAtom()</a></span> - Method in class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomPanel.html" title="class in com.compomics.util.gui.atoms">AtomPanel</a></dt>
<dd>
<div class="block">Returns the currently selected atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#getAtomChain-java.lang.String-">getAtomChain(String)</a></span> - Static method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>
<div class="block">Returns an atom chain from the input as string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#getAtomChain--">getAtomChain()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>
<div class="block">Returns the atom chain as a list of AtomImpl.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getAtomChainAdded--">getAtomChainAdded()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Returns the atom chain added.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomChainDialog.html#getAtomChainAdded--">getAtomChainAdded()</a></span> - Method in class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomChainDialog.html" title="class in com.compomics.util.gui.atoms">AtomChainDialog</a></dt>
<dd>
<div class="block">Returns the added atom chain as edited by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getAtomChainRemoved--">getAtomChainRemoved()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Returns the atom chain removed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomChainDialog.html#getAtomChainRemoved--">getAtomChainRemoved()</a></span> - Method in class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomChainDialog.html" title="class in com.compomics.util.gui.atoms">AtomChainDialog</a></dt>
<dd>
<div class="block">Returns the removed atom chain as edited by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getAtomicComposition--">getAtomicComposition()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the atomic composition.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html#getAtomSymbol--">getAtomSymbol()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html" title="class in com.compomics.util.experiment.biology.atoms">AtomImpl</a></dt>
<dd>
<div class="block">Returns the atom symbol as specified in the Atom class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/roc/DataRoc.html#getAuc--">getAuc()</a></span> - Method in class com.compomics.util.math.roc.<a href="com/compomics/util/math/roc/DataRoc.html" title="class in com.compomics.util.math.roc">DataRoc</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/roc/DistributionRoc.html#getAuc--">getAuc()</a></span> - Method in class com.compomics.util.math.roc.<a href="com/compomics/util/math/roc/DistributionRoc.html" title="class in com.compomics.util.math.roc">DistributionRoc</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/ROC.html#getAuc--">getAuc()</a></span> - Method in interface com.compomics.util.math.statistics.<a href="com/compomics/util/math/statistics/ROC.html" title="interface in com.compomics.util.math.statistics">ROC</a></dt>
<dd>
<div class="block">Returns an estimation of the area under the curve.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html#getAutoUpdate--">getAutoUpdate()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html" title="class in com.compomics.util.parameters.identification.advanced">GeneParameters</a></dt>
<dd>
<div class="block">Indicates whether the gene mappings should be automatically updated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#getAverageMassDelta-java.lang.String-">getAverageMassDelta(String)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">This method returns a double with the average mass difference
 conferred on the sequence by this modification for the specified residue.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#getAverageMassDelta-java.lang.String-">getAverageMassDelta(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This method returns a double with the average mass difference
 conferred on the sequence by this modification for the specified residue.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#getAverageMassDelta-java.lang.String-">getAverageMassDelta(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">This method returns a double with the average mass difference
 conferred on the sequence by this modification for the specified residue.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#getBackedUpModifications--">getBackedUpModifications()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Returns the modifications backed-up as a map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getBackgroundPeakWidth--">getBackgroundPeakWidth()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the current width of the background peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#getBase64String--">getBase64String()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the basicity according to PMID 14730315.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Alanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Arginine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Asparagine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">AsparticAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">B</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Cysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">GlutamicAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glutamine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glycine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Histidine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Isoleucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">J</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Leucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Lysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Methionine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Phenylalanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Proline</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Pyrrolysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Selenocysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Serine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Threonine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tryptophan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tyrosine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Valine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">X</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html#getBasicity--">getBasicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Z</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getBatchSize--">getBatchSize()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the batch size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#getBestPeak-java.util.ArrayList-com.compomics.util.experiment.identification.matches.IonMatch-">getBestPeak(ArrayList&lt;Peak&gt;, IonMatch)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the peak to retain of the matched peaks according to the ties
 resolution setting.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#getBestPeptideAssumption--">getBestPeptideAssumption()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Getter for the best peptide assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#getBestTagAssumption--">getBestTagAssumption()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Getter for the best tag assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getBin-double-">getBin(double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>
<div class="block">Returns the bin corresponding to the given m/z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#getBinary-java.lang.String-">getBinary(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">GET a file from the FTP server in Binary mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getBinMax--">getBinMax()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>
<div class="block">Returns the highest bin.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getBinMin--">getBinMin()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>
<div class="block">Returns the lowest bin.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SimpleNoiseDistribution.html#getBinnedCumulativeProbability-double-">getBinnedCumulativeProbability(double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SimpleNoiseDistribution.html" title="class in com.compomics.util.experiment.mass_spectrometry">SimpleNoiseDistribution</a></dt>
<dd>
<div class="block">Get the binned cumulative probability.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SimpleNoiseDistribution.html#getBinnedCumulativeProbabilityLog-double-">getBinnedCumulativeProbabilityLog(double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SimpleNoiseDistribution.html" title="class in com.compomics.util.experiment.mass_spectrometry">SimpleNoiseDistribution</a></dt>
<dd>
<div class="block">Get the binned logged cumulative probability.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html#getBins--">getBins()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">PrecursorMap</a></dt>
<dd>
<div class="block">Returns the bins in the map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getBins--">getBins()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>
<div class="block">Returns the bins in the map as a list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getBiomartEnsemblGenomeMappingFile-java.lang.String-">getBiomartEnsemblGenomeMappingFile(String)</a></span> - Static method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the Ensembl Genome BioMart file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getBiomartEnsemblMappingFile-java.lang.String-">getBiomartEnsemblMappingFile(String)</a></span> - Static method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the Ensembl BioMart file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getBiomartMapping--">getBiomartMapping()</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the BioMart mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/objects/BlobObject.html#getBlob">getBlob</a></span> - Variable in class com.compomics.util.db.object.objects.<a href="com/compomics/util/db/object/objects/BlobObject.html" title="class in com.compomics.util.db.object.objects">BlobObject</a></dt>
<dd>
<div class="block">Object representation of the blob.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/objects/BlobObject.html#getBlob--">getBlob()</a></span> - Method in class com.compomics.util.db.object.objects.<a href="com/compomics/util/db/object/objects/BlobObject.html" title="class in com.compomics.util.db.object.objects">BlobObject</a></dt>
<dd>
<div class="block">Returns the byte representation of the object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#getBlock--">getBlock()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Returns the block.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html#getBlocks--">getBlocks()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb">PdbParameter</a></dt>
<dd>
<div class="block">Returns the blocks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#getBlockSequence-java.lang.String-">getBlockSequence(String)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Get the blocked sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#getBorderColor--">getBorderColor()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Returns the border color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#getBorderWidth--">getBorderWidth()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Returns the border width.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#getBs--">getBs()</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Returns a histogram of the bs found in the previously interpolated
 scores.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#getCacheSize--">getCacheSize()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Returns the cache size in number of objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getCacheSize--">getCacheSize()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the spectrum cache size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html#getCategories--">getCategories()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">FeaturesMap</a></dt>
<dd>
<div class="block">Returns the different categories of the features in this map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureAbsolute.html#getCategory--">getCategory()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureAbsolute.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAIdentityFeatureAbsolute</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureRelative.html#getCategory--">getCategory()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureRelative.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAIdentityFeatureRelative</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureAbsolute.html#getCategory--">getCategory()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureAbsolute.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyFeatureAbsolute</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureRelative.html#getCategory--">getCategory()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureRelative.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyFeatureRelative</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html#getCategory--">getCategory()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonAminoAcidFeature.html#getCategory--">getCategory()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ComplementaryIonAminoAcidFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonAminoAcidFeature.html#getCategory--">getCategory()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ForwardIonAminoAcidFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.html#getCategory--">getCategory()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">IonFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html#getCategory--">getCategory()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ModificationFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideAminoAcidFeature.html#getCategory--">getCategory()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideAminoAcidFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html#getCategory--">getCategory()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/Ms2pipFeature.html#getCategory--">getCategory()</a></span> - Method in interface com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/Ms2pipFeature.html" title="interface in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">Ms2pipFeature</a></dt>
<dd>
<div class="block">Returns the category of the feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#getCharacterInfo-int-">getCharacterInfo(int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Returns the character and rank at a given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#getCharge--">getCharge()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Getter for the charge</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#getCharge--">getCharge()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method reports on the charge of the precursor ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Charge.html#getChargeAsFormattedString-int-">getChargeAsFormattedString(int)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Charge.html" title="class in com.compomics.util.experiment.biology.ions">Charge</a></dt>
<dd>
<div class="block">Returns the charge as a string of + or -.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getChargePrideCvTerm--">getChargePrideCvTerm()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the pride CV term for the ion match charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IntensityHistogram.html#getChartPanel--">getChartPanel()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IntensityHistogram.html" title="class in com.compomics.util.gui.spectrum">IntensityHistogram</a></dt>
<dd>
<div class="block">Returns the chart panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html#getChartPanel--">getChartPanel()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorBubblePlot</a></dt>
<dd>
<div class="block">Returns the chart panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorPlot.html#getChartPanel--">getChartPanel()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorPlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorPlot</a></dt>
<dd>
<div class="block">Returns the chart panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getChartPanel--">getChartPanel()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Returns the chart panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/SampleSelection.html#getChoice--">getChoice()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/SampleSelection.html" title="class in com.compomics.util.gui">SampleSelection</a></dt>
<dd>
<div class="block">Getter for the users choice.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html#getChromosome-java.lang.String-">getChromosome(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.ensembl.<a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html" title="class in com.compomics.util.experiment.biology.genes.ensembl">GeneMapping</a></dt>
<dd>
<div class="block">Returns the chromosome for a given gene.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getChromosome-java.lang.String-">getChromosome(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the chromosome corresponding to a given gene name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#getClassObjects-java.lang.Class-">getClassObjects(Class)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Returns an iterator of all objects of a given class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getClassObjects-java.lang.Class-">getClassObjects(Class)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns the keys of the objects of the given class,</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getClassSizeMultiplier--">getClassSizeMultiplier()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns the intensity class size multiplier.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#getCleavage-int-">getCleavage(int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">This method returns the residues that are used for cleavage at the respective
 locations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#getCleavage--">getCleavage()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">Simple getter for the cleavable residues of the Enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#getCleavage--">getCleavage()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">Simple getter for the cleavagable residues of the Enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/RegExEnzyme.html#getCleavage--">getCleavage()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/RegExEnzyme.html" title="class in com.compomics.util.protein">RegExEnzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#getCleavageParameter--">getCleavageParameter()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Returns the cleavage parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html#getCleavageParameters-int-">getCleavageParameters(int)</a></span> - Static method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.CleavageParameter</a></dt>
<dd>
<div class="block">Returns the cleavage parameter of the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getClipNtermMethionine--">getClipNtermMethionine()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns if the starting methionine peptides will be included both with
 and without the starting M.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/KMeansClustering.html#getClusterMembers-int-">getClusterMembers(int)</a></span> - Method in class com.compomics.util.math.clustering.<a href="com/compomics/util/math/clustering/KMeansClustering.html" title="class in com.compomics.util.math.clustering">KMeansClustering</a></dt>
<dd>
<div class="block">Get the sample names of all the members in the given cluster.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/KMeansClustering.html#getClusterMembersData-int-">getClusterMembersData(int)</a></span> - Method in class com.compomics.util.math.clustering.<a href="com/compomics/util/math/clustering/KMeansClustering.html" title="class in com.compomics.util.math.clustering">KMeansClustering</a></dt>
<dd>
<div class="block">Returns a hashmap with the values for the members in the given cluster.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#getCode--">getCode()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">This method returns the short code for the modification, eg.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#getCode--">getCode()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This method returns the short code for the modification, eg.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#getCode--">getCode()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">This method returns the short code for the modification, eg.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getCodedColumnType-java.lang.String-">getCodedColumnType(String)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method returns the coded column type for the specified column.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getCodedColumnType-int-">getCodedColumnType(int)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method returns the coded columntype for the specified column index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getCodedColumnTypes--">getCodedColumnTypes()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method reports on all the coded column types.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#getColor-java.lang.String-">getColor(String)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Returns the color used to code the given modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#getColor--">getColor()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Returns the color of the advocate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/interfaces/SpectrumAnnotation.html#getColor--">getColor()</a></span> - Method in interface com.compomics.util.gui.interfaces.<a href="com/compomics/util/gui/interfaces/SpectrumAnnotation.html" title="interface in com.compomics.util.gui.interfaces">SpectrumAnnotation</a></dt>
<dd>
<div class="block">This method returns the color for the annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ModificationProfile.html#getColor--">getColor()</a></span> - Method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ModificationProfile.html" title="class in com.compomics.util.gui.protein">ModificationProfile</a></dt>
<dd>
<div class="block">Returns the PTM color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/DefaultSpectrumAnnotation.html#getColor--">getColor()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/DefaultSpectrumAnnotation.html" title="class in com.compomics.util.gui.spectrum">DefaultSpectrumAnnotation</a></dt>
<dd>
<div class="block">This method returns the color for the annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#getColor-java.lang.String-">getColor(String)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Returns the color used to code the given modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#getColors--">getColors()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Returns the modification colors as a map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#getColummnNames--">getColummnNames()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>
<div class="block">Return the column names.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#getColumn-int-">getColumn(int)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBResultSet.html#getColumnClass-int-">getColumnClass(int)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db">DBResultSet</a></dt>
<dd>
<div class="block">Returns <code>Object.class</code> regardless of <code>columnIndex</code>.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html#getColumnClass-int-">getColumnClass(int)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html" title="class in com.compomics.util.gui.spectrum">IonLabelColorTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html#getColumnClass-int-">getColumnClass(int)</a></span> - Method in class com.compomics.util.gui.variants.aa_substitutions.<a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html" title="class in com.compomics.util.gui.variants.aa_substitutions">AaSubstitutionMatrixTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableMap.html#getColumnClass-int-">getColumnClass(int)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun">TableMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getColumnCount--">getColumnCount()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method returns the column count for this table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBResultSet.html#getColumnCount--">getColumnCount()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db">DBResultSet</a></dt>
<dd>
<div class="block">This method reports on the number of columns in the resultset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html#getColumnCount--">getColumnCount()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html" title="class in com.compomics.util.gui.spectrum">IonLabelColorTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html#getColumnCount--">getColumnCount()</a></span> - Method in class com.compomics.util.gui.variants.aa_substitutions.<a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html" title="class in com.compomics.util.gui.variants.aa_substitutions">AaSubstitutionMatrixTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableMap.html#getColumnCount--">getColumnCount()</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun">TableMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getColumnName-int-">getColumnName(int)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method returns the name of the specified column.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBResultSet.html#getColumnName-int-">getColumnName(int)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db">DBResultSet</a></dt>
<dd>
<div class="block">Returns a default name for the column using spreadsheet conventions:
  A, B, C, ...</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html#getColumnName-int-">getColumnName(int)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html" title="class in com.compomics.util.gui.spectrum">IonLabelColorTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html#getColumnName-int-">getColumnName(int)</a></span> - Method in class com.compomics.util.gui.variants.aa_substitutions.<a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html" title="class in com.compomics.util.gui.variants.aa_substitutions">AaSubstitutionMatrixTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableMap.html#getColumnName-int-">getColumnName(int)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun">TableMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getColumnNames--">getColumnNames()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method reports on all the column names.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBResultSet.html#getColumnNames--">getColumnNames()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db">DBResultSet</a></dt>
<dd>
<div class="block">This method reports on all the columnn ames.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#getColumns--">getColumns()</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getColumnSize-java.lang.String-">getColumnSize(String)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method returns the column size for the specified column.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getColumnSize-int-">getColumnSize(int)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method returns the column size for the specified column index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getColumnSizes--">getColumnSizes()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method reports on all the column sizes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#getCombination-int-int-">getCombination(int, int)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Returns the number of k-combinations in a set of n elements.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigFunctions.html#getCombination-java.math.BigInteger-java.math.BigInteger-">getCombination(BigInteger, BigInteger)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BigFunctions.html" title="class in com.compomics.util.math">BigFunctions</a></dt>
<dd>
<div class="block">Returns the number of k-combinations in a set of n elements as a big
 decimal.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getCombinations--">getCombinations()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the amino acids combinations which might represent this amino
 acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#getCombinations-java.lang.String-">getCombinations(String)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Returns a list of all combinations which can be created from a sequence
 when expanding ambiguous amino acids like Xs.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandLineUtils.html#getCommandLineArgument-java.util.ArrayList-">getCommandLineArgument(ArrayList&lt;File&gt;)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli">CommandLineUtils</a></dt>
<dd>
<div class="block">Returns the list of file as argument for the command line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandLineUtils.html#getCommandLineArgument-java.io.File-">getCommandLineArgument(File)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli">CommandLineUtils</a></dt>
<dd>
<div class="block">Returns the file as argument for the command line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html#getCommandLineDescription--">getCommandLineDescription()</a></span> - Static method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.CleavageParameter</a></dt>
<dd>
<div class="block">Returns the different options as command line description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html#getCommandLineDescription--">getCommandLineDescription()</a></span> - Static method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.Specificity</a></dt>
<dd>
<div class="block">Returns the different options as command line description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFormat.html#getCommandLineOption--">getCommandLineOption()</a></span> - Static method in enum com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFormat.html" title="enum in com.compomics.util.io.export">ExportFormat</a></dt>
<dd>
<div class="block">Returns the command line description when the format is used as command
 line argument.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html#getCommandLineOptions--">getCommandLineOptions()</a></span> - Static method in enum com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html" title="enum in com.compomics.util.experiment.identification.modification">ModificationLocalizationScore</a></dt>
<dd>
<div class="block">Returns the different implemented scores as list of command line option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html#getCommandLineOptions--">getCommandLineOptions()</a></span> - Static method in enum com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html" title="enum in com.compomics.util.experiment.io.identification">MzIdentMLVersion</a></dt>
<dd>
<div class="block">Returns a string describing the different options.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html#getCommandLineOptions--">getCommandLineOptions()</a></span> - Static method in enum com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html" title="enum in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters.MatchingType</a></dt>
<dd>
<div class="block">Returns the different matching types as command line options
 description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/searchgui/OutputParameters.html#getCommandLineOptions--">getCommandLineOptions()</a></span> - Static method in enum com.compomics.util.parameters.searchgui.<a href="com/compomics/util/parameters/searchgui/OutputParameters.html" title="enum in com.compomics.util.parameters.searchgui">OutputParameters</a></dt>
<dd>
<div class="block">Convenience method returning all possibilities in a command line option description format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html#getCommonName-java.lang.Integer-">getCommonName(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblSpecies</a></dt>
<dd>
<div class="block">Returns the common name corresponding to the given NCBI taxon.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html#getCommonName-java.lang.Integer-">getCommonName(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">UniprotTaxonomy</a></dt>
<dd>
<div class="block">Returns the common name corresponding to the given NCBI taxon.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getComparatorForItem-java.lang.String-">getComparatorForItem(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Returns the comparator set for a given filtering item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_generation/FeaturesGenerator.html#getComplementaryIonsFeatures-com.compomics.util.experiment.biology.proteins.Peptide-int-int-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getComplementaryIonsFeatures(Peptide, int, int, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_generation/FeaturesGenerator.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation">FeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the ms2pip features for the complementary ions of the given
 peptide at the given charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getComplementMzTolerance--">getComplementMzTolerance()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the complement m/z tolerance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getComplementScoreWeight--">getComplementScoreWeight()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the complement score weight.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#getComposition--">getComposition()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">The composition of the loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getComputeExactPValues--">getComputeExactPValues()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true if the exact p-values are to be computed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getComputeSpScore--">getComputeSpScore()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true of the SP score is to be computed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getComputeXCorr--">getComputeXCorr()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns true if a Sequest-like cross correlation score will be calculated
 for the top ranking hits in each spectrums result set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getConcatenatTargetDecoy--">getConcatenatTargetDecoy()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true if the target and decoy results are to be concatenated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/Filter.html#getCondition--">getCondition()</a></span> - Method in interface com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/Filter.html" title="interface in com.compomics.util.experiment.filtering">Filter</a></dt>
<dd>
<div class="block">Returns a description of the condition to match for the filter to
 validate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getCondition--">getCondition()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getConfidence--">getConfidence()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the confidence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getConfidenceLevel-int-">getConfidenceLevel(int)</a></span> - Static method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Convenience method returning the given confidence level as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#getConfidenceMargin--">getConfidenceMargin()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Returns the margin to the threshold to use as factor of the resolution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html#getConfident--">getConfident()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches">ModificationMatch</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the modification is confidently
 localized on the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#getConfidentlyLocalizedModifications--">getConfidentlyLocalizedModifications()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Returns a list of modifications presenting at least a confident site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#getConfidentModificationsAt-int-">getConfidentModificationsAt(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Returns the main potential modifications at the given amino acid index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getConfidentModificationSites-com.compomics.util.experiment.identification.IdentificationMatch-java.lang.String-">getConfidentModificationSites(IdentificationMatch, String)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns a summary of all modifications present on the sequence
 confidently assigned to an amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getConfidentModificationSites-com.compomics.util.experiment.identification.IdentificationMatch-java.lang.String-java.util.ArrayList-">getConfidentModificationSites(IdentificationMatch, String, ArrayList&lt;String&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns a summary of the modifications present on the peptide sequence
 confidently assigned to an amino acid with focus on given list of
 modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getConfidentModificationSitesNumber-com.compomics.util.experiment.identification.IdentificationMatch-">getConfidentModificationSitesNumber(IdentificationMatch)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of confidently localized variable modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getConfidentModificationSitesNumber-com.compomics.util.experiment.identification.IdentificationMatch-java.util.ArrayList-">getConfidentModificationSitesNumber(IdentificationMatch, ArrayList&lt;String&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of confidently localized variable modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getConfidentPtmLocations--">getConfidentPtmLocations()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns the confidently and very confidently localized PTMs.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#getConfidentSites--">getConfidentSites()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Returns a list of all confident modification sites.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#getConfidentSitesForModification-java.lang.String-">getConfidentSitesForModification(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Returns the confident sites for the given modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getConsecutiveIonProbability--">getConsecutiveIonProbability()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the consecutive ion probability.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#getContactGroups--">getContactGroups()</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Returns the contact groups.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ContactGroup.html#getContacts--">getContacts()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ContactGroup.html" title="class in com.compomics.util.pride.prideobjects">ContactGroup</a></dt>
<dd>
<div class="block">Returns the contacts.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getContent--">getContent()</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the content of this tag as a list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html#getContentPane--">getContentPane()</a></span> - Method in class com.compomics.util.gui.isotopic_calculator.<a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html" title="class in com.compomics.util.gui.isotopic_calculator">IsotopeDistributionGui</a></dt>
<dd>
<div class="block">This methods gives a JPanel holding everything from this frame</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#getControlSamples--">getControlSamples()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">Returns the indexes of the samples labeled as control.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getConvertedColumnType-java.lang.String-">getConvertedColumnType(String)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method returns the converted column type for the specified column.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getConvertedColumnType-int-">getConvertedColumnType(int)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method returns the converted columntype for the specified column index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getConvertedColumnTypes--">getConvertedColumnTypes()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method reports on all the converted column types.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getCopyRight--">getCopyRight()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns the copyright.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getCoreHeader--">getCoreHeader()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method reports on the core information for the header, which is
 comprised of the ID and the accession String:</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#getCorrelation-java.util.ArrayList-java.util.ArrayList-">getCorrelation(ArrayList&lt;Double&gt;, ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Returns the population Pearson correlation r between series1 and series2.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getCoverableAA-long-">getCoverableAA(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns an array of the likelihood to find identify a given amino acid in
 the protein sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#getCoveredAminoAcids-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getCoveredAminoAcids(AnnotationParameters, SpecificAnnotationParameters, Spectrum, Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the ion matches corresponding to fragment ions indexed by amino
 acid number in the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#getCreationTime--">getCreationTime()</a></span> - Method in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#getCtermAsString-boolean-int-java.lang.String:A...-">getCtermAsString(boolean, int, String[]...)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns the C-terminal annotation as string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getCTerminal-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getCTerminal(boolean, ModificationParameters, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the C-terminal tag of this tag as a string for sequence display.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getCTerminalGap--">getCTerminalGap()</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the C-terminal gap of the tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#getcTermModification--">getcTermModification()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Returns the C-term modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html#getCtermModification-com.compomics.util.experiment.identification.protein_sequences.digestion.PeptideDraft-java.lang.String-int-">getCtermModification(PeptideDraft, String, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">ProteinIteratorUtils</a></dt>
<dd>
<div class="block">Returns the c-terminal modification for the given peptide draft.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#getCTermTruncatedProtein-int-">getCTermTruncatedProtein(int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This method truncates the sequence for this protein on the C-terminus
 to the requested size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getCTermTruncatedSequence-int-">getCTermTruncatedSequence(int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method will return an AASequenceImpl that represents
 a C-terminal truncation of the current sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/Distribution.html#getCumulativeProbabilityAt-double-">getCumulativeProbabilityAt(double)</a></span> - Method in interface com.compomics.util.math.statistics.<a href="com/compomics/util/math/statistics/Distribution.html" title="interface in com.compomics.util.math.statistics">Distribution</a></dt>
<dd>
<div class="block">Returns the cumulative density function value at a given position.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html#getCumulativeProbabilityAt-double-">getCumulativeProbabilityAt(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html" title="class in com.compomics.util.math.statistics.distributions">BinomialDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getCumulativeProbabilityAt-double-">getCumulativeProbabilityAt(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html#getCumulativeProbabilityAt-double-">getCumulativeProbabilityAt(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#getCurrentAdded--">getCurrentAdded()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Getter for the current number of added objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#getCurrentAnnotation-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getCurrentAnnotation(Spectrum, AnnotationParameters, SpecificAnnotationParameters, ModificationParameters, SequenceProvider, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html#getCurrentAnnotation-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getCurrentAnnotation(Spectrum, AnnotationParameters, SpecificAnnotationParameters, ModificationParameters, SequenceProvider, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">TagSpectrumAnnotator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#getCurrentAnnotation-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getCurrentAnnotation(Spectrum, AnnotationParameters, SpecificAnnotationParameters, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the currently matched ions with the given settings using the
 intensity filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#getCurrentAnnotation-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getCurrentAnnotation(Spectrum, AnnotationParameters, SpecificAnnotationParameters, ModificationParameters, SequenceProvider, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the currently matched ions with the given settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ensembl/EnsemblVersion.html#getCurrentEnsemblVersion-com.compomics.util.experiment.biology.taxonomy.mappings.EnsemblGenomesSpecies.EnsemblGenomeDivision-">getCurrentEnsemblVersion(EnsemblGenomesSpecies.EnsemblGenomeDivision)</a></span> - Static method in class com.compomics.util.experiment.biology.genes.ensembl.<a href="com/compomics/util/experiment/biology/genes/ensembl/EnsemblVersion.html" title="class in com.compomics.util.experiment.biology.genes.ensembl">EnsemblVersion</a></dt>
<dd>
<div class="block">Returns the current Ensembl version number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#getCurrentlyLoadedPeptide--">getCurrentlyLoadedPeptide()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the currently inspected peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#getCurrentlyLoadedSpectrumKey--">getCurrentlyLoadedSpectrumKey()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the spectrum currently inspected.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#getCurrentPeptideKey--">getCurrentPeptideKey()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Returns the current peptide key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#getCurrentProteinKey--">getCurrentProteinKey()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Returns the current protein key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getCurrentVennDiagramType--">getCurrentVennDiagramType()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Returns the current Venn diagram type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#getCvTerm--">getCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Returns the CV term associated with this enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getCvTerm--">getCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Returns the CV term associated with this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PtmToPrideMap.html#getCVTerm-java.lang.String-">getCVTerm(String)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PtmToPrideMap.html" title="class in com.compomics.util.pride">PtmToPrideMap</a></dt>
<dd>
<div class="block">Returns the CV term corresponding to the given PTM name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Instrument.html#getCvTerms--">getCvTerms()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Instrument.html" title="class in com.compomics.util.pride.prideobjects">Instrument</a></dt>
<dd>
<div class="block">Returns the CV terms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Protocol.html#getCvTerms--">getCvTerms()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Protocol.html" title="class in com.compomics.util.pride.prideobjects">Protocol</a></dt>
<dd>
<div class="block">Returns the CV terms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Sample.html#getCvTerms--">getCvTerms()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Sample.html" title="class in com.compomics.util.pride.prideobjects">Sample</a></dt>
<dd>
<div class="block">Returns the CV terms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicElement.html#getDaltonDifference--">getDaltonDifference()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicElement.html" title="class in com.compomics.util.general">IsotopicElement</a></dt>
<dd>
<div class="block">Getter for the dalton difference with the natural form of this element</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBResultSet.html#getData--">getData()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db">DBResultSet</a></dt>
<dd>
<div class="block">This method reports on the data stored in the resultset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/ByteArrayRenderer.html#getData--">getData()</a></span> - Method in class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/ByteArrayRenderer.html" title="class in com.compomics.util.gui.renderers">ByteArrayRenderer</a></dt>
<dd>
<div class="block">This method returns the data that is currently cached by the renderer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getDatabaseDirectory--">getDatabaseDirectory()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns the database directory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getDatabaseType--">getDatabaseType()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the database type as inferred from the header structure.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getDatabaseTypeAsString-com.compomics.util.experiment.io.biology.protein.ProteinDatabase-">getDatabaseTypeAsString(ProteinDatabase)</a></span> - Static method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Convenience method returning the database name as a String.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getDatabaseTypesAsString--">getDatabaseTypesAsString()</a></span> - Static method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the implemented database types as an array of String.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#getDataEndian--">getDataEndian()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Returns the endian value of the binary array (mzData element
 .../data/endian).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#getDataFilters--">getDataFilters()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>
<div class="block">Return the data filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html#getDataFormats-java.lang.Boolean-java.lang.Boolean-">getDataFormats(Boolean, Boolean)</a></span> - Static method in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsFormat</a></dt>
<dd>
<div class="block">Returns a list of formats.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#getDataLength--">getDataLength()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Returns the length of the binary array (mzData element
 .../data/length).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#getDataPrecision--">getDataPrecision()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Returns the precision of the binary array (mzData element
 .../data/precision).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/BiomartMapping.html#getDataset-java.lang.String-">getDataset(String)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/BiomartMapping.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">BiomartMapping</a></dt>
<dd>
<div class="block">Returns the Ensembl dataset for the given assembly.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getDaTolerance-double-double-">getDaTolerance(double, double)</a></span> - Static method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the absolute tolerance in Dalton corresponding to the relative
 tolerance in ppm at the given reference mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#getDB--">getDB()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Getter for the persistence manager.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#getDbFile--">getDbFile()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Getter for the database file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#getDbFolder--">getDbFolder()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Getter for the database folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getDbFolder--">getDbFolder()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the last used database folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getDbSummaryFolder--">getDbSummaryFolder()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the folder where FASTA files summary statistics are stored.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#getDecodedByteArray--">getDecodedByteArray()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Returns the contents of the binary array <i>decoded</i> using the
 Base64 algorithm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getDecoyAccessions--">getDecoyAccessions()</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html#getDecoyAccessions--">getDecoyAccessions()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html" title="class in com.compomics.util.experiment.identification.protein_sequences">SingleProteinSequenceProvider</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getDecoyAccessions--">getDecoyAccessions()</a></span> - Method in interface com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></dt>
<dd>
<div class="block">Returns the decoy accessions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#getDecoyFlag--">getDecoyFlag()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Returns the decoy flag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getDecoyFormat--">getDecoyFormat()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the decoy format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getDecoyMode--">getDecoyMode()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the decoy mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/converters/DecoyConverter.html#getDecoyParameters-com.compomics.util.experiment.io.biology.protein.FastaParameters-">getDecoyParameters(FastaParameters)</a></span> - Static method in class com.compomics.util.experiment.io.biology.protein.converters.<a href="com/compomics/util/experiment/io/biology/protein/converters/DecoyConverter.html" title="class in com.compomics.util.experiment.io.biology.protein.converters">DecoyConverter</a></dt>
<dd>
<div class="block">Returns the FASTA parameters of the target-decoy database based on the parameters of the target database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getDecoySeed--">getDecoySeed()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the decoy seed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#getDecoysInFirstBin--">getDecoysInFirstBin()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Returns the minimal number of decoys to include in a bin to set the bin
 size of the score histogram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/converters/DecoyConverter.html#getDecoySummary-java.io.File-com.compomics.util.experiment.io.biology.protein.FastaSummary-">getDecoySummary(File, FastaSummary)</a></span> - Static method in class com.compomics.util.experiment.io.biology.protein.converters.<a href="com/compomics/util/experiment/io/biology/protein/converters/DecoyConverter.html" title="class in com.compomics.util.experiment.io.biology.protein.converters">DecoyConverter</a></dt>
<dd>
<div class="block">Returns the FASTA summary of the target-decoy database based on the summary of the target database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScoresEstimator.html#getDecreasingScore-com.compomics.util.experiment.biology.proteins.Peptide-java.lang.Integer-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.parameters.identification.IdentificationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.PeptideSpectrumAnnotator-int-">getDecreasingScore(Peptide, Integer, Spectrum, IdentificationParameters, SpecificAnnotationParameters, ModificationParameters, SequenceProvider, SequenceMatchingParameters, PeptideSpectrumAnnotator, int)</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScoresEstimator.html" title="class in com.compomics.util.experiment.identification.psm_scoring">PsmScoresEstimator</a></dt>
<dd>
<div class="block">Scores the match between the given peptide and spectrum using the given
 score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#getDefault--">getDefault()</a></span> - Static method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Returns an instance containing only the default enzymes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#getDefaultColor-java.lang.String-">getDefaultColor(String)</a></span> - Static method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Returns a default color based on the modification name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PtmToPrideMap.html#getDefaultCVTerm-java.lang.String-">getDefaultCVTerm(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PtmToPrideMap.html" title="class in com.compomics.util.pride">PtmToPrideMap</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">use the getCvTerm method for the PTM instead</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getDefaultDescription--">getDefaultDescription()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Indicates whether the description is automatically generated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramDialog.html#getDefaultExportFolder--">getDefaultExportFolder()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramDialog.html" title="class in com.compomics.util.gui">VennDiagramDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMapManager.html#getDefaultFeaturesMap--">getDefaultFeaturesMap()</a></span> - Static method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMapManager.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">FeaturesMapManager</a></dt>
<dd>
<div class="block">Returns the default features map used in the c implementation, see
 https://github.com/mvaudel/ms2pip_c.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Instrument.html#getDefaultInstruments--">getDefaultInstruments()</a></span> - Static method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Instrument.html" title="class in com.compomics.util.pride.prideobjects">Instrument</a></dt>
<dd>
<div class="block">Returns a list of predefined instruments.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#getDefaultLosses-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getDefaultLosses(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the possible neutral losses expected by default for a given
 peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html#getDefaultLosses-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getDefaultLosses(Tag, ModificationParameters, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">TagSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the possible neutral losses expected by default for a given tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#getDefaultLosses-com.compomics.util.experiment.identification.SpectrumIdentificationAssumption-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getDefaultLosses(SpectrumIdentificationAssumption, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the possible neutral losses expected by default for a given
 peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#getDefaultModifications--">getDefaultModifications()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Returns the names of the default modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getDefaultModifications--">getDefaultModifications()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the default modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#getDefaultModificationsOrdered--">getDefaultModificationsOrdered()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Returns the alphabetically ordered names of the default modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/IonFactory.html#getDefaultNeutralLosses--">getDefaultNeutralLosses()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/IonFactory.html" title="class in com.compomics.util.experiment.biology.ions">IonFactory</a></dt>
<dd>
<div class="block">Returns the default neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#getDefaultParameters--">getDefaultParameters()</a></span> - Static method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Returns default digestion parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html#getDefaultPeptideFDR--">getDefaultPeptideFDR()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">IdMatchValidationParameters</a></dt>
<dd>
<div class="block">Returns the default peptide FDR.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html#getDefaultProteinFDR--">getDefaultProteinFDR()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">IdMatchValidationParameters</a></dt>
<dd>
<div class="block">Returns the default protein FDR.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Protocol.html#getDefaultProtocols--">getDefaultProtocols()</a></span> - Static method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Protocol.html" title="class in com.compomics.util.pride.prideobjects">Protocol</a></dt>
<dd>
<div class="block">Returns a list of default protocols.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html#getDefaultPsmFDR--">getDefaultPsmFDR()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">IdMatchValidationParameters</a></dt>
<dd>
<div class="block">Returns the default PSM FDR.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#getDefaultReference-java.lang.String-java.lang.String-int-">getDefaultReference(String, String, int)</a></span> - Static method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">Returns the default reference for an identification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html#getDefaultReferences--">getDefaultReferences()</a></span> - Static method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html" title="class in com.compomics.util.pride.prideobjects">ReferenceGroup</a></dt>
<dd>
<div class="block">Returns the default references.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Sample.html#getDefaultSamples--">getDefaultSamples()</a></span> - Static method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Sample.html" title="class in com.compomics.util.pride.prideobjects">Sample</a></dt>
<dd>
<div class="block">Returns a list of default samples.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#getDefaultScores--">getDefaultScores()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Returns the default scores.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html#getDefaultSequenceMatching--">getDefaultSequenceMatching()</a></span> - Static method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a></dt>
<dd>
<div class="block">Returns default preferences from amino acid matching.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getDeisotopingMode--">getDeisotopingMode()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the deisotoping flag value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/amino_acids/Deletion.html#getDeletedAminoAcid--">getDeletedAminoAcid()</a></span> - Method in class com.compomics.util.experiment.biology.variants.amino_acids.<a href="com/compomics/util/experiment/biology/variants/amino_acids/Deletion.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids">Deletion</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#getDeltaMass-double-boolean-int-int-">getDeltaMass(double, boolean, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Returns the precursor mass error (in ppm or Da).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getDeltaMassWindow--">getDeltaMassWindow()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Get the size of the window to use when searching for matches in the known
 masses list when the user hovers over a second data point after clicking
 a previous data point.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getDeltaPEP--">getDeltaPEP()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the difference in identification algorithm level PEP with the
 next best peptide assumption with sequence difference across all search
 engines.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getDeltaScore-int-">getDeltaScore(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns the delta score at a given site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getDeNovoCharge--">getDeNovoCharge()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns the charge to use for the fragment ions in the de novo
 sequencing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getDeNovoGuiPath--">getDeNovoGuiPath()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the path to the DeNovoGUI installation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/Distribution.html#getDescendingCumulativeProbabilityAt-double-">getDescendingCumulativeProbabilityAt(double)</a></span> - Method in interface com.compomics.util.math.statistics.<a href="com/compomics/util/math/statistics/Distribution.html" title="interface in com.compomics.util.math.statistics">Distribution</a></dt>
<dd>
<div class="block">Returns the cumulative density function value at a given position when
 starting from the high values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html#getDescendingCumulativeProbabilityAt-double-">getDescendingCumulativeProbabilityAt(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html" title="class in com.compomics.util.math.statistics.distributions">BinomialDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getDescendingCumulativeProbabilityAt-double-">getDescendingCumulativeProbabilityAt(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html#getDescendingCumulativeProbabilityAt-double-">getDescendingCumulativeProbabilityAt(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/PathKey.html#getDescription--">getDescription()</a></span> - Method in interface com.compomics.software.settings.<a href="com/compomics/software/settings/PathKey.html" title="interface in com.compomics.software.settings">PathKey</a></dt>
<dd>
<div class="block">Returns the description of the path.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.UtilitiesPathKey.html#getDescription--">getDescription()</a></span> - Method in enum com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.UtilitiesPathKey.html" title="enum in com.compomics.software.settings">UtilitiesPathParameters.UtilitiesPathKey</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Returns the description of the cleavage of this enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Returns the description of this substitution matrix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/amino_acids/Deletion.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.biology.variants.amino_acids.<a href="com/compomics/util/experiment/biology/variants/amino_acids/Deletion.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids">Deletion</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/amino_acids/Insertion.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.biology.variants.amino_acids.<a href="com/compomics/util/experiment/biology/variants/amino_acids/Insertion.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids">Insertion</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/amino_acids/Substitution.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.biology.variants.amino_acids.<a href="com/compomics/util/experiment/biology/variants/amino_acids/Substitution.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids">Substitution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/amino_acids/Swap.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.biology.variants.amino_acids.<a href="com/compomics/util/experiment/biology/variants/amino_acids/Swap.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids">Swap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/Variant.html#getDescription--">getDescription()</a></span> - Method in interface com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/Variant.html" title="interface in com.compomics.util.experiment.biology.variants">Variant</a></dt>
<dd>
<div class="block">Returns the description of the variant as string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/Filter.html#getDescription--">getDescription()</a></span> - Method in interface com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/Filter.html" title="interface in com.compomics.util.experiment.filtering">Filter</a></dt>
<dd>
<div class="block">Returns a description for the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItem.html#getDescription--">getDescription()</a></span> - Method in interface com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItem.html" title="interface in com.compomics.util.experiment.filtering">FilterItem</a></dt>
<dd>
<div class="block">Returns a description of the item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html#getDescription--">getDescription()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">AssumptionFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html#getDescription--">getDescription()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PeptideFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html#getDescription--">getDescription()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">ProteinFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html#getDescription--">getDescription()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PsmFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureAbsolute.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureAbsolute.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAIdentityFeatureAbsolute</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureRelative.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureRelative.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAIdentityFeatureRelative</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureAbsolute.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureAbsolute.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyFeatureAbsolute</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureRelative.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureRelative.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyFeatureRelative</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonAminoAcidFeature.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ComplementaryIonAminoAcidFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonFeature.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ComplementaryIonFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonAminoAcidFeature.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ForwardIonAminoAcidFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonFeature.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ForwardIonFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ModificationFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideAminoAcidFeature.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideAminoAcidFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/Ms2pipFeature.html#getDescription--">getDescription()</a></span> - Method in interface com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/Ms2pipFeature.html" title="interface in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">Ms2pipFeature</a></dt>
<dd>
<div class="block">Returns the description of the feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getDescription-java.lang.String-">getDescription(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Returns the description for this database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getDescription-java.lang.String-">getDescription(String)</a></span> - Method in interface com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></dt>
<dd>
<div class="block">Returns the description of the protein with the given accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/patient/PatientInformation.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.experiment.patient.<a href="com/compomics/util/experiment/patient/PatientInformation.html" title="class in com.compomics.util.experiment.patient">PatientInformation</a></dt>
<dd>
<div class="block">Returns the description of the information.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFeature.html#getDescription--">getDescription()</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFeature.html" title="interface in com.compomics.util.io.export">ExportFeature</a></dt>
<dd>
<div class="block">Returns the description of the feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/DatFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/DatFileFilter.html" title="class in com.compomics.util.io.file.filefilters">DatFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/DtaFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/DtaFileFilter.html" title="class in com.compomics.util.io.file.filefilters">DtaFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FastaFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FastaFileFilter.html" title="class in com.compomics.util.io.file.filefilters">FastaFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/JpegFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/JpegFileFilter.html" title="class in com.compomics.util.io.file.filefilters">JpegFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/MgfFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/MgfFileFilter.html" title="class in com.compomics.util.io.file.filefilters">MgfFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/Ms2FileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/Ms2FileFilter.html" title="class in com.compomics.util.io.file.filefilters">Ms2FileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/MzDataFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/MzDataFileFilter.html" title="class in com.compomics.util.io.file.filefilters">MzDataFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/MzMlFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/MzMlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">MzMlFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/MzXmlFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/MzXmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">MzXmlFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/OmxFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/OmxFileFilter.html" title="class in com.compomics.util.io.file.filefilters">OmxFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/OutFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/OutFileFilter.html" title="class in com.compomics.util.io.file.filefilters">OutFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PdfFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PdfFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PdfFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PeffFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PeffFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PeffFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PepXmlFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PepXmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PepXmlFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PklFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PklFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PklFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PklSpoFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PklSpoFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PklSpoFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PkxFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PkxFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PkxFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PngFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PngFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PngFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/ProtXmlFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/ProtXmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">ProtXmlFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/SequestParamsFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/SequestParamsFileFilter.html" title="class in com.compomics.util.io.file.filefilters">SequestParamsFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/SvgFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/SvgFileFilter.html" title="class in com.compomics.util.io.file.filefilters">SvgFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/TiffFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/TiffFileFilter.html" title="class in com.compomics.util.io.file.filefilters">TiffFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/XmlFileFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/XmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">XmlFileFilter</a></dt>
<dd>
<div class="block">The description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FilenameExtensionFilter.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FilenameExtensionFilter.html" title="class in com.compomics.util.io.file">FilenameExtensionFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getDescription--">getDescription()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getDescriptionProteinName--">getDescriptionProteinName()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the protein name as inferred from the description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getDescriptionShort--">getDescriptionShort()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the short description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getDesignaledPeakList-java.util.ArrayList-">getDesignaledPeakList(ArrayList&lt;IonMatch&gt;)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the peak list of this spectrum without matched peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html#getDestinationFile--">getDestinationFile()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersInputBean</a></dt>
<dd>
<div class="block">Returns the file where to save the identification parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#getDestinationFile--">getDestinationFile()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">This method reports on the destinationfile currently used in transfer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Instrument.html#getDetector--">getDetector()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Instrument.html" title="class in com.compomics.util.pride.prideobjects">Instrument</a></dt>
<dd>
<div class="block">Returns the instrument detector.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#getDeterminant--">getDeterminant()</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html#getDifference--">getDifference()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html" title="class in com.compomics.util.pdbfinder.das.readers">AlignmentBlock</a></dt>
<dd>
<div class="block">Returns the difference.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#getDifference--">getDifference()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Returns the difference.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#getDifferenceToMonoisotopic-int-">getDifferenceToMonoisotopic(int)</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">Returns the mass difference between the given isotope and the
 monoisotopic mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getDigestionParameters--">getDigestionParameters()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the digestion preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getDigestionType--">getDigestionType()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the enzyme digestion type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#getDiscardLowQualitySpectra--">getDiscardLowQualitySpectra()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether low quality spectra shall be
 discarded.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getDisplayedTips--">getDisplayedTips()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the list of displayed tips.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/UnitOfMeasurement.html#getDisplayName--">getDisplayName()</a></span> - Method in class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/UnitOfMeasurement.html" title="class in com.compomics.util.experiment.units">UnitOfMeasurement</a></dt>
<dd>
<div class="block">Returns the name to display, e.g.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#getDisplayProgress--">getDisplayProgress()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#getDisplayProgress--">getDisplayProgress()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#getDisplayProgress--">getDisplayProgress()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#getDisplayProgress--">getDisplayProgress()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#getDisplayProgress--">getDisplayProgress()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Returns if the waiting handler is to show the progress for the current
 process or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html#getDivision-java.lang.Integer-">getDivision(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblGenomesSpecies</a></dt>
<dd>
<div class="block">Returns the division corresponding to the given NCBI taxon.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Reference.html#getDoi--">getDoi()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Reference.html" title="class in com.compomics.util.pride.prideobjects">Reference</a></dt>
<dd>
<div class="block">Returns the Digital Object Identifier (DOI) of the reference.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#getDoubleArray--">getDoubleArray()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Checks if all the necessary information is provided and then converts the
 decoded binary array into an array of double values (that for example
 could be used to draw a spectra).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getDoubleChargeWindow--">getDoubleChargeWindow()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the window size for doubly charged ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandLineUtils.html#getDoubleListFromString-java.lang.String-java.lang.String-">getDoubleListFromString(String, String)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli">CommandLineUtils</a></dt>
<dd>
<div class="block">Parses a list of doubles from a command line option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getDSites--">getDSites()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns a set of sites where a D-score is available</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getDuplicatedSpectrumTitles--">getDuplicatedSpectrumTitles()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns a map of the duplicated spectrum titles, can be null.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/Duration.html#getDuration--">getDuration()</a></span> - Method in class com.compomics.util.waiting.<a href="com/compomics/util/waiting/Duration.html" title="class in com.compomics.util.waiting">Duration</a></dt>
<dd>
<div class="block">Returns the duration in milliseconds.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#getDynamicRange--">getDynamicRange()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns the dynamic range for spectrum filtering.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicElement.html#getElement--">getElement()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicElement.html" title="class in com.compomics.util.general">IsotopicElement</a></dt>
<dd>
<div class="block">Getter for the MolecularElement</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/UnknownElementMassException.html#getElement--">getElement()</a></span> - Method in exception com.compomics.util.general.<a href="com/compomics/util/general/UnknownElementMassException.html" title="class in com.compomics.util.general">UnknownElementMassException</a></dt>
<dd>
<div class="block">Simple getter for the element variable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/MolecularFormula.html#getElementCount-com.compomics.util.enumeration.MolecularElement-">getElementCount(MolecularElement)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/MolecularFormula.html" title="class in com.compomics.util.protein">MolecularFormula</a></dt>
<dd>
<div class="block">Getter for the count of a specific element</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Contact.html#getEMail--">getEMail()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Contact.html" title="class in com.compomics.util.pride.prideobjects">Contact</a></dt>
<dd>
<div class="block">Returns the contact e-mail.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#getEnd--">getEnd()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Get the end value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getEnd--">getEnd()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the end.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#getEndBlock--">getEndBlock()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Returns the block end.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getEndLocation--">getEndLocation()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method reports on the end index of the header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/StartEndPosition.html#getEndPosition--">getEndPosition()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/StartEndPosition.html" title="class in com.compomics.util.pdbfinder.das.readers">StartEndPosition</a></dt>
<dd>
<div class="block">Returns the end position.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#getEndProtein--">getEndProtein()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Returns the protein end.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html#getEnsemblAccession-java.lang.String-">getEnsemblAccession(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.ensembl.<a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html" title="class in com.compomics.util.experiment.biology.genes.ensembl">GeneMapping</a></dt>
<dd>
<div class="block">Returns the Ensembl accession for a given gene.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getEnsemblAssembly-java.lang.Integer-">getEnsemblAssembly(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the Ensembl assembly to use for the given taxon.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getEnsemblDataset-java.lang.Integer-">getEnsemblDataset(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the Ensembl dataset to use for the given taxon.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getEnsembleSpecies--">getEnsembleSpecies()</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns a map of the species in Ensembl.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.EnsemblGenomeDivision.html#getEnsemblGenomeDivisionFromName-java.lang.String-">getEnsemblGenomeDivisionFromName(String)</a></span> - Static method in enum com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.EnsemblGenomeDivision.html" title="enum in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblGenomesSpecies.EnsemblGenomeDivision</a></dt>
<dd>
<div class="block">Returns the EnsemblGenomeDivision corresponding to the given Ensembl name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getEnsemblGenomesSpecies--">getEnsemblGenomesSpecies()</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the Ensembl genome species mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getEnsemblGenomesSpeciesFile-java.lang.String-">getEnsemblGenomesSpeciesFile(String)</a></span> - Static method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the Ensembl genome species file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getEnsemblId-java.lang.String-">getEnsemblId(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the Ensembl ID corresponding to the given gene name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ensembl/EnsemblVersion.html#getEnsemblSchemaName-com.compomics.util.experiment.biology.taxonomy.mappings.EnsemblGenomesSpecies.EnsemblGenomeDivision-">getEnsemblSchemaName(EnsemblGenomesSpecies.EnsemblGenomeDivision)</a></span> - Static method in class com.compomics.util.experiment.biology.genes.ensembl.<a href="com/compomics/util/experiment/biology/genes/ensembl/EnsemblVersion.html" title="class in com.compomics.util.experiment.biology.genes.ensembl">EnsemblVersion</a></dt>
<dd>
<div class="block">Returns the name of the Ensembl schema for BioMart queries.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getEnsemblSpecies--">getEnsemblSpecies()</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the Ensembl species mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getEnsemblSpeciesFile-java.lang.String-">getEnsemblSpeciesFile(String)</a></span> - Static method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the Ensembl species file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#getEnsemblVersion-java.lang.Integer-">getEnsemblVersion(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Returns the Ensembl version for a given species.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#getEnsemblVersionFromFile-java.io.File-java.lang.String-">getEnsemblVersionFromFile(File, String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Gets the Ensembl version of a given species from a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#getEnsemblVersionsFile--">getEnsemblVersionsFile()</a></span> - Static method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Returns the Ensembl version file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getEnsemblVersionsMap--">getEnsemblVersionsMap()</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the Ensembl version map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getEntitiesForPoint-int-int-">getEntitiesForPoint(int, int)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Returns a list of the entities at the given x, y view location.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#getEntitiesForPoint-int-int-">getEntitiesForPoint(int, int)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>
<div class="block">Returns a list of the entities at the given x, y view location.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#getEnzyme-java.lang.String-">getEnzyme(String)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Returns the enzyme corresponding to the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/pride/EnzymeParametersDialog.html#getEnzyme--">getEnzyme()</a></span> - Method in class com.compomics.util.gui.parameters.identification.pride.<a href="com/compomics/util/gui/parameters/identification/pride/EnzymeParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.pride">EnzymeParametersDialog</a></dt>
<dd>
<div class="block">Returns the selected enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/MascotEnzymeReader.html#getEnzyme-java.lang.String-">getEnzyme(String)</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/MascotEnzymeReader.html" title="class in com.compomics.util.io">MascotEnzymeReader</a></dt>
<dd>
<div class="block">This method will return a <b>copy of</b> an Enzyme instance
 for the given name, or 'null' if the enzyme was not found in the current list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#getEnzymeId--">getEnzymeId()</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Get the enzyme id.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/MascotEnzymeReader.html#getEnzymeNames--">getEnzymeNames()</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/MascotEnzymeReader.html" title="class in com.compomics.util.io">MascotEnzymeReader</a></dt>
<dd>
<div class="block">This method reports on all the known names for enzymes in this reader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#getEnzymes--">getEnzymes()</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Get the imported enzymes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#getEnzymes--">getEnzymes()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Returns the enzymes used for digestion in a list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html#getEnzymeToAdd--">getEnzymeToAdd()</a></span> - Method in class com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html" title="class in com.compomics.cli.enzymes">EnzymesCLIInputBean</a></dt>
<dd>
<div class="block">Returns the enzyme to add.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html#getEnzymeToRemove--">getEnzymeToRemove()</a></span> - Method in class com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html" title="class in com.compomics.cli.enzymes">EnzymesCLIInputBean</a></dt>
<dd>
<div class="block">Returns the name of the enzyme to remove.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getEnzymeType--">getEnzymeType()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the enzyme type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getError-boolean-int-int-">getError(boolean, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the error.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getError-boolean-">getError(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the error.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#getErrorKeys--">getErrorKeys()</a></span> - Static method in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Returns a list containing the keys of the paths where the tool is not
 able to write.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/interfaces/SpectrumAnnotation.html#getErrorMargin--">getErrorMargin()</a></span> - Method in interface com.compomics.util.gui.interfaces.<a href="com/compomics/util/gui/interfaces/SpectrumAnnotation.html" title="interface in com.compomics.util.gui.interfaces">SpectrumAnnotation</a></dt>
<dd>
<div class="block">This method returns the allowed error margin (both sides)
 for the M/Z of the annotation (eg., 0.1 means an allowed
 interval of [M/Z-0.1, M/Z+0.1].</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/DefaultSpectrumAnnotation.html#getErrorMargin--">getErrorMargin()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/DefaultSpectrumAnnotation.html" title="class in com.compomics.util.gui.spectrum">DefaultSpectrumAnnotation</a></dt>
<dd>
<div class="block">This method returns the allowed error margin (both sides)
 for the M/Z of the annotation (eg., 0.1 means an allowed
 interval of [M/Z-0.1, M/Z+0.1].</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/PrideXmlValidator.html#getErrors--">getErrors()</a></span> - Method in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/PrideXmlValidator.html" title="class in com.compomics.util.pride.validation">PrideXmlValidator</a></dt>
<dd>
<div class="block">Returns the XML validation error object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html#getErrorsAsList--">getErrorsAsList()</a></span> - Method in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html" title="class in com.compomics.util.pride.validation">XMLValidationErrorHandler</a></dt>
<dd>
<div class="block">Returns the errors as a list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/PrideXmlValidator.html#getErrorsAsString--">getErrorsAsString()</a></span> - Method in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/PrideXmlValidator.html" title="class in com.compomics.util.pride.validation">PrideXmlValidator</a></dt>
<dd>
<div class="block">Returns the errors formatted as s single string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html#getErrorsFormattedAsHTML--">getErrorsFormattedAsHTML()</a></span> - Method in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html" title="class in com.compomics.util.pride.validation">XMLValidationErrorHandler</a></dt>
<dd>
<div class="block">Returns the errors formatted as HTML.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html#getErrorsFormattedAsPlainText--">getErrorsFormattedAsPlainText()</a></span> - Method in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html" title="class in com.compomics.util.pride.validation">XMLValidationErrorHandler</a></dt>
<dd>
<div class="block">Returns the errors formatted as plain text.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#getEValueMap-java.util.ArrayList-">getEValueMap(ArrayList&lt;Double&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Returns the e-value corresponding to a list of scores in a map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#getEValueMap-java.util.ArrayList-boolean-">getEValueMap(ArrayList&lt;Double&gt;, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Returns the e-value corresponding to a list of scores in a map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#getEvenRowColor--">getEvenRowColor()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">This method returns the Color used for the even numbered rows, or 'null' if the
 default JTable background color is maintained for these rows.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getExceptions--">getExceptions()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Returns the exceptions to the rule.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/ExceptionHandler.html#getExceptionType-java.lang.Exception-">getExceptionType(Exception)</a></span> - Static method in class com.compomics.util.exceptions.<a href="com/compomics/util/exceptions/ExceptionHandler.html" title="class in com.compomics.util.exceptions">ExceptionHandler</a></dt>
<dd>
<div class="block">Returns the exception type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#getExpectedIons-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getExpectedIons(SpecificAnnotationParameters, Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the expected ions in a map indexed by the possible charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#getExpectedIons-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-java.util.HashMap-">getExpectedIons(SpecificAnnotationParameters, Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, HashMap&lt;Integer, HashMap&lt;Integer, ArrayList&lt;Ion&gt;&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the expected ions in a map indexed by the possible charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#getExpectedIons-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-">getExpectedIons(SpecificAnnotationParameters)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the expected ions in a map indexed by the possible charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#getExpectedModifications-double-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.biology.proteins.Peptide-double-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getExpectedModifications(double, ModificationParameters, Peptide, double, SequenceProvider, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">Returns the expected modifications for a given modification mass indexed by site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html#getExperiment_type--">getExperiment_type()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb">PdbParameter</a></dt>
<dd>
<div class="block">Returns the experiment type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html#getExperimentType--">getExperimentType()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAlignment</a></dt>
<dd>
<div class="block">Returns the experiment type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFactory.html#getExportFeatures-java.lang.String-boolean-">getExportFeatures(String, boolean)</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFactory.html" title="interface in com.compomics.util.io.export">ExportFactory</a></dt>
<dd>
<div class="block">Returns the export features implemented for the given section.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFeature.html#getExportFeatures-boolean-">getExportFeatures(boolean)</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFeature.html" title="interface in com.compomics.util.io.export">ExportFeature</a></dt>
<dd>
<div class="block">Returns a list of all implemented export features.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#getExportFeatures-java.lang.String-">getExportFeatures(String)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Returns the export features to be included in the given section.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#getExportFormat--">getExportFormat()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Returns the export of the format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFactory.html#getExportScheme-java.lang.String-">getExportScheme(String)</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFactory.html" title="interface in com.compomics.util.io.export">ExportFactory</a></dt>
<dd>
<div class="block">Returns the export scheme indexed by the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#getExportWriter-com.compomics.util.io.export.ExportFormat-java.io.File-java.lang.String-int-boolean-">getExportWriter(ExportFormat, File, String, int, boolean)</a></span> - Static method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Returns an export writer for the desired format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/enumeration/ImageType.html#getExtension--">getExtension()</a></span> - Method in enum com.compomics.util.enumeration.<a href="com/compomics/util/enumeration/ImageType.html" title="enum in com.compomics.util.enumeration">ImageType</a></dt>
<dd>
<div class="block">Returns the extension.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/IdfileReader.html#getExtension--">getExtension()</a></span> - Method in interface com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/IdfileReader.html" title="interface in com.compomics.util.experiment.io.identification">IdfileReader</a></dt>
<dd>
<div class="block">Returns the extension of the file for which this IdfileReader can be
 used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html#getExtension--">getExtension()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">AndromedaIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getExtension--">getExtension()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html#getExtension--">getExtension()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MascotIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html#getExtension--">getExtension()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MsAmandaIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html#getExtension--">getExtension()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MzIdentMLIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html#getExtension--">getExtension()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">NovorIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#getExtension--">getExtension()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#getExtension--">getExtension()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html#getExtension--">getExtension()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepxmlIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html#getExtension--">getExtension()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PNovoIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html#getExtension--">getExtension()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">TideIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html#getExtension--">getExtension()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">XTandemIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#getExtension-java.io.File-">getExtension(File)</a></span> - Static method in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>
<div class="block">Get the extension of a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#getExtension-java.io.File-">getExtension(File)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Returns the extensions of a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getFastaFile--">getFastaFile()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the sequence database file used for identification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/fasta/FastaParametersInputBean.html#getFastaParameters--">getFastaParameters()</a></span> - Method in class com.compomics.cli.fasta.<a href="com/compomics/cli/fasta/FastaParametersInputBean.html" title="class in com.compomics.cli.fasta">FastaParametersInputBean</a></dt>
<dd>
<div class="block">Returns the fasta parameters as parsed from the command line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html#getFastaParameters--">getFastaParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SequenceDbDetailsDialog</a></dt>
<dd>
<div class="block">Returns the fasta parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getFastaParameters--">getFastaParameters()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the parameters to use to parse the FASTA file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getFastIndexFolderName--">getFastIndexFolderName()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the name of the FASTA index folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.html#getFeature--">getFeature()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">IonFeature</a></dt>
<dd>
<div class="block">Returns the peptide feature;</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFeature.html#getFeatureFamily--">getFeatureFamily()</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFeature.html" title="interface in com.compomics.util.io.export">ExportFeature</a></dt>
<dd>
<div class="block">Returns the family type of this export feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getFeatureId--">getFeatureId()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the feature ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getFeatureLabel--">getFeatureLabel()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the feature label.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html#getFeatures-java.lang.String-">getFeatures(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">FeaturesMap</a></dt>
<dd>
<div class="block">Returns the features in the map for the given category.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#getFeed-java.lang.String-java.lang.String-java.util.ArrayList-">getFeed(String, String, ArrayList&lt;String&gt;)</a></span> - Static method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Creates a feed object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#getFeed-java.lang.String-java.lang.String-">getFeed(String, String)</a></span> - Static method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Creates a general feed with no key word.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/FileAndFileFilter.html#getFile--">getFile()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/FileAndFileFilter.html" title="class in com.compomics.util">FileAndFileFilter</a></dt>
<dd>
<div class="block">Returns the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html#getFileDetailList-java.lang.String-">getFileDetailList(String)</a></span> - Method in class com.compomics.util.io.json.marshallers.<a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html" title="class in com.compomics.util.io.json.marshallers">PrideMarshaller</a></dt>
<dd>
<div class="block">Convert from JSON to a list of FileDetails.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/FileAndFileFilter.html#getFileFilter--">getFileFilter()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/FileAndFileFilter.html" title="class in com.compomics.util">FileAndFileFilter</a></dt>
<dd>
<div class="block">Returns the file filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html#getFileIn--">getFileIn()</a></span> - Method in class com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html" title="class in com.compomics.cli.enzymes">EnzymesCLIInputBean</a></dt>
<dd>
<div class="block">Returns the file provided as input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html#getFileIn--">getFileIn()</a></span> - Method in class com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html" title="class in com.compomics.cli.modifications">ModificationsCLIInputBean</a></dt>
<dd>
<div class="block">Returns the file provided as input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getFileName--">getFileName()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns the name of the indexed file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getFileName--">getFileName()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the file name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#getFilename--">getFilename()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Getter for the filename</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#getFilename--">getFilename()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method reports on the filename for the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#getFileName--">getFileName()</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObject.html#getFileName--">getFileName()</a></span> - Method in interface com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObject.html" title="interface in com.compomics.util.pride">PrideObject</a></dt>
<dd>
<div class="block">Returns the name to use when serializing the object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Contact.html#getFileName--">getFileName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Contact.html" title="class in com.compomics.util.pride.prideobjects">Contact</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ContactGroup.html#getFileName--">getFileName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ContactGroup.html" title="class in com.compomics.util.pride.prideobjects">ContactGroup</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Instrument.html#getFileName--">getFileName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Instrument.html" title="class in com.compomics.util.pride.prideobjects">Instrument</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Protocol.html#getFileName--">getFileName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Protocol.html" title="class in com.compomics.util.pride.prideobjects">Protocol</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Reference.html#getFileName--">getFileName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Reference.html" title="class in com.compomics.util.pride.prideobjects">Reference</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html#getFileName--">getFileName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html" title="class in com.compomics.util.pride.prideobjects">ReferenceGroup</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Sample.html#getFileName--">getFileName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Sample.html" title="class in com.compomics.util.pride.prideobjects">Sample</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#getFileName-java.lang.String-">getFileName(String)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">An OS independent getName alternative.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#getFileName-java.io.File-">getFileName(File)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">An OS independent getName alternative.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html#getFileOut--">getFileOut()</a></span> - Method in class com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html" title="class in com.compomics.cli.enzymes">EnzymesCLIInputBean</a></dt>
<dd>
<div class="block">Returns the file provided as output.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html#getFileOut--">getFileOut()</a></span> - Method in class com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html" title="class in com.compomics.cli.modifications">ModificationsCLIInputBean</a></dt>
<dd>
<div class="block">Returns the file provided as output.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/IdfileReaderFactory.html#getFileReader-java.io.File-">getFileReader(File)</a></span> - Method in class com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/IdfileReaderFactory.html" title="class in com.compomics.util.experiment.io.identification">IdfileReaderFactory</a></dt>
<dd>
<div class="block">This method returns the proper identification file reader depending on
 the format of the provided file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandLineUtils.html#getFiles-java.lang.String-java.util.ArrayList-">getFiles(String, ArrayList&lt;String&gt;)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli">CommandLineUtils</a></dt>
<dd>
<div class="block">Returns a list of files as imported from the command line option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#getFileSize-java.net.URL-">getFileSize(URL)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Returns the size of the file located at the given URL.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html#getFilter-java.lang.Integer-">getFilter(Integer)</a></span> - Static method in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">ProteoWizardFilter</a></dt>
<dd>
<div class="block">Returns the filter designed by the given number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html#getFilter-java.lang.String-">getFilter(String)</a></span> - Static method in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">ProteoWizardFilter</a></dt>
<dd>
<div class="block">Returns the filter designed by the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterDialog.html#getFilter--">getFilter()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterDialog.html" title="class in com.compomics.util.gui.filtering">FilterDialog</a></dt>
<dd>
<div class="block">Returns the filter as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/MatchesImportParametersDialog.html#getFilter--">getFilter()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/MatchesImportParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">MatchesImportParametersDialog</a></dt>
<dd>
<div class="block">Returns the id filter as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/filters/ProbabilityFilter.html#getFilteredInput-java.util.ArrayList-java.util.ArrayList-double-">getFilteredInput(ArrayList&lt;Double&gt;, ArrayList&lt;Double&gt;, double)</a></span> - Static method in class com.compomics.util.math.statistics.linear_regression.filters.<a href="com/compomics/util/math/statistics/linear_regression/filters/ProbabilityFilter.html" title="class in com.compomics.util.math.statistics.linear_regression.filters">ProbabilityFilter</a></dt>
<dd>
<div class="block">Returns a list containing first the filtered xs and then the filtered ys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/filters/BinningFilter.html#getFilteredInputFixedBins-java.util.ArrayList-java.util.ArrayList-int-">getFilteredInputFixedBins(ArrayList&lt;Double&gt;, ArrayList&lt;Double&gt;, int)</a></span> - Static method in class com.compomics.util.math.statistics.linear_regression.filters.<a href="com/compomics/util/math/statistics/linear_regression/filters/BinningFilter.html" title="class in com.compomics.util.math.statistics.linear_regression.filters">BinningFilter</a></dt>
<dd>
<div class="block">Returns a list containing first the filtered xs and then the filtered ys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/filters/BinningFilter.html#getFilteredInputFixedBinsSize-java.util.ArrayList-java.util.ArrayList-int-">getFilteredInputFixedBinsSize(ArrayList&lt;Double&gt;, ArrayList&lt;Double&gt;, int)</a></span> - Static method in class com.compomics.util.math.statistics.linear_regression.filters.<a href="com/compomics/util/math/statistics/linear_regression/filters/BinningFilter.html" title="class in com.compomics.util.math.statistics.linear_regression.filters">BinningFilter</a></dt>
<dd>
<div class="block">Returns a list containing first the filtered xs and then the filtered ys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html#getFilterItem-java.lang.String-">getFilterItem(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">AssumptionFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getFilterItem-java.lang.String-">getFilterItem(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Returns the filter item corresponding to the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html#getFilterItem-java.lang.String-">getFilterItem(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html#getFilterItem-java.lang.String-">getFilterItem(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html" title="class in com.compomics.util.experiment.identification.filtering">ProteinFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html#getFilterItem-java.lang.String-">getFilterItem(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PsmFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html#getFilters--">getFilters()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html" title="class in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsConvertParameters</a></dt>
<dd>
<div class="block">Returns the index of the filters selected.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html#getFiltersMap--">getFiltersMap()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html" title="class in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsConvertParameters</a></dt>
<dd>
<div class="block">Returns the filters map, filter index - value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/DbObject.html#getFirstLevel--">getFirstLevel()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/DbObject.html" title="class in com.compomics.util.db.object">DbObject</a></dt>
<dd>
<div class="block">Gets whether an object is a first level object or not i.e.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#getFixedAaModifications--">getFixedAaModifications()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Returns the modifications at specific amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html#getFixedModificationAtAa-char-">getFixedModificationAtAa(char)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">ProteinIteratorUtils</a></dt>
<dd>
<div class="block">Returns the fixed modification that can be found at the given amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#getFixedModifications-boolean-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getFixedModifications(boolean, boolean, ModificationParameters, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Returns the fixed modifications for this sequence based on the given
 modification parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getFixedModifications-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getFixedModifications(ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the fixed modifications for this peptide based on the given
 modification parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#getFixedModifications--">getFixedModifications()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Returns the fixed modifications for the peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#getFixedModifications--">getFixedModifications()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Returns the searched fixed modifications names.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#getFixedModificationsAsString-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getFixedModificationsAsString(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns the peptide modifications as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/CommandLineParser.html#getFlags--">getFlags()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/CommandLineParser.html" title="class in com.compomics.util.general">CommandLineParser</a></dt>
<dd>
<div class="block">This method will report on all flags that have been found, or return an
 empty String[] if none were present.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getFontSize--">getFontSize()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Get the font size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getFontSizeLegend--">getFontSizeLegend()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Get the font size for the legend.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getForeignAccession--">getForeignAccession()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the foreign accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getForeignDescription--">getForeignDescription()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the foreign description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getForeignID--">getForeignID()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the foreign ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/ExportFormatSelectionDialog.html#getFormat--">getFormat()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/ExportFormatSelectionDialog.html" title="class in com.compomics.util.gui">ExportFormatSelectionDialog</a></dt>
<dd>
<div class="block">Returns the selected format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFormat.html#getFormatFromCommandLineOption-java.lang.String-">getFormatFromCommandLineOption(String)</a></span> - Static method in enum com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFormat.html" title="enum in com.compomics.util.io.export">ExportFormat</a></dt>
<dd>
<div class="block">Returns the export format designed by the given command line option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getForwardIons--">getForwardIons()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the forward ions searched as list of integers as indexed in the
 FragmentIon class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_generation/FeaturesGenerator.html#getForwardIonsFeatures-com.compomics.util.experiment.biology.proteins.Peptide-int-int-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getForwardIonsFeatures(Peptide, int, int, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_generation/FeaturesGenerator.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation">FeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the ms2pip features for the forward ions of the given peptide at
 the given charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html#getForwardStart-java.lang.String-">getForwardStart(String)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">NeutralLossesMap</a></dt>
<dd>
<div class="block">Returns the amino acid where a neutral loss should start being accounted
 for when predicting b ions (counting from N-terminus, first aa is 1).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getFoundCharges--">getFoundCharges()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the found charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getFoundModifications--">getFoundModifications()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the variable modifications found in the currently loaded dataset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getFoundModifications--">getFoundModifications()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the names of the variable modifications found in the dataset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getFractionConfidence-java.lang.String-">getFractionConfidence(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the fraction confidence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html#getFractionMolecularWeightRanges--">getFractionMolecularWeightRanges()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html" title="class in com.compomics.util.parameters.identification.advanced">FractionParameters</a></dt>
<dd>
<div class="block">Returns the user provided molecular weight ranges for the fractions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getFractionOfPeaksForChargeEstimation--">getFractionOfPeaksForChargeEstimation()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the fraction of peaks to be retained for charge &gt;1 estimation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getFractionParameters--">getFractionParameters()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the fraction parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getFractionPEP-java.lang.String-">getFractionPEP(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the fraction pep.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getFractionPEP--">getFractionPEP()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the fraction pep map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getFractionPsmMatches--">getFractionPsmMatches()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the list of fraction PSM matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getFractions--">getFractions()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns the fractions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getFractions--">getFractions()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Return the fractions where this match was found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getFractionScore-java.lang.String-">getFractionScore(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the fraction score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getFractionScore--">getFractionScore()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Return the fractions where this match was found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/FractionParametersDialog.html#getFractionSettings--">getFractionSettings()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/FractionParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">FractionParametersDialog</a></dt>
<dd>
<div class="block">Returns the fraction settings as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getFractionValidatedPeptides-java.lang.String-">getFractionValidatedPeptides(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Get the number of validated peptides in the given fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getFractionValidatedSpectra-java.lang.String-">getFractionValidatedSpectra(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Get the number of validated spectra in the given fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getFragmentAccuracyType--">getFragmentAccuracyType()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the fragment accuracy type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getFragmentationMethod--">getFragmentationMethod()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the fragmentation method used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html#getFragmentationMethod--">getFragmentationMethod()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">NovorParameters</a></dt>
<dd>
<div class="block">Returns the fragmentation method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#getFragmentationModel--">getFragmentationModel()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Returns the name of the fragmentation model.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getFragmentationRule--">getFragmentationRule()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns the fragmentation rule.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#getFragmentationType--">getFragmentationType()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Return the fragmentation ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getFragmentBinOffset--">getFragmentBinOffset()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the fragment ion bin offset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getFragmentIonAccuracy--">getFragmentIonAccuracy()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns the fragment ion accuracy.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#getFragmentIonAccuracy--">getFragmentIonAccuracy()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Returns the fragment ion accuracy.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getFragmentIonAccuracy--">getFragmentIonAccuracy()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the MS2 ion m/z tolerance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getFragmentIonAccuracyInDa-java.lang.Double-">getFragmentIonAccuracyInDa(Double)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns the fragment ion accuracy in Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#getFragmentIonAccuracyInDa-java.lang.Double-">getFragmentIonAccuracyInDa(Double)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Returns the fragment ion accuracy in Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getFragmentIonAccuracyInDaltons--">getFragmentIonAccuracyInDaltons()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the absolute fragment ion tolerance in Dalton.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getFragmentIonAccuracyInDaltons-double-">getFragmentIonAccuracyInDaltons(double)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the absolute fragment ion tolerance in Dalton.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/IonFactory.html#getFragmentIons-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getFragmentIons(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/IonFactory.html" title="class in com.compomics.util.experiment.biology.ions">IonFactory</a></dt>
<dd>
<div class="block">This method returns all the theoretic ions expected from a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/IonFactory.html#getFragmentIons-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getFragmentIons(Peptide, SpecificAnnotationParameters, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/IonFactory.html" title="class in com.compomics.util.experiment.biology.ions">IonFactory</a></dt>
<dd>
<div class="block">This method returns the theoretic ions expected from a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/IonFactory.html#getFragmentIons-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getFragmentIons(Tag, ModificationParameters, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/IonFactory.html" title="class in com.compomics.util.experiment.biology.ions">IonFactory</a></dt>
<dd>
<div class="block">This method returns the theoretic ions expected from a tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getFragmentIonTypes--">getFragmentIonTypes()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns the type of peptide fragment ions annotated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#getFragmentIonTypes--">getFragmentIonTypes()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Returns the type of peptide fragment ions annotated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/junit/TestCaseLM.html#getFullFilePath-java.lang.String-">getFullFilePath(String)</a></span> - Static method in class com.compomics.util.junit.<a href="com/compomics/util/junit/TestCaseLM.html" title="class in com.compomics.util.junit">TestCaseLM</a></dt>
<dd>
<div class="block">This method finds a file from the current classpath and attempts to reconstruct its
 full filename.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getFullHeaderWithAddenda--">getFullHeaderWithAddenda()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method reports on the full header, with the addenda (if present).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html#getFullName--">getFullName()</a></span> - Method in enum com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html" title="enum in com.compomics.util.experiment.io.biology.protein">ProteinDatabase</a></dt>
<dd>
<div class="block">Returns the full name of the database, null if not set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/UnitOfMeasurement.html#getFullName--">getFullName()</a></span> - Method in class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/UnitOfMeasurement.html" title="class in com.compomics.util.experiment.units">UnitOfMeasurement</a></dt>
<dd>
<div class="block">Returns the full name of the unit.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html#getFunction--">getFunction()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AASequenceFeature</a></dt>
<dd>
<div class="block">Returns the function used to compare the amino acid properties.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#getGeneMappingFile-java.lang.String-">getGeneMappingFile(String)</a></span> - Static method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Returns the gene mapping file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#getGeneMappingFolder--">getGeneMappingFolder()</a></span> - Static method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Returns the path to the folder containing the gene mapping files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#getGeneMaps-com.compomics.util.parameters.identification.advanced.GeneParameters-com.compomics.util.experiment.io.biology.protein.FastaSummary-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.experiment.io.biology.protein.ProteinDetailsProvider-com.compomics.util.waiting.WaitingHandler-">getGeneMaps(GeneParameters, FastaSummary, SequenceProvider, ProteinDetailsProvider, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Returns the gene maps for the given proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getGeneName-java.lang.String-">getGeneName(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getGeneName--">getGeneName()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the gene name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getGeneName-java.lang.String-">getGeneName(String)</a></span> - Method in interface com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></dt>
<dd>
<div class="block">Returns the gene name for the given protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html#getGeneNameToAccession--">getGeneNameToAccession()</a></span> - Method in class com.compomics.util.experiment.biology.genes.ensembl.<a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html" title="class in com.compomics.util.experiment.biology.genes.ensembl">GeneMapping</a></dt>
<dd>
<div class="block">Returns the gene name to protein accession map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html#getGeneNameToChromosome--">getGeneNameToChromosome()</a></span> - Method in class com.compomics.util.experiment.biology.genes.ensembl.<a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html" title="class in com.compomics.util.experiment.biology.genes.ensembl">GeneMapping</a></dt>
<dd>
<div class="block">Returns the gene name to chromosome map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getGeneNameToChromosomeMap--">getGeneNameToChromosomeMap()</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the gene name to chromosome map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getGeneNameToEnsemblIdMap--">getGeneNameToEnsemblIdMap()</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the gene name to Ensembl ID map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/GeneParametersDialog.html#getGeneParameters--">getGeneParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/GeneParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">GeneParametersDialog</a></dt>
<dd>
<div class="block">Returns the gene preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getGeneParameters--">getGeneParameters()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the gene parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/interfaces/Persistable.html#getGeneratedKeys--">getGeneratedKeys()</a></span> - Method in interface com.compomics.util.db.interfaces.<a href="com/compomics/util/db/interfaces/Persistable.html" title="interface in com.compomics.util.db.interfaces">Persistable</a></dt>
<dd>
<div class="block">This method will return the automatically generated key for the insert if
 one was triggered, or 'null' otherwise.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getGenericIon-com.compomics.util.experiment.biology.ions.Ion.IonType-int-com.compomics.util.experiment.biology.ions.NeutralLoss:A-">getGenericIon(Ion.IonType, int, NeutralLoss[])</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Convenience method returning a generic ion based on the given ion type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getGenericIon-com.compomics.util.experiment.biology.ions.Ion.IonType-int-">getGenericIon(Ion.IonType, int)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Convenience method returning a generic ion based on the given ion type
 without neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getGoAccession-java.lang.String-">getGoAccession(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the GO Term accession corresponding to the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/genes/GeneDetailsDialog.html#getGoAccessionLink-java.lang.String-">getGoAccessionLink(String)</a></span> - Method in class com.compomics.util.gui.genes.<a href="com/compomics/util/gui/genes/GeneDetailsDialog.html" title="class in com.compomics.util.gui.genes">GeneDetailsDialog</a></dt>
<dd>
<div class="block">Returns the GO accession number as a web link to the given GO term at
 QuickGO.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html#getGoAccessions-java.lang.String-">getGoAccessions(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html" title="class in com.compomics.util.experiment.biology.genes.go">GoMapping</a></dt>
<dd>
<div class="block">Returns the GO accessions linked to a given protein accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getGoAccessionToProteinMap--">getGoAccessionToProteinMap()</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the GO to protein accession map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#getGoDomainsFile--">getGoDomainsFile()</a></span> - Static method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Returns the GO domains file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#getGoMappingFile-java.lang.String-">getGoMappingFile(String)</a></span> - Static method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Returns the GO mapping file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getGoNamesForProtein-java.lang.String-">getGoNamesForProtein(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the go terms names for a protein accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getGoNamesMap--">getGoNamesMap()</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the GO accession to names map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html#getGoNamesMap--">getGoNamesMap()</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html" title="class in com.compomics.util.experiment.biology.genes.go">GoMapping</a></dt>
<dd>
<div class="block">Returns the GO accession to name map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getGoTermsForProtein-java.lang.String-">getGoTermsForProtein(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the go terms accessions for a protein accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html#getGoToProteinMap--">getGoToProteinMap()</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html" title="class in com.compomics.util.experiment.biology.genes.go">GoMapping</a></dt>
<dd>
<div class="block">Returns the GO to protein accession map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getGravy--">getGravy()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method gets the GRAVY score (Kyte &amp; Doolittle) from
 the cache, or, if it isn't cached, reconstructs it.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/MavenJarFile.html#getGroupId--">getGroupId()</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/MavenJarFile.html" title="class in com.compomics.software.autoupdater">MavenJarFile</a></dt>
<dd>
<div class="block">Returns the group id.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getGroupNames--">getGroupNames()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Returns a standard map of the group names.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html#getHeader--">getHeader()</a></span> - Static method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html" title="class in com.compomics.cli.identification_parameters">AbstractIdentificationParametersCli</a></dt>
<dd>
<div class="block">IdentificationParametersCLI header message when printing the usage.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getHeader-java.lang.String-">getHeader(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html#getHeader-java.lang.String-">getHeader(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html" title="class in com.compomics.util.experiment.identification.protein_sequences">SingleProteinSequenceProvider</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getHeader-java.lang.String-">getHeader(String)</a></span> - Method in interface com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></dt>
<dd>
<div class="block">Returns the FASTA header of the protein as found in the FASTA file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequence.html#getHeader--">getHeader()</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide">NucleotideSequence</a></dt>
<dd>
<div class="block">This method reports on the nucleotide header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html#getHeader--">getHeader()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAlignment</a></dt>
<dd>
<div class="block">Returns the header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#getHeader--">getHeader()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This method reports on the header for the current
 protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/styles/DefaultStyle.html#getHeaderHeight--">getHeaderHeight()</a></span> - Method in class com.compomics.util.io.export.styles.<a href="com/compomics/util/io/export/styles/DefaultStyle.html" title="class in com.compomics.util.io.export.styles">DefaultStyle</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/WorkbookStyle.html#getHeaderHeight--">getHeaderHeight()</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/WorkbookStyle.html" title="interface in com.compomics.util.io.export">WorkbookStyle</a></dt>
<dd>
<div class="block">Returns the height of a header row.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/styles/DefaultStyle.html#getHeaderStyle--">getHeaderStyle()</a></span> - Method in class com.compomics.util.io.export.styles.<a href="com/compomics/util/io/export/styles/DefaultStyle.html" title="class in com.compomics.util.io.export.styles">DefaultStyle</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/styles/DefaultStyle.html#getHeaderStyle-int-">getHeaderStyle(int)</a></span> - Method in class com.compomics.util.io.export.styles.<a href="com/compomics/util/io/export/styles/DefaultStyle.html" title="class in com.compomics.util.io.export.styles">DefaultStyle</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/WorkbookStyle.html#getHeaderStyle--">getHeaderStyle()</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/WorkbookStyle.html" title="interface in com.compomics.util.io.export">WorkbookStyle</a></dt>
<dd>
<div class="block">Returns the standard cell style of a header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/WorkbookStyle.html#getHeaderStyle-int-">getHeaderStyle(int)</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/WorkbookStyle.html" title="interface in com.compomics.util.io.export">WorkbookStyle</a></dt>
<dd>
<div class="block">Returns the standard cell style of a header at the given hierarchical
 depth.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the helicity according to PMID 14730315.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Alanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Arginine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Asparagine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">AsparticAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">B</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Cysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">GlutamicAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glutamine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glycine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Histidine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Isoleucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">J</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Leucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Lysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Methionine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Phenylalanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Proline</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Pyrrolysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Selenocysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Serine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Threonine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tryptophan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tyrosine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Valine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">X</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html#getHelicity--">getHelicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Z</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getHidden--">getHidden()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns whether a match is hidden or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#getHighestIntensity--">getHighestIntensity()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Method that find the intensity of the most intense peak</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#getHighestIntensity--">getHighestIntensity()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method returns the intensity of the highest intensity peak in
 this spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/MetaDataXMLParser.html#getHighestVersionNumber--">getHighestVersionNumber()</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/MetaDataXMLParser.html" title="class in com.compomics.software.autoupdater">MetaDataXMLParser</a></dt>
<dd>
<div class="block">Returns the highest version number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getHighIntensityCutOff--">getHighIntensityCutOff()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the high intensity cut-off as percentage of the most intense ion
 peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html#getHistogram-int-java.lang.Integer-int-int-">getHistogram(int, Integer, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html" title="class in com.compomics.util.experiment.identification.modification">ModificationtableContent</a></dt>
<dd>
<div class="block">Get histogram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getHitListLength--">getHitListLength()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the length of the hit list for OMSSA.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#getHitListLength--">getHitListLength()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Returns the length of the hit list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FileSystemAccessor.html#getHomeFolder--">getHomeFolder()</a></span> - Static method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FileSystemAccessor.html" title="class in com.compomics.util.io.file">FileSystemAccessor</a></dt>
<dd>
<div class="block">Returns the user home folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getHtmlTooltip--">getHtmlTooltip()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Returns information about the modification as an HTML tooltip.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the hydrophobicity according to PMID 14730315.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Alanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Arginine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Asparagine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">AsparticAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">B</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Cysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">GlutamicAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glutamine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glycine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Histidine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Isoleucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">J</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Leucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Lysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Methionine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Phenylalanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Proline</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Pyrrolysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Selenocysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Serine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Threonine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tryptophan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tyrosine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Valine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">X</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html#getHydrophobicity--">getHydrophobicity()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Z</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#getIconMap-java.lang.Class-">getIconMap(Class)</a></span> - Static method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>
<div class="block">Returns the default map of icons for the validation levels.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/PathKey.html#getId--">getId()</a></span> - Method in interface com.compomics.software.settings.<a href="com/compomics/software/settings/PathKey.html" title="interface in com.compomics.software.settings">PathKey</a></dt>
<dd>
<div class="block">Returns the id of the path.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.UtilitiesPathKey.html#getId--">getId()</a></span> - Method in enum com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.UtilitiesPathKey.html" title="enum in com.compomics.software.settings">UtilitiesPathParameters.UtilitiesPathKey</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/DbObject.html#getId--">getId()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/DbObject.html" title="class in com.compomics.util.db.object">DbObject</a></dt>
<dd>
<div class="block">Returns the id of the object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html#getId-java.lang.String-boolean-">getId(String, boolean)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">UniprotTaxonomy</a></dt>
<dd>
<div class="block">Returns the NCBI taxon corresponding to the given species name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html#getId--">getId()</a></span> - Method in enum com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html" title="enum in com.compomics.util.experiment.identification.modification">ModificationLocalizationScore</a></dt>
<dd>
<div class="block">Returns the id number of the score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getID--">getID()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/patient/Patient.html#getId--">getId()</a></span> - Method in class com.compomics.util.experiment.patient.<a href="com/compomics/util/experiment/patient/Patient.html" title="class in com.compomics.util.experiment.patient">Patient</a></dt>
<dd>
<div class="block">Returns the patient id,</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getIdentificationAlgorithmParameter-int-">getIdentificationAlgorithmParameter(int)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the algorithm specific parameters, null if not found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#getIdentificationCharge--">getIdentificationCharge()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Returns the charge used for identification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getIdentificationFeaturesCache--">getIdentificationFeaturesCache()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the identification features cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#getIdentificationFile--">getIdentificationFile()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Returns the identification file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html#getIdentificationParameters--">getIdentificationParameters()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersInputBean</a></dt>
<dd>
<div class="block">Returns the identification parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html#getIdentificationParameters-java.lang.String-">getIdentificationParameters(String)</a></span> - Method in class com.compomics.util.experiment.identification.identification_parameters.<a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html" title="class in com.compomics.util.experiment.identification.identification_parameters">IdentificationParametersFactory</a></dt>
<dd>
<div class="block">Returns the identification parameters corresponding to the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html#getIdentificationParameters--">getIdentificationParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersEditionDialog</a></dt>
<dd>
<div class="block">Returns the identification parameters as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersOverviewDialog.html#getIdentificationParameters--">getIdentificationParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersOverviewDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersOverviewDialog</a></dt>
<dd>
<div class="block">Returns the selected identification parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getIdentificationParameters-java.io.File-">getIdentificationParameters(File)</a></span> - Static method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Loads the identification parameters from a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getIdentificationParameters-java.io.File-">getIdentificationParameters(File)</a></span> - Static method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Loads the identification parameters from a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html#getIdentificationParametersFile-java.lang.String-">getIdentificationParametersFile(String)</a></span> - Static method in class com.compomics.util.experiment.identification.identification_parameters.<a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html" title="class in com.compomics.util.experiment.identification.identification_parameters">IdentificationParametersFactory</a></dt>
<dd>
<div class="block">Returns the identification parameters corresponding to the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#getIdentifier--">getIdentifier()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Returns the reference identifier.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#getIdentityMatrix-int-">getIdentityMatrix(int)</a></span> - Static method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#getIdMap--">getIdMap()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Getter for the id map mapping the hashed keys into zoo db ids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationParametersDialog.html#getIdMatchValidationPreferences--">getIdMatchValidationPreferences()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationParametersDialog</a></dt>
<dd>
<div class="block">Returns the validation settings as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getIdValidationParameters--">getIdValidationParameters()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the identification matches validation parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#getImmoniumIon-int-">getImmoniumIon(int)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Returns the immonium ion corresponding to the given subtype.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#getImmoniumIon-char-">getImmoniumIon(char)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Returns the immonium ion produced by the given amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#getImplementedAtoms-boolean-">getImplementedAtoms(boolean)</a></span> - Static method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">Returns an array of implemented atoms indicated by their short name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#getImplementedIons--">getImplementedIons()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Returns a set of possible subtypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getImplementedIonTypes--">getImplementedIonTypes()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the implemented ion types.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#getImplementedIsotopes--">getImplementedIsotopes()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">returns an unsorted list of isotopes for which a mass is available relative to the
 monoisotopic peak (+1 for carbon 13).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFactory.html#getImplementedSections--">getImplementedSections()</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFactory.html" title="interface in com.compomics.util.io.export">ExportFactory</a></dt>
<dd>
<div class="block">Returns the implemented sections.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#getIndentString--">getIndentString()</a></span> - Method in class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Returns the indent string to use, e.g.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getIndex-char-">getIndex(char)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns an index for the amino acid excluding combinations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Returns the index of a reporter ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Returns the index of the advocate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationMethod.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationMethod.html" title="class in com.compomics.util.experiment.identification">IdentificationMethod</a></dt>
<dd>
<div class="block">returns the index of the identification method</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureAbsolute.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureAbsolute.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAIdentityFeatureAbsolute</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureRelative.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureRelative.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAIdentityFeatureRelative</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureAbsolute.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureAbsolute.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyFeatureAbsolute</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureRelative.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureRelative.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyFeatureRelative</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonAminoAcidFeature.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ComplementaryIonAminoAcidFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonFeature.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ComplementaryIonFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonAminoAcidFeature.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ForwardIonAminoAcidFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonFeature.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ForwardIonFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ModificationFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideAminoAcidFeature.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideAminoAcidFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/Ms2pipFeature.html#getIndex--">getIndex()</a></span> - Method in interface com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/Ms2pipFeature.html" title="interface in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">Ms2pipFeature</a></dt>
<dd>
<div class="block">Returns an integer unique to this class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html#getIndex--">getIndex()</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a></dt>
<dd>
<div class="block">Returns the index on the protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#getIndex--">getIndex()</a></span> - Method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>
<div class="block">Returns the index associated to this possibility.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getIndex-java.lang.String-">getIndex(String)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns the index corresponding to the desired spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getIndex-java.io.File-">getIndex(File)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Deserializes the index of an mgf file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#getIndexedVariableModifications--">getIndexedVariableModifications()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Returns the variable modifications indexed by site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getIndexedVariableModifications--">getIndexedVariableModifications()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the variable modifications indexed by site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getIndexes-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getIndexes(String, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the indexes where the amino acid pattern was found in the input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getIndexes-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getIndexes(AminoAcidPattern, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the indexes where the amino acid pattern was found in the input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#getIndexes-java.lang.String-java.lang.String-">getIndexes(String, String)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Returns at which indexes a small string can be found in a big string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#getIndexMap-java.io.File-">getIndexMap(File)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Returns the index of all spectra in the given mgf file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#getIndexMap-java.io.File-com.compomics.util.waiting.WaitingHandler-">getIndexMap(File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Returns the index of all spectra in the given MGF file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html#getIndexMap-java.io.File-com.compomics.util.waiting.WaitingHandler-">getIndexMap(File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspReader</a></dt>
<dd>
<div class="block">Returns the index of all spectra in the given MSP file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getIndexName-java.lang.String-">getIndexName(String)</a></span> - Static method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the name of an index file based on the given spectrum file name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#getIndexOnProtein--">getIndexOnProtein()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Returns the peptide index on the protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html#getInferred--">getInferred()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches">ModificationMatch</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the modification is inferred from
 another peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html#getInput--">getInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">AndromedaParametersDialog</a></dt>
<dd>
<div class="block">Returns the user selection as Andromeda parameters object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html#getInput--">getInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">CometParametersDialog</a></dt>
<dd>
<div class="block">Returns the user selection as Comet parameters object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html#getInput--">getInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">DirecTagParametersDialog</a></dt>
<dd>
<div class="block">Returns the DirecTag parameters as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html#getInput--">getInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MsAmandaParametersDialog</a></dt>
<dd>
<div class="block">Returns the user selection as MS Amanda parameters object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html#getInput--">getInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MsgfParametersDialog</a></dt>
<dd>
<div class="block">Returns the user selection as MS-GF+ parameters object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html#getInput--">getInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MyriMatchParametersDialog</a></dt>
<dd>
<div class="block">Returns the user selection as MyriMatch parameters object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html#getInput--">getInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">NovorParametersDialog</a></dt>
<dd>
<div class="block">Returns the Novor parameters as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html#getInput--">getInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">OmssaParametersDialog</a></dt>
<dd>
<div class="block">Returns the user selection as OMSSA parameters object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html#getInput--">getInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">PepNovoParametersDialog</a></dt>
<dd>
<div class="block">Returns the PepNovo+ parameters as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html#getInput--">getInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">TideParametersDialog</a></dt>
<dd>
<div class="block">Returns the user selection as Tide parameters object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html#getInput--">getInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">XTandemParametersDialog</a></dt>
<dd>
<div class="block">Returns the user selection as X!tandem parameters object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html#getInputFile--">getInputFile()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersInputBean</a></dt>
<dd>
<div class="block">Returns the input parameters file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getInputFile--">getInputFile()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns the spectrum file name as found in the parameters section.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/amino_acids/Insertion.html#getInsertedAminoAcid--">getInsertedAminoAcid()</a></span> - Method in class com.compomics.util.experiment.biology.variants.amino_acids.<a href="com/compomics/util/experiment/biology/variants/amino_acids/Insertion.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids">Insertion</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#getInstance--">getInstance()</a></span> - Static method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Static method to get an instance of the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#getInstance-java.io.File-">getInstance(File)</a></span> - Static method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Static method to get an instance of the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/IonFactory.html#getInstance--">getInstance()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/IonFactory.html" title="class in com.compomics.util.experiment.biology.ions">IonFactory</a></dt>
<dd>
<div class="block">Static method which returns the instance of the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#getInstance--">getInstance()</a></span> - Static method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Static method to get the instance of the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getInstance--">getInstance()</a></span> - Static method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Static method returning the instance of the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html#getInstance--">getInstance()</a></span> - Static method in class com.compomics.util.experiment.identification.identification_parameters.<a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html" title="class in com.compomics.util.experiment.identification.identification_parameters">IdentificationParametersFactory</a></dt>
<dd>
<div class="block">Static method to get the instance of the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/IdfileReaderFactory.html#getInstance--">getInstance()</a></span> - Static method in class com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/IdfileReaderFactory.html" title="class in com.compomics.util.experiment.io.identification">IdfileReaderFactory</a></dt>
<dd>
<div class="block">A static method to retrieve the instance of the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getInstance--">getInstance()</a></span> - Static method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Static method returning the instance of the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getInstance-int-">getInstance(int)</a></span> - Static method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Static method returning the instance of the factory with a new cache
 size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html#getInstance--">getInstance()</a></span> - Static method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethodFactory</a></dt>
<dd>
<div class="block">Constructor for the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PropertiesManager.html#getInstance--">getInstance()</a></span> - Static method in class com.compomics.util.io.<a href="com/compomics/util/io/PropertiesManager.html" title="class in com.compomics.util.io">PropertiesManager</a></dt>
<dd>
<div class="block">Get the singleton instance to access properties of Computational Omics tools.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#getInstance--">getInstance()</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Method returning the instance of the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Contact.html#getInstitution--">getInstitution()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Contact.html" title="class in com.compomics.util.pride.prideobjects">Contact</a></dt>
<dd>
<div class="block">Returns the instituition.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#getInstrumentID--">getInstrumentID()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Return the instrument ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#getInstrumentID--">getInstrumentID()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Return the instrument ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#getInstruments--">getInstruments()</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Returns the instruments.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandLineUtils.html#getIntegerListFromString-java.lang.String-java.lang.String-">getIntegerListFromString(String, String)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli">CommandLineUtils</a></dt>
<dd>
<div class="block">Parses a list of integers from a command line option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#getIntegerParameter-java.lang.String-">getIntegerParameter(String)</a></span> - Method in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html#getIntensities-int-java.lang.Integer-int-">getIntensities(int, Integer, int)</a></span> - Method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html" title="class in com.compomics.util.experiment.identification.modification">ModificationtableContent</a></dt>
<dd>
<div class="block">Get intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/SequestFragmentationModel.html#getIntensity-com.compomics.util.experiment.biology.ions.Ion-">getIntensity(Ion)</a></span> - Static method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/SequestFragmentationModel.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models">SequestFragmentationModel</a></dt>
<dd>
<div class="block">Returns the intensity expected for the given ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/UniformFragmentation.html#getIntensity--">getIntensity()</a></span> - Static method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/UniformFragmentation.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models">UniformFragmentation</a></dt>
<dd>
<div class="block">Returns a default intensity of 1.0.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#getIntensity--">getIntensity()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Getter for the intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#getIntensity--">getIntensity()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Getter for the precursor intensity (here always zero)</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#getIntensity--">getIntensity()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method reports on the intensity of the precursor ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SimpleNoiseDistribution.html#getIntensityAtP-double-">getIntensityAtP(double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SimpleNoiseDistribution.html" title="class in com.compomics.util.experiment.mass_spectrometry">SimpleNoiseDistribution</a></dt>
<dd>
<div class="block">Returns the intensity at a given upper tale cumulative probability.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getIntensityCutOffIncrement--">getIntensityCutOffIncrement()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the intensity cut-off increment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getIntensityLimit-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters.IntensityThresholdType-double-">getIntensityLimit(AnnotationParameters.IntensityThresholdType, double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the limit in intensity according to the given threshold.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getIntensityLogDistribution--">getIntensityLogDistribution()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the intensity of the log of the peaks intensities.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getIntensityMap--">getIntensityMap()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the peak list in a map where peaks are indexed by their
 intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getIntensityPrideCvTerm--">getIntensityPrideCvTerm()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the pride CV term for the ion match intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getIntensityScoreWeight--">getIntensityScoreWeight()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the intensity score weight.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getIntensityThresholdType--">getIntensityThresholdType()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns the intensity threshold type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#getIntensityValues--">getIntensityValues()</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getIntensityValuesAsArray--">getIntensityValuesAsArray()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the intensity values as an array.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getIntensityValuesNormalizedAsArray--">getIntensityValuesNormalizedAsArray()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the intensity values as an array normalized against the largest
 peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/FragmentItensityPredictor.html#getIntentisy-com.compomics.util.experiment.biology.ions.Ion-">getIntentisy(Ion)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/FragmentItensityPredictor.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation">FragmentItensityPredictor</a></dt>
<dd>
<div class="block">Returns the intensity expected for the given peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getIntermediateScore-int-">getIntermediateScore(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the desired intermediate score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getIntermediateScores--">getIntermediateScores()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the intermediate scores map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#getInterpolation-java.util.ArrayList-java.lang.Double-java.lang.Double-">getInterpolation(ArrayList&lt;Double&gt;, Double, Double)</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Returns the interpolation of a list of hyperscores using a linear
 interpolation of the form result = a * log(score) + b.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#getInterpolation-double-double-double-">getInterpolation(double, double, double)</a></span> - Static method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Returns the interpolated value for a given score in log.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#getInterpolationValues-int:A-boolean-">getInterpolationValues(int[], boolean)</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Returns the interpolation values for the given scores in the form {a, b}.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#getInterpolationValues-java.util.HashMap-boolean-">getInterpolationValues(HashMap&lt;Integer, Integer&gt;, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Returns the interpolation values for the given score histogram in the
 form {a, b}.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html#getIonAtRow-int-">getIonAtRow(int)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html" title="class in com.compomics.util.gui.spectrum">IonLabelColorTableModel</a></dt>
<dd>
<div class="block">Returns the ion type at the given row.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getIonMassErrorPrideCvTerm-int-int-">getIonMassErrorPrideCvTerm(int, int)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the pride CV term for the ion match error.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotator.html#getIonMatches-com.compomics.util.experiment.mass_spectrometry.indexes.SpectrumIndex-int-">getIonMatches(SpectrumIndex, int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">FragmentAnnotator</a></dt>
<dd>
<div class="block">Returns the ions matched in the given spectrum at the given charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotatorNL.html#getIonMatches-com.compomics.util.experiment.mass_spectrometry.indexes.SpectrumIndex-int-">getIonMatches(SpectrumIndex, int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/FragmentAnnotatorNL.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">FragmentAnnotatorNL</a></dt>
<dd>
<div class="block">Returns the ions matched in the given spectrum at the given charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/ImmoniumIonAnnotator.html#getIonMatches-com.compomics.util.experiment.mass_spectrometry.indexes.SpectrumIndex-">getIonMatches(SpectrumIndex)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/ImmoniumIonAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">ImmoniumIonAnnotator</a></dt>
<dd>
<div class="block">Returns the ions matched in the given spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/PrecursorAnnotator.html#getIonMatches-com.compomics.util.experiment.mass_spectrometry.indexes.SpectrumIndex-int-int-">getIonMatches(SpectrumIndex, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/PrecursorAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">PrecursorAnnotator</a></dt>
<dd>
<div class="block">Returns the ions matched in the given spectrum at the given charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/ReporterIonAnnotator.html#getIonMatches-com.compomics.util.experiment.mass_spectrometry.indexes.SpectrumIndex-">getIonMatches(SpectrumIndex)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/ReporterIonAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">ReporterIonAnnotator</a></dt>
<dd>
<div class="block">Returns the ions matched in the given spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.html#getIonMatches-com.compomics.util.experiment.mass_spectrometry.indexes.SpectrumIndex-int-int-">getIonMatches(SpectrumIndex, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">SimplePeptideAnnotator</a></dt>
<dd>
<div class="block">Returns the ion matches for the given spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getIons--">getIons()</a></span> - Static method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Getter for the list of ion symbols used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#getIonType-java.lang.String-">getIonType(String)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Returns the ion index corresponding to the given symbol in the drop down
 menu.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getIonTypes--">getIonTypes()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns the type of ions annotated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#getIonTypes--">getIonTypes()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Returns the map of ions to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html#getIsotope--">getIsotope()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html" title="class in com.compomics.util.experiment.biology.atoms">AtomImpl</a></dt>
<dd>
<div class="block">Returns the isotope, 0 for monoisotope.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomPanel.html#getIsotope--">getIsotope()</a></span> - Method in class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomPanel.html" title="class in com.compomics.util.gui.atoms">AtomPanel</a></dt>
<dd>
<div class="block">Returns the currently selected isotope.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getIsotopeCorrection--">getIsotopeCorrection()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the isotope correction setting.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#getIsotopeMass-int-">getIsotopeMass(int)</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">Returns the mass corresponding to the given isotope number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getIsotopeMzTolerance--">getIsotopeMzTolerance()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the deisotoping m/z tolerance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html#getIsotopeNumber-java.lang.Integer-">getIsotopeNumber(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html" title="class in com.compomics.util.experiment.biology.atoms">AtomImpl</a></dt>
<dd>
<div class="block">Returns the isotope number corresponding to the given rounded mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getIsotopeNumber-int-int-">getIsotopeNumber(int, int)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the distance in number of neutrons between the experimental mass
 and theoretic mass, image of the isotope number: 1 typically indicates
 C13 isotope.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#getIsotopeNumber-double-int-int-">getIsotopeNumber(double, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Returns the precursor isotope number according to the number of protons.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getIsotopicDistribution--">getIsotopicDistribution()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method gives the IsotopicDistribution for the sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html#getItem-java.lang.String-">getItem(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">AssumptionFilterItem</a></dt>
<dd>
<div class="block">Returns the item designated by the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html#getItem-java.lang.String-">getItem(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PeptideFilterItem</a></dt>
<dd>
<div class="block">Returns the item designated by the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html#getItem-java.lang.String-">getItem(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">ProteinFilterItem</a></dt>
<dd>
<div class="block">Returns the item designated by the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html#getItem-java.lang.String-">getItem(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PsmFilterItem</a></dt>
<dd>
<div class="block">Returns the item designated by the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getItemsNames--">getItemsNames()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Returns the name of the items used to filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getIterativeReplaceEvalue--">getIterativeReplaceEvalue()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the e-value threshold to use to replace a hit for the iterative
 search.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getIterativeSequenceEvalue--">getIterativeSequenceEvalue()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the e-value threshold to use to consider a sequence for the
 iterative search.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getIterativeSpectrumEvalue--">getIterativeSpectrumEvalue()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the e-value threshold to use consider a spectrum for the
 iterative search.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getIterator-java.lang.Class-java.lang.String-">getIterator(Class, String)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns an iterator of all objects of a given class</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getiXAxisMax--">getiXAxisMax()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the maximum value of the x-axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getiXAxisMin--">getiXAxisMin()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the minimum value of the x-axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getiYAxisMax--">getiYAxisMax()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the maximum value of the y-axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getiYAxisMin--">getiYAxisMin()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the minimum value of the y-axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html#getJarFilePath--">getJarFilePath()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html" title="class in com.compomics.cli.identification_parameters">AbstractIdentificationParametersCli</a></dt>
<dd>
<div class="block">Returns the path to the jar file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLI.html#getJarFilePath--">getJarFilePath()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLI.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersCLI</a></dt>
<dd>
<div class="block">Returns the path to the jar file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/CompomicsWrapper.html#getJarFilePath-java.lang.String-java.lang.String-">getJarFilePath(String, String)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/CompomicsWrapper.html" title="class in com.compomics.software">CompomicsWrapper</a></dt>
<dd>
<div class="block">Returns the path to the jar file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/MavenJarFile.html#getJarPath--">getJarPath()</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/MavenJarFile.html" title="class in com.compomics.software.autoupdater">MavenJarFile</a></dt>
<dd>
<div class="block">Returns the path to the jar file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getJavaHome--">getJavaHome()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the Java Home folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/CompomicsWrapper.html#getJavaHomeAndOptions-java.lang.String-">getJavaHomeAndOptions(String)</a></span> - Method in class com.compomics.software.<a href="com/compomics/software/CompomicsWrapper.html" title="class in com.compomics.software">CompomicsWrapper</a></dt>
<dd>
<div class="block">Returns an array list containing the Java home plus any parameters to the
 JVM.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JOptionEditorPane.html#getJOptionEditorPane-java.lang.String-">getJOptionEditorPane(String)</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/JOptionEditorPane.html" title="class in com.compomics.util.gui">JOptionEditorPane</a></dt>
<dd>
<div class="block">Returns a JEditorPane with HTML support to be used in a JOptionsPane.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#getJsonStringFromFile-java.io.File-">getJsonStringFromFile(File)</a></span> - Method in class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">Convert JSON string from file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getKeepTerminalAminoAcids--">getKeepTerminalAminoAcids()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the option for keeping the terminal amino acids when generating
 the decoys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getKey--">getKey()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the reference key of a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getKey-java.lang.String-com.compomics.util.experiment.identification.matches.ModificationMatch:A-">getKey(String, ModificationMatch[])</a></span> - Static method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the reference key of a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationMatch.html#getKey--">getKey()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationMatch.html" title="class in com.compomics.util.experiment.identification">IdentificationMatch</a></dt>
<dd>
<div class="block">Returns the key of a match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html#getKey--">getKey()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html" title="class in com.compomics.util.experiment.identification.matches">PeptideMatch</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#getKey--">getKey()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#getKey--">getKey()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.UtilitiesPathKey.html#getKeyFromId-java.lang.String-">getKeyFromId(String)</a></span> - Static method in enum com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.UtilitiesPathKey.html" title="enum in com.compomics.software.settings">UtilitiesPathParameters.UtilitiesPathKey</a></dt>
<dd>
<div class="block">Returns the key from its id.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/gui/PathParametersDialog.html#getKeyToPathMap--">getKeyToPathMap()</a></span> - Method in class com.compomics.software.settings.gui.<a href="com/compomics/software/settings/gui/PathParametersDialog.html" title="class in com.compomics.software.settings.gui">PathParametersDialog</a></dt>
<dd>
<div class="block">Returns the path settings in a map: key | path.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#getKeyWords--">getKeyWords()</a></span> - Method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Returns the key words for this report.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getKnownMassDeltas--">getKnownMassDeltas()</a></span> - Static method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Get all the known mass deltas (if any).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/interfaces/SpectrumAnnotation.html#getLabel--">getLabel()</a></span> - Method in interface com.compomics.util.gui.interfaces.<a href="com/compomics/util/gui/interfaces/SpectrumAnnotation.html" title="interface in com.compomics.util.gui.interfaces">SpectrumAnnotation</a></dt>
<dd>
<div class="block">This method returns the label for the annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/DefaultSpectrumAnnotation.html#getLabel--">getLabel()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/DefaultSpectrumAnnotation.html" title="class in com.compomics.util.gui.spectrum">DefaultSpectrumAnnotation</a></dt>
<dd>
<div class="block">This method returns the label for the annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#getLabel--">getLabel()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Returns the label.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#getLabelColor--">getLabelColor()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Returns the label color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/TableProperties.html#getLabelWidth--">getLabelWidth()</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/TableProperties.html" title="class in com.compomics.util.gui">TableProperties</a></dt>
<dd>
<div class="block">Returns the label width for the sparklines.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html#getLastHeader--">getLastHeader()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.iterators.<a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html" title="class in com.compomics.util.experiment.io.biology.protein.iterators">FastaIterator</a></dt>
<dd>
<div class="block">Returns the header corresponding to the last protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getLastModified--">getLastModified()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns when the file was last modified.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html#getLastSelectedFolder--">getLastSelectedFolder()</a></span> - Method in class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SequenceDbDetailsDialog</a></dt>
<dd>
<div class="block">Returns the last selected folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#getLastSelectedFolder--">getLastSelectedFolder()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Returns the last selected folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/LastSelectedFolder.html#getLastSelectedFolder-java.lang.String-">getLastSelectedFolder(String)</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/LastSelectedFolder.html" title="class in com.compomics.util.io.file">LastSelectedFolder</a></dt>
<dd>
<div class="block">Returns the last selected folder according to the given use case.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/LastSelectedFolder.html#getLastSelectedFolder--">getLastSelectedFolder()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/LastSelectedFolder.html" title="class in com.compomics.util.io.file">LastSelectedFolder</a></dt>
<dd>
<div class="block">Returns the last selected folder according to the given use case.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getLastSelectedFolder--">getLastSelectedFolder()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the last selected folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/WebDAO.html#getLatestVersionNumberFromRemoteRepo-java.net.URL-">getLatestVersionNumberFromRemoteRepo(URL)</a></span> - Static method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/WebDAO.html" title="class in com.compomics.software.autoupdater">WebDAO</a></dt>
<dd>
<div class="block">Fetches the latest Maven deployed version from a Maven built repository.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html#getLatinName-java.lang.Integer-">getLatinName(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">UniprotTaxonomy</a></dt>
<dd>
<div class="block">Returns the Latin name corresponding to the given NCBI taxon.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getLatinName-java.lang.Integer-">getLatinName(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the Latin name of the species corresponding to the given taxon
 according to the UniProt mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#getLeadingAccession--">getLeadingAccession()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Returns the leading accession for this match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getLegendDatasetAFourWay--">getLegendDatasetAFourWay()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Returns the legend location of dataset A in a four way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getLegendDatasetAThreeWay--">getLegendDatasetAThreeWay()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Returns the legend location of Dataset A in a three way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getLegendDatasetBFourWay--">getLegendDatasetBFourWay()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Returns the legend location of dataset B in a four way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getLegendDatasetBThreeWay--">getLegendDatasetBThreeWay()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Returns the legend location of dataset B in a three way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getLegendDatasetCFourWay--">getLegendDatasetCFourWay()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Returns the legend location of dataset C in a four way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getLegendDatasetCThreeWay--">getLegendDatasetCThreeWay()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Returns the legend location of dataset C in a three way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#getLegendDatasetDFourWay--">getLegendDatasetDFourWay()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Returns the legend location of dataset D in a four way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Protein.html#getLength--">getLength()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Protein.html" title="class in com.compomics.util.experiment.biology.proteins">Protein</a></dt>
<dd>
<div class="block">Returns the number of amino acids in the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Sequence.html#getLength--">getLength()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Sequence.html" title="interface in com.compomics.util.interfaces">Sequence</a></dt>
<dd>
<div class="block">This method reports on the length of the current sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequence.html#getLength--">getLength()</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide">NucleotideSequence</a></dt>
<dd>
<div class="block">This method reports on the length of the sequence for the current nucleotide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html#getLength--">getLength()</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html" title="class in com.compomics.util.nucleotide">NucleotideSequenceImpl</a></dt>
<dd>
<div class="block">This method reports on the length of the current sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getLength--">getLength()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method reports on the length of the current sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#getLength--">getLength()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This method reports on the length of the sequence for the current protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideVariantMatches.html#getLengthDiff--">getLengthDiff()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideVariantMatches.html" title="class in com.compomics.util.experiment.identification.matches">PeptideVariantMatches</a></dt>
<dd>
<div class="block">Returns the length difference induced by the variants.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getLengthInAminoAcid--">getLengthInAminoAcid()</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the amino acid length of the tag when mass gaps are considered
 like one amino acid</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#getLetter--">getLetter()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">Returns the single letter code of the atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getLevel--">getLevel()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns at which level the spectrum was recorded.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getLicense--">getLicense()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns the license information of this file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html#getLimitX--">getLimitX()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a></dt>
<dd>
<div class="block">Returns the maximal share of X's a match can contain, range [0.0-1.0].</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#getLine-int-">getLine(int)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/regressions/MedianRegression.html#getLinearRegression-java.util.ArrayList-java.util.ArrayList-">getLinearRegression(ArrayList&lt;Double&gt;, ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.statistics.linear_regression.regressions.<a href="com/compomics/util/math/statistics/linear_regression/regressions/MedianRegression.html" title="class in com.compomics.util.math.statistics.linear_regression.regressions">MedianRegression</a></dt>
<dd>
<div class="block">Returns a robust linear regression based on the median of statistics.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/regressions/SimpleLinearRegression.html#getLinearRegression-java.util.ArrayList-java.util.ArrayList-">getLinearRegression(ArrayList&lt;Double&gt;, ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.statistics.linear_regression.regressions.<a href="com/compomics/util/math/statistics/linear_regression/regressions/SimpleLinearRegression.html" title="class in com.compomics.util.math.statistics.linear_regression.regressions">SimpleLinearRegression</a></dt>
<dd>
<div class="block">Returns a simple linear regression.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#getLines--">getLines()</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getLink--">getLink()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the link.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getLinkHref--">getLinkHref()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the linkHref.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/AlignedListCellRenderer.html#getListCellRendererComponent-javax.swing.JList-java.lang.Object-int-boolean-boolean-">getListCellRendererComponent(JList, Object, int, boolean, boolean)</a></span> - Method in class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/AlignedListCellRenderer.html" title="class in com.compomics.util.gui.renderers">AlignedListCellRenderer</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/ToolTipComboBoxRenderer.html#getListCellRendererComponent-javax.swing.JList-java.lang.Object-int-boolean-boolean-">getListCellRendererComponent(JList, Object, int, boolean, boolean)</a></span> - Method in class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/ToolTipComboBoxRenderer.html" title="class in com.compomics.util.gui.renderers">ToolTipComboBoxRenderer</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigMathUtils.html#getLn10-java.math.MathContext-">getLn10(MathContext)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BigMathUtils.html" title="class in com.compomics.util.math">BigMathUtils</a></dt>
<dd>
<div class="block">Returns the value of ln(10) according to the mathContext.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getLocalizationConfidence-int-">getLocalizationConfidence(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns the confidence of the PTM localization.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getLocalPrideFolder--">getLocalPrideFolder()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the local PRIDE folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#getLocation--">getLocation()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">This method returns the location of the modification in the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#getLocation--">getLocation()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This method returns the location of the modification in the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/FileDAO.html#getLocationToDownloadOnDisk-java.lang.String-">getLocationToDownloadOnDisk(String)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/FileDAO.html" title="class in com.compomics.software.autoupdater">FileDAO</a></dt>
<dd>
<div class="block">Try to find an at least somewhat sane location to download files to.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/GUIFileDAO.html#getLocationToDownloadOnDisk-java.lang.String-">getLocationToDownloadOnDisk(String)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/GUIFileDAO.html" title="class in com.compomics.software.autoupdater">GUIFileDAO</a></dt>
<dd>
<div class="block">Try to find an at least somewhat sane location to download files to.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/HeadlessFileDAO.html#getLocationToDownloadOnDisk-java.lang.String-">getLocationToDownloadOnDisk(String)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/HeadlessFileDAO.html" title="class in com.compomics.software.autoupdater">HeadlessFileDAO</a></dt>
<dd>
<div class="block">Try to find an at least somewhat sane location to download files to.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SimpleNoiseDistribution.html#getLogIntensityAtP-double-">getLogIntensityAtP(double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SimpleNoiseDistribution.html" title="class in com.compomics.util.experiment.mass_spectrometry">SimpleNoiseDistribution</a></dt>
<dd>
<div class="block">Returns the log10 intensity at a given upper tale cumulative probability.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getLongestAminoAcidSequence--">getLongestAminoAcidSequence()</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the longest amino acid sequence contained in this tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/IonFactory.html#getLossesMass-com.compomics.util.experiment.biology.ions.NeutralLoss:A-">getLossesMass(NeutralLoss[])</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/IonFactory.html" title="class in com.compomics.util.experiment.biology.ions">IonFactory</a></dt>
<dd>
<div class="block">Convenience summing the masses of various neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getLowerClearMzRange--">getLowerClearMzRange()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the lower mass value for the clear mz range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#getLowerPrecursorMass--">getLowerPrecursorMass()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Returns the lower precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getLowIntensityCutOff--">getLowIntensityCutOff()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the low intensity cut-off as percentage of the most intense ion
 peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#getMainTitle--">getMainTitle()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Returns the main title of the report.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/styles/DefaultStyle.html#getMainTitleRowHeight--">getMainTitleRowHeight()</a></span> - Method in class com.compomics.util.io.export.styles.<a href="com/compomics/util/io/export/styles/DefaultStyle.html" title="class in com.compomics.util.io.export.styles">DefaultStyle</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/WorkbookStyle.html#getMainTitleRowHeight--">getMainTitleRowHeight()</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/WorkbookStyle.html" title="interface in com.compomics.util.io.export">WorkbookStyle</a></dt>
<dd>
<div class="block">Returns the row height for the main title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/styles/DefaultStyle.html#getMainTitleStyle--">getMainTitleStyle()</a></span> - Method in class com.compomics.util.io.export.styles.<a href="com/compomics/util/io/export/styles/DefaultStyle.html" title="class in com.compomics.util.io.export.styles">DefaultStyle</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/WorkbookStyle.html#getMainTitleStyle--">getMainTitleStyle()</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/WorkbookStyle.html" title="interface in com.compomics.util.io.export">WorkbookStyle</a></dt>
<dd>
<div class="block">Returns the cell style for the main title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getManualValidation--">getManualValidation()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Returns the manually validated items.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getManualValidation--">getManualValidation()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Indicates whether the match validation was manually inspected.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html#getMap--">getMap()</a></span> - Method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html" title="class in com.compomics.util.experiment.identification.modification">ModificationtableContent</a></dt>
<dd>
<div class="block">Get the map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>
<div class="block">Returns the mass of the atomic chain as sum of the individual atoms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html" title="class in com.compomics.util.experiment.biology.atoms">AtomImpl</a></dt>
<dd>
<div class="block">Returns the mass of the atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/ElementaryElement.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/ElementaryElement.html" title="class in com.compomics.util.experiment.biology.atoms">ElementaryElement</a></dt>
<dd>
<div class="block">Returns the mass of the element.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#getMass-int-">getMass(int)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>
<div class="block">Get the glycan mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Returns the mass of the neutral loss, from the atomic composition if
 available, from the mass field otherwise.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLossCombination.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLossCombination.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLossCombination</a></dt>
<dd>
<div class="block">Returns the mass of this combination.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Getter for the mass difference induced by this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the mass, does not attempt to estimate it.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getMass-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getMass(ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the mass, estimates it if not done before.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">MassGap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the mass of the tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getMass-boolean-boolean-">getMass(boolean, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the theoretic mass of the tag, eventually without terminal gaps.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/TagComponent.html#getMass--">getMass()</a></span> - Method in interface com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/TagComponent.html" title="interface in com.compomics.util.experiment.identification.amino_acid_tags">TagComponent</a></dt>
<dd>
<div class="block">Returns the mass of the tag component.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Returns the mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html#getMass-int-">getMass(int)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">PrecursorMap</a></dt>
<dd>
<div class="block">Returns the mass associated with the given bin, the middle of the bin.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getMass-int-">getMass(int)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>
<div class="block">Returns the mass associated with the given bin, the middle of the bin.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#getMass-int-">getMass(int)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>
<div class="block">Returns the mass of the compound with the given charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#getMass-int-">getMass(int)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Returns the mass of the precursor with the given charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Sequence.html#getMass--">getMass()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Sequence.html" title="interface in com.compomics.util.interfaces">Sequence</a></dt>
<dd>
<div class="block">This method will return the mass for the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequence.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide">NucleotideSequence</a></dt>
<dd>
<div class="block">This method returns the nucleotide sequence weight in Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html" title="class in com.compomics.util.nucleotide">NucleotideSequenceImpl</a></dt>
<dd>
<div class="block">This method will return the mass for the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method calculates the mass for the current sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#getMass--">getMass()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This method returns the protein weight in Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html#getMassAnalyzer--">getMassAnalyzer()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">NovorParameters</a></dt>
<dd>
<div class="block">Returns the mass analyzer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#getMassDelta-java.lang.String-int-">getMassDelta(String, int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This method returns the mass delta for the specified residue, measured either
 monoisotopically or averaged.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#getMassDelta-java.lang.String-int-">getMassDelta(String, int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">This method returns the mass delta for the specified residue, measured either
 monoisotopically or averaged.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getMassErrorDistribution-java.lang.String-">getMassErrorDistribution(String)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the precursor mass error distribution of validated peptides in a
 spectrum file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getMassGap--">getMassGap()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Returns the mass gap comprised in this ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#getMassPlusProton-int-">getMassPlusProton(int)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Returns the mass of the precursor with the given charge plus a single
 proton.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#getMassShift--">getMassShift()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the m/z shift applied to the fragment ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#getMassShiftCTerm--">getMassShiftCTerm()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the C-terminal m/z shift applied to all reverse ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#getMassShiftNTerm--">getMassShiftNTerm()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the N-terminal m/z shift applied to all forward ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getMatchingAminoAcid-char-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getMatchingAminoAcid(char, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns a matching amino acid using the given preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getMatchingKey--">getMatchingKey()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the key accounting for sequence matching preferences</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getMatchingKey-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getMatchingKey(SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns a unique key for the peptide when considering the given matching
 preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getMatchingPeaks-double-">getMatchingPeaks(double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>
<div class="block">Returns the peaks matching the given m/z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getMatchingSequence-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getMatchingSequence(String, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the matching sequence of a given sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html#getMatchingSpectra-double-">getMatchingSpectra(double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">PrecursorMap</a></dt>
<dd>
<div class="block">Returns a list containing the precursors matching the given m/z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html#getMatchingType-int-">getMatchingType(int)</a></span> - Static method in enum com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html" title="enum in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters.MatchingType</a></dt>
<dd>
<div class="block">Returns the matching type corresponding to the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getMatchKey-com.compomics.util.experiment.biology.ions.Ion-int-">getMatchKey(Ion, int)</a></span> - Static method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the key for the ion match uniquely representing a peak
 annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getMatchKey-com.compomics.util.experiment.biology.ions.Ion-int-com.compomics.util.experiment.identification.spectrum_annotation.IonMatchKeysCache-">getMatchKey(Ion, int, IonMatchKeysCache)</a></span> - Static method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the key for the ion match uniquely representing a peak
 annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getMatchKey-int-int-int-java.lang.String-int-">getMatchKey(int, int, int, String, int)</a></span> - Static method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the key based on the different attributes of a match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/IonMatchKeysCache.html#getMatchKey-com.compomics.util.experiment.biology.ions.Ion-int-">getMatchKey(Ion, int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/IonMatchKeysCache.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">IonMatchKeysCache</a></dt>
<dd>
<div class="block">Returns the key for the ion match uniquely representing a peak
 annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getMatchValidationLevel--">getMatchValidationLevel()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the validation level of the match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#getMatchValidationLevel-java.lang.String-">getMatchValidationLevel(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>
<div class="block">Returns the match validation level associated to a given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#getMatchValidationLevel-int-">getMatchValidationLevel(int)</a></span> - Static method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>
<div class="block">Returns the match validation level associated to a given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#getMatchValidationLevel--">getMatchValidationLevel()</a></span> - Method in class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Returns the lowest validation level considered as an integer as indexed
 in the MatchValidationLevel enum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/FileDAO.html#getMavenJarFileFromFolderWithArtifactId-java.io.File-java.lang.String-">getMavenJarFileFromFolderWithArtifactId(File, String)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/FileDAO.html" title="class in com.compomics.software.autoupdater">FileDAO</a></dt>
<dd>
<div class="block">Fetches a Maven built jar file from a folder for the given artifact id
 (e.g peptideshaker or ms-lims).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxCharge--">getMaxCharge()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the maximal charge among found charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getMaxCharge--">getMaxCharge()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns the maximal charge found in the mgf file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMaxCharge-java.lang.String-">getMaxCharge(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the max precursor charge encountered for the given mgf file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMaxCharge--">getMaxCharge()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the max precursor charge encountered among all loaded mgf files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getMaxChargeSearched--">getMaxChargeSearched()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the maximal charge searched.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getMaxCombinations--">getMaxCombinations()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the maximal number of combinations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getMaxDynamicMods--">getMaxDynamicMods()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of dynamic mods.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getMaxDynamicMods--">getMaxDynamicMods()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of variable modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getMaxEValue--">getMaxEValue()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the maximal e-value searched for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#getMaxEValue--">getMaxEValue()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns the maximal e-value searched for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigFunctions.html#getMaxExp-java.math.MathContext-">getMaxExp(MathContext)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BigFunctions.html" title="class in com.compomics.util.math">BigFunctions</a></dt>
<dd>
<div class="block">Returns the estimated maximal value exp can be calculated on according to
 the mathContext.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getMaxFragmentCharge--">getMaxFragmentCharge()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the maximum fragment ion charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getMaxFragmentCharge--">getMaxFragmentCharge()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the maximal fragment charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getMaxFragmentPerSeries--">getMaxFragmentPerSeries()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the maximal number of fragments to retain per series.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getMaxHitsPerSpectrumPerCharge--">getMaxHitsPerSpectrumPerCharge()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the maximal number of hits searched per spectrum and per charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Monitorable.html#getMaximum--">getMaximum()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Monitorable.html" title="interface in com.compomics.util.interfaces">Monitorable</a></dt>
<dd>
<div class="block">This method reports on the total amount steps the task can take before
 completion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableFileInputStream.html#getMaximum--">getMaximum()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableFileInputStream.html" title="class in com.compomics.util.io.file">MonitorableFileInputStream</a></dt>
<dd>
<div class="block">This method reports on the maximum scale for the monitor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableInputStream.html#getMaximum--">getMaximum()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableInputStream.html" title="class in com.compomics.util.io.file">MonitorableInputStream</a></dt>
<dd>
<div class="block">This method reports on the total amount of bytes that can be read
 from the monitored InputStream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#getMaximumExpectationValueRefinement--">getMaximumExpectationValueRefinement()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns the maximum expectation value to use for refinement.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html#getMaximumValue--">getMaximumValue()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialog</a></dt>
<dd>
<div class="block">Returns the current maximum value for the progressbar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getMaxIndex--">getMaxIndex()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the maximal index where amino acids are found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html#getMaxIntensity--">getMaxIntensity()</a></span> - Method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html" title="class in com.compomics.util.experiment.identification.modification">ModificationtableContent</a></dt>
<dd>
<div class="block">Returns the max intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getMaxIntensity--">getMaxIntensity()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns the maximum precursor intensity in this file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getMaxIntensity--">getMaxIntensity()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the max intensity value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMaxIntensity-java.lang.String-">getMaxIntensity(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the maximum precursor intensity for the desired file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMaxIntensity--">getMaxIntensity()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the maximum precursor intensity for the whole project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#getMaxIsotopes--">getMaxIsotopes()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Returns the maximal number of isotopes allowed (inclusive).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getMaxIsotopicCorrection--">getMaxIsotopicCorrection()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the maximal isotopic correction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/KMeansClustering.html#getMaxIterations--">getMaxIterations()</a></span> - Method in class com.compomics.util.math.clustering.<a href="com/compomics/util/math/clustering/KMeansClustering.html" title="class in com.compomics.util.math.clustering">KMeansClustering</a></dt>
<dd>
<div class="block">Returns the maximum number of iterations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#getMaxLoadedProteins--">getMaxLoadedProteins()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of proteins loaded into memory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#getMaxLoadedSpectra--">getMaxLoadedSpectra()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of spectra loaded into memory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/events/RescalingEvent.html#getMaxMass--">getMaxMass()</a></span> - Method in class com.compomics.util.gui.events.<a href="com/compomics/util/gui/events/RescalingEvent.html" title="class in com.compomics.util.gui.events">RescalingEvent</a></dt>
<dd>
<div class="block">Returns the maximum mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#getMaxMissedCleavages--">getMaxMissedCleavages()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Returns the maximum number of missed cleavages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#getMaxModifications--">getMaxModifications()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of modifications per peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#getMaxModificationSites--">getMaxModificationSites()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of modifications sites per modification per
 peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxMW--">getMaxMW()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the molecular weight of the fattest protein in the dataset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getMaxMz--">getMaxMz()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns the maximum m/z in this file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html#getMaxMz--">getMaxMz()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">PrecursorMap</a></dt>
<dd>
<div class="block">Returns the maximal m/z encountered among the precursors.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getMaxMz--">getMaxMz()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the max mz value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMaxMz-java.lang.String-">getMaxMz(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the maximum m/z for the desired file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMaxMz--">getMaxMz()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the maximum m/z for the whole project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#getMaxMzDeviation--">getMaxMzDeviation()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Returns the maximal m/z deviation allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getMaxMzLadders--">getMaxMzLadders()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the maximal m/z ladder length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#getMaxNeutralLosses--">getMaxNeutralLosses()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of water and ammonia losses per peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#getMaxNeutralLossesPerModification--">getMaxNeutralLossesPerModification()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns the maximum number identical modification specific losses per
 peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxNPeptides--">getMaxNPeptides()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the maximal amount of peptides in the proteins of the dataset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getMaxNSpectra--">getMaxNSpectra()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the maximum number of spectra accounted by a single peptide Match
 all found in a protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxNSpectra--">getMaxNSpectra()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the the maximal amount of PSMs in the proteins of the dataset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getMaxNumberOfModifications--">getMaxNumberOfModifications()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the maximal number of modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getMaxPadding--">getMaxPadding()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the max padding (distance between the axes and the border of the
 panel).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getMaxPeakCount--">getMaxPeakCount()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns the maximum peak count in this file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMaxPeakCount-java.lang.String-">getMaxPeakCount(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the max peak count encountered for the given mgf file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMaxPeakCount--">getMaxPeakCount()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the max peak count encountered among all loaded mgf files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getMaxPeakCount--">getMaxPeakCount()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the maximum peak count.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getMaxPeakCount--">getMaxPeakCount()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns the max peak count.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#getMaxPepLength--">getMaxPepLength()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Returns the maximal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#getMaxPeptideLength--">getMaxPeptideLength()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Returns the maximal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getMaxPeptideLength--">getMaxPeptideLength()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns the maximal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getMaxPeptideLength--">getMaxPeptideLength()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the maximal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMaxPeptideLength--">getMaxPeptideLength()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the maximal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getMaxPeptideLengthNoEnzyme--">getMaxPeptideLengthNoEnzyme()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the maximal peptide length to use when searching with no enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getMaxPeptideMass--">getMaxPeptideMass()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the maximal peptide mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxPeptidePrecursorErrorDa--">getMaxPeptidePrecursorErrorDa()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Return the max peptide precursor mass error in Dalton.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxPeptidePrecursorErrorPpm--">getMaxPeptidePrecursorErrorPpm()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the max peptide precursor mass error in ppm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getMaxPrecursorAdjustment--">getMaxPrecursorAdjustment()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the maximum precursor adjustment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getMaxPrecursorMass--">getMaxPrecursorMass()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the maximum precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getMaxPrecursorMass--">getMaxPrecursorMass()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns the maxPrecursorMass precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMaxPrecursorMass--">getMaxPrecursorMass()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the maxPrecursorMass precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#getMaxPrimaryProgressCounter--">getMaxPrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#getMaxPrimaryProgressCounter--">getMaxPrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#getMaxPrimaryProgressCounter--">getMaxPrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#getMaxPrimaryProgressCounter--">getMaxPrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#getMaxPrimaryProgressCounter--">getMaxPrimaryProgressCounter()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Returns the max primary progress counter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxProteinAveragePrecursorIntensity--">getMaxProteinAveragePrecursorIntensity()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the maximum protein average precursor intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxProteinKeyLength--">getMaxProteinKeyLength()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the max protein key length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxProteinSummedPrecursorIntensity--">getMaxProteinSummedPrecursorIntensity()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the maximum summed protein precursor intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#getMaxRank--">getMaxRank()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns the maximum rank.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getMaxRT--">getMaxRT()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns the maximal RT in this file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMaxRT-java.lang.String-">getMaxRT(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the maximum retention time in seconds for the desired file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMaxRT--">getMaxRT()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the maximum retention time in seconds for the whole project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#getMaxSecondaryProgressCounter--">getMaxSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#getMaxSecondaryProgressCounter--">getMaxSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#getMaxSecondaryProgressCounter--">getMaxSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#getMaxSecondaryProgressCounter--">getMaxSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#getMaxSecondaryProgressCounter--">getMaxSecondaryProgressCounter()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Returns the max secondary progress counter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getMaxSpectrumChargeRange--">getMaxSpectrumChargeRange()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the maximum charge added when the charge is missing for a given
 spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#getMaxSpectrumCount--">getMaxSpectrumCount()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Returns the maximal amount of PSMs for the peptides in the current
 peptide list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxSpectrumCounting--">getMaxSpectrumCounting()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the maximal spectrum counting value of the proteins of the
 dataset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMaxSpectrumMz--">getMaxSpectrumMz()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the max spectrum m/z to search for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getMaxTagCount--">getMaxTagCount()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of tags per spectrum to be generated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxTagPrecursorErrorDa--">getMaxTagPrecursorErrorDa()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Return the max tag precursor mass error in Dalton.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxTagPrecursorErrorPpm--">getMaxTagPrecursorErrorPpm()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the max tag precursor mass error in ppm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxValidatedPeptidesPerFraction--">getMaxValidatedPeptidesPerFraction()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the maximum validated peptides at the fraction level.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getMaxValidatedSpectraPerFraction--">getMaxValidatedSpectraPerFraction()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the maximum validated spectra at the fraction level.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/Distribution.html#getMaxValueForProbability-double-">getMaxValueForProbability(double)</a></span> - Method in interface com.compomics.util.math.statistics.<a href="com/compomics/util/math/statistics/Distribution.html" title="interface in com.compomics.util.math.statistics">Distribution</a></dt>
<dd>
<div class="block">The value after which the density function will be smaller than p.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html#getMaxValueForProbability-double-">getMaxValueForProbability(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html" title="class in com.compomics.util.math.statistics.distributions">BinomialDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getMaxValueForProbability-double-">getMaxValueForProbability(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html#getMaxValueForProbability-double-">getMaxValueForProbability(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#getMaxVariableModifications--">getMaxVariableModifications()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of variable modifications per peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMaxVariableModificationsPerPeptide--">getMaxVariableModificationsPerPeptide()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of variable modifications allowed on a single
 peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMaxVariableModificationsPerTypePerPeptide--">getMaxVariableModificationsPerTypePerPeptide()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of variable modifications of each modification
 type allowed on a single peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getMaxVariableMods--">getMaxVariableMods()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of variable modifications per peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getMaxXAxisValue--">getMaxXAxisValue()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method reports on the largest x-axis value in the point collection
 across all datasets.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html#getMaxXsInSequence--">getMaxXsInSequence()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">ProteinIteratorUtils</a></dt>
<dd>
<div class="block">Returns the maximal number of Xs to account for in a sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/scores/MDScore.html#getMDScore-java.util.ArrayList-com.compomics.util.experiment.biology.proteins.Peptide-java.util.ArrayList-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-java.lang.Integer-">getMDScore(ArrayList&lt;SpectrumIdentificationAssumption&gt;, Peptide, ArrayList&lt;String&gt;, SequenceMatchingParameters, Integer)</a></span> - Static method in class com.compomics.util.experiment.identification.modification.scores.<a href="com/compomics/util/experiment/identification/modification/scores/MDScore.html" title="class in com.compomics.util.experiment.identification.modification.scores">MDScore</a></dt>
<dd>
<div class="block">Returns the MD score for the given peptide in a spectrum match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getMean--">getMean()</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>
<div class="block">Returns the mean of the distribution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/HistogramUtils.html#getMedianValue-java.util.HashMap-">getMedianValue(HashMap&lt;Double, Integer&gt;)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/HistogramUtils.html" title="class in com.compomics.util.math">HistogramUtils</a></dt>
<dd>
<div class="block">Returns the median of a histogram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/HistogramUtils.html#getMedianValue-java.util.HashMap-long-">getMedianValue(HashMap&lt;Double, Integer&gt;, long)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/HistogramUtils.html" title="class in com.compomics.util.math">HistogramUtils</a></dt>
<dd>
<div class="block">Returns the median of a histogram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getMeek--">getMeek()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method will return an estimated 'net' HPLC retention
 time for the sequence based on the table by Meek.<br>
 It does NOT take a t0 value, specific to a setup, into account.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getMemoryParameter--">getMemoryParameter()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the upper memory limit in MB.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#getMemoryShare--">getMemoryShare()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Returns the share of heap size which can be used before emptying the
 cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#getMendianA--">getMendianA()</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Returns the rounded median of the as found in the previously interpolated
 scores.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#getMendianB--">getMendianB()</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Returns the rounded median of the bs found in the previously interpolated
 scores.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/GeneratorException.html#getMessage--">getMessage()</a></span> - Method in exception com.compomics.util.db.<a href="com/compomics/util/db/GeneratorException.html" title="class in com.compomics.util.db">GeneratorException</a></dt>
<dd>
<div class="block">This method returns a String with the description for this exception.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#getMessage--">getMessage()</a></span> - Method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Returns the message.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/StreamGobbler.html#getMessages--">getMessages()</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/StreamGobbler.html" title="class in com.compomics.util.io">StreamGobbler</a></dt>
<dd>
<div class="block">Returns the messages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getMethod--">getMethod()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getMethodId--">getMethodId()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the method ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html#getMethods--">getMethods()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethodFactory</a></dt>
<dd>
<div class="block">Returns the methods implemented in the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html#getMethodsNames--">getMethodsNames()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethodFactory</a></dt>
<dd>
<div class="block">Returns the name of the methods present in the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html#getMethodsNamesAsArray--">getMethodsNamesAsArray()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethodFactory</a></dt>
<dd>
<div class="block">Returns the methods names as array.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/Quantification.html#getMethodUsed--">getMethodUsed()</a></span> - Method in class com.compomics.util.experiment.quantification.<a href="com/compomics/util/experiment/quantification/Quantification.html" title="class in com.compomics.util.experiment.quantification">Quantification</a></dt>
<dd>
<div class="block">getter for the method used</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getMetrics--">getMetrics()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the metrics.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/UnitOfMeasurement.html#getMetricsPrefix--">getMetricsPrefix()</a></span> - Method in class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/UnitOfMeasurement.html" title="class in com.compomics.util.experiment.units">UnitOfMeasurement</a></dt>
<dd>
<div class="block">Returns the metrics prefix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMgfFileFromName-java.lang.String-">getMgfFileFromName(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the file associated to the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#getMgfFileName--">getMgfFileName()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>
<div class="block">Returns the spectrum file name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html#getMgfFileName--">getMgfFileName()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PNovoIdfileReader</a></dt>
<dd>
<div class="block">Returns the spectrum file name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMgfFileNames--">getMgfFileNames()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns a list of loaded mgf files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getMgfMaxSize--">getMgfMaxSize()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the max mgf file size before splitting.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getMgfNSpectra--">getMgfNSpectra()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Get the max number of spectra in an mgf file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html#getMH--">getMH()</a></span> - Method in class com.compomics.util.experiment.refinement_parameters.<a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html" title="class in com.compomics.util.experiment.refinement_parameters">PepnovoAssumptionDetails</a></dt>
<dd>
<div class="block">Returns the PepNovo mH.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getMinAnnotatedPeaks--">getMinAnnotatedPeaks()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the minimal number of annotated peaks a peptide should have.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getMinChargeSearched--">getMinChargeSearched()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the minimal charge searched.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html#getMinCtermMass--">getMinCtermMass()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">ProteinIteratorUtils</a></dt>
<dd>
<div class="block">Returns the minimal mass to consider for a c-terminus.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#getMinFragmentMz--">getMinFragmentMz()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns the minimal fragment m/z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getMinimalChargeForMultipleChargedFragments--">getMinimalChargeForMultipleChargedFragments()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the minimal precursor charge to account for multiply charged
 fragments in OMSSA.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getMinimalLocalizationConfidence--">getMinimalLocalizationConfidence()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns the minimal confidence among the PTM sites of this scoring.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html#getMinimumValue--">getMinimumValue()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialog</a></dt>
<dd>
<div class="block">Returns the current minimum value for the progressbar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getMinIndex--">getMinIndex()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the minimal index where amino acids are found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#getMinIsotopes--">getMinIsotopes()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Returns the minimal number of isotopes allowed (inclusive).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getMinIsotopicCorrection--">getMinIsotopicCorrection()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the minimal isotopic correction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#getMinMass-char:A-">getMinMass(char[])</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Returns the minimal mass that an amino acid sequence can have taking into
 account ambiguous amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/events/RescalingEvent.html#getMinMass--">getMinMass()</a></span> - Method in class com.compomics.util.gui.events.<a href="com/compomics/util/gui/events/RescalingEvent.html" title="class in com.compomics.util.gui.events">RescalingEvent</a></dt>
<dd>
<div class="block">Returns the minimum mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#getMinMissedCleavages--">getMinMissedCleavages()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Returns the minimum number of missed cleavages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html#getMinMz--">getMinMz()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">PrecursorMap</a></dt>
<dd>
<div class="block">Returns the minimal m/z encountered among the precursors.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getMinMz--">getMinMz()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the min mz value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getMinPeakIntensity--">getMinPeakIntensity()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the minimum peak intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getMinPeaks--">getMinPeaks()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the minimum number of peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getMinPeaks--">getMinPeaks()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the minimal number of peaks a spectrum should contain.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#getMinPeaksPerSpectrum--">getMinPeaksPerSpectrum()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns the minimal number of peaks per spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#getMinPepLength--">getMinPepLength()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Returns the maximal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#getMinPeptideLength--">getMinPeptideLength()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns the minimum peptide length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#getMinPeptideLength--">getMinPeptideLength()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Sets the minimal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getMinPeptideLength--">getMinPeptideLength()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Sets the minimal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getMinPeptideLength--">getMinPeptideLength()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the minimal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMinPeptideLength--">getMinPeptideLength()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Sets the minimal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getMinPeptideLengthNoEnzyme--">getMinPeptideLengthNoEnzyme()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the minimal peptide length to use when searching with no enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getMinPrecPerSpectrum--">getMinPrecPerSpectrum()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the minimal number of precursors per spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getMinPrecursorAdjustment--">getMinPrecursorAdjustment()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the minimum precursor adjustment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getMinPrecursorMass--">getMinPrecursorMass()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the minimum precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getMinPrecursorMass--">getMinPrecursorMass()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns the minimum precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMinPrecursorMass--">getMinPrecursorMass()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the minimum precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#getMinPrecursorMass--">getMinPrecursorMass()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns the minimal precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getMinRT--">getMinRT()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns the minimum RT in this file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMinRT-java.lang.String-">getMinRT(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the minimum retention time in seconds for the desired file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMinRT--">getMinRT()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the minimum retention time in seconds for the whole project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getMinSpectrumChargeRange--">getMinSpectrumChargeRange()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the minimum charge added when the charge is missing for a given
 spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMinSpectrumMz--">getMinSpectrumMz()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Return the min spectrum m/z to search for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMinSpectrumPeaks--">getMinSpectrumPeaks()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the min number of peaks in a spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getMinTerminiCleavages--">getMinTerminiCleavages()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns the minimum number of termini cleavages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#getMinus1--">getMinus1()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Returns the isotopic correction factor at -1 Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#getMinus2--">getMinus2()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Returns the isotopic correction factor at -2 Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/Distribution.html#getMinValueForProbability-double-">getMinValueForProbability(double)</a></span> - Method in interface com.compomics.util.math.statistics.<a href="com/compomics/util/math/statistics/Distribution.html" title="interface in com.compomics.util.math.statistics">Distribution</a></dt>
<dd>
<div class="block">The value before which the density function will be smaller than p.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html#getMinValueForProbability-double-">getMinValueForProbability(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html" title="class in com.compomics.util.math.statistics.distributions">BinomialDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getMinValueForProbability-double-">getMinValueForProbability(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html#getMinValueForProbability-double-">getMinValueForProbability(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getMinXAxisValue--">getMinXAxisValue()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method reports on the smallest x-axis value in the point collection
 across all datasets.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#getMiscleavages--">getMiscleavages()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">Simple getter for the number of allowed missed cleavages for the Enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#getMissedCleavages--">getMissedCleavages()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Returns the number of missed cleavages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableMap.html#getModel--">getModel()</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun">TableMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#getModification-java.lang.String-">getModification(String)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Returns the Modification indexed by its name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html#getModification--">getModification()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches">ModificationMatch</a></dt>
<dd>
<div class="block">Getter for the theoretic PTM name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#getModification-java.lang.String-">getModification(String)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Returns the back-ed up modification with the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationFactory.html#getModification-java.lang.String-int-">getModification(String, int)</a></span> - Static method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationFactory.html" title="class in com.compomics.util.protein">ModificationFactory</a></dt>
<dd>
<div class="block">This method will return a Modification instance from a title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationFactory.html#getModification-java.lang.String-java.lang.String-int-">getModification(String, String, int)</a></span> - Static method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationFactory.html" title="class in com.compomics.util.protein">ModificationFactory</a></dt>
<dd>
<div class="block">This method takes a code and a residue (the residue for the N-terminus is NTERMINUS and for the
 C-terminus CTERMINUS!) and converts this into a Modification instance if possible.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getModificationLocalizationParameters--">getModificationLocalizationParameters()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the modification localization scoring parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html#getModificationMass-java.lang.String-">getModificationMass(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">ProteinIteratorUtils</a></dt>
<dd>
<div class="block">Returns the mass corresponding to a given modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getModificationName-int-">getModificationName(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the name of the modification indexed by the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getModificationName-int-">getModificationName(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the name of the modification indexed by the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getModificationParameters--">getModificationParameters()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the modification settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html#getModificationPattern-java.lang.String-">getModificationPattern(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">ProteinIteratorUtils</a></dt>
<dd>
<div class="block">Returns the modification pattern that is targeted by the given
 modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html#getModificationPlotData-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.biology.modifications.Modification-int-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getModificationPlotData(Peptide, Modification, int, Spectrum, AnnotationParameters, SpecificAnnotationParameters, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html" title="class in com.compomics.util.experiment.identification.modification">ModificationtableContent</a></dt>
<dd>
<div class="block">Returns the modification plot series in the JFreechart format for one
 PSM.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html#getModificationProfile--">getModificationProfile()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">XTandemParametersDialog</a></dt>
<dd>
<div class="block">Returns the modification profile corresponding to the input by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html#getModificationProfileKey-int:A-">getModificationProfileKey(int[])</a></span> - Static method in class com.compomics.util.experiment.identification.modification.scores.<a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html" title="class in com.compomics.util.experiment.identification.modification.scores">PhosphoRS</a></dt>
<dd>
<div class="block">Returns a key as concatenated string for the given sites.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#getModifications--">getModifications()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Returns the names of all imported Modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getModifications--">getModifications()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">Simple getter for the modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#getModificationsAtRepresentativeSite-int-">getModificationsAtRepresentativeSite(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Returns the modifications which have a representative ambiguous site at
 the given site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#getModificationScoring-java.lang.String-">getModificationScoring(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Returns the modification scoring for the desired modification (null if
 none found).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html#getModificationTableContent-com.compomics.util.experiment.identification.spectrum_assumptions.PeptideAssumption-com.compomics.util.experiment.biology.modifications.Modification-int-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getModificationTableContent(PeptideAssumption, Modification, int, Spectrum, AnnotationParameters, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html" title="class in com.compomics.util.experiment.identification.modification">ModificationtableContent</a></dt>
<dd>
<div class="block">Get the table content.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html#getModificationToAdd--">getModificationToAdd()</a></span> - Method in class com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html" title="class in com.compomics.cli.modifications">ModificationsCLIInputBean</a></dt>
<dd>
<div class="block">Returns the modification to add.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html#getModificationToRemove--">getModificationToRemove()</a></span> - Method in class com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html" title="class in com.compomics.cli.modifications">ModificationsCLIInputBean</a></dt>
<dd>
<div class="block">Returns the name of the modification to remove.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getModificationType--">getModificationType()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Returns the modification type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getModifiedSequence-com.compomics.util.experiment.identification.IdentificationMatch-java.lang.String-">getModifiedSequence(IdentificationMatch, String)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the match sequence annotated with modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getModifiedSequence--">getModifiedSequence()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method will return the sequence with annotated modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getMolecularFormula--">getMolecularFormula()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method calculates the molecular formula based on the sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getMonoisotopicAtomChain--">getMonoisotopicAtomChain()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the monoisotopic atom chain representing this amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getMonoisotopicMass--">getMonoisotopicMass()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the mass of the amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html#getMonoisotopicMass--">getMonoisotopicMass()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">B</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html#getMonoisotopicMass--">getMonoisotopicMass()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">X</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html#getMonoisotopicMass--">getMonoisotopicMass()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Z</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#getMonoisotopicMass--">getMonoisotopicMass()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">Returns the monoisotopic mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#getMonoisotopicMassDelta-java.lang.String-">getMonoisotopicMassDelta(String)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">This method returns a double with the monoisotopic mass difference
 conferred on the sequence by this modification for the specified residue.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#getMonoisotopicMassDelta-java.lang.String-">getMonoisotopicMassDelta(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This method returns a double with the monoisotopic mass difference
 conferred on the sequence by this modification for the specified residue.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#getMonoisotopicMassDelta-java.lang.String-">getMonoisotopicMassDelta(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">This method returns a double with the monoisotopic mass difference
 conferred on the sequence by this modification for the specified residue.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMonoisotopicPrecursor--">getMonoisotopicPrecursor()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true if the precursor mass is monoisotopic, false if average.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/gui/MsConvertParametersDialog.html#getMsConvertParameters--">getMsConvertParameters()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.proteowizard.gui.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/gui/MsConvertParametersDialog.html" title="class in com.compomics.util.experiment.mass_spectrometry.proteowizard.gui">MsConvertParametersDialog</a></dt>
<dd>
<div class="block">Returns the parameters as created by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html#getMsFormat--">getMsFormat()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html" title="class in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsConvertParameters</a></dt>
<dd>
<div class="block">Returns the format to convert to.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#getMyriMatchFormat--">getMyriMatchFormat()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Get the MyriMatch enzyme format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#getMz--">getMz()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Getter for the m/z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/interfaces/SpectrumAnnotation.html#getMZ--">getMZ()</a></span> - Method in interface com.compomics.util.gui.interfaces.<a href="com/compomics/util/gui/interfaces/SpectrumAnnotation.html" title="interface in com.compomics.util.gui.interfaces">SpectrumAnnotation</a></dt>
<dd>
<div class="block">This method returns the M/Z of the feature to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/DefaultSpectrumAnnotation.html#getMZ--">getMZ()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/DefaultSpectrumAnnotation.html" title="class in com.compomics.util.gui.spectrum">DefaultSpectrumAnnotation</a></dt>
<dd>
<div class="block">This method returns the M/Z of the feature to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getMz-int-">getMz(int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method calculates the mass over charge ratio for a given charge
 for the current sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getMzAndIntensityAsArray--">getMzAndIntensityAsArray()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the m/z and intensity values as an array in increasing order
 sorted on m/z value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMzBinOffset--">getMzBinOffset()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the m/z bin offset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMzBinWidth--">getMzBinWidth()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the m/z bin width.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html#getMzErrorType-int-">getMzErrorType(int)</a></span> - Static method in enum com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html" title="enum in com.compomics.util.experiment.identification.matches">IonMatch.MzErrorType</a></dt>
<dd>
<div class="block">Returns the error type corresponding to the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getMzFidelityScoreWeight--">getMzFidelityScoreWeight()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the mzFidelity score weight.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html#getMzIdentMLVersion-int-">getMzIdentMLVersion(int)</a></span> - Static method in enum com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html" title="enum in com.compomics.util.experiment.io.identification">MzIdentMLVersion</a></dt>
<dd>
<div class="block">Returns the mzIdentML format corresponding to the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getMzidOutput--">getMzidOutput()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true if the mzid output is to be created.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getMzMLFileNames--">getMzMLFileNames()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns a list of loaded mzML files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getMZPrideCvTerm--">getMZPrideCvTerm()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the pride CV term for the ion match m/z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#getMzValues--">getMzValues()</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#getnAaDeletions--">getnAaDeletions()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Returns the number of amino acid deletions allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#getnAaInsertions--">getnAaInsertions()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Returns the number of amino acid insertions allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#getnAaSubstitutions--">getnAaSubstitutions()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Returns the number of amino acid substitutions allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#getnAaSwap--">getnAaSwap()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Returns the number of amino acid swaps allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#getName--">getName()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Returns the database name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/enumeration/CompomicsTools.html#getName--">getName()</a></span> - Method in enum com.compomics.util.enumeration.<a href="com/compomics/util/enumeration/CompomicsTools.html" title="enum in com.compomics.util.enumeration">CompomicsTools</a></dt>
<dd>
<div class="block">Getter for the name of the tool or library.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">Returns the name of the atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/ElementaryElement.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/ElementaryElement.html" title="class in com.compomics.util.experiment.biology.atoms">ElementaryElement</a></dt>
<dd>
<div class="block">Returns the name of the element.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Get the enzyme name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">This method returns the name of the reporter ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the name of the ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Getter for the modification name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html#getName-java.lang.Integer-">getName(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblGenomesSpecies</a></dt>
<dd>
<div class="block">Returns the name corresponding to the given NCBI taxon.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getName-java.lang.Integer-">getName(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the name of the species corresponding to the given taxon
 according to the UniProt mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Returns the name of this substitution matrix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/Filter.html#getName--">getName()</a></span> - Method in interface com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/Filter.html" title="interface in com.compomics.util.experiment.filtering">Filter</a></dt>
<dd>
<div class="block">Returns the name of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItem.html#getName--">getName()</a></span> - Method in interface com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItem.html" title="interface in com.compomics.util.experiment.filtering">FilterItem</a></dt>
<dd>
<div class="block">Returns the name of the item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Returns the name of the advocate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html#getName--">getName()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">AssumptionFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html#getName--">getName()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PeptideFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html#getName--">getName()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">ProteinFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html#getName--">getName()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PsmFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html#getName--">getName()</a></span> - Method in enum com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html" title="enum in com.compomics.util.experiment.identification.modification">ModificationLocalizationScore</a></dt>
<dd>
<div class="block">Returns the name of the score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns the name of the inspected protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#getName--">getName()</a></span> - Method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>
<div class="block">Returns the name of this possibility.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Returns the name of the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Returns the name of the reagent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html#getName--">getName()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethod</a></dt>
<dd>
<div class="block">Returns the name of the method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#getName--">getName()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Returns the name of the scheme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getName--">getName()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the name of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/CvTerm.html#getName--">getName()</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/CvTerm.html" title="class in com.compomics.util.pride">CvTerm</a></dt>
<dd>
<div class="block">Returns the name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Contact.html#getName--">getName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Contact.html" title="class in com.compomics.util.pride.prideobjects">Contact</a></dt>
<dd>
<div class="block">Returns the contact name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ContactGroup.html#getName--">getName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ContactGroup.html" title="class in com.compomics.util.pride.prideobjects">ContactGroup</a></dt>
<dd>
<div class="block">Returns the contact group name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Instrument.html#getName--">getName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Instrument.html" title="class in com.compomics.util.pride.prideobjects">Instrument</a></dt>
<dd>
<div class="block">Returns the instrument name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Protocol.html#getName--">getName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Protocol.html" title="class in com.compomics.util.pride.prideobjects">Protocol</a></dt>
<dd>
<div class="block">Returns the protocol name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html#getName--">getName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html" title="class in com.compomics.util.pride.prideobjects">ReferenceGroup</a></dt>
<dd>
<div class="block">Returns the references group name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Sample.html#getName--">getName()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Sample.html" title="class in com.compomics.util.pride.prideobjects">Sample</a></dt>
<dd>
<div class="block">Returns the name of the sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getNameForGoTerm-java.lang.String-">getNameForGoTerm(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the name of a GO term.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getNameWithGapAndNumber--">getNameWithGapAndNumber()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Returns the name with number and mass gap.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#getNameWithNumber--">getNameWithNumber()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Returns the name with number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getNameWithNumber--">getNameWithNumber()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Returns the name with number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getnAnnotatedMostIntensePeaks--">getnAnnotatedMostIntensePeaks()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the minimal number of annotated most intense peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html#getNCleavageSites-java.lang.String-java.util.ArrayList-">getNCleavageSites(String, ArrayList&lt;Enzyme&gt;)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html" title="class in com.compomics.util.experiment.identification.utils">ProteinUtils</a></dt>
<dd>
<div class="block">Returns the number of cleavage sites.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/settings/KMeansClusteringSettings.html#getnClusters--">getnClusters()</a></span> - Method in class com.compomics.util.math.clustering.settings.<a href="com/compomics/util/math/clustering/settings/KMeansClusteringSettings.html" title="class in com.compomics.util.math.clustering.settings">KMeansClusteringSettings</a></dt>
<dd>
<div class="block">Returns the number of clusters to use.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#getNColumns--">getNColumns()</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNConfidentPeptides-long-">getNConfidentPeptides(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of confident peptides for a given protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNConfidentProteins--">getNConfidentProteins()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of confident proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getnConfidentProteins--">getnConfidentProteins()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the number of confident proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNConfidentSpectra-long-">getNConfidentSpectra(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of confident spectra for a given protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNConfidentSpectraForPeptide-long-">getNConfidentSpectraForPeptide(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of confident spectra for a given peptide match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#getNEnzymaticTermini-int-int-java.lang.String-com.compomics.util.experiment.biology.enzymes.Enzyme-">getNEnzymaticTermini(int, int, String, Enzyme)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns the number of enzymatic termini for the given enzyme on this
 protein at the given location.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/GeneratorException.html#getNestedException--">getNestedException()</a></span> - Method in exception com.compomics.util.db.<a href="com/compomics/util/db/GeneratorException.html" title="class in com.compomics.util.db">GeneratorException</a></dt>
<dd>
<div class="block">This method returns the nested exception for this exception, or
 'null' if none is present.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#getNeutralLoss-java.lang.String-">getNeutralLoss(String)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Returns the neutral loss associated to the given name in the static map
 of the class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLossCombination.html#getNeutralLossCombination--">getNeutralLossCombination()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLossCombination.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLossCombination</a></dt>
<dd>
<div class="block">Returns the neutral losses in this combination.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#getNeutralLosses--">getNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#getNeutralLosses--">getNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#getNeutralLosses--">getNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#getNeutralLosses--">getNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#getNeutralLosses--">getNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#getNeutralLosses--">getNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#getNeutralLosses--">getNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getNeutralLosses--">getNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getNeutralLosses--">getNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the possible neutral losses of this ion type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/IonFactory.html#getNeutralLosses-com.compomics.util.parameters.identification.search.ModificationParameters-">getNeutralLosses(ModificationParameters)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/IonFactory.html" title="class in com.compomics.util.experiment.biology.ions">IonFactory</a></dt>
<dd>
<div class="block">Returns a list containing the default neutral losses and the losses found
 in the given modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getNeutralLosses--">getNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Returns the neutral losses possibly encountered with this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getNeutralLosses--">getNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns the considered neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getNeutralLossesAsString--">getNeutralLossesAsString()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the neutral loss (if any), the empty string if no loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getNeutralLossesAsString-com.compomics.util.experiment.biology.ions.NeutralLoss:A-">getNeutralLossesAsString(NeutralLoss[])</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the neutral loss (if any), the empty string if no loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/IonFactory.html#getNeutralLossesCombinations-java.util.HashSet-">getNeutralLossesCombinations(HashSet&lt;String&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/IonFactory.html" title="class in com.compomics.util.experiment.biology.ions">IonFactory</a></dt>
<dd>
<div class="block">Returns the possible neutral losses combinations as array of arrays of
 neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#getNeutralLossesMap--">getNeutralLossesMap()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Returns the map of neutral losses to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/ElementaryElement.html#getNeutronMassMultiple-int-">getNeutronMassMultiple(int)</a></span> - Static method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/ElementaryElement.html" title="class in com.compomics.util.experiment.biology.atoms">ElementaryElement</a></dt>
<dd>
<div class="block">Returns the mass of the neutron multiplied by i.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getNeutronThreshold--">getNeutronThreshold()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the mass after which exact neutron mass should be considered in
 the calculation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html#getNew--">getNew()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">AssumptionFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getNew--">getNew()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Returns a new empty filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html#getNew--">getNew()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html#getNew--">getNew()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html" title="class in com.compomics.util.experiment.identification.filtering">ProteinFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html#getNew--">getNew()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PsmFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAnnotationServerResultReader.html#getNextFeature--">getNextFeature()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAnnotationServerResultReader.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAnnotationServerResultReader</a></dt>
<dd>
<div class="block">Get the next feature in the XML string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/iterators/HeaderIterator.html#getNextHeader--">getNextHeader()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.iterators.<a href="com/compomics/util/experiment/io/biology/protein/iterators/HeaderIterator.html" title="class in com.compomics.util.experiment.io.biology.protein.iterators">HeaderIterator</a></dt>
<dd>
<div class="block">Returns the next header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/NoDigestionCombinationIterator.html#getNextPeptide--">getNextPeptide()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.iterators.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/NoDigestionCombinationIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">NoDigestionCombinationIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/NoDigestionIterator.html#getNextPeptide--">getNextPeptide()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.iterators.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/NoDigestionIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">NoDigestionIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/SpecificSingleEnzymeCombinationIterator.html#getNextPeptide--">getNextPeptide()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.iterators.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/SpecificSingleEnzymeCombinationIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">SpecificSingleEnzymeCombinationIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/SpecificSingleEnzymeIterator.html#getNextPeptide--">getNextPeptide()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.iterators.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/SpecificSingleEnzymeIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">SpecificSingleEnzymeIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/UnspecificCombinationIterator.html#getNextPeptide--">getNextPeptide()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.iterators.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/UnspecificCombinationIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">UnspecificCombinationIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/UnspecificIterator.html#getNextPeptide--">getNextPeptide()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.iterators.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/UnspecificIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">UnspecificIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/SequenceIterator.html#getNextPeptide--">getNextPeptide()</a></span> - Method in interface com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/SequenceIterator.html" title="interface in com.compomics.util.experiment.identification.protein_sequences.digestion">SequenceIterator</a></dt>
<dd>
<div class="block">Returns the next peptide that can be generated from the iterator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html#getNextProtein--">getNextProtein()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.iterators.<a href="com/compomics/util/experiment/io/biology/protein/iterators/FastaIterator.html" title="class in com.compomics.util.experiment.io.biology.protein.iterators">FastaIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinIterator.html#getNextProtein--">getNextProtein()</a></span> - Method in interface com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinIterator.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinIterator</a></dt>
<dd>
<div class="block">Returns the next protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/AmbiguousSequenceIterator.html#getNextSequence--">getNextSequence()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/AmbiguousSequenceIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences">AmbiguousSequenceIterator</a></dt>
<dd>
<div class="block">Returns the next sequence, null if none.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html#getnFeatures--">getnFeatures()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">FeaturesMap</a></dt>
<dd>
<div class="block">Returns the number of features in the map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#getNLines--">getNLines()</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#getNmissedCleavages-java.lang.String-">getNmissedCleavages(String)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Returns the number of missed cleavages in an amino acid sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getNMissedCleavages-com.compomics.util.experiment.biology.enzymes.Enzyme-">getNMissedCleavages(Enzyme)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the number of missed cleavages using the specified enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getNMissedCleavages-com.compomics.util.parameters.identification.search.DigestionParameters-">getNMissedCleavages(DigestionParameters)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the number of missed cleavages using the digestion preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#getnMissedCleavages-java.lang.String-">getnMissedCleavages(String)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Returns the number of allowed missed cleavages for the given enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getNoModPeptide-java.util.HashSet-">getNoModPeptide(HashSet&lt;String&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns a version of the peptide which does not contain the given list of
 modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#getNonDiagonalScore--">getNonDiagonalScore()</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>
<div class="block">Returns a score based on the non-diagonal values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNonEnzymatic-long-com.compomics.util.parameters.identification.search.DigestionParameters-">getNonEnzymatic(long, DigestionParameters)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns a list of non-enzymatic peptides for a given protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getNoProlineRuleSeries--">getNoProlineRuleSeries()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the id numbers of ion series to apply no product ions at proline
 rule at.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html#getNormalDistribution-java.util.ArrayList-">getNormalDistribution(ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NormalDistribution</a></dt>
<dd>
<div class="block">Returns the normal distribution corresponding to a given list of double
 calibrated on mean and standard deviation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramDialog.html#getNormalIcon--">getNormalIcon()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramDialog.html" title="class in com.compomics.util.gui">VennDiagramDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#getNormalIcon--">getNormalIcon()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#getNormalizationFactors--">getNormalizationFactors()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">Returns the normalization factors.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#getNormalize--">getNormalize()</a></span> - Method in class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Indicates whether the spectrum counting index should be normalized.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNormalizedSpectrumCounting-long-">getNormalizedSpectrumCounting(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the spectrum counting metric of the protein match of interest
 using the preference settings normalized to the injected protein amount
 using the spectrum counting preferences of the identification features
 generator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNormalizedSpectrumCounting-long-com.compomics.util.parameters.quantification.spectrum_counting.SpectrumCountingParameters-com.compomics.util.experiment.identification.peptide_shaker.Metrics-">getNormalizedSpectrumCounting(long, SpectrumCountingParameters, Metrics)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the spectrum counting metric of the protein match of interest
 using the preference settings normalized to the injected protein amount
 using the given spectrum counting preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNormalizedSpectrumCounting-long-com.compomics.util.experiment.units.UnitOfMeasurement-com.compomics.util.experiment.quantification.spectrumcounting.SpectrumCountingMethod-">getNormalizedSpectrumCounting(long, UnitOfMeasurement, SpectrumCountingMethod)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the spectrum counting metric of the protein match of interest
 using the preference settings normalized to the injected protein amount.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNormalizedSpectrumCounting-long-com.compomics.util.experiment.identification.peptide_shaker.Metrics-com.compomics.util.experiment.units.UnitOfMeasurement-java.lang.Double-com.compomics.util.experiment.quantification.spectrumcounting.SpectrumCountingMethod-">getNormalizedSpectrumCounting(long, Metrics, UnitOfMeasurement, Double, SpectrumCountingMethod)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the spectrum counting metric of the protein match of interest
 using the preference settings normalized to the injected protein amount.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getNote--">getNote()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the note.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/TableProperties.html#getNotSelectedRowHtmlTagFontColor--">getNotSelectedRowHtmlTagFontColor()</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/TableProperties.html" title="class in com.compomics.util.gui">TableProperties</a></dt>
<dd>
<div class="block">Returns the color to use for the HTML tags for the rows that are not
 selected, in HTML color code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#getNoVariantPreferences--">getNoVariantPreferences()</a></span> - Static method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Returns the preferences corresponding to no variants allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html#getNovorPtmMap--">getNovorPtmMap()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">NovorParameters</a></dt>
<dd>
<div class="block">Returns the Novor to utilities PTM map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getNPeaks--">getNPeaks()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the number of peaks in the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#getnPeaks--">getnPeaks()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns the number of most intense peaks to consider.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getnPeaksInDoubleChargeWindow--">getnPeaksInDoubleChargeWindow()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the number of peaks in doubly charged windows.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getnPeaksInSingleChargeWindow--">getnPeaksInSingleChargeWindow()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the number of peaks in singly charged windows.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getnProcessingNode--">getnProcessingNode()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns the number of processing nodes used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.Property.html#getNProperties--">getNProperties()</a></span> - Static method in enum com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.Property.html" title="enum in com.compomics.util.experiment.biology.aminoacids">AminoAcid.Property</a></dt>
<dd>
<div class="block">Returns the number of implemented properties.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#getNProteins--">getNProteins()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Returns the number of proteins for this match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNSpectra-long-">getNSpectra(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Estimates the number of spectra for the given protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getNSpectra--">getNSpectra()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns the number of imported spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getNSpectra-java.lang.String-">getNSpectra(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the number of spectra in the desired file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getNSpectra--">getNSpectra()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the total number of spectra in all files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getNTargetedAA-int-">getNTargetedAA(int)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the number of targeted amino acids at the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#getNtermAsString-boolean-java.lang.String:A...-">getNtermAsString(boolean, String[]...)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns the N-terminal annotation as string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getNTerminal-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getNTerminal(boolean, ModificationParameters, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the N-terminal tag of this tag as a string for sequence display.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getNTerminalGap--">getNTerminalGap()</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the N-terminal gap of the tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#getnTermModification--">getnTermModification()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Returns the N-term modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html#getNtermModification-boolean-char-java.lang.String-">getNtermModification(boolean, char, String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">ProteinIteratorUtils</a></dt>
<dd>
<div class="block">Returns the N-term modification for the given amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#getNTermTruncatedProtein-int-">getNTermTruncatedProtein(int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This method truncates the sequence for this protein on the N-terminus
 to the requested size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getNTermTruncatedSequence-int-">getNTermTruncatedSequence(int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method will return an AASequenceImpl that represents
 an N-terminal truncation of the current sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/tools/ProcessingParameters.html#getnThreads--">getnThreads()</a></span> - Method in class com.compomics.util.parameters.tools.<a href="com/compomics/util/parameters/tools/ProcessingParameters.html" title="class in com.compomics.util.parameters.tools">ProcessingParameters</a></dt>
<dd>
<div class="block">Returns the number or threads to use.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#getNumber-java.lang.Class-">getNumber(Class)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Returns the number of instances of a given class stored in the db.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#getNumber--">getNumber()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Returns the number of the fragment in the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getNumber--">getNumber()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Returns the number of the fragment in the tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getNumber-java.lang.Class-">getNumber(Class)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns the number of objects of a given class</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getNumberOfBatches--">getNumberOfBatches()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Set the number of batches per node to strive for when using the MPI-based
 parallelization features.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getNumberOfCandidates--">getNumberOfCandidates()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the number of candidates.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/KMeansClustering.html#getNumberOfClusters--">getNumberOfClusters()</a></span> - Method in class com.compomics.util.math.clustering.<a href="com/compomics/util/math/clustering/KMeansClustering.html" title="class in com.compomics.util.math.clustering">KMeansClustering</a></dt>
<dd>
<div class="block">Returns the number of clusters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html#getNumberOfDataPointsInPlot--">getNumberOfDataPointsInPlot()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorBubblePlot</a></dt>
<dd>
<div class="block">Returns the current number of data points in the mass error plot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorPlot.html#getNumberOfDataPointsInPlot--">getNumberOfDataPointsInPlot()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorPlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorPlot</a></dt>
<dd>
<div class="block">Returns the current number of data points in the mass error plot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#getNumberOfLines-java.io.File-">getNumberOfLines(File)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Returns the number of lines in the given file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#getNumberOfModificationsPerPeptide--">getNumberOfModificationsPerPeptide()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of modifications per peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#getNumberOfPeptides--">getNumberOfPeptides()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Returns the number of peptides to report per spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getNumberOfSpectrumMatches--">getNumberOfSpectrumMatches()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of spectrum matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#getNumberOfSpectrumMatches--">getNumberOfSpectrumMatches()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of spectrum matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getNumberOfSpectrumMatches--">getNumberOfSpectrumMatches()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns the maximum number of spectrum matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getNumberOfSpectrumMatches--">getNumberOfSpectrumMatches()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the number of PSMs to report per spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#getNumberTolerableTermini--">getNumberTolerableTermini()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Returns the number of tolerable termini.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getNumChargeStates--">getNumChargeStates()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the number of charge states.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#getNumericParameter-java.lang.String-">getNumericParameter(String)</a></span> - Method in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getNumIntensityClasses--">getNumIntensityClasses()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the number of intensity classes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getNumIntensityClasses--">getNumIntensityClasses()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns the number of intensity classes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getNUnique--">getNUnique()</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the number of amino acids excluding combinations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNUniquePeptides-long-">getNUniquePeptides(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of unique peptides for this protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNUniqueValidatedPeptides-long-">getNUniqueValidatedPeptides(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of unique validated peptides for this protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNValidatedPeptides-long-">getNValidatedPeptides(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of validated peptides for a given protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNValidatedProteinGroups-com.compomics.util.experiment.biology.proteins.Peptide-">getNValidatedProteinGroups(Peptide)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Indicates whether a peptide is found in a single protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNValidatedProteinGroups-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.waiting.WaitingHandler-">getNValidatedProteinGroups(Peptide, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Indicates whether a peptide is found in a single protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNValidatedProteins--">getNValidatedProteins()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of validated proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getnValidatedProteins--">getnValidatedProteins()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the number of validated proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#getnValidatedPsms--">getnValidatedPsms()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Returns the number of validated PSMs for the currently selected peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNValidatedPsms--">getNValidatedPsms()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of validated PSMs for the last selected peptide /!\
 This value is only available after getSortedPsmKeys has been called.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNValidatedSpectra-long-">getNValidatedSpectra(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of validated spectra for a given protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getNValidatedSpectraForPeptide-long-">getNValidatedSpectraForPeptide(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the number of validated spectra for a given peptide match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getNVariableModifications-double-">getNVariableModifications(double)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the number of variable modifications found with the given mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getNVariableModifications--">getNVariableModifications()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the number of modifications carried by this peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#getnVariants--">getnVariants()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Returns the total number of variants allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#getnX--">getnX()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Returns the number of Xs already considered in this draft.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#getObject-java.lang.Long-">getObject(Long)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Returns the objects if present in the cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#getObject-com.compomics.util.experiment.identification.IdentificationFeaturesCache.ObjectType-long-">getObject(IdentificationFeaturesCache.ObjectType, long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Returns an object if present in the cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/BoxedObject.html#getObject--">getObject()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/BoxedObject.html" title="class in com.compomics.util.general">BoxedObject</a></dt>
<dd>
<div class="block">Returns the boxed object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#getObjectsCache--">getObjectsCache()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Returns the cache used by this database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getObjectsDB--">getObjectsDB()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns the objects database used in this class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#getObjectsIterator-java.lang.Class-java.lang.String-">getObjectsIterator(Class, String)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Returns an iterator of all objects of a given class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html#getObservableAminoAcids-java.lang.String-java.util.ArrayList-double-">getObservableAminoAcids(String, ArrayList&lt;Enzyme&gt;, double)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html" title="class in com.compomics.util.experiment.identification.utils">ProteinUtils</a></dt>
<dd>
<div class="block">Returns the observable amino acids in the sequence when using the given
 enzymes with the given maximal peptide length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getObservableCoverage-long-">getObservableCoverage(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the best protein coverage possible according to the given
 cleavage settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html#getObservableLength-java.lang.String-java.util.ArrayList-double-">getObservableLength(String, ArrayList&lt;Enzyme&gt;, double)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html" title="class in com.compomics.util.experiment.identification.utils">ProteinUtils</a></dt>
<dd>
<div class="block">Returns the number of observable amino acids in the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getObservedFractionalMassesAll--">getObservedFractionalMassesAll()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the observed molecular masses in kDa for each fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#getOccurrence-com.compomics.util.experiment.biology.atoms.Atom-java.lang.Integer-">getOccurrence(Atom, Integer)</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>
<div class="block">Returns the occurrence of a given atom in the chain.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicElement.html#getOccurrence--">getOccurrence()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicElement.html" title="class in com.compomics.util.general">IsotopicElement</a></dt>
<dd>
<div class="block">Getter for the occurrence of this element</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomPanel.html#getOccurrence--">getOccurrence()</a></span> - Method in class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomPanel.html" title="class in com.compomics.util.gui.atoms">AtomPanel</a></dt>
<dd>
<div class="block">Returns the occurrence of the selected atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#getOccurrence-java.lang.String-char-">getOccurrence(String, char)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Returns the occurrence of a character in a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/modification/ModificationsDialog.html#getOlsAccessionLink-java.lang.String-">getOlsAccessionLink(String)</a></span> - Method in class com.compomics.util.gui.modification.<a href="com/compomics/util/gui/modification/ModificationsDialog.html" title="class in com.compomics.util.gui.modification">ModificationsDialog</a></dt>
<dd>
<div class="block">Returns a web link to the given PSI-MOD CV term at
 https://www.ebi.ac.uk/ontology-lookup.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getOMSSAIndexes-java.io.File-">getOMSSAIndexes(File)</a></span> - Static method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Imports the OMSSA indexes from an XML file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getOmssaOutputTypes--">getOmssaOutputTypes()</a></span> - Static method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the output types available.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/CvTerm.html#getOntology--">getOntology()</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/CvTerm.html" title="class in com.compomics.util.pride">CvTerm</a></dt>
<dd>
<div class="block">Returns the ontology.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FileSystemAccessor.html#getOperatingSystem--">getOperatingSystem()</a></span> - Static method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FileSystemAccessor.html" title="class in com.compomics.util.io.file">FileSystemAccessor</a></dt>
<dd>
<div class="block">Get the operating system.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/CommandLineParser.html#getOptionParameter-java.lang.String-">getOptionParameter(String)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/CommandLineParser.html" title="class in com.compomics.util.general">CommandLineParser</a></dt>
<dd>
<div class="block">This method will report the option parameter for an option that can take
 a parameter itself.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/CommandLineParser.html#getOptions--">getOptions()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/CommandLineParser.html" title="class in com.compomics.util.general">CommandLineParser</a></dt>
<dd>
<div class="block">This method will report on all options that have been found, or return an
 empty String[] if none were present.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIParams.html#getOptionsAsString--">getOptionsAsString()</a></span> - Static method in enum com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIParams.html" title="enum in com.compomics.cli.enzymes">EnzymesCLIParams</a></dt>
<dd>
<div class="block">Returns the options as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/fasta/FastaParametersCLIParams.html#getOptionsAsString--">getOptionsAsString()</a></span> - Static method in enum com.compomics.cli.fasta.<a href="com/compomics/cli/fasta/FastaParametersCLIParams.html" title="enum in com.compomics.cli.fasta">FastaParametersCLIParams</a></dt>
<dd>
<div class="block">Returns the options as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html#getOptionsAsString--">getOptionsAsString()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html" title="class in com.compomics.cli.identification_parameters">AbstractIdentificationParametersCli</a></dt>
<dd>
<div class="block">Returns the options left to the user as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLI.html#getOptionsAsString--">getOptionsAsString()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLI.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersCLI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html#getOptionsAsString--">getOptionsAsString()</a></span> - Static method in enum com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html" title="enum in com.compomics.cli.identification_parameters">IdentificationParametersCLIParams</a></dt>
<dd>
<div class="block">Returns the options as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIParams.html#getOptionsAsString--">getOptionsAsString()</a></span> - Static method in enum com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIParams.html" title="enum in com.compomics.cli.modifications">ModificationsCLIParams</a></dt>
<dd>
<div class="block">Returns the options as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getOrderedDSites--">getOrderedDSites()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns an ordered list of sites where the D-score was used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getOrderedMzValues--">getOrderedMzValues()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns an array of the m/z values sorted in ascending order.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getOrderedProbabilisticSites--">getOrderedProbabilisticSites()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns an ordered list of sites where the probabilistic score was used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getOrderedPtmLocations--">getOrderedPtmLocations()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns sites of all localized PTMs ordered increasingly.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getOrientation--">getOrientation()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the orientation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/amino_acids/Substitution.html#getOriginalAminoAcid--">getOriginalAminoAcid()</a></span> - Method in class com.compomics.util.experiment.biology.variants.amino_acids.<a href="com/compomics/util/experiment/biology/variants/amino_acids/Substitution.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids">Substitution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#getOriginalAminoAcids-java.lang.Character-">getOriginalAminoAcids(Character)</a></span> - Method in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Returns the possible original amino acids for the given substituted amino
 acid as a list of their single letter code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#getOriginalAminoAcids--">getOriginalAminoAcids()</a></span> - Method in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Returns the amino acids where a substitution has been registered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getOutputFolderName--">getOutputFolderName()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the name of the output folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#getOutputFormat--">getOutputFormat()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns the output format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getOutputFormat--">getOutputFormat()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns the output format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/searchgui/OutputParameters.html#getOutputOptions--">getOutputOptions()</a></span> - Static method in enum com.compomics.util.parameters.searchgui.<a href="com/compomics/util/parameters/searchgui/OutputParameters.html" title="enum in com.compomics.util.parameters.searchgui">OutputParameters</a></dt>
<dd>
<div class="block">Returns an array of the indexes of the different output options.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/searchgui/OutputParameters.html#getOutputParameters-int-">getOutputParameters(int)</a></span> - Static method in enum com.compomics.util.parameters.searchgui.<a href="com/compomics/util/parameters/searchgui/OutputParameters.html" title="enum in com.compomics.util.parameters.searchgui">OutputParameters</a></dt>
<dd>
<div class="block">Returns the output option of the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/searchgui/OutputParameters.html#getOutputParametersNames--">getOutputParametersNames()</a></span> - Static method in enum com.compomics.util.parameters.searchgui.<a href="com/compomics/util/parameters/searchgui/OutputParameters.html" title="enum in com.compomics.util.parameters.searchgui">OutputParameters</a></dt>
<dd>
<div class="block">Returns an array of the names of the different options.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#getOutputResults--">getOutputResults()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns the output results filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getOutputSuffix--">getOutputSuffix()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the output tags suffix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/AaOccurrence.html#getP-char-">getP(char)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/AaOccurrence.html" title="class in com.compomics.util.experiment.identification.protein_sequences">AaOccurrence</a></dt>
<dd>
<div class="block">Returns the log10 of the share of the given amino acid in the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/AaOccurrence.html#getP-char:A-int-int-double-">getP(char[], int, int, double)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/AaOccurrence.html" title="class in com.compomics.util.experiment.identification.protein_sequences">AaOccurrence</a></dt>
<dd>
<div class="block">Returns the sum of the log10 of the share of the given amino acids in the
 database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/enumeration/CompomicsTools.html#getPackageName--">getPackageName()</a></span> - Method in enum com.compomics.util.enumeration.<a href="com/compomics/util/enumeration/CompomicsTools.html" title="enum in com.compomics.util.enumeration">CompomicsTools</a></dt>
<dd>
<div class="block">Return the package name of the tool.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomPanel.html#getPanelIndex--">getPanelIndex()</a></span> - Method in class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomPanel.html" title="class in com.compomics.util.gui.atoms">AtomPanel</a></dt>
<dd>
<div class="block">Returns the panel index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#getParameterKey--">getParameterKey()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getParameterKey--">getParameterKey()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html#getParameterKey--">getParameterKey()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">TagAssumption</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getParameterKey--">getParameterKey()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/personalization/UrParameter.html#getParameterKey--">getParameterKey()</a></span> - Method in interface com.compomics.util.experiment.personalization.<a href="com/compomics/util/experiment/personalization/UrParameter.html" title="interface in com.compomics.util.experiment.personalization">UrParameter</a></dt>
<dd>
<div class="block">This method returns the key of the parameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html#getParameterKey--">getParameterKey()</a></span> - Method in class com.compomics.util.experiment.refinement_parameters.<a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html" title="class in com.compomics.util.experiment.refinement_parameters">PepnovoAssumptionDetails</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/CommandLineParser.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/CommandLineParser.html" title="class in com.compomics.util.general">CommandLineParser</a></dt>
<dd>
<div class="block">This method will report on all parameters that have been found, or return
 an empty String[] if none were present.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">AndromedaParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">CometParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">DirecTagParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MsAmandaParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MsgfParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MyriMatchParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">NovorParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">OmssaParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">PepNovoParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">PNovoParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">TideParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html#getParameters--">getParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">XTandemParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/AlgorithmParametersDialog.html#getParameters--">getParameters()</a></span> - Method in interface com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/AlgorithmParametersDialog.html" title="interface in com.compomics.util.gui.parameters.identification">AlgorithmParametersDialog</a></dt>
<dd>
<div class="block">Returns the parameters as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html#getParametersList--">getParametersList()</a></span> - Method in class com.compomics.util.experiment.identification.identification_parameters.<a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html" title="class in com.compomics.util.experiment.identification.identification_parameters">IdentificationParametersFactory</a></dt>
<dd>
<div class="block">Returns an ordered list of the names of the implemented parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html#getParametersOptionsAsString--">getParametersOptionsAsString()</a></span> - Static method in enum com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html" title="enum in com.compomics.cli.identification_parameters">IdentificationParametersCLIParams</a></dt>
<dd>
<div class="block">Returns the options as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html#getParentFolder--">getParentFolder()</a></span> - Static method in class com.compomics.util.experiment.identification.identification_parameters.<a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html" title="class in com.compomics.util.experiment.identification.identification_parameters">IdentificationParametersFactory</a></dt>
<dd>
<div class="block">Returns the parent folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#getParentMonoisotopicMassIsotopeError--">getParentMonoisotopicMassIsotopeError()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns true if the parent ion mass tolerance is expanded by opening up
 multiple tolerance windows centered on the first and second 13C isotope
 peaks for a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#getPath-java.lang.String-">getPath(String)</a></span> - Static method in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Returns the path at the given line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#getPath--">getPath()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Returns the path to the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#getPathID-java.lang.String-">getPathID(String)</a></span> - Static method in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Returns the path id line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FileSystemAccessor.html#getPathOfClass-java.lang.String-">getPathOfClass(String)</a></span> - Static method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FileSystemAccessor.html" title="class in com.compomics.util.io.file">FileSystemAccessor</a></dt>
<dd>
<div class="block">Returns the absolute path of the underlying file system of the given
 class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#getPathParameter-com.compomics.software.settings.UtilitiesPathParameters.UtilitiesPathKey-">getPathParameter(UtilitiesPathParameters.UtilitiesPathKey)</a></span> - Static method in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Returns the path according to the given key and path.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/patient/Patient.html#getPatientInformation-com.compomics.util.experiment.patient.PatientInformation-">getPatientInformation(PatientInformation)</a></span> - Method in class com.compomics.util.experiment.patient.<a href="com/compomics/util/experiment/patient/Patient.html" title="class in com.compomics.util.experiment.patient">Patient</a></dt>
<dd>
<div class="block">Returns patient information,</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getPattern--">getPattern()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Returns the amino acid pattern targeted by this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/AminoAcidPatternDialog.html#getPattern--">getPattern()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/AminoAcidPatternDialog.html" title="class in com.compomics.util.gui">AminoAcidPatternDialog</a></dt>
<dd>
<div class="block">Returns the pattern as edited by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html#getPdbAccession--">getPdbAccession()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html" title="class in com.compomics.util.pdbfinder.das.readers">AlignmentBlock</a></dt>
<dd>
<div class="block">Returns the PDB accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html#getPdbAccession--">getPdbAccession()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAlignment</a></dt>
<dd>
<div class="block">Returns the PDB accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html#getPdbaccession--">getPdbaccession()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb">PdbParameter</a></dt>
<dd>
<div class="block">Returns the PDB accessions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html#getPdbEnd--">getPdbEnd()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html" title="class in com.compomics.util.pdbfinder.das.readers">AlignmentBlock</a></dt>
<dd>
<div class="block">Returns the PDB end.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html#getPdbGroup--">getPdbGroup()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAlignment</a></dt>
<dd>
<div class="block">Returns the PDB groups.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/FindPdbForUniprotAccessions.html#getPdbs--">getPdbs()</a></span> - Method in class com.compomics.util.pdbfinder.<a href="com/compomics/util/pdbfinder/FindPdbForUniprotAccessions.html" title="class in com.compomics.util.pdbfinder">FindPdbForUniprotAccessions</a></dt>
<dd>
<div class="block">Returns a vector of the PDB files mapped to the given protein accession
 number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html#getPdbStart--">getPdbStart()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html" title="class in com.compomics.util.pdbfinder.das.readers">AlignmentBlock</a></dt>
<dd>
<div class="block">Returns the PDB start.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getPeakAnnotation--">getPeakAnnotation()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the annotation to use for the ion match as a String.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getPeakAnnotation-com.compomics.util.experiment.biology.ions.Ion-int-">getPeakAnnotation(Ion, int)</a></span> - Static method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the annotation to use for a given ion and charge as a String.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getPeakAnnotation-boolean-com.compomics.util.experiment.biology.ions.Ion-int-">getPeakAnnotation(boolean, Ion, int)</a></span> - Static method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the annotation to use for a given ion and charge as a String.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getPeakAnnotation-boolean-">getPeakAnnotation(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Returns the annotation to use for the given ion match as a String.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getPeakList--">getPeakList()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the peak list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getPeakListAsString--">getPeakListAsString()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the peak list as an array list formatted as text, e.g.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getPeakMap--">getPeakMap()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns a peak map where peaks are indexed by their m/z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#getPeaks--">getPeaks()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Getter for the peaks</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#getPeaks--">getPeaks()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method reports on the peaks in the spectrum, with the
 Doubles for the masses as keys in the HashMap, and the intensities
 for each peak as Double value for that mass key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getPeaksAboveIntensityThreshold-double-">getPeaksAboveIntensityThreshold(double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns a DoubleStream of the intensity of all peaks strictly above the
 provided threshold.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getPeaksInBin-java.lang.Integer-">getPeaksInBin(Integer)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>
<div class="block">Returns the peaks at the given bin indexed by m/z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getPeaksMap--">getPeaksMap()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getPeakWaterMarkColor--">getPeakWaterMarkColor()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Get the peak water mark color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getPeakWidth--">getPeakWidth()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the current width of the peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#getPepNovoPtmMap--">getPepNovoPtmMap()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Returns the PepNovo to utilities PTM map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html#getPeptide--">getPeptide()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html" title="class in com.compomics.util.experiment.identification.matches">PeptideMatch</a></dt>
<dd>
<div class="block">Getter for the peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#getPeptide-double-double-">getPeptide(double, double)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Returns a peptide from the peptide draft.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#getPeptide-double-double-com.compomics.util.general.BoxedObject-">getPeptide(double, double, BoxedObject&lt;Boolean&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Returns a peptide from the peptide draft.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideWithPosition.html#getPeptide--">getPeptide()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideWithPosition.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideWithPosition</a></dt>
<dd>
<div class="block">Returns the peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html#getPeptide--">getPeptide()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">PeptideAssumption</a></dt>
<dd>
<div class="block">Get the theoretic peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getPeptideAssumptionFilter--">getPeptideAssumptionFilter()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the filter used when importing PSMs.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#getPeptideAssumptionsMap--">getPeptideAssumptionsMap()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Returns the peptide assumptions map: advocate id &gt; score &gt; list of
 assumptions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#getPeptideCount--">getPeptideCount()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Returns the number of peptides found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html#getPeptideDetailList-java.lang.String-">getPeptideDetailList(String)</a></span> - Method in class com.compomics.util.io.json.marshallers.<a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html" title="class in com.compomics.util.io.json.marshallers">PrideMarshaller</a></dt>
<dd>
<div class="block">Convert from JSON to a list of ProteinDetails.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getPeptideEnd-java.lang.String-int-">getPeptideEnd(String, int)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the 0 based end index of the peptide on the protein sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#getPeptideFilters--">getPeptideFilters()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Returns the list of peptide quality filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html#getPeptideFromProtein-char:A-int-double-double-">getPeptideFromProtein(char[], int, double, double)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">ProteinIteratorUtils</a></dt>
<dd>
<div class="block">Returns a peptide from the given sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html#getPeptideFromProtein-char:A-java.lang.String-int-java.lang.Double-java.lang.Double-">getPeptideFromProtein(char[], String, int, Double, Double)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">ProteinIteratorUtils</a></dt>
<dd>
<div class="block">Returns a peptide from the given sequence on the given protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html#getPeptideFromProtein-char:A-java.lang.String-int-java.lang.Double-java.lang.Double-com.compomics.util.general.BoxedObject-">getPeptideFromProtein(char[], String, int, Double, Double, BoxedObject&lt;Boolean&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">ProteinIteratorUtils</a></dt>
<dd>
<div class="block">Returns a peptide from the given sequence on the given protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#getPeptideHideFilters--">getPeptideHideFilters()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Returns the peptide hide filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getPeptideIdentification--">getPeptideIdentification()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns a list of the keys of all encountered peptides.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getPeptideLengthDistribution--">getPeptideLengthDistribution()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the distribution of validated peptide lengths.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#getPeptideList--">getPeptideList()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Returns the current peptide list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getPeptideMatch-long-">getPeptideMatch(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns the peptide match with the given key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getPeptideMatchesIterator-long:A-com.compomics.util.waiting.WaitingHandler-">getPeptideMatchesIterator(long[], WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns a peptide matches iterator for a given key list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getPeptideMatchesIterator-com.compomics.util.waiting.WaitingHandler-">getPeptideMatchesIterator(WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns a peptide matches iterator for all PeptideMatches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#getPeptideMatchesKeys--">getPeptideMatchesKeys()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Getter for the peptide keys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#getPeptideNormalisationFactor-java.lang.String-">getPeptideNormalisationFactor(String)</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Returns the peptide normalization factor for the given sample, 1.0 if not
 set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html#getPeptideProteinIndexesMap-java.util.ArrayList-">getPeptideProteinIndexesMap(ArrayList&lt;PeptideProteinMapping&gt;)</a></span> - Static method in class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a></dt>
<dd>
<div class="block">Returns a map made from the given mappings containing the indexes of the
 peptides in the protein sequences indexed by peptide sequence and protein
 accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html#getPeptideProteinMap-java.util.ArrayList-">getPeptideProteinMap(ArrayList&lt;PeptideProteinMapping&gt;)</a></span> - Static method in class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a></dt>
<dd>
<div class="block">Returns a map made from the given mappings containing protein accessions
 for every peptide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html#getPeptides-java.util.ArrayList-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getPeptides(ArrayList&lt;PeptideProteinMapping&gt;, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a></dt>
<dd>
<div class="block">Aggregates the given mapping into a list of peptides.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getPeptideSelected--">getPeptideSelected()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the color for a selected peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html#getPeptideSequence--">getPeptideSequence()</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a></dt>
<dd>
<div class="block">Returns the peptide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IsotopicDistributionPanel.html#getPeptideSequences--">getPeptideSequences()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IsotopicDistributionPanel.html" title="class in com.compomics.util.gui.spectrum">IsotopicDistributionPanel</a></dt>
<dd>
<div class="block">Get the set of peptide sequences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getPeptideShakerPath--">getPeptideShakerPath()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the path to the PeptideShaker installation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#getPeptideStarFilters--">getPeptideStarFilters()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Returns the peptide star filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html#getPeptideVariantMatches--">getPeptideVariantMatches()</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a></dt>
<dd>
<div class="block">Returns eventual variants found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getPeptideVariantsParameters--">getPeptideVariantsParameters()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the peptide variant parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/PeptideVariantsParametersDialog.html#getPeptideVariantsPreferences--">getPeptideVariantsPreferences()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/PeptideVariantsParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">PeptideVariantsParametersDialog</a></dt>
<dd>
<div class="block">Returns the peptide variants settings as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getPepXmlOutput--">getPepXmlOutput()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true if the pepxml output is to be created.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#getPercentLength--">getPercentLength()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Returns the length of the reference area in percent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/MonitorDimension.html#getPercentualScreenDimension-javax.swing.JFrame-double-">getPercentualScreenDimension(JFrame, double)</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/MonitorDimension.html" title="class in com.compomics.util.gui">MonitorDimension</a></dt>
<dd>
<div class="block">This method returns a Dimension with a percentual width and height in relation to the JFrame.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/MonitorDimension.html#getPercentualScreenDimension-javax.swing.JPanel-double-">getPercentualScreenDimension(JPanel, double)</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/MonitorDimension.html" title="class in com.compomics.util.gui">MonitorDimension</a></dt>
<dd>
<div class="block">This method returns a Dimension with a percentual width and height in realtion to the JPanel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistribution.html#getPercMax--">getPercMax()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistribution.html" title="class in com.compomics.util.general">IsotopicDistribution</a></dt>
<dd>
<div class="block">Getter for result of the isotopic distributions calculation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistribution.html#getPercTot--">getPercTot()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistribution.html" title="class in com.compomics.util.general">IsotopicDistribution</a></dt>
<dd>
<div class="block">Getter for result of the isotopic distributions calculation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getPhase--">getPhase()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the phase.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the pI.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Alanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Arginine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Asparagine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">AsparticAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">B</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Cysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">GlutamicAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glutamine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glycine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Histidine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Isoleucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">J</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Leucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Lysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Methionine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Phenylalanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Proline</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Pyrrolysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Selenocysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Serine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Threonine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tryptophan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tyrosine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Valine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">X</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html#getPI--">getPI()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Z</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getPinOutput--">getPinOutput()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true if the Percolator output is to be created.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the pK1.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Alanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Arginine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Asparagine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">AsparticAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">B</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Cysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">GlutamicAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glutamine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glycine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Histidine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Isoleucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">J</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Leucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Lysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Methionine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Phenylalanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Proline</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Pyrrolysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Selenocysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Serine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Threonine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tryptophan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tyrosine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Valine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">X</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html#getPK1--">getPK1()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Z</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the pK2.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Alanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Arginine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Asparagine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">AsparticAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">B</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Cysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">GlutamicAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glutamine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glycine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Histidine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Isoleucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">J</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Leucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Lysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Methionine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Phenylalanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Proline</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Pyrrolysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Selenocysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Serine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Threonine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tryptophan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tyrosine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Valine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">X</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html#getPK2--">getPK2()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Z</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the pKa.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Alanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Arginine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Asparagine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">AsparticAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">B</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Cysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">GlutamicAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glutamine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glycine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Histidine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Isoleucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">J</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Leucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Lysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Methionine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Phenylalanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Proline</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Pyrrolysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Selenocysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Serine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Threonine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tryptophan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tyrosine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Valine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">X</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html#getPKa--">getPKa()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Z</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#getPlus1--">getPlus1()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Returns the isotopic correction factor at +1 Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#getPlus2--">getPlus2()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Returns the isotopic correction factor at +2 Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#getPmid--">getPmid()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Returns the PubMed id of the reference of the advocate of interest.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html#getPmid--">getPmid()</a></span> - Method in enum com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html" title="enum in com.compomics.util.experiment.io.biology.protein">ProteinDatabase</a></dt>
<dd>
<div class="block">Returns the PubMed id of the database, null if not set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Reference.html#getPmid--">getPmid()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Reference.html" title="class in com.compomics.util.pride.prideobjects">Reference</a></dt>
<dd>
<div class="block">Returns the PubMed ID of the reference.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html#getPNovoParameters--">getPNovoParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">PNovoParametersDialog</a></dt>
<dd>
<div class="block">Returns the pNovo parameters as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#getPNovoPtmMap--">getPNovoPtmMap()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Returns the pNovo to utilities PTM map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#getPNovoPtmResiduesMap--">getPNovoPtmResiduesMap()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Returns the pNovo to original amino acids residue map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideWithPosition.html#getPosition--">getPosition()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideWithPosition.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideWithPosition</a></dt>
<dd>
<div class="block">Returns the position on the protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#getPosition--">getPosition()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">Simple getter for the cleavage position of the Enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItem.html#getPossibilities--">getPossibilities()</a></span> - Method in interface com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItem.html" title="interface in com.compomics.util.experiment.filtering">FilterItem</a></dt>
<dd>
<div class="block">In case the filter has predefined values, return the possible values, null otherwise.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html#getPossibilities--">getPossibilities()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">AssumptionFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html#getPossibilities--">getPossibilities()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PeptideFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html#getPossibilities--">getPossibilities()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">ProteinFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html#getPossibilities--">getPossibilities()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PsmFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#getPossibleCharges--">getPossibleCharges()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Getter for the possible charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#getPossibleChargesAsString--">getPossibleChargesAsString()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Returns the possible charges as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getPossibleConfidenceLevels--">getPossibleConfidenceLevels()</a></span> - Static method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Convenience method returning all confidence levels as string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html#getPossibleFilterItems--">getPossibleFilterItems()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">AssumptionFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getPossibleFilterItems--">getPossibleFilterItems()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Returns the filter items accepted by this filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html#getPossibleFilterItems--">getPossibleFilterItems()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html#getPossibleFilterItems--">getPossibleFilterItems()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html" title="class in com.compomics.util.experiment.identification.filtering">ProteinFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html#getPossibleFilterItems--">getPossibleFilterItems()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PsmFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getPossibleFilterItemsNames--">getPossibleFilterItemsNames()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Returns the filter items accepted by this filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#getPossibleModificationSites-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.biology.modifications.Modification-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getPossibleModificationSites(Peptide, Modification, SequenceProvider, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">Returns an array of the possible modification sites for the given
 modification on the given peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#getPossibleModificationSites-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidSequence-boolean-boolean-com.compomics.util.experiment.biology.modifications.Modification-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getPossibleModificationSites(AminoAcidSequence, boolean, boolean, Modification, SequenceMatchingParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">Returns an array of the possible modification sites for the given
 modification on the given peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#getPossibleSubtypes--">getPossibleSubtypes()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>
<div class="block">Returns the possible subtypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#getPossibleSubtypes--">getPossibleSubtypes()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>
<div class="block">Returns the possible subtypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#getPossibleSubtypes--">getPossibleSubtypes()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Returns the possible subtypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#getPossibleSubtypes--">getPossibleSubtypes()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Returns the possible subtypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#getPossibleSubtypes--">getPossibleSubtypes()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>
<div class="block">Returns the possible subtypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#getPossibleSubtypes--">getPossibleSubtypes()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Returns the possible subtypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#getPossibleSubtypes--">getPossibleSubtypes()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Returns the possible subtypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getPossibleSubtypes--">getPossibleSubtypes()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Returns the possible subtypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getPossibleSubtypes-com.compomics.util.experiment.biology.ions.Ion.IonType-">getPossibleSubtypes(Ion.IonType)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns an array of possible subtypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getPossibleSubtypesAsSet-com.compomics.util.experiment.biology.ions.Ion.IonType-">getPossibleSubtypesAsSet(Ion.IonType)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns a hashset of possible subtypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html#getPossibleTags-boolean-int-int-int-">getPossibleTags(boolean, int, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">TagAssumption</a></dt>
<dd>
<div class="block">Computes the possible tag assumptions which can be obtained from this one
 by accounting for other charges and isotopes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItem.html#getPossibleValues--">getPossibleValues()</a></span> - Method in interface com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItem.html" title="interface in com.compomics.util.experiment.filtering">FilterItem</a></dt>
<dd>
<div class="block">Returns the possible items of this kind.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html#getPossibleValues--">getPossibleValues()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">AssumptionFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html#getPossibleValues--">getPossibleValues()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PeptideFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html#getPossibleValues--">getPossibleValues()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">ProteinFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html#getPossibleValues--">getPossibleValues()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PsmFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getPotentialModificationSites-com.compomics.util.experiment.biology.modifications.Modification-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getPotentialModificationSites(Modification, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the potential modification sites as an ordered list of string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getPotentialModificationSitesNoCombination-com.compomics.util.experiment.biology.modifications.Modification-java.lang.String-int-">getPotentialModificationSitesNoCombination(Modification, String, int)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the potential modification sites as an ordered list of sites.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getPpm--">getPpm()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#getPpmError-double-double-">getPpmError(double, double)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Returns the ppm value of the given mass error relative to its theoretical
 m/z value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getPpmTolerance-double-double-">getPpmTolerance(double, double)</a></span> - Static method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the relative tolerance in ppm corresponding to the absolute
 tolerance in Dalton at the given reference mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#getPrecurorCharge--">getPrecurorCharge()</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#getPrecursor-uk.ac.ebi.pride.tools.braf.BufferedRandomAccessFile-java.lang.Long-java.lang.String-">getPrecursor(BufferedRandomAccessFile, Long, String)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Returns the next precursor starting from the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html#getPrecursor-uk.ac.ebi.pride.tools.braf.BufferedRandomAccessFile-java.lang.Long-java.lang.String-">getPrecursor(BufferedRandomAccessFile, Long, String)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspReader</a></dt>
<dd>
<div class="block">Returns the next precursor starting from the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getPrecursor--">getPrecursor()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the precursor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getPrecursor-java.lang.String-java.lang.String-">getPrecursor(String, String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the precursor of the desired spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getPrecursor-java.lang.String-java.lang.String-boolean-">getPrecursor(String, String, boolean)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the precursor of the desired spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getPrecursor-java.lang.String-">getPrecursor(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the precursor of the desired spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getPrecursor-java.lang.String-boolean-">getPrecursor(String, boolean)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the precursor of the desired spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getPrecursorAccuracy--">getPrecursorAccuracy()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the precursor tolerance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getPrecursorAccuracyType--">getPrecursorAccuracyType()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the precursor accuracy type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getPrecursorAdjustmentStep--">getPrecursorAdjustmentStep()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the precursor adjustment step.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#getPrecursorCharge--">getPrecursorCharge()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Returns the charge of the precursor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#getPrecursorIntensity--">getPrecursorIntensity()</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getPrecursorIntensityAveragePerFraction-java.lang.String-">getPrecursorIntensityAveragePerFraction(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Get the average precursor intensity in the given fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getPrecursorIntensityAveragePerFraction--">getPrecursorIntensityAveragePerFraction()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the fraction precursor intensity average map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getPrecursorIntensityPerFraction-java.lang.String-">getPrecursorIntensityPerFraction(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Get the precursor intensity in the given fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getPrecursorIntensityPerFraction--">getPrecursorIntensityPerFraction()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the precursor intensity per fraction map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getPrecursorIntensitySummedPerFraction-java.lang.String-">getPrecursorIntensitySummedPerFraction(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Get the summed precursor intensity in the given fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getPrecursorIntensitySummedPerFraction--">getPrecursorIntensitySummedPerFraction()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the fraction summed intensity map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getPrecursorMap-java.lang.String-">getPrecursorMap(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns a map containing all the precursors of a gven file indexed by
 spectrum title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#getPrecursorMatch-com.compomics.util.experiment.mass_spectrometry.spectra.Peak-">getPrecursorMatch(Peak)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Returns the ion match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getPrecursorMz-int-">getPrecursorMz(int)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns the precursor mz for the spectrum at the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getPrecursorMz-java.lang.String-">getPrecursorMz(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the precursor mz of the desired spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#getPrecursorMZ--">getPrecursorMZ()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Getter for the precursor MZ</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#getPrecursorMZ--">getPrecursorMZ()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method reports on the precursor M/Z</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#getPrecursorMz--">getPrecursorMz()</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html#getPrecursorsInBin-int-">getPrecursorsInBin(int)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">PrecursorMap</a></dt>
<dd>
<div class="block">Returns the precursors at the given bin indexed by mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getPrecursorToleance--">getPrecursorToleance()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/genes/GeneDetailsDialog.html#getPreferredAccessionColumnWidth-javax.swing.JTable-int-int-">getPreferredAccessionColumnWidth(JTable, int, int)</a></span> - Method in class com.compomics.util.gui.genes.<a href="com/compomics/util/gui/genes/GeneDetailsDialog.html" title="class in com.compomics.util.gui.genes">GeneDetailsDialog</a></dt>
<dd>
<div class="block">Gets the preferred width of the column specified by colIndex.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/examples/UtilitiesDemo.html#getPreferredColumnWidth-javax.swing.JTable-int-int-">getPreferredColumnWidth(JTable, int, int)</a></span> - Method in class com.compomics.util.examples.<a href="com/compomics/util/examples/UtilitiesDemo.html" title="class in com.compomics.util.examples">UtilitiesDemo</a></dt>
<dd>
<div class="block">Gets the preferred width of the column specified by vColIndex.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/GuiUtilities.html#getPreferredWidthOfCell-javax.swing.JTable-int-int-">getPreferredWidthOfCell(JTable, int, int)</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/GuiUtilities.html" title="class in com.compomics.util.gui">GuiUtilities</a></dt>
<dd>
<div class="block">Returns the preferred width of a given cell in a table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#getPrideCvTerm--">getPrideCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#getPrideCvTerm--">getPrideCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#getPrideCvTerm--">getPrideCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#getPrideCvTerm--">getPrideCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#getPrideCvTerm--">getPrideCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#getPrideCvTerm--">getPrideCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#getPrideCvTerm--">getPrideCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getPrideCvTerm--">getPrideCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getPrideCvTerm--">getPrideCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the CV term adapted to the fragment ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#getPrideFolder--">getPrideFolder()</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Returns the folder where pride annotation information should be saved.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getPrimaryKeyColumns--">getPrimaryKeyColumns()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method reports on the primary key columns for the table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#getPrimaryProgressCounter--">getPrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#getPrimaryProgressCounter--">getPrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#getPrimaryProgressCounter--">getPrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#getPrimaryProgressCounter--">getPrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#getPrimaryProgressCounter--">getPrimaryProgressCounter()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Returns the primary progress counter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getPrintExpectScore--">getPrintExpectScore()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns true if the export score is to be printed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getPrintPeptides--">getPrintPeptides()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true of a list of the peptides should be printed to the output
 folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getPrintProgressIndicatorSize--">getPrintProgressIndicatorSize()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the progress indicator size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getProbabilisticScore-int-">getProbabilisticScore(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns the probabilistic score at a given site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#getProbabilisticScoreThreshold--">getProbabilisticScoreThreshold()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Returns the probabilistic score threshold.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getProbabilisticSites--">getProbabilisticSites()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns a set of sites where a probabilistic score is available</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getProbability--">getProbability()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the match probability.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/Distribution.html#getProbabilityAt-double-">getProbabilityAt(double)</a></span> - Method in interface com.compomics.util.math.statistics.<a href="com/compomics/util/math/statistics/Distribution.html" title="interface in com.compomics.util.math.statistics">Distribution</a></dt>
<dd>
<div class="block">Returns the density function value at a given position.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html#getProbabilityAt-double-">getProbabilityAt(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html" title="class in com.compomics.util.math.statistics.distributions">BinomialDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getProbabilityAt-double-">getProbabilityAt(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html#getProbabilityAt-double-">getProbabilityAt(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getProcessedProteinKeys-com.compomics.util.waiting.WaitingHandler-com.compomics.util.gui.filtering.FilterParameters-">getProcessedProteinKeys(WaitingHandler, FilterParameters)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the sorted list of protein keys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/tools/ProcessingParametersDialog.html#getProcessingParameters--">getProcessingParameters()</a></span> - Method in class com.compomics.util.gui.parameters.tools.<a href="com/compomics/util/gui/parameters/tools/ProcessingParametersDialog.html" title="class in com.compomics.util.gui.parameters.tools">ProcessingParametersDialog</a></dt>
<dd>
<div class="block">Returns the processing parameters as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/tools/ProcessingParameters.html#getProcessingType--">getProcessingType()</a></span> - Method in class com.compomics.util.parameters.tools.<a href="com/compomics/util/parameters/tools/ProcessingParameters.html" title="class in com.compomics.util.parameters.tools">ProcessingParameters</a></dt>
<dd>
<div class="block">Returns the processing type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ModificationProfile.html#getProfile--">getProfile()</a></span> - Method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ModificationProfile.html" title="class in com.compomics.util.gui.protein">ModificationProfile</a></dt>
<dd>
<div class="block">Returns the profile.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html#getProgress--">getProgress()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialog</a></dt>
<dd>
<div class="block">This method reports on the current progress of the progressdialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#getProgressBar--">getProgressBar()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Returns the progress bar for updates from external processes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getProjectCount-java.lang.String-com.compomics.util.pride.prideobjects.webservice.query.PrideFilter...-">getProjectCount(String, PrideFilter...)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a project count for the given criteria and filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html#getProjectDetail-java.lang.String-">getProjectDetail(String)</a></span> - Method in class com.compomics.util.io.json.marshallers.<a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html" title="class in com.compomics.util.io.json.marshallers">PrideMarshaller</a></dt>
<dd>
<div class="block">Convert from JSON to a list of ProjectDetails.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getProjectDetail-java.lang.String-">getProjectDetail(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns the project details for a given accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getProjectFileCount-java.lang.String-">getProjectFileCount(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a count of file details for a given project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getProjectFileDetails-java.lang.String-">getProjectFileDetails(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a list of file details for a given project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html#getProjectSummaryList-java.lang.String-">getProjectSummaryList(String)</a></span> - Method in class com.compomics.util.io.json.marshallers.<a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html" title="class in com.compomics.util.io.json.marshallers">PrideMarshaller</a></dt>
<dd>
<div class="block">Convert from JSON to a list of ProjectDetails.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getProjectSummaryList-java.lang.String-com.compomics.util.pride.prideobjects.webservice.query.PrideFilter...-">getProjectSummaryList(String, PrideFilter...)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a ProjectSummaryList for the given query and filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#getProjectUniqueName--">getProjectUniqueName()</a></span> - Method in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#getProjectUniqueName-java.lang.String-">getProjectUniqueName(String)</a></span> - Method in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PropertiesManager.html#getProperties-com.compomics.util.enumeration.CompomicsTools-java.lang.String-">getProperties(CompomicsTools, String)</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PropertiesManager.html" title="class in com.compomics.util.io">PropertiesManager</a></dt>
<dd>
<div class="block">Get a Properties instance for the parameter properties filename.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/junit/TestCaseLM.html#getPropertiesFile-java.lang.String-">getPropertiesFile(String)</a></span> - Static method in class com.compomics.util.junit.<a href="com/compomics/util/junit/TestCaseLM.html" title="class in com.compomics.util.junit">TestCaseLM</a></dt>
<dd>
<div class="block">This method will read a Properties file from the current ClassLoader
 and return a Properties instance with the Properties from the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getProperty-com.compomics.util.experiment.biology.aminoacids.AminoAcid.Property-">getProperty(AminoAcid.Property)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns a property of the amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.Property.html#getProperty-int-">getProperty(int)</a></span> - Method in enum com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.Property.html" title="enum in com.compomics.util.experiment.biology.aminoacids">AminoAcid.Property</a></dt>
<dd>
<div class="block">Returns the property at index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html#getProperty--">getProperty()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ModificationFeature</a></dt>
<dd>
<div class="block">Returns the feature property.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html#getProperty--">getProperty()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideFeature</a></dt>
<dd>
<div class="block">Returns the feature property;</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getPrositeFormat--">getPrositeFormat()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the pattern in the PROSITE format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html#getProteinAccession--">getProteinAccession()</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a></dt>
<dd>
<div class="block">Returns the accession of the protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html#getProteinAccessions-java.lang.String-">getProteinAccessions(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html" title="class in com.compomics.util.experiment.biology.genes.go">GoMapping</a></dt>
<dd>
<div class="block">Returns the protein accessions linked to a given GO term.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html#getProteinConfidenceMwPlots--">getProteinConfidenceMwPlots()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html" title="class in com.compomics.util.parameters.identification.advanced">FractionParameters</a></dt>
<dd>
<div class="block">Returns the protein confidence for inclusion in MW plots.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinDatabase-java.lang.String-">getProteinDatabase(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html#getProteinDatabase-java.lang.String-">getProteinDatabase(String)</a></span> - Method in enum com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html" title="enum in com.compomics.util.experiment.io.biology.protein">ProteinDatabase</a></dt>
<dd>
<div class="block">Returns the protein database with the given full name, null if not found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getProteinDatabase-java.lang.String-">getProteinDatabase(String)</a></span> - Method in interface com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></dt>
<dd>
<div class="block">Returns the the protein database for the given protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html#getProteinDetailList-java.lang.String-">getProteinDetailList(String)</a></span> - Method in class com.compomics.util.io.json.marshallers.<a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html" title="class in com.compomics.util.io.json.marshallers">PrideMarshaller</a></dt>
<dd>
<div class="block">Convert from JSON to a list of ProteinDetails.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getProteinEvidencAsString-java.lang.Integer-">getProteinEvidencAsString(Integer)</a></span> - Static method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Return the UniProt protein evidence type as text.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinEvidence-java.lang.String-">getProteinEvidence(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getProteinEvidence--">getProteinEvidence()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the protein evidence level as indexed in UniProt.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getProteinEvidence-java.lang.String-">getProteinEvidence(String)</a></span> - Method in interface com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></dt>
<dd>
<div class="block">Returns an integer representing the protein evidence level as indexed by uniprot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#getProteinFilters--">getProteinFilters()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Returns the list of protein quality filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#getProteinHideFilters--">getProteinHideFilters()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Returns the protein hide filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getProteinIdentification--">getProteinIdentification()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns a list of the keys of all encountered proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getProteinIdentificationByAssay-java.lang.String-">getProteinIdentificationByAssay(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a list of protein details for a given assay.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getProteinIdentificationByProject-java.lang.String-">getProteinIdentificationByProject(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a list of protein details for a given project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getProteinIdentificationCountByAssay-java.lang.String-">getProteinIdentificationCountByAssay(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a count of protein details for a given assay.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getProteinIdentificationCountByProject-java.lang.String-">getProteinIdentificationCountByProject(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a count of protein details for a given project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getProteinIdentificationsByProjectAndProtein-java.lang.String-java.lang.String-">getProteinIdentificationsByProjectAndProtein(String, String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a list of protein details for a given project and a given protein
 accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getProteinIdentificationsCountByProjectAndProtein-java.lang.String-java.lang.String-">getProteinIdentificationsCountByProjectAndProtein(String, String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a count of protein details for a given project and a given
 protein accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getProteinInferenceClassAsString--">getProteinInferenceClassAsString()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the protein inference class as a string for the given
 integer-based class</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getProteinInferenceClassAsString-int-">getProteinInferenceClassAsString(int)</a></span> - Static method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the protein inference class as a string for the given
 integer-based class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getProteinInferenceGroupClass--">getProteinInferenceGroupClass()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the protein inference class of the protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getProteinInferenceParameters--">getProteinInferenceParameters()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the protein inference parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ProteinInferenceParametersDialog.html#getProteinInferencePreferences--">getProteinInferencePreferences()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ProteinInferenceParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ProteinInferenceParametersDialog</a></dt>
<dd>
<div class="block">Returns the protein inference preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Protein.html#getProteinKey--">getProteinKey()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Protein.html" title="class in com.compomics.util.experiment.biology.proteins">Protein</a></dt>
<dd>
<div class="block">Returns the key for protein indexing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getProteinKeys-com.compomics.util.waiting.WaitingHandler-com.compomics.util.gui.filtering.FilterParameters-">getProteinKeys(WaitingHandler, FilterParameters)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the ordered protein keys to display when no filtering is applied.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getProteinKeys--">getProteinKeys()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the list of ordered protein keys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#getProteinList--">getProteinList()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Returns the protein list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#getProteinListAfterHiding--">getProteinListAfterHiding()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Returns the protein list after all hiding filters have been used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getProteinMap--">getProteinMap()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns a map of all the protein matches which can be ascribed to a
 protein indexed by its accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getProteinMapping--">getProteinMapping()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the proteins mapping as a map of 0 based indexes for every
 protein accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/FastaMapper.html#getProteinMapping-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getProteinMapping(String, SequenceMatchingParameters)</a></span> - Method in interface com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/FastaMapper.html" title="interface in com.compomics.util.experiment.identification.protein_inference">FastaMapper</a></dt>
<dd>
<div class="block">Returns the protein mapping in the FASTA file loaded in the sequence
 factory for the given peptide sequence in a map: peptide sequence found
 in the FASTA file | protein accession | list of indexes of the peptide
 sequence on the protein sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/FastaMapper.html#getProteinMapping-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getProteinMapping(Tag, SequenceMatchingParameters)</a></span> - Method in interface com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/FastaMapper.html" title="interface in com.compomics.util.experiment.identification.protein_inference">FastaMapper</a></dt>
<dd>
<div class="block">Returns the protein mappings for the given peptide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMapping-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getProteinMapping(String, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMapping-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getProteinMapping(Tag, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMappingWithoutVariants-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">getProteinMappingWithoutVariants(String, SequenceMatchingParameters, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Exact mapping peptides against the proteome.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMappingWithoutVariants-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">getProteinMappingWithoutVariants(Tag, SequenceMatchingParameters, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Mapping tags against proteome without variants.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMappingWithVariants-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">getProteinMappingWithVariants(Tag, SequenceMatchingParameters, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Mapping tags against proteome with variants.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMappingWithVariantsGeneric-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">getProteinMappingWithVariantsGeneric(String, SequenceMatchingParameters, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Variant tolerant mapping peptides against the proteome.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getProteinMappingWithVariantsSpecific-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">getProteinMappingWithVariantsSpecific(String, SequenceMatchingParameters, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Variant tolerant mapping peptides against the proteome.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getProteinMatch-long-">getProteinMatch(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns the protein match with the given key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getProteinMatches-com.compomics.util.experiment.biology.proteins.Peptide-">getProteinMatches(Peptide)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns the keys of the protein matches where a peptide can be found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getProteinMatchesIterator-long:A-com.compomics.util.waiting.WaitingHandler-">getProteinMatchesIterator(long[], WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns a protein matches iterator for a given key list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getProteinMatchesIterator-com.compomics.util.waiting.WaitingHandler-">getProteinMatchesIterator(WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns a protein matches iterator for all PeptideMatches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#getProteinMatchKey-com.compomics.util.experiment.biology.proteins.Peptide-">getProteinMatchKey(Peptide)</a></span> - Static method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Convenience method which returns the protein key from a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#getProteinNormalisationFactor-java.lang.String-">getProteinNormalisationFactor(String)</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Returns the protein normalization factor for the given sample, 1.0 if not
 set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#getProteinPtmComplexity--">getProteinPtmComplexity()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns the proteinPtmComplexity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getProteinsForGoTerm-java.lang.String-">getProteinsForGoTerm(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the protein accessions for a GO accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#getProteinStarFilters--">getProteinStarFilters()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Returns the protein star filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#getProteinToGoMap--">getProteinToGoMap()</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Returns the protein to GO terms accession map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html#getProteinToGoMap--">getProteinToGoMap()</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html" title="class in com.compomics.util.experiment.biology.genes.go">GoMapping</a></dt>
<dd>
<div class="block">Returns the protein to GO accession map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getProteoWizardPath--">getProteoWizardPath()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the path to ProteoWizard.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#getProtocol--">getProtocol()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Returns the protocol ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#getProtocols--">getProtocols()</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Returns the protocols.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#getProtonMassMultiple-int-">getProtonMassMultiple(int)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>
<div class="block">Returns the mass of the proton multiplied by i.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#getPsiMsCvTerm--">getPsiMsCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#getPsiMsCvTerm--">getPsiMsCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#getPsiMsCvTerm--">getPsiMsCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#getPsiMsCvTerm--">getPsiMsCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#getPsiMsCvTerm--">getPsiMsCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#getPsiMsCvTerm--">getPsiMsCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#getPsiMsCvTerm--">getPsiMsCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getPsiMsCvTerm--">getPsiMsCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getPsiMsCvTerm--">getPsiMsCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the CV term adapted to the fragment ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#getPsiMsCvTerm--">getPsiMsCvTerm()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Returns the CV term for the neutral loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getPSMCountByAssay-java.lang.String-">getPSMCountByAssay(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a count of PSM details for a given assay.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getPSMCountByAssayAndSequence-java.lang.String-java.lang.String-">getPSMCountByAssayAndSequence(String, String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a count of PSM details for a given assay and a given peptide
 sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getPSMCountByProject-java.lang.String-">getPSMCountByProject(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a count of PSM details for a given project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getPSMCountByProjectAndSequence-java.lang.String-java.lang.String-">getPSMCountByProjectAndSequence(String, String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a count of PSM details for a given project and a given peptide
 sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#getPsmFilters--">getPsmFilters()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Returns the list of PSM quality filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#getPsmHideFilters--">getPsmHideFilters()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Returns the psm hide filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#getPsmList--">getPsmList()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Returns the PSM list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#getPsmNormalisationFactor-java.lang.String-">getPsmNormalisationFactor(String)</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Returns the PSM normalization factor for the given sample, 1.0 if not
 set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getPSMsByAssay-java.lang.String-">getPSMsByAssay(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a list of PSM details for a given assay.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getPSMsByAssayAndSequence-java.lang.String-java.lang.String-">getPSMsByAssayAndSequence(String, String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a list of PSM details for a given assay and a given peptide
 sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getPSMsByProject-java.lang.String-">getPSMsByProject(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a list of PSM details for a given project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#getPSMsByProjectAndSequence-java.lang.String-java.lang.String-">getPSMsByProjectAndSequence(String, String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>
<div class="block">Returns a list of PSM details for a given project and a given peptide
 sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getPsmScoringParameters--">getPsmScoringParameters()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the scoring parameters used when scoring PSMs.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/PsmScoringParametersDialog.html#getPsmScoringPreferences--">getPsmScoringPreferences()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/PsmScoringParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">PsmScoringParametersDialog</a></dt>
<dd>
<div class="block">Returns the PSM scoring preferences as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#getPsmStarFilters--">getPsmStarFilters()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Returns the psm star filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#getPTM-com.compomics.util.parameters.identification.tool_specific.PepnovoParameters-java.lang.String-">getPTM(PepnovoParameters, String)</a></span> - Static method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>
<div class="block">Get a PTM.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getPtmIndex-java.lang.String-">getPtmIndex(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the index of a given modification, null if not found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getPtmIndex-java.lang.String-">getPtmIndex(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the index of a given modification</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getPtmIndexes--">getPtmIndexes()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the PTM indexes as a map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getPtmIndexes--">getPtmIndexes()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the PTM indexes as a map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getPtmLocationAtAA--">getPtmLocationAtAA()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns the map of the localization.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getPtmLocations-int-">getPtmLocations(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns the PTM locations at a given confidence level (see static
 fields).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ModificationProfile.html#getPtmName--">getPtmName()</a></span> - Method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ModificationProfile.html" title="class in com.compomics.util.gui.protein">ModificationProfile</a></dt>
<dd>
<div class="block">Returns the PTM name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#getPtmResidue-java.lang.Character-">getPtmResidue(Character)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Returns the original amino acids residue corresponding to the given pNovo
 PTM character.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ModificationLocalizationParametersDialog.html#getPtmScoringPreferences--">getPtmScoringPreferences()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ModificationLocalizationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ModificationLocalizationParametersDialog</a></dt>
<dd>
<div class="block">Returns the PTM scoring preferences as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#getPtmToPrideMap--">getPtmToPrideMap()</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Returns the utilities PTM to pride map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getQcCriteria--">getQcCriteria()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the list of qc checks made for this match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getQcFilters--">getQcFilters()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the qc filters map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html#getQuantile-int-java.lang.Integer-int-double-">getQuantile(int, Integer, int, double)</a></span> - Method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html" title="class in com.compomics.util.experiment.identification.modification">ModificationtableContent</a></dt>
<dd>
<div class="block">Get the quantile.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/webservice/PrideQuery.html#getQueryTemplate-boolean-">getQueryTemplate(boolean)</a></span> - Method in enum com.compomics.util.pride.prideobjects.webservice.<a href="com/compomics/util/pride/prideobjects/webservice/PrideQuery.html" title="enum in com.compomics.util.pride.prideobjects.webservice">PrideQuery</a></dt>
<dd>
<div class="block">Returns the actual query template.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandLineUtils.html#getQuoteType--">getQuoteType()</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli">CommandLineUtils</a></dt>
<dd>
<div class="block">Returns the quote type to use.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#getRandomIndexes-int-int-int-">getRandomIndexes(int, int, int)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Returns a list of n random indexes between min and max included.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#getRandomInteger-int-int-">getRandomInteger(int, int)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Returns an integer randomly chosen between min and max included.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html#getRank-int-boolean-">getRank(int, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">Rank</a></dt>
<dd>
<div class="block">Returns the rank.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#getRank-int-int-">getRank(int, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Returns the number of occurrences of a given character until position
 index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#getRank--">getRank()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Get the identification rank.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html#getRankOne-int-">getRankOne(int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">Rank</a></dt>
<dd>
<div class="block">Returns the rank of ones.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#getRankRecursive-int-int-">getRankRecursive(int, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Returns the number of occurrences of a given character until position
 index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html#getRankScore--">getRankScore()</a></span> - Method in class com.compomics.util.experiment.refinement_parameters.<a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html" title="class in com.compomics.util.experiment.refinement_parameters">PepnovoAssumptionDetails</a></dt>
<dd>
<div class="block">Returns the PepNovo rank score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html#getRankZero-int-">getRankZero(int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">Rank</a></dt>
<dd>
<div class="block">Returns the rank of zeros.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getRawBins--">getRawBins()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>
<div class="block">Returns the bins in the map as collection of keys from the map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getRawHeader--">getRawHeader()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the entire header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#getRawScore--">getRawScore()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Returns the raw score as provided by the identification algorithm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getReadTweets--">getReadTweets()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the list of read tweets.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html#getReagent-java.lang.String-">getReagent(String)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethod</a></dt>
<dd>
<div class="block">Returns the reagent of the given name, null if not found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html#getReagentNames--">getReagentNames()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethod</a></dt>
<dd>
<div class="block">Returns the reagents available in this method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html#getReagentsSortedByMass--">getReagentsSortedByMass()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethod</a></dt>
<dd>
<div class="block">Returns the list of reagents sorted by ascending mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getRecalibratedPeakList-java.util.HashMap-">getRecalibratedPeakList(HashMap&lt;Double, Double&gt;)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns a recalibrated peak list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#getRecalibratedPrecursor-double-double-">getRecalibratedPrecursor(double, double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Returns a recalibrated precursor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#getRef--">getRef()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Returns the value used as reference for the correction factors.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Reference.html#getReference--">getReference()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Reference.html" title="class in com.compomics.util.pride.prideobjects">Reference</a></dt>
<dd>
<div class="block">Returns the reference as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#getReferenceGroups--">getReferenceGroups()</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Returns the reference groups.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#getReferenceMass--">getReferenceMass()</a></span> - Method in class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Returns the reference total mass to use for normalization.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html#getReferences--">getReferences()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html" title="class in com.compomics.util.pride.prideobjects">ReferenceGroup</a></dt>
<dd>
<div class="block">Returns the references.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#getRefinementFixedModifications--">getRefinementFixedModifications()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Return the refinement fixed modifications used for the second pass
 search.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#getRefinementVariableModifications--">getRefinementVariableModifications()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Return the refinement variable modifications used for the second pass
 search.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getRefMass--">getRefMass()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the reference mass used to convert ppm to Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getRefMass--">getRefMass()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the reference mass for the conversion of the fragment ion
 tolerance from ppm to Dalton.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#getRelatedIons-char-">getRelatedIons(char)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Returns the list of related ions for the given amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#getRelatedIons-com.compomics.util.experiment.biology.aminoacids.AminoAcid-">getRelatedIons(AminoAcid)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Returns the list of related ions for the given amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getRelatedIons--">getRelatedIons()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Indicates whether related ions should be annotated by default.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html#getRelationship--">getRelationship()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature</a></dt>
<dd>
<div class="block">Returns the relationship between the features.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getRelativeError--">getRelativeError()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Get the relative m/z matching error in ppm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#getRelativeError-int-int-">getRelativeError(int, int)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Get the relative m/z matching error in ppm after isotope removal.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getRelimsPath--">getRelimsPath()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the path to the Relims installation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getRemoveMethionine--">getRemoveMethionine()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns true if n-term methionine is to be removed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getRemovePrecursor--">getRemovePrecursor()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns if the precursor is to be removed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getRemovePrecursor--">getRemovePrecursor()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true of the precursor peak is to be removed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getRemovePrecursorTolerance--">getRemovePrecursorTolerance()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the precursor removal tolerance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getRemovePrecursorTolerance--">getRemovePrecursorTolerance()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the tolerance with which to remove the precursor peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getRemoveTempFolders--">getRemoveTempFolders()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true if the output and index folders are to be removed when the
 search has completed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/Filter.html#getReport-boolean-">getReport(boolean)</a></span> - Method in interface com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/Filter.html" title="interface in com.compomics.util.experiment.filtering">Filter</a></dt>
<dd>
<div class="block">Returns a filter report depending on whether the condition was met.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getReport-boolean-">getReport(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#getReport-java.io.File-">getReport(File)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Returns the report.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#getReporterIndex-java.lang.String-">getReporterIndex(String)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">This method returns the reporter ion name associated to the given sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#getReporterIon-int-">getReporterIon(int)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Returns the reporter ion indexed by the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#getReporterIon-java.lang.String-">getReporterIon(String)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Returns the reporter ion corresponding to the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#getReporterIon--">getReporterIon()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Returns the reporter ion to look for in the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html#getReporterIon-java.lang.String-">getReporterIon(String)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethod</a></dt>
<dd>
<div class="block">Returns the reporter ion of the given name, null if not found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html#getReporterIonNames--">getReporterIonNames()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethod</a></dt>
<dd>
<div class="block">Returns a list containing the names of the reporter ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/IonFactory.html#getReporterIons-com.compomics.util.parameters.identification.search.ModificationParameters-">getReporterIons(ModificationParameters)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/IonFactory.html" title="class in com.compomics.util.experiment.biology.ions">IonFactory</a></dt>
<dd>
<div class="block">Returns the reporter ions to annotate with the given PTM settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getReporterIons--">getReporterIons()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Returns the reporter ions possibly encountered with this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getReporterIons--">getReporterIons()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Indicates whether reporter ions should be annotated by default.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#getReporterMethod--">getReporterMethod()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">Returns the reporter method used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html#getReporterMethod-java.lang.String-">getReporterMethod(String)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethodFactory</a></dt>
<dd>
<div class="block">Returns the reporter methods corresponding to the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getReporterPath--">getReporterPath()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the path to the Reporter installation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#getRepresentativeSites--">getRepresentativeSites()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Returns a list of all representative sites of ambiguously localized modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getRequireVariableMods--">getRequireVariableMods()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns true if at least one variable modification is required per
 peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#getResidues--">getResidues()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">This method reports on all the residues that can be modified by this Modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#getResidues--">getResidues()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This method reports on all the residues that can be modified by this Modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#getResidues--">getResidues()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">This method reports on all the residues that can be modified by this Modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html#getResolution--">getResolution()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAlignment</a></dt>
<dd>
<div class="block">Returns the resolution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html#getResolution--">getResolution()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb">PdbParameter</a></dt>
<dd>
<div class="block">Returns the resolution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#getResponseString--">getResponseString()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Returns all server response strings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#getResponseStringNoReset--">getResponseStringNoReset()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">This method allows the caller to read the response strings from
 the server, without resetting the internal buffer (and thus not clearing the
 messages read by this messages).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getRest--">getRest()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the rest of the header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#getRestrict--">getRestrict()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">Simple getter for the restricting residues of the Enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#getRestrictionAfter--">getRestrictionAfter()</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Getter for the amino acids restricting when following the cleavage.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#getRestrictionBefore--">getRestrictionBefore()</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Getter for the amino acids restricting when preceding the cleavage.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html#getReverseComplementary--">getReverseComplementary()</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html" title="class in com.compomics.util.nucleotide">NucleotideSequenceImpl</a></dt>
<dd>
<div class="block">This method returns the reverse complementary strand for the
 sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getRewindIons--">getRewindIons()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the rewind ions searched as list of integers as indexed in the
 FragmentIon class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html#getRewindStart-java.lang.String-">getRewindStart(String)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">NeutralLossesMap</a></dt>
<dd>
<div class="block">Returns the amino acid where a neutral loss should start being accounted
 for when predicting b ions (counting from N-terminus, first aa is 1).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#getRobustCorrelation-java.util.ArrayList-java.util.ArrayList-">getRobustCorrelation(ArrayList&lt;Double&gt;, ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Returns the population Pearson correlation r between series1 and series2.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/LinearRegression.html#getRobustLinearRegression-java.util.ArrayList-java.util.ArrayList-">getRobustLinearRegression(ArrayList&lt;Double&gt;, ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.statistics.linear_regression.<a href="com/compomics/util/math/statistics/linear_regression/LinearRegression.html" title="class in com.compomics.util.math.statistics.linear_regression">LinearRegression</a></dt>
<dd>
<div class="block">Returns a robust linear regression based on the median.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getRobustNonSymmetricalNormalDistribution-java.util.ArrayList-">getRobustNonSymmetricalNormalDistribution(ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>
<div class="block">Returns the non-symmetrical distribution of the input list of double
 calibrated on the median, 15.9% and 84.1% percentiles.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getRobustNonSymmetricalNormalDistributionFromSortedList-java.util.ArrayList-">getRobustNonSymmetricalNormalDistributionFromSortedList(ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>
<div class="block">Returns the non-symmetrical distribution of the input list of double
 calibrated on the median, 15.9% and 84.1% percentiles.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html#getRobustNormalDistribution-java.util.ArrayList-">getRobustNormalDistribution(ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NormalDistribution</a></dt>
<dd>
<div class="block">Returns the normal distribution corresponding to a given list of double
 calibrated on median and 34.1% percentile to median distance</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getRoundedMass-int-">getRoundedMass(int)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Getter for the rounded mass difference induced by this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getRoundedMass--">getRoundedMass()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Getter for the rounded mass difference induced by this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBResultSet.html#getRowCount--">getRowCount()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db">DBResultSet</a></dt>
<dd>
<div class="block">This method reports on the number of rows in the resultset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html#getRowCount--">getRowCount()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html" title="class in com.compomics.util.gui.spectrum">IonLabelColorTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html#getRowCount--">getRowCount()</a></span> - Method in class com.compomics.util.gui.variants.aa_substitutions.<a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html" title="class in com.compomics.util.gui.variants.aa_substitutions">AaSubstitutionMatrixTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableMap.html#getRowCount--">getRowCount()</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun">TableMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#getRowNumber-int-">getRowNumber(int)</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Returns the row number of the given view index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#getRt--">getRt()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Getter for the retention time in seconds.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#getRtInMinutes--">getRtInMinutes()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Returns the retention time in minutes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#getRtWindow--">getRtWindow()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Returns an array containing the min and max of the RT window.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#getSameMassNotFixedModifications-double-">getSameMassNotFixedModifications(double)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Returns a list containing all not fixed modifications with the same mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#getSample-java.lang.String-">getSample(String)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">This method returns the sample associated to the given ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#getSampleIndex-java.lang.String-">getSampleIndex(String)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">Returns the sample index for the given sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#getSampleIndexes--">getSampleIndexes()</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">Returns a set containing the indexes of every sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#getSamples--">getSamples()</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Returns the samples.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getScalingFactor--">getScalingFactor()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getScanNumber--">getScanNumber()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Getter for the scan number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getScanStartTime--">getScanStartTime()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the scan start time.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html#getScientificName-java.lang.Integer-">getScientificName(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblSpecies</a></dt>
<dd>
<div class="block">Returns the scientific name corresponding to the given NCBI taxon.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html#getScore-int-">getScore(int)</a></span> - Static method in enum com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html" title="enum in com.compomics.util.experiment.identification.modification">ModificationLocalizationScore</a></dt>
<dd>
<div class="block">Returns the PTM score indexed by the given id.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html#getScore-java.lang.String-">getScore(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html" title="enum in com.compomics.util.experiment.identification.modification">ModificationLocalizationScore</a></dt>
<dd>
<div class="block">Returns the PTM score of the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getScore--">getScore()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#getScore-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.PeptideSpectrumAnnotator-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getScore(Peptide, Spectrum, AnnotationParameters, SpecificAnnotationParameters, PeptideSpectrumAnnotator, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Returns the hyperscore.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#getScore-com.compomics.util.experiment.biology.proteins.Peptide-int-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.matches.IonMatch:A-">getScore(Peptide, int, Spectrum, IonMatch[])</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Returns the hyperscore.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/PrecursorAccuracy.html#getScore-com.compomics.util.experiment.biology.proteins.Peptide-int-com.compomics.util.experiment.mass_spectrometry.spectra.Precursor-boolean-int-int-">getScore(Peptide, int, Precursor, boolean, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/PrecursorAccuracy.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">PrecursorAccuracy</a></dt>
<dd>
<div class="block">Scores the match between the given peptide and spectrum using the
 precursor m/z accuracy.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/SnrScore.html#getScore-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.PeptideSpectrumAnnotator-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getScore(Peptide, Spectrum, AnnotationParameters, SpecificAnnotationParameters, PeptideSpectrumAnnotator, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/SnrScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">SnrScore</a></dt>
<dd>
<div class="block">Returns the score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/SnrScore.html#getScore-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.matches.IonMatch:A-">getScore(Peptide, Spectrum, IonMatch[])</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/SnrScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">SnrScore</a></dt>
<dd>
<div class="block">Returns the score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/SnrScore.html#getScore-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-java.util.HashMap-">getScore(Peptide, Spectrum, HashMap&lt;Double, ArrayList&lt;IonMatch&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/SnrScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">SnrScore</a></dt>
<dd>
<div class="block">Returns the score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html#getScore-int-">getScore(int)</a></span> - Static method in enum com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html" title="enum in com.compomics.util.experiment.identification.psm_scoring">PsmScore</a></dt>
<dd>
<div class="block">Returns the PSM score of the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html#getScore-java.lang.String-">getScore(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html" title="enum in com.compomics.util.experiment.identification.psm_scoring">PsmScore</a></dt>
<dd>
<div class="block">Returns the PSM score of the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScoresEstimator.html#getScore-com.compomics.util.experiment.biology.proteins.Peptide-java.lang.Integer-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.parameters.identification.IdentificationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.PeptideSpectrumAnnotator-int-">getScore(Peptide, Integer, Spectrum, IdentificationParameters, SpecificAnnotationParameters, ModificationParameters, SequenceProvider, SequenceMatchingParameters, PeptideSpectrumAnnotator, int)</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScoresEstimator.html" title="class in com.compomics.util.experiment.identification.psm_scoring">PsmScoresEstimator</a></dt>
<dd>
<div class="block">Scores the match between the given peptide and spectrum using the given
 score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScoresEstimator.html#getScore-com.compomics.util.experiment.biology.proteins.Peptide-java.lang.Integer-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.parameters.identification.IdentificationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.PeptideSpectrumAnnotator-com.compomics.util.experiment.identification.psm_scoring.PsmScore-">getScore(Peptide, Integer, Spectrum, IdentificationParameters, SpecificAnnotationParameters, ModificationParameters, SequenceProvider, SequenceMatchingParameters, PeptideSpectrumAnnotator, PsmScore)</a></span> - Method in class com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScoresEstimator.html" title="class in com.compomics.util.experiment.identification.psm_scoring">PsmScoresEstimator</a></dt>
<dd>
<div class="block">Scores the match between the given peptide and spectrum using the given
 score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#getScore--">getScore()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Returns the score assigned by the advocate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getScore--">getScore()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method will attribute a score to the current header, based on the
 following scoring list:   SwissProt : 4   IPI, SwissProt
 reference : 3   IPI, TrEMBL or REFSEQ_NP reference : 2 
  IPI, without SwissProt, TrEMBL or REFSEQ_NP reference : 1  
 NCBI, SwissProt reference : 2  NCBI, other reference : 1
  Unknown header format : 0 </div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getScore--">getScore()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#getScoredModifications--">getScoredModifications()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Returns a list of scored modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getScoredSites--">getScoredSites()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns a set of sites where a score is available.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#getScoreForAlgorithm-int-">getScoreForAlgorithm(int)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Returns the scores set for a given algorithm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSearchGuiOutputParameters--">getSearchGuiOutputParameters()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the selected SearchGUI output option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSearchGuiPath--">getSearchGuiPath()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the path to the SearchGUI installation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/MzIdentMLIdfileSearchParametersConverter.html#getSearchParameters-java.io.File-com.compomics.util.parameters.identification.search.SearchParameters-java.lang.String-com.compomics.util.waiting.WaitingHandler-">getSearchParameters(File, SearchParameters, String, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/MzIdentMLIdfileSearchParametersConverter.html" title="class in com.compomics.util.experiment.io.identification">MzIdentMLIdfileSearchParametersConverter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html#getSearchParameters--">getSearchParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SearchParametersDialog</a></dt>
<dd>
<div class="block">Returns a SearchParameters instance based on the user input in the GUI.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getSearchParameters--">getSearchParameters()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the parameters used for the spectrum matching.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#getSecondaryProgressBar--">getSecondaryProgressBar()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Returns the secondary progress bar for updates from external processes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#getSecondaryProgressCounter--">getSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#getSecondaryProgressCounter--">getSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#getSecondaryProgressCounter--">getSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#getSecondaryProgressCounter--">getSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#getSecondaryProgressCounter--">getSecondaryProgressCounter()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Returns the secondary progress counter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#getSecondaryPtmLocations--">getSecondaryPtmLocations()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Returns the not found, randomly or doubtfully localized PTMs.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#getSections--">getSections()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">returns the list of sections to be included in the scheme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html#getSelectedBackgroundSpecies--">getSelectedBackgroundSpecies()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html" title="class in com.compomics.util.parameters.identification.advanced">GeneParameters</a></dt>
<dd>
<div class="block">Returns the taxon of the species selected as background species.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#getSelectedCharges--">getSelectedCharges()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Returns the charges selected for annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/DateChooser.html#getSelectedDate--">getSelectedDate()</a></span> - Method in class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/DateChooser.html" title="class in com.compomics.util.gui.utils">DateChooser</a></dt>
<dd>
<div class="block">This method returns a Calendar set at the selected date,
 or 'null' if no date was selected.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html#getSelectedFastaFile--">getSelectedFastaFile()</a></span> - Method in class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SequenceDbDetailsDialog</a></dt>
<dd>
<div class="block">Returns the selected fasta file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/file_handling/FileSelectionDialog.html#getSelectedFile--">getSelectedFile()</a></span> - Method in class com.compomics.util.gui.file_handling.<a href="com/compomics/util/gui/file_handling/FileSelectionDialog.html" title="class in com.compomics.util.gui.file_handling">FileSelectionDialog</a></dt>
<dd>
<div class="block">Returns the file selected by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/file_handling/FileDisplayDialog.html#getSelectedFiles--">getSelectedFiles()</a></span> - Method in class com.compomics.util.gui.file_handling.<a href="com/compomics/util/gui/file_handling/FileDisplayDialog.html" title="class in com.compomics.util.gui.file_handling">FileDisplayDialog</a></dt>
<dd>
<div class="block">Returns the list of selected files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/ListChooser.html#getSelectedItem--">getSelectedItem()</a></span> - Method in class com.compomics.util.gui.utils.user_choice.<a href="com/compomics/util/gui/utils/user_choice/ListChooser.html" title="class in com.compomics.util.gui.utils.user_choice">ListChooser</a></dt>
<dd>
<div class="block">Returns the item selected by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/ListChooser.html#getSelectedItems--">getSelectedItems()</a></span> - Method in class com.compomics.util.gui.utils.user_choice.<a href="com/compomics/util/gui/utils/user_choice/ListChooser.html" title="class in com.compomics.util.gui.utils.user_choice">ListChooser</a></dt>
<dd>
<div class="block">Returns the items selected by the user in a list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#getSelectedMethod--">getSelectedMethod()</a></span> - Method in class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Returns the current spectrum counting method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getSelectedOutput--">getSelectedOutput()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the selected output type, omx, csv or pepXML.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getSelectedOutputFormat--">getSelectedOutputFormat()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the selected output format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#getSelectedPositions--">getSelectedPositions()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Returns the selected positions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#getSelectedProbabilisticScore--">getSelectedProbabilisticScore()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Returns the selected probabilistic score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/TableProperties.html#getSelectedRowHtmlTagFontColor--">getSelectedRowHtmlTagFontColor()</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/TableProperties.html" title="class in com.compomics.util.gui">TableProperties</a></dt>
<dd>
<div class="block">Returns the color to use for the HTML tags for the selected rows, in HTML
 color code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/genes/SpeciesDialog.html#getSelectedSpecies--">getSelectedSpecies()</a></span> - Method in class com.compomics.util.gui.genes.<a href="com/compomics/util/gui/genes/SpeciesDialog.html" title="class in com.compomics.util.gui.genes">SpeciesDialog</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">Returns the species selected in the species drop down menu.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#getSelection--">getSelection()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Returns true if there is a selection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#getSeparationLines--">getSeparationLines()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Returns the number of lines to be used to separate the sections.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#getSeparator--">getSeparator()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Returns the column separator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#getSequence--">getSequence()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Returns the sequence as String.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getSequence--">getSequence()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns for the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Protein.html#getSequence--">getSequence()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Protein.html" title="class in com.compomics.util.experiment.biology.proteins">Protein</a></dt>
<dd>
<div class="block">Getter for the protein sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getSequence-java.lang.String-">getSequence(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#getSequence--">getSequence()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Returns the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html#getSequence-java.lang.String-">getSequence(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html" title="class in com.compomics.util.experiment.identification.protein_sequences">SingleProteinSequenceProvider</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getSequence-java.lang.String-">getSequence(String)</a></span> - Method in interface com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></dt>
<dd>
<div class="block">Returns the protein sequence for the given accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Sequence.html#getSequence--">getSequence()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Sequence.html" title="interface in com.compomics.util.interfaces">Sequence</a></dt>
<dd>
<div class="block">This method will retrieve the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequence.html#getSequence--">getSequence()</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide">NucleotideSequence</a></dt>
<dd>
<div class="block">This method reports on the nucleotide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html#getSequence--">getSequence()</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html" title="class in com.compomics.util.nucleotide">NucleotideSequenceImpl</a></dt>
<dd>
<div class="block">This method will retrieve the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getSequence--">getSequence()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">Simple getter for the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#getSequence--">getSequence()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This method reports on the sequence for the current
 protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getSequenceCoverage-long-">getSequenceCoverage(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the sequence coverage of the protein of interest.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/IteratorFactory.html#getSequenceIterator-java.lang.String-com.compomics.util.parameters.identification.search.DigestionParameters-double-double-">getSequenceIterator(String, DigestionParameters, double, double)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/IteratorFactory.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">IteratorFactory</a></dt>
<dd>
<div class="block">Returns a sequence iterator for the given protein sequence and digestion
 preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#getSequenceMatchingParameters--">getSequenceMatchingParameters()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Returns the sequence matching preferences to use when mapping modifications on
 amino acid sequences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getSequenceMatchingParameters--">getSequenceMatchingParameters()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns the sequence matching parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/SequenceMatchingParametersDialog.html#getSequenceMatchingPreferences--">getSequenceMatchingPreferences()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/SequenceMatchingParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">SequenceMatchingParametersDialog</a></dt>
<dd>
<div class="block">Returns the sequence matching settings as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html#getSequenceMatchingType--">getSequenceMatchingType()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a></dt>
<dd>
<div class="block">Returns the sequence matching type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html#getSequenceProbabilities-com.compomics.util.experiment.biology.proteins.Peptide-java.util.ArrayList-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-boolean-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.PeptideSpectrumAnnotator-">getSequenceProbabilities(Peptide, ArrayList&lt;Modification&gt;, ModificationParameters, Spectrum, SequenceProvider, AnnotationParameters, SpecificAnnotationParameters, boolean, SequenceMatchingParameters, SequenceMatchingParameters, PeptideSpectrumAnnotator)</a></span> - Static method in class com.compomics.util.experiment.identification.modification.scores.<a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html" title="class in com.compomics.util.experiment.identification.modification.scores">PhosphoRS</a></dt>
<dd>
<div class="block">Returns the PhosphoRS sequence probabilities for the modification
 possible locations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#getSerializationFile--">getSerializationFile()</a></span> - Static method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Returns the file where to save the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#getSerializationFolder--">getSerializationFolder()</a></span> - Static method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Returns the folder where the factory is saved.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html" title="class in com.compomics.util.parameters.identification.advanced">FractionParameters</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html" title="class in com.compomics.util.parameters.identification.advanced">GeneParameters</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">IdMatchValidationParameters</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getShortDescription--">getShortDescription()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns a short description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#getShortName--">getShortName()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Getter for the short modification name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getSimpleDescription-java.lang.String-">getSimpleDescription(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getSimpleDescription-java.lang.String-">getSimpleDescription(String)</a></span> - Method in interface com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></dt>
<dd>
<div class="block">Returns the simple description of the protein with the given accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/LinearRegression.html#getSimpleLinearRegression-java.util.ArrayList-java.util.ArrayList-">getSimpleLinearRegression(ArrayList&lt;Double&gt;, ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.statistics.linear_regression.<a href="com/compomics/util/math/statistics/linear_regression/LinearRegression.html" title="class in com.compomics.util.math.statistics.linear_regression">LinearRegression</a></dt>
<dd>
<div class="block">Returns a simple linear regression.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/LinearRegression.html#getSimpleLinearRegressionOutlierRemoval-java.util.ArrayList-java.util.ArrayList-java.lang.Double-">getSimpleLinearRegressionOutlierRemoval(ArrayList&lt;Double&gt;, ArrayList&lt;Double&gt;, Double)</a></span> - Static method in class com.compomics.util.math.statistics.linear_regression.<a href="com/compomics/util/math/statistics/linear_regression/LinearRegression.html" title="class in com.compomics.util.math.statistics.linear_regression">LinearRegression</a></dt>
<dd>
<div class="block">Returns a simple linear regression performed after outlier removal.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getSimpleProteinDescription--">getSimpleProteinDescription()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns a simplified protein description for a UniProt header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#getSimplifyGroups--">getSimplifyGroups()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the protein groups should be
 simplified.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#getSimplifyGroupsEnzymaticity--">getSimplifyGroupsEnzymaticity()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the protein groups should be
 simplified based on the peptide enzymaticity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#getSimplifyGroupsEvidence--">getSimplifyGroupsEvidence()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the protein groups should be
 simplified based on the Uniprot evidence level.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#getSimplifyGroupsUncharacterized--">getSimplifyGroupsUncharacterized()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the protein groups consisting of
 uncharacterized proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#getSimplifyGroupsVariants--">getSimplifyGroupsVariants()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the protein groups should be
 simplified based on the peptide variant matching.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#getSingleAAModification-com.compomics.util.experiment.biology.modifications.Modification-">getSingleAAModification(Modification)</a></span> - Static method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Returns a clone of the given Modification targeting a single amino acid instead of
 a pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#getSingleAAModification-java.lang.String-">getSingleAAModification(String)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Returns a clone of the given Modification targeting a single amino acid instead of
 a pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#getSingleChargeWindow--">getSingleChargeWindow()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Returns the window size for singly charged ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getSingleLetterCodeAsChar--">getSingleLetterCodeAsChar()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the single letter code as character.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html#getSite--">getSite()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches">ModificationMatch</a></dt>
<dd>
<div class="block">Getter for the modification site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#getSite-int-int-">getSite(int, int)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">Returns the 1-based index on the peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#getSkylinePath--">getSkylinePath()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns the skyline path.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/Distribution.html#getSmallestCumulativeProbabilityAt-double-">getSmallestCumulativeProbabilityAt(double)</a></span> - Method in interface com.compomics.util.math.statistics.<a href="com/compomics/util/math/statistics/Distribution.html" title="interface in com.compomics.util.math.statistics">Distribution</a></dt>
<dd>
<div class="block">Returns the cumulative density function value at a given position, starting from the low values if before the median, from the high otherwise.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html#getSmallestCumulativeProbabilityAt-double-">getSmallestCumulativeProbabilityAt(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html" title="class in com.compomics.util.math.statistics.distributions">BinomialDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getSmallestCumulativeProbabilityAt-double-">getSmallestCumulativeProbabilityAt(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html#getSmallestCumulativeProbabilityAt-double-">getSmallestCumulativeProbabilityAt(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/IdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in interface com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/IdfileReader.html" title="interface in com.compomics.util.experiment.io.identification">IdfileReader</a></dt>
<dd>
<div class="block">Returns the names and versions of the software used to generate the
 identification file in a map, e.g., Mascot &gt; (2.2 and 2.3) and X!Tandem
 &gt; Sledgehammer (2013.09.01.1).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">AndromedaIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MascotIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MsAmandaIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MzIdentMLIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">NovorIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepxmlIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PNovoIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">TideIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html#getSoftwareVersions--">getSoftwareVersions()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">XTandemIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html#getSortedFeaturesList--">getSortedFeaturesList()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">FeaturesMap</a></dt>
<dd>
<div class="block">Returns a sorted list of features categories.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#getSortedImplementedIons--">getSortedImplementedIons()</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Returns an ordered list of possible subtypes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getSortedPeptideKeys-long-">getSortedPeptideKeys(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns a sorted list of peptide keys from the protein of interest.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getSortedPsmKeys-long-boolean-boolean-">getSortedPsmKeys(long, boolean, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the ordered list of spectrum keys for a given peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html#getSortedTermNames--">getSortedTermNames()</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html" title="class in com.compomics.util.experiment.biology.genes.go">GoMapping</a></dt>
<dd>
<div class="block">Returns a sorted list of all GO Terms names linked to proteins in the
 proteinToGoMap.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSortPsmsOnRt--">getSortPsmsOnRt()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns true if the PSMs are sorted on retention time, false sorts on PSM
 score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/events/RescalingEvent.html#getSource--">getSource()</a></span> - Method in class com.compomics.util.gui.events.<a href="com/compomics/util/gui/events/RescalingEvent.html" title="class in com.compomics.util.gui.events">RescalingEvent</a></dt>
<dd>
<div class="block">Returns the JPanel source.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Instrument.html#getSource--">getSource()</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Instrument.html" title="class in com.compomics.util.pride.prideobjects">Instrument</a></dt>
<dd>
<div class="block">Returns the instrument source.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#getSourceFile--">getSourceFile()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">This method reports on the sourcefile currently used in transfer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html#getSpAccession--">getSpAccession()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html" title="class in com.compomics.util.pdbfinder.das.readers">AlignmentBlock</a></dt>
<dd>
<div class="block">Returns the SP accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html#getSpAccession--">getSpAccession()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAlignment</a></dt>
<dd>
<div class="block">Returns the SP accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSparklineColor--">getSparklineColor()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Getter for the sparkline color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSparklineColorDoubtful--">getSparklineColorDoubtful()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the color for a doubtful sparkline bar chart plots.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSparklineColorFalsePositives--">getSparklineColorFalsePositives()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the color for false positives in sparkline bar chart plots.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSparklineColorNonValidated--">getSparklineColorNonValidated()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Getter for the non-validated sparkline color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSparklineColorNotFound--">getSparklineColorNotFound()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the color for a not found sparkline bar chart plots.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSparklineColorPossible--">getSparklineColorPossible()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the color for a possible sparkline bar chart plots.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getSpeciesDescription-java.util.TreeMap-">getSpeciesDescription(TreeMap&lt;String, Integer&gt;)</a></span> - Static method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns a listing of the species occurrence map provided.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getSpecificAnnotationParameters-java.lang.String-com.compomics.util.experiment.identification.SpectrumIdentificationAssumption-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getSpecificAnnotationParameters(String, SpectrumIdentificationAssumption, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns the annotation preferences specific to a spectrum and an
 identification assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#getSpecificity-java.lang.String-">getSpecificity(String)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Returns the expected specificity of the given enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html#getSpecificity-int-">getSpecificity(int)</a></span> - Static method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.Specificity</a></dt>
<dd>
<div class="block">Returns the specificity of the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/roc/DataRoc.html#getSpecificityAt-double-">getSpecificityAt(double)</a></span> - Method in class com.compomics.util.math.roc.<a href="com/compomics/util/math/roc/DataRoc.html" title="class in com.compomics.util.math.roc">DataRoc</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/roc/DistributionRoc.html#getSpecificityAt-double-">getSpecificityAt(double)</a></span> - Method in class com.compomics.util.math.roc.<a href="com/compomics/util/math/roc/DistributionRoc.html" title="class in com.compomics.util.math.roc">DistributionRoc</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/ROC.html#getSpecificityAt-double-">getSpecificityAt(double)</a></span> - Method in interface com.compomics.util.math.statistics.<a href="com/compomics/util/math/statistics/ROC.html" title="interface in com.compomics.util.math.statistics">ROC</a></dt>
<dd>
<div class="block">Returns the 1-specificity at a given sensitivity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#getSpectra-java.io.File-">getSpectra(File)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Reads an MGF file and retrieves a list of spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html#getSpectra-java.io.File-">getSpectra(File)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspReader</a></dt>
<dd>
<div class="block">Reads an MSP file and retrieves a list of spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#getSpectrum-java.io.BufferedReader-java.lang.String-">getSpectrum(BufferedReader, String)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Returns the next spectrum found in the mgf file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#getSpectrum-uk.ac.ebi.pride.tools.braf.BufferedRandomAccessFile-long-java.lang.String-">getSpectrum(BufferedRandomAccessFile, long, String)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Returns the next spectrum starting from the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html#getSpectrum-java.io.BufferedReader-java.lang.String-">getSpectrum(BufferedReader, String)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspReader</a></dt>
<dd>
<div class="block">The function returns the next spectrum found in the msp file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html#getSpectrum-uk.ac.ebi.pride.tools.braf.BufferedRandomAccessFile-long-java.lang.String-">getSpectrum(BufferedRandomAccessFile, long, String)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspReader</a></dt>
<dd>
<div class="block">Returns the next spectrum starting from the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getSpectrum-java.lang.String-java.lang.String-">getSpectrum(String, String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the desired spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getSpectrum-java.lang.String-">getSpectrum(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the desired spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSpectrumAnnotatedMirroredPeakColor--">getSpectrumAnnotatedMirroredPeakColor()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the color to use for the annotated mirrored peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSpectrumAnnotatedPeakColor--">getSpectrumAnnotatedPeakColor()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the color to use for the annotated peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSpectrumAnnotatedPeakWidth--">getSpectrumAnnotatedPeakWidth()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the width of the annotated peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#getSpectrumAnnotation-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getSpectrumAnnotation(AnnotationParameters, SpecificAnnotationParameters, Spectrum, Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the spectrum annotations of a spectrum in a list of IonMatches
 using the intensity filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#getSpectrumAnnotation-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getSpectrumAnnotation(AnnotationParameters, SpecificAnnotationParameters, Spectrum, Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the spectrum annotations of a spectrum in a list of IonMatches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html#getSpectrumAnnotation-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.amino_acid_tags.Tag-">getSpectrumAnnotation(AnnotationParameters, ModificationParameters, SequenceMatchingParameters, SpecificAnnotationParameters, Spectrum, Tag)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">TagSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the spectrum annotations of a spectrum in a list of IonMatches using an intensity filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html#getSpectrumAnnotation-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.amino_acid_tags.Tag-boolean-">getSpectrumAnnotation(AnnotationParameters, ModificationParameters, SequenceMatchingParameters, SpecificAnnotationParameters, Spectrum, Tag, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">TagSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the spectrum annotations of a spectrum in a list of IonMatches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#getSpectrumAnnotation-com.compomics.util.experiment.identification.matches.IonMatch:A-">getSpectrumAnnotation(IonMatch[])</a></span> - Static method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Translates the list of ion matches into a vector of annotations which can
 be read by the SpectrumPanel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#getSpectrumAnnotationStream-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">getSpectrumAnnotationStream(AnnotationParameters, SpecificAnnotationParameters, Spectrum, Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the spectrum annotations of a spectrum in a list of IonMatches
 using the intensity filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#getSpectrumAnnotationStream-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">getSpectrumAnnotationStream(AnnotationParameters, SpecificAnnotationParameters, Spectrum, Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the spectrum annotations of a spectrum in a list of IonMatches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#getSpectrumAnnotationStream-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-java.util.HashMap-boolean-">getSpectrumAnnotationStream(AnnotationParameters, SpecificAnnotationParameters, Spectrum, Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, HashMap&lt;Integer, HashMap&lt;Integer, ArrayList&lt;Ion&gt;&gt;&gt;, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the spectrum annotations of a spectrum in a list of IonMatches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSpectrumBackgroundPeakColor--">getSpectrumBackgroundPeakColor()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the color to use for the background peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getSpectrumBackgroundPeakWidth--">getSpectrumBackgroundPeakWidth()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the width of the background peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getSpectrumCharges--">getSpectrumCharges()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the spectrum charges to search for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getSpectrumComponent-int-java.lang.String-">getSpectrumComponent(int, String)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns a component in a spectrum line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getSpectrumComponentNames--">getSpectrumComponentNames()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns all the spectrum components names found in the header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html#getSpectrumCount--">getSpectrumCount()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html" title="class in com.compomics.util.experiment.identification.matches">PeptideMatch</a></dt>
<dd>
<div class="block">Returns the number of spectra matched.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getSpectrumCounting-long-">getSpectrumCounting(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the spectrum counting metric of the protein match of interest
 using the preference settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getSpectrumCounting-long-com.compomics.util.experiment.quantification.spectrumcounting.SpectrumCountingMethod-">getSpectrumCounting(long, SpectrumCountingMethod)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the spectrum counting metric of the protein match of interest for
 the given method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getSpectrumFile-java.lang.String-">getSpectrumFile(String)</a></span> - Static method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Convenience method to retrieve the name of a file from the spectrum key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getSpectrumFileFromIdName-java.lang.String-">getSpectrumFileFromIdName(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the spectrum file corresponding to the name of the file used for
 identification</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#getSpectrumFileId--">getSpectrumFileId()</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getSpectrumIdentification--">getSpectrumIdentification()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns a map of the spectrum matches keys indexed by spectrum file name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#getSpectrumIdentificationAssumption--">getSpectrumIdentificationAssumption()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Returns the spectrum identification assumption to annotate with.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getSpectrumIdentificationSize--">getSpectrumIdentificationSize()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns the number of spectrum identifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getSpectrumIds--">getSpectrumIds()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns all the spectrum IDs found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getSpectrumIndex-java.lang.String-">getSpectrumIndex(String)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns the spectrum index corresponding to the desired spectrum, i.e.,
 returns 0 for the first spectrum in the file, 1 for the second, etc.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getSpectrumIndex-java.lang.String-java.lang.String-">getSpectrumIndex(String, String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the spectrum index of the given spectrum in the given file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#getSpectrumKey--">getSpectrumKey()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Returns the key of the spectrum corresponding to this match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#getSpectrumKey--">getSpectrumKey()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Returns the key of the spectrum to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getSpectrumKey-java.lang.String-java.lang.String-">getSpectrumKey(String, String)</a></span> - Static method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Convenience method returning the key for a spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getSpectrumKey--">getSpectrumKey()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the key of the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getSpectrumMatch-long-">getSpectrumMatch(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns the spectrum match with the given key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getSpectrumMatchesIterator-long:A-com.compomics.util.waiting.WaitingHandler-">getSpectrumMatchesIterator(long[], WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns a spectrum matches iterator for a given key list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getSpectrumMatchesIterator-com.compomics.util.waiting.WaitingHandler-">getSpectrumMatchesIterator(WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns a spectrum matches iterator for all SpectrumMatches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#getSpectrumMatchesIterator-com.compomics.util.waiting.WaitingHandler-java.lang.String-">getSpectrumMatchesIterator(WaitingHandler, String)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns a psm iterator for all SpectrumMatches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html#getSpectrumMatchesKeys--">getSpectrumMatchesKeys()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html" title="class in com.compomics.util.experiment.identification.matches">PeptideMatch</a></dt>
<dd>
<div class="block">Returns the keys of all spectra matched.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#getSpectrumNumber--">getSpectrumNumber()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Returns the spectrum number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getSpectrumTitle-int-">getSpectrumTitle(int)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns the spectrum title corresponding to the given spectrum number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getSpectrumTitle-java.lang.String-">getSpectrumTitle(String)</a></span> - Static method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Convenience method to retrieve the name of a spectrum from the spectrum
 key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getSpectrumTitle--">getSpectrumTitle()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the spectrum title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getSpectrumTitle-java.lang.String-int-">getSpectrumTitle(String, int)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns the spectrum title of the spectrum of the given number in the
 given file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#getSpectrumTitles--">getSpectrumTitles()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns an ordered list of all spectrum titles.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#getSpectrumTitles-java.lang.String-">getSpectrumTitles(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns a list of titles from indexed spectra in the given file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html#getSpEnd--">getSpEnd()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html" title="class in com.compomics.util.pdbfinder.das.readers">AlignmentBlock</a></dt>
<dd>
<div class="block">Returns the SP end.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html#getSpStart--">getSpStart()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/AlignmentBlock.html" title="class in com.compomics.util.pdbfinder.das.readers">AlignmentBlock</a></dt>
<dd>
<div class="block">Returns the SP start.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getSqtOutput--">getSqtOutput()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true if the sqt output is to be created.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getStandardGeneticCode--">getStandardGeneticCode()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the standard genetic triplets associated to this amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html#getStandardGeneticCode--">getStandardGeneticCode()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">X</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getStandardGeneticCodeForCombination--">getStandardGeneticCodeForCombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the genetic code as combination of the sub amino acid genetic
 codes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/styles/DefaultStyle.html#getStandardHeight--">getStandardHeight()</a></span> - Method in class com.compomics.util.io.export.styles.<a href="com/compomics/util/io/export/styles/DefaultStyle.html" title="class in com.compomics.util.io.export.styles">DefaultStyle</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/WorkbookStyle.html#getStandardHeight--">getStandardHeight()</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/WorkbookStyle.html" title="interface in com.compomics.util.io.export">WorkbookStyle</a></dt>
<dd>
<div class="block">Returns the height of a standard line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getStandardSearchPattern--">getStandardSearchPattern()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Computes a pattern which can be searched by standard search engines,
 i.e., a pattern targeting a single amino acid and not a complex pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/styles/DefaultStyle.html#getStandardStyle--">getStandardStyle()</a></span> - Method in class com.compomics.util.io.export.styles.<a href="com/compomics/util/io/export/styles/DefaultStyle.html" title="class in com.compomics.util.io.export.styles">DefaultStyle</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/styles/DefaultStyle.html#getStandardStyle-int-">getStandardStyle(int)</a></span> - Method in class com.compomics.util.io.export.styles.<a href="com/compomics/util/io/export/styles/DefaultStyle.html" title="class in com.compomics.util.io.export.styles">DefaultStyle</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/WorkbookStyle.html#getStandardStyle--">getStandardStyle()</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/WorkbookStyle.html" title="interface in com.compomics.util.io.export">WorkbookStyle</a></dt>
<dd>
<div class="block">Returns the standard cell style.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/WorkbookStyle.html#getStandardStyle-int-">getStandardStyle(int)</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/WorkbookStyle.html" title="interface in com.compomics.util.io.export">WorkbookStyle</a></dt>
<dd>
<div class="block">Returns the standard cell style for the given hierarchical depth.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/StandardUnit.html#getStandardUnit-java.lang.String-">getStandardUnit(String)</a></span> - Static method in enum com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/StandardUnit.html" title="enum in com.compomics.util.experiment.units">StandardUnit</a></dt>
<dd>
<div class="block">Returns the standard unit having the given full name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getStarred--">getStarred()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns whether a match is starred or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#getStart--">getStart()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Get the start value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getStart--">getStart()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the start.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#getStartBlock--">getStartBlock()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Returns the block start.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getStartLocation--">getStartLocation()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method reports on the start index of the header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/StartEndPosition.html#getStartPosition--">getStartPosition()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/StartEndPosition.html" title="class in com.compomics.util.pdbfinder.das.readers">StartEndPosition</a></dt>
<dd>
<div class="block">Returns the start position.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#getStartProtein--">getStartProtein()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Returns the protein start.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getStdDown--">getStdDown()</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>
<div class="block">Returns the standard deviation to the left of the distribution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getStdUp--">getStdUp()</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>
<div class="block">Returns the standard deviation to the right of the distribution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getStoreSpectraFileName--">getStoreSpectraFileName()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the name of the file where the binary spectra are to be stored.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html#getStringMatching--">getStringMatching()</a></span> - Static method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a></dt>
<dd>
<div class="block">Returns preferences for simple string matching.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#getStringParameter-java.lang.String-">getStringParameter(String)</a></span> - Method in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/styles/DefaultStyle.html#getStyle-com.compomics.util.io.export.writers.ExcelWriter-">getStyle(ExcelWriter)</a></span> - Static method in class com.compomics.util.io.export.styles.<a href="com/compomics/util/io/export/styles/DefaultStyle.html" title="class in com.compomics.util.io.export.styles">DefaultStyle</a></dt>
<dd>
<div class="block">Returns the style attached to that writer or create a new one if none
 found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getSubAminoAcids--">getSubAminoAcids()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">In case of a combination of amino acids, returns the comprised amino
 acids or amino acid groups represented by their single letter code
 including sub combinations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getSubAminoAcids-boolean-">getSubAminoAcids(boolean)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">In case of a combination of amino acids, returns the comprised amino
 acids or amino acid groups represented by their single letter code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#getSubMatrix-int-int-int-int-">getSubMatrix(int, int, int, int)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getSubNumber--">getSubNumber()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Returns the number of the fragment in the current amino acid sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getSubPattern-int-int-boolean-">getSubPattern(int, int, boolean)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns a sub pattern of the pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getSubPattern-int-boolean-">getSubPattern(int, boolean)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns a sub pattern of the pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getSubsequence-java.lang.String-int-int-">getSubsequence(String, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html#getSubsequence-java.lang.String-int-int-">getSubsequence(String, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html" title="class in com.compomics.util.experiment.identification.protein_sequences">SingleProteinSequenceProvider</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/SequenceProvider.html#getSubsequence-java.lang.String-int-int-">getSubsequence(String, int, int)</a></span> - Method in interface com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">SequenceProvider</a></dt>
<dd>
<div class="block">Returns the subsequence of the sequence of a given protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getSubSpectrum-double-double-">getSubSpectrum(double, double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the part of the spectrum contained between mzMin (inclusive) and
 mzMax (exclusive) as a peak list</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/amino_acids/Substitution.html#getSubstitutedAminoAcid--">getSubstitutedAminoAcid()</a></span> - Method in class com.compomics.util.experiment.biology.variants.amino_acids.<a href="com/compomics/util/experiment/biology/variants/amino_acids/Substitution.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids">Substitution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#getSubstitutionAminoAcids-java.lang.Character-">getSubstitutionAminoAcids(Character)</a></span> - Method in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Returns the possible substituted amino acids for the given amino acid as
 a list of their single letter code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#getSubstitutionAminoAcids--">getSubstitutionAminoAcids()</a></span> - Method in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Returns the possible substituted amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#getSubType--">getSubType()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#getSubType--">getSubType()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#getSubType--">getSubType()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#getSubType--">getSubType()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#getSubType--">getSubType()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#getSubType--">getSubType()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#getSubType--">getSubType()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getSubType--">getSubType()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getSubType--">getSubType()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the ion subtype.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#getSubTypeAsString--">getSubTypeAsString()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#getSubTypeAsString--">getSubTypeAsString()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#getSubTypeAsString--">getSubTypeAsString()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#getSubTypeAsString--">getSubTypeAsString()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#getSubTypeAsString-int-">getSubTypeAsString(int)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Returns the type of fragment ion as a letter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#getSubTypeAsString--">getSubTypeAsString()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#getSubTypeAsString--">getSubTypeAsString()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#getSubTypeAsString--">getSubTypeAsString()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getSubTypeAsString--">getSubTypeAsString()</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#getSubTypeAsString-int-">getSubTypeAsString(int)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Returns the type of fragment ion as a letter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getSubTypeAsString--">getSubTypeAsString()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the subtype as string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html#getSummary-java.io.File-com.compomics.util.experiment.io.biology.protein.FastaParameters-com.compomics.util.waiting.WaitingHandler-">getSummary(File, FastaParameters, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaSummary</a></dt>
<dd>
<div class="block">Gathers summary data on the FASTA file content.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/ListChooser.html#getTable--">getTable()</a></span> - Method in class com.compomics.util.gui.utils.user_choice.<a href="com/compomics/util/gui/utils/user_choice/ListChooser.html" title="class in com.compomics.util.gui.utils.user_choice">ListChooser</a></dt>
<dd>
<div class="block">Returns the JTable containing the items.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VisibleTableColumnsDialogParent.html#getTable--">getTable()</a></span> - Method in interface com.compomics.util.gui.<a href="com/compomics/util/gui/VisibleTableColumnsDialogParent.html" title="interface in com.compomics.util.gui">VisibleTableColumnsDialogParent</a></dt>
<dd>
<div class="block">Returns the JTable for which to hide/show the column for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#getTable--">getTable()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/AlignedTableCellRenderer.html#getTableCellRendererComponent-javax.swing.JTable-java.lang.Object-boolean-boolean-int-int-">getTableCellRendererComponent(JTable, Object, boolean, boolean, int, int)</a></span> - Method in class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/AlignedTableCellRenderer.html" title="class in com.compomics.util.gui.renderers">AlignedTableCellRenderer</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/ByteArrayRenderer.html#getTableCellRendererComponent-javax.swing.JTable-java.lang.Object-boolean-boolean-int-int-">getTableCellRendererComponent(JTable, Object, boolean, boolean, int, int)</a></span> - Method in class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/ByteArrayRenderer.html" title="class in com.compomics.util.gui.renderers">ByteArrayRenderer</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/FragmentIonTableCellRenderer.html#getTableCellRendererComponent-javax.swing.JTable-java.lang.Object-boolean-boolean-int-int-">getTableCellRendererComponent(JTable, Object, boolean, boolean, int, int)</a></span> - Method in class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/FragmentIonTableCellRenderer.html" title="class in com.compomics.util.gui.renderers">FragmentIonTableCellRenderer</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/GraphicsPanelTableCellRenderer.html#getTableCellRendererComponent-javax.swing.JTable-java.lang.Object-boolean-boolean-int-int-">getTableCellRendererComponent(JTable, Object, boolean, boolean, int, int)</a></span> - Method in class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/GraphicsPanelTableCellRenderer.html" title="class in com.compomics.util.gui.renderers">GraphicsPanelTableCellRenderer</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/TimestampRenderer.html#getTableCellRendererComponent-javax.swing.JTable-java.lang.Object-boolean-boolean-int-int-">getTableCellRendererComponent(JTable, Object, boolean, boolean, int, int)</a></span> - Method in class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/TimestampRenderer.html" title="class in com.compomics.util.gui.renderers">TimestampRenderer</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#getTableName--">getTableName()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method reports on the tablename.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/ListChooser.html#getTableTooltips--">getTableTooltips()</a></span> - Method in class com.compomics.util.gui.utils.user_choice.<a href="com/compomics/util/gui/utils/user_choice/ListChooser.html" title="class in com.compomics.util.gui.utils.user_choice">ListChooser</a></dt>
<dd>
<div class="block">Returns the tooltips used for the column headers.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html#getTag--">getTag()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">TagSpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns the tag to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html#getTag--">getTag()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">TagAssumption</a></dt>
<dd>
<div class="block">Returns the tag of this assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#getTagAssumptionsMap--">getTagAssumptionsMap()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Returns the tag assumptions map: advocate id &gt; score &gt; list of
 assumptions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getTaggedModifiedSequence-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-boolean-boolean-java.util.HashSet-">getTaggedModifiedSequence(ModificationParameters, SequenceProvider, SequenceMatchingParameters, boolean, boolean, boolean, HashSet&lt;String&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the modified sequence as an tagged string with potential
 modification sites color coded or with Modification tags, e.g,
 &lt;mox&gt;.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getTaggedModifiedSequence-com.compomics.util.parameters.identification.search.ModificationParameters-boolean-boolean-boolean-boolean-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-java.util.HashSet-">getTaggedModifiedSequence(ModificationParameters, boolean, boolean, boolean, boolean, SequenceMatchingParameters, HashSet&lt;String&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the modified sequence as an tagged string with potential
 modification sites color coded or with modification tags, e.g,
 &lt;mox&gt;.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getTaggedModifiedSequence-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.identification.amino_acid_tags.Tag-boolean-boolean-boolean-boolean-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-java.util.HashSet-">getTaggedModifiedSequence(ModificationParameters, Tag, boolean, boolean, boolean, boolean, SequenceMatchingParameters, HashSet&lt;String&gt;)</a></span> - Static method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the modified sequence as an tagged string with potential
 modification sites color coded or with modification tags, e.g,
 &lt;mox&gt;.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#getTaggedModifiedSequence-com.compomics.util.parameters.identification.search.ModificationParameters-java.lang.String-java.lang.String:A-java.lang.String:A-java.lang.String:A-java.lang.String:A-boolean-boolean-">getTaggedModifiedSequence(ModificationParameters, String, String[], String[], String[], String[], boolean, boolean)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">Returns the modified sequence as an tagged string with potential
 modification sites color coded or with PTM tags, e.g, &lt;mox&gt;.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#getTaggedModifiedSequence-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-java.lang.String:A-java.lang.String:A-java.lang.String:A-java.lang.String:A-boolean-boolean-boolean-">getTaggedModifiedSequence(Peptide, ModificationParameters, String[], String[], String[], String[], boolean, boolean, boolean)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns the modified sequence as an tagged string with potential
 modification sites color coded or with Modification tags, e.g,
 &lt;mox&gt;.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getTaggingTimeSeconds--">getTaggingTimeSeconds()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns the tagging time in seconds as listed in the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getTagLength--">getTagLength()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the tag length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#getTagModificationsAsString-com.compomics.util.experiment.identification.amino_acid_tags.Tag-">getTagModificationsAsString(Tag)</a></span> - Static method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns the tag modifications as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getTagParameter-java.lang.String-">getTagParameter(String)</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns the tagging parameter corresponding to a given parameter name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getTagsGenerator--">getTagsGenerator()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns the tags generator used to create the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getTagsGeneratorVersion--">getTagsGeneratorVersion()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns the version of the tags generator used to create the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getTagsParametersNames--">getTagsParametersNames()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns the name of the different parameters names found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getTarget--">getTarget()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the index of the amino acid of interest in the pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getTargetDecoyFileNameSuffix--">getTargetDecoyFileNameSuffix()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the target-decoy file name suffix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getTargetedAA-int-">getTargetedAA(int)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the targeted amino acids at a given index in the pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#getTaxonomy-java.lang.String-">getTaxonomy(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#getTaxonomy--">getTaxonomy()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Returns the taxonomy.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html#getTaxonomy-java.lang.String-">getTaxonomy(String)</a></span> - Method in interface com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein">ProteinDetailsProvider</a></dt>
<dd>
<div class="block">Returns the taxonomy for the given protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html#getTaxons--">getTaxons()</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblGenomesSpecies</a></dt>
<dd>
<div class="block">Returns the taxons in this map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html#getTaxons--">getTaxons()</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblSpecies</a></dt>
<dd>
<div class="block">Returns the taxons in this map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html#getTermAccession-java.lang.String-">getTermAccession(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html" title="class in com.compomics.util.experiment.biology.genes.go">GoMapping</a></dt>
<dd>
<div class="block">Returns the accession of a GO term.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html#getTermDomain-java.lang.String-">getTermDomain(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html" title="class in com.compomics.util.experiment.biology.genes.go">GoDomains</a></dt>
<dd>
<div class="block">Returns the description of a GO term.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html#getTermName-java.lang.String-">getTermName(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html" title="class in com.compomics.util.experiment.biology.genes.go">GoMapping</a></dt>
<dd>
<div class="block">Returns the name of a GO term.</div>
</dd>
<dt><a href="com/compomics/util/db/components/GettersAndSetters.html" title="class in com.compomics.util.db.components"><span class="typeNameLink">GettersAndSetters</span></a> - Class in <a href="com/compomics/util/db/components/package-summary.html">com.compomics.util.db.components</a></dt>
<dd>
<div class="block">This class will generate (and subsequently output upon request) all relevant getters and setters
 for a DBAccessor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/GettersAndSetters.html#GettersAndSetters-com.compomics.util.db.DBMetaData-">GettersAndSetters(DBMetaData)</a></span> - Constructor for class com.compomics.util.db.components.<a href="com/compomics/util/db/components/GettersAndSetters.html" title="class in com.compomics.util.db.components">GettersAndSetters</a></dt>
<dd>
<div class="block">The constructor takes care of generating all getters and setters,
 based on the speicifed metadata.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getTextOutput--">getTextOutput()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true if the text output is to be created.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getTheoreticalFragmentIonsSumOnly--">getTheoreticalFragmentIonsSumOnly()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns true if the theoretical fragment ions are calculated on the sum
 in the bin alone.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getTheoreticMass--">getTheoreticMass()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the theoretic mass, from the atomic composition if available,
 from the theoreticMass field otherwise.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html#getTheoreticMass--">getTheoreticMass()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">PeptideAssumption</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html#getTheoreticMass--">getTheoreticMass()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">TagAssumption</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html#getTheoreticMass-boolean-boolean-">getTheoreticMass(boolean, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">TagAssumption</a></dt>
<dd>
<div class="block">Returns the theoretic mass of the tag, eventually without terminal gaps.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#getTheoreticMass--">getTheoreticMass()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Returns the theoretic mass of the given assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getTheoreticMz-java.lang.Integer-">getTheoreticMz(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the m/z expected for this ion at the given charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html#getTheoreticMz-boolean-boolean-">getTheoreticMz(boolean, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">TagAssumption</a></dt>
<dd>
<div class="block">Returns the theoretic mass of the tag, eventually without terminal gaps.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#getTheoreticMz--">getTheoreticMz()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Returns the theoretic m/z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getTicCutoffPercentage--">getTicCutoffPercentage()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the TIC cutoff in percent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getTicCutoffPercentage--">getTicCutoffPercentage()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns the TicCutoffPercentage.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#getTiesResolution--">getTiesResolution()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns the ties resolution method to use when multiple peaks can be
 assigned to an ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getTimeEnd--">getTimeEnd()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns the ending time of the tagging as given in the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#getTimeStart--">getTimeStart()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">Returns the starting time of the tagging as given in the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#getTip-java.lang.String-java.lang.String-java.util.ArrayList-">getTip(String, String, ArrayList&lt;String&gt;)</a></span> - Static method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Creates a tip object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#getTip-java.lang.String-java.lang.String-">getTip(String, String)</a></span> - Static method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Creates a general tip with no key word.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#getTitle--">getTitle()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">This method returns the title of the modification,
 eg.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFeature.html#getTitle--">getTitle()</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFeature.html" title="interface in com.compomics.util.io.export">ExportFeature</a></dt>
<dd>
<div class="block">Returns the column title of the feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#getTitle--">getTitle()</a></span> - Method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Returns the title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html#getTitle--">getTitle()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasAlignment.html" title="class in com.compomics.util.pdbfinder.das.readers">DasAlignment</a></dt>
<dd>
<div class="block">Returns the title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html#getTitle--">getTitle()</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb">PdbParameter</a></dt>
<dd>
<div class="block">Returns the title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#getTitle--">getTitle()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">Simple getter for the title (name) of the Enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#getTitle--">getTitle()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This method returns the title of the modification,
 eg.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#getTitle--">getTitle()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">This method returns the title of the modification,
 eg.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#getTooltipMap--">getTooltipMap()</a></span> - Static method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>
<div class="block">Returns the default map of icons for the validation levels.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getTopPeaks--">getTopPeaks()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the top peaks number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#getTopPeaksWindow--">getTopPeaksWindow()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns the top peaks window size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#getTotalIntensity--">getTotalIntensity()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>
<div class="block">Returns the total intensity of the peaks above the intensity threshold.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#getTotalIntensity--">getTotalIntensity()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns the total intensity of the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#getTotalIntensity--">getTotalIntensity()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Method calculates the total intensity of the peaks</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#getTotalIntensity--">getTotalIntensity()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method returns the total intensity contributed by all the ions
 in this spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getTotalPeptidesPerFraction--">getTotalPeptidesPerFraction()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the total number of peptides per fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getTotalSpectrumCounting--">getTotalSpectrumCounting()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the total spectrum counting.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getTotalSpectrumCountingMass--">getTotalSpectrumCountingMass()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the total spectrum counting masses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#getTrace--">getTrace()</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getTransformedScore--">getTransformedScore()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the log transformed score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#getTruncatedSequence-int-int-">getTruncatedSequence(int, int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method will return an AASequenceImpl that represents
 an internal truncation of the current sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#getTruncationPosition--">getTruncationPosition()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This method reports on the position of the truncation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#getTrypsinExample--">getTrypsinExample()</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the trypsin example as amino acid pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getType--">getType()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the ion type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#getType--">getType()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Returns the type of advocate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getType--">getType()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Return the type of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationMatch.html#getType--">getType()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationMatch.html" title="class in com.compomics.util.experiment.identification">IdentificationMatch</a></dt>
<dd>
<div class="block">Returns the type of match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html#getType--">getType()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html" title="class in com.compomics.util.experiment.identification.matches">PeptideMatch</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#getType--">getType()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#getType--">getType()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/parameters/DummyParameters.html#getType--">getType()</a></span> - Method in class com.compomics.util.experiment.io.parameters.<a href="com/compomics/util/experiment/io/parameters/DummyParameters.html" title="class in com.compomics.util.experiment.io.parameters">DummyParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/parameters/MarshallableParameter.html#getType--">getType()</a></span> - Method in interface com.compomics.util.experiment.io.parameters.<a href="com/compomics/util/experiment/io/parameters/MarshallableParameter.html" title="interface in com.compomics.util.experiment.io.parameters">MarshallableParameter</a></dt>
<dd>
<div class="block">Returns the type of marshalled parameter from an unmarshalled object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/patient/PatientInformation.html#getType--">getType()</a></span> - Method in class com.compomics.util.experiment.patient.<a href="com/compomics/util/experiment/patient/PatientInformation.html" title="class in com.compomics.util.experiment.patient">PatientInformation</a></dt>
<dd>
<div class="block">Returns the type of information.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#getType--">getType()</a></span> - Method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Returns the feedback type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#getType--">getType()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#getType--">getType()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getType--">getType()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/webservice/query/PrideFilter.html#getType--">getType()</a></span> - Method in class com.compomics.util.pride.prideobjects.webservice.query.<a href="com/compomics/util/pride/prideobjects/webservice/query/PrideFilter.html" title="class in com.compomics.util.pride.prideobjects.webservice.query">PrideFilter</a></dt>
<dd>
<div class="block">Returns the filter type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getTypeAsString--">getTypeAsString()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the type of ion as string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#getTypeAsString-com.compomics.util.experiment.biology.ions.Ion.IonType-">getTypeAsString(Ion.IonType)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the type of ion as string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html#getTypeAsString--">getTypeAsString()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaSummary</a></dt>
<dd>
<div class="block">Returns a string with the different database types found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getTypeCategory--">getTypeCategory()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the type category.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/genes/SpeciesDialog.html#getTypeForGenomeDivision-com.compomics.util.experiment.biology.taxonomy.mappings.EnsemblGenomesSpecies.EnsemblGenomeDivision-">getTypeForGenomeDivision(EnsemblGenomesSpecies.EnsemblGenomeDivision)</a></span> - Method in class com.compomics.util.gui.genes.<a href="com/compomics/util/gui/genes/SpeciesDialog.html" title="class in com.compomics.util.gui.genes">SpeciesDialog</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">Returns the display name of an Ensembl genome division.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getTypeId--">getTypeId()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the type ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getTypeReference--">getTypeReference()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the type reference.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getTypeSubparts--">getTypeSubparts()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the type subparts.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#getTypeSuperparts--">getTypeSuperparts()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns the type super parts.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#getUnevenRowColor--">getUnevenRowColor()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">This method returns the Color used for the uneven numbered rows, or 'null' if the
 default JTable background color is maintained for these rows.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/modification/ModificationsDialog.html#getUniModAccessionLink-java.lang.String-">getUniModAccessionLink(String)</a></span> - Method in class com.compomics.util.gui.modification.<a href="com/compomics/util/gui/modification/ModificationsDialog.html" title="class in com.compomics.util.gui.modification">ModificationsDialog</a></dt>
<dd>
<div class="block">Returns a web link to the given Unimod CV term at
 http://www.unimod.org.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getUniprotTaxonomy--">getUniprotTaxonomy()</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the UniProt taxonomy mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#getUniprotTaxonomyFile-java.lang.String-">getUniprotTaxonomyFile(String)</a></span> - Static method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Returns the UniProt taxonomy file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getUniqueAminoAcids--">getUniqueAminoAcids()</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns a char array of the one letter code of amino acids without
 combinations of amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getUniqueIndex-char-">getUniqueIndex(char)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns an index for the amino acid excluding combinations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#getUnit--">getUnit()</a></span> - Method in class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Returns the unit used for normalization.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getUpperClearMzRange--">getUpperClearMzRange()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Returns the upper mass value for the clear mz range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#getUpperPrecursorMass--">getUpperPrecursorMass()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Returns the upper precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/WebDAO.html#getUrlOfZippedVersion-java.net.URL-java.lang.String-boolean-">getUrlOfZippedVersion(URL, String, boolean)</a></span> - Static method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/WebDAO.html" title="class in com.compomics.software.autoupdater">WebDAO</a></dt>
<dd>
<div class="block">Gets the first zip file from a URL, in case of a Maven repo deploy this
 should be the only zip in the folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/personalization/ExperimentObject.html#getUrParam-com.compomics.util.experiment.personalization.UrParameter-">getUrParam(UrParameter)</a></span> - Method in class com.compomics.util.experiment.personalization.<a href="com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></dt>
<dd>
<div class="block">Returns the refinement parameter of the same type than the one provided.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/personalization/ExperimentObject.html#getUrParams--">getUrParams()</a></span> - Method in class com.compomics.util.experiment.personalization.<a href="com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></dt>
<dd>
<div class="block">Returns the user parameters map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getUseFlankingPeaks--">getUseFlankingPeaks()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true if flanking peaks are to be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html#getUseGeneMapping--">getUseGeneMapping()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html" title="class in com.compomics.util.parameters.identification.advanced">GeneParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether gene mappings should be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getUseNeutralLossPeaks--">getUseNeutralLossPeaks()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns true if the neutral loss peaks are to be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#getUserAdvocates--">getUserAdvocates()</a></span> - Static method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Returns the map of user advocates imported.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#getUserModifications--">getUserModifications()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Returns the names of the user defined modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#getUserModificationsOrdered--">getUserModificationsOrdered()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Returns the alphabetically ordered names of the user defined
 modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getUserParametersFile--">getUserParametersFile()</a></span> - Static method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the user preferences file to be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#getUserParametersFolder--">getUserParametersFolder()</a></span> - Static method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns the user preferences file to be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#getUserSelectedFile-java.awt.Component-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String-boolean-">getUserSelectedFile(Component, String, String, String, String, String, boolean)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Returns the file selected by the user, or null if no file was selected.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#getUserSelectedFile-java.awt.Component-java.lang.String:A-java.lang.String:A-java.lang.String-java.lang.String-java.lang.String-boolean-boolean-boolean-int-">getUserSelectedFile(Component, String[], String[], String, String, String, boolean, boolean, boolean, int)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Returns the file selected by the user, or null if no file was selected.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#getUserSelectedFolder-java.awt.Component-java.lang.String-java.lang.String-java.lang.String-java.lang.String-boolean-">getUserSelectedFolder(Component, String, String, String, String, boolean)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Returns the folder selected by the user, or null if no folder was
 selected.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#getUseSmartPlusThreeModel--">getUseSmartPlusThreeModel()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Returns true if the UseSmartPlusThreeModel is to be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#getUseSparseMatrix--">getUseSparseMatrix()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">from Comet release 2015.02 all searches use this internal
 data representation by default</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#getUseSpecificCount--">getUseSpecificCount()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Returns whether the specific variant count limitations should be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#getUtilitiesEnzyme-java.lang.String-">getUtilitiesEnzyme(String)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Returns the enzyme associated to the given cvTerm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#getUtilitiesModificationName-int-">getUtilitiesModificationName(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns the name of the modification indexed by the given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html#getUtilitiesPtmName-java.lang.String-">getUtilitiesPtmName(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">NovorParameters</a></dt>
<dd>
<div class="block">Returns the utilities PTM name corresponding to the given Novor PTM
 short name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#getUtilitiesPtmName-java.lang.String-">getUtilitiesPtmName(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Returns the utilities PTM name corresponding to the given PepNovo PTM
 name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#getUtilitiesPtmName-java.lang.Character-">getUtilitiesPtmName(Character)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Returns the utilities PTM name corresponding to the given pNovo PTM
 character.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/JavaHomeOrMemoryDialogParent.html#getUtilitiesUserParameters--">getUtilitiesUserParameters()</a></span> - Method in interface com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/JavaHomeOrMemoryDialogParent.html" title="interface in com.compomics.software.dialogs">JavaHomeOrMemoryDialogParent</a></dt>
<dd>
<div class="block">Returns the utilities user preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getValidatedPeptidesPerFraction--">getValidatedPeptidesPerFraction()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the number of validated peptides per fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#getValidatedProteinList--">getValidatedProteinList()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Returns a list of validated proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getValidatedProteins-com.compomics.util.gui.filtering.FilterParameters-">getValidatedProteins(FilterParameters)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the list of validated protein keys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getValidatedProteins-com.compomics.util.waiting.WaitingHandler-com.compomics.util.gui.filtering.FilterParameters-">getValidatedProteins(WaitingHandler, FilterParameters)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the list of validated protein keys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#getValidatedSequenceCoverage-long-">getValidatedSequenceCoverage(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns the sequence coverage of the protein of interest.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#getValidatedSpectraPerFraction--">getValidatedSpectraPerFraction()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns the number of validated spectra per fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#getValidatedTargetProteinKeys--">getValidatedTargetProteinKeys()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Returns the keys of the validated target protein matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#getValidationLevelIndexes--">getValidationLevelIndexes()</a></span> - Static method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>
<div class="block">Returns an array of the validation level indexes in ascending order of
 confidence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#getValidationLevels--">getValidationLevels()</a></span> - Static method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>
<div class="block">Returns the implemented validation levels.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#getValidationLevelsNames--">getValidationLevelsNames()</a></span> - Static method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>
<div class="block">Returns a list of the names of the implemented validation levels.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html#getValidationQCParameters--">getValidationQCParameters()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationQCParametersDialog</a></dt>
<dd>
<div class="block">Returns the validation QC preferences as set by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html#getValidationQCParameters--">getValidationQCParameters()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">IdMatchValidationParameters</a></dt>
<dd>
<div class="block">Returns the validation QC preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#getValue-java.lang.String-">getValue(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Returns the value used for comparison for a given filtering item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html#getValue-java.lang.Integer-">getValue(Integer)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html" title="class in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsConvertParameters</a></dt>
<dd>
<div class="block">Returns the value set for a given filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/TableValueWrapper.html#getValue--">getValue()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/TableValueWrapper.html" title="interface in com.compomics.util.interfaces">TableValueWrapper</a></dt>
<dd>
<div class="block">This method returns the Object that this instance wraps around.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/CvTerm.html#getValue--">getValue()</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/CvTerm.html" title="class in com.compomics.util.pride">CvTerm</a></dt>
<dd>
<div class="block">Returns the value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/webservice/query/PrideFilter.html#getValue--">getValue()</a></span> - Method in class com.compomics.util.pride.prideobjects.webservice.query.<a href="com/compomics/util/pride/prideobjects/webservice/query/PrideFilter.html" title="class in com.compomics.util.pride.prideobjects.webservice.query">PrideFilter</a></dt>
<dd>
<div class="block">Returns the filter value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/SwingWorker.html#getValue--">getValue()</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/SwingWorker.html" title="class in com.compomics.util.sun">SwingWorker</a></dt>
<dd>
<div class="block">Get the value produced by the worker thread, or null if it 
 hasn't been constructed yet.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBResultSet.html#getValueAt-int-int-">getValueAt(int, int)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db">DBResultSet</a></dt>
<dd>
<div class="block">Returns the value for the cell at <code>columnIndex</code> and
 <code>rowIndex</code>.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html#getValueAt-int-int-">getValueAt(int, int)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html" title="class in com.compomics.util.gui.spectrum">IonLabelColorTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html#getValueAt-int-int-">getValueAt(int, int)</a></span> - Method in class com.compomics.util.gui.variants.aa_substitutions.<a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html" title="class in com.compomics.util.gui.variants.aa_substitutions">AaSubstitutionMatrixTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#getValueAt-int-int-">getValueAt(int, int)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>
<div class="block">Returns the value of the matrix at given indexes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/roc/DataRoc.html#getValueAt-double-">getValueAt(double)</a></span> - Method in class com.compomics.util.math.roc.<a href="com/compomics/util/math/roc/DataRoc.html" title="class in com.compomics.util.math.roc">DataRoc</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/roc/DistributionRoc.html#getValueAt-double-">getValueAt(double)</a></span> - Method in class com.compomics.util.math.roc.<a href="com/compomics/util/math/roc/DistributionRoc.html" title="class in com.compomics.util.math.roc">DistributionRoc</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/ROC.html#getValueAt-double-">getValueAt(double)</a></span> - Method in interface com.compomics.util.math.statistics.<a href="com/compomics/util/math/statistics/ROC.html" title="interface in com.compomics.util.math.statistics">ROC</a></dt>
<dd>
<div class="block">Returns the sensitivity at a given 1-specificity, i.e., 1-type 2 error,
 the number of true healthy for a given type 1 error, the number of false
 healthy.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableMap.html#getValueAt-int-int-">getValueAt(int, int)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun">TableMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#getValueAt-int-int-">getValueAt(int, int)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/Distribution.html#getValueAtCumulativeProbability-double-">getValueAtCumulativeProbability(double)</a></span> - Method in interface com.compomics.util.math.statistics.<a href="com/compomics/util/math/statistics/Distribution.html" title="interface in com.compomics.util.math.statistics">Distribution</a></dt>
<dd>
<div class="block">The value after which the cumulative density function will be smaller
 than p.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html#getValueAtCumulativeProbability-double-">getValueAtCumulativeProbability(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html" title="class in com.compomics.util.math.statistics.distributions">BinomialDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getValueAtCumulativeProbability-double-">getValueAtCumulativeProbability(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html#getValueAtCumulativeProbability-double-">getValueAtCumulativeProbability(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/Distribution.html#getValueAtDescendingCumulativeProbability-double-">getValueAtDescendingCumulativeProbability(double)</a></span> - Method in interface com.compomics.util.math.statistics.<a href="com/compomics/util/math/statistics/Distribution.html" title="interface in com.compomics.util.math.statistics">Distribution</a></dt>
<dd>
<div class="block">The value after which the cumulative density function will be smaller
 than p when starting from high values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html#getValueAtDescendingCumulativeProbability-double-">getValueAtDescendingCumulativeProbability(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html" title="class in com.compomics.util.math.statistics.distributions">BinomialDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#getValueAtDescendingCumulativeProbability-double-">getValueAtDescendingCumulativeProbability(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html#getValueAtDescendingCumulativeProbability-double-">getValueAtDescendingCumulativeProbability(double)</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NormalDistribution</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns the van der Waals volume in 3.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Alanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Arginine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Asparagine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">AsparticAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">B</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Cysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">GlutamicAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glutamine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glycine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Histidine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Isoleucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">J</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Leucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Lysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Methionine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Phenylalanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Proline</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Pyrrolysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Selenocysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Serine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Threonine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tryptophan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tyrosine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Valine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">X</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html#getVanDerWaalsVolume--">getVanDerWaalsVolume()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Z</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#getVariableModifications--">getVariableModifications()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Getter for the modifications carried by this sequence in a map: aa number
 &gt; modification matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getVariableModifications--">getVariableModifications()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the variable modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html#getVariableModifications--">getVariableModifications()</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a></dt>
<dd>
<div class="block">Returns eventual modifications found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#getVariableModifications--">getVariableModifications()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Returns the expected variable modification names included in this
 profile.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#getVariableModificationsAsString-com.compomics.util.experiment.biology.proteins.Peptide-">getVariableModificationsAsString(Peptide)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns the peptide modifications as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#getVariantMatches--">getVariantMatches()</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns the sequence variant matches of this peptide indexed by protein
 accession and peptide start.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideVariantMatches.html#getVariantMatches--">getVariantMatches()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideVariantMatches.html" title="class in com.compomics.util.experiment.identification.matches">PeptideVariantMatches</a></dt>
<dd>
<div class="block">Returns the map of variants indexed by 0 based position on the peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html#getVariantMatches-java.util.ArrayList-">getVariantMatches(ArrayList&lt;PeptideProteinMapping&gt;)</a></span> - Static method in class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a></dt>
<dd>
<div class="block">Returns the variant matches summarized in a map indexed by protein
 accession and peptide index on the protein sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramDialog.html#getVennDiagramPanel--">getVennDiagramPanel()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramDialog.html" title="class in com.compomics.util.gui">VennDiagramDialog</a></dt>
<dd>
<div class="block">Returns the Venn diagram panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#getVerbosity--">getVerbosity()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Returns the verbosity type of progress output.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html#getVersion--">getVersion()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/FeaturesMap.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">FeaturesMap</a></dt>
<dd>
<div class="block">Returns the version of the features used to create the map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#getVersion--">getVersion()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Returns the database version.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#getVersion--">getVersion()</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Retrieves the version number set in the pom file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/MavenJarFile.html#getVersionNumber--">getVersionNumber()</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/MavenJarFile.html" title="class in com.compomics.software.autoupdater">MavenJarFile</a></dt>
<dd>
<div class="block">Returns the version number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#getViewIndex-int-">getViewIndex(int)</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Returns the view index of the given row.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VisibleTableColumnsDialogParent.html#getVisibleColumns--">getVisibleColumns()</a></span> - Method in interface com.compomics.util.gui.<a href="com/compomics/util/gui/VisibleTableColumnsDialogParent.html" title="interface in com.compomics.util.gui">VisibleTableColumnsDialogParent</a></dt>
<dd>
<div class="block">Returns the list of visible columns.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#getVisibleColumns--">getVisibleColumns()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramDialog.html#getWaitingIcon--">getWaitingIcon()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramDialog.html" title="class in com.compomics.util.gui">VennDiagramDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#getWaitingIcon--">getWaitingIcon()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#getWarning-java.lang.String-java.lang.String-">getWarning(String, String)</a></span> - Static method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Creates a warning object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#getWorkbook--">getWorkbook()</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>
<div class="block">Return the workBook.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#getWorkbookStyle--">getWorkbookStyle()</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>
<div class="block">Returns the workbook style.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/XYZDataPoint.html#getX--">getX()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/XYZDataPoint.html" title="class in com.compomics.util">XYZDataPoint</a></dt>
<dd>
<div class="block">Returns the x value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getXAxisData--">getXAxisData()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the x-axis data.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getXAxisZoomRangeLowerValue--">getXAxisZoomRangeLowerValue()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the lower range for the current zoom range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getXAxisZoomRangeUpperValue--">getXAxisZoomRangeUpperValue()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the upper range for the current zoom range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#getXTandemFormat--">getXTandemFormat()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Get the X!Tandem enzyme format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/roc/DataRoc.html#getxYValues--">getxYValues()</a></span> - Method in class com.compomics.util.math.roc.<a href="com/compomics/util/math/roc/DataRoc.html" title="class in com.compomics.util.math.roc">DataRoc</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/roc/DistributionRoc.html#getxYValues--">getxYValues()</a></span> - Method in class com.compomics.util.math.roc.<a href="com/compomics/util/math/roc/DistributionRoc.html" title="class in com.compomics.util.math.roc">DistributionRoc</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/ROC.html#getxYValues--">getxYValues()</a></span> - Method in interface com.compomics.util.math.statistics.<a href="com/compomics/util/math/statistics/ROC.html" title="interface in com.compomics.util.math.statistics">ROC</a></dt>
<dd>
<div class="block">Returns xy values to draw the curve.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/XYZDataPoint.html#getY--">getY()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/XYZDataPoint.html" title="class in com.compomics.util">XYZDataPoint</a></dt>
<dd>
<div class="block">Returns the y value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#getYAxisData--">getYAxisData()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns the y-axis data.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/XYZDataPoint.html#getZ--">getZ()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/XYZDataPoint.html" title="class in com.compomics.util">XYZDataPoint</a></dt>
<dd>
<div class="block">Returns the z value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#gif">gif</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#GIF">GIF</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#GLUTAMIC_ACID">GLUTAMIC_ACID</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Glutamic acid immonium ion.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">GlutamicAcid</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Glutamic Acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html#GlutamicAcid--">GlutamicAcid()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">GlutamicAcid</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Glutamine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Glutamine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html#Glutamine--">Glutamine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glutamine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#GLUTAMINE">GLUTAMINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Glutamine immonium ion.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl"><span class="typeNameLink">Glycan</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/impl/package-summary.html">com.compomics.util.experiment.biology.ions.impl</a></dt>
<dd>
<div class="block">This class represents a glycan.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#Glycan-java.lang.String-java.lang.String-">Glycan(String, String)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>
<div class="block">Constructor for a glycan.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Glycine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Glycine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html#Glycine--">Glycine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glycine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#GLYCINE">GLYCINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Glycine immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#GO_MAPPING_FILE_SUFFIX">GO_MAPPING_FILE_SUFFIX</a></span> - Static variable in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">The suffix to use for files containing GO mappings.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html" title="class in com.compomics.util.experiment.biology.genes.go"><span class="typeNameLink">GoDomains</span></a> - Class in <a href="com/compomics/util/experiment/biology/genes/go/package-summary.html">com.compomics.util.experiment.biology.genes.go</a></dt>
<dd>
<div class="block">Class parsing go domains and storing them in a map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html#GoDomains--">GoDomains()</a></span> - Constructor for class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html" title="class in com.compomics.util.experiment.biology.genes.go">GoDomains</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html" title="class in com.compomics.util.experiment.biology.genes.go"><span class="typeNameLink">GoMapping</span></a> - Class in <a href="com/compomics/util/experiment/biology/genes/go/package-summary.html">com.compomics.util.experiment.biology.genes.go</a></dt>
<dd>
<div class="block">Class parsing BioMart protein go mappings and storing them in maps.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html#GoMapping--">GoMapping()</a></span> - Constructor for class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html" title="class in com.compomics.util.experiment.biology.genes.go">GoMapping</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum"><span class="typeNameLink">GraphicsPanel</span></a> - Class in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">This class presents a JPanel that will hold and display a mass spectrum or a
 chromatogram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#GraphicsPanel--">GraphicsPanel()</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/spectrum/GraphicsPanel.DrawingStyle.html" title="enum in com.compomics.util.gui.spectrum"><span class="typeNameLink">GraphicsPanel.DrawingStyle</span></a> - Enum in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">An enumerator of the possible GraphicsPanel types.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/GraphicsPanel.GraphicsPanelType.html" title="enum in com.compomics.util.gui.spectrum"><span class="typeNameLink">GraphicsPanel.GraphicsPanelType</span></a> - Enum in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">An enumerator of the possible GraphicsPanel types.</div>
</dd>
<dt><a href="com/compomics/util/gui/renderers/GraphicsPanelTableCellRenderer.html" title="class in com.compomics.util.gui.renderers"><span class="typeNameLink">GraphicsPanelTableCellRenderer</span></a> - Class in <a href="com/compomics/util/gui/renderers/package-summary.html">com.compomics.util.gui.renderers</a></dt>
<dd>
<div class="block">A table cell renderer for GraphicsPanels.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/GraphicsPanelTableCellRenderer.html#GraphicsPanelTableCellRenderer-double-double-">GraphicsPanelTableCellRenderer(double, double)</a></span> - Constructor for class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/GraphicsPanelTableCellRenderer.html" title="class in com.compomics.util.gui.renderers">GraphicsPanelTableCellRenderer</a></dt>
<dd>
<div class="block">Create a new GraphicsPanelTableCellRenderer with a given x-axis range
 to ensure that all spectra in the column have the same range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#gson">gson</a></span> - Variable in class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">GSON parser instance to convert JSON to Java objects and back.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFormat.html#guiDefaultOption">guiDefaultOption</a></span> - Static variable in enum com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFormat.html" title="enum in com.compomics.util.io.export">ExportFormat</a></dt>
<dd>
<div class="block">The default export format to use for GUI exports.</div>
</dd>
<dt><a href="com/compomics/software/autoupdater/GUIFileDAO.html" title="class in com.compomics.software.autoupdater"><span class="typeNameLink">GUIFileDAO</span></a> - Class in <a href="com/compomics/software/autoupdater/package-summary.html">com.compomics.software.autoupdater</a></dt>
<dd>
<div class="block">GUIFileDAO.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/GUIFileDAO.html#GUIFileDAO--">GUIFileDAO()</a></span> - Constructor for class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/GUIFileDAO.html" title="class in com.compomics.software.autoupdater">GUIFileDAO</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/GuiUtilities.html" title="class in com.compomics.util.gui"><span class="typeNameLink">GuiUtilities</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">A class containing simple GUI helper methods.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/GuiUtilities.html#GuiUtilities--">GuiUtilities()</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/GuiUtilities.html" title="class in com.compomics.util.gui">GuiUtilities</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="I:H">
<!--   -->
</a>
<h2 class="title">H</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#H">H</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#H">H</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The hydrogen atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#H2O">H2O</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">H2O loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#H3PO4">H3PO4</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">H3PO4 loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/memory/MemoryConsumptionStatus.html#halfGbFree--">halfGbFree()</a></span> - Static method in class com.compomics.util.memory.<a href="com/compomics/util/memory/MemoryConsumptionStatus.html" title="class in com.compomics.util.memory">MemoryConsumptionStatus</a></dt>
<dd>
<div class="block">Indicates whether a GB of memory is free.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#hasAddenda--">hasAddenda()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method reports on the presence of addenda for this header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIParams.html#hasArgument">hasArgument</a></span> - Variable in enum com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIParams.html" title="enum in com.compomics.cli.enzymes">EnzymesCLIParams</a></dt>
<dd>
<div class="block">Boolean indicating whether this command line option needs an argument.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/fasta/FastaParametersCLIParams.html#hasArgument">hasArgument</a></span> - Variable in enum com.compomics.cli.fasta.<a href="com/compomics/cli/fasta/FastaParametersCLIParams.html" title="enum in com.compomics.cli.fasta">FastaParametersCLIParams</a></dt>
<dd>
<div class="block">Boolean indicating whether this command line option needs an argument.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html#hasArgument">hasArgument</a></span> - Variable in enum com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html" title="enum in com.compomics.cli.identification_parameters">IdentificationParametersCLIParams</a></dt>
<dd>
<div class="block">Boolean indicating whether this command line option needs an argument.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIParams.html#hasArgument">hasArgument</a></span> - Variable in enum com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIParams.html" title="enum in com.compomics.cli.modifications">ModificationsCLIParams</a></dt>
<dd>
<div class="block">Boolean indicating whether this command line option needs an argument.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/CommandLineParser.html#hasArguments--">hasArguments()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/CommandLineParser.html" title="class in com.compomics.util.general">CommandLineParser</a></dt>
<dd>
<div class="block">This method returns 'true' if any arguments are present, 'false'
 otherwise.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#hasCombination-java.lang.String-">hasCombination(String)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Indicates whether the given sequence contains an amino acid which is in
 fact a combination of amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#hasCombination-java.util.stream.IntStream-">hasCombination(IntStream)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Indicates whether the given sequence contains an amino acid which is in
 fact a combination of amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#hasCombination-char:A-">hasCombination(char[])</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Indicates whether the given sequence contains an amino acid which is in
 fact a combination of amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/IdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in interface com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/IdfileReader.html" title="interface in com.compomics.util.experiment.io.identification">IdfileReader</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the file contains de novo results as tags.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/AndromedaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">AndromedaIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MascotIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MsAmandaIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MzIdentMLIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">NovorIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepxmlIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PNovoIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">TideIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html#hasDeNovoTags--">hasDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">XTandemIdfileReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#hasEnzymaticPeptides-long-">hasEnzymaticPeptides(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Returns true if the leading protein of the given group has any enzymatic
 peptides.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#hasEnzymes--">hasEnzymes()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether enzyme settings were set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/CommandLineParser.html#hasFlag-java.lang.String-">hasFlag(String)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/CommandLineParser.html" title="class in com.compomics.util.general">CommandLineParser</a></dt>
<dd>
<div class="block">This method test whether the specified flag was set on the command line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#hasGoMappings--">hasGoMappings()</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Indicates whether the GO maps are populated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#hashCode--">hashCode()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#hashCode--">hashCode()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#hashCode--">hashCode()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">Returns a hash code value for the object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#hashCode--">hashCode()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">Returns a hash code value for the object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterDialog.html#hasInput--">hasInput()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterDialog.html" title="class in com.compomics.util.gui.filtering">FilterDialog</a></dt>
<dd>
<div class="block">Indicates whether something was input for the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#hasModificationIndexes--">hasModificationIndexes()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Indicates whether the modification profile has PTM indexes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#hasModificationIndexes--">hasModificationIndexes()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Indicates whether the modification profile has PTM indexes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#hasNeutralLosses--">hasNeutralLosses()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Indicates whether the ion has a neutral loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfFileIterator.html#hasNext--">hasNext()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfFileIterator.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfFileIterator</a></dt>
<dd>
<div class="block">Indicates whether the file contains another spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspFileIterator.html#hasNext--">hasNext()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspFileIterator.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspFileIterator</a></dt>
<dd>
<div class="block">Indicates whether the file contains another spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#hasNormalizationFactors--">hasNormalizationFactors()</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Indicates whether normalization factors are set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#hasPeptideAssumption--">hasPeptideAssumption()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Indicates whether the spectrum match contains a peptide assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#hasPeptideAssumption-int-">hasPeptideAssumption(int)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Indicates whether the spectrum match contains a peptide assumption for
 the given advocate (see the Advocate class).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#hasPeptideNormalisationFactors--">hasPeptideNormalisationFactors()</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Indicates whether the peptide normalization factors are set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#hasProteinNormalisationFactors--">hasProteinNormalisationFactors()</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Indicates whether the protein normalization factors are set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#hasPsmNormalisationFactors--">hasPsmNormalisationFactors()</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Indicates whether the PSM normalization factors are set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#hasQcFilters--">hasQcFilters()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Indicates whether QC filters were implemented for this match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#hasRTWindow--">hasRTWindow()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the retention time window was
 implemented.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#hasTagAssumption--">hasTagAssumption()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Indicates whether the spectrum match contains a tag assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#hasTagAssumption-int-">hasTagAssumption(int)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Indicates whether the spectrum match contains a tag assumption for
 the given advocate (see the Advocate class).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#He">He</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The helium atom.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein"><span class="typeNameLink">Header</span></a> - Class in <a href="com/compomics/util/experiment/io/biology/protein/package-summary.html">com.compomics.util.experiment.io.biology.protein</a></dt>
<dd>
<div class="block">This class represents the header for a Protein instance.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/iterators/HeaderIterator.html" title="class in com.compomics.util.experiment.io.biology.protein.iterators"><span class="typeNameLink">HeaderIterator</span></a> - Class in <a href="com/compomics/util/experiment/io/biology/protein/iterators/package-summary.html">com.compomics.util.experiment.io.biology.protein.iterators</a></dt>
<dd>
<div class="block">Iterator for the headers of a fasta file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/iterators/HeaderIterator.html#HeaderIterator-java.io.File-">HeaderIterator(File)</a></span> - Constructor for class com.compomics.util.experiment.io.biology.protein.iterators.<a href="com/compomics/util/experiment/io/biology/protein/iterators/HeaderIterator.html" title="class in com.compomics.util.experiment.io.biology.protein.iterators">HeaderIterator</a></dt>
<dd>
<div class="block">Constructor without sanity check.</div>
</dd>
<dt><a href="com/compomics/software/autoupdater/HeadlessFileDAO.html" title="class in com.compomics.software.autoupdater"><span class="typeNameLink">HeadlessFileDAO</span></a> - Class in <a href="com/compomics/software/autoupdater/package-summary.html">com.compomics.software.autoupdater</a></dt>
<dd>
<div class="block">HeadlessFileDAO.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/HeadlessFileDAO.html#HeadlessFileDAO--">HeadlessFileDAO()</a></span> - Constructor for class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/HeadlessFileDAO.html" title="class in com.compomics.software.autoupdater">HeadlessFileDAO</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/Helium.html" title="class in com.compomics.util.experiment.biology.atoms.impl"><span class="typeNameLink">Helium</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a></dt>
<dd>
<div class="block">The helium atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/impl/Helium.html#Helium--">Helium()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.impl.<a href="com/compomics/util/experiment/biology/atoms/impl/Helium.html" title="class in com.compomics.util.experiment.biology.atoms.impl">Helium</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/error_handlers/HelpDialog.html" title="class in com.compomics.util.gui.error_handlers"><span class="typeNameLink">HelpDialog</span></a> - Class in <a href="com/compomics/util/gui/error_handlers/package-summary.html">com.compomics.util.gui.error_handlers</a></dt>
<dd>
<div class="block">A dialog used to display help text in HTML format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/HelpDialog.html#HelpDialog-java.awt.Frame-java.net.URL-java.awt.Image-java.awt.Image-java.lang.String-">HelpDialog(Frame, URL, Image, Image, String)</a></span> - Constructor for class com.compomics.util.gui.error_handlers.<a href="com/compomics/util/gui/error_handlers/HelpDialog.html" title="class in com.compomics.util.gui.error_handlers">HelpDialog</a></dt>
<dd>
<div class="block">Creates a new HelpDialog with a JFrame as a parent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/HelpDialog.html#HelpDialog-java.awt.Frame-java.net.URL-java.awt.Image-java.awt.Image-java.lang.String-int-int-">HelpDialog(Frame, URL, Image, Image, String, int, int)</a></span> - Constructor for class com.compomics.util.gui.error_handlers.<a href="com/compomics/util/gui/error_handlers/HelpDialog.html" title="class in com.compomics.util.gui.error_handlers">HelpDialog</a></dt>
<dd>
<div class="block">Creates a new HelpDialog with a JFrame as a parent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/HelpDialog.html#HelpDialog-java.awt.Frame-java.net.URL-java.lang.String-java.awt.Image-java.awt.Image-java.lang.String-">HelpDialog(Frame, URL, String, Image, Image, String)</a></span> - Constructor for class com.compomics.util.gui.error_handlers.<a href="com/compomics/util/gui/error_handlers/HelpDialog.html" title="class in com.compomics.util.gui.error_handlers">HelpDialog</a></dt>
<dd>
<div class="block">Creates a new HelpDialog with a JFrame as a parent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/HelpDialog.html#HelpDialog-java.awt.Frame-java.net.URL-java.lang.String-java.awt.Image-java.awt.Image-java.lang.String-int-int-">HelpDialog(Frame, URL, String, Image, Image, String, int, int)</a></span> - Constructor for class com.compomics.util.gui.error_handlers.<a href="com/compomics/util/gui/error_handlers/HelpDialog.html" title="class in com.compomics.util.gui.error_handlers">HelpDialog</a></dt>
<dd>
<div class="block">Creates a new HelpDialog with a JFrame as a parent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/HelpDialog.html#HelpDialog-javax.swing.JDialog-java.net.URL-java.awt.Image-java.awt.Image-java.lang.String-">HelpDialog(JDialog, URL, Image, Image, String)</a></span> - Constructor for class com.compomics.util.gui.error_handlers.<a href="com/compomics/util/gui/error_handlers/HelpDialog.html" title="class in com.compomics.util.gui.error_handlers">HelpDialog</a></dt>
<dd>
<div class="block">Creates a new HelpDialog object with a JDialog as a parent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/HelpDialog.html#HelpDialog-javax.swing.JDialog-java.net.URL-java.awt.Image-java.awt.Image-java.lang.String-int-int-">HelpDialog(JDialog, URL, Image, Image, String, int, int)</a></span> - Constructor for class com.compomics.util.gui.error_handlers.<a href="com/compomics/util/gui/error_handlers/HelpDialog.html" title="class in com.compomics.util.gui.error_handlers">HelpDialog</a></dt>
<dd>
<div class="block">Creates a new HelpDialog object with a JDialog as a parent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/HelpDialog.html#HelpDialog-javax.swing.JDialog-java.net.URL-java.lang.String-java.awt.Image-java.awt.Image-java.lang.String-">HelpDialog(JDialog, URL, String, Image, Image, String)</a></span> - Constructor for class com.compomics.util.gui.error_handlers.<a href="com/compomics/util/gui/error_handlers/HelpDialog.html" title="class in com.compomics.util.gui.error_handlers">HelpDialog</a></dt>
<dd>
<div class="block">Creates a new HelpDialog with a JDialog as a parent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/HelpDialog.html#HelpDialog-javax.swing.JDialog-java.net.URL-java.lang.String-java.awt.Image-java.awt.Image-java.lang.String-int-int-">HelpDialog(JDialog, URL, String, Image, Image, String, int, int)</a></span> - Constructor for class com.compomics.util.gui.error_handlers.<a href="com/compomics/util/gui/error_handlers/HelpDialog.html" title="class in com.compomics.util.gui.error_handlers">HelpDialog</a></dt>
<dd>
<div class="block">Creates a new HelpDialog with a JDialog as a parent.</div>
</dd>
<dt><a href="com/compomics/util/examples/HelpWindow.html" title="class in com.compomics.util.examples"><span class="typeNameLink">HelpWindow</span></a> - Class in <a href="com/compomics/util/examples/package-summary.html">com.compomics.util.examples</a></dt>
<dd>
<div class="block">A window used to display help text in HTML format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/examples/HelpWindow.html#HelpWindow-javax.swing.JFrame-java.net.URL-">HelpWindow(JFrame, URL)</a></span> - Constructor for class com.compomics.util.examples.<a href="com/compomics/util/examples/HelpWindow.html" title="class in com.compomics.util.examples">HelpWindow</a></dt>
<dd>
<div class="block">Creates a new HelpWindow object with a Frame as a parent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/examples/HelpWindow.html#HelpWindow-javax.swing.JDialog-java.net.URL-">HelpWindow(JDialog, URL)</a></span> - Constructor for class com.compomics.util.examples.<a href="com/compomics/util/examples/HelpWindow.html" title="class in com.compomics.util.examples">HelpWindow</a></dt>
<dd>
<div class="block">Creates a new HelpWindow-object with a JDialog as a parent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#highLight-int-int-java.awt.Graphics-java.awt.Color-java.lang.String-int-boolean-int-boolean-">highLight(int, int, Graphics, Color, String, int, boolean, int, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method will highlight the specified point in the specified color by
 drawing a floating triangle + x-value above it.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#highlightClicked-int-int-java.awt.Graphics-boolean-">highlightClicked(int, int, Graphics, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method will draw a highlighting triangle + x-value on top of the
 clicked marked point.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#highLightPeak-int-int-java.awt.Graphics-boolean-">highLightPeak(int, int, Graphics, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method will draw a highlighting triangle + x-value on top of the
 marked point.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Histidine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Histidine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html#Histidine--">Histidine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Histidine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#HISTIDINE">HISTIDINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Histidine immonium ion.</div>
</dd>
<dt><a href="com/compomics/util/math/HistogramUtils.html" title="class in com.compomics.util.math"><span class="typeNameLink">HistogramUtils</span></a> - Class in <a href="com/compomics/util/math/package-summary.html">com.compomics.util.math</a></dt>
<dd>
<div class="block">Simple functions to manipulate histograms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/HistogramUtils.html#HistogramUtils--">HistogramUtils()</a></span> - Constructor for class com.compomics.util.math.<a href="com/compomics/util/math/HistogramUtils.html" title="class in com.compomics.util.math">HistogramUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#HOST">HOST</a></span> - Static variable in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#HPO3">HPO3</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">H3PO3 loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.HuffmanNode.html#HuffmanNode-int-int-">HuffmanNode(int, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.HuffmanNode.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree.HuffmanNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.HuffmanNode.html#HuffmanNode-com.compomics.util.experiment.identification.protein_inference.fm_index.WaveletTree.HuffmanNode-com.compomics.util.experiment.identification.protein_inference.fm_index.WaveletTree.HuffmanNode-">HuffmanNode(WaveletTree.HuffmanNode, WaveletTree.HuffmanNode)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.HuffmanNode.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree.HuffmanNode</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/Hydrogen.html" title="class in com.compomics.util.experiment.biology.atoms.impl"><span class="typeNameLink">Hydrogen</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a></dt>
<dd>
<div class="block">The hydrogen atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/impl/Hydrogen.html#Hydrogen--">Hydrogen()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.impl.<a href="com/compomics/util/experiment/biology/atoms/impl/Hydrogen.html" title="class in com.compomics.util.experiment.biology.atoms.impl">Hydrogen</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores"><span class="typeNameLink">HyperScore</span></a> - Class in <a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/package-summary.html">com.compomics.util.experiment.identification.psm_scoring.psm_scores</a></dt>
<dd>
<div class="block">Hyperscore as variation of the score implemented in X!Tandem
 www.thegpm.org/tandem.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#HyperScore-com.compomics.util.experiment.identification.peptide_fragmentation.PeptideFragmentationModel-">HyperScore(PeptideFragmentationModel)</a></span> - Constructor for class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html#HyperScore--">HyperScore()</a></span> - Constructor for class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/HyperScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">HyperScore</a></dt>
<dd>
<div class="block">Constructor using a uniform fragmentation.</div>
</dd>
</dl>
<a name="I:I">
<!--   -->
</a>
<h2 class="title">I</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#I">I</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#I">I</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The Iodine atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iAnnotations">iAnnotations</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The Vector that holds a set of Annotation instances.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iAnnotationsMirroredSpectra">iAnnotationsMirroredSpectra</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The Vector that holds a set of Annotation instances for the mirrored
 spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iAnnotationYAxisThreshold">iAnnotationYAxisThreshold</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The percent non-inclusive, minimal y-axis value (compared to the highest
 point in the spectrum) a point should have before being eligible for
 annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iAreaUnderCurveColor">iAreaUnderCurveColor</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Colors in which the chromatogram polyline is rendered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iAreaUnderCurveColorMirroredSpectra">iAreaUnderCurveColorMirroredSpectra</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Colors in which the chromatogram polyline is rendered for the mirrored
 spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#iArtifact">iArtifact</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">Boolean to indicate whether this modification is an artifact.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#iBase64String">iBase64String</a></span> - Variable in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Represents the binary contents of the array as an array of bytes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#iCleavables">iCleavables</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">Lookup cache for the cleavable residues.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#iCleavage">iCleavage</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This char[] holds the residues after which cleavage will occur.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iClicked">iClicked</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Boolean that indicates whether a point has been marked by clicking.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iClickedDataSetIndex">iClickedDataSetIndex</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Int that indicates which dataset contains the clicked point.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iClickedDataSetIndexMirrored">iClickedDataSetIndexMirrored</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Int that indicates which dataset contains the clicked point in the
 mirrored spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iClickedIndex">iClickedIndex</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Int that indicates which point was clicked.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iClickedIndexMirrored">iClickedIndexMirrored</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Int that indicates which point was clicked in the mirrored spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iClickedList">iClickedList</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The Vector that holds all points clicked up to now.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iClickedListDatasetIndices">iClickedListDatasetIndices</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The Vector that holds the dataset indices of all points clicked up to
 now.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iClickedListDatasetIndicesMirrored">iClickedListDatasetIndicesMirrored</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The Vector that holds the dataset indices of all points clicked up to now
 in the mirrored spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iClickedListMirrored">iClickedListMirrored</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The Vector that holds all points clicked up to now in the mirrored
 spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iClickedMirrored">iClickedMirrored</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Boolean that indicates whether a point has been marked by clicking in the
 mirrored spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#iCode">iCode</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">The code for this modification (eg., Mox).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iCurrentDrawStyle">iCurrentDrawStyle</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This variable holds the current drawing style.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIParams.html#id">id</a></span> - Variable in enum com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIParams.html" title="enum in com.compomics.cli.enzymes">EnzymesCLIParams</a></dt>
<dd>
<div class="block">Short Id for the CLI parameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/fasta/FastaParametersCLIParams.html#id">id</a></span> - Variable in enum com.compomics.cli.fasta.<a href="com/compomics/cli/fasta/FastaParametersCLIParams.html" title="enum in com.compomics.cli.fasta">FastaParametersCLIParams</a></dt>
<dd>
<div class="block">Short Id for the CLI parameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html#id">id</a></span> - Variable in enum com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html" title="enum in com.compomics.cli.identification_parameters">IdentificationParametersCLIParams</a></dt>
<dd>
<div class="block">Short Id for the CLI parameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIParams.html#id">id</a></span> - Variable in enum com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIParams.html" title="enum in com.compomics.cli.modifications">ModificationsCLIParams</a></dt>
<dd>
<div class="block">Short Id for the CLI parameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/searchgui/OutputParameters.html#id">id</a></span> - Variable in enum com.compomics.util.parameters.searchgui.<a href="com/compomics/util/parameters/searchgui/OutputParameters.html" title="enum in com.compomics.util.parameters.searchgui">OutputParameters</a></dt>
<dd>
<div class="block">The index of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#iDataEndian">iDataEndian</a></span> - Variable in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Represents&nbsp;the endian value of the binary array (mzData element
 .../data/endian).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#iDataLength">iDataLength</a></span> - Variable in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Represents the length of the binary array (mzData element
 .../data/length).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iDataPointAndLineColor">iDataPointAndLineColor</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Colors in which the data points and peaks are rendered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iDataPointAndLineColorMirroredSpectra">iDataPointAndLineColorMirroredSpectra</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Colors in which the data points and peaks are rendered for the mirrored
 spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#iDataPrecision">iDataPrecision</a></span> - Variable in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Represents the precision of the binary array (mzData element
 .../data/precision).</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification"><span class="typeNameLink">Identification</span></a> - Class in <a href="com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a></dt>
<dd>
<div class="block">This class interacts with the back-end database to manage identification
 objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#Identification-com.compomics.util.db.object.ObjectsDB-">Identification(ObjectsDB)</a></span> - Constructor for class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/IdentificationAlgorithmParameter.html" title="interface in com.compomics.util.gui.parameters.identification"><span class="typeNameLink">IdentificationAlgorithmParameter</span></a> - Interface in <a href="com/compomics/util/gui/parameters/identification/package-summary.html">com.compomics.util.gui.parameters.identification</a></dt>
<dd>
<div class="block">Interface for the algorithm specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#identificationCharge">identificationCharge</a></span> - Variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">The charge used for identification.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification"><span class="typeNameLink">IdentificationFeaturesCache</span></a> - Class in <a href="com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a></dt>
<dd>
<div class="block">This class caches the identification features calculated by the
 IdentificationFeaturesGenerator for later reuse.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#IdentificationFeaturesCache--">IdentificationFeaturesCache()</a></span> - Constructor for class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.ObjectType.html" title="enum in com.compomics.util.experiment.identification"><span class="typeNameLink">IdentificationFeaturesCache.ObjectType</span></a> - Enum in <a href="com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a></dt>
<dd>
<div class="block">An enumerator of the supported object types.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification"><span class="typeNameLink">IdentificationFeaturesGenerator</span></a> - Class in <a href="com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a></dt>
<dd>
<div class="block">This class provides identification features and stores them in cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#IdentificationFeaturesGenerator-com.compomics.util.experiment.identification.Identification-com.compomics.util.parameters.identification.IdentificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.experiment.identification.peptide_shaker.Metrics-com.compomics.util.parameters.quantification.spectrum_counting.SpectrumCountingParameters-">IdentificationFeaturesGenerator(Identification, IdentificationParameters, SequenceProvider, Metrics, SpectrumCountingParameters)</a></span> - Constructor for class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#identificationFile">identificationFile</a></span> - Variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">The identification file.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/IdentificationMatch.html" title="class in com.compomics.util.experiment.identification"><span class="typeNameLink">IdentificationMatch</span></a> - Class in <a href="com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a></dt>
<dd>
<div class="block">This is an abstract class for an identification match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationMatch.html#IdentificationMatch--">IdentificationMatch()</a></span> - Constructor for class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationMatch.html" title="class in com.compomics.util.experiment.identification">IdentificationMatch</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/IdentificationMatch.MatchType.html" title="enum in com.compomics.util.experiment.identification"><span class="typeNameLink">IdentificationMatch.MatchType</span></a> - Enum in <a href="com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a></dt>
<dd>
<div class="block">The type of match.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/IdentificationMethod.html" title="class in com.compomics.util.experiment.identification"><span class="typeNameLink">IdentificationMethod</span></a> - Class in <a href="com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a></dt>
<dd>
<div class="block">This class will contain all methods used to obtain identifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationMethod.html#IdentificationMethod-int-">IdentificationMethod(int)</a></span> - Constructor for class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationMethod.html" title="class in com.compomics.util.experiment.identification">IdentificationMethod</a></dt>
<dd>
<div class="block">Constructor for the identification method</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification"><span class="typeNameLink">IdentificationParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/package-summary.html">com.compomics.util.parameters.identification</a></dt>
<dd>
<div class="block">Generic class grouping the parameters used for protein identification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#IdentificationParameters--">IdentificationParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Creates empty identification parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#IdentificationParameters-com.compomics.util.parameters.identification.search.SearchParameters-">IdentificationParameters(SearchParameters)</a></span> - Constructor for class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Creates default identification parameters from the given search
 parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#IdentificationParameters-java.lang.String-java.lang.String-com.compomics.util.parameters.identification.search.SearchParameters-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters-com.compomics.util.parameters.identification.advanced.GeneParameters-com.compomics.util.parameters.identification.advanced.PsmScoringParameters-com.compomics.util.experiment.identification.filtering.PeptideAssumptionFilter-com.compomics.util.parameters.identification.advanced.ModificationLocalizationParameters-com.compomics.util.parameters.identification.advanced.ProteinInferenceParameters-com.compomics.util.parameters.identification.advanced.IdMatchValidationParameters-com.compomics.util.parameters.identification.advanced.FractionParameters-">IdentificationParameters(String, String, SearchParameters, AnnotationParameters, SequenceMatchingParameters, PeptideVariantsParameters, GeneParameters, PsmScoringParameters, PeptideAssumptionFilter, ModificationLocalizationParameters, ProteinInferenceParameters, IdMatchValidationParameters, FractionParameters)</a></span> - Constructor for class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLI.html" title="class in com.compomics.cli.identification_parameters"><span class="typeNameLink">IdentificationParametersCLI</span></a> - Class in <a href="com/compomics/cli/identification_parameters/package-summary.html">com.compomics.cli.identification_parameters</a></dt>
<dd>
<div class="block">The SearchParametersCLI allows creating search parameters files using command
 line arguments.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLI.html#IdentificationParametersCLI-java.lang.String:A-">IdentificationParametersCLI(String[])</a></span> - Constructor for class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLI.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersCLI</a></dt>
<dd>
<div class="block">Construct a new SearchParametersCLI runnable from a list of arguments.</div>
</dd>
<dt><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html" title="enum in com.compomics.cli.identification_parameters"><span class="typeNameLink">IdentificationParametersCLIParams</span></a> - Enum in <a href="com/compomics/cli/identification_parameters/package-summary.html">com.compomics.cli.identification_parameters</a></dt>
<dd>
<div class="block">Enum class specifying the SearchParameter command line option parameters to
 create a SearchParameters object</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html" title="class in com.compomics.util.gui.parameters.identification"><span class="typeNameLink">IdentificationParametersEditionDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/package-summary.html">com.compomics.util.gui.parameters.identification</a></dt>
<dd>
<div class="block">IdentificationParametersEditionDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html#IdentificationParametersEditionDialog-java.awt.Frame-com.compomics.util.parameters.identification.IdentificationParameters-java.awt.Image-java.awt.Image-com.compomics.util.io.file.LastSelectedFolder-boolean-">IdentificationParametersEditionDialog(Frame, IdentificationParameters, Image, Image, LastSelectedFolder, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersEditionDialog</a></dt>
<dd>
<div class="block">Creates a new IdentificationParametersEditionDialog with a frame as
 owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html#IdentificationParametersEditionDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.IdentificationParameters-java.awt.Image-java.awt.Image-com.compomics.util.io.file.LastSelectedFolder-boolean-">IdentificationParametersEditionDialog(Dialog, Frame, IdentificationParameters, Image, Image, LastSelectedFolder, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersEditionDialog</a></dt>
<dd>
<div class="block">Creates a new IdentificationParametersEditionDialog with a dialog as
 owner.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html" title="class in com.compomics.util.experiment.identification.identification_parameters"><span class="typeNameLink">IdentificationParametersFactory</span></a> - Class in <a href="com/compomics/util/experiment/identification/identification_parameters/package-summary.html">com.compomics.util.experiment.identification.identification_parameters</a></dt>
<dd>
<div class="block">Factory for the identification parameters.</div>
</dd>
<dt><a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html" title="class in com.compomics.cli.identification_parameters"><span class="typeNameLink">IdentificationParametersInputBean</span></a> - Class in <a href="com/compomics/cli/identification_parameters/package-summary.html">com.compomics.cli.identification_parameters</a></dt>
<dd>
<div class="block">This class parses the parameters from a command line and stores them in a
 SearchParameters object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html#IdentificationParametersInputBean-org.apache.commons.cli.CommandLine-">IdentificationParametersInputBean(CommandLine)</a></span> - Constructor for class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersInputBean</a></dt>
<dd>
<div class="block">Takes all the arguments from a command line.</div>
</dd>
<dt><a href="com/compomics/util/io/json/marshallers/IdentificationParametersMarshaller.html" title="class in com.compomics.util.io.json.marshallers"><span class="typeNameLink">IdentificationParametersMarshaller</span></a> - Class in <a href="com/compomics/util/io/json/marshallers/package-summary.html">com.compomics.util.io.json.marshallers</a></dt>
<dd>
<div class="block">This class is a convenience class to have a DefaultJsonConverter with the
 identification parameters interfaces.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/marshallers/IdentificationParametersMarshaller.html#IdentificationParametersMarshaller--">IdentificationParametersMarshaller()</a></span> - Constructor for class com.compomics.util.io.json.marshallers.<a href="com/compomics/util/io/json/marshallers/IdentificationParametersMarshaller.html" title="class in com.compomics.util.io.json.marshallers">IdentificationParametersMarshaller</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersNameDialog.html" title="class in com.compomics.util.gui.parameters.identification"><span class="typeNameLink">IdentificationParametersNameDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/package-summary.html">com.compomics.util.gui.parameters.identification</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersNameDialog.html#IdentificationParametersNameDialog-java.awt.Frame-com.compomics.util.parameters.identification.IdentificationParameters-boolean-">IdentificationParametersNameDialog(Frame, IdentificationParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersNameDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersNameDialog</a></dt>
<dd>
<div class="block">Creates a new IdentificationParametersNameDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersNameDialog.html#IdentificationParametersNameDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.IdentificationParameters-boolean-">IdentificationParametersNameDialog(Dialog, Frame, IdentificationParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersNameDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersNameDialog</a></dt>
<dd>
<div class="block">Creates a new IdentificationParametersNameDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersOverviewDialog.html" title="class in com.compomics.util.gui.parameters.identification"><span class="typeNameLink">IdentificationParametersOverviewDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/package-summary.html">com.compomics.util.gui.parameters.identification</a></dt>
<dd>
<div class="block">IdentificationParametersOverviewDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersOverviewDialog.html#IdentificationParametersOverviewDialog-java.awt.Dialog-">IdentificationParametersOverviewDialog(Dialog)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersOverviewDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersOverviewDialog</a></dt>
<dd>
<div class="block">Constructor with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersOverviewDialog.html#IdentificationParametersOverviewDialog-java.awt.Frame-">IdentificationParametersOverviewDialog(Frame)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersOverviewDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersOverviewDialog</a></dt>
<dd>
<div class="block">Constructor with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#identiPy">identiPy</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The IdentiPy search engine.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/IdfileReader.html" title="interface in com.compomics.util.experiment.io.identification"><span class="typeNameLink">IdfileReader</span></a> - Interface in <a href="com/compomics/util/experiment/io/identification/package-summary.html">com.compomics.util.experiment.io.identification</a></dt>
<dd>
<div class="block">Generic interface for the parser of a file containing PSMs.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/IdfileReaderFactory.html" title="class in com.compomics.util.experiment.io.identification"><span class="typeNameLink">IdfileReaderFactory</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/package-summary.html">com.compomics.util.experiment.io.identification</a></dt>
<dd>
<div class="block">This factory will provide the appropriate identification file reader for each
 type of file.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced"><span class="typeNameLink">IdMatchValidationParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/advanced/package-summary.html">com.compomics.util.parameters.identification.advanced</a></dt>
<dd>
<div class="block">Generic class grouping the identification matches validation preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html#IdMatchValidationParameters--">IdMatchValidationParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">IdMatchValidationParameters</a></dt>
<dd>
<div class="block">Constructor for default settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html#IdMatchValidationParameters-com.compomics.util.parameters.identification.advanced.IdMatchValidationParameters-">IdMatchValidationParameters(IdMatchValidationParameters)</a></span> - Constructor for class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">IdMatchValidationParameters</a></dt>
<dd>
<div class="block">Creates a new IdMatchValidationPreferences based on the values of the
 given IdMatchValidationPreferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iDotRadius">iDotRadius</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This variable holds the dot radius; only used when drawing style is DOTS
 style.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iDragged">iDragged</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This boolean is set to 'true' when dragging is performed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iDragXLoc">iDragXLoc</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The current dragging location.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iEndXLoc">iEndXLoc</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">When the mouse is dragged, this represents the X-coordinate of the ending
 location.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iFilename">iFilename</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The spectrum or chromatogram filename.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iFilenameColor">iFilenameColor</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This is the color the filename should be presented in.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iFilenameMirrorredSpectrum">iFilenameMirrorredSpectrum</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The spectrum or chromatogram filename for the mirrored spectrum or
 chromatogram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/ExceptionHandler.html#ignoreExceptions">ignoreExceptions</a></span> - Variable in class com.compomics.util.exceptions.<a href="com/compomics/util/exceptions/ExceptionHandler.html" title="class in com.compomics.util.exceptions">ExceptionHandler</a></dt>
<dd>
<div class="block">If true exceptions will be ignored.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iHighLight">iHighLight</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Boolean that will be 'true' when a point needs highlighting.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iHighLightDatasetIndex">iHighLightDatasetIndex</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Index of the dataset containing the point that needs to be highlighted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iHighLightDatasetIndexMirrored">iHighLightDatasetIndexMirrored</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Index of the dataset containing the point that needs to be highlighted in
 the mirrored spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iHighLightIndex">iHighLightIndex</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Index of the point that needs to be highlighted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iHighLightIndexMirrored">iHighLightIndexMirrored</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Index of the point that needs to be highlighted in the mirrored spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iHighLightMirrored">iHighLightMirrored</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Boolean that will be 'true' when a point needs highlighting in the
 mirrored spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iKnownMassDeltas">iKnownMassDeltas</a></span> - Static variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This HashMap instance holds all the known mass deltas (if any).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#iLocation">iLocation</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">The location for the modification in the parent sequence.</div>
</dd>
<dt><a href="com/compomics/util/enumeration/ImageType.html" title="enum in com.compomics.util.enumeration"><span class="typeNameLink">ImageType</span></a> - Enum in <a href="com/compomics/util/enumeration/package-summary.html">com.compomics.util.enumeration</a></dt>
<dd>
<div class="block">Enumerator for supported export plot file types.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#iMassDeltas">iMassDeltas</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">This HashMap will contain the following 'key-value' mappings: (key &gt; value) <br>
 (residue &gt; double[]{MONOISOTOPIC_DELTA, AVERAGE_DELTA})</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iMinDrag">iMinDrag</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Minimal dragging distance in pixels.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#iMiscleavages">iMiscleavages</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This variable holds the number of supported missed cleavages.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl"><span class="typeNameLink">ImmoniumIon</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/impl/package-summary.html">com.compomics.util.experiment.biology.ions.impl</a></dt>
<dd>
<div class="block">Represents an immonium ion.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/ImmoniumIonAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators"><span class="typeNameLink">ImmoniumIonAnnotator</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators</a></dt>
<dd>
<div class="block">Annotator for immonium and related ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/ImmoniumIonAnnotator.html#ImmoniumIonAnnotator-char:A-">ImmoniumIonAnnotator(char[])</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/ImmoniumIonAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">ImmoniumIonAnnotator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/ImmoniumIonAnnotator.html#ImmoniumIonAnnotator-char:A-boolean-boolean-">ImmoniumIonAnnotator(char[], boolean, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/ImmoniumIonAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">ImmoniumIonAnnotator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/Ms2pipFeature.html#implementations">implementations</a></span> - Static variable in interface com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/Ms2pipFeature.html" title="interface in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">Ms2pipFeature</a></dt>
<dd>
<div class="block">Array of the implemented features.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#implementedForwardIons">implementedForwardIons</a></span> - Static variable in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Convenience array for forward ion type selection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#implementedRewindIons">implementedRewindIons</a></span> - Static variable in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Convenience array for rewind ion type selection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html#importFromFile-java.io.File-com.compomics.util.waiting.WaitingHandler-">importFromFile(File, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.biology.genes.ensembl.<a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html" title="class in com.compomics.util.experiment.biology.genes.ensembl">GeneMapping</a></dt>
<dd>
<div class="block">Reads go mappings from a BioMart file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#importMaps-com.compomics.util.experiment.biology.genes.ensembl.GeneMapping-">importMaps(GeneMapping)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Imports the gene maps from a gene Mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html#importMethods-java.io.File-">importMethods(File)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethodFactory</a></dt>
<dd>
<div class="block">Imports the methods from an XML file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iMSLevel">iMSLevel</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The ms level of the current spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#inCache-long-">inCache(long)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Check if key in cache</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#inCache-long-">inCache(long)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Indicates whether an object is loaded in cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.SelectedValuesTableFilter.html#include-javax.swing.RowFilter.Entry-">include(RowFilter.Entry&lt;? extends DefaultTableModel, ? extends Integer&gt;)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.SelectedValuesTableFilter.html" title="class in com.compomics.util.gui">XYPlottingDialog.SelectedValuesTableFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#increaseDepth--">increaseDepth()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Notifies the writer that data of a higher hierarchical depth will be
 written, e.g.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#increaseDepth--">increaseDepth()</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/TextWriter.html#increaseDepth--">increaseDepth()</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/TextWriter.html" title="class in com.compomics.util.io.export.writers">TextWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#increaseIndent--">increaseIndent()</a></span> - Method in class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Increases the indent counter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#increaseMissedCleavages--">increaseMissedCleavages()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Increases the number of missed cleavages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#increaseNX--">increaseNX()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Increases the number of Xs already considered in this draft.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#increasePrimaryProgressCounter--">increasePrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#increasePrimaryProgressCounter-int-">increasePrimaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#increasePrimaryProgressCounter--">increasePrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Increase the progress bar value by one "counter".</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#increasePrimaryProgressCounter-int-">increasePrimaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Increase the progress bar value by the given amount.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#increasePrimaryProgressCounter--">increasePrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#increasePrimaryProgressCounter-int-">increasePrimaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#increasePrimaryProgressCounter--">increasePrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#increasePrimaryProgressCounter-int-">increasePrimaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#increasePrimaryProgressCounter--">increasePrimaryProgressCounter()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Increase the primary progress counter by one "counter".</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#increasePrimaryProgressCounter-int-">increasePrimaryProgressCounter(int)</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Increase the primary progress counter by the given increment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#increaseRWCounter--">increaseRWCounter()</a></span> - Static method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Function for increasing the counter of processes accessing objects from
 the db.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#increaseSecondaryProgressCounter--">increaseSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Increases the progress bar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#increaseSecondaryProgressCounter-int-">increaseSecondaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Increases the value of the progress bar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#increaseSecondaryProgressCounter--">increaseSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Increase the secondary progress bar value by one "counter".</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#increaseSecondaryProgressCounter-int-">increaseSecondaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Increase the secondary progress bar value by the given amount.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#increaseSecondaryProgressCounter--">increaseSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#increaseSecondaryProgressCounter-int-">increaseSecondaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#increaseSecondaryProgressCounter--">increaseSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#increaseSecondaryProgressCounter-int-">increaseSecondaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#increaseSecondaryProgressCounter--">increaseSecondaryProgressCounter()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Increase the secondary progress counter by one "counter".</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#increaseSecondaryProgressCounter-int-">increaseSecondaryProgressCounter(int)</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Increase the secondary progress counter by the given amount.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html#increasing">increasing</a></span> - Variable in enum com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html" title="enum in com.compomics.util.experiment.identification.psm_scoring">PsmScore</a></dt>
<dd>
<div class="block">Indicates whether the score increases with the quality of the match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#inDB-long-">inDB(long)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Indicates whether an object is in the cache or the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#indentCounter">indentCounter</a></span> - Variable in class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Integer keeping track of the number of indents to include at the
 beginning of each line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.IonType.html#index">index</a></span> - Variable in enum com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.IonType.html" title="enum in com.compomics.util.experiment.biology.ions">Ion.IonType</a></dt>
<dd>
<div class="block">The index of the type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationType.html#index">index</a></span> - Variable in enum com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationType.html" title="enum in com.compomics.util.experiment.biology.modifications">ModificationType</a></dt>
<dd>
<div class="block">The index of the type, must be the index in the values array.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureAbsolute.html#index">index</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureAbsolute.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAIdentityFeatureAbsolute</a></dt>
<dd>
<div class="block">The index of this ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureRelative.html#index">index</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAIdentityFeatureRelative.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAIdentityFeatureRelative</a></dt>
<dd>
<div class="block">The index of this ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureAbsolute.html#index">index</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureAbsolute.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyFeatureAbsolute</a></dt>
<dd>
<div class="block">The index of this ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureRelative.html#index">index</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyFeatureRelative.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyFeatureRelative</a></dt>
<dd>
<div class="block">The index of this ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html#index">index</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature</a></dt>
<dd>
<div class="block">The index of this ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonAminoAcidFeature.html#index">index</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ComplementaryIonAminoAcidFeature</a></dt>
<dd>
<div class="block">The index of this ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonFeature.html#index">index</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ComplementaryIonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ComplementaryIonFeature</a></dt>
<dd>
<div class="block">The index of this ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonAminoAcidFeature.html#index">index</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ForwardIonAminoAcidFeature</a></dt>
<dd>
<div class="block">The index of this ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonFeature.html#index">index</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ForwardIonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ForwardIonFeature</a></dt>
<dd>
<div class="block">The index of this ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.Property.html#index">index</a></span> - Variable in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">IonFeature.Property</a></dt>
<dd>
<div class="block">The index of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html#index">index</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ModificationFeature</a></dt>
<dd>
<div class="block">The index of this ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideAminoAcidFeature.html#index">index</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideAminoAcidFeature</a></dt>
<dd>
<div class="block">The index of this ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html#index">index</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideFeature</a></dt>
<dd>
<div class="block">The index of this ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/PeptideFragmentationModel.html#index">index</a></span> - Variable in enum com.compomics.util.experiment.identification.peptide_fragmentation.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/PeptideFragmentationModel.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation">PeptideFragmentationModel</a></dt>
<dd>
<div class="block">The index of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html#index">index</a></span> - Variable in enum com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html" title="enum in com.compomics.util.experiment.identification.psm_scoring">PsmScore</a></dt>
<dd>
<div class="block">The index of the score of interest.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html#index">index</a></span> - Variable in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsFormat</a></dt>
<dd>
<div class="block">The index of the format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFormat.html#index">index</a></span> - Variable in enum com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFormat.html" title="enum in com.compomics.util.io.export">ExportFormat</a></dt>
<dd>
<div class="block">The index of the format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html#index">index</a></span> - Variable in enum com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html" title="enum in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters.MatchingType</a></dt>
<dd>
<div class="block">The index of the type as integer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html#index">index</a></span> - Variable in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.CleavageParameter</a></dt>
<dd>
<div class="block">The index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html#index">index</a></span> - Variable in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.Specificity</a></dt>
<dd>
<div class="block">The index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/peptide_shaker/ProjectType.html#index">index</a></span> - Variable in enum com.compomics.util.parameters.peptide_shaker.<a href="com/compomics/util/parameters/peptide_shaker/ProjectType.html" title="enum in com.compomics.util.parameters.peptide_shaker">ProjectType</a></dt>
<dd>
<div class="block">The index of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.MassIndexMap.html#indexes">indexes</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.MassIndexMap.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex.MassIndexMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#indexStringLengths">indexStringLengths</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Length of the indexed string (all concatenated protein sequences).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#inDoubleRange-java.lang.String-java.lang.String-double-double-">inDoubleRange(String, String, double, double)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true if the input is a double value inside the given range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#inferParameters-java.io.File-">inferParameters(File)</a></span> - Static method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Infers the parameters used to parse the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#inferParameters-java.io.File-com.compomics.util.waiting.WaitingHandler-">inferParameters(File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Infers the parameters used to parse the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#inIntegerRange-java.lang.String-java.lang.String-int-int-">inIntegerRange(String, String, int, int)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true if the input is an integer value inside the given range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#init--">init()</a></span> - Method in class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">Initializes the marshaller with (custom) type adapters and date format</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html#init--">init()</a></span> - Method in class com.compomics.util.io.json.marshallers.<a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html" title="class in com.compomics.util.io.json.marshallers">PrideMarshaller</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#initialize-java.lang.String-">initialize(String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Initializes the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html#initiate-java.lang.String:A-">initiate(String[])</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html" title="class in com.compomics.cli.identification_parameters">AbstractIdentificationParametersCli</a></dt>
<dd>
<div class="block">Initiates the IdentificationParametersCli.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#initiate-java.lang.String-">initiate(String)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Initiates the factory using the files of the static fields.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#initiateSorter--">initiateSorter()</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Initiates the sorter to the current order of the table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#insert-int-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidSequence-">insert(int, AminoAcidSequence)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Inserts another sequence in this sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#insert-int-java.lang.String-">insert(int, String)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Inserts another sequence in this sequence.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/variants/amino_acids/Insertion.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids"><span class="typeNameLink">Insertion</span></a> - Class in <a href="com/compomics/util/experiment/biology/variants/amino_acids/package-summary.html">com.compomics.util.experiment.biology.variants.amino_acids</a></dt>
<dd>
<div class="block">Class representing an amino acid insertion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/amino_acids/Insertion.html#Insertion-char-">Insertion(char)</a></span> - Constructor for class com.compomics.util.experiment.biology.variants.amino_acids.<a href="com/compomics/util/experiment/biology/variants/amino_acids/Insertion.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids">Insertion</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#insertObject-long-java.lang.Object-">insertObject(long, Object)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Stores an object in the desired table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#insertObjects-java.util.HashMap-com.compomics.util.waiting.WaitingHandler-boolean-">insertObjects(HashMap&lt;Long, Object&gt;, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Inserts a set of objects in the given table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/GuiUtilities.html#installEscapeCloseOperation-javax.swing.JDialog-">installEscapeCloseOperation(JDialog)</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/GuiUtilities.html" title="class in com.compomics.util.gui">GuiUtilities</a></dt>
<dd>
<div class="block">Close a dialog using the escape key.</div>
</dd>
<dt><a href="com/compomics/util/db/components/InstanceVariables.html" title="class in com.compomics.util.db.components"><span class="typeNameLink">InstanceVariables</span></a> - Class in <a href="com/compomics/util/db/components/package-summary.html">com.compomics.util.db.components</a></dt>
<dd>
<div class="block">This class represents the instance variables for a generated DBAccessor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/InstanceVariables.html#InstanceVariables-com.compomics.util.db.DBMetaData-">InstanceVariables(DBMetaData)</a></span> - Constructor for class com.compomics.util.db.components.<a href="com/compomics/util/db/components/InstanceVariables.html" title="class in com.compomics.util.db.components">InstanceVariables</a></dt>
<dd>
<div class="block">This constructor will create all the code for the instance variables from the given
 DBMetaData.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/Instrument.html" title="class in com.compomics.util.pride.prideobjects"><span class="typeNameLink">Instrument</span></a> - Class in <a href="com/compomics/util/pride/prideobjects/package-summary.html">com.compomics.util.pride.prideobjects</a></dt>
<dd>
<div class="block">An object for storing Instrument details.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Instrument.html#Instrument-java.lang.String-com.compomics.util.pride.CvTerm-com.compomics.util.pride.CvTerm-java.util.ArrayList-">Instrument(String, CvTerm, CvTerm, ArrayList&lt;CvTerm&gt;)</a></span> - Constructor for class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Instrument.html" title="class in com.compomics.util.pride.prideobjects">Instrument</a></dt>
<dd>
<div class="block">Create a new Instrument object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#intensity">intensity</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>
<div class="block">The intensity of the peak.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/IntensityHistogram.html" title="class in com.compomics.util.gui.spectrum"><span class="typeNameLink">IntensityHistogram</span></a> - Class in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">Creates an IntensityHistogram plot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IntensityHistogram.html#IntensityHistogram-com.compomics.util.experiment.identification.matches.IonMatch:A-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters.IntensityThresholdType-double-">IntensityHistogram(IonMatch[], Spectrum, AnnotationParameters.IntensityThresholdType, double)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IntensityHistogram.html" title="class in com.compomics.util.gui.spectrum">IntensityHistogram</a></dt>
<dd>
<div class="block">Creates an IntensityHistogram plot</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#intensityLimit">intensityLimit</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">The intensity limit to use.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#intensityLimit">intensityLimit</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>
<div class="block">The intensity limit used for the index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#intensityPeakMap">intensityPeakMap</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Intensity indexed Peak map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#INTERACTIVE_STATUS">INTERACTIVE_STATUS</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This status indicates that no annotation will be displayed, but the user
 will have a fully functional interface (point clicking, selecting,
 sequencing etc.)</div>
</dd>
<dt><a href="com/compomics/util/io/json/adapter/InterfaceAdapter.html" title="class in com.compomics.util.io.json.adapter"><span class="typeNameLink">InterfaceAdapter</span></a>&lt;<a href="com/compomics/util/io/json/adapter/InterfaceAdapter.html" title="type parameter in InterfaceAdapter">T</a>&gt; - Class in <a href="com/compomics/util/io/json/adapter/package-summary.html">com.compomics.util.io.json.adapter</a></dt>
<dd>
<div class="block">A generic adapter to parse used interfaces in a class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/adapter/InterfaceAdapter.html#InterfaceAdapter--">InterfaceAdapter()</a></span> - Constructor for class com.compomics.util.io.json.adapter.<a href="com/compomics/util/io/json/adapter/InterfaceAdapter.html" title="class in com.compomics.util.io.json.adapter">InterfaceAdapter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/SwingWorker.html#interrupt--">interrupt()</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/SwingWorker.html" title="class in com.compomics.util.sun">SwingWorker</a></dt>
<dd>
<div class="block">A new method that interrupts the worker thread.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/Iodine.html" title="class in com.compomics.util.experiment.biology.atoms.impl"><span class="typeNameLink">Iodine</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a></dt>
<dd>
<div class="block">Iodine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/impl/Iodine.html#Iodine--">Iodine()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.impl.<a href="com/compomics/util/experiment/biology/atoms/impl/Iodine.html" title="class in com.compomics.util.experiment.biology.atoms.impl">Iodine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions"><span class="typeNameLink">Ion</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/package-summary.html">com.compomics.util.experiment.biology.ions</a></dt>
<dd>
<div class="block">This class models an ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#Ion--">Ion()</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#ion">ion</a></span> - Variable in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">The matching ion.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/Ion.IonType.html" title="enum in com.compomics.util.experiment.biology.ions"><span class="typeNameLink">Ion.IonType</span></a> - Enum in <a href="com/compomics/util/experiment/biology/ions/package-summary.html">com.compomics.util.experiment.biology.ions</a></dt>
<dd>
<div class="block">An enumerator of the supported ion types.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/IonFactory.html" title="class in com.compomics.util.experiment.biology.ions"><span class="typeNameLink">IonFactory</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/package-summary.html">com.compomics.util.experiment.biology.ions</a></dt>
<dd>
<div class="block">This factory generates the expected ions from a peptide.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic"><span class="typeNameLink">IonFeature</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic</a></dt>
<dd>
<div class="block">Feature based on a property of an ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.html#IonFeature--">IonFeature()</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">IonFeature</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic"><span class="typeNameLink">IonFeature.Property</span></a> - Enum in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic</a></dt>
<dd>
<div class="block">Enum for the possible features.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html" title="class in com.compomics.util.gui.spectrum"><span class="typeNameLink">IonLabelColorTableModel</span></a> - Class in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">A table model to use for the ion label annotation colors.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html#IonLabelColorTableModel-java.util.HashMap-java.util.HashSet-">IonLabelColorTableModel(HashMap&lt;Ion.IonType, HashSet&lt;Integer&gt;&gt;, HashSet&lt;String&gt;)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html" title="class in com.compomics.util.gui.spectrum">IonLabelColorTableModel</a></dt>
<dd>
<div class="block">Constructor which sets a new table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html#IonLabelColorTableModel--">IonLabelColorTableModel()</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html" title="class in com.compomics.util.gui.spectrum">IonLabelColorTableModel</a></dt>
<dd>
<div class="block">Constructor which sets a new empty table.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches"><span class="typeNameLink">IonMatch</span></a> - Class in <a href="com/compomics/util/experiment/identification/matches/package-summary.html">com.compomics.util.experiment.identification.matches</a></dt>
<dd>
<div class="block">This class represents the assignment of a peak to a theoretical ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#IonMatch-com.compomics.util.experiment.mass_spectrometry.spectra.Peak-com.compomics.util.experiment.biology.ions.Ion-java.lang.Integer-">IonMatch(Peak, Ion, Integer)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">Constructor for an ion peak.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html" title="enum in com.compomics.util.experiment.identification.matches"><span class="typeNameLink">IonMatch.MzErrorType</span></a> - Enum in <a href="com/compomics/util/experiment/identification/matches/package-summary.html">com.compomics.util.experiment.identification.matches</a></dt>
<dd>
<div class="block">Enum of the supported error types.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/IonMatchKeysCache.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation"><span class="typeNameLink">IonMatchKeysCache</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation</a></dt>
<dd>
<div class="block">Cache for the keys of the ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/IonMatchKeysCache.html#IonMatchKeysCache--">IonMatchKeysCache()</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/IonMatchKeysCache.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">IonMatchKeysCache</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#ionMatchKeysCache">ionMatchKeysCache</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">The cache to use for the ion match keys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iPointDetectionTolerance">iPointDetectionTolerance</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The deviation (both left and right) allowed for point highlighting
 detection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iPointSize">iPointSize</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Size for the point on a polygon.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#iPosition">iPosition</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This integer holds the position marker for the cleavage direction for
 this Enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iPrecursorCharge">iPrecursorCharge</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This String holds the charge for the precursor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iPrecursorChargeMirorredSpectrum">iPrecursorChargeMirorredSpectrum</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This String holds the charge for the precursor for the mirrored spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iPrecursorMZ">iPrecursorMZ</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This variable holds the precursor M/Z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iPrecursorMZMirroredSpectrum">iPrecursorMZMirroredSpectrum</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This variable holds the precursor M/Z for the mirrored spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#iReport">iReport</a></span> - Variable in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>
<div class="block">The report to append.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#iRestrict">iRestrict</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This char[] holds the residues that will restrict cleavage when present
 after a cleavable residue.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#iRestrictors">iRestrictors</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">Lookup cache for the restricting residues.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/CompomicsWrapper.html#is64BitJava--">is64BitJava()</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/CompomicsWrapper.html" title="class in com.compomics.software">CompomicsWrapper</a></dt>
<dd>
<div class="block">Returns if the Java version used is 64 bit.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#isAa-char-">isAa(char)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the given character is a supported
 amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#isAboveXAxis--">isAboveXAxis()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Returns true if the reference area is added above the x-axis, false adds
 the reference are below the x-axis</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#isAccounted-com.compomics.util.experiment.identification.spectrum_annotation.NeutralLossesMap-com.compomics.util.experiment.biology.ions.NeutralLoss-com.compomics.util.experiment.biology.ions.Ion-">isAccounted(NeutralLossesMap, NeutralLoss, Ion)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the neutral loss should be accounted
 for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#isActive--">isActive()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Indicates whether the filter is active.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#isAdditionalOutput--">isAdditionalOutput()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Returns if additional output is to be included.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#isAdjustPrecursorMass--">isAdjustPrecursorMass()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Returns if the precursor mass is to be adjusted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFeature.html#isAdvanced--">isAdvanced()</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFeature.html" title="interface in com.compomics.util.io.export">ExportFeature</a></dt>
<dd>
<div class="block">Indicates whether a feature is for advanced user only.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#isAnnotateHighestPeak--">isAnnotateHighestPeak()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns true if the most intense of possible peaks to annotate is to be
 selected, false if the most accurate is to be selected.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#isArtifact--">isArtifact()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">This method indicates whether this modification  is considered an artifact.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#isArtifact--">isArtifact()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This method indicates whether this modification  is considered an artifact.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#isArtifact--">isArtifact()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">This method indicates whether this modification  is considered an artifact.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#isAutomaticAnnotation--">isAutomaticAnnotation()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns whether the annotation settings should be automatically inferred.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#isAutoUpdate--">isAutoUpdate()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Indicates whether the tools should use the auto update function.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#isBooleanInput-java.lang.String-java.lang.String-">isBooleanInput(String, String)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true of the input is 0 or 1.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html#isCacheEmpty--">isCacheEmpty()</a></span> - Method in class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/BinomialDistribution.html" title="class in com.compomics.util.math.statistics.distributions">BinomialDistribution</a></dt>
<dd>
<div class="block">Indicates whether all caches are empty.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/gui/PathParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.software.settings.gui.<a href="com/compomics/software/settings/gui/PathParametersDialog.html" title="class in com.compomics.software.settings.gui">PathParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the action was canceled by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/gui/MsConvertParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.proteowizard.gui.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/gui/MsConvertParametersDialog.html" title="class in com.compomics.util.experiment.mass_spectrometry.proteowizard.gui">MsConvertParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the editing was canceled by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/AminoAcidPatternDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/AminoAcidPatternDialog.html" title="class in com.compomics.util.gui">AminoAcidPatternDialog</a></dt>
<dd>
<div class="block">indicates whether the changes have been canceled</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomChainDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomChainDialog.html" title="class in com.compomics.util.gui.atoms">AtomChainDialog</a></dt>
<dd>
<div class="block">Indicates whether the edition has been canceled by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/ExportFormatSelectionDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/ExportFormatSelectionDialog.html" title="class in com.compomics.util.gui">ExportFormatSelectionDialog</a></dt>
<dd>
<div class="block">Returns true of the user canceled the dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/file_handling/FileSelectionDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.file_handling.<a href="com/compomics/util/gui/file_handling/FileSelectionDialog.html" title="class in com.compomics.util.gui.file_handling">FileSelectionDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterDialog.html" title="class in com.compomics.util.gui.filtering">FilterDialog</a></dt>
<dd>
<div class="block">Indicates whether the editing was canceled by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/genes/SpeciesDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.genes.<a href="com/compomics/util/gui/genes/SpeciesDialog.html" title="class in com.compomics.util.gui.genes">SpeciesDialog</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/modification/ModificationDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.modification.<a href="com/compomics/util/gui/modification/ModificationDialog.html" title="class in com.compomics.util.gui.modification">ModificationDialog</a></dt>
<dd>
<div class="block">Indicates whether the edition was canceled by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/AnnotationParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/AnnotationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">AnnotationParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/FractionParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/FractionParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">FractionParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/GeneParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/GeneParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">GeneParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/MatchesImportParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/MatchesImportParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">MatchesImportParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ModificationLocalizationParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ModificationLocalizationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ModificationLocalizationParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/PeptideVariantsParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/PeptideVariantsParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">PeptideVariantsParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ProteinInferenceParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ProteinInferenceParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ProteinInferenceParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/PsmScoringParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/PsmScoringParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">PsmScoringParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/SequenceMatchingParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/SequenceMatchingParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">SequenceMatchingParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationQCParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the preference edition was canceled by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersEditionDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersNameDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersNameDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersNameDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SearchParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the cancel button was pressed by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SequenceDbDetailsDialog</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the database selection was canceled
 by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/tools/ProcessingParametersDialog.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.parameters.tools.<a href="com/compomics/util/gui/parameters/tools/ProcessingParametersDialog.html" title="class in com.compomics.util.gui.parameters.tools">ProcessingParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/ListChooser.html#isCanceled--">isCanceled()</a></span> - Method in class com.compomics.util.gui.utils.user_choice.<a href="com/compomics/util/gui/utils/user_choice/ListChooser.html" title="class in com.compomics.util.gui.utils.user_choice">ListChooser</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the selection was canceled by the
 user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">AndromedaParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">CometParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">DirecTagParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MsAmandaParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MsgfParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MyriMatchParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">NovorParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">OmssaParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">PepNovoParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">PNovoParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">TideParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">XTandemParametersDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/AlgorithmParametersDialog.html#isCancelled--">isCancelled()</a></span> - Method in interface com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/AlgorithmParametersDialog.html" title="interface in com.compomics.util.gui.parameters.identification">AlgorithmParametersDialog</a></dt>
<dd>
<div class="block">Indicates whether the user canceled the editing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html#isCellEditable-int-int-">isCellEditable(int, int)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IonLabelColorTableModel.html" title="class in com.compomics.util.gui.spectrum">IonLabelColorTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html#isCellEditable-int-int-">isCellEditable(int, int)</a></span> - Method in class com.compomics.util.gui.variants.aa_substitutions.<a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html" title="class in com.compomics.util.gui.variants.aa_substitutions">AaSubstitutionMatrixTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableMap.html#isCellEditable-int-int-">isCellEditable(int, int)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun">TableMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/interfaces/DBElement.html#isChanged--">isChanged()</a></span> - Method in interface com.compomics.util.db.interfaces.<a href="com/compomics/util/db/interfaces/DBElement.html" title="interface in com.compomics.util.db.interfaces">DBElement</a></dt>
<dd>
<div class="block">This method signals whether the DBElement has changes that should be persisted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#isCheckSpectrumCharges--">isCheckSpectrumCharges()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns whether the spectra are to be checked for missing charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#isCleavageSite-java.lang.String-java.lang.String-">isCleavageSite(String, String)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the given amino acids represent a
 cleavage site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#isCleavageSite-char-char-">isCleavageSite(char, char)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the given amino acids represent a
 cleavage site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#isCleavageSiteNoCombination-java.lang.Character-java.lang.Character-">isCleavageSiteNoCombination(Character, Character)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the given amino acids represent a
 cleavage site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#isCleaveNterMethionine--">isCleaveNterMethionine()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Indicates whether N-terminal methionines should be cleaved.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Indicates whether the amino acid object refers to a combination of amino
 acids like B, J, Z or X.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Alanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Alanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Arginine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Arginine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Asparagine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Asparagine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/AsparticAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">AsparticAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/B.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">B</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Cysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Cysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/GlutamicAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">GlutamicAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glutamine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glutamine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Glycine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Glycine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Histidine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Histidine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Isoleucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">J</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Leucine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Lysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Methionine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Phenylalanine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Proline</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Pyrrolysine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Selenocysteine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Serine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Threonine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tryptophan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tyrosine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Valine</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">X</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html#iscombination--">iscombination()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Z</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#isConfidentModificationSite-int-java.lang.String-">isConfidentModificationSite(int, String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Indicates whether a site is already registered as confident modification
 site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#isConnectionActive--">isConnectionActive()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Indicates whether the connection to the DB is active.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#isConnectionActive--">isConnectionActive()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Indicates whether the connection to the DB is active.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#isCorrectPrecursorMass--">isCorrectPrecursorMass()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the precursor mass shall be
 corrected (TagDB setting).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationType.html#isCTerm--">isCTerm()</a></span> - Method in enum com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationType.html" title="enum in com.compomics.util.experiment.biology.modifications">ModificationType</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the modification type targets the C-terminus of a protein or of a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#isCterm-java.lang.String-int-">isCterm(String, int)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the peptide is at the C-terminus of
 the given protein sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#isDbSize--">isDbSize()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Indicates whether the database size should be checked.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#isDecoy--">isDecoy()</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the given match is decoy.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#isDecoy-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.io.biology.protein.SequenceProvider-">isDecoy(Peptide, SequenceProvider)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the peptide matches a decoy
 sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html#isDecoy-java.lang.String-com.compomics.util.experiment.io.biology.protein.FastaParameters-">isDecoy(String, FastaParameters)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html" title="class in com.compomics.util.experiment.identification.utils">ProteinUtils</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the given accession corresponds to a decoy sequence according to the given fasta parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html#isDecoy-java.lang.String-com.compomics.util.experiment.io.biology.protein.SequenceProvider-">isDecoy(String, SequenceProvider)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html" title="class in com.compomics.util.experiment.identification.utils">ProteinUtils</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the given accession corresponds to a decoy sequence according to the given sequence provider.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#isDecoySuffix--">isDecoySuffix()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the decoy flag is a suffix or a
 prefix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/interfaces/DBElement.html#isDeleted--">isDeleted()</a></span> - Method in interface com.compomics.util.db.interfaces.<a href="com/compomics/util/db/interfaces/DBElement.html" title="interface in com.compomics.util.db.interfaces">DBElement</a></dt>
<dd>
<div class="block">This method signals whether the DBElement has been deleted in the persistent
 store.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#isDependentLosses--">isDependentLosses()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns whether neutral losses should be sequence dependent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#isDetermineChargePlusOneAlgorithmically--">isDetermineChargePlusOneAlgorithmically()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Indicates whether charge plus one should be determined algorithmically.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/PeptideShakerSetupDialog.html#isDialogCanceled--">isDialogCanceled()</a></span> - Method in class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/PeptideShakerSetupDialog.html" title="class in com.compomics.software.dialogs">PeptideShakerSetupDialog</a></dt>
<dd>
<div class="block">Returns true of the dialog was canceled by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/ProteoWizardSetupDialog.html#isDialogCanceled--">isDialogCanceled()</a></span> - Method in class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/ProteoWizardSetupDialog.html" title="class in com.compomics.software.dialogs">ProteoWizardSetupDialog</a></dt>
<dd>
<div class="block">Returns true of the dialog was canceled by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/ReporterSetupDialog.html#isDialogCanceled--">isDialogCanceled()</a></span> - Method in class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/ReporterSetupDialog.html" title="class in com.compomics.software.dialogs">ReporterSetupDialog</a></dt>
<dd>
<div class="block">Returns true of the dialog was canceled by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/SearchGuiSetupDialog.html#isDialogCanceled--">isDialogCanceled()</a></span> - Method in class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/SearchGuiSetupDialog.html" title="class in com.compomics.software.dialogs">SearchGuiSetupDialog</a></dt>
<dd>
<div class="block">Returns true of the dialog was canceled by the user.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#isDouble-java.lang.String-java.lang.String-">isDouble(String, String)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true if the argument can be parsed as a double value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#isDuplicateSpectra--">isDuplicateSpectra()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Indicates whether to use duplicate spectra option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#isEditable--">isEditable()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Indicates whether the scheme is editable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html#isElement-java.lang.String-">isElement(String)</a></span> - Method in class com.compomics.util.gui.isotopic_calculator.<a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html" title="class in com.compomics.util.gui.isotopic_calculator">IsotopeDistributionGui</a></dt>
<dd>
<div class="block">Method that checks if a given string is an element we can calculate an isotopic distribution for</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IsotopicDistributionPanel.html#isElement-java.lang.String-">isElement(String)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IsotopicDistributionPanel.html" title="class in com.compomics.util.gui.spectrum">IsotopicDistributionPanel</a></dt>
<dd>
<div class="block">Method that checks if a given string is an element we can calculate an isotopic distribution for</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#isEmpiricalCorrection--">isEmpiricalCorrection()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Indicates whether empirical correction should be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#isEmpty--">isEmpty()</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Indicates whether the cache is empty.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html#isEmpty--">isEmpty()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">NeutralLossesMap</a></dt>
<dd>
<div class="block">Returns a boolean indicating if the mapping is empty.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#isEmpty--">isEmpty()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the spectrum is empty.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#isEnding-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isEnding(AminoAcidPattern, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether the given amino acid sequence ends with the pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#isEnding-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isEnding(String, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether the given amino acid sequence ends with the pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#isEnzymatic-com.compomics.util.experiment.biology.proteins.Peptide-java.lang.String-java.lang.String-java.util.ArrayList-">isEnzymatic(Peptide, String, String, ArrayList&lt;Enzyme&gt;)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the peptide is enzymatic using one
 of the given enzymes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#isEnzymatic-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.experiment.io.biology.protein.SequenceProvider-java.util.ArrayList-">isEnzymatic(Peptide, SequenceProvider, ArrayList&lt;Enzyme&gt;)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the peptide is enzymatic in at least
 one protein using one of the given enzymes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#isEnzymaticProduct-java.lang.String-int-int-">isEnzymaticProduct(String, int, int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">This method reports on the possibility that the presented subsequence
 (represented by the start and end location in the parent) is the result
 of enzymatic activity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#isEnzymaticProduct-java.lang.String-java.lang.String-">isEnzymaticProduct(String, String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method reports on the possibility that the presented subsequence is
 the result of enzymatic activity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#isEnzymaticProduct-java.lang.String-int-int-">isEnzymaticProduct(String, int, int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method reports on the possibility that the presented subsequence
 (represented by the start and end location in the parent) is the result
 of enzymatic activity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/RegExEnzyme.html#isEnzymaticProduct-java.lang.String-java.lang.String-">isEnzymaticProduct(String, String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/RegExEnzyme.html" title="class in com.compomics.util.protein">RegExEnzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/RegExEnzyme.html#isEnzymaticProduct-java.lang.String-int-int-">isEnzymaticProduct(String, int, int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/RegExEnzyme.html" title="class in com.compomics.util.protein">RegExEnzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#isEqualIL--">isEqualIL()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Indicates whether I and L should be considered indistinguishable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#isEstimateCharge--">isEstimateCharge()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Indicates whether the precursor charge shall be estimated for OMSSA.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#isEstimateCharge--">isEstimateCharge()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Indicates whether the precursor charge shall be estimated for PepNovo.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#isFiltered--">isFiltered()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Indicates whether the protein list is filtered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#isFirstDecoy--">isFirstDecoy()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Indicates whether the number of hits before the first decoy should be
 checked.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#isFixed--">isFixed()</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the neutral loss is fixed or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#isForward-int-">isForward(int)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Indicates whether the given subtype refers to a forward or a rewind ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#isFragmentAll--">isFragmentAll()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Indicates whether the fragment all option should be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#isFragmentIonPpm--">isFragmentIonPpm()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Indicates whether the fragment ion accuracy is in ppm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#isFragmentIonPpm--">isFragmentIonPpm()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Indicates whether the fragment ion accuracy is in ppm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#isHeader--">isHeader()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Indicates whether column header shall be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#isHigherCharge--">isHigherCharge()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Indicates whether higher charge should be considered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#isIncludeAmmonia--">isIncludeAmmonia()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether ammonia losses should be accounted
 for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#isIncludeDateInOutputName--">isIncludeDateInOutputName()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Indicates whether the date should be included in the SearchGUI output
 name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#isIncludeDecoy--">isIncludeDecoy()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Indicates whether decoy hits should be included.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#isIncludeSectionTitles--">isIncludeSectionTitles()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Indicates whether section titles shall be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#isIncludeWater--">isIncludeWater()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether water losses should be accounted
 for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html#isIndeterminate--">isIndeterminate()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialog</a></dt>
<dd>
<div class="block">This method signals whether the progressdialog is currently in
 indeterminate ('true') or determinate ('false') mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#isIndexes--">isIndexes()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Indicates whether lines shall be indexed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#isInList-java.lang.String-java.lang.String-java.util.List-">isInList(String, String, List&lt;String&gt;)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true of the input is in the provided list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#isInList-java.lang.String-java.lang.String-java.lang.String:A-">isInList(String, String, String[])</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true of the input is in the provided list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#isInList-java.lang.String-java.lang.String-java.util.stream.Stream-">isInList(String, String, Stream&lt;String&gt;)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true of the input is in the provided stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#isInteger-java.lang.String-java.lang.String-">isInteger(String, String)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true if the argument can be parsed as an integer value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#isIsPpm--">isIsPpm()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Indicates whether the mass tolerance is in ppm (true) or Dalton (false).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html#isList--">isList()</a></span> - Method in class com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html" title="class in com.compomics.cli.enzymes">EnzymesCLIInputBean</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the list of implemented enzymes
 should be printed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html#isList--">isList()</a></span> - Method in class com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html" title="class in com.compomics.cli.modifications">ModificationsCLIInputBean</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the list of implemented enzymes
 should be printed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html#isListEnzymes--">isListEnzymes()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersInputBean</a></dt>
<dd>
<div class="block">Indicates whether the enzymes should be printed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html#isListMods--">isListMods()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersInputBean</a></dt>
<dd>
<div class="block">Indicates whether the modifications should be printed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#isLowMemoryMode--">isLowMemoryMode()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">use getMaxLoadedProteins and getMaxLoadedSpectra instead</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#isMemoryMappedSequenceLibraries--">isMemoryMappedSequenceLibraries()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Indicates whether sequence libraries should be mapped in memory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#isMiniature--">isMiniature()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns true if the graphics panel is to be drawn in a miniature form.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#isModificationRefinement--">isModificationRefinement()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Returns whether the protein mapping should be refined based on terminal and pattern modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#isMonoIsotopic--">isMonoIsotopic()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns whether monoisotopic mass values shall be used (in contrast to
 average mass values).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#isNeutralLossesAuto--">isNeutralLossesAuto()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Indicates whether neutral losses should be automatically selected.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationType.html#isNTerm--">isNTerm()</a></span> - Method in enum com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationType.html" title="enum in com.compomics.util.experiment.biology.modifications">ModificationType</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the modification type targets the N-terminus of a protein or of a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#isNterm-java.lang.String-int-">isNterm(String, int)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the peptide is at the N-terminus of
 the given protein sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItem.html#isNumber--">isNumber()</a></span> - Method in interface com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItem.html" title="interface in com.compomics.util.experiment.filtering">FilterItem</a></dt>
<dd>
<div class="block">Indicates whether the item expects a number as threshold.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html#isNumber--">isNumber()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">AssumptionFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html#isNumber--">isNumber()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PeptideFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html#isNumber--">isNumber()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">ProteinFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html#isNumber--">isNumber()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PsmFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Isoleucine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Isoleucine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html#Isoleucine--">Isoleucine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Isoleucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Isoleucine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#ISOLEUCINE">ISOLEUCINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Isoleucine immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html#isOne-int-">isOne(int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">Rank</a></dt>
<dd>
<div class="block">Returns true if the value is equal to one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html#isOneInt-int-">isOneInt(int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">Rank</a></dt>
<dd>
<div class="block">Returns true if the value is equal to one.</div>
</dd>
<dt><a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html" title="class in com.compomics.util.gui.isotopic_calculator"><span class="typeNameLink">IsotopeDistributionGui</span></a> - Class in <a href="com/compomics/util/gui/isotopic_calculator/package-summary.html">com.compomics.util.gui.isotopic_calculator</a></dt>
<dd>
<div class="block">This class is a GUI that visualizes the isotopic calculator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html#IsotopeDistributionGui-boolean-">IsotopeDistributionGui(boolean)</a></span> - Constructor for class com.compomics.util.gui.isotopic_calculator.<a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html" title="class in com.compomics.util.gui.isotopic_calculator">IsotopeDistributionGui</a></dt>
<dd>
<div class="block">The constructor</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#isotopeMap">isotopeMap</a></span> - Variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">Map of the isotope masses relative to the monoisotopic peak (+1 for
 carbon 13).</div>
</dd>
<dt><a href="com/compomics/util/general/IsotopicDistribution.html" title="class in com.compomics.util.general"><span class="typeNameLink">IsotopicDistribution</span></a> - Class in <a href="com/compomics/util/general/package-summary.html">com.compomics.util.general</a></dt>
<dd>
<div class="block">This class calculates the isotopic distribution based on a molecular formula.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistribution.html#IsotopicDistribution-com.compomics.util.protein.MolecularFormula-">IsotopicDistribution(MolecularFormula)</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistribution.html" title="class in com.compomics.util.general">IsotopicDistribution</a></dt>
<dd>
<div class="block">Constructor</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistribution.html#IsotopicDistribution-com.compomics.util.protein.MolecularFormula-int-">IsotopicDistribution(MolecularFormula, int)</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistribution.html" title="class in com.compomics.util.general">IsotopicDistribution</a></dt>
<dd>
<div class="block">Constructor</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistribution.html#IsotopicDistribution-int-int-int-int-int-">IsotopicDistribution(int, int, int, int, int)</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistribution.html" title="class in com.compomics.util.general">IsotopicDistribution</a></dt>
<dd>
<div class="block">This will calculate the isotopic distribution pattern for the given elements.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/IsotopicDistributionPanel.html" title="class in com.compomics.util.gui.spectrum"><span class="typeNameLink">IsotopicDistributionPanel</span></a> - Class in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">This class provides a JPanel that can display a peptide isotopic distribution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/IsotopicDistributionPanel.html#IsotopicDistributionPanel-java.lang.String-java.lang.Integer-boolean-int-">IsotopicDistributionPanel(String, Integer, boolean, int)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/IsotopicDistributionPanel.html" title="class in com.compomics.util.gui.spectrum">IsotopicDistributionPanel</a></dt>
<dd>
<div class="block">This constructor creates an IsotopicDistributionPanel based on the passed parameters.</div>
</dd>
<dt><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general"><span class="typeNameLink">IsotopicDistributionSpectrum</span></a> - Class in <a href="com/compomics/util/general/package-summary.html">com.compomics.util.general</a></dt>
<dd>
<div class="block">This class is an implementation of the SpectrumFile specific for the isotopic distribution
 
 Created by IntelliJ IDEA.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#IsotopicDistributionSpectrum--">IsotopicDistributionSpectrum()</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/general/IsotopicElement.html" title="class in com.compomics.util.general"><span class="typeNameLink">IsotopicElement</span></a> - Class in <a href="com/compomics/util/general/package-summary.html">com.compomics.util.general</a></dt>
<dd>
<div class="block">This class represents the isotopically different element with the occurrence 
 and the dalton difference between this and the natural variant.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isOutputHistograms--">isOutputHistograms()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Indicates whether histograms should be written in the result file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isOutputProteins--">isOutputProteins()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Indicates whether the protein bloc should be included in the export.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isOutputSequences--">isOutputSequences()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns whether the protein sequences should be included in the protein
 block of the export.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isOutputSpectra--">isOutputSpectra()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Indicate whether the spectra should be exported in the result file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#isPeakPicked--">isPeakPicked()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns true if the indexed file seems to contain only peak picked
 spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iSpecPanelListeners">iSpecPanelListeners</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The list of SpectrumPanelListeners.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#isPerformDeisotoping--">isPerformDeisotoping()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Returns whether deisotoping is to be performed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#isPositiveDouble-java.lang.String-java.lang.String-boolean-">isPositiveDouble(String, String, boolean)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true if the argument can be parsed as a positive double value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#isPositiveInteger-java.lang.String-java.lang.String-boolean-">isPositiveInteger(String, String, boolean)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true if the argument can be parsed as a positive integer value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isPotentialModificationsForFullRefinment--">isPotentialModificationsForFullRefinment()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Indicates whether the refinement modifications should be used for the
 full refinement.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#isPpm">isPpm</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Boolean indicating whether the tolerance is in ppm (true) or in Dalton
 (false).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#isPrecursorAccuracyTypePpm--">isPrecursorAccuracyTypePpm()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns true if the current precursor accuracy type is ppm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#isPrecursorChargesMissing--">isPrecursorChargesMissing()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Returns true if the at least one spectrum is missing the precursor
 charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#isProbabilisticScoreCalculation--">isProbabilisticScoreCalculation()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Indicates whether a probabilistic modification score is required.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#isProbabilisticScoreNeutralLosses--">isProbabilisticScoreNeutralLosses()</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Indicates whether the neutral losses shall be taken into account for
 spectrum annotation when calculating the probabilistic score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html#isProgressBarValid--">isProgressBarValid()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialog</a></dt>
<dd>
<div class="block">This method reports on whether this instance can be controlled in a
 multithreaded environment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isProteinQuickAcetyl--">isProteinQuickAcetyl()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Indicates whether the protein quick acetylation option should be
 triggered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#isQcPassed-java.lang.String-">isQcPassed(String)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Indicates whether the given QC check was passed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isQuickPyrolidone--">isQuickPyrolidone()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns whether the quick pyrolidone option should be triggered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isRefine--">isRefine()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns whether the second pass search should be triggered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isRefinePointMutations--">isRefinePointMutations()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Indicates whether point mutations should be looked for during the
 refinement process.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isRefineSemi--">isRefineSemi()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Indicates whether the semi enzymatic option of the second pass search
 should be triggered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isRefineSnaps--">isRefineSnaps()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether snAPs should be used during the refinement process.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isRefineSpectrumSynthesis--">isRefineSpectrumSynthesis()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Indicates whether the spectrum synthesis option should be used during the
 refinement process.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isRefineUnanticipatedCleavages--">isRefineUnanticipatedCleavages()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns whether the unanticipated cleavages option should be used during
 the refinement process.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#isRemovePrecursor--">isRemovePrecursor()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Indicates whether the precursor shall be removed for OMSSA.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#isReport--">isReport()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#isReport--">isReport()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#isReport--">isReport()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#isReport--">isReport()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#isReport--">isReport()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Indicates whether this waiting handler supports reports.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#isRunCanceled--">isRunCanceled()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#isRunCanceled--">isRunCanceled()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Returns true if the run is canceled.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#isRunCanceled--">isRunCanceled()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#isRunCanceled--">isRunCanceled()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#isRunCanceled--">isRunCanceled()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Returns true if the run is canceled.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#isRunFinished--">isRunFinished()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#isRunFinished--">isRunFinished()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#isRunFinished--">isRunFinished()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#isRunFinished--">isRunFinished()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#isRunFinished--">isRunFinished()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Returns true if the process is finished.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#isRunning--">isRunning()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>
<div class="block">This method can be consulted to find out whether the monitor is running.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#isSameAs-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameAs(AminoAcidPattern, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether another AminoAcidPattern targets the same pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#isSameAs-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidSequence-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameAs(AminoAcidSequence, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Indicates whether another sequence has a matching sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#isSameAs-com.compomics.util.experiment.identification.amino_acid_tags.TagComponent-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameAs(TagComponent, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html#isSameAs-com.compomics.util.experiment.biology.atoms.AtomImpl-">isSameAs(AtomImpl)</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html" title="class in com.compomics.util.experiment.biology.atoms">AtomImpl</a></dt>
<dd>
<div class="block">Indicates whether another atom is the same as this one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#isSameAs-com.compomics.util.experiment.biology.ions.Ion-">isSameAs(Ion)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#isSameAs-com.compomics.util.experiment.biology.ions.Ion-">isSameAs(Ion)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#isSameAs-com.compomics.util.experiment.biology.ions.Ion-">isSameAs(Ion)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#isSameAs-com.compomics.util.experiment.biology.ions.Ion-">isSameAs(Ion)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#isSameAs-com.compomics.util.experiment.biology.ions.Ion-">isSameAs(Ion)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#isSameAs-com.compomics.util.experiment.biology.ions.impl.RelatedIon-">isSameAs(RelatedIon)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Compares the current related ion with another one based on their
 composition.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#isSameAs-com.compomics.util.experiment.biology.ions.Ion-">isSameAs(Ion)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#isSameAs-com.compomics.util.experiment.biology.ions.impl.ReporterIon-">isSameAs(ReporterIon)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Compares the current reporter ion with another one based on their masses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#isSameAs-com.compomics.util.experiment.biology.ions.Ion-">isSameAs(Ion)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#isSameAs-com.compomics.util.experiment.biology.ions.Ion-">isSameAs(Ion)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#isSameAs-com.compomics.util.experiment.biology.ions.Ion-">isSameAs(Ion)</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the ion is the same as another ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#isSameAs-com.compomics.util.experiment.biology.ions.NeutralLoss-">isSameAs(NeutralLoss)</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Method indicating whether another neutral loss is the same as the one
 considered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#isSameAs-com.compomics.util.experiment.biology.modifications.Modification-">isSameAs(Modification)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Returns true if the modification is the same as another one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#isSameAs-com.compomics.util.experiment.biology.variants.AaSubstitutionMatrix-">isSameAs(AaSubstitutionMatrix)</a></span> - Method in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Indicates whether the given AaSubstitutionMatrix is the same as this one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/Filter.html#isSameAs-com.compomics.util.experiment.filtering.Filter-">isSameAs(Filter)</a></span> - Method in interface com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/Filter.html" title="interface in com.compomics.util.experiment.filtering">Filter</a></dt>
<dd>
<div class="block">Indicates whether another filter is the same as the current filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html#isSameAs-com.compomics.util.experiment.identification.amino_acid_tags.TagComponent-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameAs(TagComponent, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">MassGap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#isSameAs-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameAs(Tag, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Indicates whether this tag is the same as another tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/TagComponent.html#isSameAs-com.compomics.util.experiment.identification.amino_acid_tags.TagComponent-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameAs(TagComponent, SequenceMatchingParameters)</a></span> - Method in interface com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/TagComponent.html" title="interface in com.compomics.util.experiment.identification.amino_acid_tags">TagComponent</a></dt>
<dd>
<div class="block">Indicates whether another component is the same as the component of
 interest.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#isSameAs-com.compomics.util.experiment.filtering.Filter-">isSameAs(Filter)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#isSameAs-com.compomics.util.experiment.identification.filtering.PeptideAssumptionFilter-">isSameAs(PeptideAssumptionFilter)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Indicates whether this filter is the same as another one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html#isSameAs-com.compomics.util.experiment.identification.matches.ModificationMatch-">isSameAs(ModificationMatch)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches">ModificationMatch</a></dt>
<dd>
<div class="block">Indicates whether this modification match is the same of another one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#isSameAs-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-">isSameAs(AnnotationParameters)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the given annotation settings are
 the same as these ones.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#isSameAs-com.compomics.util.experiment.io.biology.protein.FastaParameters-">isSameAs(FastaParameters)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the parsing parameters are the same
 as the given parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#isSameAs-com.compomics.util.experiment.mass_spectrometry.spectra.Peak-">isSameAs(Peak)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>
<div class="block">Returns true if the peak has the same mz and intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/UnitOfMeasurement.html#isSameAs-com.compomics.util.experiment.units.UnitOfMeasurement-">isSameAs(UnitOfMeasurement)</a></span> - Method in class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/UnitOfMeasurement.html" title="class in com.compomics.util.experiment.units">UnitOfMeasurement</a></dt>
<dd>
<div class="block">Indicates whether the given unit of measurement is the same as another.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html#isSameAs-com.compomics.util.parameters.identification.advanced.FractionParameters-">isSameAs(FractionParameters)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html" title="class in com.compomics.util.parameters.identification.advanced">FractionParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether other given settings are the same as
 these.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#isSameAs-com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters-">isSameAs(PeptideVariantsParameters)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Indicates whether another peptide variant preferences is the same as this
 one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html#isSameAs-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameAs(SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a></dt>
<dd>
<div class="block">Indicates whether another sequence matching preferences is the same as
 this one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#isSameAs-com.compomics.util.parameters.identification.advanced.ValidationQcParameters-">isSameAs(ValidationQcParameters)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Returns true if the two ValidationQCPreferences are the same.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#isSameAs-com.compomics.util.parameters.identification.search.DigestionParameters-">isSameAs(DigestionParameters)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether these digestion parameters are the
 same as the given other parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#isSameAs-com.compomics.util.parameters.quantification.spectrum_counting.SpectrumCountingParameters-">isSameAs(SpectrumCountingParameters)</a></span> - Method in class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Compares two spectrum counting preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#isSameAtomicComposition-com.compomics.util.experiment.biology.modifications.Modification-">isSameAtomicComposition(Modification)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Returns true if the atomic composition of the modification is the same as another
 one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#isSameCompositionAs-com.compomics.util.experiment.biology.atoms.AtomChain-">isSameCompositionAs(AtomChain)</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>
<div class="block">Indicates whether two atom chains are of the same composition by
 comparing their string and type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#isSameModificationStatus-com.compomics.util.experiment.biology.proteins.Peptide-">isSameModificationStatus(Peptide)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Indicates whether another peptide has the same variable modifications as
 this peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#isSamePattern-com.compomics.util.experiment.biology.modifications.Modification-">isSamePattern(Modification)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Returns true if the targeted pattern of the modification is the same as another
 one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#isSameSequence-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameSequence(Peptide, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether another peptide has the same
 sequence as the given peptide</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#isSameSequenceAndModificationStatus-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameSequenceAndModificationStatus(Peptide, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Indicates whether another peptide has the same sequence and modification
 status without accounting for modification localization.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#isSameSequenceAndModificationStatusAs-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidSequence-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameSequenceAndModificationStatusAs(AminoAcidSequence, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Indicates whether another sequence targets the same sequence without
 accounting for PTM localization.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#isSameSequenceAndModificationStatusAs-com.compomics.util.experiment.identification.amino_acid_tags.TagComponent-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameSequenceAndModificationStatusAs(TagComponent, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html#isSameSequenceAndModificationStatusAs-com.compomics.util.experiment.identification.amino_acid_tags.TagComponent-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameSequenceAndModificationStatusAs(TagComponent, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">MassGap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#isSameSequenceAndModificationStatusAs-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameSequenceAndModificationStatusAs(Tag, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Indicates whether this tag is the same as another tag without accounting
 for modification localization.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/TagComponent.html#isSameSequenceAndModificationStatusAs-com.compomics.util.experiment.identification.amino_acid_tags.TagComponent-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isSameSequenceAndModificationStatusAs(TagComponent, SequenceMatchingParameters)</a></span> - Method in interface com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/TagComponent.html" title="interface in com.compomics.util.experiment.identification.amino_acid_tags">TagComponent</a></dt>
<dd>
<div class="block">Indicates whether another component is the same as the component of
 interest.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#isScalePrecursor--">isScalePrecursor()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Indicates whether the precursor shall be scaled for OMSSA.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#isScoringNeeded-int-">isScoringNeeded(int)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Indicates whether a score computation is needed for the given advocate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#isScoringNeeded-java.util.ArrayList-">isScoringNeeded(ArrayList&lt;Integer&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Indicates whether a score computation is needed for the given advocates.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#isScrolling">isScrolling</a></span> - Variable in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">When true this indicates that the user is currently scrolling in the
 table and that the table should not update.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#isScrolling--">isScrolling()</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Indicates whether the table is currently being scrolled.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#isSearchForwardFragmentFirst--">isSearchForwardFragmentFirst()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Indicates whether forward ions (b1) should be searched first.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#isSearchPositiveIons--">isSearchPositiveIons()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Indicates whether positive ions are searched.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#isSearchRewindFragments--">isSearchRewindFragments()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Indicates whether C-terminal fragments should be searched.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#isSelfUpdating--">isSelfUpdating()</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Indicates whether the table is in self update mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandParameter.html#isSequenceMatchingType-java.lang.String-java.lang.String-">isSequenceMatchingType(String, String)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandParameter.html" title="class in com.compomics.software.cli">CommandParameter</a></dt>
<dd>
<div class="block">Returns true of the input is in the provided list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#isSquare--">isSquare()</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/FlamableJFrame.html#isStandAlone--">isStandAlone()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/FlamableJFrame.html" title="class in com.compomics.util.gui">FlamableJFrame</a></dt>
<dd>
<div class="block">Returns true if frame is stand alone.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#isStandardSearch--">isStandardSearch()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Indicates whether a modification can be searched with standard search
 engines, i.e., true if it targets a single amino acid position, false if
 it targets a complex pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#isStarting-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isStarting(String, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether the given amino acid sequence starts with the pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#isStarting-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isStarting(AminoAcidPattern, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether the given amino acid sequence starts with the pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isStpBias--">isStpBias()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Returns whether the stP bias should be triggered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#isSubscriptAnnotationNumbers--">isSubscriptAnnotationNumbers()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns true if the numbers in the peak annotations are to be
 subscripted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#issueCommand-java.lang.String-">issueCommand(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">This method allows the caller to issue a command to the server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#issueCommandCheck-java.lang.String-">issueCommandCheck(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">This method will issue the specified command and throw an exception
 whenever the reply is not equal to success! It basically converts an
 FTP error code into an FtpProtocolException.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#isTargetDecoy--">isTargetDecoy()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the database is concatenated target
 decoy.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#isTargetDecoyNeededForPsmScoring-java.util.ArrayList-">isTargetDecoyNeededForPsmScoring(ArrayList&lt;Integer&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Indicates whether target decoy databases are needed for PSM scoring.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#isTargeted-java.lang.Character-int-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">isTargeted(Character, int, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether the given amino acid at the given index of the pattern
 is targeted without accounting for mutations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iStartXLoc">iStartXLoc</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">When the mouse is dragged, this represents the X-coordinate of the
 starting location.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iStartYLoc">iStartYLoc</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">When the mouse is dragged, this represents the Y-coordinate of the
 starting location.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/interfaces/DBElement.html#isToBeDeleted--">isToBeDeleted()</a></span> - Method in interface com.compomics.util.db.interfaces.<a href="com/compomics/util/db/interfaces/DBElement.html" title="interface in com.compomics.util.db.interfaces">DBElement</a></dt>
<dd>
<div class="block">This method signals whether the DBElement is marked for deletion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iStoredSequence">iStoredSequence</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The Vector that holds a set of stored points from a previously
 established list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iStoredSequenceDatasetIndices">iStoredSequenceDatasetIndices</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The Vector that holds the dataset indices of stored points from a
 previously established list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iStoredSequenceDatasetIndicesMirrored">iStoredSequenceDatasetIndicesMirrored</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The Vector that holds the dataset indices of stored points from a
 previously established list in the mirrored spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iStoredSequenceMirrored">iStoredSequenceMirrored</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The Vector that holds a set of stored points from a previously
 established list in the mirrored spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#isTruncated--">isTruncated()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This method reports on the 'truncatedness' of the protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#isUniqueAa-char-">isUniqueAa(char)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the given character is a supported
 amino acid excluding combinations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#isUseChargeStateFromMS--">isUseChargeStateFromMS()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Indicates whether the charge state from the spectrum should be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#isUseCorrelationCorrectionScore--">isUseCorrelationCorrectionScore()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Indicates whether the correlation correction score should be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#isUseNoiseSuppression--">isUseNoiseSuppression()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Indicates whether noise suppression should be used when importing
 spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#isUserDefined-java.lang.String-">isUserDefined(String)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Convenience method returning a boolean indicating whether a Modification is user
 defined or default.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#isValid--">isValid()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>
<div class="block">Returns true if valid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html#isValidated-java.lang.String-com.compomics.util.experiment.filtering.FilterItemComparator-java.lang.Object-long-com.compomics.util.experiment.identification.Identification-com.compomics.util.experiment.biology.genes.GeneMaps-com.compomics.util.experiment.identification.IdentificationFeaturesGenerator-com.compomics.util.parameters.identification.IdentificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.experiment.io.biology.protein.ProteinDetailsProvider-">isValidated(String, FilterItemComparator, Object, long, Identification, GeneMaps, IdentificationFeaturesGenerator, IdentificationParameters, SequenceProvider, ProteinDetailsProvider)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">AssumptionFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html#isValidated-long-java.lang.String-com.compomics.util.experiment.identification.spectrum_assumptions.PeptideAssumption-com.compomics.util.experiment.identification.Identification-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.experiment.identification.IdentificationFeaturesGenerator-com.compomics.util.parameters.identification.IdentificationParameters-">isValidated(long, String, PeptideAssumption, Identification, SequenceProvider, IdentificationFeaturesGenerator, IdentificationParameters)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">AssumptionFilter</a></dt>
<dd>
<div class="block">Tests whether a match is validated by this filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html#isValidated-java.lang.String-com.compomics.util.experiment.filtering.FilterItemComparator-java.lang.Object-long-java.lang.String-com.compomics.util.experiment.identification.spectrum_assumptions.PeptideAssumption-com.compomics.util.experiment.identification.Identification-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.experiment.identification.IdentificationFeaturesGenerator-com.compomics.util.parameters.identification.IdentificationParameters-">isValidated(String, FilterItemComparator, Object, long, String, PeptideAssumption, Identification, SequenceProvider, IdentificationFeaturesGenerator, IdentificationParameters)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/AssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">AssumptionFilter</a></dt>
<dd>
<div class="block">Indicates whether the match designated by the match key validates the
 given item using the given comparator and value threshold.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#isValidated-long-com.compomics.util.experiment.identification.Identification-com.compomics.util.experiment.biology.genes.GeneMaps-com.compomics.util.experiment.identification.IdentificationFeaturesGenerator-com.compomics.util.parameters.identification.IdentificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.experiment.io.biology.protein.ProteinDetailsProvider-">isValidated(long, Identification, GeneMaps, IdentificationFeaturesGenerator, IdentificationParameters, SequenceProvider, ProteinDetailsProvider)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Tests whether a match is validated by this filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#isValidated-java.lang.String-com.compomics.util.experiment.filtering.FilterItemComparator-java.lang.Object-long-com.compomics.util.experiment.identification.Identification-com.compomics.util.experiment.biology.genes.GeneMaps-com.compomics.util.experiment.identification.IdentificationFeaturesGenerator-com.compomics.util.parameters.identification.IdentificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.experiment.io.biology.protein.ProteinDetailsProvider-">isValidated(String, FilterItemComparator, Object, long, Identification, GeneMaps, IdentificationFeaturesGenerator, IdentificationParameters, SequenceProvider, ProteinDetailsProvider)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Indicates whether the match designated by the match key validates the
 given item using the given comparator and value threshold.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html#isValidated-java.lang.String-com.compomics.util.experiment.filtering.FilterItemComparator-java.lang.Object-long-com.compomics.util.experiment.identification.Identification-com.compomics.util.experiment.biology.genes.GeneMaps-com.compomics.util.experiment.identification.IdentificationFeaturesGenerator-com.compomics.util.parameters.identification.IdentificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.experiment.io.biology.protein.ProteinDetailsProvider-">isValidated(String, FilterItemComparator, Object, long, Identification, GeneMaps, IdentificationFeaturesGenerator, IdentificationParameters, SequenceProvider, ProteinDetailsProvider)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html#isValidated-java.lang.String-com.compomics.util.experiment.filtering.FilterItemComparator-java.lang.Object-long-com.compomics.util.experiment.identification.Identification-com.compomics.util.experiment.biology.genes.GeneMaps-com.compomics.util.experiment.identification.IdentificationFeaturesGenerator-com.compomics.util.parameters.identification.IdentificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.experiment.io.biology.protein.ProteinDetailsProvider-">isValidated(String, FilterItemComparator, Object, long, Identification, GeneMaps, IdentificationFeaturesGenerator, IdentificationParameters, SequenceProvider, ProteinDetailsProvider)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html" title="class in com.compomics.util.experiment.identification.filtering">ProteinFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html#isValidated-java.lang.String-com.compomics.util.experiment.filtering.FilterItemComparator-java.lang.Object-long-com.compomics.util.experiment.identification.Identification-com.compomics.util.experiment.biology.genes.GeneMaps-com.compomics.util.experiment.identification.IdentificationFeaturesGenerator-com.compomics.util.parameters.identification.IdentificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.experiment.io.biology.protein.ProteinDetailsProvider-">isValidated(String, FilterItemComparator, Object, long, Identification, GeneMaps, IdentificationFeaturesGenerator, IdentificationParameters, SequenceProvider, ProteinDetailsProvider)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PsmFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#isValidated--">isValidated()</a></span> - Method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>
<div class="block">Indicates whether this level is considered as validated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#isValidatedOnly--">isValidatedOnly()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Indicates whether only validated results should be exported.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html#isValidModifications-org.apache.commons.cli.CommandLine-">isValidModifications(CommandLine)</a></span> - Static method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersInputBean</a></dt>
<dd>
<div class="block">Verifies that modifications are correctly recognized.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html#isValidStartup-org.apache.commons.cli.CommandLine-">isValidStartup(CommandLine)</a></span> - Static method in class com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIInputBean.html" title="class in com.compomics.cli.enzymes">EnzymesCLIInputBean</a></dt>
<dd>
<div class="block">Verifies the command line start parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/fasta/FastaParametersInputBean.html#isValidStartup-org.apache.commons.cli.CommandLine-">isValidStartup(CommandLine)</a></span> - Static method in class com.compomics.cli.fasta.<a href="com/compomics/cli/fasta/FastaParametersInputBean.html" title="class in com.compomics.cli.fasta">FastaParametersInputBean</a></dt>
<dd>
<div class="block">Verifies the command line start parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html#isValidStartup-org.apache.commons.cli.CommandLine-boolean-">isValidStartup(CommandLine, boolean)</a></span> - Static method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersInputBean</a></dt>
<dd>
<div class="block">Verifies the command line start parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html#isValidStartup-org.apache.commons.cli.CommandLine-">isValidStartup(CommandLine)</a></span> - Static method in class com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html" title="class in com.compomics.cli.modifications">ModificationsCLIInputBean</a></dt>
<dd>
<div class="block">Verifies the command line start parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#isVariant-com.compomics.util.experiment.biology.proteins.Peptide-java.lang.String-">isVariant(Peptide, String)</a></span> - Static method in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the peptide needs variants to be
 mapped to the given protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#iTemplate">iTemplate</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">The ModificationTemplate which holds all the shared information for a modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/ListChooser.html#items">items</a></span> - Variable in class com.compomics.util.gui.utils.user_choice.<a href="com/compomics/util/gui/utils/user_choice/ListChooser.html" title="class in com.compomics.util.gui.utils.user_choice">ListChooser</a></dt>
<dd>
<div class="block">The list of items the user will choose from.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/IteratorFactory.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion"><span class="typeNameLink">IteratorFactory</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/digestion/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion</a></dt>
<dd>
<div class="block">The iterator goes through a sequence and lists possible peptides with their
 fixed modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/IteratorFactory.html#IteratorFactory-java.util.ArrayList-java.lang.Integer-">IteratorFactory(ArrayList&lt;String&gt;, Integer)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/IteratorFactory.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">IteratorFactory</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/IteratorFactory.html#IteratorFactory-java.util.ArrayList-">IteratorFactory(ArrayList&lt;String&gt;)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/IteratorFactory.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">IteratorFactory</a></dt>
<dd>
<div class="block">Constructor with 2 Xs allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#iTitle">iTitle</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This String holds the title (or name) for the enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#iTitle">iTitle</a></span> - Variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">The title for this modification (eg., Oxidation Met).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iTopPadding">iTopPadding</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Effective distance from the panel top border to 5 pixels above the top of
 the highest point (or y-tick mark).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ4Plex_114">iTRAQ4Plex_114</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ 4Plex 114.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ4Plex_115">iTRAQ4Plex_115</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ 4Plex 115.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ4Plex_116">iTRAQ4Plex_116</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ 4Plex 116.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ4Plex_117">iTRAQ4Plex_117</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ 4Plex 117.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ8Plex_113">iTRAQ8Plex_113</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ 8Plex 113.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ8Plex_114">iTRAQ8Plex_114</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ 8Plex 114.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ8Plex_115">iTRAQ8Plex_115</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ 8Plex 115.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ8Plex_116">iTRAQ8Plex_116</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ 8Plex 116.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ8Plex_117">iTRAQ8Plex_117</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ 8Plex 117.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ8Plex_118">iTRAQ8Plex_118</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ 8Plex 118.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ8Plex_119">iTRAQ8Plex_119</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ 8Plex 119.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ8Plex_121">iTRAQ8Plex_121</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ 8Plex 121.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ_145">iTRAQ_145</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ (reporter + balancer).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#iTRAQ_305">iTRAQ_305</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion iTRAQ (reporter + balancer).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iXAxisData">iXAxisData</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">All the x-axis data points.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iXAxisDataInPixels">iXAxisDataInPixels</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This array will hold the x-coordinates in pixels for all the x-axis
 values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iXAxisDataInPixelsMirroredSpectrum">iXAxisDataInPixelsMirroredSpectrum</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This array will hold the x-coordinates in pixels for all the x-axis
 values of the mirrored spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iXAxisDataMirroredSpectrum">iXAxisDataMirroredSpectrum</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">All the x-axis data points for the mirrored spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iXAxisLabel">iXAxisLabel</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The label (and unit between brackets, if available) for the x-axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iXAxisMax">iXAxisMax</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The maximum x-axis value to display.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iXAxisMin">iXAxisMin</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The minimum x-axis value to display.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iXAxisStartAtZero">iXAxisStartAtZero</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This boolean is set to 'true' if the x-axis should start at zero.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iXPadding">iXPadding</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Effective distance from the x-axis to the panel border.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iXScaleUnit">iXScaleUnit</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Scale unit for the X axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iXUnit">iXUnit</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Graphical unit for the X axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iYAxisData">iYAxisData</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">All the y-axis values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iYAxisDataInPixels">iYAxisDataInPixels</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This array will hold the y-coordinates in pixels for all the y-axis
 values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iYAxisDataInPixelsMirroredSpectrum">iYAxisDataInPixelsMirroredSpectrum</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This array will hold the y-coordinates in pixels for all the y-axis
 values of the mirrored spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iYAxisDataMirroredSpectrum">iYAxisDataMirroredSpectrum</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">All the y-axis values for the mirrored spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iYAxisLabel">iYAxisLabel</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The label (and unit between brackets, if available) for the y-axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iYAxisMax">iYAxisMax</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The maximum y-axis value to display.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iYAxisMin">iYAxisMin</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The minimum y-axis value to display.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iYScaleUnit">iYScaleUnit</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Scale unit for the Y axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#iYUnit">iYUnit</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Graphical unit for the Y axis.</div>
</dd>
</dl>
<a name="I:J">
<!--   -->
</a>
<h2 class="title">J</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#J">J</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">J</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Isoleucine or Leucine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html#J--">J()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/J.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">J</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/software/autoupdater/JarFileFilter.html" title="class in com.compomics.software.autoupdater"><span class="typeNameLink">JarFileFilter</span></a> - Class in <a href="com/compomics/software/autoupdater/package-summary.html">com.compomics.software.autoupdater</a></dt>
<dd>
<div class="block">JarFileFilter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/JarFileFilter.html#JarFileFilter--">JarFileFilter()</a></span> - Constructor for class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/JarFileFilter.html" title="class in com.compomics.software.autoupdater">JarFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/software/dialogs/JavaHomeDialog.html" title="class in com.compomics.software.dialogs"><span class="typeNameLink">JavaHomeDialog</span></a> - Class in <a href="com/compomics/software/dialogs/package-summary.html">com.compomics.software.dialogs</a></dt>
<dd>
<div class="block">A dialog for changing the Java Home.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/JavaHomeDialog.html#JavaHomeDialog-javax.swing.JFrame-com.compomics.software.dialogs.JavaHomeOrMemoryDialogParent-javax.swing.JDialog-java.lang.String-">JavaHomeDialog(JFrame, JavaHomeOrMemoryDialogParent, JDialog, String)</a></span> - Constructor for class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/JavaHomeDialog.html" title="class in com.compomics.software.dialogs">JavaHomeDialog</a></dt>
<dd>
<div class="block">Creates a new JavaHomeDialog.</div>
</dd>
<dt><a href="com/compomics/software/dialogs/JavaHomeOrMemoryDialogParent.html" title="interface in com.compomics.software.dialogs"><span class="typeNameLink">JavaHomeOrMemoryDialogParent</span></a> - Interface in <a href="com/compomics/software/dialogs/package-summary.html">com.compomics.software.dialogs</a></dt>
<dd>
<div class="block">Interface for parents of JavaMemoryDialog and JavaHomeDialog.</div>
</dd>
<dt><a href="com/compomics/software/dialogs/JavaMemoryDialog.html" title="class in com.compomics.software.dialogs"><span class="typeNameLink">JavaMemoryDialog</span></a> - Class in <a href="com/compomics/software/dialogs/package-summary.html">com.compomics.software.dialogs</a></dt>
<dd>
<div class="block">A dialog for changing the Java memory setting.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/JavaMemoryDialog.html#JavaMemoryDialog-javax.swing.JFrame-com.compomics.software.dialogs.JavaHomeOrMemoryDialogParent-javax.swing.JDialog-java.lang.String-">JavaMemoryDialog(JFrame, JavaHomeOrMemoryDialogParent, JDialog, String)</a></span> - Constructor for class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/JavaMemoryDialog.html" title="class in com.compomics.software.dialogs">JavaMemoryDialog</a></dt>
<dd>
<div class="block">Creates a new JavaMemoryDialog.</div>
</dd>
<dt><a href="com/compomics/software/dialogs/JavaParametersDialog.html" title="class in com.compomics.software.dialogs"><span class="typeNameLink">JavaParametersDialog</span></a> - Class in <a href="com/compomics/software/dialogs/package-summary.html">com.compomics.software.dialogs</a></dt>
<dd>
<div class="block">A dialog for showing Java memory settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/JavaParametersDialog.html#JavaParametersDialog-javax.swing.JFrame-com.compomics.software.dialogs.JavaHomeOrMemoryDialogParent-javax.swing.JDialog-java.lang.String-boolean-">JavaParametersDialog(JFrame, JavaHomeOrMemoryDialogParent, JDialog, String, boolean)</a></span> - Constructor for class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/JavaParametersDialog.html" title="class in com.compomics.software.dialogs">JavaParametersDialog</a></dt>
<dd>
<div class="block">Creates a new JavaSettingsDialog.</div>
</dd>
<dt><a href="com/compomics/util/gui/JExceptionDialog.html" title="class in com.compomics.util.gui"><span class="typeNameLink">JExceptionDialog</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">This class implements a JDialog for the specific purpose of showing unrecoverable
 errors or exceptions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JExceptionDialog.html#JExceptionDialog-java.awt.Frame-java.lang.String-java.lang.String:A-java.lang.Throwable-">JExceptionDialog(Frame, String, String[], Throwable)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/JExceptionDialog.html" title="class in com.compomics.util.gui">JExceptionDialog</a></dt>
<dd>
<div class="block">This constructor takes all required parameters for the construction, initialization
 and execution of a JExceptionDialog.</div>
</dd>
<dt><a href="com/compomics/util/gui/JFrameForBytes.html" title="class in com.compomics.util.gui"><span class="typeNameLink">JFrameForBytes</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">This class implements a JFrame, made specifically for the
 display of binary data.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JFrameForBytes.html#JFrameForBytes-java.lang.String-byte:A-">JFrameForBytes(String, byte[])</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/JFrameForBytes.html" title="class in com.compomics.util.gui">JFrameForBytes</a></dt>
<dd>
<div class="block">This constructor creates a JFrame for display of the
 binary data (specified as well).</div>
</dd>
<dt><a href="com/compomics/util/gui/JLabelAndComponentPanel.html" title="class in com.compomics.util.gui"><span class="typeNameLink">JLabelAndComponentPanel</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">This class implements a JPanel that lays out a set of JLabel and a JComponent
 next to each other.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JLabelAndComponentPanel.html#JLabelAndComponentPanel-javax.swing.JLabel:A-javax.swing.JComponent:A-">JLabelAndComponentPanel(JLabel[], JComponent[])</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/JLabelAndComponentPanel.html" title="class in com.compomics.util.gui">JLabelAndComponentPanel</a></dt>
<dd>
<div class="block">Constructor which allows the specification of the labels and components
 to lay out.</div>
</dd>
<dt><a href="com/compomics/util/gui/JOptionEditorPane.html" title="class in com.compomics.util.gui"><span class="typeNameLink">JOptionEditorPane</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">A dialog to show HTML content with clickable links inside a JOptionsPane.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JOptionEditorPane.html#JOptionEditorPane--">JOptionEditorPane()</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/JOptionEditorPane.html" title="class in com.compomics.util.gui">JOptionEditorPane</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#jpeg">jpeg</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#JPEG">JPEG</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/JpegFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">JpegFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.jpeg files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/JpegFileFilter.html#JpegFileFilter--">JpegFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/JpegFileFilter.html" title="class in com.compomics.util.io.file.filefilters">JpegFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#jpg">jpg</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#JPG">JPG</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json"><span class="typeNameLink">JsonMarshaller</span></a> - Class in <a href="com/compomics/util/io/json/package-summary.html">com.compomics.util.io.json</a></dt>
<dd>
<div class="block">This class converts non specific objects to the JSON format and vice versa.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#JsonMarshaller--">JsonMarshaller()</a></span> - Constructor for class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">Default constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#JsonMarshaller-java.lang.Class...-">JsonMarshaller(Class...)</a></span> - Constructor for class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/JTableForDB.html" title="class in com.compomics.util.gui"><span class="typeNameLink">JTableForDB</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">This class extends a JTable with specific cellrenderers for fields,
 retrieved from DB.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JTableForDB.html#JTableForDB--">JTableForDB()</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/JTableForDB.html" title="class in com.compomics.util.gui">JTableForDB</a></dt>
<dd>
<div class="block">Constructs a default <code>JTable</code> that is initialized with a default
 data model, a default column model, and a default selection
 model.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JTableForDB.html#JTableForDB-int-int-">JTableForDB(int, int)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/JTableForDB.html" title="class in com.compomics.util.gui">JTableForDB</a></dt>
<dd>
<div class="block">Constructs a <code>JTable</code> with <code>numRows</code>
 and <code>numColumns</code> of empty cells using
 <code>DefaultTableModel</code>.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JTableForDB.html#JTableForDB-javax.swing.table.TableModel-javax.swing.table.TableColumnModel-">JTableForDB(TableModel, TableColumnModel)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/JTableForDB.html" title="class in com.compomics.util.gui">JTableForDB</a></dt>
<dd>
<div class="block">Constructs a <code>JTable</code> that is initialized with
 <code>dm</code> as the data model, <code>cm</code>
 as the column model, and a default selection model.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JTableForDB.html#JTableForDB-javax.swing.table.TableModel-">JTableForDB(TableModel)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/JTableForDB.html" title="class in com.compomics.util.gui">JTableForDB</a></dt>
<dd>
<div class="block">Constructs a <code>JTable</code> that is initialized with
 <code>dm</code> as the data model, a default column model,
 and a default selection model.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JTableForDB.html#JTableForDB-javax.swing.table.TableModel-javax.swing.table.TableColumnModel-javax.swing.ListSelectionModel-">JTableForDB(TableModel, TableColumnModel, ListSelectionModel)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/JTableForDB.html" title="class in com.compomics.util.gui">JTableForDB</a></dt>
<dd>
<div class="block">Constructs a <code>JTable</code> that is initialized with
 <code>dm</code> as the data model, <code>cm</code> as the
 column model, and <code>sm</code> as the selection model.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JTableForDB.html#JTableForDB-java.util.Vector-java.util.Vector-">JTableForDB(Vector, Vector)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/JTableForDB.html" title="class in com.compomics.util.gui">JTableForDB</a></dt>
<dd>
<div class="block">Constructs a <code>JTable</code> to display the values in the
 <code>Vector</code> of <code>Vectors</code>, <code>rowData</code>,
 with column names, <code>columnNames</code>.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JTableForDB.html#JTableForDB-java.lang.Object:A:A-java.lang.Object:A-">JTableForDB(Object[][], Object[])</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/JTableForDB.html" title="class in com.compomics.util.gui">JTableForDB</a></dt>
<dd>
<div class="block">Constructs a <code>JTable</code> to display the values in the two dimensional array,
 <code>rowData</code>, with column names, <code>columnNames</code>.</div>
</dd>
</dl>
<a name="I:K">
<!--   -->
</a>
<h2 class="title">K</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#K">K</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#key">key</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Spectrum key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#key">key</a></span> - Static variable in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/KMeansClustering.html#kMeanCluster-com.compomics.util.waiting.WaitingHandler-">kMeanCluster(WaitingHandler)</a></span> - Method in class com.compomics.util.math.clustering.<a href="com/compomics/util/math/clustering/KMeansClustering.html" title="class in com.compomics.util.math.clustering">KMeansClustering</a></dt>
<dd>
<div class="block">Run the k-means clustering.</div>
</dd>
<dt><a href="com/compomics/util/math/clustering/KMeansClustering.html" title="class in com.compomics.util.math.clustering"><span class="typeNameLink">KMeansClustering</span></a> - Class in <a href="com/compomics/util/math/clustering/package-summary.html">com.compomics.util.math.clustering</a></dt>
<dd>
<div class="block">K-means clustering.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/KMeansClustering.html#KMeansClustering-double:A:A-java.lang.String:A-int-">KMeansClustering(double[][], String[], int)</a></span> - Constructor for class com.compomics.util.math.clustering.<a href="com/compomics/util/math/clustering/KMeansClustering.html" title="class in com.compomics.util.math.clustering">KMeansClustering</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/KMeansClustering.html#KMeansClustering-java.io.File-int-">KMeansClustering(File, int)</a></span> - Constructor for class com.compomics.util.math.clustering.<a href="com/compomics/util/math/clustering/KMeansClustering.html" title="class in com.compomics.util.math.clustering">KMeansClustering</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/math/clustering/settings/KMeansClusteringSettings.html" title="class in com.compomics.util.math.clustering.settings"><span class="typeNameLink">KMeansClusteringSettings</span></a> - Class in <a href="com/compomics/util/math/clustering/settings/package-summary.html">com.compomics.util.math.clustering.settings</a></dt>
<dd>
<div class="block">Settings for k-means clustering.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/settings/KMeansClusteringSettings.html#KMeansClusteringSettings--">KMeansClusteringSettings()</a></span> - Constructor for class com.compomics.util.math.clustering.settings.<a href="com/compomics/util/math/clustering/settings/KMeansClusteringSettings.html" title="class in com.compomics.util.math.clustering.settings">KMeansClusteringSettings</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
</dl>
<a name="I:L">
<!--   -->
</a>
<h2 class="title">L</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#L">L</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html#laodMappingFromFile-java.io.File-com.compomics.util.waiting.WaitingHandler-">laodMappingFromFile(File, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html" title="class in com.compomics.util.experiment.biology.genes.go">GoDomains</a></dt>
<dd>
<div class="block">Reads go mappings from a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.ObjectType.html#large">large</a></span> - Variable in enum com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.ObjectType.html" title="enum in com.compomics.util.experiment.identification">IdentificationFeaturesCache.ObjectType</a></dt>
<dd>
<div class="block">Boolean indicating whether this category contains large objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html#lastFolderKey">lastFolderKey</a></span> - Static variable in class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SequenceDbDetailsDialog</a></dt>
<dd>
<div class="block">The key to use to store FASTA files paths.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#lastFolderKey">lastFolderKey</a></span> - Static variable in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Key to store the last export folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html#lastModified">lastModified</a></span> - Variable in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaSummary</a></dt>
<dd>
<div class="block">The last time the file was modified.</div>
</dd>
<dt><a href="com/compomics/util/io/file/LastSelectedFolder.html" title="class in com.compomics.util.io.file"><span class="typeNameLink">LastSelectedFolder</span></a> - Class in <a href="com/compomics/util/io/file/package-summary.html">com.compomics.util.io.file</a></dt>
<dd>
<div class="block">Convenience class keeping class of the last selected folders.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/LastSelectedFolder.html#LastSelectedFolder-java.lang.String-">LastSelectedFolder(String)</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/LastSelectedFolder.html" title="class in com.compomics.util.io.file">LastSelectedFolder</a></dt>
<dd>
<div class="block">Constructs a new last selected folder class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/LastSelectedFolder.html#LastSelectedFolder--">LastSelectedFolder()</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/LastSelectedFolder.html" title="class in com.compomics.util.io.file">LastSelectedFolder</a></dt>
<dd>
<div class="block">Constructs a new last selected folder class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/BugReport.html#lastSelectedFolderKey">lastSelectedFolderKey</a></span> - Static variable in class com.compomics.util.gui.error_handlers.<a href="com/compomics/util/gui/error_handlers/BugReport.html" title="class in com.compomics.util.gui.error_handlers">BugReport</a></dt>
<dd>
<div class="block">The specific key for bug reports.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/CompomicsWrapper.html#launchTool-java.lang.String-java.io.File-java.lang.String-java.lang.String-">launchTool(String, File, String, String)</a></span> - Method in class com.compomics.software.<a href="com/compomics/software/CompomicsWrapper.html" title="class in com.compomics.software">CompomicsWrapper</a></dt>
<dd>
<div class="block">Starts the launcher by calling the launch method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/CompomicsWrapper.html#launchTool-java.lang.String-java.io.File-java.lang.String-java.lang.String-java.lang.String:A-">launchTool(String, File, String, String, String[])</a></span> - Method in class com.compomics.software.<a href="com/compomics/software/CompomicsWrapper.html" title="class in com.compomics.software">CompomicsWrapper</a></dt>
<dd>
<div class="block">Starts the launcher by calling the launch method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#left">left</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Left index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#length--">length()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns the length of the pattern in amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#length--">length()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Returns the length of the sequence in amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#length">length</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Current peptide sequence length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html#length">length</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">Rank</a></dt>
<dd>
<div class="block">The length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#length--">length()</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Returns the length of the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#lessTablesPrimary">lessTablesPrimary</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Less table for doing an update step according to the LF step.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#lessTablesReversed">lessTablesReversed</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Less table for doing an update step according to the LF step reversed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#letter">letter</a></span> - Variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The single letter code of the atom.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Leucine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Leucine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html#Leucine--">Leucine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Leucine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Leucine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#LEUCINE">LEUCINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Leucine immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#level">level</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">The MS level.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#Li">Li</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The phosphorus atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#LIGHT_AQUA">LIGHT_AQUA</a></span> - Static variable in class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">A soft-tone AQUA colour (r=219, g=238, b=244).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#LIMIT">LIMIT</a></span> - Static variable in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/math/statistics/linear_regression/LinearRegression.html" title="class in com.compomics.util.math.statistics.linear_regression"><span class="typeNameLink">LinearRegression</span></a> - Class in <a href="com/compomics/util/math/statistics/linear_regression/package-summary.html">com.compomics.util.math.statistics.linear_regression</a></dt>
<dd>
<div class="block">Enum of the different implementations for a regression.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/LinearRegression.html#LinearRegression--">LinearRegression()</a></span> - Constructor for class com.compomics.util.math.statistics.linear_regression.<a href="com/compomics/util/math/statistics/linear_regression/LinearRegression.html" title="class in com.compomics.util.math.statistics.linear_regression">LinearRegression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#linePermutation-int-int-">linePermutation(int, int)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#list--">list()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">LIST files on a remote FTP server.</div>
</dd>
<dt><a href="com/compomics/util/gui/utils/user_choice/ListChooser.html" title="class in com.compomics.util.gui.utils.user_choice"><span class="typeNameLink">ListChooser</span></a> - Class in <a href="com/compomics/util/gui/utils/user_choice/package-summary.html">com.compomics.util.gui.utils.user_choice</a></dt>
<dd>
<div class="block">Dialog for choosing an item in a list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/ListChooser.html#ListChooser-java.awt.Frame-java.util.ArrayList-java.lang.String-java.lang.String-java.lang.String-boolean-">ListChooser(Frame, ArrayList&lt;String&gt;, String, String, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.utils.user_choice.<a href="com/compomics/util/gui/utils/user_choice/ListChooser.html" title="class in com.compomics.util.gui.utils.user_choice">ListChooser</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/ListChooser.html#ListChooser-javax.swing.JDialog-java.util.ArrayList-java.lang.String-java.lang.String-java.lang.String-boolean-">ListChooser(JDialog, ArrayList&lt;String&gt;, String, String, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.utils.user_choice.<a href="com/compomics/util/gui/utils/user_choice/ListChooser.html" title="class in com.compomics.util.gui.utils.user_choice">ListChooser</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/Lithium.html" title="class in com.compomics.util.experiment.biology.atoms.impl"><span class="typeNameLink">Lithium</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a></dt>
<dd>
<div class="block">The Lithium atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/impl/Lithium.html#Lithium--">Lithium()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.impl.<a href="com/compomics/util/experiment/biology/atoms/impl/Lithium.html" title="class in com.compomics.util.experiment.biology.atoms.impl">Lithium</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#LITTLE_ENDIAN_LABEL">LITTLE_ENDIAN_LABEL</a></span> - Static variable in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Defines the valid String indicating little endian byte order.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigFunctions.html#ln-java.math.BigDecimal-java.math.MathContext-">ln(BigDecimal, MathContext)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BigFunctions.html" title="class in com.compomics.util.math">BigFunctions</a></dt>
<dd>
<div class="block">Returns the natural logarithm of a big decimal.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigFunctions.html#lnBD-java.math.BigDecimal-java.math.MathContext-">lnBD(BigDecimal, MathContext)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BigFunctions.html" title="class in com.compomics.util.math">BigFunctions</a></dt>
<dd>
<div class="block">Returns the log of a big decimal.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/interfaces/DBElement.html#load-java.sql.Connection-java.util.HashMap-">load(Connection, HashMap)</a></span> - Method in interface com.compomics.util.db.interfaces.<a href="com/compomics/util/db/interfaces/DBElement.html" title="interface in com.compomics.util.db.interfaces">DBElement</a></dt>
<dd>
<div class="block">This method should be called to fill the DBElement with current data
 as present in the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#loadBackedUpModifications-com.compomics.util.parameters.identification.search.SearchParameters-boolean-">loadBackedUpModifications(SearchParameters, boolean)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Verifies that the modifications backed-up in the search parameters are
 loaded and alerts the user in case conflicts are found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#loadDataForRows-java.util.ArrayList-com.compomics.util.waiting.WaitingHandler-">loadDataForRows(ArrayList&lt;Integer&gt;, WaitingHandler)</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Loads the data needed for objects at rows of the given view indexes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#loadEnsemblSpeciesVersions-java.io.File-">loadEnsemblSpeciesVersions(File)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Loads the given Ensembl species file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#loadFromFile-java.io.File-">loadFromFile(File)</a></span> - Static method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Loads an enzyme factory from a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#loadFromFile-java.io.File-">loadFromFile(File)</a></span> - Static method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Loads an enzyme factory from a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/BiomartMapping.html#loadMapping-java.io.File-java.io.File-">loadMapping(File, File)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/BiomartMapping.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">BiomartMapping</a></dt>
<dd>
<div class="block">Loads the species mapping from a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html#loadMapping-java.io.File-">loadMapping(File)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblGenomesSpecies</a></dt>
<dd>
<div class="block">Loads the species mapping from a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html#loadMapping-java.io.File-">loadMapping(File)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblSpecies</a></dt>
<dd>
<div class="block">Loads the species mapping from a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html#loadMapping-java.io.File-">loadMapping(File)</a></span> - Method in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">UniprotTaxonomy</a></dt>
<dd>
<div class="block">Loads the species mapping from a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html#loadMappingsFromFile-java.io.File-com.compomics.util.waiting.WaitingHandler-">loadMappingsFromFile(File, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html" title="class in com.compomics.util.experiment.biology.genes.go">GoMapping</a></dt>
<dd>
<div class="block">Reads go mappings from a BioMart file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#loadObjects-java.util.Collection-com.compomics.util.waiting.WaitingHandler-boolean-">loadObjects(Collection&lt;Long&gt;, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Loads objects from a table in the cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#loadObjects-java.lang.Class-com.compomics.util.waiting.WaitingHandler-boolean-">loadObjects(Class, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Loads all objects from a given class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#loadObjects-java.lang.Class-com.compomics.util.waiting.WaitingHandler-boolean-">loadObjects(Class, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Loads all objects of the class in cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#loadObjects-java.util.ArrayList-com.compomics.util.waiting.WaitingHandler-boolean-">loadObjects(ArrayList&lt;Long&gt;, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Loads all objects of given keys in cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#loadPathParameterFromLine-java.lang.String-">loadPathParameterFromLine(String)</a></span> - Static method in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Loads a path to be set from a line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#loadPathParametersFromFile-java.io.File-">loadPathParametersFromFile(File)</a></span> - Static method in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Loads the path preferences from a text file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PtmToPrideMap.html#loadPtmToPrideMap-com.compomics.util.parameters.identification.search.SearchParameters-">loadPtmToPrideMap(SearchParameters)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PtmToPrideMap.html" title="class in com.compomics.util.pride">PtmToPrideMap</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">the CV term is now part of the PTM object</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#loadUserParameters--">loadUserParameters()</a></span> - Static method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Loads the user parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#log-double-double-">log(double, double)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Returns the log of the input in the desired base.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigFunctions.html#log-java.math.BigDecimal-double-java.math.MathContext-">log(BigDecimal, double, MathContext)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BigFunctions.html" title="class in com.compomics.util.math">BigFunctions</a></dt>
<dd>
<div class="block">Returns the log of the input in the desired base.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/DefaultOutputLoggerImplementation.html#logExceptionalEvent-java.lang.String-">logExceptionalEvent(String)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/DefaultOutputLoggerImplementation.html" title="class in com.compomics.util.general">DefaultOutputLoggerImplementation</a></dt>
<dd>
<div class="block">This method allows the logging of an exceptional event.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/FileLoggerImplementation.html#logExceptionalEvent-java.lang.String-">logExceptionalEvent(String)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/FileLoggerImplementation.html" title="class in com.compomics.util.general">FileLoggerImplementation</a></dt>
<dd>
<div class="block">This method allows the logging of an exceptional event.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Logger.html#logExceptionalEvent-java.lang.String-">logExceptionalEvent(String)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Logger.html" title="interface in com.compomics.util.interfaces">Logger</a></dt>
<dd>
<div class="block">This method allows the logging of an exceptional event.</div>
</dd>
<dt><a href="com/compomics/util/interfaces/Logger.html" title="interface in com.compomics.util.interfaces"><span class="typeNameLink">Logger</span></a> - Interface in <a href="com/compomics/util/interfaces/package-summary.html">com.compomics.util.interfaces</a></dt>
<dd>
<div class="block">This interface describes the behaviour for a class that can log messages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#login-java.lang.String-java.lang.String-">login(String, String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Login user to a host with username <i>user</i> and password
 <i>password</i>.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#login-java.lang.String-">login(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Login user to a host with username <i>user</i> and no password
 such as HP server which uses the form "&lt;username&gt;/&lt;password&gt;,user.&lt;group&gt;.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/DefaultOutputLoggerImplementation.html#logNormalEvent-java.lang.String-">logNormalEvent(String)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/DefaultOutputLoggerImplementation.html" title="class in com.compomics.util.general">DefaultOutputLoggerImplementation</a></dt>
<dd>
<div class="block">This method allows the logging of a 'normal' event.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/FileLoggerImplementation.html#logNormalEvent-java.lang.String-">logNormalEvent(String)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/FileLoggerImplementation.html" title="class in com.compomics.util.general">FileLoggerImplementation</a></dt>
<dd>
<div class="block">This method allows the logging of a 'normal' event.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Logger.html#logNormalEvent-java.lang.String-">logNormalEvent(String)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Logger.html" title="interface in com.compomics.util.interfaces">Logger</a></dt>
<dd>
<div class="block">This method allows the logging of a 'normal' event.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/DefaultOutputLoggerImplementation.html#logTime-java.lang.String-">logTime(String)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/DefaultOutputLoggerImplementation.html" title="class in com.compomics.util.general">DefaultOutputLoggerImplementation</a></dt>
<dd>
<div class="block">This method allows the caller to have the logger record a timestamp (implementation
 specific format and content, although 'date - time' is a good rule of thumb) along with the
 specified message.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/FileLoggerImplementation.html#logTime-java.lang.String-">logTime(String)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/FileLoggerImplementation.html" title="class in com.compomics.util.general">FileLoggerImplementation</a></dt>
<dd>
<div class="block">This method allows the caller to have the logger record a timestamp (implementation
 specific format and content, although 'date - time' is a good rule of thumb) along with the
 specified message.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Logger.html#logTime-java.lang.String-">logTime(String)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Logger.html" title="interface in com.compomics.util.interfaces">Logger</a></dt>
<dd>
<div class="block">This method allows the caller to have the logger record a timestamp (implementation
 specific format and content, although 'date - time' is a good rule of thumb) along with the
 specified message.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#lossesValidated-com.compomics.util.experiment.identification.spectrum_annotation.NeutralLossesMap-com.compomics.util.experiment.biology.ions.Ion-">lossesValidated(NeutralLossesMap, Ion)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the neutral losses of the given
 fragment ion fit the requirement of the given neutral losses map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#ls--">ls()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Folder-list files on a remote FTP server.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Lysine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Lysine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html#Lysine--">Lysine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Lysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Lysine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#LYSINE">LYSINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Lysine immonium ion.</div>
</dd>
</dl>
<a name="I:M">
<!--   -->
</a>
<h2 class="title">M</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#M">M</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#m_strDestination">m_strDestination</a></span> - Variable in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#m_strSource">m_strSource</a></span> - Variable in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#mad-double:A-">mad(double[])</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Method estimating the median absolute deviation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#mad-java.util.ArrayList-">mad(ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Method estimating the median absolute deviation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLI.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLI.html" title="class in com.compomics.cli.enzymes">EnzymesCLI</a></dt>
<dd>
<div class="block">Main method for the EnzymeCLI.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLI.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLI.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersCLI</a></dt>
<dd>
<div class="block">Starts the launcher by calling the launch method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLI.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLI.html" title="class in com.compomics.cli.modifications">ModificationsCLI</a></dt>
<dd>
<div class="block">Main method for the ModificationsCLI.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/peptide_mapper/PeptideMapperCLI.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.cli.peptide_mapper.<a href="com/compomics/cli/peptide_mapper/PeptideMapperCLI.html" title="class in com.compomics.cli.peptide_mapper">PeptideMapperCLI</a></dt>
<dd>
<div class="block">Main class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/scripts_marc/ExtractFirstLines.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.scripts_marc.<a href="com/compomics/scripts_marc/ExtractFirstLines.html" title="class in com.compomics.scripts_marc">ExtractFirstLines</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/SplashScreen.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/SplashScreen.html" title="class in com.compomics.software">SplashScreen</a></dt>
<dd>
<div class="block">The main method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBAccessorGenerator.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.db.<a href="com/compomics/util/db/DBAccessorGenerator.html" title="class in com.compomics.util.db">DBAccessorGenerator</a></dt>
<dd>
<div class="block">The main method allows to run this tool from the command-line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/examples/MirroredSpectraTestDialog.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.examples.<a href="com/compomics/util/examples/MirroredSpectraTestDialog.html" title="class in com.compomics.util.examples">MirroredSpectraTestDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/examples/UtilitiesDemo.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.examples.<a href="com/compomics/util/examples/UtilitiesDemo.html" title="class in com.compomics.util.examples">UtilitiesDemo</a></dt>
<dd>
<div class="block">Starts the UtilitiesDemo.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MzIdentMLIdfileReader</a></dt>
<dd>
<div class="block">Main method for testing purposes only.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/MassCalc.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.general.<a href="com/compomics/util/general/MassCalc.html" title="class in com.compomics.util.general">MassCalc</a></dt>
<dd>
<div class="block">The main method can be used for command-line usage of this class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/Translate.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.general.<a href="com/compomics/util/general/Translate.html" title="class in com.compomics.util.general">Translate</a></dt>
<dd>
<div class="block">Translate a DNA sequence into 6 reading frames.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/AaMass.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/AaMass.html" title="class in com.compomics.util.gui">AaMass</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomChainDialog.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomChainDialog.html" title="class in com.compomics.util.gui.atoms">AtomChainDialog</a></dt>
<dd>
<div class="block">Main method for testing purposes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.gui.isotopic_calculator.<a href="com/compomics/util/gui/isotopic_calculator/IsotopeDistributionGui.html" title="class in com.compomics.util.gui.isotopic_calculator">IsotopeDistributionGui</a></dt>
<dd>
<div class="block">Main method</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JExceptionDialog.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/JExceptionDialog.html" title="class in com.compomics.util.gui">JExceptionDialog</a></dt>
<dd>
<div class="block">This main method is for testing purposes only.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/DateChooser.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/DateChooser.html" title="class in com.compomics.util.gui.utils">DateChooser</a></dt>
<dd>
<div class="block">Creates and opens a new DateChooser dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramDialog.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramDialog.html" title="class in com.compomics.util.gui">VennDiagramDialog</a></dt>
<dd>
<div class="block">Main method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>
<div class="block">Main method starting the foldermonitor that looks for changes to a monitored folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PropertiesManager.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.io.<a href="com/compomics/util/io/PropertiesManager.html" title="class in com.compomics.util.io">PropertiesManager</a></dt>
<dd>
<div class="block">DO NOT RUN.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/KMeansClustering.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.math.clustering.<a href="com/compomics/util/math/clustering/KMeansClustering.html" title="class in com.compomics.util.math.clustering">KMeansClustering</a></dt>
<dd>
<div class="block">Main method for testing purposes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/FindPdbForUniprotAccessions.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.pdbfinder.<a href="com/compomics/util/pdbfinder/FindPdbForUniprotAccessions.html" title="class in com.compomics.util.pdbfinder">FindPdbForUniprotAccessions</a></dt>
<dd>
<div class="block">Main method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#main-java.lang.String:A-">main(String[])</a></span> - Static method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">Tests the DualEnzyme by digesting a hardcoded protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html#makeSequenceIndependant--">makeSequenceIndependant()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">NeutralLossesMap</a></dt>
<dd>
<div class="block">Makes the neutral losses sequence independent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#makeUnique-char:A-">makeUnique(char[])</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Returns an array containing the unique characters of the given array.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIParams.html#mandatory">mandatory</a></span> - Variable in enum com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIParams.html" title="enum in com.compomics.cli.enzymes">EnzymesCLIParams</a></dt>
<dd>
<div class="block">Boolean indicating whether the parameter is mandatory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/fasta/FastaParametersCLIParams.html#mandatory">mandatory</a></span> - Variable in enum com.compomics.cli.fasta.<a href="com/compomics/cli/fasta/FastaParametersCLIParams.html" title="enum in com.compomics.cli.fasta">FastaParametersCLIParams</a></dt>
<dd>
<div class="block">Boolean indicating whether the parameter is mandatory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html#mandatory">mandatory</a></span> - Variable in enum com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html" title="enum in com.compomics.cli.identification_parameters">IdentificationParametersCLIParams</a></dt>
<dd>
<div class="block">Boolean indicating whether the parameter is mandatory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIParams.html#mandatory">mandatory</a></span> - Variable in enum com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIParams.html" title="enum in com.compomics.cli.modifications">ModificationsCLIParams</a></dt>
<dd>
<div class="block">Boolean indicating whether the parameter is mandatory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#MANUAL_SELECTION">MANUAL_SELECTION</a></span> - Static variable in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Name of the manual selection filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#manualValidation">manualValidation</a></span> - Variable in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">The key of the manually validated matches.</div>
</dd>
<dt><a href="com/compomics/util/threading/MapMutex.html" title="class in com.compomics.util.threading"><span class="typeNameLink">MapMutex</span></a>&lt;<a href="com/compomics/util/threading/MapMutex.html" title="type parameter in MapMutex">K</a>&gt; - Class in <a href="com/compomics/util/threading/package-summary.html">com.compomics.util.threading</a></dt>
<dd>
<div class="block">Map of semaphores acting like a map mutex to block some parts of the maps.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/threading/MapMutex.html#MapMutex-java.lang.Integer-java.lang.Integer-java.lang.Integer-">MapMutex(Integer, Integer, Integer)</a></span> - Constructor for class com.compomics.util.threading.<a href="com/compomics/util/threading/MapMutex.html" title="class in com.compomics.util.threading">MapMutex</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/threading/MapMutex.html#MapMutex--">MapMutex()</a></span> - Constructor for class com.compomics.util.threading.<a href="com/compomics/util/threading/MapMutex.html" title="class in com.compomics.util.threading">MapMutex</a></dt>
<dd>
<div class="block">Constructor with one permit per key and no cache limit size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/threading/MapMutex.html#MapMutex-int-">MapMutex(int)</a></span> - Constructor for class com.compomics.util.threading.<a href="com/compomics/util/threading/MapMutex.html" title="class in com.compomics.util.threading">MapMutex</a></dt>
<dd>
<div class="block">Constructor with no cache limit size.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/parameters/MarshallableParameter.html" title="interface in com.compomics.util.experiment.io.parameters"><span class="typeNameLink">MarshallableParameter</span></a> - Interface in <a href="com/compomics/util/experiment/io/parameters/package-summary.html">com.compomics.util.experiment.io.parameters</a></dt>
<dd>
<div class="block">Interface for a parameter which can be marshalled as .par file.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/parameters/MarshallableParameter.Type.html" title="enum in com.compomics.util.experiment.io.parameters"><span class="typeNameLink">MarshallableParameter.Type</span></a> - Enum in <a href="com/compomics/util/experiment/io/parameters/package-summary.html">com.compomics.util.experiment.io.parameters</a></dt>
<dd>
<div class="block">Enum of the type of parameter which can be encountered in a .par file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#mascot">mascot</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Mascot search engine.</div>
</dd>
<dt><a href="com/compomics/util/io/MascotEnzymeReader.html" title="class in com.compomics.util.io"><span class="typeNameLink">MascotEnzymeReader</span></a> - Class in <a href="com/compomics/util/io/package-summary.html">com.compomics.util.io</a></dt>
<dd>
<div class="block">This class will load Enzyme properties from a Mascot
 (<a href="http://www.matrixscience.com" target="_blank">www.matrixscience.com</a>)
 formatted text file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/MascotEnzymeReader.html#MascotEnzymeReader-java.lang.String-">MascotEnzymeReader(String)</a></span> - Constructor for class com.compomics.util.io.<a href="com/compomics/util/io/MascotEnzymeReader.html" title="class in com.compomics.util.io">MascotEnzymeReader</a></dt>
<dd>
<div class="block">The constructor requires that you specify the file from which to load
 the enzyme information.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/MascotEnzymeReader.html#MascotEnzymeReader-java.io.InputStream-">MascotEnzymeReader(InputStream)</a></span> - Constructor for class com.compomics.util.io.<a href="com/compomics/util/io/MascotEnzymeReader.html" title="class in com.compomics.util.io">MascotEnzymeReader</a></dt>
<dd>
<div class="block">The constructor allows you specify to specify an inputstream from which to load
 the enzyme information.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">MascotIdfileReader</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">This IdfileReader reads identifications from a Mascot results file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html#MascotIdfileReader--">MascotIdfileReader()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MascotIdfileReader</a></dt>
<dd>
<div class="block">Default constructor for the purpose of instantiation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html#MascotIdfileReader-java.io.File-">MascotIdfileReader(File)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MascotIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an Mascot dat result file reader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html#MascotIdfileReader-java.io.File-com.compomics.util.waiting.WaitingHandler-">MascotIdfileReader(File, WaitingHandler)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MascotIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MascotIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an Mascot dat csv result file reader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.MassIndexMap.html#mass">mass</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.MassIndexMap.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex.MassIndexMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#mass">mass</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Current mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/utils/StandardMasses.html#mass">mass</a></span> - Variable in enum com.compomics.util.experiment.mass_spectrometry.utils.<a href="com/compomics/util/experiment/mass_spectrometry/utils/StandardMasses.html" title="enum in com.compomics.util.experiment.mass_spectrometry.utils">StandardMasses</a></dt>
<dd>
<div class="block">The mass.</div>
</dd>
<dt><a href="com/compomics/util/general/MassCalc.html" title="class in com.compomics.util.general"><span class="typeNameLink">MassCalc</span></a> - Class in <a href="com/compomics/util/general/package-summary.html">com.compomics.util.general</a></dt>
<dd>
<div class="block">This class takes care of mass calculation, based on a sequence 
 in IUPAC format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/MassCalc.html#MassCalc--">MassCalc()</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/MassCalc.html" title="class in com.compomics.util.general">MassCalc</a></dt>
<dd>
<div class="block">Default constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/MassCalc.html#MassCalc-int-">MassCalc(int)</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/MassCalc.html" title="class in com.compomics.util.general">MassCalc</a></dt>
<dd>
<div class="block">This constructor allows you to specify an identifier to select a
 element list to use for calculating a mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/MassCalc.html#MassCalc-java.lang.String-">MassCalc(String)</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/MassCalc.html" title="class in com.compomics.util.general">MassCalc</a></dt>
<dd>
<div class="block">This constructor allows the caller to use an elementlist of
 its own making.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/MassCalc.html#MassCalc-java.util.HashMap-">MassCalc(HashMap)</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/MassCalc.html" title="class in com.compomics.util.general">MassCalc</a></dt>
<dd>
<div class="block">This constructor allows the caller to initialize the elementlist
 with a HashMap of its own design.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/MassCalc.html#MassCalc-int-java.util.HashMap-">MassCalc(int, HashMap)</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/MassCalc.html" title="class in com.compomics.util.general">MassCalc</a></dt>
<dd>
<div class="block">This constructor allows the caller to supplement (or to replace
 elements in) a built-in elementlist with a HashMap of its own
 design.</div>
</dd>
<dt><a href="com/compomics/util/general/servlet/MassCalcServlet.html" title="class in com.compomics.util.general.servlet"><span class="typeNameLink">MassCalcServlet</span></a> - Class in <a href="com/compomics/util/general/servlet/package-summary.html">com.compomics.util.general.servlet</a></dt>
<dd>
<div class="block">This class implements the MassCalc class as a servlet.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/servlet/MassCalcServlet.html#MassCalcServlet--">MassCalcServlet()</a></span> - Constructor for class com.compomics.util.general.servlet.<a href="com/compomics/util/general/servlet/MassCalcServlet.html" title="class in com.compomics.util.general.servlet">MassCalcServlet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html" title="class in com.compomics.util.gui.spectrum"><span class="typeNameLink">MassErrorBubblePlot</span></a> - Class in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">Creates a MassErrorBubblePlot displaying the mz values vs the mass error with
 the intensity as the size of the bubbles.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html#MassErrorBubblePlot-java.util.ArrayList-java.util.ArrayList-java.util.ArrayList-double-boolean-boolean-">MassErrorBubblePlot(ArrayList&lt;String&gt;, ArrayList&lt;IonMatch[]&gt;, ArrayList&lt;Spectrum&gt;, double, boolean, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorBubblePlot</a></dt>
<dd>
<div class="block">Creates a new MassErrorBubblePlot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html#MassErrorBubblePlot-java.util.ArrayList-java.util.ArrayList-java.util.ArrayList-double-boolean-boolean-boolean-">MassErrorBubblePlot(ArrayList&lt;String&gt;, ArrayList&lt;IonMatch[]&gt;, ArrayList&lt;Spectrum&gt;, double, boolean, boolean, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorBubblePlot</a></dt>
<dd>
<div class="block">Creates a new MassErrorBubblePlot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html#MassErrorBubblePlot-java.util.ArrayList-java.util.ArrayList-java.util.ArrayList-double-double-boolean-boolean-boolean-">MassErrorBubblePlot(ArrayList&lt;String&gt;, ArrayList&lt;IonMatch[]&gt;, ArrayList&lt;Spectrum&gt;, double, double, boolean, boolean, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorBubblePlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorBubblePlot</a></dt>
<dd>
<div class="block">Creates a new MassErrorBubblePlot.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/MassErrorPlot.html" title="class in com.compomics.util.gui.spectrum"><span class="typeNameLink">MassErrorPlot</span></a> - Class in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">Creates a MassErrorPlot displaying the mz values vs the mass error.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorPlot.html#MassErrorPlot-com.compomics.util.experiment.identification.matches.IonMatch:A-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-double-">MassErrorPlot(IonMatch[], Spectrum, double)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorPlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorPlot</a></dt>
<dd>
<div class="block">Creates a new MassErrorPlot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/MassErrorPlot.html#MassErrorPlot-com.compomics.util.experiment.identification.matches.IonMatch:A-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-double-boolean-">MassErrorPlot(IonMatch[], Spectrum, double, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/MassErrorPlot.html" title="class in com.compomics.util.gui.spectrum">MassErrorPlot</a></dt>
<dd>
<div class="block">Creates a new MassErrorPlot.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags"><span class="typeNameLink">MassGap</span></a> - Class in <a href="com/compomics/util/experiment/identification/amino_acid_tags/package-summary.html">com.compomics.util.experiment.identification.amino_acid_tags</a></dt>
<dd>
<div class="block">An undefined mass gap.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html#MassGap-double-">MassGap(double)</a></span> - Constructor for class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">MassGap</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.MassIndexMap.html#MassIndexMap-double-int:A-">MassIndexMap(double, int[])</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.MassIndexMap.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex.MassIndexMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#massShift">massShift</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">m/z shift applied to all theoretic peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#massShiftCTerm">massShiftCTerm</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">C-terminal m/z shift applied to all reverse ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#massShiftNTerm">massShiftNTerm</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">N-terminal m/z shift applied to all forward ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#matches-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">matches(String, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether the pattern matches the given amino acid sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#matches-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">matches(AminoAcidPattern, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether the pattern matches the given amino acid sequence</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#matches-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">matches(String, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Indicates whether the sequence matches the given amino acid sequence in
 size and according to the given matching preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#matches-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidSequence-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">matches(AminoAcidSequence, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Indicates whether the sequence matches the given amino acid sequence in
 size and according to the given matching preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#matchesAt-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">matchesAt(String, SequenceMatchingParameters, int)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether the pattern is found in the given amino acid sequence
 at the given index, where 0 is the first amino acid.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/advanced/MatchesImportParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced"><span class="typeNameLink">MatchesImportParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/advanced/package-summary.html">com.compomics.util.gui.parameters.identification.advanced</a></dt>
<dd>
<div class="block">The PeptideShaker import settings dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/MatchesImportParametersDialog.html#MatchesImportParametersDialog-java.awt.Frame-com.compomics.util.experiment.identification.filtering.PeptideAssumptionFilter-boolean-">MatchesImportParametersDialog(Frame, PeptideAssumptionFilter, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/MatchesImportParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">MatchesImportParametersDialog</a></dt>
<dd>
<div class="block">Creates a new ImportSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/MatchesImportParametersDialog.html#MatchesImportParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.experiment.identification.filtering.PeptideAssumptionFilter-boolean-">MatchesImportParametersDialog(Dialog, Frame, PeptideAssumptionFilter, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/MatchesImportParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">MatchesImportParametersDialog</a></dt>
<dd>
<div class="block">Creates a new ImportSettingsDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#matchesIn-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">matchesIn(String, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether the pattern is found in the given amino acid sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#matchesIn-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">matchesIn(AminoAcidPattern, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Indicates whether the pattern is found in the given amino acid sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#matchesIn-java.lang.String-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">matchesIn(String, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Indicates whether the sequence is found in the given amino acid sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#matchesIn-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidSequence-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">matchesIn(AminoAcidSequence, SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Indicates whether the sequence is found in the given amino acid sequence.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/matches_iterators/MatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators"><span class="typeNameLink">MatchesIterator</span></a> - Class in <a href="com/compomics/util/experiment/identification/matches_iterators/package-summary.html">com.compomics.util.experiment.identification.matches_iterators</a></dt>
<dd>
<div class="block">An abstract iterator class to iterate identification matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/MatchesIterator.html#MatchesIterator-java.lang.Class-com.compomics.util.experiment.identification.Identification-com.compomics.util.waiting.WaitingHandler-boolean-">MatchesIterator(Class, Identification, WaitingHandler, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/MatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">MatchesIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/MatchesIterator.html#MatchesIterator-long:A-java.lang.Class-com.compomics.util.experiment.identification.Identification-com.compomics.util.waiting.WaitingHandler-boolean-java.lang.String-">MatchesIterator(long[], Class, Identification, WaitingHandler, boolean, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/MatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">MatchesIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering"><span class="typeNameLink">MatchFilter</span></a> - Class in <a href="com/compomics/util/experiment/identification/filtering/package-summary.html">com.compomics.util.experiment.identification.filtering</a></dt>
<dd>
<div class="block">Abstract representing a filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#MatchFilter--">MatchFilter()</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.FilterType.html" title="enum in com.compomics.util.experiment.identification.filtering"><span class="typeNameLink">MatchFilter.FilterType</span></a> - Enum in <a href="com/compomics/util/experiment/identification/filtering/package-summary.html">com.compomics.util.experiment.identification.filtering</a></dt>
<dd>
<div class="block">Enum for the type of possible filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#matchInSpectrum-com.compomics.util.experiment.biology.ions.Ion-java.lang.Integer-">matchInSpectrum(Ion, Integer)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Matches a theoretic ion in the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#matchPeak-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.mass_spectrometry.spectra.Peak-">matchPeak(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, SpecificAnnotationParameters, Peak)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">This method matches the potential fragment ions of a given peptide with a
 given peak according to the annotation settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#matchPeak-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-com.compomics.util.experiment.mass_spectrometry.spectra.Peak-">matchPeak(SpecificAnnotationParameters, Peak)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">This method matches the potential fragment ions of a given peptide with a
 given peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#matchReporterIon-com.compomics.util.experiment.biology.ions.Ion-int-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-double-">matchReporterIon(Ion, int, Spectrum, double)</a></span> - Static method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Convenience method to match a reporter ion in a spectrum.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation"><span class="typeNameLink">MatchValidationLevel</span></a> - Enum in <a href="com/compomics/util/experiment/identification/validation/package-summary.html">com.compomics.util.experiment.identification.validation</a></dt>
<dd>
<div class="block">Enum listing the different levels of match validation.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index"><span class="typeNameLink">MatrixContent</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_inference/fm_index/package-summary.html">com.compomics.util.experiment.identification.protein_inference.fm_index</a></dt>
<dd>
<div class="block">Element for the matrix necessary in pattern search of the FMIndex.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-">MatrixContent(int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor almost empty.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-int-int-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-int-">MatrixContent(int, int, int, MatrixContent, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor for simple sequence mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-int-int-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-int-double-int-">MatrixContent(int, int, int, MatrixContent, int, double, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor for simple tag mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-int-int-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-int-double-int-java.lang.String-">MatrixContent(int, int, int, MatrixContent, int, double, int, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor for simple tag mapping with peptide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-int-int-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-int-int-int-char-">MatrixContent(int, int, int, MatrixContent, int, int, int, char)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor for sequence with variants.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-int-int-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-int-int-int:A-char-">MatrixContent(int, int, int, MatrixContent, int, int, int[], char)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor for sequence with variants.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-int-int-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-double-int-int-int-int-int-">MatrixContent(int, int, int, MatrixContent, double, int, int, int, int, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-int-int-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-double-java.lang.String-java.lang.String-int-int-com.compomics.util.experiment.identification.matches.ModificationMatch-java.util.ArrayList-int-int-">MatrixContent(int, int, int, MatrixContent, double, String, String, int, int, ModificationMatch, ArrayList&lt;ModificationMatch&gt;, int, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-int-int-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-double-java.lang.String-java.lang.String-int-int-int-com.compomics.util.experiment.identification.matches.ModificationMatch-java.util.ArrayList-int-">MatrixContent(int, int, int, MatrixContent, double, String, String, int, int, int, ModificationMatch, ArrayList&lt;ModificationMatch&gt;, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-int-int-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-double-int-int-int-int-char-java.lang.String-">MatrixContent(int, int, int, MatrixContent, double, int, int, int, int, char, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-int-int-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-double-int-int-int-int:A-char-java.lang.String-">MatrixContent(int, int, int, MatrixContent, double, int, int, int, int[], char, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-int-int-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-double-java.lang.String-int-int-com.compomics.util.experiment.identification.matches.ModificationMatch-java.util.ArrayList-int-int-char-java.lang.String-">MatrixContent(int, int, int, MatrixContent, double, String, int, int, ModificationMatch, ArrayList&lt;ModificationMatch&gt;, int, int, char, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-int-int-int-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-double-java.lang.String-int-int-com.compomics.util.experiment.identification.matches.ModificationMatch-java.util.ArrayList-int-int:A-char-java.lang.String-">MatrixContent(int, int, int, MatrixContent, double, String, int, int, ModificationMatch, ArrayList&lt;ModificationMatch&gt;, int, int[], char, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#MatrixContent-com.compomics.util.experiment.identification.protein_inference.fm_index.MatrixContent-">MatrixContent(MatrixContent)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Copy constructor.</div>
</dd>
<dt><a href="com/compomics/software/autoupdater/MavenJarFile.html" title="class in com.compomics.software.autoupdater"><span class="typeNameLink">MavenJarFile</span></a> - Class in <a href="com/compomics/software/autoupdater/package-summary.html">com.compomics.software.autoupdater</a></dt>
<dd>
<div class="block">MavenJarFile.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/MavenJarFile.html#MavenJarFile-java.net.URI-">MavenJarFile(URI)</a></span> - Constructor for class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/MavenJarFile.html" title="class in com.compomics.software.autoupdater">MavenJarFile</a></dt>
<dd>
<div class="block">Create a new MavenJarFile object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/MavenJarFile.html#MavenJarFile-java.io.File-">MavenJarFile(File)</a></span> - Constructor for class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/MavenJarFile.html" title="class in com.compomics.software.autoupdater">MavenJarFile</a></dt>
<dd>
<div class="block">Create a new MavenJarFile object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html#MAX_DEPTH">MAX_DEPTH</a></span> - Static variable in class com.compomics.util.experiment.identification.modification.scores.<a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html" title="class in com.compomics.util.experiment.identification.modification.scores">PhosphoRS</a></dt>
<dd>
<div class="block">The maximal depth to use per window.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#MAX_MS_AMANDA_FASTA_FILE_NAME_LENGTH">MAX_MS_AMANDA_FASTA_FILE_NAME_LENGTH</a></span> - Static variable in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">The maximum allowed length of the FASTA file name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigMathUtils.html#maxDouble">maxDouble</a></span> - Static variable in class com.compomics.util.math.<a href="com/compomics/util/math/BigMathUtils.html" title="class in com.compomics.util.math">BigMathUtils</a></dt>
<dd>
<div class="block">Big decimal value of the maximal value of a double.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#maxIsotopicCorrection">maxIsotopicCorrection</a></span> - Static variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">The maximal isotope correction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#maxPadding">maxPadding</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The maximum padding (distance between the axes and the border of the
 panel).</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/modification/scores/MDScore.html" title="class in com.compomics.util.experiment.identification.modification.scores"><span class="typeNameLink">MDScore</span></a> - Class in <a href="com/compomics/util/experiment/identification/modification/scores/package-summary.html">com.compomics.util.experiment.identification.modification.scores</a></dt>
<dd>
<div class="block">This class estimates the MD score as described in
 https://www.ncbi.nlm.nih.gov/pubmed/21057138 Note: this implementation of the
 MD score is not restricted to phosphorylation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/scores/MDScore.html#MDScore--">MDScore()</a></span> - Constructor for class com.compomics.util.experiment.identification.modification.scores.<a href="com/compomics/util/experiment/identification/modification/scores/MDScore.html" title="class in com.compomics.util.experiment.identification.modification.scores">MDScore</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#mean-java.util.ArrayList-">mean(ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Convenience method returning the mean of a list of doubles.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html#meanDistance">meanDistance</a></span> - Variable in class com.compomics.util.math.statistics.linear_regression.<a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html" title="class in com.compomics.util.math.statistics.linear_regression">RegressionStatistics</a></dt>
<dd>
<div class="block">The mean of the squared distance of the points to the regression line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#median-double:A-">median(double[])</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Method to estimate the median.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#median-java.util.ArrayList-">median(ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Method to estimate the median.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html#medianDistance">medianDistance</a></span> - Variable in class com.compomics.util.math.statistics.linear_regression.<a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html" title="class in com.compomics.util.math.statistics.linear_regression">RegressionStatistics</a></dt>
<dd>
<div class="block">The median of the squared distance of the points to the regression line.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/linear_regression/regressions/MedianRegression.html" title="class in com.compomics.util.math.statistics.linear_regression.regressions"><span class="typeNameLink">MedianRegression</span></a> - Class in <a href="com/compomics/util/math/statistics/linear_regression/regressions/package-summary.html">com.compomics.util.math.statistics.linear_regression.regressions</a></dt>
<dd>
<div class="block">Performs a robust linear regression.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/regressions/MedianRegression.html#MedianRegression--">MedianRegression()</a></span> - Constructor for class com.compomics.util.math.statistics.linear_regression.regressions.<a href="com/compomics/util/math/statistics/linear_regression/regressions/MedianRegression.html" title="class in com.compomics.util.math.statistics.linear_regression.regressions">MedianRegression</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#medianSorted-java.util.ArrayList-">medianSorted(ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Method to estimate the median of a sorted list.</div>
</dd>
<dt><a href="com/compomics/util/memory/MemoryConsumptionStatus.html" title="class in com.compomics.util.memory"><span class="typeNameLink">MemoryConsumptionStatus</span></a> - Class in <a href="com/compomics/util/memory/package-summary.html">com.compomics.util.memory</a></dt>
<dd>
<div class="block">This class provides information on the memory consumption status.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/memory/MemoryConsumptionStatus.html#MemoryConsumptionStatus--">MemoryConsumptionStatus()</a></span> - Constructor for class com.compomics.util.memory.<a href="com/compomics/util/memory/MemoryConsumptionStatus.html" title="class in com.compomics.util.memory">MemoryConsumptionStatus</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/memory/MemoryConsumptionStatus.html#memoryUsed--">memoryUsed()</a></span> - Static method in class com.compomics.util.memory.<a href="com/compomics/util/memory/MemoryConsumptionStatus.html" title="class in com.compomics.util.memory">MemoryConsumptionStatus</a></dt>
<dd>
<div class="block">Returns the share of memory being used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#merge-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-">merge(AminoAcidPattern)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Simple merger for two patterns.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#merge-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-">merge(AminoAcidPattern, AminoAcidPattern)</a></span> - Static method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Convenience method merging two different patterns (see public void
 merge(AminoAcidPattern otherPattern) for detailed information of the
 merging procedure).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#mergeCharArrays-char:A-char:A-">mergeCharArrays(char[], char[])</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Simple method to merge two character arrays.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/HistogramUtils.html#mergeHistograms-java.util.Collection-">mergeHistograms(Collection&lt;HashMap&lt;Double, Integer&gt;&gt;)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/HistogramUtils.html" title="class in com.compomics.util.math">HistogramUtils</a></dt>
<dd>
<div class="block">Merges histograms in a single histogram by adding the values in each bin.</div>
</dd>
<dt><a href="com/compomics/software/autoupdater/MetaDataXMLParser.html" title="class in com.compomics.software.autoupdater"><span class="typeNameLink">MetaDataXMLParser</span></a> - Class in <a href="com/compomics/software/autoupdater/package-summary.html">com.compomics.software.autoupdater</a></dt>
<dd>
<div class="block">MetaDataXMLParser.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/MetaDataXMLParser.html#MetaDataXMLParser-javax.xml.stream.XMLEventReader-">MetaDataXMLParser(XMLEventReader)</a></span> - Constructor for class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/MetaDataXMLParser.html" title="class in com.compomics.software.autoupdater">MetaDataXMLParser</a></dt>
<dd>
<div class="block">Create a new MetaDataXMLParser.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#metaMorpheus">metaMorpheus</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The MetaMorpheus search engine.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Methionine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Methionine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html#Methionine--">Methionine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Methionine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Methionine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#METHIONINE">METHIONINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Methionine immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/Quantification.html#methodUsed">methodUsed</a></span> - Variable in class com.compomics.util.experiment.quantification.<a href="com/compomics/util/experiment/quantification/Quantification.html" title="class in com.compomics.util.experiment.quantification">Quantification</a></dt>
<dd>
<div class="block">The quantification method used</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#METHYL_R_112">METHYL_R_112</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for methylation of R (PMID: 16335983).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#METHYL_R_115">METHYL_R_115</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for methylation of R (PMID: 16335983).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#METHYL_R_143">METHYL_R_143</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for methylation of R (PMID: 16335983).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#METHYL_R_70">METHYL_R_70</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for methylation of R (PMID: 16335983).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#METHYL_R_87">METHYL_R_87</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for methylation of R.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker"><span class="typeNameLink">Metrics</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_shaker/package-summary.html">com.compomics.util.experiment.identification.peptide_shaker</a></dt>
<dd>
<div class="block">This class contains metrics gathered during the creation of a PeptideShaker project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#Metrics--">Metrics()</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/units/MetricsPrefix.html" title="enum in com.compomics.util.experiment.units"><span class="typeNameLink">MetricsPrefix</span></a> - Enum in <a href="com/compomics/util/experiment/units/package-summary.html">com.compomics.util.experiment.units</a></dt>
<dd>
<div class="block">Enumeration of the metrics prefixes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#mgf">mgf</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#MGF">MGF</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/MgfFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">MgfFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.mgf files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/MgfFileFilter.html#MgfFileFilter--">MgfFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/MgfFileFilter.html" title="class in com.compomics.util.io.file.filefilters">MgfFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfFileIterator.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf"><span class="typeNameLink">MgfFileIterator</span></a> - Class in <a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/package-summary.html">com.compomics.util.experiment.io.mass_spectrometry.mgf</a></dt>
<dd>
<div class="block">An iterator of the spectra in an mgf file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfFileIterator.html#MgfFileIterator-java.io.File-">MgfFileIterator(File)</a></span> - Constructor for class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfFileIterator.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfFileIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf"><span class="typeNameLink">MgfIndex</span></a> - Class in <a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/package-summary.html">com.compomics.util.experiment.io.mass_spectrometry.mgf</a></dt>
<dd>
<div class="block">This class contains the indexes of an mgf file after indexing mapped with the
 title of the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#MgfIndex-java.util.ArrayList-java.util.HashMap-java.util.HashMap-java.util.HashMap-java.lang.String-double-double-double-double-int-int-boolean-boolean-long-">MgfIndex(ArrayList&lt;String&gt;, HashMap&lt;String, Long&gt;, HashMap&lt;String, Integer&gt;, HashMap&lt;Integer, Double&gt;, String, double, double, double, double, int, int, boolean, boolean, long)</a></span> - Constructor for class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#MgfIndex-java.util.ArrayList-java.util.HashMap-java.util.HashMap-java.util.HashMap-java.util.HashMap-java.lang.String-double-double-double-double-int-int-boolean-boolean-long-int-">MgfIndex(ArrayList&lt;String&gt;, HashMap&lt;String, Integer&gt;, HashMap&lt;String, Long&gt;, HashMap&lt;String, Integer&gt;, HashMap&lt;Integer, Double&gt;, String, double, double, double, double, int, int, boolean, boolean, long, int)</a></span> - Constructor for class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf"><span class="typeNameLink">MgfReader</span></a> - Class in <a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/package-summary.html">com.compomics.util.experiment.io.mass_spectrometry.mgf</a></dt>
<dd>
<div class="block">This class will read an MGF file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#MgfReader--">MgfReader()</a></span> - Constructor for class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">General constructor for an mgf reader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/export/AplExporter.html#mgfToApl-java.io.File-java.io.File-com.compomics.util.experiment.mass_spectrometry.FragmentationMethod-int-int-">mgfToApl(File, File, FragmentationMethod, int, int)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.export.<a href="com/compomics/util/experiment/io/mass_spectrometry/export/AplExporter.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.export">AplExporter</a></dt>
<dd>
<div class="block">Writes the content of the given file name into the destination file in
 the APL format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/export/Ms2Exporter.html#mgfToMs2-java.io.File-java.io.File-boolean-">mgfToMs2(File, File, boolean)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.export.<a href="com/compomics/util/experiment/io/mass_spectrometry/export/Ms2Exporter.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.export">Ms2Exporter</a></dt>
<dd>
<div class="block">Writes the content of the given file name into the destination file in
 the ms2 format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html#MIN_DEPTH">MIN_DEPTH</a></span> - Static variable in class com.compomics.util.experiment.identification.modification.scores.<a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html" title="class in com.compomics.util.experiment.identification.modification.scores">PhosphoRS</a></dt>
<dd>
<div class="block">The minimal depth to use per window.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#miniature">miniature</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">If set to true, the y-axis is removed, the y- and x-axis tags are
 removed, and any annotations are hidden.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#minIsotopicCorrection">minIsotopicCorrection</a></span> - Static variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">The minimal isotope correction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigMathUtils.html#minNormalDouble">minNormalDouble</a></span> - Static variable in class com.compomics.util.math.<a href="com/compomics/util/math/BigMathUtils.html" title="class in com.compomics.util.math">BigMathUtils</a></dt>
<dd>
<div class="block">Big decimal value of the minimal normal value of a double.</div>
</dd>
<dt><a href="com/compomics/util/examples/MirroredSpectraTestDialog.html" title="class in com.compomics.util.examples"><span class="typeNameLink">MirroredSpectraTestDialog</span></a> - Class in <a href="com/compomics/util/examples/package-summary.html">com.compomics.util.examples</a></dt>
<dd>
<div class="block">Mirrored spectra test class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/examples/MirroredSpectraTestDialog.html#MirroredSpectraTestDialog-java.awt.Frame-boolean-">MirroredSpectraTestDialog(Frame, boolean)</a></span> - Constructor for class com.compomics.util.examples.<a href="com/compomics/util/examples/MirroredSpectraTestDialog.html" title="class in com.compomics.util.examples">MirroredSpectraTestDialog</a></dt>
<dd>
<div class="block">Creates a new MirroredSpectraTestDialog</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#mkdir-java.lang.String-">mkdir(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Create a directory on the remote system</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableMap.html#model">model</a></span> - Variable in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun">TableMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications"><span class="typeNameLink">Modification</span></a> - Class in <a href="com/compomics/util/experiment/biology/modifications/package-summary.html">com.compomics.util.experiment.biology.modifications</a></dt>
<dd>
<div class="block">This class models a peptide or protein modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#Modification--">Modification()</a></span> - Constructor for class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Constructor for the modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#Modification-com.compomics.util.experiment.biology.modifications.ModificationType-java.lang.String-java.lang.String-com.compomics.util.experiment.biology.atoms.AtomChain-com.compomics.util.experiment.biology.atoms.AtomChain-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-">Modification(ModificationType, String, String, AtomChain, AtomChain, AminoAcidPattern)</a></span> - Constructor for class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Constructor for a reference modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#Modification-com.compomics.util.experiment.biology.modifications.ModificationType-java.lang.String-java.lang.String-com.compomics.util.experiment.biology.atoms.AtomChain-com.compomics.util.experiment.biology.atoms.AtomChain-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-com.compomics.util.pride.CvTerm-">Modification(ModificationType, String, String, AtomChain, AtomChain, AminoAcidPattern, CvTerm)</a></span> - Constructor for class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Constructor for a reference modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#Modification-com.compomics.util.experiment.biology.modifications.ModificationType-java.lang.String-java.lang.Double-java.util.ArrayList-">Modification(ModificationType, String, Double, ArrayList&lt;String&gt;)</a></span> - Constructor for class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Simple constructor for a modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#modification">modification</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Index to the modifications list.</div>
</dd>
<dt><a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces"><span class="typeNameLink">Modification</span></a> - Interface in <a href="com/compomics/util/interfaces/package-summary.html">com.compomics.util.interfaces</a></dt>
<dd>
<div class="block">This interface describes the behaviour for a modification on a peptide or aminoacid sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#MODIFICATION_LOCALIZATION_SEPARATOR">MODIFICATION_LOCALIZATION_SEPARATOR</a></span> - Static variable in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Separator preceding confident localization of the confident localization
 of a modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#MODIFICATION_SEPARATOR">MODIFICATION_SEPARATOR</a></span> - Static variable in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Separator used to separate modifications in peptide keys as string.</div>
</dd>
<dt><a href="com/compomics/util/gui/utils/user_choice/list_choosers/ModificationChooser.html" title="class in com.compomics.util.gui.utils.user_choice.list_choosers"><span class="typeNameLink">ModificationChooser</span></a> - Class in <a href="com/compomics/util/gui/utils/user_choice/list_choosers/package-summary.html">com.compomics.util.gui.utils.user_choice.list_choosers</a></dt>
<dd>
<div class="block">Dialog for choosing an item in a list of modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/list_choosers/ModificationChooser.html#ModificationChooser-java.awt.Frame-java.util.ArrayList-java.lang.String-java.lang.String-java.lang.String-boolean-">ModificationChooser(Frame, ArrayList&lt;String&gt;, String, String, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.utils.user_choice.list_choosers.<a href="com/compomics/util/gui/utils/user_choice/list_choosers/ModificationChooser.html" title="class in com.compomics.util.gui.utils.user_choice.list_choosers">ModificationChooser</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/list_choosers/ModificationChooser.html#ModificationChooser-javax.swing.JDialog-java.util.ArrayList-java.lang.String-java.lang.String-java.lang.String-boolean-">ModificationChooser(JDialog, ArrayList&lt;String&gt;, String, String, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.utils.user_choice.list_choosers.<a href="com/compomics/util/gui/utils/user_choice/list_choosers/ModificationChooser.html" title="class in com.compomics.util.gui.utils.user_choice.list_choosers">ModificationChooser</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/list_choosers/ModificationChooser.html#ModificationChooser-java.awt.Frame-java.util.ArrayList-boolean-">ModificationChooser(Frame, ArrayList&lt;String&gt;, boolean)</a></span> - Constructor for class com.compomics.util.gui.utils.user_choice.list_choosers.<a href="com/compomics/util/gui/utils/user_choice/list_choosers/ModificationChooser.html" title="class in com.compomics.util.gui.utils.user_choice.list_choosers">ModificationChooser</a></dt>
<dd>
<div class="block">Constructor with default values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/list_choosers/ModificationChooser.html#ModificationChooser-javax.swing.JDialog-java.util.ArrayList-boolean-">ModificationChooser(JDialog, ArrayList&lt;String&gt;, boolean)</a></span> - Constructor for class com.compomics.util.gui.utils.user_choice.list_choosers.<a href="com/compomics/util/gui/utils/user_choice/list_choosers/ModificationChooser.html" title="class in com.compomics.util.gui.utils.user_choice.list_choosers">ModificationChooser</a></dt>
<dd>
<div class="block">Constructor with default values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationFactory.html#modificationConversionToString--">modificationConversionToString()</a></span> - Static method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationFactory.html" title="class in com.compomics.util.protein">ModificationFactory</a></dt>
<dd>
<div class="block">Returns a String representation of the modification title to code mappings.</div>
</dd>
<dt><a href="com/compomics/util/gui/modification/ModificationDialog.html" title="class in com.compomics.util.gui.modification"><span class="typeNameLink">ModificationDialog</span></a> - Class in <a href="com/compomics/util/gui/modification/package-summary.html">com.compomics.util.gui.modification</a></dt>
<dd>
<div class="block">This dialog allows the user to create/edit Modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/modification/ModificationDialog.html#ModificationDialog-javax.swing.JDialog-com.compomics.util.experiment.biology.modifications.Modification-boolean-">ModificationDialog(JDialog, Modification, boolean)</a></span> - Constructor for class com.compomics.util.gui.modification.<a href="com/compomics/util/gui/modification/ModificationDialog.html" title="class in com.compomics.util.gui.modification">ModificationDialog</a></dt>
<dd>
<div class="block">Creates a new Modification dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/modification/ModificationDialog.html#ModificationDialog-javax.swing.JFrame-com.compomics.util.pride.PtmToPrideMap-com.compomics.util.experiment.biology.modifications.Modification-boolean-">ModificationDialog(JFrame, PtmToPrideMap, Modification, boolean)</a></span> - Constructor for class com.compomics.util.gui.modification.<a href="com/compomics/util/gui/modification/ModificationDialog.html" title="class in com.compomics.util.gui.modification">ModificationDialog</a></dt>
<dd>
<div class="block">Creates a new Modification dialog.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications"><span class="typeNameLink">ModificationFactory</span></a> - Class in <a href="com/compomics/util/experiment/biology/modifications/package-summary.html">com.compomics.util.experiment.biology.modifications</a></dt>
<dd>
<div class="block">This factory will load Modification from an XML file and provide them on demand as
 standard class.</div>
</dd>
<dt><a href="com/compomics/util/protein/ModificationFactory.html" title="class in com.compomics.util.protein"><span class="typeNameLink">ModificationFactory</span></a> - Class in <a href="com/compomics/util/protein/package-summary.html">com.compomics.util.protein</a></dt>
<dd>
<div class="block">This class can be used to generate a Modification instance from a code or title.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">ModificationFeature</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">Feature based on the modification status of an amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html#ModificationFeature-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.ModificationFeature.Property-">ModificationFeature(ModificationFeature.Property)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ModificationFeature</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">ModificationFeature.Property</span></a> - Enum in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">Enum for the possible features.</div>
</dd>
<dt><a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein"><span class="typeNameLink">ModificationImplementation</span></a> - Class in <a href="com/compomics/util/protein/package-summary.html">com.compomics.util.protein</a></dt>
<dd>
<div class="block">This class implements a specific Modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#ModificationImplementation-com.compomics.util.protein.ModificationTemplate-int-">ModificationImplementation(ModificationTemplate, int)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This constructor allows initialization of all the properties for the modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#ModificationImplementation-java.lang.String-java.lang.String-java.util.HashMap-int-">ModificationImplementation(String, String, HashMap, int)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This constructor allows initialization of all the properties for the modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#ModificationImplementation-java.lang.String-java.lang.String-java.util.HashMap-boolean-int-">ModificationImplementation(String, String, HashMap, boolean, int)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This constructor allows initialization of all the properties for the modification.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced"><span class="typeNameLink">ModificationLocalizationParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/advanced/package-summary.html">com.compomics.util.parameters.identification.advanced</a></dt>
<dd>
<div class="block">This class contains the modification localization preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#ModificationLocalizationParameters--">ModificationLocalizationParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/advanced/ModificationLocalizationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced"><span class="typeNameLink">ModificationLocalizationParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/advanced/package-summary.html">com.compomics.util.gui.parameters.identification.advanced</a></dt>
<dd>
<div class="block">PTMLocalizationParametersDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ModificationLocalizationParametersDialog.html#ModificationLocalizationParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.ModificationLocalizationParameters-boolean-">ModificationLocalizationParametersDialog(Frame, ModificationLocalizationParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ModificationLocalizationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ModificationLocalizationParametersDialog</a></dt>
<dd>
<div class="block">Creates a new PTMLocalizationParametersDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ModificationLocalizationParametersDialog.html#ModificationLocalizationParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.ModificationLocalizationParameters-boolean-">ModificationLocalizationParametersDialog(Dialog, Frame, ModificationLocalizationParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ModificationLocalizationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ModificationLocalizationParametersDialog</a></dt>
<dd>
<div class="block">Creates a new PTMLocalizationParametersDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html" title="enum in com.compomics.util.experiment.identification.modification"><span class="typeNameLink">ModificationLocalizationScore</span></a> - Enum in <a href="com/compomics/util/experiment/identification/modification/package-summary.html">com.compomics.util.experiment.identification.modification</a></dt>
<dd>
<div class="block">An enum of the modification localization scores.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches"><span class="typeNameLink">ModificationMatch</span></a> - Class in <a href="com/compomics/util/experiment/identification/matches/package-summary.html">com.compomics.util.experiment.identification.matches</a></dt>
<dd>
<div class="block">This class models the match between theoretic PTM and identification results.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html#ModificationMatch-java.lang.String-int-">ModificationMatch(String, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches">ModificationMatch</a></dt>
<dd>
<div class="block">Constructor for a modification match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html#ModificationMatch--">ModificationMatch()</a></span> - Constructor for class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches">ModificationMatch</a></dt>
<dd>
<div class="block">Default constructor for a modification match.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search"><span class="typeNameLink">ModificationParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/search/package-summary.html">com.compomics.util.parameters.identification.search</a></dt>
<dd>
<div class="block">This class stores the information about the modification parameters (usage,
 colors, names).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#ModificationParameters--">ModificationParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#ModificationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-">ModificationParameters(ModificationParameters)</a></span> - Constructor for class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Constructor creating a new Modification profile based on the given one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#modificationPos">modificationPos</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">List of all modifications.</div>
</dd>
<dt><a href="com/compomics/util/gui/protein/ModificationProfile.html" title="class in com.compomics.util.gui.protein"><span class="typeNameLink">ModificationProfile</span></a> - Class in <a href="com/compomics/util/gui/protein/package-summary.html">com.compomics.util.gui.protein</a></dt>
<dd>
<div class="block">A simple modification profile object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ModificationProfile.html#ModificationProfile-java.lang.String-double:A:A-java.awt.Color-">ModificationProfile(String, double[][], Color)</a></span> - Constructor for class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ModificationProfile.html" title="class in com.compomics.util.gui.protein">ModificationProfile</a></dt>
<dd>
<div class="block">Modification profile constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#modifications">modifications</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">List of all modifications.</div>
</dd>
<dt><a href="com/compomics/cli/modifications/ModificationsCLI.html" title="class in com.compomics.cli.modifications"><span class="typeNameLink">ModificationsCLI</span></a> - Class in <a href="com/compomics/cli/modifications/package-summary.html">com.compomics.cli.modifications</a></dt>
<dd>
<div class="block">Command line to manage the modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLI.html#ModificationsCLI-com.compomics.cli.modifications.ModificationsCLIInputBean-">ModificationsCLI(ModificationsCLIInputBean)</a></span> - Constructor for class com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLI.html" title="class in com.compomics.cli.modifications">ModificationsCLI</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html" title="class in com.compomics.cli.modifications"><span class="typeNameLink">ModificationsCLIInputBean</span></a> - Class in <a href="com/compomics/cli/modifications/package-summary.html">com.compomics.cli.modifications</a></dt>
<dd>
<div class="block">This class parses the parameters from an EnzymeCLI.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html#ModificationsCLIInputBean-org.apache.commons.cli.CommandLine-">ModificationsCLIInputBean(CommandLine)</a></span> - Constructor for class com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIInputBean.html" title="class in com.compomics.cli.modifications">ModificationsCLIInputBean</a></dt>
<dd>
<div class="block">Parses all the arguments from a command line.</div>
</dd>
<dt><a href="com/compomics/cli/modifications/ModificationsCLIParams.html" title="enum in com.compomics.cli.modifications"><span class="typeNameLink">ModificationsCLIParams</span></a> - Enum in <a href="com/compomics/cli/modifications/package-summary.html">com.compomics.cli.modifications</a></dt>
<dd>
<div class="block">Enum class specifying the ModificationsCLI parameters.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker"><span class="typeNameLink">ModificationScoring</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_shaker/package-summary.html">com.compomics.util.experiment.identification.peptide_shaker</a></dt>
<dd>
<div class="block">This class contains stores the modification localization scores.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#ModificationScoring--">ModificationScoring()</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Default constructor Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#ModificationScoring-java.lang.String-">ModificationScoring(String)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/modification/ModificationsDialog.html" title="class in com.compomics.util.gui.modification"><span class="typeNameLink">ModificationsDialog</span></a> - Class in <a href="com/compomics/util/gui/modification/package-summary.html">com.compomics.util.gui.modification</a></dt>
<dd>
<div class="block">A dialog where the modification details can be modified.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/modification/ModificationsDialog.html#ModificationsDialog-java.awt.Frame-boolean-">ModificationsDialog(Frame, boolean)</a></span> - Constructor for class com.compomics.util.gui.modification.<a href="com/compomics/util/gui/modification/ModificationsDialog.html" title="class in com.compomics.util.gui.modification">ModificationsDialog</a></dt>
<dd>
<div class="block">Creates a new ModificationsDialog.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/modification/ModificationSiteMapping.html" title="class in com.compomics.util.experiment.identification.modification"><span class="typeNameLink">ModificationSiteMapping</span></a> - Class in <a href="com/compomics/util/experiment/identification/modification/package-summary.html">com.compomics.util.experiment.identification.modification</a></dt>
<dd>
<div class="block">This class contains convenience methods to map modifications on potential sites.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationSiteMapping.html#ModificationSiteMapping--">ModificationSiteMapping()</a></span> - Constructor for class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationSiteMapping.html" title="class in com.compomics.util.experiment.identification.modification">ModificationSiteMapping</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationFactory.html#modificationsToString--">modificationsToString()</a></span> - Static method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationFactory.html" title="class in com.compomics.util.protein">ModificationFactory</a></dt>
<dd>
<div class="block">Returns a String representation of the modifications.</div>
</dd>
<dt><a href="com/compomics/util/gui/modification/ModificationTable.html" title="class in com.compomics.util.gui.modification"><span class="typeNameLink">ModificationTable</span></a> - Class in <a href="com/compomics/util/gui/modification/package-summary.html">com.compomics.util.gui.modification</a></dt>
<dd>
<div class="block">Table containing information about the peak annotation of a modified peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/modification/ModificationTable.html#ModificationTable-com.compomics.util.experiment.identification.Identification-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-com.compomics.util.experiment.identification.matches.PeptideMatch-com.compomics.util.experiment.biology.modifications.Modification-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">ModificationTable(Identification, AnnotationParameters, PeptideMatch, Modification, boolean, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Constructor for class com.compomics.util.gui.modification.<a href="com/compomics/util/gui/modification/ModificationTable.html" title="class in com.compomics.util.gui.modification">ModificationTable</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html" title="class in com.compomics.util.experiment.identification.modification"><span class="typeNameLink">ModificationtableContent</span></a> - Class in <a href="com/compomics/util/experiment/identification/modification/package-summary.html">com.compomics.util.experiment.identification.modification</a></dt>
<dd>
<div class="block">Convenience class for the content of a PTM table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html#ModificationtableContent--">ModificationtableContent()</a></span> - Constructor for class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html" title="class in com.compomics.util.experiment.identification.modification">ModificationtableContent</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein"><span class="typeNameLink">ModificationTemplate</span></a> - Class in <a href="com/compomics/util/protein/package-summary.html">com.compomics.util.protein</a></dt>
<dd>
<div class="block">This class holds a template for a Modification, meaning that it contains all shared characteristics of
 a Modification, but not the instance-specific one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#ModificationTemplate-java.lang.String-java.lang.String-java.util.HashMap-">ModificationTemplate(String, String, HashMap)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">This constructor allows initialization of all the properties for the modification template.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#ModificationTemplate-java.lang.String-java.lang.String-java.util.HashMap-boolean-">ModificationTemplate(String, String, HashMap, boolean)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">This constructor allows initialization of all the properties for the modification template.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/modifications/ModificationType.html" title="enum in com.compomics.util.experiment.biology.modifications"><span class="typeNameLink">ModificationType</span></a> - Enum in <a href="com/compomics/util/experiment/biology/modifications/package-summary.html">com.compomics.util.experiment.biology.modifications</a></dt>
<dd>
<div class="block">Enum for the different types of modifications supported.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils"><span class="typeNameLink">ModificationUtils</span></a> - Class in <a href="com/compomics/util/experiment/identification/utils/package-summary.html">com.compomics.util.experiment.identification.utils</a></dt>
<dd>
<div class="block">This class groups functions that can be used to work with modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#ModificationUtils--">ModificationUtils()</a></span> - Constructor for class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html#modProfileEdited--">modProfileEdited()</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">XTandemParametersDialog</a></dt>
<dd>
<div class="block">Returns true if the user edited the modification profile.</div>
</dd>
<dt><a href="com/compomics/util/enumeration/MolecularElement.html" title="enum in com.compomics.util.enumeration"><span class="typeNameLink">MolecularElement</span></a> - Enum in <a href="com/compomics/util/enumeration/package-summary.html">com.compomics.util.enumeration</a></dt>
<dd>
<div class="block">This ENUM types the distinct molecular elements.</div>
</dd>
<dt><a href="com/compomics/util/protein/MolecularFormula.html" title="class in com.compomics.util.protein"><span class="typeNameLink">MolecularFormula</span></a> - Class in <a href="com/compomics/util/protein/package-summary.html">com.compomics.util.protein</a></dt>
<dd>
<div class="block">This class represents the molecular formula.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/MolecularFormula.html#MolecularFormula--">MolecularFormula()</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/MolecularFormula.html" title="class in com.compomics.util.protein">MolecularFormula</a></dt>
<dd>
<div class="block">Default constructor</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/MolecularFormula.html#MolecularFormula-com.compomics.util.interfaces.Sequence-">MolecularFormula(Sequence)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/MolecularFormula.html" title="class in com.compomics.util.protein">MolecularFormula</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/interfaces/Monitorable.html" title="interface in com.compomics.util.interfaces"><span class="typeNameLink">Monitorable</span></a> - Interface in <a href="com/compomics/util/interfaces/package-summary.html">com.compomics.util.interfaces</a></dt>
<dd>
<div class="block">This interface describes the behaviour for any Object whose progressing state
 can be monitored.</div>
</dd>
<dt><a href="com/compomics/util/io/file/MonitorableFileInputStream.html" title="class in com.compomics.util.io.file"><span class="typeNameLink">MonitorableFileInputStream</span></a> - Class in <a href="com/compomics/util/io/file/package-summary.html">com.compomics.util.io.file</a></dt>
<dd>
<div class="block">This class implements an InputStream that reports on the progress of reading
 a file in integer quantities, even though the underlying file may have a binary size that is outside
 of integer range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableFileInputStream.html#MonitorableFileInputStream-java.io.File-">MonitorableFileInputStream(File)</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableFileInputStream.html" title="class in com.compomics.util.io.file">MonitorableFileInputStream</a></dt>
<dd>
<div class="block">This constructor takes a file to load the FileInputStream from.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableFileInputStream.html#MonitorableFileInputStream-java.lang.String-">MonitorableFileInputStream(String)</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableFileInputStream.html" title="class in com.compomics.util.io.file">MonitorableFileInputStream</a></dt>
<dd>
<div class="block">This constructor takes a file to load the FileInputStream from.</div>
</dd>
<dt><a href="com/compomics/util/io/file/MonitorableInputStream.html" title="class in com.compomics.util.io.file"><span class="typeNameLink">MonitorableInputStream</span></a> - Class in <a href="com/compomics/util/io/file/package-summary.html">com.compomics.util.io.file</a></dt>
<dd>
<div class="block">This class provides a specific FilterInputStream that can be monitored for progress.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableInputStream.html#MonitorableInputStream-java.io.InputStream-">MonitorableInputStream(InputStream)</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableInputStream.html" title="class in com.compomics.util.io.file">MonitorableInputStream</a></dt>
<dd>
<div class="block">A MonitorableInputStream can only be constructed around an existing InputStream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableInputStream.html#MonitorableInputStream-java.io.InputStream-int-">MonitorableInputStream(InputStream, int)</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableInputStream.html" title="class in com.compomics.util.io.file">MonitorableInputStream</a></dt>
<dd>
<div class="block">A MonitorableInputStream can only be constructed around an existing InputStream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableInputStream.html#MonitorableInputStream-java.io.InputStream-boolean-">MonitorableInputStream(InputStream, boolean)</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableInputStream.html" title="class in com.compomics.util.io.file">MonitorableInputStream</a></dt>
<dd>
<div class="block">A MonitorableInputStream can only be constructed around an existing InputStream.</div>
</dd>
<dt><a href="com/compomics/util/gui/MonitorDimension.html" title="class in com.compomics.util.gui"><span class="typeNameLink">MonitorDimension</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">This class makes it possible to set the dimmensions of a JFrame of JPanel
 to a certain percentage of the screen size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/MonitorDimension.html#MonitorDimension--">MonitorDimension()</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/MonitorDimension.html" title="class in com.compomics.util.gui">MonitorDimension</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Monitorable.html#monitorProgress--">monitorProgress()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Monitorable.html" title="interface in com.compomics.util.interfaces">Monitorable</a></dt>
<dd>
<div class="block">This method returns the progress of the task.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableFileInputStream.html#monitorProgress--">monitorProgress()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableFileInputStream.html" title="class in com.compomics.util.io.file">MonitorableFileInputStream</a></dt>
<dd>
<div class="block">This method returns the progress (as  measured by taking the value
 of iMaximum, minus the current progress as reported by the cache).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableInputStream.html#monitorProgress--">monitorProgress()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableInputStream.html" title="class in com.compomics.util.io.file">MonitorableInputStream</a></dt>
<dd>
<div class="block">This method returns the progress (as  measured by taking the value
 of iMaximum, minus the current available bytes as reported by the underlying
 InputStream - variable iRemainder).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/MassCalc.html#MONOAA">MONOAA</a></span> - Static variable in class com.compomics.util.general.<a href="com/compomics/util/general/MassCalc.html" title="class in com.compomics.util.general">MassCalc</a></dt>
<dd>
<div class="block">Index for the monoisotopic aminoacid masses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/MassCalc.html#MONOELEMENTS">MONOELEMENTS</a></span> - Static variable in class com.compomics.util.general.<a href="com/compomics/util/general/MassCalc.html" title="class in com.compomics.util.general">MassCalc</a></dt>
<dd>
<div class="block">Index for the monoisotopic masses of the biochemically relevant
 elements.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#MONOISOTOPIC">MONOISOTOPIC</a></span> - Static variable in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">Constant to indicate the position of the monoisotopic mass in the mass array.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#monoisotopicAtomChain">monoisotopicAtomChain</a></span> - Variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">The monoisotopic atom chain.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#monoisotopicMass">monoisotopicMass</a></span> - Variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The monoisotopic mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/MassCalc.html#MONONUCLEOTIDES">MONONUCLEOTIDES</a></span> - Static variable in class com.compomics.util.general.<a href="com/compomics/util/general/MassCalc.html" title="class in com.compomics.util.general">MassCalc</a></dt>
<dd>
<div class="block">Index for the monoisotopic nucleotide masses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#morpheus">morpheus</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Morpheus search engine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/TableMouseWheelListener.html#mouseWheelMoved-java.awt.event.MouseWheelEvent-">mouseWheelMoved(MouseWheelEvent)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/TableMouseWheelListener.html" title="class in com.compomics.util.gui">TableMouseWheelListener</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#ms2">ms2</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#MS2">MS2</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationMethod.html#MS2_IDENTIFICATION">MS2_IDENTIFICATION</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationMethod.html" title="class in com.compomics.util.experiment.identification">IdentificationMethod</a></dt>
<dd>
<div class="block">index for identification method based on MS2 fragment ion matching</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/mass_spectrometry/export/Ms2Exporter.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.export"><span class="typeNameLink">Ms2Exporter</span></a> - Class in <a href="com/compomics/util/experiment/io/mass_spectrometry/export/package-summary.html">com.compomics.util.experiment.io.mass_spectrometry.export</a></dt>
<dd>
<div class="block">This converter writes spectrum files in MS2 format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/export/Ms2Exporter.html#Ms2Exporter--">Ms2Exporter()</a></span> - Constructor for class com.compomics.util.experiment.io.mass_spectrometry.export.<a href="com/compomics/util/experiment/io/mass_spectrometry/export/Ms2Exporter.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.export">Ms2Exporter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/Ms2FileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">Ms2FileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.ms2 files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/Ms2FileFilter.html#Ms2FileFilter--">Ms2FileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/Ms2FileFilter.html" title="class in com.compomics.util.io.file.filefilters">Ms2FileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/Ms2pipFeature.html" title="interface in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration"><span class="typeNameLink">Ms2pipFeature</span></a> - Interface in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration</a></dt>
<dd>
<div class="block">Interface for an ms2pip feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#msAmanda">msAmanda</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The MS Amanda search engine.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">MsAmandaIdfileReader</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">This IdfileReader reads identifications from an MS Amanda csv result file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html#MsAmandaIdfileReader--">MsAmandaIdfileReader()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MsAmandaIdfileReader</a></dt>
<dd>
<div class="block">Default constructor for the purpose of instantiation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html#MsAmandaIdfileReader-java.io.File-">MsAmandaIdfileReader(File)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MsAmandaIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an MS Amanda csv result file reader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html#MsAmandaIdfileReader-java.io.File-com.compomics.util.waiting.WaitingHandler-">MsAmandaIdfileReader(File, WaitingHandler)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MsAmandaIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MsAmandaIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an MS Amanda csv result file reader.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">MsAmandaParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">The MS Amanda specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#MsAmandaParameters--">MsAmandaParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm"><span class="typeNameLink">MsAmandaParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a></dt>
<dd>
<div class="block">Dialog for the MS Amanda specific settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html#MsAmandaParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.MsAmandaParameters-boolean-">MsAmandaParametersDialog(Frame, MsAmandaParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MsAmandaParametersDialog</a></dt>
<dd>
<div class="block">Creates new form MsAmandaSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html#MsAmandaParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.MsAmandaParameters-boolean-">MsAmandaParametersDialog(Dialog, Frame, MsAmandaParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MsAmandaParametersDialog</a></dt>
<dd>
<div class="block">Creates new form MsAmandaSettingsDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html" title="class in com.compomics.util.experiment.mass_spectrometry.proteowizard"><span class="typeNameLink">MsConvertParameters</span></a> - Class in <a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/package-summary.html">com.compomics.util.experiment.mass_spectrometry.proteowizard</a></dt>
<dd>
<div class="block">The parameters to use when running msconvert.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html#MsConvertParameters--">MsConvertParameters()</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html" title="class in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsConvertParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/gui/MsConvertParametersDialog.html" title="class in com.compomics.util.experiment.mass_spectrometry.proteowizard.gui"><span class="typeNameLink">MsConvertParametersDialog</span></a> - Class in <a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/gui/package-summary.html">com.compomics.util.experiment.mass_spectrometry.proteowizard.gui</a></dt>
<dd>
<div class="block">Dialog for the creation and edition of msconvert parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/gui/MsConvertParametersDialog.html#MsConvertParametersDialog-java.awt.Frame-com.compomics.util.experiment.mass_spectrometry.proteowizard.MsConvertParameters-">MsConvertParametersDialog(Frame, MsConvertParameters)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.proteowizard.gui.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/gui/MsConvertParametersDialog.html" title="class in com.compomics.util.experiment.mass_spectrometry.proteowizard.gui">MsConvertParametersDialog</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#msFit">msFit</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The MS-Fit search engine, old school peptide mass fingerprinting.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard"><span class="typeNameLink">MsFormat</span></a> - Enum in <a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/package-summary.html">com.compomics.util.experiment.mass_spectrometry.proteowizard</a></dt>
<dd>
<div class="block">The mass spectrometry formats supported by ProteoWizard.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#msgf">msgf</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The MS-GF+ search engine.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">MsgfParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">The MS-GF+ specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#MsgfParameters--">MsgfParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm"><span class="typeNameLink">MsgfParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a></dt>
<dd>
<div class="block">Dialog for the MS-GF+ specific settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html#MsgfParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.MsgfParameters-boolean-">MsgfParametersDialog(Frame, MsgfParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MsgfParametersDialog</a></dt>
<dd>
<div class="block">Creates new form MsgfSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html#MsgfParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.MsgfParameters-boolean-">MsgfParametersDialog(Dialog, Frame, MsgfParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MsgfParametersDialog</a></dt>
<dd>
<div class="block">Creates new form MsgfSettingsDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspFileIterator.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp"><span class="typeNameLink">MspFileIterator</span></a> - Class in <a href="com/compomics/util/experiment/io/mass_spectrometry/msp/package-summary.html">com.compomics.util.experiment.io.mass_spectrometry.msp</a></dt>
<dd>
<div class="block">An iterator of the spectra in an mgf file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspFileIterator.html#MspFileIterator-java.io.File-">MspFileIterator(File)</a></span> - Constructor for class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspFileIterator.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspFileIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp"><span class="typeNameLink">MspReader</span></a> - Class in <a href="com/compomics/util/experiment/io/mass_spectrometry/msp/package-summary.html">com.compomics.util.experiment.io.mass_spectrometry.msp</a></dt>
<dd>
<div class="block">This class reads NIST .msp file</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html#MspReader--">MspReader()</a></span> - Constructor for class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspReader</a></dt>
<dd>
<div class="block">General Constructor for .msp file format reader</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/MultipleAAPropertyFeature.html" title="interface in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration"><span class="typeNameLink">MultipleAAPropertyFeature</span></a> - Interface in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration</a></dt>
<dd>
<div class="block">Interface for the ms2pip features having multiple amino acid properties.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#myriMatch">myriMatch</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The MyriMatch search engine, old school peptide mass fingerprinting.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">MyriMatchParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">The MyriMatch specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#MyriMatchParameters--">MyriMatchParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm"><span class="typeNameLink">MyriMatchParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a></dt>
<dd>
<div class="block">Dialog for the MyriMatch specific settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html#MyriMatchParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.MyriMatchParameters-boolean-">MyriMatchParametersDialog(Frame, MyriMatchParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MyriMatchParametersDialog</a></dt>
<dd>
<div class="block">Creates new form MyriMatchSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html#MyriMatchParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.MyriMatchParameters-boolean-">MyriMatchParametersDialog(Dialog, Frame, MyriMatchParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MyriMatchParametersDialog</a></dt>
<dd>
<div class="block">Creates new form MyriMatchSettingsDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#mz">mz</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>
<div class="block">The mass over charge ratio of the peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#mzdata">mzdata</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#mzData">mzData</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#mzDATA">mzDATA</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#MZDATA">MZDATA</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/MzDataFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">MzDataFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.mzData files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/MzDataFileFilter.html#MzDataFileFilter--">MzDataFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/MzDataFileFilter.html" title="class in com.compomics.util.io.file.filefilters">MzDataFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">MzIdentMLIdfileReader</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">This IdfileReader reads identifications from an mzIdentML result file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html#MzIdentMLIdfileReader--">MzIdentMLIdfileReader()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MzIdentMLIdfileReader</a></dt>
<dd>
<div class="block">Default constructor for the purpose of instantiation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html#MzIdentMLIdfileReader-java.io.File-">MzIdentMLIdfileReader(File)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MzIdentMLIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an mzIdentML result file reader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html#MzIdentMLIdfileReader-java.io.File-com.compomics.util.waiting.WaitingHandler-">MzIdentMLIdfileReader(File, WaitingHandler)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">MzIdentMLIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an mzIdentML result file reader.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.RawValueConversionType.html" title="enum in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">MzIdentMLIdfileReader.RawValueConversionType</span></a> - Enum in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">Enum for the raw value to e-value conversion.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/MzIdentMLIdfileSearchParametersConverter.html" title="class in com.compomics.util.experiment.io.identification"><span class="typeNameLink">MzIdentMLIdfileSearchParametersConverter</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/package-summary.html">com.compomics.util.experiment.io.identification</a></dt>
<dd>
<div class="block">Reads search parameters from a mzIdentML result files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/MzIdentMLIdfileSearchParametersConverter.html#MzIdentMLIdfileSearchParametersConverter--">MzIdentMLIdfileSearchParametersConverter()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/MzIdentMLIdfileSearchParametersConverter.html" title="class in com.compomics.util.experiment.io.identification">MzIdentMLIdfileSearchParametersConverter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html" title="enum in com.compomics.util.experiment.io.identification"><span class="typeNameLink">MzIdentMLVersion</span></a> - Enum in <a href="com/compomics/util/experiment/io/identification/package-summary.html">com.compomics.util.experiment.io.identification</a></dt>
<dd>
<div class="block">Enum of the versions of the mzIdentML format supported.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#mzML">mzML</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#mzml">mzml</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#MZML">MZML</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/MzMlFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">MzMlFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.mzML files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/MzMlFileFilter.html#MzMlFileFilter--">MzMlFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/MzMlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">MzMlFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#mzTolerance">mzTolerance</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">The m/z tolerance for peak matching.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#mzXML">mzXML</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#mzxml">mzxml</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#MZXML">MZXML</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/MzXmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">MzXmlFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.mzXML files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/MzXmlFileFilter.html#MzXmlFileFilter--">MzXmlFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/MzXmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">MzXmlFileFilter</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="I:N">
<!--   -->
</a>
<h2 class="title">N</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#N">N</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#N">N</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The nitrogen atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#n2sort--">n2sort()</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#N_TERM_ENZYMATIC">N_TERM_ENZYMATIC</a></span> - Static variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#Na">Na</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The sodium atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#name">name</a></span> - Variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Name of the amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.Property.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.Property.html" title="enum in com.compomics.util.experiment.biology.aminoacids">AminoAcid.Property</a></dt>
<dd>
<div class="block">The name of the property.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#name">name</a></span> - Variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The name of the atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#name">name</a></span> - Variable in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">The name of the neutral loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItemComparator.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItemComparator.html" title="enum in com.compomics.util.experiment.filtering">FilterItemComparator</a></dt>
<dd>
<div class="block">The symbol to use.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">AssumptionFilterItem</a></dt>
<dd>
<div class="block">The name of the filtering item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PeptideFilterItem</a></dt>
<dd>
<div class="block">The name of the filtering item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">ProteinFilterItem</a></dt>
<dd>
<div class="block">The name of the filtering item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PsmFilterItem</a></dt>
<dd>
<div class="block">The name of the filtering item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#name">name</a></span> - Variable in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Name of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html" title="enum in com.compomics.util.experiment.identification.matches">IonMatch.MzErrorType</a></dt>
<dd>
<div class="block">The name of the error type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.Property.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">IonFeature.Property</a></dt>
<dd>
<div class="block">The name of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/PeptideFragmentationModel.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.identification.peptide_fragmentation.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/PeptideFragmentationModel.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation">PeptideFragmentationModel</a></dt>
<dd>
<div class="block">The name of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html" title="enum in com.compomics.util.experiment.identification.psm_scoring">PsmScore</a></dt>
<dd>
<div class="block">The name of the score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html" title="enum in com.compomics.util.experiment.io.identification">MzIdentMLVersion</a></dt>
<dd>
<div class="block">The name of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/FragmentationMethod.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/FragmentationMethod.html" title="enum in com.compomics.util.experiment.mass_spectrometry">FragmentationMethod</a></dt>
<dd>
<div class="block">The name of the fragmentation method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsFormat</a></dt>
<dd>
<div class="block">The name of the format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html#name">name</a></span> - Variable in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">ProteoWizardFilter</a></dt>
<dd>
<div class="block">The name of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFormat.html#name">name</a></span> - Variable in enum com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFormat.html" title="enum in com.compomics.util.io.export">ExportFormat</a></dt>
<dd>
<div class="block">The name of the format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html#name">name</a></span> - Variable in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.CleavageParameter</a></dt>
<dd>
<div class="block">The name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html#name">name</a></span> - Variable in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.Specificity</a></dt>
<dd>
<div class="block">The name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/peptide_shaker/ProjectType.html#name">name</a></span> - Variable in enum com.compomics.util.parameters.peptide_shaker.<a href="com/compomics/util/parameters/peptide_shaker/ProjectType.html" title="enum in com.compomics.util.parameters.peptide_shaker">ProjectType</a></dt>
<dd>
<div class="block">The name of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/searchgui/OutputParameters.html#name">name</a></span> - Variable in enum com.compomics.util.parameters.searchgui.<a href="com/compomics/util/parameters/searchgui/OutputParameters.html" title="enum in com.compomics.util.parameters.searchgui">OutputParameters</a></dt>
<dd>
<div class="block">Name of the option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItem.html#needsModifications--">needsModifications()</a></span> - Method in interface com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItem.html" title="interface in com.compomics.util.experiment.filtering">FilterItem</a></dt>
<dd>
<div class="block">Indicates whether the item expects a list of modifications to filter on.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html#needsModifications--">needsModifications()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">AssumptionFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html#needsModifications--">needsModifications()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PeptideFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html#needsModifications--">needsModifications()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">ProteinFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html#needsModifications--">needsModifications()</a></span> - Method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PsmFilterItem</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#needsUpdate-int-java.lang.String-">needsUpdate(int, String)</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Indicates whether the given column needs an update, i.e.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions"><span class="typeNameLink">NeutralLoss</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/package-summary.html">com.compomics.util.experiment.biology.ions</a></dt>
<dd>
<div class="block">This class represents a neutral loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#NeutralLoss-java.lang.String-com.compomics.util.experiment.biology.atoms.AtomChain-boolean-char:A-">NeutralLoss(String, AtomChain, boolean, char[])</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Constructor for a user defined neutral loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#NeutralLoss-java.lang.String-com.compomics.util.experiment.biology.atoms.AtomChain-boolean-">NeutralLoss(String, AtomChain, boolean)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Constructor for a user defined neutral loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#NeutralLoss-java.lang.String-com.compomics.util.experiment.biology.atoms.AtomChain-boolean-char:A-boolean-">NeutralLoss(String, AtomChain, boolean, char[], boolean)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Constructor for a user defined neutral loss.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/NeutralLossCombination.html" title="class in com.compomics.util.experiment.biology.ions"><span class="typeNameLink">NeutralLossCombination</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/package-summary.html">com.compomics.util.experiment.biology.ions</a></dt>
<dd>
<div class="block">This class represents a combination of multiple neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLossCombination.html#NeutralLossCombination-com.compomics.util.experiment.biology.ions.NeutralLoss:A-">NeutralLossCombination(NeutralLoss[])</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLossCombination.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLossCombination</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation"><span class="typeNameLink">NeutralLossesMap</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation</a></dt>
<dd>
<div class="block">This class contains the informations relative to the accounting of neutral
 losses</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html#NeutralLossesMap--">NeutralLossesMap()</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/NeutralLossesMap.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">NeutralLossesMap</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/ElementaryElement.html#neutron">neutron</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/ElementaryElement.html" title="class in com.compomics.util.experiment.biology.atoms">ElementaryElement</a></dt>
<dd>
<div class="block">A neutron.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/ElementaryElement.html#neutronMassMultiples">neutronMassMultiples</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/ElementaryElement.html" title="class in com.compomics.util.experiment.biology.atoms">ElementaryElement</a></dt>
<dd>
<div class="block">Cache for the multiples of the neutron mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#newLine--">newLine()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Adds a separator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#newLine--">newLine()</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/TextWriter.html#newLine--">newLine()</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/TextWriter.html" title="class in com.compomics.util.io.export.writers">TextWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#newLine--">newLine()</a></span> - Method in class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Adds an end of line character.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#newVersionExists-java.lang.Integer-">newVersionExists(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Returns true if a newer version of the species mapping exists in Ensembl.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/WebDAO.html#newVersionReleased-com.compomics.software.autoupdater.MavenJarFile-java.net.URL-">newVersionReleased(MavenJarFile, URL)</a></span> - Static method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/WebDAO.html" title="class in com.compomics.software.autoupdater">WebDAO</a></dt>
<dd>
<div class="block">Returns true if a new version is available.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/PeptideMatchesIterator.html#next--">next()</a></span> - Method in class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/PeptideMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">PeptideMatchesIterator</a></dt>
<dd>
<div class="block">Returns the next match and updates the buffer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/ProteinMatchesIterator.html#next--">next()</a></span> - Method in class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/ProteinMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">ProteinMatchesIterator</a></dt>
<dd>
<div class="block">Returns the next match and updates the buffer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/SpectrumMatchesIterator.html#next--">next()</a></span> - Method in class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/SpectrumMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">SpectrumMatchesIterator</a></dt>
<dd>
<div class="block">Returns the next match and updates the buffer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfFileIterator.html#next--">next()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfFileIterator.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfFileIterator</a></dt>
<dd>
<div class="block">Returns the next spectrum in the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspFileIterator.html#next--">next()</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspFileIterator.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspFileIterator</a></dt>
<dd>
<div class="block">Returns the next spectrum in the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/MatchesIterator.html#nextObject--">nextObject()</a></span> - Method in class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/MatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">MatchesIterator</a></dt>
<dd>
<div class="block">Returns the next match and updates the buffer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#NH3">NH3</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">NH3 loss.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/Nitrogen.html" title="class in com.compomics.util.experiment.biology.atoms.impl"><span class="typeNameLink">Nitrogen</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a></dt>
<dd>
<div class="block">The nitrogen atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/impl/Nitrogen.html#Nitrogen--">Nitrogen()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.impl.<a href="com/compomics/util/experiment/biology/atoms/impl/Nitrogen.html" title="class in com.compomics.util.experiment.biology.atoms.impl">Nitrogen</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#nlist--">nlist()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">NLIST files on a remote FTP server</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/personalization/ExperimentObject.html#NO_KEY">NO_KEY</a></span> - Static variable in class com.compomics.util.experiment.personalization.<a href="com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></dt>
<dd>
<div class="block">Value for a key not set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/genes/SpeciesDialog.html#NO_SPECIES_TAG">NO_SPECIES_TAG</a></span> - Static variable in class com.compomics.util.gui.genes.<a href="com/compomics/util/gui/genes/SpeciesDialog.html" title="class in com.compomics.util.gui.genes">SpeciesDialog</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">The text to use for no species selected.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/NoDigestionCombinationIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators"><span class="typeNameLink">NoDigestionCombinationIterator</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion.iterators</a></dt>
<dd>
<div class="block">Iterator for no digestion of a sequence containing amino acid combinations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/NoDigestionCombinationIterator.html#NoDigestionCombinationIterator-com.compomics.util.experiment.identification.protein_sequences.digestion.ProteinIteratorUtils-java.lang.String-java.lang.Double-java.lang.Double-">NoDigestionCombinationIterator(ProteinIteratorUtils, String, Double, Double)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.iterators.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/NoDigestionCombinationIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">NoDigestionCombinationIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/NoDigestionIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators"><span class="typeNameLink">NoDigestionIterator</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion.iterators</a></dt>
<dd>
<div class="block">Iterator for no digestion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/NoDigestionIterator.html#NoDigestionIterator-com.compomics.util.experiment.identification.protein_sequences.digestion.ProteinIteratorUtils-java.lang.String-double-double-">NoDigestionIterator(ProteinIteratorUtils, String, double, double)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.iterators.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/NoDigestionIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">NoDigestionIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html#noErrors--">noErrors()</a></span> - Method in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html" title="class in com.compomics.util.pride.validation">XMLValidationErrorHandler</a></dt>
<dd>
<div class="block">Returns true of there were no errors.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions"><span class="typeNameLink">NonSymmetricalNormalDistribution</span></a> - Class in <a href="com/compomics/util/math/statistics/distributions/package-summary.html">com.compomics.util.math.statistics.distributions</a></dt>
<dd>
<div class="block">This class represents a non symmetrical normal distribution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html#NonSymmetricalNormalDistribution-double-double-double-">NonSymmetricalNormalDistribution(double, double, double)</a></span> - Constructor for class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NonSymmetricalNormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NonSymmetricalNormalDistribution</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions"><span class="typeNameLink">NormalDistribution</span></a> - Class in <a href="com/compomics/util/math/statistics/distributions/package-summary.html">com.compomics.util.math.statistics.distributions</a></dt>
<dd>
<div class="block">This class represents a normal distribution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html#NormalDistribution-double-double-">NormalDistribution(double, double)</a></span> - Constructor for class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalDistribution.html" title="class in com.compomics.util.math.statistics.distributions">NormalDistribution</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization"><span class="typeNameLink">NormalizationFactors</span></a> - Class in <a href="com/compomics/util/experiment/normalization/package-summary.html">com.compomics.util.experiment.normalization</a></dt>
<dd>
<div class="block">Class grouping the normalization factors used to normalize quantification
 results.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#NormalizationFactors--">NormalizationFactors()</a></span> - Constructor for class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html#normalize--">normalize()</a></span> - Method in class com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationtableContent.html" title="class in com.compomics.util.experiment.identification.modification">ModificationtableContent</a></dt>
<dd>
<div class="block">Normalize intensities.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/distributions/NormalKernelDensityEstimator.html" title="class in com.compomics.util.math.statistics.distributions"><span class="typeNameLink">NormalKernelDensityEstimator</span></a> - Class in <a href="com/compomics/util/math/statistics/distributions/package-summary.html">com.compomics.util.math.statistics.distributions</a></dt>
<dd>
<div class="block">This class makes use of "SSJ: Stochastic Simulation in Java" library from
 iro.umontreal.ca to estimate probability density function of an array of
 double.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/distributions/NormalKernelDensityEstimator.html#NormalKernelDensityEstimator--">NormalKernelDensityEstimator()</a></span> - Constructor for class com.compomics.util.math.statistics.distributions.<a href="com/compomics/util/math/statistics/distributions/NormalKernelDensityEstimator.html" title="class in com.compomics.util.math.statistics.distributions">NormalKernelDensityEstimator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#noSubstitution">noSubstitution</a></span> - Static variable in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Empty substitution matrix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#NOT_FOUND">NOT_FOUND</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Index indicating that the modification was not found.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#NOT_GROUP">NOT_GROUP</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Static index for a protein inference group: 0 - not a protein group or
 unique peptide of single protein group.</div>
</dd>
<dt><a href="com/compomics/util/gui/error_handlers/notification/NotesDialog.html" title="class in com.compomics.util.gui.error_handlers.notification"><span class="typeNameLink">NotesDialog</span></a> - Class in <a href="com/compomics/util/gui/error_handlers/notification/package-summary.html">com.compomics.util.gui.error_handlers.notification</a></dt>
<dd>
<div class="block">A simple dialog for displaying notes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/notification/NotesDialog.html#NotesDialog-java.awt.Frame-boolean-java.util.ArrayList-">NotesDialog(Frame, boolean, ArrayList&lt;String&gt;)</a></span> - Constructor for class com.compomics.util.gui.error_handlers.notification.<a href="com/compomics/util/gui/error_handlers/notification/NotesDialog.html" title="class in com.compomics.util.gui.error_handlers.notification">NotesDialog</a></dt>
<dd>
<div class="block">Creates a new NotesDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/notification/NotificationDialogParent.html#notificationClicked-java.lang.String-">notificationClicked(String)</a></span> - Method in interface com.compomics.util.gui.error_handlers.notification.<a href="com/compomics/util/gui/error_handlers/notification/NotificationDialogParent.html" title="interface in com.compomics.util.gui.error_handlers.notification">NotificationDialogParent</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/error_handlers/notification/NotificationDialog.html" title="class in com.compomics.util.gui.error_handlers.notification"><span class="typeNameLink">NotificationDialog</span></a> - Class in <a href="com/compomics/util/gui/error_handlers/notification/package-summary.html">com.compomics.util.gui.error_handlers.notification</a></dt>
<dd>
<div class="block">An undecorated dialog for displaying notifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/error_handlers/notification/NotificationDialog.html#NotificationDialog-java.awt.Frame-com.compomics.util.gui.error_handlers.notification.NotificationDialogParent-boolean-int-java.lang.String-">NotificationDialog(Frame, NotificationDialogParent, boolean, int, String)</a></span> - Constructor for class com.compomics.util.gui.error_handlers.notification.<a href="com/compomics/util/gui/error_handlers/notification/NotificationDialog.html" title="class in com.compomics.util.gui.error_handlers.notification">NotificationDialog</a></dt>
<dd>
<div class="block">Creates a new NotificationDialog.</div>
</dd>
<dt><a href="com/compomics/util/gui/error_handlers/notification/NotificationDialogParent.html" title="interface in com.compomics.util.gui.error_handlers.notification"><span class="typeNameLink">NotificationDialogParent</span></a> - Interface in <a href="com/compomics/util/gui/error_handlers/notification/package-summary.html">com.compomics.util.gui.error_handlers.notification</a></dt>
<dd>
<div class="block">A NotificationDialog parent interface.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/exception_handlers/CommandLineExceptionHandler.html#notifyUser-java.lang.Exception-">notifyUser(Exception)</a></span> - Method in class com.compomics.util.exceptions.exception_handlers.<a href="com/compomics/util/exceptions/exception_handlers/CommandLineExceptionHandler.html" title="class in com.compomics.util.exceptions.exception_handlers">CommandLineExceptionHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/exception_handlers/FrameExceptionHandler.html#notifyUser-java.lang.Exception-">notifyUser(Exception)</a></span> - Method in class com.compomics.util.exceptions.exception_handlers.<a href="com/compomics/util/exceptions/exception_handlers/FrameExceptionHandler.html" title="class in com.compomics.util.exceptions.exception_handlers">FrameExceptionHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/exception_handlers/WaitingDialogExceptionHandler.html#notifyUser-java.lang.Exception-">notifyUser(Exception)</a></span> - Method in class com.compomics.util.exceptions.exception_handlers.<a href="com/compomics/util/exceptions/exception_handlers/WaitingDialogExceptionHandler.html" title="class in com.compomics.util.exceptions.exception_handlers">WaitingDialogExceptionHandler</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/ExceptionHandler.html#notifyUser-java.lang.Exception-">notifyUser(Exception)</a></span> - Method in class com.compomics.util.exceptions.<a href="com/compomics/util/exceptions/ExceptionHandler.html" title="class in com.compomics.util.exceptions">ExceptionHandler</a></dt>
<dd>
<div class="block">Notifies the user that an exception was caught.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#novor">novor</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Novor de novo sequencing algorithm.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">NovorIdfileReader</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">This IdfileReader reads identifications from a Novor csv result file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html#NovorIdfileReader--">NovorIdfileReader()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">NovorIdfileReader</a></dt>
<dd>
<div class="block">Default constructor for the purpose of instantiation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html#NovorIdfileReader-java.io.File-">NovorIdfileReader(File)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">NovorIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an Novor csv result file reader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html#NovorIdfileReader-java.io.File-com.compomics.util.waiting.WaitingHandler-">NovorIdfileReader(File, WaitingHandler)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/NovorIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">NovorIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an Novor csv result file reader.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">NovorParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">Novor specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html#NovorParameters--">NovorParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">NovorParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm"><span class="typeNameLink">NovorParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a></dt>
<dd>
<div class="block">Dialog for editing Novor advanced settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html#NovorParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.NovorParameters-boolean-">NovorParametersDialog(Frame, NovorParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">NovorParametersDialog</a></dt>
<dd>
<div class="block">Creates a new NovorSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html#NovorParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.NovorParameters-boolean-">NovorParametersDialog(Dialog, Frame, NovorParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">NovorParametersDialog</a></dt>
<dd>
<div class="block">Creates a new NovorSettingsDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html#nSequences">nSequences</a></span> - Variable in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaSummary</a></dt>
<dd>
<div class="block">The number of sequences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#nSpectraInCache-long-">nSpectraInCache(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Indicates whether the number of spectra for a given protein match is in
 cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html#nTarget">nTarget</a></span> - Variable in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaSummary</a></dt>
<dd>
<div class="block">The number of target sequences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#NTERM">NTERM</a></span> - Static variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html#nTermCorrection">nTermCorrection</a></span> - Variable in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/DirecTagIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">DirecTagIdfileReader</a></dt>
<dd>
<div class="block">The mass to add to the N-terminal gap so that is corresponds to a peptide
 fragment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#nTermCorrection">nTermCorrection</a></span> - Variable in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>
<div class="block">The mass to add to the N-terminal gap so that is corresponds to a peptide
 fragment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#NTERMINAL">NTERMINAL</a></span> - Static variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">The code for N-terminal position in the resultant peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#NTERMINUS">NTERMINUS</a></span> - Static variable in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">The residue code for the N-terminus.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#NTERMTRUNC">NTERMTRUNC</a></span> - Static variable in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">The code for an N-terminal truncation.</div>
</dd>
<dt><a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide"><span class="typeNameLink">NucleotideSequence</span></a> - Class in <a href="com/compomics/util/nucleotide/package-summary.html">com.compomics.util.nucleotide</a></dt>
<dd>
<div class="block">This class combines a protein Header with a nucleotide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequence.html#NucleotideSequence-com.compomics.util.nucleotide.NucleotideSequenceImpl-">NucleotideSequence(NucleotideSequenceImpl)</a></span> - Constructor for class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide">NucleotideSequence</a></dt>
<dd>
<div class="block">This constructor requires a NucleotideSequenceImpl as argument.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequence.html#NucleotideSequence-com.compomics.util.experiment.io.biology.protein.Header-com.compomics.util.nucleotide.NucleotideSequenceImpl-">NucleotideSequence(Header, NucleotideSequenceImpl)</a></span> - Constructor for class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide">NucleotideSequence</a></dt>
<dd>
<div class="block">This constructor allows the passing of a Header, as well as an
 AASequenceImpl for this Protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequence.html#NucleotideSequence-java.lang.String-">NucleotideSequence(String)</a></span> - Constructor for class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide">NucleotideSequence</a></dt>
<dd>
<div class="block">This constructor allows for the construction of a NucleotideSequence instance
 by passing a FASTA entry.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequence.html#NucleotideSequence-java.lang.String-java.lang.String-">NucleotideSequence(String, String)</a></span> - Constructor for class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide">NucleotideSequence</a></dt>
<dd>
<div class="block">This constructor allows for the construction of a NucleotideSequence instance through the
 passing of a NucleotideSequence String and a Sequence String.</div>
</dd>
<dt><a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html" title="class in com.compomics.util.nucleotide"><span class="typeNameLink">NucleotideSequenceImpl</span></a> - Class in <a href="com/compomics/util/nucleotide/package-summary.html">com.compomics.util.nucleotide</a></dt>
<dd>
<div class="block">This class represents a nucleotide sequence (RNA or DNA).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html#NucleotideSequenceImpl-java.lang.String-">NucleotideSequenceImpl(String)</a></span> - Constructor for class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html" title="class in com.compomics.util.nucleotide">NucleotideSequenceImpl</a></dt>
<dd>
<div class="block">This constructor allows the construction of an Object,
 wrapping a nucleotidesequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html#NucleotideSequenceImpl-java.lang.String-java.util.Properties-">NucleotideSequenceImpl(String, Properties)</a></span> - Constructor for class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html" title="class in com.compomics.util.nucleotide">NucleotideSequenceImpl</a></dt>
<dd>
<div class="block">This constructor allows the construction of an Object,
 wrapping a nucleotidesequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html#NucleotideSequenceImpl-java.lang.String-java.lang.String-">NucleotideSequenceImpl(String, String)</a></span> - Constructor for class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html" title="class in com.compomics.util.nucleotide">NucleotideSequenceImpl</a></dt>
<dd>
<div class="block">This constructor allows the construction of an Object,
 wrapping a nucleotidesequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html#number">number</a></span> - Variable in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">ProteoWizardFilter</a></dt>
<dd>
<div class="block">The index of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#numSpecificVariants">numSpecificVariants</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">The specific variants.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#numVariants">numVariants</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">List of all modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#numX">numX</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Current number of contained X's.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#nValidatedPeptidesInCache-long-">nValidatedPeptidesInCache(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Indicates whether the number of validated peptides is in cache for a
 given protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#nValidatedSpectraForPeptideInCache-long-">nValidatedSpectraForPeptideInCache(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Indicates whether the number of validated spectra for a peptide match is
 in cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#nValidatedSpectraInCache-long-">nValidatedSpectraInCache(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Indicates whether the number of validated spectra is in cache for the
 given protein match.</div>
</dd>
</dl>
<a name="I:O">
<!--   -->
</a>
<h2 class="title">O</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#O">O</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#O">O</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The oxygen atom.</div>
</dd>
<dt><a href="com/compomics/util/threading/ObjectMutex.html" title="class in com.compomics.util.threading"><span class="typeNameLink">ObjectMutex</span></a> - Class in <a href="com/compomics/util/threading/package-summary.html">com.compomics.util.threading</a></dt>
<dd>
<div class="block">This mutex can be used to manage threads editing experiment objects using
 their key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/threading/ObjectMutex.html#ObjectMutex--">ObjectMutex()</a></span> - Constructor for class com.compomics.util.threading.<a href="com/compomics/util/threading/ObjectMutex.html" title="class in com.compomics.util.threading">ObjectMutex</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object"><span class="typeNameLink">ObjectsCache</span></a> - Class in <a href="com/compomics/util/db/object/package-summary.html">com.compomics.util.db.object</a></dt>
<dd>
<div class="block">An object cache can be combined to an ObjectDB to improve its performance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#ObjectsCache-com.compomics.util.db.object.ObjectsDB-">ObjectsCache(ObjectsDB)</a></span> - Constructor for class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object"><span class="typeNameLink">ObjectsDB</span></a> - Class in <a href="com/compomics/util/db/object/package-summary.html">com.compomics.util.db.object</a></dt>
<dd>
<div class="block">A database which can easily be used to store objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#ObjectsDB-java.lang.String-java.lang.String-">ObjectsDB(String, String)</a></span> - Constructor for class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#ObjectsDB-java.lang.String-java.lang.String-boolean-">ObjectsDB(String, String, boolean)</a></span> - Constructor for class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#observableCoverageInCache-long-">observableCoverageInCache(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Indicates whether the observable coverage of a protein match is in cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#occurrenceTablesPrimary">occurrenceTablesPrimary</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Wavelet tree for storing the burrows wheeler transform.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#occurrenceTablesReversed">occurrenceTablesReversed</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Wavelet tree for storing the burrows wheeler transform reversed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#oldCleave-com.compomics.util.protein.Protein-">oldCleave(Protein)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">This cleave method will process sequence XDYRZ solely into
 YR peptides.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#omssa">omssa</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The OMSSA search engine.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">OmssaParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">The OMSSA specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#OmssaParameters--">OmssaParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm"><span class="typeNameLink">OmssaParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a></dt>
<dd>
<div class="block">Dialog for the OMSSA specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html#OmssaParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.OmssaParameters-boolean-">OmssaParametersDialog(Frame, OmssaParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">OmssaParametersDialog</a></dt>
<dd>
<div class="block">Creates new form OmssaParametersDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html#OmssaParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.OmssaParameters-boolean-">OmssaParametersDialog(Dialog, Frame, OmssaParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">OmssaParametersDialog</a></dt>
<dd>
<div class="block">Creates new form OmssaParametersDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#omx">omx</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#OMX">OMX</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/OmxFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">OmxFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.omx files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/OmxFileFilter.html#OmxFileFilter--">OmxFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/OmxFileFilter.html" title="class in com.compomics.util.io.file.filefilters">OmxFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#onyaseEngine">onyaseEngine</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Onyase engine script.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">OnyaseIdfileReader</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">Id file reader for the scripts in Onyase.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#OnyaseIdfileReader--">OnyaseIdfileReader()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>
<div class="block">Default constructor instantiation purposes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#OnyaseIdfileReader-java.io.File-">OnyaseIdfileReader(File)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an onyase file reader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#openDataConnection-java.lang.String-">openDataConnection(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">This method will set up the networking for client-server data transfer and
 it will send the specified command to the server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#openServer-java.lang.String-">openServer(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">open a FTP connection to host <i>host</i>.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#openServer-java.lang.String-int-">openServer(String, int)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Open an FTP connection to host <i>host</i> on port <i>port</i>.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/examples/BareBonesBrowserLaunch.html#openURL-java.lang.String-">openURL(String)</a></span> - Static method in class com.compomics.util.examples.<a href="com/compomics/util/examples/BareBonesBrowserLaunch.html" title="class in com.compomics.util.examples">BareBonesBrowserLaunch</a></dt>
<dd>
<div class="block">Tries to opens the given URL in the default browser.</div>
</dd>
<dt><a href="com/compomics/util/enumeration/OperatingSystemEnum.html" title="enum in com.compomics.util.enumeration"><span class="typeNameLink">OperatingSystemEnum</span></a> - Enum in <a href="com/compomics/util/enumeration/package-summary.html">com.compomics.util.enumeration</a></dt>
<dd>
<div class="block">This ENUM types the distinct OS systems.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AminoAcidSubstitution.html#originalAminoAcid">originalAminoAcid</a></span> - Variable in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AminoAcidSubstitution.html" title="class in com.compomics.util.experiment.biology.variants">AminoAcidSubstitution</a></dt>
<dd>
<div class="block">The original amino acid represented by its single letter code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#out">out</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#OUT">OUT</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/OutFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">OutFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.out files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/OutFileFilter.html#OutFileFilter--">OutFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/OutFileFilter.html" title="class in com.compomics.util.io.file.filefilters">OutFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#outputData--">outputData()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Indicates whether data should be copied along with the identification
 files in the SearchGUI output.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#outputFolderOption">outputFolderOption</a></span> - Static variable in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">The command line argument for an output folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html#outputFormat">outputFormat</a></span> - Variable in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsFormat</a></dt>
<dd>
<div class="block">Boolean indicating if this in a format that can be used as output.</div>
</dd>
<dt><a href="com/compomics/util/parameters/searchgui/OutputParameters.html" title="enum in com.compomics.util.parameters.searchgui"><span class="typeNameLink">OutputParameters</span></a> - Enum in <a href="com/compomics/util/parameters/searchgui/package-summary.html">com.compomics.util.parameters.searchgui</a></dt>
<dd>
<div class="block">This enum lists the possible output options.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/Oxygen.html" title="class in com.compomics.util.experiment.biology.atoms.impl"><span class="typeNameLink">Oxygen</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a></dt>
<dd>
<div class="block">The oxygen atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/impl/Oxygen.html#Oxygen--">Oxygen()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.impl.<a href="com/compomics/util/experiment/biology/atoms/impl/Oxygen.html" title="class in com.compomics.util.experiment.biology.atoms.impl">Oxygen</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
</dl>
<a name="I:P">
<!--   -->
</a>
<h2 class="title">P</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#P">P</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#P">P</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The phosphorus atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#padding">padding</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The padding (distance between the axes and the border of the panel).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/SequenceModificationPanel.html#paint-java.awt.Graphics-">paint(Graphics)</a></span> - Method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/SequenceModificationPanel.html" title="class in com.compomics.util.gui.protein">SequenceModificationPanel</a></dt>
<dd>
<div class="block">Paints the SequenceModificationPanel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#paint-java.awt.Graphics-">paint(Graphics)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Invoked by Swing to draw components.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SequenceFragmentationPanel.html#paint-java.awt.Graphics-">paint(Graphics)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SequenceFragmentationPanel.html" title="class in com.compomics.util.gui.spectrum">SequenceFragmentationPanel</a></dt>
<dd>
<div class="block">Paints the SequenceFragmentationPanel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html#PARAMETERS_EXTENSION">PARAMETERS_EXTENSION</a></span> - Static variable in class com.compomics.util.experiment.identification.identification_parameters.<a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html" title="class in com.compomics.util.experiment.identification.identification_parameters">IdentificationParametersFactory</a></dt>
<dd>
<div class="block">The extension for a parameters file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html#PARAMETERS_FOLDER">PARAMETERS_FOLDER</a></span> - Static variable in class com.compomics.util.experiment.identification.identification_parameters.<a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html" title="class in com.compomics.util.experiment.identification.identification_parameters">IdentificationParametersFactory</a></dt>
<dd>
<div class="block">The identification parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#paramsTag">paramsTag</a></span> - Static variable in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>
<div class="block">Tag for the parameters file path.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#parseFromFASTA-java.lang.String-">parseFromFASTA(String)</a></span> - Static method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Factory method that constructs a Header instance based on a FASTA header
 line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#parsePeptideFromAnnotatedSequence-java.lang.String-">parsePeptideFromAnnotatedSequence(String)</a></span> - Static method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method allows the construction of an AASequenceImpl object, complete with modifications
 from an annotated sequence String (eg., something like: 'NH2-YS&lt;P&gt;FVATER-COOH' or 'Ace-MATHM&lt;Mox&gt;PIR-COOH').</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#parseSequenceAndModificationsFromString-java.lang.String-">parseSequenceAndModificationsFromString(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">This method is designed to load a sequence and it's set of modifications from a String
 which holds the sequence, annotated with all the modifications applied to it.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Connectable.html#passConnection-java.sql.Connection-java.lang.String-">passConnection(Connection, String)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Connectable.html" title="interface in com.compomics.util.interfaces">Connectable</a></dt>
<dd>
<div class="block">This method will be called by the class actually making the connection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItemComparator.html#passes-java.lang.String-double-">passes(String, double)</a></span> - Method in enum com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItemComparator.html" title="enum in com.compomics.util.experiment.filtering">FilterItemComparator</a></dt>
<dd>
<div class="block">Indicates whether a given value passes a threshold using this comparator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItemComparator.html#passes-java.lang.String-java.lang.String-">passes(String, String)</a></span> - Method in enum com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItemComparator.html" title="enum in com.compomics.util.experiment.filtering">FilterItemComparator</a></dt>
<dd>
<div class="block">Indicates whether a given value passes a threshold using this comparator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItemComparator.html#passes-java.lang.String-java.util.Collection-">passes(String, Collection&lt;String&gt;)</a></span> - Method in enum com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItemComparator.html" title="enum in com.compomics.util.experiment.filtering">FilterItemComparator</a></dt>
<dd>
<div class="block">Indicates whether a set of values passes a threshold using this
 comparator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItemComparator.html#passes-java.lang.String-java.lang.String:A-">passes(String, String[])</a></span> - Method in enum com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItemComparator.html" title="enum in com.compomics.util.experiment.filtering">FilterItemComparator</a></dt>
<dd>
<div class="block">Indicates whether a set of values passes a threshold using this
 comparator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItemComparator.html#passes-java.lang.String-java.util.stream.Stream-">passes(String, Stream&lt;String&gt;)</a></span> - Method in enum com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItemComparator.html" title="enum in com.compomics.util.experiment.filtering">FilterItemComparator</a></dt>
<dd>
<div class="block">Indicates whether a set of values passes a threshold using this
 comparator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/FlamableJFrame.html#passHotPotato-java.lang.Throwable-">passHotPotato(Throwable)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/FlamableJFrame.html" title="class in com.compomics.util.gui">FlamableJFrame</a></dt>
<dd>
<div class="block">This method will handle all errors thrown from child threads.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/FlamableJFrame.html#passHotPotato-java.lang.Throwable-java.lang.String-">passHotPotato(Throwable, String)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/FlamableJFrame.html" title="class in com.compomics.util.gui">FlamableJFrame</a></dt>
<dd>
<div class="block">This method takes care of any unrecoverable exception or error, thrown by
 a child thread.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Flamable.html#passHotPotato-java.lang.Throwable-">passHotPotato(Throwable)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Flamable.html" title="interface in com.compomics.util.interfaces">Flamable</a></dt>
<dd>
<div class="block">This method takes care of any unrecoverable exception or error, thrown by a child thread.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Flamable.html#passHotPotato-java.lang.Throwable-java.lang.String-">passHotPotato(Throwable, String)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Flamable.html" title="interface in com.compomics.util.interfaces">Flamable</a></dt>
<dd>
<div class="block">This method takes care of any unrecoverable exception or error, thrown by a child thread.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#PASSWORD">PASSWORD</a></span> - Static variable in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/software/settings/PathKey.html" title="interface in com.compomics.software.settings"><span class="typeNameLink">PathKey</span></a> - Interface in <a href="com/compomics/software/settings/package-summary.html">com.compomics.software.settings</a></dt>
<dd>
<div class="block">Interface for a path key.</div>
</dd>
<dt><a href="com/compomics/software/settings/gui/PathParametersDialog.html" title="class in com.compomics.software.settings.gui"><span class="typeNameLink">PathParametersDialog</span></a> - Class in <a href="com/compomics/software/settings/gui/package-summary.html">com.compomics.software.settings.gui</a></dt>
<dd>
<div class="block">Dialog used to set paths.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/gui/PathParametersDialog.html#PathParametersDialog-java.awt.Frame-java.lang.String-java.util.HashMap-">PathParametersDialog(Frame, String, HashMap&lt;PathKey, String&gt;)</a></span> - Constructor for class com.compomics.software.settings.gui.<a href="com/compomics/software/settings/gui/PathParametersDialog.html" title="class in com.compomics.software.settings.gui">PathParametersDialog</a></dt>
<dd>
<div class="block">Creates a new PathSettingsDialog.</div>
</dd>
<dt><a href="com/compomics/util/experiment/patient/Patient.html" title="class in com.compomics.util.experiment.patient"><span class="typeNameLink">Patient</span></a> - Class in <a href="com/compomics/util/experiment/patient/package-summary.html">com.compomics.util.experiment.patient</a></dt>
<dd>
<div class="block">This class groups the information about the patient from whom the sample is
 taken</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/patient/Patient.html#Patient-java.lang.String-">Patient(String)</a></span> - Constructor for class com.compomics.util.experiment.patient.<a href="com/compomics/util/experiment/patient/Patient.html" title="class in com.compomics.util.experiment.patient">Patient</a></dt>
<dd>
<div class="block">Constructor,</div>
</dd>
<dt><a href="com/compomics/util/experiment/patient/PatientInformation.html" title="class in com.compomics.util.experiment.patient"><span class="typeNameLink">PatientInformation</span></a> - Class in <a href="com/compomics/util/experiment/patient/package-summary.html">com.compomics.util.experiment.patient</a></dt>
<dd>
<div class="block">Information about the patient.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/patient/PatientInformation.html#PatientInformation-java.lang.String-java.lang.String-">PatientInformation(String, String)</a></span> - Constructor for class com.compomics.util.experiment.patient.<a href="com/compomics/util/experiment/patient/PatientInformation.html" title="class in com.compomics.util.experiment.patient">PatientInformation</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb"><span class="typeNameLink">PdbBlock</span></a> - Class in <a href="com/compomics/util/pdbfinder/pdb/package-summary.html">com.compomics.util.pdbfinder.pdb</a></dt>
<dd>
<div class="block">PdbBlock.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#PdbBlock-java.lang.String-int-int-int-int-">PdbBlock(String, int, int, int, int)</a></span> - Constructor for class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb"><span class="typeNameLink">PdbParameter</span></a> - Class in <a href="com/compomics/util/pdbfinder/pdb/package-summary.html">com.compomics.util.pdbfinder.pdb</a></dt>
<dd>
<div class="block">PdbParameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html#PdbParameter-java.lang.String-java.lang.String-java.lang.String-java.lang.String-">PdbParameter(String, String, String, String)</a></span> - Constructor for class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb">PdbParameter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#pdf">pdf</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#PDF">PDF</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/PdfFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">PdfFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.pdf files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PdfFileFilter.html#PdfFileFilter--">PdfFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PdfFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PdfFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.html#peak">peak</a></span> - Variable in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.html" title="class in com.compomics.util.experiment.identification.matches">IonMatch</a></dt>
<dd>
<div class="block">The matched peak.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra"><span class="typeNameLink">Peak</span></a> - Class in <a href="com/compomics/util/experiment/mass_spectrometry/spectra/package-summary.html">com.compomics.util.experiment.mass_spectrometry.spectra</a></dt>
<dd>
<div class="block">This class represents a peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#Peak-double-double-">Peak(double, double)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>
<div class="block">Constructor for a peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#Peak-double-double-double-">Peak(double, double, double)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>
<div class="block">Constructor for a peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#peakMap">peakMap</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">mz indexed Peak map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#peaks">peaks</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The PEAKS sequencing algorithm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#PEFF">PEFF</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#peff">peff</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/PeffFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">PeffFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.peff files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PeffFileFilter.html#PeffFileFilter--">PeffFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PeffFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PeffFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#pep_xml">pep_xml</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#PEP_XML">PEP_XML</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#pepMass-java.lang.String-">pepMass(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Computing the mass of a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#pepnovo">pepnovo</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The PepNovo+ de novo sequencing algorithm.</div>
</dd>
<dt><a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html" title="class in com.compomics.util.experiment.refinement_parameters"><span class="typeNameLink">PepnovoAssumptionDetails</span></a> - Class in <a href="com/compomics/util/experiment/refinement_parameters/package-summary.html">com.compomics.util.experiment.refinement_parameters</a></dt>
<dd>
<div class="block">This class contains pepnovo assumption details which are not contained in the
 tag assumption class which will be saved as additional parameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html#PepnovoAssumptionDetails--">PepnovoAssumptionDetails()</a></span> - Constructor for class com.compomics.util.experiment.refinement_parameters.<a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html" title="class in com.compomics.util.experiment.refinement_parameters">PepnovoAssumptionDetails</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">PepNovoIdfileReader</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">This class can be used to parse PepNovo identification files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#PepNovoIdfileReader--">PepNovoIdfileReader()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>
<div class="block">Default constructor for the purpose of instantiation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#PepNovoIdfileReader-java.io.File-">PepNovoIdfileReader(File)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>
<div class="block">Constructor, initiate the parser.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#PepNovoIdfileReader-java.io.File-com.compomics.util.waiting.WaitingHandler-">PepNovoIdfileReader(File, WaitingHandler)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>
<div class="block">Constructor, initiate the parser.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">PepnovoParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">PepNovo specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#PepnovoParameters--">PepnovoParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm"><span class="typeNameLink">PepNovoParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a></dt>
<dd>
<div class="block">Dialog for editing the PepNovo advanced settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html#PepNovoParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.PepnovoParameters-boolean-">PepNovoParametersDialog(Frame, PepnovoParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">PepNovoParametersDialog</a></dt>
<dd>
<div class="block">Creates a new PepNovoSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html#PepNovoParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.PepnovoParameters-boolean-">PepNovoParametersDialog(Dialog, Frame, PepnovoParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">PepNovoParametersDialog</a></dt>
<dd>
<div class="block">Creates a new PepNovoSettingsDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins"><span class="typeNameLink">Peptide</span></a> - Class in <a href="com/compomics/util/experiment/biology/proteins/package-summary.html">com.compomics.util.experiment.biology.proteins</a></dt>
<dd>
<div class="block">This class represents a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#Peptide--">Peptide()</a></span> - Constructor for class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Constructor for the peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#Peptide-java.lang.String-com.compomics.util.experiment.identification.matches.ModificationMatch:A-java.util.HashMap-boolean-double-">Peptide(String, ModificationMatch[], HashMap&lt;String, HashMap&lt;Integer, PeptideVariantMatches&gt;&gt;, boolean, double)</a></span> - Constructor for class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#Peptide-java.lang.String-com.compomics.util.experiment.identification.matches.ModificationMatch:A-boolean-double-">Peptide(String, ModificationMatch[], boolean, double)</a></span> - Constructor for class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#Peptide-java.lang.String-com.compomics.util.experiment.identification.matches.ModificationMatch:A-boolean-">Peptide(String, ModificationMatch[], boolean)</a></span> - Constructor for class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#Peptide-java.lang.String-">Peptide(String)</a></span> - Constructor for class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#Peptide-java.lang.String-com.compomics.util.experiment.identification.matches.ModificationMatch:A-">Peptide(String, ModificationMatch[])</a></span> - Constructor for class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#Peptide-java.lang.String-com.compomics.util.experiment.identification.matches.ModificationMatch:A-java.util.HashMap-boolean-">Peptide(String, ModificationMatch[], HashMap&lt;String, HashMap&lt;Integer, PeptideVariantMatches&gt;&gt;, boolean)</a></span> - Constructor for class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Constructor for the peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationMethod.html#PEPTIDE_FINGERPRINTING">PEPTIDE_FINGERPRINTING</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationMethod.html" title="class in com.compomics.util.experiment.identification">IdentificationMethod</a></dt>
<dd>
<div class="block">index for identification method based on peptide mass fingerprinting</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">PeptideAminoAcidFeature</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">Feature based on the amino acid properties of a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideAminoAcidFeature.html#PeptideAminoAcidFeature-com.compomics.util.experiment.biology.aminoacids.AminoAcid.Property-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.AASequenceFeature.Function-">PeptideAminoAcidFeature(AminoAcid.Property, AASequenceFeature.Function)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideAminoAcidFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideAminoAcidFeature</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_inference/PeptideAndProteinBuilder.html" title="class in com.compomics.util.experiment.identification.protein_inference"><span class="typeNameLink">PeptideAndProteinBuilder</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_inference/package-summary.html">com.compomics.util.experiment.identification.protein_inference</a></dt>
<dd>
<div class="block">This class builds peptides and proteins based on PSMs.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideAndProteinBuilder.html#PeptideAndProteinBuilder-com.compomics.util.experiment.identification.Identification-">PeptideAndProteinBuilder(Identification)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideAndProteinBuilder.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideAndProteinBuilder</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions"><span class="typeNameLink">PeptideAssumption</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_assumptions/package-summary.html">com.compomics.util.experiment.identification.spectrum_assumptions</a></dt>
<dd>
<div class="block">This object will models the assumption made by an advocate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html#PeptideAssumption-com.compomics.util.experiment.biology.proteins.Peptide-int-int-int-double-java.lang.String-">PeptideAssumption(Peptide, int, int, int, double, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">PeptideAssumption</a></dt>
<dd>
<div class="block">Constructor for a peptide assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html#PeptideAssumption-com.compomics.util.experiment.biology.proteins.Peptide-int-int-int-double-">PeptideAssumption(Peptide, int, int, int, double)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">PeptideAssumption</a></dt>
<dd>
<div class="block">Constructor for a peptide assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html#PeptideAssumption-com.compomics.util.experiment.biology.proteins.Peptide-int-">PeptideAssumption(Peptide, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">PeptideAssumption</a></dt>
<dd>
<div class="block">Constructor for a simple peptide assumption containing only the
 information necessary for spectrum annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html#PeptideAssumption--">PeptideAssumption()</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">PeptideAssumption</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering"><span class="typeNameLink">PeptideAssumptionFilter</span></a> - Class in <a href="com/compomics/util/experiment/identification/filtering/package-summary.html">com.compomics.util.experiment.identification.filtering</a></dt>
<dd>
<div class="block">This class filters peptide assumptions based on various properties.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#PeptideAssumptionFilter--">PeptideAssumptionFilter()</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Constructor with default settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#PeptideAssumptionFilter-int-int-double-boolean-boolean-java.lang.Integer-java.lang.Integer-java.lang.Integer-java.lang.Integer-">PeptideAssumptionFilter(int, int, double, boolean, boolean, Integer, Integer, Integer, Integer)</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Constructor for an Identification filter.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion"><span class="typeNameLink">PeptideDraft</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/digestion/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion</a></dt>
<dd>
<div class="block">Draft of a peptide used during protein sequence digestion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#PeptideDraft-char:A-">PeptideDraft(char[])</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#PeptideDraft-char:A-java.lang.String-java.util.HashMap-double-">PeptideDraft(char[], String, HashMap&lt;Integer, String&gt;, double)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#PeptideDraft-char:A-java.lang.String-java.util.HashMap-double-int-">PeptideDraft(char[], String, HashMap&lt;Integer, String&gt;, double, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#PeptideDraft-char:A-java.lang.String-java.lang.String-java.util.HashMap-double-">PeptideDraft(char[], String, String, HashMap&lt;Integer, String&gt;, double)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#PeptideDraft-char:A-java.lang.String-java.lang.String-java.util.HashMap-double-int-">PeptideDraft(char[], String, String, HashMap&lt;Integer, String&gt;, double, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">PeptideFeature</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">Feature based on a peptide property.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html#PeptideFeature-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.PeptideFeature.Property-">PeptideFeature(PeptideFeature.Property)</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideFeature</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features"><span class="typeNameLink">PeptideFeature.Property</span></a> - Enum in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></dt>
<dd>
<div class="block">Enum for the possible features.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html" title="class in com.compomics.util.experiment.identification.filtering"><span class="typeNameLink">PeptideFilter</span></a> - Class in <a href="com/compomics/util/experiment/identification/filtering/package-summary.html">com.compomics.util.experiment.identification.filtering</a></dt>
<dd>
<div class="block">Peptide filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html#PeptideFilter--">PeptideFilter()</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html#PeptideFilter-java.lang.String-">PeptideFilter(String)</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html#PeptideFilter-java.lang.String-java.lang.String-">PeptideFilter(String, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html#PeptideFilter-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String-">PeptideFilter(String, String, String, String, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items"><span class="typeNameLink">PeptideFilterItem</span></a> - Enum in <a href="com/compomics/util/experiment/identification/filtering/items/package-summary.html">com.compomics.util.experiment.identification.filtering.items</a></dt>
<dd>
<div class="block">Enum of the different items a peptide filter can filter on.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/PeptideFragmentationModel.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation"><span class="typeNameLink">PeptideFragmentationModel</span></a> - Enum in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation</a></dt>
<dd>
<div class="block">Enum for the different peptide fragmentation models implemented.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl"><span class="typeNameLink">PeptideFragmentIon</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/impl/package-summary.html">com.compomics.util.experiment.biology.ions.impl</a></dt>
<dd>
<div class="block">This class models a peptide fragment ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#PeptideFragmentIon-int-int-double-com.compomics.util.experiment.biology.ions.NeutralLoss:A-">PeptideFragmentIon(int, int, double, NeutralLoss[])</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#PeptideFragmentIon-int-com.compomics.util.experiment.biology.ions.NeutralLoss:A-">PeptideFragmentIon(int, NeutralLoss[])</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Constructor for a generic ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#PeptideFragmentIon-int-">PeptideFragmentIon(int)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Constructor for a generic ion without neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#peptideIdentification">peptideIdentification</a></span> - Variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">List of the keys of all imported peptides.</div>
</dd>
<dt><a href="com/compomics/cli/peptide_mapper/PeptideMapperCLI.html" title="class in com.compomics.cli.peptide_mapper"><span class="typeNameLink">PeptideMapperCLI</span></a> - Class in <a href="com/compomics/cli/peptide_mapper/package-summary.html">com.compomics.cli.peptide_mapper</a></dt>
<dd>
<div class="block">Command line peptide mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/peptide_mapper/PeptideMapperCLI.html#PeptideMapperCLI--">PeptideMapperCLI()</a></span> - Constructor for class com.compomics.cli.peptide_mapper.<a href="com/compomics/cli/peptide_mapper/PeptideMapperCLI.html" title="class in com.compomics.cli.peptide_mapper">PeptideMapperCLI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html" title="class in com.compomics.util.experiment.identification.matches"><span class="typeNameLink">PeptideMatch</span></a> - Class in <a href="com/compomics/util/experiment/identification/matches/package-summary.html">com.compomics.util.experiment.identification.matches</a></dt>
<dd>
<div class="block">This class models a peptide match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html#PeptideMatch-com.compomics.util.experiment.biology.proteins.Peptide-long-long-">PeptideMatch(Peptide, long, long)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html" title="class in com.compomics.util.experiment.identification.matches">PeptideMatch</a></dt>
<dd>
<div class="block">Constructor for the peptide match.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/matches_iterators/PeptideMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators"><span class="typeNameLink">PeptideMatchesIterator</span></a> - Class in <a href="com/compomics/util/experiment/identification/matches_iterators/package-summary.html">com.compomics.util.experiment.identification.matches_iterators</a></dt>
<dd>
<div class="block">An iterator for peptide matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/PeptideMatchesIterator.html#PeptideMatchesIterator-com.compomics.util.experiment.identification.Identification-com.compomics.util.waiting.WaitingHandler-boolean-">PeptideMatchesIterator(Identification, WaitingHandler, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/PeptideMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">PeptideMatchesIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/PeptideMatchesIterator.html#PeptideMatchesIterator-long:A-com.compomics.util.experiment.identification.Identification-com.compomics.util.waiting.WaitingHandler-boolean-">PeptideMatchesIterator(long[], Identification, WaitingHandler, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/PeptideMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">PeptideMatchesIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference"><span class="typeNameLink">PeptideProteinMapping</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_inference/package-summary.html">com.compomics.util.experiment.identification.protein_inference</a></dt>
<dd>
<div class="block">Class used to model the mapping of a peptide to a protein sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html#PeptideProteinMapping-java.lang.String-java.lang.String-int-com.compomics.util.experiment.identification.matches.ModificationMatch:A-com.compomics.util.experiment.identification.matches.PeptideVariantMatches-">PeptideProteinMapping(String, String, int, ModificationMatch[], PeptideVariantMatches)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html#PeptideProteinMapping-java.lang.String-java.lang.String-int-">PeptideProteinMapping(String, String, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html#PeptideProteinMapping-java.lang.String-java.lang.String-int-com.compomics.util.experiment.identification.matches.ModificationMatch:A-">PeptideProteinMapping(String, String, int, ModificationMatch[])</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.<a href="com/compomics/util/experiment/identification/protein_inference/PeptideProteinMapping.html" title="class in com.compomics.util.experiment.identification.protein_inference">PeptideProteinMapping</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#peptideSequence">peptideSequence</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Current peptide sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#peptideSequenceSearch">peptideSequenceSearch</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Current peptide sequence needed for search.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#peptideShaker">peptideShaker</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The PeptideShaker multiple algorithm software.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#peptideShakerFileOption">peptideShakerFileOption</a></span> - Static variable in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">The command line argument for a cps file for PeptideShaker.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#peptideShakerPxAccessionOption">peptideShakerPxAccessionOption</a></span> - Static variable in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">The command line argument to open a given PX accession in PRIDE Reshake.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#peptideShakerPxAccessionPrivateOption">peptideShakerPxAccessionPrivateOption</a></span> - Static variable in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">The command line argument to indicate that the PX accession to open in
 PRIDE Reshake is private.</div>
</dd>
<dt><a href="com/compomics/software/dialogs/PeptideShakerSetupDialog.html" title="class in com.compomics.software.dialogs"><span class="typeNameLink">PeptideShakerSetupDialog</span></a> - Class in <a href="com/compomics/software/dialogs/package-summary.html">com.compomics.software.dialogs</a></dt>
<dd>
<div class="block">A dialog used to set up the connection to PeptideShaker.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/PeptideShakerSetupDialog.html#PeptideShakerSetupDialog-javax.swing.JFrame-boolean-">PeptideShakerSetupDialog(JFrame, boolean)</a></span> - Constructor for class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/PeptideShakerSetupDialog.html" title="class in com.compomics.software.dialogs">PeptideShakerSetupDialog</a></dt>
<dd>
<div class="block">Creates a new PeptideShakerSetupDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/PeptideShakerSetupDialog.html#PeptideShakerSetupDialog-javax.swing.JDialog-boolean-">PeptideShakerSetupDialog(JDialog, boolean)</a></span> - Constructor for class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/PeptideShakerSetupDialog.html" title="class in com.compomics.software.dialogs">PeptideShakerSetupDialog</a></dt>
<dd>
<div class="block">Creates a new PeptideShakerSetupDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#peptideShakerUrlDownloadFolderOption">peptideShakerUrlDownloadFolderOption</a></span> - Static variable in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">The command line argument for the download folder for the URL for
 PeptideShaker.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#peptideShakerUrlOption">peptideShakerUrlOption</a></span> - Static variable in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">The command line argument for a zipped cps URL for PeptideShaker.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators"><span class="typeNameLink">PeptideSpectrumAnnotator</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators</a></dt>
<dd>
<div class="block">Annotates a spectrum with peptide fragments.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#PeptideSpectrumAnnotator--">PeptideSpectrumAnnotator()</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils"><span class="typeNameLink">PeptideUtils</span></a> - Class in <a href="com/compomics/util/experiment/identification/utils/package-summary.html">com.compomics.util.experiment.identification.utils</a></dt>
<dd>
<div class="block">This class groups functions that can be used to work with peptides.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html#PeptideUtils--">PeptideUtils()</a></span> - Constructor for class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/PeptideUtils.html" title="class in com.compomics.util.experiment.identification.utils">PeptideUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/matches/PeptideVariantMatches.html" title="class in com.compomics.util.experiment.identification.matches"><span class="typeNameLink">PeptideVariantMatches</span></a> - Class in <a href="com/compomics/util/experiment/identification/matches/package-summary.html">com.compomics.util.experiment.identification.matches</a></dt>
<dd>
<div class="block">This class represents a set of variants found on a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideVariantMatches.html#PeptideVariantMatches-java.util.HashMap-int-">PeptideVariantMatches(HashMap&lt;Integer, Variant&gt;, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideVariantMatches.html" title="class in com.compomics.util.experiment.identification.matches">PeptideVariantMatches</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced"><span class="typeNameLink">PeptideVariantsParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/advanced/package-summary.html">com.compomics.util.parameters.identification.advanced</a></dt>
<dd>
<div class="block">Preferences for the allowed variants in peptide sequences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#PeptideVariantsParameters--">PeptideVariantsParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/advanced/PeptideVariantsParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced"><span class="typeNameLink">PeptideVariantsParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/advanced/package-summary.html">com.compomics.util.gui.parameters.identification.advanced</a></dt>
<dd>
<div class="block">FractionSettingsDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/PeptideVariantsParametersDialog.html#PeptideVariantsParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters-boolean-">PeptideVariantsParametersDialog(Frame, PeptideVariantsParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/PeptideVariantsParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">PeptideVariantsParametersDialog</a></dt>
<dd>
<div class="block">Creates a new FractionSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/PeptideVariantsParametersDialog.html#PeptideVariantsParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters-boolean-">PeptideVariantsParametersDialog(Dialog, Frame, PeptideVariantsParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/PeptideVariantsParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">PeptideVariantsParametersDialog</a></dt>
<dd>
<div class="block">Creates a new FractionSettingsDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideWithPosition.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion"><span class="typeNameLink">PeptideWithPosition</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/digestion/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion</a></dt>
<dd>
<div class="block">This class packages together a peptide and its position on the protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideWithPosition.html#PeptideWithPosition-com.compomics.util.experiment.biology.proteins.Peptide-int-">PeptideWithPosition(Peptide, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideWithPosition.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideWithPosition</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#pepxml">pepxml</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#PEPXML">PEPXML</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/PepXmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">PepXmlFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.pep.xml, *.pepxml and *.xml files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PepXmlFileFilter.html#PepXmlFileFilter--">PepXmlFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PepXmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PepXmlFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">PepxmlIdfileReader</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">Simple IdfileReader for Pepxml files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html#PepxmlIdfileReader--">PepxmlIdfileReader()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepxmlIdfileReader</a></dt>
<dd>
<div class="block">Blank constructor for instantiation purposes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html#PepxmlIdfileReader-java.io.File-">PepxmlIdfileReader(File)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepxmlIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepxmlIdfileReader</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#PERACETYLATED_MASS">PERACETYLATED_MASS</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>
<div class="block">Final index for peracetylated masses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/Units.html#percent">percent</a></span> - Static variable in class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/Units.html" title="class in com.compomics.util.experiment.units">Units</a></dt>
<dd>
<div class="block">percent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#percentile-double:A-double-">percentile(double[], double)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Returns the desired percentile in a given array of double.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#percentile-java.util.ArrayList-double-">percentile(ArrayList&lt;Double&gt;, double)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Returns the desired percentile in a given list of double.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#percentileSorted-java.util.ArrayList-double-">percentileSorted(ArrayList&lt;Double&gt;, double)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Returns the desired percentile in a given list of double.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#percolator">percolator</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Percolator rescoring algorithm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#PERMETHYLATED_MASS">PERMETHYLATED_MASS</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>
<div class="block">Final index for permethylated masses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/threading/MapMutex.html#permits">permits</a></span> - Variable in class com.compomics.util.threading.<a href="com/compomics/util/threading/MapMutex.html" title="class in com.compomics.util.threading">MapMutex</a></dt>
<dd>
<div class="block">The number of permits per key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/interfaces/Persistable.html#persist-java.sql.Connection-">persist(Connection)</a></span> - Method in interface com.compomics.util.db.interfaces.<a href="com/compomics/util/db/interfaces/Persistable.html" title="interface in com.compomics.util.db.interfaces">Persistable</a></dt>
<dd>
<div class="block">This method can be called to store the implemnted objects state to a persistent
 store for the first time (equivalent to SQL 'INSERT').</div>
</dd>
<dt><a href="com/compomics/util/db/interfaces/Persistable.html" title="interface in com.compomics.util.db.interfaces"><span class="typeNameLink">Persistable</span></a> - Interface in <a href="com/compomics/util/db/interfaces/package-summary.html">com.compomics.util.db.interfaces</a></dt>
<dd>
<div class="block">This interface indicates that the implementing class can be persisted in permanent storage.</div>
</dd>
<dt><a href="com/compomics/util/db/components/PersistableCode.html" title="class in com.compomics.util.db.components"><span class="typeNameLink">PersistableCode</span></a> - Class in <a href="com/compomics/util/db/components/package-summary.html">com.compomics.util.db.components</a></dt>
<dd>
<div class="block">This class generates the code to make a DBAccessor an implementation of the Persistable
 interface.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/PersistableCode.html#PersistableCode-com.compomics.util.db.DBMetaData-">PersistableCode(DBMetaData)</a></span> - Constructor for class com.compomics.util.db.components.<a href="com/compomics/util/db/components/PersistableCode.html" title="class in com.compomics.util.db.components">PersistableCode</a></dt>
<dd>
<div class="block">This constructor allows the generation of the code for the implementation of
 the Persistable interface for a DBAccessor class, based on the specified metadata.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Phenylalanine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Phenylalanine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html#Phenylalanine--">Phenylalanine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Phenylalanine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Phenylalanine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#PHENYLALANINE">PHENYLALANINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Phenylananine immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#phenyx">phenyx</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Phenyx search engine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#PHOSPHO_Y">PHOSPHO_Y</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for phosphorylation of tyrosine (PMID: 11473401).</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html" title="class in com.compomics.util.experiment.identification.modification.scores"><span class="typeNameLink">PhosphoRS</span></a> - Class in <a href="com/compomics/util/experiment/identification/modification/scores/package-summary.html">com.compomics.util.experiment.identification.modification.scores</a></dt>
<dd>
<div class="block">This class estimates the PhosphoRS score as described in
 https://www.ncbi.nlm.nih.gov/pubmed/22073976.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html#PhosphoRS--">PhosphoRS()</a></span> - Constructor for class com.compomics.util.experiment.identification.modification.scores.<a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html" title="class in com.compomics.util.experiment.identification.modification.scores">PhosphoRS</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/Phosphorus.html" title="class in com.compomics.util.experiment.biology.atoms.impl"><span class="typeNameLink">Phosphorus</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a></dt>
<dd>
<div class="block">The phosphorus atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/impl/Phosphorus.html#Phosphorus--">Phosphorus()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.impl.<a href="com/compomics/util/experiment/biology/atoms/impl/Phosphorus.html" title="class in com.compomics.util.experiment.biology.atoms.impl">Phosphorus</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/interfaces/PickUp.html" title="interface in com.compomics.util.interfaces"><span class="typeNameLink">PickUp</span></a> - Interface in <a href="com/compomics/util/interfaces/package-summary.html">com.compomics.util.interfaces</a></dt>
<dd>
<div class="block">This interface describes the behaviour for a class that can pick up results.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#PICKUP">PICKUP</a></span> - Static variable in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#pkl">pkl</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#PKL">PKL</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io"><span class="typeNameLink">PklFile</span></a> - Class in <a href="com/compomics/util/io/package-summary.html">com.compomics.util.io</a></dt>
<dd>
<div class="block">Contains information about the contents of one PKL file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#PklFile-java.io.File-">PklFile(File)</a></span> - Constructor for class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>
<div class="block">Parse a PKL file and store the details in the PKLFile object.</div>
</dd>
<dt><a href="com/compomics/util/io/file/filefilters/PklFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">PklFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.pkl files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PklFileFilter.html#PklFileFilter--">PklFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PklFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PklFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/PklSpoFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">PklSpoFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.pkl.spo files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PklSpoFileFilter.html#PklSpoFileFilter--">PklSpoFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PklSpoFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PklSpoFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#pkx">pkx</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#PKX">PKX</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/PkxFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">PkxFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.pkx files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PkxFileFilter.html#PkxFileFilter--">PkxFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PkxFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PkxFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html#pmid">pmid</a></span> - Variable in enum com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html" title="enum in com.compomics.util.experiment.io.biology.protein">ProteinDatabase</a></dt>
<dd>
<div class="block">The PubMed id of the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#png">png</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#PNG">PNG</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/PngFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">PngFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.png files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/PngFileFilter.html#PngFileFilter--">PngFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/PngFileFilter.html" title="class in com.compomics.util.io.file.filefilters">PngFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#pNovo">pNovo</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The pNovo+ de novo sequencing algorithm.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">PNovoIdfileReader</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">This class can be used to parse pNovo identification files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html#PNovoIdfileReader--">PNovoIdfileReader()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PNovoIdfileReader</a></dt>
<dd>
<div class="block">Default constructor for the purpose of instantiation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html#PNovoIdfileReader-java.io.File-">PNovoIdfileReader(File)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PNovoIdfileReader</a></dt>
<dd>
<div class="block">Constructor, initiate the parser.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html#PNovoIdfileReader-java.io.File-com.compomics.util.waiting.WaitingHandler-">PNovoIdfileReader(File, WaitingHandler)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PNovoIdfileReader</a></dt>
<dd>
<div class="block">Constructor, initiate the parser.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">PNovoParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">pNovo specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#PNovoParameters--">PNovoParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm"><span class="typeNameLink">PNovoParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a></dt>
<dd>
<div class="block">Dialog for editing pNovo advanced settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html#PNovoParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.PNovoParameters-boolean-">PNovoParametersDialog(Frame, PNovoParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">PNovoParametersDialog</a></dt>
<dd>
<div class="block">Creates a new PNovoSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html#PNovoParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.PNovoParameters-boolean-">PNovoParametersDialog(Dialog, Frame, PNovoParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">PNovoParametersDialog</a></dt>
<dd>
<div class="block">Creates a new PNovoSettingsDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigFunctions.html#pow-java.math.BigDecimal-java.math.BigDecimal-java.math.MathContext-">pow(BigDecimal, BigDecimal, MathContext)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BigFunctions.html" title="class in com.compomics.util.math">BigFunctions</a></dt>
<dd>
<div class="block">Returns the first big decimal power the second using the given math
 context.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/MetricsPrefix.html#POWER">POWER</a></span> - Variable in enum com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/MetricsPrefix.html" title="enum in com.compomics.util.experiment.units">MetricsPrefix</a></dt>
<dd>
<div class="block">The power of 10 to use.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/Units.html#ppm">ppm</a></span> - Static variable in class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/Units.html" title="class in com.compomics.util.experiment.units">Units</a></dt>
<dd>
<div class="block">ppm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#PRECURSOR">PRECURSOR</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>
<div class="block">For now only one type of precursor implemented.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.PrecursorWithTitle.html#precursor">precursor</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.PrecursorWithTitle.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">PrecursorMap.PrecursorWithTitle</a></dt>
<dd>
<div class="block">The precursor</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra"><span class="typeNameLink">Precursor</span></a> - Class in <a href="com/compomics/util/experiment/mass_spectrometry/spectra/package-summary.html">com.compomics.util.experiment.mass_spectrometry.spectra</a></dt>
<dd>
<div class="block">This class models a precursor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#Precursor-double-double-java.util.ArrayList-">Precursor(double, double, ArrayList&lt;Integer&gt;)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Constructor for the precursor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#Precursor-double-double-double-java.util.ArrayList-double-double-">Precursor(double, double, double, ArrayList&lt;Integer&gt;, double, double)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Constructor with retention time window.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#Precursor-double-double-java.util.ArrayList-double-double-">Precursor(double, double, ArrayList&lt;Integer&gt;, double, double)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Constructor with retention time window and no reference retention time.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html#Precursor-double-double-double-java.util.ArrayList-">Precursor(double, double, double, ArrayList&lt;Integer&gt;)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Precursor.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Precursor</a></dt>
<dd>
<div class="block">Constructor for the precursor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/PrecursorAccuracy.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores"><span class="typeNameLink">PrecursorAccuracy</span></a> - Class in <a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/package-summary.html">com.compomics.util.experiment.identification.psm_scoring.psm_scores</a></dt>
<dd>
<div class="block">The precursor accuracy as a score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/PrecursorAccuracy.html#PrecursorAccuracy--">PrecursorAccuracy()</a></span> - Constructor for class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/PrecursorAccuracy.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">PrecursorAccuracy</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/PrecursorAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators"><span class="typeNameLink">PrecursorAnnotator</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators</a></dt>
<dd>
<div class="block">Annotator for the precursor peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/PrecursorAnnotator.html#PrecursorAnnotator-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">PrecursorAnnotator(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/PrecursorAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">PrecursorAnnotator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#precursorCharge">precursorCharge</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">The precursor charge as deduced by the search engine.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl"><span class="typeNameLink">PrecursorIon</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/impl/package-summary.html">com.compomics.util.experiment.biology.ions.impl</a></dt>
<dd>
<div class="block">A precursor ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#PrecursorIon-double-com.compomics.util.experiment.biology.ions.NeutralLoss:A-">PrecursorIon(double, NeutralLoss[])</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#PrecursorIon-com.compomics.util.experiment.biology.ions.NeutralLoss:A-">PrecursorIon(NeutralLoss[])</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>
<div class="block">Constructor for a generic ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#PrecursorIon--">PrecursorIon()</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>
<div class="block">Constructor for a generic ion without neutral losses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html#PrecursorIon-double-">PrecursorIon(double)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PrecursorIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PrecursorIon</a></dt>
<dd>
<div class="block">Constructor for a generic ion without neutral losses.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes"><span class="typeNameLink">PrecursorMap</span></a> - Class in <a href="com/compomics/util/experiment/mass_spectrometry/indexes/package-summary.html">com.compomics.util.experiment.mass_spectrometry.indexes</a></dt>
<dd>
<div class="block">This map stores the precursors indexed by mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html#PrecursorMap-java.util.HashMap-double-boolean-">PrecursorMap(HashMap&lt;String, Precursor&gt;, double, boolean)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">PrecursorMap</a></dt>
<dd>
<div class="block">Builds a precursor map.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.PrecursorWithTitle.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes"><span class="typeNameLink">PrecursorMap.PrecursorWithTitle</span></a> - Class in <a href="com/compomics/util/experiment/mass_spectrometry/indexes/package-summary.html">com.compomics.util.experiment.mass_spectrometry.indexes</a></dt>
<dd>
<div class="block">Convenience class storing the precursor and corresponding spectrum title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.PrecursorWithTitle.html#PrecursorWithTitle-com.compomics.util.experiment.mass_spectrometry.spectra.Precursor-java.lang.String-">PrecursorWithTitle(Precursor, String)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.PrecursorWithTitle.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">PrecursorMap.PrecursorWithTitle</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#preferredMaxVariableModifications">preferredMaxVariableModifications</a></span> - Static variable in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">The preferred maximal number of variable modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#preferredMinSequences">preferredMinSequences</a></span> - Static variable in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">The preferred minimal number of protein sequences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/MetricsPrefix.html#PREFIX">PREFIX</a></span> - Variable in enum com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/MetricsPrefix.html" title="enum in com.compomics.util.experiment.units">MetricsPrefix</a></dt>
<dd>
<div class="block">The prefix in full letters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#prepareRenderer-javax.swing.table.TableCellRenderer-int-int-">prepareRenderer(TableCellRenderer, int, int)</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">Prepares the renderer by querying the data model for the
 value and selection state
 of the cell at <code>row</code>, <code>column</code>.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#previousContent">previousContent</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Index of the originating entry of a particular cell with the pattern
 searching matrix.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/webservice/query/PrideFilter.html" title="class in com.compomics.util.pride.prideobjects.webservice.query"><span class="typeNameLink">PrideFilter</span></a> - Class in <a href="com/compomics/util/pride/prideobjects/webservice/query/package-summary.html">com.compomics.util.pride.prideobjects.webservice.query</a></dt>
<dd>
<div class="block">An object to set up a filter for use in a query to the PRIDE webservice.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/webservice/query/PrideFilter.html#PrideFilter-com.compomics.util.pride.prideobjects.webservice.query.PrideFilterType-java.lang.String-">PrideFilter(PrideFilterType, String)</a></span> - Constructor for class com.compomics.util.pride.prideobjects.webservice.query.<a href="com/compomics/util/pride/prideobjects/webservice/query/PrideFilter.html" title="class in com.compomics.util.pride.prideobjects.webservice.query">PrideFilter</a></dt>
<dd>
<div class="block">Creates a new PrideFilter instance.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/webservice/query/PrideFilterType.html" title="enum in com.compomics.util.pride.prideobjects.webservice.query"><span class="typeNameLink">PrideFilterType</span></a> - Enum in <a href="com/compomics/util/pride/prideobjects/webservice/query/package-summary.html">com.compomics.util.pride.prideobjects.webservice.query</a></dt>
<dd>
<div class="block">An enum of the possible filters for the PRIDE Filter options.</div>
</dd>
<dt><a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html" title="class in com.compomics.util.io.json.marshallers"><span class="typeNameLink">PrideMarshaller</span></a> - Class in <a href="com/compomics/util/io/json/marshallers/package-summary.html">com.compomics.util.io.json.marshallers</a></dt>
<dd>
<div class="block">A json marshaller for use with the PRIDE webservice.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html#PrideMarshaller--">PrideMarshaller()</a></span> - Constructor for class com.compomics.util.io.json.marshallers.<a href="com/compomics/util/io/json/marshallers/PrideMarshaller.html" title="class in com.compomics.util.io.json.marshallers">PrideMarshaller</a></dt>
<dd>
<div class="block">Create a PRIDE marshaller.</div>
</dd>
<dt><a href="com/compomics/util/pride/PrideObject.html" title="interface in com.compomics.util.pride"><span class="typeNameLink">PrideObject</span></a> - Interface in <a href="com/compomics/util/pride/package-summary.html">com.compomics.util.pride</a></dt>
<dd>
<div class="block">Interface for a PRIDE object.</div>
</dd>
<dt><a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride"><span class="typeNameLink">PrideObjectsFactory</span></a> - Class in <a href="com/compomics/util/pride/package-summary.html">com.compomics.util.pride</a></dt>
<dd>
<div class="block">This factory manages the pride objects saved in the user folder.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/webservice/PrideQuery.html" title="enum in com.compomics.util.pride.prideobjects.webservice"><span class="typeNameLink">PrideQuery</span></a> - Enum in <a href="com/compomics/util/pride/prideobjects/webservice/package-summary.html">com.compomics.util.pride.prideobjects.webservice</a></dt>
<dd>
<div class="block">An enum that constructs the template queries for the PRIDE webservice.</div>
</dd>
<dt><a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride"><span class="typeNameLink">PrideWebService</span></a> - Class in <a href="com/compomics/util/pride/package-summary.html">com.compomics.util.pride</a></dt>
<dd>
<div class="block">A java representation of the PRIDE web service.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideWebService.html#PrideWebService--">PrideWebService()</a></span> - Constructor for class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideWebService.html" title="class in com.compomics.util.pride">PrideWebService</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/pride/validation/PrideXmlValidator.html" title="class in com.compomics.util.pride.validation"><span class="typeNameLink">PrideXmlValidator</span></a> - Class in <a href="com/compomics/util/pride/validation/package-summary.html">com.compomics.util.pride.validation</a></dt>
<dd>
<div class="block">PRIDE XML schema validation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/PrideXmlValidator.html#PrideXmlValidator--">PrideXmlValidator()</a></span> - Constructor for class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/PrideXmlValidator.html" title="class in com.compomics.util.pride.validation">PrideXmlValidator</a></dt>
<dd>
<div class="block">Set up a PrideXmlValidator using the default schema.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/PrideXmlValidator.html#PrideXmlValidator-java.net.URL-">PrideXmlValidator(URL)</a></span> - Constructor for class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/PrideXmlValidator.html" title="class in com.compomics.util.pride.validation">PrideXmlValidator</a></dt>
<dd>
<div class="block">Set up a PrideXmlValidator using the provided schema.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/KMeansClustering.html#printCentroids--">printCentroids()</a></span> - Method in class com.compomics.util.math.clustering.<a href="com/compomics/util/math/clustering/KMeansClustering.html" title="class in com.compomics.util.math.clustering">KMeansClustering</a></dt>
<dd>
<div class="block">Print the centroids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/KMeansClustering.html#printClusters--">printClusters()</a></span> - Method in class com.compomics.util.math.clustering.<a href="com/compomics/util/math/clustering/KMeansClustering.html" title="class in com.compomics.util.math.clustering">KMeansClustering</a></dt>
<dd>
<div class="block">Print the current clusters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html#printEnzymes--">printEnzymes()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html" title="class in com.compomics.cli.identification_parameters">AbstractIdentificationParametersCli</a></dt>
<dd>
<div class="block">Prints the available enzymes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html#printModifications--">printModifications()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/AbstractIdentificationParametersCli.html" title="class in com.compomics.cli.identification_parameters">AbstractIdentificationParametersCli</a></dt>
<dd>
<div class="block">Prints the available modifications.</div>
</dd>
<dt><a href="com/compomics/util/gui/PrivacyParametersDialog.html" title="class in com.compomics.util.gui"><span class="typeNameLink">PrivacyParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">The privacy settings dialog allows the user to set the privacy settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/PrivacyParametersDialog.html#PrivacyParametersDialog-java.awt.Frame-java.awt.Image-">PrivacyParametersDialog(Frame, Image)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/PrivacyParametersDialog.html" title="class in com.compomics.util.gui">PrivacyParametersDialog</a></dt>
<dd>
<div class="block">Creates a new PrivacySettingsDialog.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/linear_regression/filters/ProbabilityFilter.html" title="class in com.compomics.util.math.statistics.linear_regression.filters"><span class="typeNameLink">ProbabilityFilter</span></a> - Class in <a href="com/compomics/util/math/statistics/linear_regression/filters/package-summary.html">com.compomics.util.math.statistics.linear_regression.filters</a></dt>
<dd>
<div class="block">Filter removing points with outlying slope.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/filters/ProbabilityFilter.html#ProbabilityFilter--">ProbabilityFilter()</a></span> - Constructor for class com.compomics.util.math.statistics.linear_regression.filters.<a href="com/compomics/util/math/statistics/linear_regression/filters/ProbabilityFilter.html" title="class in com.compomics.util.math.statistics.linear_regression.filters">ProbabilityFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/parameters/tools/ProcessingParameters.html" title="class in com.compomics.util.parameters.tools"><span class="typeNameLink">ProcessingParameters</span></a> - Class in <a href="com/compomics/util/parameters/tools/package-summary.html">com.compomics.util.parameters.tools</a></dt>
<dd>
<div class="block">ProcessingPreferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/tools/ProcessingParameters.html#ProcessingParameters--">ProcessingParameters()</a></span> - Constructor for class com.compomics.util.parameters.tools.<a href="com/compomics/util/parameters/tools/ProcessingParameters.html" title="class in com.compomics.util.parameters.tools">ProcessingParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/parameters/tools/ProcessingParameters.ProcessingType.html" title="enum in com.compomics.util.parameters.tools"><span class="typeNameLink">ProcessingParameters.ProcessingType</span></a> - Enum in <a href="com/compomics/util/parameters/tools/package-summary.html">com.compomics.util.parameters.tools</a></dt>
<dd>
<div class="block">Class indicating the type of processing.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/tools/ProcessingParametersDialog.html" title="class in com.compomics.util.gui.parameters.tools"><span class="typeNameLink">ProcessingParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/tools/package-summary.html">com.compomics.util.gui.parameters.tools</a></dt>
<dd>
<div class="block">Dialog to edit the processing parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/tools/ProcessingParametersDialog.html#ProcessingParametersDialog-java.awt.Frame-com.compomics.util.parameters.tools.ProcessingParameters-boolean-">ProcessingParametersDialog(Frame, ProcessingParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.tools.<a href="com/compomics/util/gui/parameters/tools/ProcessingParametersDialog.html" title="class in com.compomics.util.gui.parameters.tools">ProcessingParametersDialog</a></dt>
<dd>
<div class="block">Creates a new ProcessingPreferencesDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/tools/ProcessingParametersDialog.html#ProcessingParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.tools.ProcessingParameters-boolean-">ProcessingParametersDialog(Dialog, Frame, ProcessingParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.tools.<a href="com/compomics/util/gui/parameters/tools/ProcessingParametersDialog.html" title="class in com.compomics.util.gui.parameters.tools">ProcessingParametersDialog</a></dt>
<dd>
<div class="block">Creates a new ProcessingPreferencesDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#processMirroredXAndYData-double:A-double:A-java.awt.Color-java.awt.Color-">processMirroredXAndYData(double[], double[], Color, Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method reads the x and y values from the specified arrays and stores
 these internally for drawing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#processXAndYData-double:A-double:A-java.awt.Color-java.awt.Color-">processXAndYData(double[], double[], Color, Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method reads the x and y values from the specified arrays and stores
 these internally for drawing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#proFound">proFound</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Profound search engine.</div>
</dd>
<dt><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers"><span class="typeNameLink">ProgressDialog</span></a> - Class in <a href="com/compomics/util/gui/waiting/waitinghandlers/package-summary.html">com.compomics.util.gui.waiting.waitinghandlers</a></dt>
<dd>
<div class="block">This class implements a modal dialog that is displayed during a longer-lasting task.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html#ProgressDialog-javax.swing.JFrame-java.lang.String-java.lang.String-">ProgressDialog(JFrame, String, String)</a></span> - Constructor for class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialog</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">use ProgressDialogX instead</span></div>
</div>
</dd>
<dt><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers"><span class="typeNameLink">ProgressDialogX</span></a> - Class in <a href="com/compomics/util/gui/waiting/waitinghandlers/package-summary.html">com.compomics.util.gui.waiting.waitinghandlers</a></dt>
<dd>
<div class="block">A dialog for displaying information about progress.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#ProgressDialogX-java.awt.Frame-java.awt.Image-java.awt.Image-boolean-">ProgressDialogX(Frame, Image, Image, boolean)</a></span> - Constructor for class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Opens a new ProgressDialogX with a Frame as a parent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#ProgressDialogX-javax.swing.JDialog-java.awt.Frame-java.awt.Image-java.awt.Image-boolean-">ProgressDialogX(JDialog, Frame, Image, Image, boolean)</a></span> - Constructor for class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Opens a new ProgressDialog with a JDialog as a parent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#ProgressDialogX-boolean-">ProgressDialogX(boolean)</a></span> - Constructor for class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Opens a new ProgressDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#projectNameOption">projectNameOption</a></span> - Static variable in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">The command line argument for the PeptideShaker project name.</div>
</dd>
<dt><a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment"><span class="typeNameLink">ProjectParameters</span></a> - Class in <a href="com/compomics/util/experiment/package-summary.html">com.compomics.util.experiment</a></dt>
<dd>
<div class="block">This class contains project specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#ProjectParameters--">ProjectParameters()</a></span> - Constructor for class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#ProjectParameters-java.lang.String-">ProjectParameters(String)</a></span> - Constructor for class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/parameters/peptide_shaker/ProjectType.html" title="enum in com.compomics.util.parameters.peptide_shaker"><span class="typeNameLink">ProjectType</span></a> - Enum in <a href="com/compomics/util/parameters/peptide_shaker/package-summary.html">com.compomics.util.parameters.peptide_shaker</a></dt>
<dd>
<div class="block">This enum lists the different types of projects that can be created using PeptideShaker.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Proline</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Proline.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html#Proline--">Proline()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Proline.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Proline</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#PROLINE">PROLINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Proline immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#properties">properties</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#PROPERTIES">PROPERTIES</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/PropertiesManager.html" title="class in com.compomics.util.io"><span class="typeNameLink">PropertiesManager</span></a> - Class in <a href="com/compomics/util/io/package-summary.html">com.compomics.util.io</a></dt>
<dd>
<div class="block">This class holds all user properties for ms_lims.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.html#property">property</a></span> - Variable in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">IonFeature</a></dt>
<dd>
<div class="block">The peptide feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#prot_xml">prot_xml</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#PROT_XML">PROT_XML</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/proteins/Protein.html" title="class in com.compomics.util.experiment.biology.proteins"><span class="typeNameLink">Protein</span></a> - Class in <a href="com/compomics/util/experiment/biology/proteins/package-summary.html">com.compomics.util.experiment.biology.proteins</a></dt>
<dd>
<div class="block">This class models a protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Protein.html#Protein--">Protein()</a></span> - Constructor for class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Protein.html" title="class in com.compomics.util.experiment.biology.proteins">Protein</a></dt>
<dd>
<div class="block">Constructor for a protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Protein.html#Protein-java.lang.String-">Protein(String)</a></span> - Constructor for class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Protein.html" title="class in com.compomics.util.experiment.biology.proteins">Protein</a></dt>
<dd>
<div class="block">Simplistic constructor for a protein (typically used when loading
 identification files).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Protein.html#Protein-java.lang.String-java.lang.String-">Protein(String, String)</a></span> - Constructor for class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Protein.html" title="class in com.compomics.util.experiment.biology.proteins">Protein</a></dt>
<dd>
<div class="block">Constructor for a protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Protein.html#Protein-java.lang.String-java.lang.String-boolean-">Protein(String, String, boolean)</a></span> - Constructor for class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Protein.html" title="class in com.compomics.util.experiment.biology.proteins">Protein</a></dt>
<dd>
<div class="block">Constructor for a protein.</div>
</dd>
<dt><a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein"><span class="typeNameLink">Protein</span></a> - Class in <a href="com/compomics/util/protein/package-summary.html">com.compomics.util.protein</a></dt>
<dd>
<div class="block">This class implements the behaviour for a Protein instance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#Protein-com.compomics.util.protein.AASequenceImpl-">Protein(AASequenceImpl)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This constructor requires an AASequenceImpl as argument.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#Protein-com.compomics.util.experiment.io.biology.protein.Header-com.compomics.util.protein.AASequenceImpl-">Protein(Header, AASequenceImpl)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This constructor allows the passing of a Header, as well as an
 AASequenceImpl for this Protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#Protein-com.compomics.util.experiment.io.biology.protein.Header-com.compomics.util.protein.AASequenceImpl-boolean-int-">Protein(Header, AASequenceImpl, boolean, int)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This constructor allows the passing of a Header, as well as an
 AASequenceImpl for this Protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#Protein-java.lang.String-">Protein(String)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This constructor allows for the construction of a Protein instance
 by passing a FASTA entry.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#Protein-java.lang.String-boolean-int-">Protein(String, boolean, int)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This constructor allows for the construction of a Protein instance
 by passing a FASTA entry and boolean flag for truncation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#Protein-java.lang.String-java.lang.String-">Protein(String, String)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This constructor allows for the construction of a Protein instance through the
 passing of a Header String and a Sequence String.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#Protein-java.lang.String-java.lang.String-boolean-int-">Protein(String, String, boolean, int)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This constructor allows for the construction of a Protein instance through the
 passing of a Header String and a Sequence String.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html" title="enum in com.compomics.util.experiment.io.biology.protein"><span class="typeNameLink">ProteinDatabase</span></a> - Enum in <a href="com/compomics/util/experiment/io/biology/protein/package-summary.html">com.compomics.util.experiment.io.biology.protein</a></dt>
<dd>
<div class="block">Enum of the supported protein databases.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/ProteinDetailsProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein"><span class="typeNameLink">ProteinDetailsProvider</span></a> - Interface in <a href="com/compomics/util/experiment/io/biology/protein/package-summary.html">com.compomics.util.experiment.io.biology.protein</a></dt>
<dd>
<div class="block">Interface for a class able to provide metadata on proteins.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html" title="class in com.compomics.util.experiment.identification.filtering"><span class="typeNameLink">ProteinFilter</span></a> - Class in <a href="com/compomics/util/experiment/identification/filtering/package-summary.html">com.compomics.util.experiment.identification.filtering</a></dt>
<dd>
<div class="block">Protein filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html#ProteinFilter--">ProteinFilter()</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html" title="class in com.compomics.util.experiment.identification.filtering">ProteinFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html#ProteinFilter-java.lang.String-">ProteinFilter(String)</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html" title="class in com.compomics.util.experiment.identification.filtering">ProteinFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html#ProteinFilter-java.lang.String-java.lang.String-">ProteinFilter(String, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html" title="class in com.compomics.util.experiment.identification.filtering">ProteinFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html#ProteinFilter-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String-">ProteinFilter(String, String, String, String, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/ProteinFilter.html" title="class in com.compomics.util.experiment.identification.filtering">ProteinFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items"><span class="typeNameLink">ProteinFilterItem</span></a> - Enum in <a href="com/compomics/util/experiment/identification/filtering/items/package-summary.html">com.compomics.util.experiment.identification.filtering.items</a></dt>
<dd>
<div class="block">Enum of the different items a protein filter can filter on.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes"><span class="typeNameLink">ProteinGeneDetailsProvider</span></a> - Class in <a href="com/compomics/util/experiment/biology/genes/package-summary.html">com.compomics.util.experiment.biology.genes</a></dt>
<dd>
<div class="block">Class used to map proteins to gene information.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#ProteinGeneDetailsProvider--">ProteinGeneDetailsProvider()</a></span> - Constructor for class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#proteinIdentification">proteinIdentification</a></span> - Variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">List of the keys of all imported proteins.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced"><span class="typeNameLink">ProteinInferenceParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/advanced/package-summary.html">com.compomics.util.parameters.identification.advanced</a></dt>
<dd>
<div class="block">Generic class grouping the protein inference preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#ProteinInferenceParameters--">ProteinInferenceParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/advanced/ProteinInferenceParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced"><span class="typeNameLink">ProteinInferenceParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/advanced/package-summary.html">com.compomics.util.gui.parameters.identification.advanced</a></dt>
<dd>
<div class="block">Dialog for the edition of the protein inference settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ProteinInferenceParametersDialog.html#ProteinInferenceParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.ProteinInferenceParameters-java.awt.Image-java.awt.Image-boolean-">ProteinInferenceParametersDialog(Frame, ProteinInferenceParameters, Image, Image, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ProteinInferenceParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ProteinInferenceParametersDialog</a></dt>
<dd>
<div class="block">Creates a new ProteinInferenceSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ProteinInferenceParametersDialog.html#ProteinInferenceParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.ProteinInferenceParameters-java.awt.Image-java.awt.Image-boolean-">ProteinInferenceParametersDialog(Dialog, Frame, ProteinInferenceParameters, Image, Image, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ProteinInferenceParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ProteinInferenceParametersDialog</a></dt>
<dd>
<div class="block">Creates a new ProteinInferenceSettingsDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/ProteinIterator.html" title="interface in com.compomics.util.experiment.io.biology.protein"><span class="typeNameLink">ProteinIterator</span></a> - Interface in <a href="com/compomics/util/experiment/io/biology/protein/package-summary.html">com.compomics.util.experiment.io.biology.protein</a></dt>
<dd>
<div class="block">Interface for a protein iterator.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion"><span class="typeNameLink">ProteinIteratorUtils</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/digestion/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion</a></dt>
<dd>
<div class="block">This class contains utilities used during the iteration of protein sequences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html#ProteinIteratorUtils-java.util.ArrayList-java.lang.Integer-">ProteinIteratorUtils(ArrayList&lt;String&gt;, Integer)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">ProteinIteratorUtils</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#proteinLynx">proteinLynx</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The ProteinLynx search engine, waters instruments.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches"><span class="typeNameLink">ProteinMatch</span></a> - Class in <a href="com/compomics/util/experiment/identification/matches/package-summary.html">com.compomics.util.experiment.identification.matches</a></dt>
<dd>
<div class="block">This class models a protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#ProteinMatch--">ProteinMatch()</a></span> - Constructor for class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Constructor for the protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#ProteinMatch-java.lang.String-">ProteinMatch(String)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Constructor for the protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#ProteinMatch-com.compomics.util.experiment.biology.proteins.Peptide-long-">ProteinMatch(Peptide, long)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Constructor for the protein match.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/matches_iterators/ProteinMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators"><span class="typeNameLink">ProteinMatchesIterator</span></a> - Class in <a href="com/compomics/util/experiment/identification/matches_iterators/package-summary.html">com.compomics.util.experiment.identification.matches_iterators</a></dt>
<dd>
<div class="block">An iterator of protein matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/ProteinMatchesIterator.html#ProteinMatchesIterator-com.compomics.util.experiment.identification.Identification-com.compomics.util.waiting.WaitingHandler-boolean-">ProteinMatchesIterator(Identification, WaitingHandler, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/ProteinMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">ProteinMatchesIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/ProteinMatchesIterator.html#ProteinMatchesIterator-long:A-com.compomics.util.experiment.identification.Identification-com.compomics.util.waiting.WaitingHandler-boolean-">ProteinMatchesIterator(long[], Identification, WaitingHandler, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/ProteinMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">ProteinMatchesIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#proteinPilot">proteinPilot</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The ProteinPilot search engine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#proteinProspector">proteinProspector</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The search engine results of Protein Prospector.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#proteinScape">proteinScape</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The search engine integrated in Protein Scape, Bruker instruments.</div>
</dd>
<dt><a href="com/compomics/util/gui/protein/ProteinSequencePane.html" title="class in com.compomics.util.gui.protein"><span class="typeNameLink">ProteinSequencePane</span></a> - Class in <a href="com/compomics/util/gui/protein/package-summary.html">com.compomics.util.gui.protein</a></dt>
<dd>
<div class="block">This class contains a method that formats a given protein sequence such that 
 both the covered parts of the sequence and the peptide selected in the peptide 
 table is highlighted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ProteinSequencePane.html#ProteinSequencePane--">ProteinSequencePane()</a></span> - Constructor for class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ProteinSequencePane.html" title="class in com.compomics.util.gui.protein">ProteinSequencePane</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html" title="class in com.compomics.util.experiment.identification.utils"><span class="typeNameLink">ProteinUtils</span></a> - Class in <a href="com/compomics/util/experiment/identification/utils/package-summary.html">com.compomics.util.experiment.identification.utils</a></dt>
<dd>
<div class="block">Convenience functions for the handling of proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html#ProteinUtils--">ProteinUtils()</a></span> - Constructor for class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ProteinUtils.html" title="class in com.compomics.util.experiment.identification.utils">ProteinUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard"><span class="typeNameLink">ProteoWizardFilter</span></a> - Enum in <a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/package-summary.html">com.compomics.util.experiment.mass_spectrometry.proteowizard</a></dt>
<dd>
<div class="block">Filters which can be applied to msconvert.</div>
</dd>
<dt><a href="com/compomics/software/dialogs/ProteoWizardSetupDialog.html" title="class in com.compomics.software.dialogs"><span class="typeNameLink">ProteoWizardSetupDialog</span></a> - Class in <a href="com/compomics/software/dialogs/package-summary.html">com.compomics.software.dialogs</a></dt>
<dd>
<div class="block">A dialog used to set up the connection to ProteoWizard.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/ProteoWizardSetupDialog.html#ProteoWizardSetupDialog-javax.swing.JFrame-boolean-">ProteoWizardSetupDialog(JFrame, boolean)</a></span> - Constructor for class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/ProteoWizardSetupDialog.html" title="class in com.compomics.software.dialogs">ProteoWizardSetupDialog</a></dt>
<dd>
<div class="block">Creates a new ProteoWizardSetupDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/ProteoWizardSetupDialog.html#ProteoWizardSetupDialog-javax.swing.JDialog-boolean-">ProteoWizardSetupDialog(JDialog, boolean)</a></span> - Constructor for class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/ProteoWizardSetupDialog.html" title="class in com.compomics.software.dialogs">ProteoWizardSetupDialog</a></dt>
<dd>
<div class="block">Creates a new ProteoWizardSetupDialog.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/Protocol.html" title="class in com.compomics.util.pride.prideobjects"><span class="typeNameLink">Protocol</span></a> - Class in <a href="com/compomics/util/pride/prideobjects/package-summary.html">com.compomics.util.pride.prideobjects</a></dt>
<dd>
<div class="block">An object for storing Protocol details.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Protocol.html#Protocol-java.lang.String-java.util.ArrayList-">Protocol(String, ArrayList&lt;CvTerm&gt;)</a></span> - Constructor for class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Protocol.html" title="class in com.compomics.util.pride.prideobjects">Protocol</a></dt>
<dd>
<div class="block">Create a new Protocol object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#PROTON">PROTON</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>
<div class="block">Subtype int for a proton.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#proton">proton</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>
<div class="block">A proton.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html#protonMassMultiples">protonMassMultiples</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ElementaryIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ElementaryIon</a></dt>
<dd>
<div class="block">Cache for the multiples of the proton mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#protxml">protxml</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#PROTXML">PROTXML</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/ProtXmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">ProtXmlFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.prot.xml, *.protxml and *.xml files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/ProtXmlFileFilter.html#ProtXmlFileFilter--">ProtXmlFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/ProtXmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">ProtXmlFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html" title="class in com.compomics.util.experiment.identification.filtering"><span class="typeNameLink">PsmFilter</span></a> - Class in <a href="com/compomics/util/experiment/identification/filtering/package-summary.html">com.compomics.util.experiment.identification.filtering</a></dt>
<dd>
<div class="block">PSM filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html#PsmFilter-java.lang.String-">PsmFilter(String)</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PsmFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html#PsmFilter--">PsmFilter()</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PsmFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html#PsmFilter-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String-">PsmFilter(String, String, String, String, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PsmFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PsmFilter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items"><span class="typeNameLink">PsmFilterItem</span></a> - Enum in <a href="com/compomics/util/experiment/identification/filtering/items/package-summary.html">com.compomics.util.experiment.identification.filtering.items</a></dt>
<dd>
<div class="block">Enum of the different items a PSM filter can filter on.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker"><span class="typeNameLink">PSModificationScores</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_shaker/package-summary.html">com.compomics.util.experiment.identification.peptide_shaker</a></dt>
<dd>
<div class="block">This class contains the scores for the locations of the possible
 modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html#PSModificationScores--">PSModificationScores()</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSModificationScores.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSModificationScores</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html" title="enum in com.compomics.util.experiment.identification.psm_scoring"><span class="typeNameLink">PsmScore</span></a> - Enum in <a href="com/compomics/util/experiment/identification/psm_scoring/package-summary.html">com.compomics.util.experiment.identification.psm_scoring</a></dt>
<dd>
<div class="block">Enum listing the PSM scores implemented in compomics utilities.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScoresEstimator.html" title="class in com.compomics.util.experiment.identification.psm_scoring"><span class="typeNameLink">PsmScoresEstimator</span></a> - Class in <a href="com/compomics/util/experiment/identification/psm_scoring/package-summary.html">com.compomics.util.experiment.identification.psm_scoring</a></dt>
<dd>
<div class="block">This class can be used to estimate PSM scores.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScoresEstimator.html#PsmScoresEstimator-com.compomics.util.experiment.identification.peptide_fragmentation.PeptideFragmentationModel-">PsmScoresEstimator(PeptideFragmentationModel)</a></span> - Constructor for class com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScoresEstimator.html" title="class in com.compomics.util.experiment.identification.psm_scoring">PsmScoresEstimator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScoresEstimator.html#PsmScoresEstimator--">PsmScoresEstimator()</a></span> - Constructor for class com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScoresEstimator.html" title="class in com.compomics.util.experiment.identification.psm_scoring">PsmScoresEstimator</a></dt>
<dd>
<div class="block">Constructor using a uniform fragmentation.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced"><span class="typeNameLink">PsmScoringParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/advanced/package-summary.html">com.compomics.util.parameters.identification.advanced</a></dt>
<dd>
<div class="block">Generic class for peptide spectrum match scoring.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#PsmScoringParameters--">PsmScoringParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/advanced/PsmScoringParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced"><span class="typeNameLink">PsmScoringParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/advanced/package-summary.html">com.compomics.util.gui.parameters.identification.advanced</a></dt>
<dd>
<div class="block">Dialog for the edition of the PSM scoring settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/PsmScoringParametersDialog.html#PsmScoringParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.PsmScoringParameters-boolean-">PsmScoringParametersDialog(Frame, PsmScoringParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/PsmScoringParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">PsmScoringParametersDialog</a></dt>
<dd>
<div class="block">Creates a new PsmScoringSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/PsmScoringParametersDialog.html#PsmScoringParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.PsmScoringParameters-boolean-">PsmScoringParametersDialog(Dialog, Frame, PsmScoringParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/PsmScoringParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">PsmScoringParametersDialog</a></dt>
<dd>
<div class="block">Creates a new PsmScoringSettingsDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker"><span class="typeNameLink">PSParameter</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_shaker/package-summary.html">com.compomics.util.experiment.identification.peptide_shaker</a></dt>
<dd>
<div class="block">PeptideShaker compomics utilities experiment customizable parameter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#PSParameter--">PSParameter()</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/io/json/marshallers/PTMFactoryMarshaller.html" title="class in com.compomics.util.io.json.marshallers"><span class="typeNameLink">PTMFactoryMarshaller</span></a> - Class in <a href="com/compomics/util/io/json/marshallers/package-summary.html">com.compomics.util.io.json.marshallers</a></dt>
<dd>
<div class="block">This class is a convenience class to have a DefaultJsonConverter with the
 search parameter interfaces.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/marshallers/PTMFactoryMarshaller.html#PTMFactoryMarshaller--">PTMFactoryMarshaller()</a></span> - Constructor for class com.compomics.util.io.json.marshallers.<a href="com/compomics/util/io/json/marshallers/PTMFactoryMarshaller.html" title="class in com.compomics.util.io.json.marshallers">PTMFactoryMarshaller</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/pride/PtmToPrideMap.html" title="class in com.compomics.util.pride"><span class="typeNameLink">PtmToPrideMap</span></a> - Class in <a href="com/compomics/util/pride/package-summary.html">com.compomics.util.pride</a></dt>
<dd>
<div class="block">Map linking user modification names to Unimod CV terms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PtmToPrideMap.html#PtmToPrideMap--">PtmToPrideMap()</a></span> - Constructor for class com.compomics.util.pride.<a href="com/compomics/util/pride/PtmToPrideMap.html" title="class in com.compomics.util.pride">PtmToPrideMap</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/io/PushBackStringReader.html" title="class in com.compomics.util.io"><span class="typeNameLink">PushBackStringReader</span></a> - Class in <a href="com/compomics/util/io/package-summary.html">com.compomics.util.io</a></dt>
<dd>
<div class="block">This class implements a PushBackReader for Strings which allows the reading
 of entire lines as well as the pushing back of entire lines.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PushBackStringReader.html#PushBackStringReader-java.lang.String-">PushBackStringReader(String)</a></span> - Constructor for class com.compomics.util.io.<a href="com/compomics/util/io/PushBackStringReader.html" title="class in com.compomics.util.io">PushBackStringReader</a></dt>
<dd>
<div class="block">This constructor requires the passing of a String from which to
 construct the PushBackStringReader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#putAscii-java.lang.String-">putAscii(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">PUT a file on the FTP server in Ascii mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#putBinary-java.lang.String-">putBinary(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">PUT a file to the FTP server in Binary mode</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PtmToPrideMap.html#putCVTerm-java.lang.String-com.compomics.util.pride.CvTerm-">putCVTerm(String, CvTerm)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PtmToPrideMap.html" title="class in com.compomics.util.pride">PtmToPrideMap</a></dt>
<dd>
<div class="block">Puts a new mapping in the map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#pwd--">pwd()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Get the name of the present working directory on the ftp server file system.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Pyrrolysine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Pyrrolysine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html#Pyrrolysine--">Pyrrolysine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Pyrrolysine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Pyrrolysine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#PYRROLYSINE">PYRROLYSINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Pyrrolysine immonium ion.</div>
</dd>
</dl>
<a name="I:Q">
<!--   -->
</a>
<h2 class="title">Q</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#Q">Q</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#QQ">QQ</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for SUMO-2/3 Q87R.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#QQ_H2O">QQ_H2O</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for SUMO-2/3 Q87R.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#QQT">QQT</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for SUMO-2/3 Q87R.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#QQT_H2O">QQT_H2O</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for SUMO-2/3 Q87R.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#QQTG">QQTG</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for SUMO-2/3 Q87R.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#QQTG_H2O">QQTG_H2O</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for SUMO-2/3 Q87R.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#QQTGG">QQTGG</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for SUMO-2/3 Q87R.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#QQTGG_H2O">QQTGG_H2O</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion for SUMO-2/3 Q87R.</div>
</dd>
<dt><a href="com/compomics/util/experiment/quantification/Quantification.html" title="class in com.compomics.util.experiment.quantification"><span class="typeNameLink">Quantification</span></a> - Class in <a href="com/compomics/util/experiment/quantification/package-summary.html">com.compomics.util.experiment.quantification</a></dt>
<dd>
<div class="block">An abstract quantification class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/Quantification.html#Quantification--">Quantification()</a></span> - Constructor for class com.compomics.util.experiment.quantification.<a href="com/compomics/util/experiment/quantification/Quantification.html" title="class in com.compomics.util.experiment.quantification">Quantification</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/quantification/Quantification.QuantificationMethod.html" title="enum in com.compomics.util.experiment.quantification"><span class="typeNameLink">Quantification.QuantificationMethod</span></a> - Enum in <a href="com/compomics/util/experiment/quantification/package-summary.html">com.compomics.util.experiment.quantification</a></dt>
<dd>
<div class="block">The implemented quantification methods.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#queryEnsembl-java.lang.String-java.io.File-java.lang.String-">queryEnsembl(String, File, String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Sends an XML query to Ensembl and writes the result in a text file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#queryEnsembl-java.lang.String-java.io.File-java.lang.String-com.compomics.util.waiting.WaitingHandler-">queryEnsembl(String, File, String, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Sends an XML query to Ensembl and writes the result in a text file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#queryEnsembl-java.lang.String-java.lang.String-java.io.File-java.lang.String-com.compomics.util.waiting.WaitingHandler-">queryEnsembl(String, String, File, String, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Sends an XML query to Ensembl and writes the result in a text file.</div>
</dd>
</dl>
<a name="I:R">
<!--   -->
</a>
<h2 class="title">R</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#R">R</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#RANDOM">RANDOM</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Index for a random location choice.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#rangeQuery-int-int-">rangeQuery(int, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Returns a list of character and new left/right index for a given range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#rangeQuery-int-int-int:A:A-">rangeQuery(int, int, int[][])</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Fills a list of character and new left/right index for a given range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#rangeQueryOneValue-int-int:A:A-">rangeQueryOneValue(int, int[][])</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Fills a list of character and new left/right index for a given index.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index"><span class="typeNameLink">Rank</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_inference/fm_index/package-summary.html">com.compomics.util.experiment.identification.protein_inference.fm_index</a></dt>
<dd>
<div class="block">Rank as used in the FM index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html#Rank-byte:A-long:A-">Rank(byte[], long[])</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">Rank</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html#Rank-long:A-int-">Rank(long[], int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/Rank.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">Rank</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#rank">rank</a></span> - Variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">The rank of the peptide assumption for the concerned spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html#rawFormat">rawFormat</a></span> - Variable in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsFormat</a></dt>
<dd>
<div class="block">Boolean indicating if this in a format for raw data, false means that it
 is a peak list format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#rawScore">rawScore</a></span> - Variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">The raw score as provided by the identification algorithm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableFileInputStream.html#read-byte:A-">read(byte[])</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableFileInputStream.html" title="class in com.compomics.util.io.file">MonitorableFileInputStream</a></dt>
<dd>
<div class="block">Reads up to <code>byte.length</code> bytes of data from this
 input stream into an array of bytes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableFileInputStream.html#read--">read()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableFileInputStream.html" title="class in com.compomics.util.io.file">MonitorableFileInputStream</a></dt>
<dd>
<div class="block">Reads the next byte of data from this input stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableFileInputStream.html#read-byte:A-int-int-">read(byte[], int, int)</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableFileInputStream.html" title="class in com.compomics.util.io.file">MonitorableFileInputStream</a></dt>
<dd>
<div class="block">Reads up to <code>len</code> bytes of data from this input stream
 into an array of bytes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableInputStream.html#read-byte:A-">read(byte[])</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableInputStream.html" title="class in com.compomics.util.io.file">MonitorableInputStream</a></dt>
<dd>
<div class="block">Reads up to <code>byte.length</code> bytes of data from this
 input stream into an array of bytes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableInputStream.html#read--">read()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableInputStream.html" title="class in com.compomics.util.io.file">MonitorableInputStream</a></dt>
<dd>
<div class="block">Reads the next byte of data from this input stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableInputStream.html#read-byte:A-int-int-">read(byte[], int, int)</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableInputStream.html" title="class in com.compomics.util.io.file">MonitorableInputStream</a></dt>
<dd>
<div class="block">Reads up to <code>len</code> bytes of data from this input stream
 into an array of bytes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/adapter/FileAdapter.html#read-com.google.gson.stream.JsonReader-">read(JsonReader)</a></span> - Method in class com.compomics.util.io.json.adapter.<a href="com/compomics/util/io/json/adapter/FileAdapter.html" title="class in com.compomics.util.io.json.adapter">FileAdapter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#readDoubleAsString-java.lang.String-">readDoubleAsString(String)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Method for reading a double value as a string which uses either "," or
 "." as the decimal symbol.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#readFasta-java.lang.String-java.lang.String-">readFasta(String, String)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Read a FASTA.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PushBackStringReader.html#readLine--">readLine()</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PushBackStringReader.html" title="class in com.compomics.util.io">PushBackStringReader</a></dt>
<dd>
<div class="block">This method reads a line of text from the buffer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/SerializationUtils.html#readObject-java.io.File-">readObject(File)</a></span> - Static method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/SerializationUtils.html" title="class in com.compomics.util.io.file">SerializationUtils</a></dt>
<dd>
<div class="block">Reads an object from a serialized file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#readReply--">readReply()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">This method attempts to read a reply from the FTP server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#readServerResponse--">readServerResponse()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">This method allows the caller to capture the server response.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#readUrl-java.lang.String-java.lang.String-">readUrl(String, String)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Read a URL.</div>
</dd>
<dt><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion"><span class="typeNameLink">Reagent</span></a> - Class in <a href="com/compomics/util/experiment/quantification/reporterion/package-summary.html">com.compomics.util.experiment.quantification.reporterion</a></dt>
<dd>
<div class="block">A reporter ion labeling reagent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#Reagent--">Reagent()</a></span> - Constructor for class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#reallocateIndexes--">reallocateIndexes()</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/pride/prideobjects/Reference.html" title="class in com.compomics.util.pride.prideobjects"><span class="typeNameLink">Reference</span></a> - Class in <a href="com/compomics/util/pride/prideobjects/package-summary.html">com.compomics.util.pride.prideobjects</a></dt>
<dd>
<div class="block">A simple publication reference object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Reference.html#Reference-java.lang.String-java.lang.String-java.lang.String-">Reference(String, String, String)</a></span> - Constructor for class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Reference.html" title="class in com.compomics.util.pride.prideobjects">Reference</a></dt>
<dd>
<div class="block">Create a new Reference object.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum"><span class="typeNameLink">ReferenceArea</span></a> - Class in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">A reference area to be added to a GraphicsPanel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#ReferenceArea-java.lang.String-java.lang.String-double-double-java.awt.Color-float-boolean-boolean-boolean-">ReferenceArea(String, String, double, double, Color, float, boolean, boolean, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Creates a new ReferenceArea.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#ReferenceArea-java.lang.String-java.lang.String-double-double-java.awt.Color-float-boolean-boolean-java.awt.Color-boolean-java.awt.Color-float-double-boolean-">ReferenceArea(String, String, double, double, Color, float, boolean, boolean, Color, boolean, Color, float, double, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Creates a new ReferenceArea.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html" title="class in com.compomics.util.pride.prideobjects"><span class="typeNameLink">ReferenceGroup</span></a> - Class in <a href="com/compomics/util/pride/prideobjects/package-summary.html">com.compomics.util.pride.prideobjects</a></dt>
<dd>
<div class="block">An object for storing ReferenceGroup details.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html#ReferenceGroup-java.util.ArrayList-java.lang.String-">ReferenceGroup(ArrayList&lt;Reference&gt;, String)</a></span> - Constructor for class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html" title="class in com.compomics.util.pride.prideobjects">ReferenceGroup</a></dt>
<dd>
<div class="block">Create a new ReferenceGroup object.</div>
</dd>
<dt><a href="com/compomics/util/protein/RegExEnzyme.html" title="class in com.compomics.util.protein"><span class="typeNameLink">RegExEnzyme</span></a> - Class in <a href="com/compomics/util/protein/package-summary.html">com.compomics.util.protein</a></dt>
<dd>
<div class="block">This class implements the functionality of an Enzyme by
 simulating digestion based on a regular expression.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/RegExEnzyme.html#RegExEnzyme-java.lang.String-java.lang.String-java.lang.String-java.lang.String-">RegExEnzyme(String, String, String, String)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/RegExEnzyme.html" title="class in com.compomics.util.protein">RegExEnzyme</a></dt>
<dd>
<div class="block">Create a new RegExEnzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/RegExEnzyme.html#RegExEnzyme-java.lang.String-java.lang.String-java.lang.String-java.lang.String-int-">RegExEnzyme(String, String, String, String, int)</a></span> - Constructor for class com.compomics.util.protein.<a href="com/compomics/util/protein/RegExEnzyme.html" title="class in com.compomics.util.protein">RegExEnzyme</a></dt>
<dd>
<div class="block">Create a new RegExEnzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/IdfileReaderFactory.html#registerIdFileReader-java.lang.Class-java.lang.String-">registerIdFileReader(Class, String)</a></span> - Static method in class com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/IdfileReaderFactory.html" title="class in com.compomics.util.experiment.io.identification">IdfileReaderFactory</a></dt>
<dd>
<div class="block">This method registers a new IdfileReader Class, and the file extension it
 can read from.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/file_handling/TempFilesManager.html#registerTempFolder-java.io.File-">registerTempFolder(File)</a></span> - Static method in class com.compomics.util.gui.file_handling.<a href="com/compomics/util/gui/file_handling/TempFilesManager.html" title="class in com.compomics.util.gui.file_handling">TempFilesManager</a></dt>
<dd>
<div class="block">Adds a temp folder to the references temp folders.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html" title="class in com.compomics.util.math.statistics.linear_regression"><span class="typeNameLink">RegressionStatistics</span></a> - Class in <a href="com/compomics/util/math/statistics/linear_regression/package-summary.html">com.compomics.util.math.statistics.linear_regression</a></dt>
<dd>
<div class="block">Results of a linear regression of equation y = a.x + b and Pearson
 correlation r.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html#RegressionStatistics-java.lang.Double-java.lang.Double-java.lang.Double-java.lang.Double-java.lang.Double-">RegressionStatistics(Double, Double, Double, Double, Double)</a></span> - Constructor for class com.compomics.util.math.statistics.linear_regression.<a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html" title="class in com.compomics.util.math.statistics.linear_regression">RegressionStatistics</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#RELATED">RELATED</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Static index for a protein group: 1 - related proteins or peptide from
 related protein groups (not necessarily unique to the group).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#RELATED_AND_UNRELATED">RELATED_AND_UNRELATED</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Static index for a protein group: 2 - related and a unrelated proteins or
 peptide shared by related and unrelated proteins (not necessarily unique
 to the group).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_C_1">RELATED_C_1</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for C.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_C_2">RELATED_C_2</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for C.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_D_1">RELATED_D_1</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for D.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_E_1">RELATED_E_1</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for E.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_F_1">RELATED_F_1</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for F.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_H_1">RELATED_H_1</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for H (C4H5N2).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_H_2">RELATED_H_2</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for H.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_K_1">RELATED_K_1</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for K.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_K_2">RELATED_K_2</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for K.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_K_3">RELATED_K_3</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for K.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_M_1">RELATED_M_1</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for M (C2H4S).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_N_1">RELATED_N_1</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for N.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_Q_1">RELATED_Q_1</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for Q.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_Q_2">RELATED_Q_2</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for Q.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_R_1">RELATED_R_1</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for R.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_R_2">RELATED_R_2</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for R.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_R_3">RELATED_R_3</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for R.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_R_4">RELATED_R_4</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for R (C5H10N2O).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_W_1">RELATED_W_1</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for W (C9H7N).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RELATED_Y_1">RELATED_Y_1</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Related ion for Y (C7H7O).</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl"><span class="typeNameLink">RelatedIon</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/impl/package-summary.html">com.compomics.util.experiment.biology.ions.impl</a></dt>
<dd>
<div class="block">Represents a related ion, i.e., an ion that is related to a given amino acid,
 and is its own factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RelatedIon-com.compomics.util.experiment.biology.aminoacids.AminoAcid-com.compomics.util.experiment.biology.atoms.AtomChain-int-">RelatedIon(AminoAcid, AtomChain, int)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Constructor for a related ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html#RelatedIon-com.compomics.util.experiment.biology.aminoacids.AminoAcid-com.compomics.util.experiment.biology.atoms.AtomChain-int-boolean-">RelatedIon(AminoAcid, AtomChain, int, boolean)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/RelatedIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">RelatedIon</a></dt>
<dd>
<div class="block">Constructor for a related ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/threading/MapMutex.html#release-K-">release(K)</a></span> - Method in class com.compomics.util.threading.<a href="com/compomics/util/threading/MapMutex.html" title="class in com.compomics.util.threading">MapMutex</a></dt>
<dd>
<div class="block">Releases the permit for the given key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/threading/ObjectMutex.html#release-long-">release(long)</a></span> - Method in class com.compomics.util.threading.<a href="com/compomics/util/threading/ObjectMutex.html" title="class in com.compomics.util.threading">ObjectMutex</a></dt>
<dd>
<div class="block">Release function for the given key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationFactory.html#reLoadAllData--">reLoadAllData()</a></span> - Static method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationFactory.html" title="class in com.compomics.util.protein">ModificationFactory</a></dt>
<dd>
<div class="block">This method can be used to refresh all data from the data store indicated in the ModificationFactory.properties
 file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#reloadFactory--">reloadFactory()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Reloads the factory getInstance() needs to be called afterwards.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/interfaces/DBElement.html#remove-java.sql.Connection-">remove(Connection)</a></span> - Method in interface com.compomics.util.db.interfaces.<a href="com/compomics/util/db/interfaces/DBElement.html" title="interface in com.compomics.util.db.interfaces">DBElement</a></dt>
<dd>
<div class="block">This method provides a convenient short-cut to delete the DBElement from
 the persistent store.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#remove-com.compomics.util.experiment.biology.atoms.Atom-java.lang.Integer-">remove(Atom, Integer)</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>
<div class="block">Removes all the occurrences of the given atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#removeAA-int-">removeAA(int)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Removes an amino acid index from the pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#removeAllReferenceAreasXAxis--">removeAllReferenceAreasXAxis()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Removes all the x-axis reference areas.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#removeAllReferenceAreasYAxis--">removeAllReferenceAreasYAxis()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Removes all the y-axis reference areas.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#removeCharacters-java.lang.String-char-">removeCharacters(String, char)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Removes characters from a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#removeCharacters-java.lang.String-java.lang.String:A-">removeCharacters(String, String[])</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Removes characters from a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#removeDuplicateSpectrumTitles-java.io.File-com.compomics.util.waiting.WaitingHandler-">removeDuplicateSpectrumTitles(File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Removes duplicate spectrum titles (the first occurrence is kept).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomChainDialog.html#removeElementsPanel-int-">removeElementsPanel(int)</a></span> - Method in class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomChainDialog.html" title="class in com.compomics.util.gui.atoms">AtomChainDialog</a></dt>
<dd>
<div class="block">Remove the given elements panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#removeEnzyme-java.lang.String-">removeEnzyme(String)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Removes an enzyme from the mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#removeException-java.lang.Long-">removeException(Long)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Removes an exception.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFactory.html#removeExportScheme-java.lang.String-">removeExportScheme(String)</a></span> - Method in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFactory.html" title="interface in com.compomics.util.io.export">ExportFactory</a></dt>
<dd>
<div class="block">Removes a user scheme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#removeExtension-java.lang.String-">removeExtension(String)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Removes the extension from a file name or path.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#removeFilterItem-java.lang.String-">removeFilterItem(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Removes an item from the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#removeFixedModification-java.lang.String-">removeFixedModification(String)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Removes a modification from the list of fixed modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#removeForbiddenCharacters-java.lang.String-">removeForbiddenCharacters(String)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Removes the forbidden characters from a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html#removeIdentificationParameters-java.lang.String-">removeIdentificationParameters(String)</a></span> - Method in class com.compomics.util.experiment.identification.identification_parameters.<a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html" title="class in com.compomics.util.experiment.identification.identification_parameters">IdentificationParametersFactory</a></dt>
<dd>
<div class="block">Deletes the identification parameters of the given name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#removeManualValidation-java.lang.Long-">removeManualValidation(Long)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Removes a manually validated Match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#removeMouseListenerToHeaderInTable-javax.swing.JTable-">removeMouseListenerToHeaderInTable(JTable)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#removeNeutralLoss-java.lang.String-">removeNeutralLoss(String)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Removes the neutral loss associated to the given name in the static map
 of the class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#removeObject-long-">removeObject(long)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Removes an object from the cache</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#removeObject-long-">removeObject(long)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Removing an object from the cache and database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#removeObject-long-">removeObject(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Removes an object from the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#removeObjects-java.util.Collection-com.compomics.util.waiting.WaitingHandler-boolean-">removeObjects(Collection&lt;Long&gt;, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Removing an object from the cache and database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#removeObjects-java.util.ArrayList-com.compomics.util.waiting.WaitingHandler-boolean-">removeObjects(ArrayList&lt;Long&gt;, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Remove a list of objects from the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#removeObjects-com.compomics.util.experiment.identification.IdentificationFeaturesCache.ObjectType-">removeObjects(IdentificationFeaturesCache.ObjectType)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Clears all objects of the given type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#removePeptideAssumption-com.compomics.util.experiment.identification.spectrum_assumptions.PeptideAssumption-">removePeptideAssumption(PeptideAssumption)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Removes an assumption from the mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#removeReferenceAreaXAxis-java.lang.String-">removeReferenceAreaXAxis(String)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Removes the x-axis reference area with the given identifier.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#removeReferenceAreaYAxis-java.lang.String-">removeReferenceAreaYAxis(String)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Removes the y-axis reference area with the given identifier.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#removeRefinementFixedModification-java.lang.String-">removeRefinementFixedModification(String)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Removes a fixed modification from the list of refinement modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#removeRefinementVariableModification-java.lang.String-">removeRefinementVariableModification(String)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Removes a variable modification from the list of refinement
 modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#removeSection-java.lang.String-">removeSection(String)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Removes an entire section from the mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#removeSubString-java.lang.String-java.lang.String-">removeSubString(String, String)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Removes characters from a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#removeTagAssumption-com.compomics.util.experiment.identification.spectrum_assumptions.TagAssumption-">removeTagAssumption(TagAssumption)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Removes an assumption from the mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#removeUnknownModifications--">removeUnknownModifications()</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether unknown modifications shall be
 removed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/personalization/ExperimentObject.html#removeUrParam-long-">removeUrParam(long)</a></span> - Method in class com.compomics.util.experiment.personalization.<a href="com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></dt>
<dd>
<div class="block">Removes a user parameter from the user parameters map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#removeUserPtm-java.lang.String-">removeUserPtm(String)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Removes a user Modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#removeVariableModification-java.lang.String-">removeVariableModification(String)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Removes a modification from the list of variable modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#removeZeroes-java.io.File-com.compomics.util.waiting.WaitingHandler-">removeZeroes(File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Removes zero intensity peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html#removeZeroes-java.io.File-com.compomics.util.waiting.WaitingHandler-">removeZeroes(File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspReader</a></dt>
<dd>
<div class="block">Removes zero intensity peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#rename-java.lang.String-java.lang.String-">rename(String, String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Rename a file on the remote server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#renameDuplicateSpectrumTitles-java.io.File-com.compomics.util.waiting.WaitingHandler-">renameDuplicateSpectrumTitles(File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Renames duplicate spectrum titles.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html#renameDuplicateSpectrumTitles-java.io.File-com.compomics.util.waiting.WaitingHandler-">renameDuplicateSpectrumTitles(File, WaitingHandler)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspReader</a></dt>
<dd>
<div class="block">Renames duplicate spectrum titles.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html#renameFastaFileName-java.io.File-">renameFastaFileName(File)</a></span> - Method in class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SequenceDbDetailsDialog</a></dt>
<dd>
<div class="block">Copies the content of the FASTA file to a new file and replaces any white
 space in the file name with '_' instead.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#renameXTandemFile--">renameXTandemFile()</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Returns true if the X! Tandem file should be renamed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#replaceAll-java.lang.String-char-char-">replaceAll(String, char, char)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Replaces all characters equals to a by b in the given string.</div>
</dd>
<dt><a href="com/compomics/util/gui/export/report/ReportEditor.html" title="class in com.compomics.util.gui.export.report"><span class="typeNameLink">ReportEditor</span></a> - Class in <a href="com/compomics/util/gui/export/report/package-summary.html">com.compomics.util.gui.export.report</a></dt>
<dd>
<div class="block">Dialog for editing reports.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/export/report/ReportEditor.html#ReportEditor-java.awt.Frame-com.compomics.util.io.export.ExportFactory-java.lang.String-boolean-">ReportEditor(Frame, ExportFactory, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.export.report.<a href="com/compomics/util/gui/export/report/ReportEditor.html" title="class in com.compomics.util.gui.export.report">ReportEditor</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/export/report/ReportEditor.html#ReportEditor-java.awt.Frame-com.compomics.util.io.export.ExportFactory-">ReportEditor(Frame, ExportFactory)</a></span> - Constructor for class com.compomics.util.gui.export.report.<a href="com/compomics/util/gui/export/report/ReportEditor.html" title="class in com.compomics.util.gui.export.report">ReportEditor</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl"><span class="typeNameLink">ReporterIon</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/impl/package-summary.html">com.compomics.util.experiment.biology.ions.impl</a></dt>
<dd>
<div class="block">This class models a reporter ion and is its own factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#ReporterIon-java.lang.String-double-">ReporterIon(String, double)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Constructor for a user-defined reporter ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#ReporterIon-java.lang.String-double-boolean-">ReporterIon(String, double, boolean)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Constructor for a user-defined reporter ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#ReporterIon-java.lang.String-com.compomics.util.experiment.biology.atoms.AtomChain-">ReporterIon(String, AtomChain)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Constructor for a user-defined reporter ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#ReporterIon-java.lang.String-com.compomics.util.experiment.biology.atoms.AtomChain-boolean-">ReporterIon(String, AtomChain, boolean)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Constructor for a user-defined reporter ion.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/ReporterIonAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators"><span class="typeNameLink">ReporterIonAnnotator</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators</a></dt>
<dd>
<div class="block">Annotator for reporter ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/ReporterIonAnnotator.html#ReporterIonAnnotator-com.compomics.util.experiment.biology.ions.impl.ReporterIon:A-">ReporterIonAnnotator(ReporterIon[])</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/ReporterIonAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">ReporterIonAnnotator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion"><span class="typeNameLink">ReporterIonQuantification</span></a> - Class in <a href="com/compomics/util/experiment/quantification/reporterion/package-summary.html">com.compomics.util.experiment.quantification.reporterion</a></dt>
<dd>
<div class="block">This class contains quantification parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#ReporterIonQuantification-com.compomics.util.experiment.quantification.Quantification.QuantificationMethod-">ReporterIonQuantification(Quantification.QuantificationMethod)</a></span> - Constructor for class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">Constructor for the reporter ion quantification.</div>
</dd>
<dt><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html" title="class in com.compomics.util.experiment.quantification.reporterion"><span class="typeNameLink">ReporterMethod</span></a> - Class in <a href="com/compomics/util/experiment/quantification/reporterion/package-summary.html">com.compomics.util.experiment.quantification.reporterion</a></dt>
<dd>
<div class="block">This class contains information relative to a reporter quantification method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html#ReporterMethod-java.lang.String-java.util.ArrayList-">ReporterMethod(String, ArrayList&lt;Reagent&gt;)</a></span> - Constructor for class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethod.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethod</a></dt>
<dd>
<div class="block">Constructor for a reporter method.</div>
</dd>
<dt><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html" title="class in com.compomics.util.experiment.quantification.reporterion"><span class="typeNameLink">ReporterMethodFactory</span></a> - Class in <a href="com/compomics/util/experiment/quantification/reporterion/package-summary.html">com.compomics.util.experiment.quantification.reporterion</a></dt>
<dd>
<div class="block">This factory imports reporter methods details from an XMl file.</div>
</dd>
<dt><a href="com/compomics/software/dialogs/ReporterSetupDialog.html" title="class in com.compomics.software.dialogs"><span class="typeNameLink">ReporterSetupDialog</span></a> - Class in <a href="com/compomics/software/dialogs/package-summary.html">com.compomics.software.dialogs</a></dt>
<dd>
<div class="block">A dialog used to set up the connection to Reporter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/ReporterSetupDialog.html#ReporterSetupDialog-javax.swing.JFrame-boolean-">ReporterSetupDialog(JFrame, boolean)</a></span> - Constructor for class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/ReporterSetupDialog.html" title="class in com.compomics.software.dialogs">ReporterSetupDialog</a></dt>
<dd>
<div class="block">Creates a new ReporterSetupDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#reportFailed">reportFailed</a></span> - Variable in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Report when the filter is not passed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#reportPassed">reportPassed</a></span> - Variable in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Report when the filter is passed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#representativeComposition">representativeComposition</a></span> - Variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">Map of the isotope representative composition of the stable isotopes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#rescale-double-double-">rescale(double, double)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method rescales the x-axis while notifying the observers.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#rescale-double-double-boolean-">rescale(double, double, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method rescales the x-axis, allowing the caller to specify whether
 the observers need be notified.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/interfaces/SpectrumPanelListener.html#rescaled-com.compomics.util.gui.events.RescalingEvent-">rescaled(RescalingEvent)</a></span> - Method in interface com.compomics.util.gui.interfaces.<a href="com/compomics/util/gui/interfaces/SpectrumPanelListener.html" title="interface in com.compomics.util.gui.interfaces">SpectrumPanelListener</a></dt>
<dd>
<div class="block">This method will be called whenever the SpectrumPanel is rescaled.</div>
</dd>
<dt><a href="com/compomics/util/gui/events/RescalingEvent.html" title="class in com.compomics.util.gui.events"><span class="typeNameLink">RescalingEvent</span></a> - Class in <a href="com/compomics/util/gui/events/package-summary.html">com.compomics.util.gui.events</a></dt>
<dd>
<div class="block">This class implements a resizing event that occurred on a SpectrumPanel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/events/RescalingEvent.html#RescalingEvent-javax.swing.JPanel-double-double-">RescalingEvent(JPanel, double, double)</a></span> - Constructor for class com.compomics.util.gui.events.<a href="com/compomics/util/gui/events/RescalingEvent.html" title="class in com.compomics.util.gui.events">RescalingEvent</a></dt>
<dd>
<div class="block">The constructor takes the resizing event X coordinates in mass units.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableFileInputStream.html#reset--">reset()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableFileInputStream.html" title="class in com.compomics.util.io.file">MonitorableFileInputStream</a></dt>
<dd>
<div class="block">Repositions this stream to the position at the time the
 <code>mark</code> method was last called on this input stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableInputStream.html#reset--">reset()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableInputStream.html" title="class in com.compomics.util.io.file">MonitorableInputStream</a></dt>
<dd>
<div class="block">Repositions this stream to the position at the time the
 <code>mark</code> method was last called on this input stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#resetPeptideNormalisationFactors--">resetPeptideNormalisationFactors()</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Resets the peptide normalization factors.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#resetPrimaryProgressCounter--">resetPrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#resetPrimaryProgressCounter--">resetPrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Reset the primary progress bar value to 0.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#resetPrimaryProgressCounter--">resetPrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#resetPrimaryProgressCounter--">resetPrimaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#resetPrimaryProgressCounter--">resetPrimaryProgressCounter()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Reset the primary progress counter to 0.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#resetProteinNormalisationFactors--">resetProteinNormalisationFactors()</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Resets the protein normalization factors.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/normalization/NormalizationFactors.html#resetPsmNormalisationFactors--">resetPsmNormalisationFactors()</a></span> - Method in class com.compomics.util.experiment.normalization.<a href="com/compomics/util/experiment/normalization/NormalizationFactors.html" title="class in com.compomics.util.experiment.normalization">NormalizationFactors</a></dt>
<dd>
<div class="block">Resets the PSM normalization factors.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#resetQcResults--">resetQcResults()</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Resets the results of the QC filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#resetSecondaryProgressCounter--">resetSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Resets the value of the progress bar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#resetSecondaryProgressCounter--">resetSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Reset the secondary progress bar value to 0.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#resetSecondaryProgressCounter--">resetSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#resetSecondaryProgressCounter--">resetSecondaryProgressCounter()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#resetSecondaryProgressCounter--">resetSecondaryProgressCounter()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Reset the secondary progress counter to 0.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#resetSorting-com.compomics.util.gui.waiting.waitinghandlers.ProgressDialogX-">resetSorting(ProgressDialogX)</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Sorts the table according to a given column using the built in sorter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#resizePlot-int-int-">resizePlot(int, int)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/JavaHomeOrMemoryDialogParent.html#restart--">restart()</a></span> - Method in interface com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/JavaHomeOrMemoryDialogParent.html" title="interface in com.compomics.software.dialogs">JavaHomeOrMemoryDialogParent</a></dt>
<dd>
<div class="block">Restart the given tool with the new Java options.</div>
</dd>
<dt><a href="com/compomics/util/db/interfaces/Retrievable.html" title="interface in com.compomics.util.db.interfaces"><span class="typeNameLink">Retrievable</span></a> - Interface in <a href="com/compomics/util/db/interfaces/package-summary.html">com.compomics.util.db.interfaces</a></dt>
<dd>
<div class="block">This interface indicates that the implementing class can be retrieved from permanent storage.</div>
</dd>
<dt><a href="com/compomics/util/db/components/RetrievableCode.html" title="class in com.compomics.util.db.components"><span class="typeNameLink">RetrievableCode</span></a> - Class in <a href="com/compomics/util/db/components/package-summary.html">com.compomics.util.db.components</a></dt>
<dd>
<div class="block">This class generates the code to make a DBAccessor an implementation of the Retrieveable
 interface.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/RetrievableCode.html#RetrievableCode-com.compomics.util.db.DBMetaData-">RetrievableCode(DBMetaData)</a></span> - Constructor for class com.compomics.util.db.components.<a href="com/compomics/util/db/components/RetrievableCode.html" title="class in com.compomics.util.db.components">RetrievableCode</a></dt>
<dd>
<div class="block">This constructor allows the generation of the code for the implementation of
 the Retrievable interface for a DBAccessor class, based on the specified metadata.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/interfaces/Retrievable.html#retrieve-java.sql.Connection-java.util.HashMap-">retrieve(Connection, HashMap)</a></span> - Method in interface com.compomics.util.db.interfaces.<a href="com/compomics/util/db/interfaces/Retrievable.html" title="interface in com.compomics.util.db.interfaces">Retrievable</a></dt>
<dd>
<div class="block">This method allows the implemented object to retrieve its state from the persistent
 store.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#retrieveObject-long-">retrieveObject(long)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Retrieves some objects from the database or cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#retrieveObject-long-">retrieveObject(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns an array of all objects of a given list of keys</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#retrieveObjects-java.util.Collection-com.compomics.util.waiting.WaitingHandler-boolean-">retrieveObjects(Collection&lt;Long&gt;, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">retrieves some objects from the database or cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#retrieveObjects-java.lang.Class-com.compomics.util.waiting.WaitingHandler-boolean-">retrieveObjects(Class, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Retrieves all objects from a given class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#retrieveObjects-java.util.Collection-com.compomics.util.waiting.WaitingHandler-boolean-">retrieveObjects(Collection&lt;Long&gt;, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns an array of all objects of a given list of keys</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#retrieveObjects-java.lang.Class-com.compomics.util.waiting.WaitingHandler-boolean-">retrieveObjects(Class, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Returns an array of all objects of a given class</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#reverse--">reverse()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Returns an amino acid pattern which is a reversed version of the current
 pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#reverse--">reverse()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Returns an amino acid sequence which is a reversed version of the current
 pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#reverse-boolean-">reverse(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Returns a new tag instance which is a reversed version of the current
 tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html#reverse-boolean-">reverse(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">TagAssumption</a></dt>
<dd>
<div class="block">Retunrs a new TagAssumption instance where the tag is a reversed version
 of this tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#right">right</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Right index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#rmdir-java.lang.String-">rmdir(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Delete the specified directory from the ftp server file system.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/ROC.html" title="interface in com.compomics.util.math.statistics"><span class="typeNameLink">ROC</span></a> - Interface in <a href="com/compomics/util/math/statistics/package-summary.html">com.compomics.util.math.statistics</a></dt>
<dd>
<div class="block">This class can be used to draw ROC curves.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#roundDouble-double-int-">roundDouble(double, int)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Rounds a double value to the wanted number of decimal places.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html#rSquared">rSquared</a></span> - Variable in class com.compomics.util.math.statistics.linear_regression.<a href="com/compomics/util/math/statistics/linear_regression/RegressionStatistics.html" title="class in com.compomics.util.math.statistics.linear_regression">RegressionStatistics</a></dt>
<dd>
<div class="block">The coefficient of determination.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#rt">rt</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>
<div class="block">The retention time when the peak was recorded.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#run--">run()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>
<div class="block">When an object implementing interface <code>Runnable</code> is used
 to create a thread, starting the thread causes the object's
 <code>run</code> method to be called in that separately executing
 thread.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/StreamGobbler.html#run--">run()</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/StreamGobbler.html" title="class in com.compomics.util.io">StreamGobbler</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="I:S">
<!--   -->
</a>
<h2 class="title">S</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#S">S</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#S">S</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The sulfur atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#sameLists-java.util.ArrayList-java.util.ArrayList-">sameLists(ArrayList, ArrayList)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Convenience methods indicating whether the content of two lists have the
 same content.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#sameListsFirstListSorted-java.util.ArrayList-java.util.ArrayList-">sameListsFirstListSorted(ArrayList, ArrayList)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Convenience methods indicating whether the content of two lists have the
 same content.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#sameModificationsAs-com.compomics.util.experiment.biology.proteins.Peptide-java.util.ArrayList-">sameModificationsAs(Peptide, ArrayList&lt;String&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Indicates whether another peptide has the same variable modifications at
 the same localization as this peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#sameModificationsAs-com.compomics.util.experiment.biology.proteins.Peptide-">sameModificationsAs(Peptide)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Indicates whether another peptide has the same modifications at the same
 localization as this peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#sameSets-java.util.HashSet-java.util.HashSet-">sameSets(HashSet, HashSet)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Convenience methods indicating whether the content of two sets have the
 same content.</div>
</dd>
<dt><a href="com/compomics/util/pride/prideobjects/Sample.html" title="class in com.compomics.util.pride.prideobjects"><span class="typeNameLink">Sample</span></a> - Class in <a href="com/compomics/util/pride/prideobjects/package-summary.html">com.compomics.util.pride.prideobjects</a></dt>
<dd>
<div class="block">An object for storing Sample details.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Sample.html#Sample-java.lang.String-java.util.ArrayList-">Sample(String, ArrayList&lt;CvTerm&gt;)</a></span> - Constructor for class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Sample.html" title="class in com.compomics.util.pride.prideobjects">Sample</a></dt>
<dd>
<div class="block">Create a new Sample object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html#sampleBatchSize">sampleBatchSize</a></span> - Static variable in class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SequenceDbDetailsDialog</a></dt>
<dd>
<div class="block">The batch size of proteins to sample.</div>
</dd>
<dt><a href="com/compomics/util/gui/SampleSelection.html" title="class in com.compomics.util.gui"><span class="typeNameLink">SampleSelection</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">Class which allows the selection of the right identification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/SampleSelection.html#SampleSelection-java.awt.Frame-boolean-java.lang.String:A-java.lang.String-">SampleSelection(Frame, boolean, String[], String)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/SampleSelection.html" title="class in com.compomics.util.gui">SampleSelection</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/interfaces/DBElement.html#save-java.sql.Connection-">save(Connection)</a></span> - Method in interface com.compomics.util.db.interfaces.<a href="com/compomics/util/db/interfaces/DBElement.html" title="interface in com.compomics.util.db.interfaces">DBElement</a></dt>
<dd>
<div class="block">This method can be called when changes need to be persisted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#saveCache-com.compomics.util.waiting.WaitingHandler-boolean-">saveCache(WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Saves the cache content in the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#saveFactory--">saveFactory()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Saves the factory in the user folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html#saveFile-java.io.File-">saveFile(File)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterMethodFactory.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterMethodFactory</a></dt>
<dd>
<div class="block">Save to file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#saveIdentificationParameters-com.compomics.util.parameters.identification.IdentificationParameters-java.io.File-">saveIdentificationParameters(IdentificationParameters, File)</a></span> - Static method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Saves the identification parameters to a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#saveIdentificationParameters-com.compomics.util.parameters.identification.search.SearchParameters-java.io.File-">saveIdentificationParameters(SearchParameters, File)</a></span> - Static method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Saves the identification parameters to a serialized file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#saveIdentificationParametersAsTextFile-java.io.File-">saveIdentificationParametersAsTextFile(File)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Saves the identification parameters as a human readable text file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html#saveMapping-java.io.File-">saveMapping(File)</a></span> - Method in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html" title="class in com.compomics.util.experiment.biology.genes.go">GoDomains</a></dt>
<dd>
<div class="block">Saves the mapping to the given file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#saveObjects-int-">saveObjects(int)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Saves an entry in the database if modified and clears it from the cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#saveObjects-int-com.compomics.util.waiting.WaitingHandler-">saveObjects(int, WaitingHandler)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Saves an entry in the database if modified and clears it from the cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#saveObjects-int-com.compomics.util.waiting.WaitingHandler-boolean-">saveObjects(int, WaitingHandler, boolean)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Saves an entry in the database if modified.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#saveObjectToJson-java.lang.Object-java.io.File-">saveObjectToJson(Object, File)</a></span> - Method in class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">Save an object to JSON.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#saveToFile-com.compomics.util.experiment.biology.enzymes.EnzymeFactory-java.io.File-">saveToFile(EnzymeFactory, File)</a></span> - Static method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Saves en enzyme factory to a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#saveToFile-com.compomics.util.experiment.biology.modifications.ModificationFactory-java.io.File-">saveToFile(ModificationFactory, File)</a></span> - Static method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Saves a Modification factory to a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#saveUrl-java.io.File-java.lang.String-int-java.lang.String-java.lang.String-com.compomics.util.waiting.WaitingHandler-">saveUrl(File, String, int, String, String, WaitingHandler)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Save a file from a URL.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#saveUserParameters-com.compomics.util.parameters.UtilitiesUserParameters-">saveUserParameters(UtilitiesUserParameters)</a></span> - Static method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Convenience method saving the user parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#scaffold">scaffold</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Scaffold multiple search engine data interpretation software.</div>
</dd>
<dt><a href="com/compomics/util/experiment/quantification/spectrumcounting/ScalingFactorsEstimators.html" title="class in com.compomics.util.experiment.quantification.spectrumcounting"><span class="typeNameLink">ScalingFactorsEstimators</span></a> - Class in <a href="com/compomics/util/experiment/quantification/spectrumcounting/package-summary.html">com.compomics.util.experiment.quantification.spectrumcounting</a></dt>
<dd>
<div class="block">This class estimates spectrum counting scaling factors.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/spectrumcounting/ScalingFactorsEstimators.html#ScalingFactorsEstimators-com.compomics.util.parameters.quantification.spectrum_counting.SpectrumCountingParameters-">ScalingFactorsEstimators(SpectrumCountingParameters)</a></span> - Constructor for class com.compomics.util.experiment.quantification.spectrumcounting.<a href="com/compomics/util/experiment/quantification/spectrumcounting/ScalingFactorsEstimators.html" title="class in com.compomics.util.experiment.quantification.spectrumcounting">ScalingFactorsEstimators</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#scanNumber">scanNumber</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Scan number or range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#scanStartTime">scanStartTime</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">The time point when the spectrum was recorded (scan start time in mzML
 files).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#score">score</a></span> - Variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">The (advocate specific) score used to rank this assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ModificationProfile.html#SCORE_1_ROW_INDEX">SCORE_1_ROW_INDEX</a></span> - Static variable in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ModificationProfile.html" title="class in com.compomics.util.gui.protein">ModificationProfile</a></dt>
<dd>
<div class="block">The score above the sequence row index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ModificationProfile.html#SCORE_2_ROW_INDEX">SCORE_2_ROW_INDEX</a></span> - Static variable in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ModificationProfile.html" title="class in com.compomics.util.gui.protein">ModificationProfile</a></dt>
<dd>
<div class="block">The score under the sequence row index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#Se">Se</a></span> - Static variable in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>
<div class="block">The selenium atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#searchDecoyDatabase--">searchDecoyDatabase()</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Returns whether MS-GF+ is to create and search a decoy database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#searchGuiParametersFileOption">searchGuiParametersFileOption</a></span> - Static variable in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">The command line argument for a parameters file for SearchGUI.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#searchGuiRawFileOption">searchGuiRawFileOption</a></span> - Static variable in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">The command line argument for raw files for SearchGUI.</div>
</dd>
<dt><a href="com/compomics/software/dialogs/SearchGuiSetupDialog.html" title="class in com.compomics.software.dialogs"><span class="typeNameLink">SearchGuiSetupDialog</span></a> - Class in <a href="com/compomics/software/dialogs/package-summary.html">com.compomics.software.dialogs</a></dt>
<dd>
<div class="block">A dialog used to set up the connection between PeptideShaker and SearchGUI.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/SearchGuiSetupDialog.html#SearchGuiSetupDialog-javax.swing.JFrame-boolean-">SearchGuiSetupDialog(JFrame, boolean)</a></span> - Constructor for class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/SearchGuiSetupDialog.html" title="class in com.compomics.software.dialogs">SearchGuiSetupDialog</a></dt>
<dd>
<div class="block">Creates a new SearchGuiSetupDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/dialogs/SearchGuiSetupDialog.html#SearchGuiSetupDialog-javax.swing.JDialog-boolean-">SearchGuiSetupDialog(JDialog, boolean)</a></span> - Constructor for class com.compomics.software.dialogs.<a href="com/compomics/software/dialogs/SearchGuiSetupDialog.html" title="class in com.compomics.software.dialogs">SearchGuiSetupDialog</a></dt>
<dd>
<div class="block">Creates a new SearchGuiSetupDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#searchGuiSpectrumFileOption">searchGuiSpectrumFileOption</a></span> - Static variable in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">The command line argument for mgf files for SearchGUI.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search"><span class="typeNameLink">SearchParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/search/package-summary.html">com.compomics.util.parameters.identification.search</a></dt>
<dd>
<div class="block">This class groups the parameters used for identification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#SearchParameters--">SearchParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#SearchParameters-com.compomics.util.parameters.identification.search.SearchParameters-">SearchParameters(SearchParameters)</a></span> - Constructor for class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/search/SearchParameters.MassAccuracyType.html" title="enum in com.compomics.util.parameters.identification.search"><span class="typeNameLink">SearchParameters.MassAccuracyType</span></a> - Enum in <a href="com/compomics/util/parameters/identification/search/package-summary.html">com.compomics.util.parameters.identification.search</a></dt>
<dd>
<div class="block">Possible mass accuracy types.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.search"><span class="typeNameLink">SearchParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/search/package-summary.html">com.compomics.util.gui.parameters.identification.search</a></dt>
<dd>
<div class="block">The search settings dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html#SearchParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.search.SearchParameters-java.awt.Image-java.awt.Image-boolean-boolean-com.compomics.util.io.file.LastSelectedFolder-java.lang.String-boolean-">SearchParametersDialog(Frame, SearchParameters, Image, Image, boolean, boolean, LastSelectedFolder, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SearchParametersDialog</a></dt>
<dd>
<div class="block">Creates a new SearchSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html#SearchParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.search.SearchParameters-java.awt.Image-java.awt.Image-boolean-boolean-com.compomics.util.io.file.LastSelectedFolder-java.lang.String-boolean-">SearchParametersDialog(Dialog, Frame, SearchParameters, Image, Image, boolean, boolean, LastSelectedFolder, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SearchParametersDialog</a></dt>
<dd>
<div class="block">Creates a new SearchSettingsDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/genes/SpeciesDialog.html#SELECT_SPECIES_TAG">SELECT_SPECIES_TAG</a></span> - Static variable in class com.compomics.util.gui.genes.<a href="com/compomics/util/gui/genes/SpeciesDialog.html" title="class in com.compomics.util.gui.genes">SpeciesDialog</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">The text to use to tell the user to please select a species in the list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html#selectDB-boolean-">selectDB(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SequenceDbDetailsDialog</a></dt>
<dd>
<div class="block">Allows the user to select a fasta file, loads its information, and
 returns a boolean indicating whether the process loading was successful.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.SelectedValuesTableFilter.html#SelectedValuesTableFilter--">SelectedValuesTableFilter()</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.SelectedValuesTableFilter.html" title="class in com.compomics.util.gui">XYPlottingDialog.SelectedValuesTableFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/Selenium.html" title="class in com.compomics.util.experiment.biology.atoms.impl"><span class="typeNameLink">Selenium</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a></dt>
<dd>
<div class="block">Selenium.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/impl/Selenium.html#Selenium--">Selenium()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.impl.<a href="com/compomics/util/experiment/biology/atoms/impl/Selenium.html" title="class in com.compomics.util.experiment.biology.atoms.impl">Selenium</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Selenocysteine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">SeC (U) (Mascot)-</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html#Selenocysteine--">Selenocysteine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Selenocysteine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Selenocysteine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#SELENOCYSTEINE">SELENOCYSTEINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Selenocysteine immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/MassCalc.html#SELFDEFINED">SELFDEFINED</a></span> - Static variable in class com.compomics.util.general.<a href="com/compomics/util/general/MassCalc.html" title="class in com.compomics.util.general">MassCalc</a></dt>
<dd>
<div class="block">Value for the self-defined masses.</div>
</dd>
<dt><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels"><span class="typeNameLink">SelfUpdatingTableModel</span></a> - Class in <a href="com/compomics/util/gui/tablemodels/package-summary.html">com.compomics.util.gui.tablemodels</a></dt>
<dd>
<div class="block">These table models include a self updating function.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#SelfUpdatingTableModel--">SelfUpdatingTableModel()</a></span> - Constructor for class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTPClient.html#sendBinaryFile-java.lang.String-">sendBinaryFile(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTPClient.html" title="class in com.compomics.util.io.ftp">FTPClient</a></dt>
<dd>
<div class="block">This method sends a binary file to the default FTP location on the server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTPClient.html#sendFiles-java.lang.String:A-boolean:A-">sendFiles(String[], boolean[])</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTPClient.html" title="class in com.compomics.util.io.ftp">FTPClient</a></dt>
<dd>
<div class="block">This method sends a group of files to the default FTP location on the server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTPClient.html#sendFiles-java.lang.String:A-boolean-">sendFiles(String[], boolean)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTPClient.html" title="class in com.compomics.util.io.ftp">FTPClient</a></dt>
<dd>
<div class="block">This method sends a group of files to the default FTP location on the server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#sendGAUpdate-java.lang.String-java.lang.String-java.lang.String-">sendGAUpdate(String, String, String)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Function for sending event requests to Google analytics.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/PickUp.html#sendIncoming-java.lang.Object-">sendIncoming(Object)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/PickUp.html" title="interface in com.compomics.util.interfaces">PickUp</a></dt>
<dd>
<div class="block">This method should be called by the notifier when appropriate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#sendServer-java.lang.String-">sendServer(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Sends command <i>cmd</i> to the server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTPClient.html#sendTextFile-java.lang.String-">sendTextFile(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTPClient.html" title="class in com.compomics.util.io.ftp">FTPClient</a></dt>
<dd>
<div class="block">This method sends a text file to the default FTP location on the server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandLineUtils.html#SEPARATOR">SEPARATOR</a></span> - Static variable in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli">CommandLineUtils</a></dt>
<dd>
<div class="block">The command line argument separator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#separator">separator</a></span> - Static variable in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">The separator between a path ID and a path.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html#SEPARATOR">SEPARATOR</a></span> - Static variable in class com.compomics.util.experiment.biology.genes.ensembl.<a href="com/compomics/util/experiment/biology/genes/ensembl/GeneMapping.html" title="class in com.compomics.util.experiment.biology.genes.ensembl">GeneMapping</a></dt>
<dd>
<div class="block">The separator used to separate line contents.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html#SEPARATOR">SEPARATOR</a></span> - Static variable in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoDomains.html" title="class in com.compomics.util.experiment.biology.genes.go">GoDomains</a></dt>
<dd>
<div class="block">The separator used to separate line contents.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html#SEPARATOR">SEPARATOR</a></span> - Static variable in class com.compomics.util.experiment.biology.genes.go.<a href="com/compomics/util/experiment/biology/genes/go/GoMapping.html" title="class in com.compomics.util.experiment.biology.genes.go">GoMapping</a></dt>
<dd>
<div class="block">The separator used to separate line contents.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#SEPARATOR">SEPARATOR</a></span> - Static variable in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">The separator used to separate line contents.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/BiomartMapping.html#SEPARATOR">SEPARATOR</a></span> - Static variable in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/BiomartMapping.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">BiomartMapping</a></dt>
<dd>
<div class="block">The separator used to separate line contents.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html#SEPARATOR">SEPARATOR</a></span> - Static variable in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblGenomesSpecies</a></dt>
<dd>
<div class="block">The separator used to separate line contents.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html#SEPARATOR">SEPARATOR</a></span> - Static variable in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblSpecies.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblSpecies</a></dt>
<dd>
<div class="block">The separator used to separate line contents.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html#SEPARATOR">SEPARATOR</a></span> - Static variable in class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">UniprotTaxonomy</a></dt>
<dd>
<div class="block">The separator used to separate line contents.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#separator">separator</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">The separator used to separate locations in the modification location
 key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#separator">separator</a></span> - Static variable in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>
<div class="block">The columns separator.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#separator">separator</a></span> - Static variable in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Separator which can be used to separate things.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#separators">separators</a></span> - Static variable in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">The decoy separators used to infer the FASTA parameters.</div>
</dd>
<dt><a href="com/compomics/util/interfaces/Sequence.html" title="interface in com.compomics.util.interfaces"><span class="typeNameLink">Sequence</span></a> - Interface in <a href="com/compomics/util/interfaces/package-summary.html">com.compomics.util.interfaces</a></dt>
<dd>
<div class="block">This interface defines the default behavior of any kind of 
 sequence with a certain mass - be it DNA, protein or something 
 else.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#sequenceCoverageInCache-long-">sequenceCoverageInCache(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Indicates whether the sequence coverage is in cache.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html" title="class in com.compomics.util.gui.parameters.identification.search"><span class="typeNameLink">SequenceDbDetailsDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/search/package-summary.html">com.compomics.util.gui.parameters.identification.search</a></dt>
<dd>
<div class="block">This dialog displays information about a sequence database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html#SequenceDbDetailsDialog-java.awt.Dialog-java.awt.Frame-java.io.File-com.compomics.util.experiment.io.biology.protein.FastaParameters-com.compomics.util.io.file.LastSelectedFolder-boolean-java.awt.Image-java.awt.Image-">SequenceDbDetailsDialog(Dialog, Frame, File, FastaParameters, LastSelectedFolder, boolean, Image, Image)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SequenceDbDetailsDialog</a></dt>
<dd>
<div class="block">Creates a new SequenceDbDetailsDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html#SequenceDbDetailsDialog-java.awt.Frame-java.io.File-com.compomics.util.experiment.io.biology.protein.FastaParameters-com.compomics.util.io.file.LastSelectedFolder-boolean-java.awt.Image-java.awt.Image-">SequenceDbDetailsDialog(Frame, File, FastaParameters, LastSelectedFolder, boolean, Image, Image)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SequenceDbDetailsDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SequenceDbDetailsDialog</a></dt>
<dd>
<div class="block">Creates a new SequenceDbDetailsDialog.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/SequenceFragmentationPanel.html" title="class in com.compomics.util.gui.spectrum"><span class="typeNameLink">SequenceFragmentationPanel</span></a> - Class in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">This class was imported from the Peptizer and MascotDatfile parser, and was
 developed to display fragmentation information on the modified sequence as
 inspired by X!Tandem.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SequenceFragmentationPanel.html#SequenceFragmentationPanel-java.lang.String-com.compomics.util.experiment.identification.matches.IonMatch:A-boolean-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-int-int-">SequenceFragmentationPanel(String, IonMatch[], boolean, boolean, ModificationParameters, int, int)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SequenceFragmentationPanel.html" title="class in com.compomics.util.gui.spectrum">SequenceFragmentationPanel</a></dt>
<dd>
<div class="block">Creates a new SequenceFragmentationPanel working with B and Y ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SequenceFragmentationPanel.html#SequenceFragmentationPanel-java.lang.String-com.compomics.util.experiment.identification.matches.IonMatch:A-boolean-com.compomics.util.parameters.identification.search.ModificationParameters-int-int-">SequenceFragmentationPanel(String, IonMatch[], boolean, ModificationParameters, int, int)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SequenceFragmentationPanel.html" title="class in com.compomics.util.gui.spectrum">SequenceFragmentationPanel</a></dt>
<dd>
<div class="block">Creates a new SequenceFragmentationPanel working with B and Y ions.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/SequenceIterator.html" title="interface in com.compomics.util.experiment.identification.protein_sequences.digestion"><span class="typeNameLink">SequenceIterator</span></a> - Interface in <a href="com/compomics/util/experiment/identification/protein_sequences/digestion/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion</a></dt>
<dd>
<div class="block">Interface for an iterator returning peptides along a protein sequence.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced"><span class="typeNameLink">SequenceMatchingParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/advanced/package-summary.html">com.compomics.util.parameters.identification.advanced</a></dt>
<dd>
<div class="block">The sequence matching options.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html#SequenceMatchingParameters--">SequenceMatchingParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a></dt>
<dd>
<div class="block">Constructor for empty preferences.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html" title="enum in com.compomics.util.parameters.identification.advanced"><span class="typeNameLink">SequenceMatchingParameters.MatchingType</span></a> - Enum in <a href="com/compomics/util/parameters/identification/advanced/package-summary.html">com.compomics.util.parameters.identification.advanced</a></dt>
<dd>
<div class="block">The different types of amino acid matching.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/advanced/SequenceMatchingParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced"><span class="typeNameLink">SequenceMatchingParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/advanced/package-summary.html">com.compomics.util.gui.parameters.identification.advanced</a></dt>
<dd>
<div class="block">Dialog for the edition of the sequence matching settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/SequenceMatchingParametersDialog.html#SequenceMatchingParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">SequenceMatchingParametersDialog(Frame, SequenceMatchingParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/SequenceMatchingParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">SequenceMatchingParametersDialog</a></dt>
<dd>
<div class="block">Creates a new SequenceMatchingSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/SequenceMatchingParametersDialog.html#SequenceMatchingParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-boolean-">SequenceMatchingParametersDialog(Dialog, Frame, SequenceMatchingParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/SequenceMatchingParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">SequenceMatchingParametersDialog</a></dt>
<dd>
<div class="block">Creates a new SequenceMatchingSettingsDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/gui/protein/SequenceModificationPanel.html" title="class in com.compomics.util.gui.protein"><span class="typeNameLink">SequenceModificationPanel</span></a> - Class in <a href="com/compomics/util/gui/protein/package-summary.html">com.compomics.util.gui.protein</a></dt>
<dd>
<div class="block">A panel for displaying modification profiles.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/SequenceModificationPanel.html#SequenceModificationPanel-java.lang.String-java.util.ArrayList-boolean-java.lang.String-java.lang.String-">SequenceModificationPanel(String, ArrayList&lt;ModificationProfile&gt;, boolean, String, String)</a></span> - Constructor for class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/SequenceModificationPanel.html" title="class in com.compomics.util.gui.protein">SequenceModificationPanel</a></dt>
<dd>
<div class="block">Creates a new SequenceFragmentationPanel.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/biology/protein/SequenceProvider.html" title="interface in com.compomics.util.experiment.io.biology.protein"><span class="typeNameLink">SequenceProvider</span></a> - Interface in <a href="com/compomics/util/experiment/io/biology/protein/package-summary.html">com.compomics.util.experiment.io.biology.protein</a></dt>
<dd>
<div class="block">Interface for a class able to retrieve the sequence of a given protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#sequest">sequest</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Sequest search engine.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/SequestFragmentationModel.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models"><span class="typeNameLink">SequestFragmentationModel</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models</a></dt>
<dd>
<div class="block">Fragmentation model originally described in the Sequest algorithm
 (https://www.ncbi.nlm.nih.gov/pubmed/24226387) adapted to utilities objects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/SequestFragmentationModel.html#SequestFragmentationModel--">SequestFragmentationModel()</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/SequestFragmentationModel.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models">SequestFragmentationModel</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/SequestParamsFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">SequestParamsFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for sequest.params files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/SequestParamsFileFilter.html#SequestParamsFileFilter--">SequestParamsFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/SequestParamsFileFilter.html" title="class in com.compomics.util.io.file.filefilters">SequestParamsFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/SerializationUtils.html" title="class in com.compomics.util.io.file"><span class="typeNameLink">SerializationUtils</span></a> - Class in <a href="com/compomics/util/io/file/package-summary.html">com.compomics.util.io.file</a></dt>
<dd>
<div class="block">This class implements convenience methods for serialization and
 deserialization.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/SerializationUtils.html#SerializationUtils--">SerializationUtils()</a></span> - Constructor for class com.compomics.util.io.file.<a href="com/compomics/util/io/file/SerializationUtils.html" title="class in com.compomics.util.io.file">SerializationUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/adapter/InterfaceAdapter.html#serialize-T-java.lang.reflect.Type-com.google.gson.JsonSerializationContext-">serialize(T, Type, JsonSerializationContext)</a></span> - Method in class com.compomics.util.io.json.adapter.<a href="com/compomics/util/io/json/adapter/InterfaceAdapter.html" title="class in com.compomics.util.io.json.adapter">InterfaceAdapter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFeature.html#serialVersionUID">serialVersionUID</a></span> - Static variable in interface com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFeature.html" title="interface in com.compomics.util.io.export">ExportFeature</a></dt>
<dd>
<div class="block">Serial number for backward compatibility.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Serine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Serine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html#Serine--">Serine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Serine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Serine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#SERINE">SERINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Serine immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#serverInput">serverInput</a></span> - Variable in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Buffered stream for reading replies from server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#serverIsOpen--">serverIsOpen()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Return server connection status</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#serverOutput">serverOutput</a></span> - Variable in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Stream for printing to the server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#serverResponse">serverResponse</a></span> - Variable in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Array of strings (usually 1 entry) for the last reply from the server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#setAaAtIndex-int-char-">setAaAtIndex(int, char)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">replaces the amino acid at the given position by the given amino acid
 represented by its single letter code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html#setAaIndex-int-">setAaIndex(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAIdentityFeature</a></dt>
<dd>
<div class="block">Sets the index on the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html#setAaIndex-int-">setAaIndex(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAPropertyFeature</a></dt>
<dd>
<div class="block">Sets the index on the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html#setAaSubstitutionMatrix-com.compomics.util.experiment.biology.variants.AaSubstitutionMatrix-">setAaSubstitutionMatrix(AaSubstitutionMatrix)</a></span> - Method in class com.compomics.util.gui.variants.aa_substitutions.<a href="com/compomics/util/gui/variants/aa_substitutions/AaSubstitutionMatrixTableModel.html" title="class in com.compomics.util.gui.variants.aa_substitutions">AaSubstitutionMatrixTableModel</a></dt>
<dd>
<div class="block">Sets the substitution matrix to display.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#setAaSubstitutionMatrix-com.compomics.util.experiment.biology.variants.AaSubstitutionMatrix-">setAaSubstitutionMatrix(AaSubstitutionMatrix)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Sets the amino acid substitution matrix to use.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setAboveXAxis-boolean-">setAboveXAxis(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Set if the reference area is to be added above the x-axis, false adds the
 reference are below the x-axis</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setAccession-java.lang.String-">setAccession(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/CvTerm.html#setAccession-java.lang.String-">setAccession(String)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/CvTerm.html" title="class in com.compomics.util.pride">CvTerm</a></dt>
<dd>
<div class="block">Sets the accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#setAccessions-java.lang.String:A-">setAccessions(String[])</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Sets the accessions of the proteins in this group.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#setActicationType-java.lang.String-">setActicationType(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Sets the activation type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#setActive-boolean-">setActive(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Sets whether the filter is active.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#setAdditionalOutput-boolean-">setAdditionalOutput(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Set if additional output is to be included.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setAdjustPrecursorMass-boolean-">setAdjustPrecursorMass(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Set if the precursor mass is to be adjusted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#setAdvocate-int-">setAdvocate(int)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setAlgorithmDeltaPEP-java.lang.Double-">setAlgorithmDeltaPEP(Double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Sets the difference in identification algorithm level PEP with the next
 best peptide assumption with sequence difference for the given search
 engine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#setAlignNonConfidentModifications-boolean-">setAlignNonConfidentModifications(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Sets whether the non confidently localized modifications should be aligned on the
 confident sites.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#setAllPathsIn-java.lang.String-">setAllPathsIn(String)</a></span> - Static method in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Sets all the paths inside a given folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setAlpha-float-">setAlpha(float)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Set the alpha level (transparency).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setAlphaLevel-float-">setAlphaLevel(float)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Sets the alpha level</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html#setAminoAcid-char-">setAminoAcid(char)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAIdentityFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAIdentityFeature</a></dt>
<dd>
<div class="block">Sets the amino acid targeted represented as single letter code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html#setAminoAcidFeature1-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.AAPropertyFeature-">setAminoAcidFeature1(AAPropertyFeature)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature</a></dt>
<dd>
<div class="block">Sets the first amino acid feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html#setAminoAcidFeature2-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.AAPropertyFeature-">setAminoAcidFeature2(AAPropertyFeature)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature</a></dt>
<dd>
<div class="block">Sets the second amino acid feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html#setAminoAcidProperty-com.compomics.util.experiment.biology.aminoacids.AminoAcid.Property-">setAminoAcidProperty(AminoAcid.Property)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AAPropertyFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AAPropertyFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html#setAminoAcidProperty-com.compomics.util.experiment.biology.aminoacids.AminoAcid.Property-">setAminoAcidProperty(AminoAcid.Property)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AASequenceFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/SingleAAPropertyFeature.html#setAminoAcidProperty-com.compomics.util.experiment.biology.aminoacids.AminoAcid.Property-">setAminoAcidProperty(AminoAcid.Property)</a></span> - Method in interface com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/SingleAAPropertyFeature.html" title="interface in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">SingleAAPropertyFeature</a></dt>
<dd>
<div class="block">Sets the amino acid property to consider.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#setAminoAcidScores-java.util.ArrayList-">setAminoAcidScores(ArrayList&lt;double[]&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Set the amino acid scores.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setAnnotateHighestPeak-boolean-">setAnnotateHighestPeak(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set if the most intense of possible peaks to annotate is to be selected,
 false if the most accurate is to be selected.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setAnnotationParameters-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-">setAnnotationParameters(AnnotationParameters)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the annotation parameters used for identification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setAnnotations-java.util.List-boolean-">setAnnotations(List&lt;SpectrumAnnotation&gt;, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method sets all the annotations on this instance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setAnnotations-java.util.List-">setAnnotations(List&lt;SpectrumAnnotation&gt;)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method sets all the annotations on this instance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setAnnotationsMirrored-java.util.List-boolean-">setAnnotationsMirrored(List&lt;SpectrumAnnotation&gt;, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method sets all the annotations for the mirrored spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setAnnotationsMirrored-java.util.List-">setAnnotationsMirrored(List&lt;SpectrumAnnotation&gt;)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method sets all the annotations for the mirrored spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setAnnotationYAxisThreshold-double-">setAnnotationYAxisThreshold(double)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method allows the caller to set the minimal non-inclusive y-axis
 value threshold in percent (compared to the highest point in the spectrum
 or chromatogram) a point must pass before being eligible for annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setAreaColor-java.awt.Color-">setAreaColor(Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Set the area color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setAreaUnderCurveColor-java.awt.Color-int-">setAreaUnderCurveColor(Color, int)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Sets the color of the area under the curve for profile chromatograms and
 spectra for the dataset with the given dataset index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#setAtomChainAdded-com.compomics.util.experiment.biology.atoms.AtomChain-">setAtomChainAdded(AtomChain)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Sets the atom chain added.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#setAtomChainRemoved-com.compomics.util.experiment.biology.atoms.AtomChain-">setAtomChainRemoved(AtomChain)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Sets the atom chain removed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#setAtomicComposition-com.compomics.util.experiment.biology.atoms.AtomChain-">setAtomicComposition(AtomChain)</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Returns the atomic composition.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html#setAtomSymbol-java.lang.String-">setAtomSymbol(String)</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html" title="class in com.compomics.util.experiment.biology.atoms">AtomImpl</a></dt>
<dd>
<div class="block">Sets the atom symbol as specified in the Atom class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setAutomaticAnnotation-boolean-">setAutomaticAnnotation(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Sets whether the annotation settings should be automatically inferred.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html#setAutoUpdate-java.lang.Boolean-">setAutoUpdate(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html" title="class in com.compomics.util.parameters.identification.advanced">GeneParameters</a></dt>
<dd>
<div class="block">Sets whether the gene mappings should be automatically updated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setAutoUpdate-java.lang.Boolean-">setAutoUpdate(Boolean)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Sets whether the tools should use the auto update function.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setBackgroundPeakWidth-float-">setBackgroundPeakWidth(float)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set the backgroundPeakWidth peak width.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setBatchSize-java.lang.Integer-">setBatchSize(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the batch size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#setBestPeptideAssumption-com.compomics.util.experiment.identification.spectrum_assumptions.PeptideAssumption-">setBestPeptideAssumption(PeptideAssumption)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Setter for the best peptide assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#setBestTagAssumption-com.compomics.util.experiment.identification.spectrum_assumptions.TagAssumption-">setBestTagAssumption(TagAssumption)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Setter for the best tag assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#setBinMax-java.lang.Integer-">setBinMax(Integer)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#setBinMin-java.lang.Integer-">setBinMin(Integer)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/objects/BlobObject.html#setBlob-byte:A-">setBlob(byte[])</a></span> - Method in class com.compomics.util.db.object.objects.<a href="com/compomics/util/db/object/objects/BlobObject.html" title="class in com.compomics.util.db.object.objects">BlobObject</a></dt>
<dd>
<div class="block">Sets the byte representation of the object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#setBlock-java.lang.String-">setBlock(String)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Sets the block.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html#setBlocks-com.compomics.util.pdbfinder.pdb.PdbBlock:A-">setBlocks(PdbBlock[])</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb">PdbParameter</a></dt>
<dd>
<div class="block">Set the blocks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setBoldFont-boolean-">setBoldFont(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Set if the label is to be in bold.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setBorderColor-java.awt.Color-">setBorderColor(Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Set the border color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setBorderWidth-float-">setBorderWidth(float)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Set the border width.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#setCacheSize-int-">setCacheSize(int)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Sets the spectrum cache size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#setCharge-int-">setCharge(int)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Setter for the charge</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#setCharge-int-">setCharge(int)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method sets the charge of the precursor ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setCheckDuplicateTitles-boolean-">setCheckDuplicateTitles(boolean)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set if the spectra should be checked for duplicate titles or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setCheckMgfSize-boolean-">setCheckMgfSize(boolean)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set if the mgf should be checked for size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setCheckPeakPicking-boolean-">setCheckPeakPicking(boolean)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set if the spectra should be checked for peak picking or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setCheckSpectrumCharges-boolean-">setCheckSpectrumCharges(boolean)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set if the spectra are to be checked for missing charges.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setClassSizeMultiplier-java.lang.Integer-">setClassSizeMultiplier(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Set the intensity class size multiplier.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#setCleavage-char:A-">setCleavage(char[])</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">This method allows the caller to specify the cleavable residues.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#setCleavage-java.lang.String-">setCleavage(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">This method allows the caller to specify the cleavable residues.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#setCleavage-char:A-int-">setCleavage(char[], int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">This method allows the caller to specify the cleavable residues.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#setCleavage-java.lang.String-int-">setCleavage(String, int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">This method allows the caller to specify the cleavable residus.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#setCleavage-char:A-">setCleavage(char[])</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method allows the caller to specify the cleavable residues.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#setCleavage-java.lang.String-">setCleavage(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method allows the caller to specify the cleavable residues.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/RegExEnzyme.html#setCleavage-java.lang.String-">setCleavage(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/RegExEnzyme.html" title="class in com.compomics.util.protein">RegExEnzyme</a></dt>
<dd>
<div class="block">This method can be used to set the cleavage pattern for this RegExEnzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/RegExEnzyme.html#setCleavage-char:A-">setCleavage(char[])</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/RegExEnzyme.html" title="class in com.compomics.util.protein">RegExEnzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#setCleavageParameter-com.compomics.util.parameters.identification.search.DigestionParameters.CleavageParameter-">setCleavageParameter(DigestionParameters.CleavageParameter)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Sets the cleavage parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setCleaveNterMethionine-boolean-">setCleaveNterMethionine(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets whether N-terminal methionines should be cleaved.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setClipNtermMethionine-java.lang.Boolean-">setClipNtermMethionine(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the starting methionine peptides will be included both with and
 without the starting M.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setCloseDialogWhenImportCompletes-boolean-boolean-">setCloseDialogWhenImportCompletes(boolean, boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Set if the value for if the dialog is to be closed when completed and if
 the option is enabled or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#setColor-java.lang.String-java.awt.Color-">setColor(String, Color)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Sets a new color for the given expected modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ModificationProfile.html#setColor-java.awt.Color-">setColor(Color)</a></span> - Method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ModificationProfile.html" title="class in com.compomics.util.gui.protein">ModificationProfile</a></dt>
<dd>
<div class="block">Sets the PTM color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/ModificationParameters.html#setColor-java.lang.String-java.awt.Color-">setColor(String, Color)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/ModificationParameters.html" title="class in com.compomics.util.parameters.identification.search">ModificationParameters</a></dt>
<dd>
<div class="block">Sets a new color for the given expected modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#setColumn-int-java.util.ArrayList-">setColumn(int, ArrayList&lt;Double&gt;)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#setComparatorForItem-java.lang.String-com.compomics.util.experiment.filtering.FilterItemComparator-">setComparatorForItem(String, FilterItemComparator)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Sets the comparator for a given item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setComplementMzTolerance-double-">setComplementMzTolerance(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Sets the complement m/z tolerance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setComplementScoreWeight-double-">setComplementScoreWeight(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Set the complement score weight.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#setComposition-com.compomics.util.experiment.biology.atoms.AtomChain-">setComposition(AtomChain)</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Sets the composition of the neutral loss.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setComputeExactPValues-java.lang.Boolean-">setComputeExactPValues(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the exact p-values are to be computed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setComputeSpScore-java.lang.Boolean-">setComputeSpScore(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the SP score is to be computed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setComputeXCorr-boolean-">setComputeXCorr(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Sets if a Sequest-like cross correlation score will be calculated for the
 top ranking hits in each spectrums result set.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setConcatenatTargetDecoy-java.lang.Boolean-">setConcatenatTargetDecoy(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the target and decoy results are to be concatenated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#setCondition-java.lang.String-">setCondition(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Sets the description of the condition to meet.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#setConfidenceMargin-java.lang.Double-">setConfidenceMargin(Double)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Sets the margin to the threshold to use as factor of the resolution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html#setConfident-boolean-">setConfident(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches">ModificationMatch</a></dt>
<dd>
<div class="block">Sets whether the modification is confidently localized on the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setConsecutiveIonProbability-double-">setConsecutiveIonProbability(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Set the consecutive ion probability.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ContactGroup.html#setContacts-java.util.ArrayList-">setContacts(ArrayList&lt;Contact&gt;)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ContactGroup.html" title="class in com.compomics.util.pride.prideobjects">ContactGroup</a></dt>
<dd>
<div class="block">Set the contacts.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#setContent-java.util.ArrayList-">setContent(ArrayList&lt;TagComponent&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Sets the content for the given tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/StreamGobbler.html#setContinueReading-boolean-">setContinueReading(boolean)</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/StreamGobbler.html" title="class in com.compomics.util.io">StreamGobbler</a></dt>
<dd>
<div class="block">Set if the reading is to continue or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#setControlSamples-java.util.ArrayList-">setControlSamples(ArrayList&lt;String&gt;)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">Sets the indexes of the samples labeled as control.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#setCorrectPrecursorMass-java.lang.Boolean-">setCorrectPrecursorMass(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Sets whether the precursor mass shall be corrected (TagDB setting).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#setCreationTime-java.util.Date-">setCreationTime(Date)</a></span> - Method in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#setcTermModification-java.lang.String-">setcTermModification(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Sets the C-term modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#setCurrentPeptideKey-long-">setCurrentPeptideKey(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Sets the current peptide key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#setCurrentProteinKey-long-">setCurrentProteinKey(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Sets the current protein key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html#setCvTerm-com.compomics.util.pride.CvTerm-">setCvTerm(CvTerm)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/Enzyme.html" title="class in com.compomics.util.experiment.biology.enzymes">Enzyme</a></dt>
<dd>
<div class="block">Sets the CV term associated with this enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#setCvTerm-com.compomics.util.pride.CvTerm-">setCvTerm(CvTerm)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Set the CV term associated with this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Instrument.html#setCvTerms-java.util.ArrayList-">setCvTerms(ArrayList&lt;CvTerm&gt;)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Instrument.html" title="class in com.compomics.util.pride.prideobjects">Instrument</a></dt>
<dd>
<div class="block">Set the CV terms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Protocol.html#setCvTerms-java.util.ArrayList-">setCvTerms(ArrayList&lt;CvTerm&gt;)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Protocol.html" title="class in com.compomics.util.pride.prideobjects">Protocol</a></dt>
<dd>
<div class="block">Set the CV terms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Sample.html#setCvTerms-java.util.ArrayList-">setCvTerms(ArrayList&lt;CvTerm&gt;)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Sample.html" title="class in com.compomics.util.pride.prideobjects">Sample</a></dt>
<dd>
<div class="block">Set the CV terms.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setDatabaseType-com.compomics.util.experiment.io.biology.protein.ProteinDatabase-">setDatabaseType(ProteinDatabase)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the database type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#setDataFilters-java.util.HashMap-">setDataFilters(HashMap&lt;String, String&gt;)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>
<div class="block">Set the data filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setDataPointAndLineColor-java.awt.Color-int-">setDataPointAndLineColor(Color, int)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Sets the color of data points and line for the dataset with the given
 dataset index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JTableForDB.html#setDateTimeFormat-java.lang.String-">setDateTimeFormat(String)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/JTableForDB.html" title="class in com.compomics.util.gui">JTableForDB</a></dt>
<dd>
<div class="block">This method allows the setting of the date /time format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setDbFolder-java.io.File-">setDbFolder(File)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Sets the last used database folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#setDbSize-boolean-">setDbSize(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Sets whether the database size should be checked.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#setDebugInteractions-boolean-">setDebugInteractions(boolean)</a></span> - Static method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Turn the debugging of interactions on or off.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#setDecoy-boolean-">setDecoy(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Sets whether the given match is decoy</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#setDecoyFlag-java.lang.String-">setDecoyFlag(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Sets the decoy flag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setDecoyFormat-java.lang.String-">setDecoyFormat(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the decoy format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setDecoyMode-com.compomics.util.parameters.identification.tool_specific.AndromedaParameters.AndromedaDecoyMode-">setDecoyMode(AndromedaParameters.AndromedaDecoyMode)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Set the decoy mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setDecoySeed-java.lang.Integer-">setDecoySeed(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the decoy seed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html#setDecoysInFirstBin-java.lang.Integer-">setDecoysInFirstBin(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PsmScoringParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PsmScoringParameters</a></dt>
<dd>
<div class="block">Sets the minimal number of decoys to include in a bin to set the bin size
 of the score histogram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#setDecoySuffix-boolean-">setDecoySuffix(boolean)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Sets whether the decoy flag is a suffix or a prefix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setDefaultAdvancedSettings--">setDefaultAdvancedSettings()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Set the advanced settings to the default values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setDefaultAdvancedSettings-com.compomics.util.parameters.identification.search.SearchParameters-">setDefaultAdvancedSettings(SearchParameters)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Set the advanced settings to the values in the given search parameters
 object or to the default values of the advanced settings are not set for
 a given advocate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#setDefaultAttributes-java.io.File-">setDefaultAttributes(File)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Sets default name and version from the given fasta file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setDefaultModifications-java.util.HashSet-">setDefaultModifications(HashSet&lt;String&gt;)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Sets the default modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html#setDefaultPeptideFDR-double-">setDefaultPeptideFDR(double)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">IdMatchValidationParameters</a></dt>
<dd>
<div class="block">Sets the default peptide FDR.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html#setDefaultProteinFDR-double-">setDefaultProteinFDR(double)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">IdMatchValidationParameters</a></dt>
<dd>
<div class="block">Sets the default protein FDR.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html#setDefaultPsmFDR-double-">setDefaultPsmFDR(double)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">IdMatchValidationParameters</a></dt>
<dd>
<div class="block">Sets the default PSM FDR.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setDeisotopingMode-int-">setDeisotopingMode(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Sets the deisotoping flag value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setDeltaMassWindow-double-">setDeltaMassWindow(double)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set the size of the window to use when searching for matches in the known
 masses list when the user hovers over a second data point after clicking
 a previous data point.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setDeltaPEP-java.lang.Double-">setDeltaPEP(Double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Sets the difference in identification algorithm level PEP with the next
 best peptide assumption with sequence difference across all search
 engines.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#setDeltaScore-int-double-">setDeltaScore(int, double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Sets the delta score at a given site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setDeNovoCharge-int-">setDeNovoCharge(int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Set the charge to use for the fragment ions in the de novo sequencing</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setDeNovoGuiPath-java.lang.String-">setDeNovoGuiPath(String)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the path to the DeNovoGUI installation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setDependentLosses-boolean-">setDependentLosses(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets whether neutral losses should be sequence dependent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#setDescription-java.lang.String-">setDescription(String)</a></span> - Method in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Sets the description of this substitution matrix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#setDescription-java.lang.String-">setDescription(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Sets the description of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#setDescription-java.lang.String-">setDescription(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Sets the description for this database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setDescription-java.lang.String-">setDescription(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/patient/PatientInformation.html#setDescription-java.lang.String-">setDescription(String)</a></span> - Method in class com.compomics.util.experiment.patient.<a href="com/compomics/util/experiment/patient/PatientInformation.html" title="class in com.compomics.util.experiment.patient">PatientInformation</a></dt>
<dd>
<div class="block">Sets the description of the information.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setDescription-java.lang.String-boolean-">setDescription(String, boolean)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the description of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setDescriptionProteinName-java.lang.String-">setDescriptionProteinName(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the protein name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setDescriptionShort-java.lang.String-">setDescriptionShort(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the short description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#setDestinationFile-java.lang.String-">setDestinationFile(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">This method allows the specification of the destinationfile.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Instrument.html#setDetector-com.compomics.util.pride.CvTerm-">setDetector(CvTerm)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Instrument.html" title="class in com.compomics.util.pride.prideobjects">Instrument</a></dt>
<dd>
<div class="block">Set the instrument detector.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html#setDeterminate-int-int-">setDeterminate(int, int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialog</a></dt>
<dd>
<div class="block">This method sets the progressbar to determinate mode, with the specified
 minimum and maximum values.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setDetermineChargePlusOneAlgorithmically-boolean-">setDetermineChargePlusOneAlgorithmically(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets whether charge plus one should be determined algorithmically.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setDigestionParameters-com.compomics.util.parameters.identification.search.DigestionParameters-">setDigestionParameters(DigestionParameters)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the digestion preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setDigestionType-java.lang.String-">setDigestionType(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the enzyme digestion type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#setDiscardLowQualitySpectra-java.lang.Boolean-">setDiscardLowQualitySpectra(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Sets whether low quality spectra shall be discarded.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setDisplayedTips-java.util.ArrayList-">setDisplayedTips(ArrayList&lt;String&gt;)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the list of displayed tips.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setDisplayProgress-boolean-">setDisplayProgress(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setDisplayProgress-boolean-">setDisplayProgress(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#setDisplayProgress-boolean-">setDisplayProgress(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#setDisplayProgress-boolean-">setDisplayProgress(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#setDisplayProgress-boolean-">setDisplayProgress(boolean)</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Set if the waiting handler is to show the progress for the current
 process or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Reference.html#setDoi-java.lang.String-">setDoi(String)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Reference.html" title="class in com.compomics.util.pride.prideobjects">Reference</a></dt>
<dd>
<div class="block">Set the Digital Object Identifier (DOI) of the reference.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setDoubleChargeWindow-int-">setDoubleChargeWindow(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the window size for doubly charged ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setDrawLabel-boolean-">setDrawLabel(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Set to true if the label is to be drawn, false otherwise.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setDrawOnTop-boolean-">setDrawOnTop(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Set to true if the area is to be drawn in front of the data, false
 otherwise.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setDuplicateSpectra-boolean-">setDuplicateSpectra(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Sets whether to use duplicate spectra option.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setDynamicRange-double-">setDynamicRange(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets the dynamic range for spectrum filtering.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#setEditable-boolean-">setEditable(boolean)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Sets whether the scheme is editable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Contact.html#setEMail-java.lang.String-">setEMail(String)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Contact.html" title="class in com.compomics.util.pride.prideobjects">Contact</a></dt>
<dd>
<div class="block">Set the contact e-mail.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setEmpiricalCorrection-boolean-">setEmpiricalCorrection(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets whether empirical correction should be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setEnd-double-">setEnd(double)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Set the end value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#setEndBlock-int-">setEndBlock(int)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Set the block end.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#setEndProtein-int-">setEndProtein(int)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Set the protein end.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#setEnsemblVersionsMap-java.util.HashMap-">setEnsemblVersionsMap(HashMap&lt;String, String&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Sets the Ensembl version map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#setEnzymes-java.util.ArrayList-">setEnzymes(ArrayList&lt;Enzyme&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Sets the enzymes used for digestion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setEnzymeType-java.lang.Integer-">setEnzymeType(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the enzyme type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setEqualIL-boolean-">setEqualIL(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets whether I and L should be considered indistinguishable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setEstimateCharge-boolean-">setEstimateCharge(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets whether the precursor charge shall be estimated for OMSSA.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#setEstimateCharge-java.lang.Boolean-">setEstimateCharge(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Sets whether the precursor charge shall be estimated for PepNovo.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#setEvenRowColor-java.awt.Color-">setEvenRowColor(Color)</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">Sets the colour of the even numbered rows; by default, the LIGHT_AQUA background color will be used..</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#setExceptions-java.util.HashSet-">setExceptions(HashSet&lt;Long&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Sets the excepted matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#setExcluded-int-java.util.ArrayList-">setExcluded(int, ArrayList&lt;Character&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Excludes the given amino acids from the targeted amino acids at the given
 index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html#setExperiment_type-java.lang.String-">setExperiment_type(String)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb">PdbParameter</a></dt>
<dd>
<div class="block">Set the experiment type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#setExportFeatures-java.lang.String-java.util.ArrayList-">setExportFeatures(String, ArrayList&lt;ExportFeature&gt;)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Sets the export features of a given section.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setFastaFile-java.io.File-">setFastaFile(File)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the sequence database file used for identification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setFastaParameters-com.compomics.util.experiment.io.biology.protein.FastaParameters-">setFastaParameters(FastaParameters)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the parameters to use to parse the FASTA file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setFastIndexFolderName-java.lang.String-">setFastIndexFolderName(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the name of the FASTA index folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.html#setFeature-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.IonFeature.Property-">setFeature(IonFeature.Property)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">IonFeature</a></dt>
<dd>
<div class="block">Sets the peptide feature.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/FileAndFileFilter.html#setFile-java.io.File-">setFile(File)</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/FileAndFileFilter.html" title="class in com.compomics.util">FileAndFileFilter</a></dt>
<dd>
<div class="block">Set the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/FileAndFileFilter.html#setFileFilter-javax.swing.filechooser.FileFilter-">setFileFilter(FileFilter)</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/FileAndFileFilter.html" title="class in com.compomics.util">FileAndFileFilter</a></dt>
<dd>
<div class="block">Set the file filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#SetFileMode-int-">SetFileMode(int)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">This method sets the file transfer mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#setFileName-java.lang.String-">setFileName(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Sets the file name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#setFilename-java.lang.String-">setFilename(String)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Setter for the filename</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#setFilename-java.lang.String-">setFilename(String)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method sets the filename for the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#setFilename-java.lang.String-">setFilename(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">This method sets the name of the file to up- or download.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#setFileName-java.lang.String-">setFileName(String)</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setFilenameColor-java.awt.Color-">setFilenameColor(Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method sets the display color for the filename on the panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterDialog.html#setFilter--">setFilter()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterDialog.html" title="class in com.compomics.util.gui.filtering">FilterDialog</a></dt>
<dd>
<div class="block">/**
 Modifies the filter according to the user selection.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#setFiltered-boolean-">setFiltered(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Sets whether the protein list is filtered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#setFilterFromSearchParameters-com.compomics.util.parameters.identification.search.SearchParameters-">setFilterFromSearchParameters(SearchParameters)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Updates the filter based on the search parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#setFilterItem-java.lang.String-com.compomics.util.experiment.filtering.FilterItemComparator-java.lang.Object-">setFilterItem(String, FilterItemComparator, Object)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Sets an item to the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#setFirstDecoy-boolean-">setFirstDecoy(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Sets whether the number of hits before the first decoy should be checked.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/DbObject.html#setFirstLevel-boolean-">setFirstLevel(boolean)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/DbObject.html" title="class in com.compomics.util.db.object">DbObject</a></dt>
<dd>
<div class="block">Sets whether an object is a first level object or not i.e.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html#setFixed-boolean-">setFixed(boolean)</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/NeutralLoss.html" title="class in com.compomics.util.experiment.biology.ions">NeutralLoss</a></dt>
<dd>
<div class="block">Sets whether the loss is fixed or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#setFixedAaModifications-java.util.HashMap-">setFixedAaModifications(HashMap&lt;Integer, String&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Sets the modifications at specific amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#setFontSize-int-">setFontSize(int)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Set the font size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#setFontSizeLegend-int-">setFontSizeLegend(int)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Set the font size of the legend.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setForeignAccession-java.lang.String-">setForeignAccession(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the foreign accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setForeignDescription-java.lang.String-">setForeignDescription(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the foreign description.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setForeignID-java.lang.String-">setForeignID(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the foreign ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setForwardIons-java.util.ArrayList-">setForwardIons(ArrayList&lt;Integer&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the forward ions searched as list of integers as indexed in the
 FragmentIon class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setFoundModifications-java.util.TreeSet-">setFoundModifications(TreeSet&lt;String&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the list of variable modifications found in the dataset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html#setFractionMolecularWeightRanges-java.util.HashMap-">setFractionMolecularWeightRanges(HashMap&lt;String, XYDataPoint&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html" title="class in com.compomics.util.parameters.identification.advanced">FractionParameters</a></dt>
<dd>
<div class="block">Set the user provided molecular weight ranges for the fractions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setFractionOfPeaksForChargeEstimation-double-">setFractionOfPeaksForChargeEstimation(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the fraction of peaks to be retained for charge &gt;1 estimation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setFractionParameters-com.compomics.util.parameters.identification.advanced.FractionParameters-">setFractionParameters(FractionParameters)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the fraction parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setFractionPEP-java.lang.String-java.lang.Double-">setFractionPEP(String, Double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Sets the fraction confidence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setFractionPEP-java.util.HashMap-">setFractionPEP(HashMap&lt;String, Double&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setFractionPsmMatches-java.util.HashMap-">setFractionPsmMatches(HashMap&lt;String, ArrayList&lt;Long&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Set the fraction PSM matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Identification.html#setFractions-java.util.ArrayList-">setFractions(ArrayList&lt;String&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Identification.html" title="class in com.compomics.util.experiment.identification">Identification</a></dt>
<dd>
<div class="block">Sets the fractions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setFractionScore-java.lang.String-java.lang.Double-">setFractionScore(String, Double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Sets the fraction confidence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setFractionScore-java.util.HashMap-">setFractionScore(HashMap&lt;String, Double&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Sets the fraction score map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setFragmentAccuracyType-com.compomics.util.parameters.identification.search.SearchParameters.MassAccuracyType-">setFragmentAccuracyType(SearchParameters.MassAccuracyType)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the fragment accuracy type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setFragmentAll-boolean-">setFragmentAll(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets whether the fragment all option should be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setFragmentationMethod-com.compomics.util.experiment.mass_spectrometry.FragmentationMethod-">setFragmentationMethod(FragmentationMethod)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets the fragmentation method used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html#setFragmentationMethod-java.lang.String-">setFragmentationMethod(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">NovorParameters</a></dt>
<dd>
<div class="block">Set the fragmentation method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#setFragmentationModel-java.lang.String-">setFragmentationModel(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Sets the name of the fragmentation model.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setFragmentationRule-java.lang.String-">setFragmentationRule(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Set the fragmentation rule.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#setFragmentationType-int-">setFragmentationType(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Set the fragmentation ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setFragmentBinOffset-java.lang.Double-">setFragmentBinOffset(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the fragment ion bin offset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setFragmentIonAccuracy-double-">setFragmentIonAccuracy(double)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Sets the fragment ion accuracy.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#setFragmentIonAccuracy-double-">setFragmentIonAccuracy(double)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Sets the fragment ion accuracy.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setFragmentIonAccuracy-double-">setFragmentIonAccuracy(double)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the fragment ion m/z tolerance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setFragmentIonPpm-boolean-">setFragmentIonPpm(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Sets whether the fragment ion accuracy is in ppm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#setFragmentIonPpm-boolean-">setFragmentIonPpm(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Sets whether the fragment ion accuracy is in ppm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html#setFunction-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.AASequenceFeature.Function-">setFunction(AASequenceFeature.Function)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AASequenceFeature</a></dt>
<dd>
<div class="block">Sets the function used to compare the amino acid properties.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#setGeneMappingFolder-java.lang.String-">setGeneMappingFolder(String)</a></span> - Static method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Sets the folder where gene mappings are saved.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setGeneName-java.lang.String-">setGeneName(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Set the gene name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#setGeneNameToChromosomeMap-java.util.HashMap-">setGeneNameToChromosomeMap(HashMap&lt;String, String&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Sets the gene name to chromosome map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#setGeneNameToEnsemblIdMap-java.util.HashMap-">setGeneNameToEnsemblIdMap(HashMap&lt;String, String&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Sets the gene name to Ensembl ID map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setGeneParameters-com.compomics.util.parameters.identification.advanced.GeneParameters-">setGeneParameters(GeneParameters)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the gene parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setGenerateDecoyDatabase-boolean-">setGenerateDecoyDatabase(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set whether a decoy database shall be created and searched against.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#setGenerateQuery-java.lang.Boolean-">setGenerateQuery(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Sets a boolean indicating whether a blast query shall be generated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#setGoAccessionToProteinMap-java.util.HashMap-">setGoAccessionToProteinMap(HashMap&lt;String, HashSet&lt;String&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Sets the GO to protein accession map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#setGoNamesMap-java.util.HashMap-">setGoNamesMap(HashMap&lt;String, String&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Sets the GO accession to names map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setGroupClass-int-">setGroupClass(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequence.html#setHeader-com.compomics.util.experiment.io.biology.protein.Header-">setHeader(Header)</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide">NucleotideSequence</a></dt>
<dd>
<div class="block">This method sets the header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#setHeader-com.compomics.util.experiment.io.biology.protein.Header-">setHeader(Header)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">Simple setter for the header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setHidden-boolean-">setHidden(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Hide/Unhide a match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setHigherCharge-boolean-">setHigherCharge(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets whether higher charge should be considered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setHighIntensityCutOff-double-">setHighIntensityCutOff(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the high intensity cut-off as percentage of the most intense ion
 peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setHitListLength-int-">setHitListLength(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the length of the hit list for OMSSA.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#setHitListLength-java.lang.Integer-">setHitListLength(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Sets the length of the hit list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/DbObject.html#setId-long-">setId(long)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/DbObject.html" title="class in com.compomics.util.db.object">DbObject</a></dt>
<dd>
<div class="block">Sets the id of the object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setID-java.lang.String-">setID(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setIdentificationAlgorithmParameter-int-com.compomics.util.gui.parameters.identification.IdentificationAlgorithmParameter-">setIdentificationAlgorithmParameter(int, IdentificationAlgorithmParameter)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Adds identification algorithm specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#setIdentificationCharge-int-">setIdentificationCharge(int)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Sets the identification charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#setIdentificationFeaturesCache-com.compomics.util.experiment.identification.IdentificationFeaturesCache-">setIdentificationFeaturesCache(IdentificationFeaturesCache)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Sets the the identification features cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#setIdentificationFile-java.lang.String-">setIdentificationFile(String)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html#setIdentificationParameters-com.compomics.util.parameters.identification.IdentificationParameters-">setIdentificationParameters(IdentificationParameters)</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersInputBean</a></dt>
<dd>
<div class="block">Sets the identification parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setIdentifier-java.lang.String-">setIdentifier(String)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Sets the references identifier.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setIdValidationParameters-com.compomics.util.parameters.identification.advanced.IdMatchValidationParameters-">setIdValidationParameters(IdMatchValidationParameters)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the identification matches validation parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/ExceptionHandler.html#setIgnoreExceptions-boolean-">setIgnoreExceptions(boolean)</a></span> - Method in class com.compomics.util.exceptions.<a href="com/compomics/util/exceptions/ExceptionHandler.html" title="class in com.compomics.util.exceptions">ExceptionHandler</a></dt>
<dd>
<div class="block">Sets whether exceptions should be ignored.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setIncludeAmmonia-boolean-">setIncludeAmmonia(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets whether ammonia losses should be accounted for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setIncludeDateInOutputName-boolean-">setIncludeDateInOutputName(boolean)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Sets whether the date should be included in the SearchGUI output name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#setIncludeDecoy-java.lang.Boolean-">setIncludeDecoy(Boolean)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Sets whether decoy hits should be included.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setIncludeWater-boolean-">setIncludeWater(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets whether water losses should be accounted for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#setIndentString-java.lang.String-">setIndentString(String)</a></span> - Method in class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Sets the indent string to use, e.g.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html#setIndeterminate--">setIndeterminate()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialog</a></dt>
<dd>
<div class="block">This method makes the progressbar indeterminate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#setIndexOnProtein-int-">setIndexOnProtein(int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Sets the peptide index on the protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html#setInferred-boolean-">setInferred(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches">ModificationMatch</a></dt>
<dd>
<div class="block">Sets whether the modification is inferred from another peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Contact.html#setInstitution-java.lang.String-">setInstitution(String)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Contact.html" title="class in com.compomics.util.pride.prideobjects">Contact</a></dt>
<dd>
<div class="block">Set the instituttion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setInstrumentID-java.lang.String-">setInstrumentID(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set the instrument ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#setInstrumentID-int-">setInstrumentID(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Set the instrument ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#setIntegerParameter-java.lang.String-java.lang.Integer-">setIntegerParameter(String, Integer)</a></span> - Method in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#setIntensity-double-">setIntensity(double)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Setter for the intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#setIntensity-double-">setIntensity(double)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method sets the intensity of the precursor ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setIntensityCutOffIncrement-double-">setIntensityCutOffIncrement(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the intensity cut-off increment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setIntensityLimit-double-">setIntensityLimit(double)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Sets the intensity percentile to consider for annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setIntensityScoreWeight-double-">setIntensityScoreWeight(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Set the intensity score weight.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setIntensityThresholdType-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters.IntensityThresholdType-">setIntensityThresholdType(AnnotationParameters.IntensityThresholdType)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Sets the intensity threshold type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#setIntensityValues-double:A-">setIntensityValues(double[])</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#setIntensityValuesAsArray-double:A-">setIntensityValuesAsArray(double[])</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Setter for the intensityValuesAsArray.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setIntermediateScore-java.lang.Integer-java.lang.Double-">setIntermediateScore(Integer, Double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Adds an intermediate score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setIntermediateScores-java.util.HashMap-">setIntermediateScores(HashMap&lt;Integer, Double&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#setIonColor-com.compomics.util.experiment.biology.ions.Ion-java.awt.Color-">setIonColor(Ion, Color)</a></span> - Static method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Sets an annotation color for the given ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SequenceFragmentationPanel.html#setIonMatches-com.compomics.util.experiment.identification.matches.IonMatch:A-">setIonMatches(IonMatch[])</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SequenceFragmentationPanel.html" title="class in com.compomics.util.gui.spectrum">SequenceFragmentationPanel</a></dt>
<dd>
<div class="block">Set the ArrayList with FragmentIon matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html#setIsotope-java.lang.Integer-">setIsotope(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html" title="class in com.compomics.util.experiment.biology.atoms">AtomImpl</a></dt>
<dd>
<div class="block">Sets the isotope, 0 for monoisotope.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setIsotopeCorrection-java.lang.Integer-">setIsotopeCorrection(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the isotope correction setting.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setIsotopeMzTolerance-double-">setIsotopeMzTolerance(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Sets the deisotoping m/z tolerance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#setIsPpm-boolean-">setIsPpm(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Sets whether the mass tolerance is in ppm (true) or Dalton (false).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#setIsScrolling-boolean-">setIsScrolling(boolean)</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Set if the user is currently scrolling or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setIterativeReplaceEvalue-double-">setIterativeReplaceEvalue(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the e-value threshold to use to replace a hit for the iterative
 search.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setIterativeSequenceEvalue-double-">setIterativeSequenceEvalue(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the e-value threshold to use to consider a sequence for the
 iterative search.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setIterativeSpectrumEvalue-double-">setIterativeSpectrumEvalue(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the e-value threshold to use consider a spectrum for the iterative
 search.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setiXAxisMax-double-">setiXAxisMax(double)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Sets the maximum value for the x-axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setiXAxisMin-double-">setiXAxisMin(double)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Sets the minimum value for the x-axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setiYAxisMax-double-">setiYAxisMax(double)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Sets the maximum value for the y-axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setiYAxisMin-double-">setiYAxisMin(double)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Sets the minimum value for the y-axis.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setJavaHome-java.lang.String-">setJavaHome(String)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the Java Home folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setKeepTerminalAminoAcids-java.lang.String-">setKeepTerminalAminoAcids(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the option for keeping the terminal amino acids when generating the
 decoys</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#setKey-long-">setKey(long)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Sets the object key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html#setKey-long-">setKey(long)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html" title="class in com.compomics.util.experiment.identification.matches">PeptideMatch</a></dt>
<dd>
<div class="block">Sets a new key for the match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#setKeyWords-java.util.ArrayList-">setKeyWords(ArrayList&lt;String&gt;)</a></span> - Method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Sets the the key words for this report.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setKnownMassDeltas-java.util.HashMap-">setKnownMassDeltas(HashMap&lt;Double, String&gt;)</a></span> - Static method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set all the known mass deltas (if any).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setLabel-java.lang.String-">setLabel(String)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Sets the label.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setLabelColor-java.awt.Color-">setLabelColor(Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Set the label color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistribution.html#setLabelDifference-int-">setLabelDifference(int)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistribution.html" title="class in com.compomics.util.general">IsotopicDistribution</a></dt>
<dd>
<div class="block">This method set the label dalton difference</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setLastSelectedFolder-java.lang.String-">setLastSelectedFolder(String)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Sets the last selected folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/LastSelectedFolder.html#setLastSelectedFolder-java.lang.String-java.lang.String-">setLastSelectedFolder(String, String)</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/LastSelectedFolder.html" title="class in com.compomics.util.io.file">LastSelectedFolder</a></dt>
<dd>
<div class="block">Sets the last selected folder of the given use case.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/LastSelectedFolder.html#setLastSelectedFolder-java.lang.String-">setLastSelectedFolder(String)</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/LastSelectedFolder.html" title="class in com.compomics.util.io.file">LastSelectedFolder</a></dt>
<dd>
<div class="block">Sets the last selected folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setLastSelectedFolder-com.compomics.util.io.file.LastSelectedFolder-">setLastSelectedFolder(LastSelectedFolder)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Sets the last selected folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#setLeadingAccession-java.lang.String-">setLeadingAccession(String)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Sets the leading accession for this match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#setLegendDatasetAFourWay-no.uib.jsparklines.data.XYDataPoint-">setLegendDatasetAFourWay(XYDataPoint)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Set the legend location of Dataset A in a four way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#setLegendDatasetAThreeWay-no.uib.jsparklines.data.XYDataPoint-">setLegendDatasetAThreeWay(XYDataPoint)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Set the legend location of Dataset A in a three way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#setLegendDatasetBFourWay-no.uib.jsparklines.data.XYDataPoint-">setLegendDatasetBFourWay(XYDataPoint)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Set the legend location of Dataset B in a four way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#setLegendDatasetBThreeWay-no.uib.jsparklines.data.XYDataPoint-">setLegendDatasetBThreeWay(XYDataPoint)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Set the legend location of Dataset B in a three way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#setLegendDatasetCFourWay-no.uib.jsparklines.data.XYDataPoint-">setLegendDatasetCFourWay(XYDataPoint)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Set the legend location of Dataset C in a four way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#setLegendDatasetCThreeWay-no.uib.jsparklines.data.XYDataPoint-">setLegendDatasetCThreeWay(XYDataPoint)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Set the legend location of Dataset C in a three way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#setLegendDatasetDFourWay-no.uib.jsparklines.data.XYDataPoint-">setLegendDatasetDFourWay(XYDataPoint)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Set the legend location of Dataset D in a four way Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html#setLimitX-double-">setLimitX(double)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a></dt>
<dd>
<div class="block">Sets the maximal share of X's a match can contain, range [0.0-1.0].</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#setLine-int-java.util.ArrayList-">setLine(int, ArrayList&lt;Double&gt;)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setLocalPrideFolder-java.lang.String-">setLocalPrideFolder(String)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the local PRIDE folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setLocation-int-int-">setLocation(int, int)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method allows the caller to add information to the header about
 location of the sequence in a certain master sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#setLocation-int-">setLocation(int)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">This method allows the setting of the location for this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#setLocation-int-">setLocation(int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This method allows the setting of the location for this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/UtilitiesGUIDefaults.html#setLookAndFeel--">setLookAndFeel()</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/UtilitiesGUIDefaults.html" title="class in com.compomics.util.gui">UtilitiesGUIDefaults</a></dt>
<dd>
<div class="block">Sets the look and feel to the default utilities look and feel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setLowerClearMzRange-java.lang.Double-">setLowerClearMzRange(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the lower mass value for the clear mz range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#setLowerPrecursorMass-int-">setLowerPrecursorMass(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Set the lower precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setLowIntensityCutOff-double-">setLowIntensityCutOff(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the low intensity cut-off as percentage of the most intense ion
 peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setLowMemoryMode-boolean-">setLowMemoryMode(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">use setMaxLoadedProteins and setMaxLoadedSpectra instead</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#setManualValidation-java.util.HashSet-">setManualValidation(HashSet&lt;Long&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Sets the list of manually validated keys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setManualValidation-boolean-">setManualValidation(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Sets whether the match validation was manually inspected.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#setMaps-com.compomics.util.experiment.biology.genes.go.GoMapping-">setMaps(GoMapping)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Imports the GO maps from a GO mapping.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#setMass-double-">setMass(double)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Method to set the mass of the reporter ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#setMass-double-">setMass(double)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Sets the mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html#setMass-double-">setMass(double)</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/MassGap.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">MassGap</a></dt>
<dd>
<div class="block">Sets the mass of the gap.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#setMass-double-">setMass(double)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Sets the mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html#setMassAnalyzer-java.lang.String-">setMassAnalyzer(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">NovorParameters</a></dt>
<dd>
<div class="block">Set the mass analyzer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#setMassErrorDistribution-java.lang.String-java.util.ArrayList-">setMassErrorDistribution(String, ArrayList&lt;Double&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Sets a mass error distribution in the massErrorDistribution map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#setMassShift-double-">setMassShift(double)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Sets an m/z shift on all ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#setMassShifts-double-double-double-">setMassShifts(double, double, double)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Sets the m/z shifts.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#setMassTolerance-double-boolean-com.compomics.util.experiment.identification.spectrum_annotation.SpectrumAnnotator.TiesResolution-">setMassTolerance(double, boolean, SpectrumAnnotator.TiesResolution)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Sets a new m/z tolerance for peak matching.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#setMatchingKey-long-">setMatchingKey(long)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Sets the key accounting for sequence matching preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setMatchValidationLevel-com.compomics.util.experiment.identification.validation.MatchValidationLevel-">setMatchValidationLevel(MatchValidationLevel)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Sets the validation level of the match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#setMatchValidationLevel-java.lang.Integer-">setMatchValidationLevel(Integer)</a></span> - Method in class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Sets the lowest validation level to consider as an integer as indexed in
 the MatchValidationLevel enum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#setMaxCharge-java.lang.Integer-">setMaxCharge(Integer)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Sets the maximum charge in this file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setMaxChargeSearched-int-">setMaxChargeSearched(int)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the maximal charge searched.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setMaxCombinations-int-">setMaxCombinations(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets the maximal number of combinations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setMaxDynamicMods-int-">setMaxDynamicMods(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Sets the maximum number of dynamic mods.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setMaxDynamicMods-java.lang.Integer-">setMaxDynamicMods(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Set the maximum number of variable modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setMaxEValue-double-">setMaxEValue(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the maximal e-value searched for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setMaxEValue-double-">setMaxEValue(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets the maximal e-value searched for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setMaxFragmentCharge-java.lang.Integer-">setMaxFragmentCharge(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the maximum fragment ion charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setMaxFragmentCharge-int-">setMaxFragmentCharge(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the maximal fragment charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setMaxFragmentPerSeries-int-">setMaxFragmentPerSeries(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the maximal number of fragments to retain per series.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setMaxHitsPerSpectrumPerCharge-int-">setMaxHitsPerSpectrumPerCharge(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Set the maximal number of hits searched per spectrum and per charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setMaximumExpectationValueRefinement-double-">setMaximumExpectationValueRefinement(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets the maximum expectation value to use for refinement.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#setMaxIntensity-java.lang.Double-">setMaxIntensity(Double)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Sets the maximum precursor intensity in this file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#setMaxIsotopes-java.lang.Integer-">setMaxIsotopes(Integer)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Sets the maximal number of isotopes allowed (inclusive).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setMaxIsotopicCorrection-int-">setMaxIsotopicCorrection(int)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the maximal isotopic correction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/KMeansClustering.html#setMaxIterations-int-">setMaxIterations(int)</a></span> - Method in class com.compomics.util.math.clustering.<a href="com/compomics/util/math/clustering/KMeansClustering.html" title="class in com.compomics.util.math.clustering">KMeansClustering</a></dt>
<dd>
<div class="block">Set the maximum number of iterations.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setMaxLoadedProteins-java.lang.Integer-">setMaxLoadedProteins(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set the maximum number of proteins loaded into memory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setMaxLoadedSpectra-java.lang.Integer-">setMaxLoadedSpectra(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set the maximum number of spectra loaded into memory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#setMaxMissedCleavages-java.lang.Integer-">setMaxMissedCleavages(Integer)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Set the maximum number of missed cleavages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setMaxModifications-java.lang.Integer-">setMaxModifications(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set the maximum number of modifications per peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setMaxModificationSites-java.lang.Integer-">setMaxModificationSites(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set the maximum number of modifications sites per modification per
 peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxMW-java.lang.Double-">setMaxMW(Double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the molecular weight of the fattest protein in the dataset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#setMaxMz-java.lang.Double-">setMaxMz(Double)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Sets the maximum m/z in this file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#setMaxMzDeviation-double-">setMaxMzDeviation(double)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Sets the maximal m/z deviation allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setMaxMzLadders-int-">setMaxMzLadders(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the maximal m/z ladder length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setMaxNeutralLosses-java.lang.Integer-">setMaxNeutralLosses(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set the maximum number of water and ammonia losses per peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setMaxNeutralLossesPerModification-java.lang.Integer-">setMaxNeutralLossesPerModification(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set the maximum number identical modification specific losses per
 peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxNPeptides-java.lang.Integer-">setMaxNPeptides(Integer)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the maximal amount of peptides in the proteins of the dataset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxNSpectra-java.lang.Integer-">setMaxNSpectra(Integer)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the the maximal amount of PSMs in the proteins of the dataset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setMaxNumberOfModifications-int-">setMaxNumberOfModifications(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets the maximal number of modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setMaxPadding-int-">setMaxPadding(int)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set the max padding (distance between the axes and the border of the
 panel).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#setMaxPeakCount-java.lang.Integer-">setMaxPeakCount(Integer)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Sets the maximum peak count in this file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setMaxPeakCount-int-">setMaxPeakCount(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Set the maximum peak count.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setMaxPeakCount-java.lang.Integer-">setMaxPeakCount(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Set the max peak count.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#setMaxPepLength-int-">setMaxPepLength(int)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Sets the maximal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#setMaxPeptideLength-java.lang.Integer-">setMaxPeptideLength(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Sets the maximal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setMaxPeptideLength-java.lang.Integer-">setMaxPeptideLength(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Sets the maximal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setMaxPeptideLength-int-">setMaxPeptideLength(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the maximal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMaxPeptideLength-java.lang.Integer-">setMaxPeptideLength(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Sets the maximal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setMaxPeptideLengthNoEnzyme-int-">setMaxPeptideLengthNoEnzyme(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets the maximal peptide length to use when searching with no enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setMaxPeptideMass-java.lang.Double-">setMaxPeptideMass(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets the maximal peptide mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxPeptidePrecursorErrorDa-double-">setMaxPeptidePrecursorErrorDa(double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Set the max peptide precursor mass error in Dalton.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxPeptidePrecursorErrorPpm-double-">setMaxPeptidePrecursorErrorPpm(double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Set the max peptide precursor mass error in ppm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setMaxPrecursorAdjustment-double-">setMaxPrecursorAdjustment(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Set the maximum precursor adjustment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setMaxPrecursorMass-java.lang.Double-">setMaxPrecursorMass(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the maximum precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setMaxPrecursorMass-java.lang.Double-">setMaxPrecursorMass(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Sets the maximum precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMaxPrecursorMass-java.lang.Double-">setMaxPrecursorMass(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Sets the maximum precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setMaxPrimaryProgressCounter-int-">setMaxPrimaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setMaxPrimaryProgressCounter-int-">setMaxPrimaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Set the maximum value of the progress bar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#setMaxPrimaryProgressCounter-int-">setMaxPrimaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#setMaxPrimaryProgressCounter-int-">setMaxPrimaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#setMaxPrimaryProgressCounter-int-">setMaxPrimaryProgressCounter(int)</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Set the maximum value for the primary progress counter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxProteinAveragePrecursorIntensity-java.lang.Double-">setMaxProteinAveragePrecursorIntensity(Double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Set the maximum protein average precursor intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxProteinKeyLength-java.lang.Integer-">setMaxProteinKeyLength(Integer)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the max protein key length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxProteinSummedPrecursorIntensity-java.lang.Double-">setMaxProteinSummedPrecursorIntensity(Double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Set the maximum summed protein precursor intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setMaxRank-java.lang.Integer-">setMaxRank(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set the maximum rank.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#setMaxRT-java.lang.Double-">setMaxRT(Double)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Sets the maximal RT in this file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setMaxSecondaryProgressCounter-int-">setMaxSecondaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Sets the maximal value of the progress bar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setMaxSecondaryProgressCounter-int-">setMaxSecondaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Set the maximum value of the secondary progress bar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#setMaxSecondaryProgressCounter-int-">setMaxSecondaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#setMaxSecondaryProgressCounter-int-">setMaxSecondaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#setMaxSecondaryProgressCounter-int-">setMaxSecondaryProgressCounter(int)</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Set the maximum value for the secondary progress counter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setMaxSpectrumChargeRange-int-">setMaxSpectrumChargeRange(int)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the maximum charge added when the charge is missing for a given
 spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#setMaxSpectrumCount-int-">setMaxSpectrumCount(int)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Sets the maximal amount of PSMs for the peptides in the current peptide
 list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxSpectrumCounting-java.lang.Double-">setMaxSpectrumCounting(Double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the maximal spectrum counting value of the proteins of the dataset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMaxSpectrumMz-java.lang.Double-">setMaxSpectrumMz(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the max spectrum m/z to search for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setMaxTagCount-int-">setMaxTagCount(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Sets the maximum number of tags per spectrum to be generated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxTagPrecursorErrorDa-double-">setMaxTagPrecursorErrorDa(double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Set the max tag precursor mass error in Dalton.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxTagPrecursorErrorPpm-double-">setMaxTagPrecursorErrorPpm(double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Set the max tag precursor mass error in ppm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxValidatedPeptidesPerFraction-java.lang.Integer-">setMaxValidatedPeptidesPerFraction(Integer)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Set the maximum validated peptides at the fraction level.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setMaxValidatedSpectraPerFraction-java.lang.Integer-">setMaxValidatedSpectraPerFraction(Integer)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Set the maximum validated spectra at the fraction level.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setMaxVariableModifications-java.lang.Integer-">setMaxVariableModifications(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set the maximum number of variable modifications per peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMaxVariableModificationsPerPeptide-java.lang.Integer-">setMaxVariableModificationsPerPeptide(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the maximum number of variable modifications of allowed on a single
 peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMaxVariableModificationsPerTypePerPeptide-java.lang.Integer-">setMaxVariableModificationsPerTypePerPeptide(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the maximum number of variable modifications of each modification
 type allowed on a single peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setMaxVariableMods-java.lang.Integer-">setMaxVariableMods(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the maximum number of variable modifications per peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setMemoryMappedSequenceLibraries-boolean-">setMemoryMappedSequenceLibraries(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets whether sequence libraries should be mapped in memory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setMemoryParameter-int-">setMemoryParameter(int)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Sets the upper memory limit.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#setMemoryShare-double-">setMemoryShare(double)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Sets the share of heap size which can be used before emptying the cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html#setMessage-java.lang.String-">setMessage(String)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialog</a></dt>
<dd>
<div class="block">This method allows the caller to set a message on the progressbar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#setMessage-java.lang.String-">setMessage(String)</a></span> - Method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Sets the message.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#setMethod-com.compomics.util.experiment.quantification.reporterion.ReporterMethod-">setMethod(ReporterMethod)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">Sets the reporter method used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/Quantification.html#setMethodUsed-com.compomics.util.experiment.quantification.Quantification.QuantificationMethod-">setMethodUsed(Quantification.QuantificationMethod)</a></span> - Method in class com.compomics.util.experiment.quantification.<a href="com/compomics/util/experiment/quantification/Quantification.html" title="class in com.compomics.util.experiment.quantification">Quantification</a></dt>
<dd>
<div class="block">setter for the method used</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setMgfMaxSize-double-">setMgfMaxSize(double)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the max mgf file size before splitting.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setMgfNSpectra-int-">setMgfNSpectra(int)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the max number of spectra in an mgf file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html#setMH-double-">setMH(double)</a></span> - Method in class com.compomics.util.experiment.refinement_parameters.<a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html" title="class in com.compomics.util.experiment.refinement_parameters">PepnovoAssumptionDetails</a></dt>
<dd>
<div class="block">Sets the PepNovo provided mH.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setMinAnnotatedPeaks-int-">setMinAnnotatedPeaks(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the minimal number of annotated peaks a peptide should have.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setMinChargeSearched-int-">setMinChargeSearched(int)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the minimal charge searched.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setMinFragmentMz-double-">setMinFragmentMz(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets the minimal fragment m/z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setMiniature-boolean-">setMiniature(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set if the graphics panel is to be drawn in a miniature form.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setMinimalChargeForMultipleChargedFragments-int-">setMinimalChargeForMultipleChargedFragments(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the minimal precursor charge to account for multiply charged
 fragments in OMSSA.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#setMinIsotopes-java.lang.Integer-">setMinIsotopes(Integer)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Sets the minimal number of isotopes allowed (inclusive).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setMinIsotopicCorrection-int-">setMinIsotopicCorrection(int)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the minimal isotopic correction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#setMinMissedCleavages-java.lang.Integer-">setMinMissedCleavages(Integer)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Set the minimum number of missed cleavages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setMinPeakIntensity-java.lang.Double-">setMinPeakIntensity(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the minimum peak intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setMinPeaks-java.lang.Integer-">setMinPeaks(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the minimum number of peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setMinPeaks-int-">setMinPeaks(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the minimal number of peaks a spectrum should contain.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setMinPeaksPerSpectrum-int-">setMinPeaksPerSpectrum(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets the minimal number of peaks per spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#setMinPepLength-int-">setMinPepLength(int)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Sets the maximal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setMinPeptideLength-java.lang.Integer-">setMinPeptideLength(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set the minimum peptide length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#setMinPeptideLength-java.lang.Integer-">setMinPeptideLength(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Sets the minimal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setMinPeptideLength-java.lang.Integer-">setMinPeptideLength(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Sets the minimal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setMinPeptideLength-int-">setMinPeptideLength(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the minimal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMinPeptideLength-java.lang.Integer-">setMinPeptideLength(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Sets the minimal peptide length allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setMinPeptideLengthNoEnzyme-int-">setMinPeptideLengthNoEnzyme(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets the minimal peptide length to use when searching with no enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setMinPrecPerSpectrum-int-">setMinPrecPerSpectrum(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the minimal number of precursors per spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setMinPrecursorAdjustment-double-">setMinPrecursorAdjustment(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Set the minimum precursor adjustment.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setMinPrecursorMass-java.lang.Double-">setMinPrecursorMass(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the minimum precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setMinPrecursorMass-java.lang.Double-">setMinPrecursorMass(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Sets the minimum precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMinPrecursorMass-java.lang.Double-">setMinPrecursorMass(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Sets the minimum precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setMinPrecursorMass-double-">setMinPrecursorMass(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets the minimal precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#setMinRT-java.lang.Double-">setMinRT(Double)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Sets the minimum RT in this file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setMinSpectrumChargeRange-int-">setMinSpectrumChargeRange(int)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the minimum charge added when the charge is missing for a given
 spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMinSpectrumMz-java.lang.Double-">setMinSpectrumMz(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the min spectrum m/z to search for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMinSpectrumPeaks-java.lang.Integer-">setMinSpectrumPeaks(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the min number of peaks in a spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setMinTerminiCleavages-java.lang.Integer-">setMinTerminiCleavages(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Set the minimum number of termini cleavages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#setMinus1-double-">setMinus1(double)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Sets the isotopic correction factor at -1 Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#setMinus2-double-">setMinus2(double)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Sets the isotopic correction factor at -2 Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#setMiscleavages-int-">setMiscleavages(int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method allows the caller to specify the number of allowed missed
 cleavages for this enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JTableForDB.html#setModel-javax.swing.table.TableModel-boolean-">setModel(TableModel, boolean)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/JTableForDB.html" title="class in com.compomics.util.gui">JTableForDB</a></dt>
<dd>
<div class="block">This method allows you to specify a tablemodel and a boolean
 which indicates whether or not the table should be sortable by
 clicking the column headers.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/JTableForDB.html#setModel-javax.swing.table.TableModel-">setModel(TableModel)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/JTableForDB.html" title="class in com.compomics.util.gui">JTableForDB</a></dt>
<dd>
<div class="block">Sets a TableModel for the table and defaults it to being sortable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableMap.html#setModel-javax.swing.table.TableModel-">setModel(TableModel)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun">TableMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#setModel-javax.swing.table.TableModel-">setModel(TableModel)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html#setModification-java.lang.String-">setModification(String)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches">ModificationMatch</a></dt>
<dd>
<div class="block">Sets the theoretic PTM.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setModificationLocalizationParameters-com.compomics.util.parameters.identification.advanced.ModificationLocalizationParameters-">setModificationLocalizationParameters(ModificationLocalizationParameters)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the modification localization parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setModificationParameters-com.compomics.util.parameters.identification.search.ModificationParameters-">setModificationParameters(ModificationParameters)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the modification settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#setModificationRefinement-boolean-">setModificationRefinement(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Sets whether the protein mapping should be refined based on terminal and pattern modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setModifications-java.util.ArrayList-">setModifications(ArrayList&lt;String&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Sets the variable modifications searched.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#setModifications-java.util.Vector-">setModifications(Vector)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">Simple setter for the modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setMonoIsotopic-boolean-">setMonoIsotopic(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set whether monoisotopic mass values shall be used (in contrast to
 average mass values).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMonoisotopicPrecursor-java.lang.Boolean-">setMonoisotopicPrecursor(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the precursor mass is monoisotopic.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html#setMsFormat-com.compomics.util.experiment.mass_spectrometry.proteowizard.MsFormat-">setMsFormat(MsFormat)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsConvertParameters.html" title="class in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsConvertParameters</a></dt>
<dd>
<div class="block">Sets the format to convert to.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMzBinOffset-java.lang.Double-">setMzBinOffset(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the m/z bin offset.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMzBinWidth-java.lang.Double-">setMzBinWidth(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the m/z bin width.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setMzFidelityScoreWeight-double-">setMzFidelityScoreWeight(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Set the mzFidelity score weight.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setMzidOutput-java.lang.Boolean-">setMzidOutput(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the mzid output is to be created.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#setMzValues-double:A-">setMzValues(double[])</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#setnAaDeletions-int-">setnAaDeletions(int)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Sets the number of amino acid deletions allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#setnAaInsertions-int-">setnAaInsertions(int)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Sets the number of amino acid insertions allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#setnAaSubstitutions-int-">setnAaSubstitutions(int)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Sets the number of amino acid substitutions allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#setnAaSwap-int-">setnAaSwap(int)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Sets the number of amino acid swaps allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Setter for the ion name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Sets the modification name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Sets the name of this substitution matrix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Sets the name of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Sets a new name for the database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Sets the name of the reagent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Sets the name of the scheme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the name of the parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/CvTerm.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/CvTerm.html" title="class in com.compomics.util.pride">CvTerm</a></dt>
<dd>
<div class="block">Sets the name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Contact.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Contact.html" title="class in com.compomics.util.pride.prideobjects">Contact</a></dt>
<dd>
<div class="block">Set the contact name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ContactGroup.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ContactGroup.html" title="class in com.compomics.util.pride.prideobjects">ContactGroup</a></dt>
<dd>
<div class="block">Set the contact group name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Instrument.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Instrument.html" title="class in com.compomics.util.pride.prideobjects">Instrument</a></dt>
<dd>
<div class="block">Set the instrument name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Protocol.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Protocol.html" title="class in com.compomics.util.pride.prideobjects">Protocol</a></dt>
<dd>
<div class="block">Set the protocol name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html" title="class in com.compomics.util.pride.prideobjects">ReferenceGroup</a></dt>
<dd>
<div class="block">Set the references group name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Sample.html#setName-java.lang.String-">setName(String)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Sample.html" title="class in com.compomics.util.pride.prideobjects">Sample</a></dt>
<dd>
<div class="block">Set the name of the sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setnAnnotatedMostIntensePeaks-int-">setnAnnotatedMostIntensePeaks(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the minimal number of annotated most intense peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/clustering/settings/KMeansClusteringSettings.html#setnClusters-int-">setnClusters(int)</a></span> - Method in class com.compomics.util.math.clustering.settings.<a href="com/compomics/util/math/clustering/settings/KMeansClusteringSettings.html" title="class in com.compomics.util.math.clustering.settings">KMeansClusteringSettings</a></dt>
<dd>
<div class="block">Sets the number of clusters to use.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setnConfidentProteins-int-">setnConfidentProteins(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the number of confident proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#setNeutralLosses-java.util.ArrayList-">setNeutralLosses(ArrayList&lt;NeutralLoss&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Sets the neutral losses possibly encountered with this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#setNeutralLossesAuto-boolean-">setNeutralLossesAuto(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Sets whether neutral losses should be automatically selected.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#setNeutralLossesMap-com.compomics.util.experiment.identification.spectrum_annotation.NeutralLossesMap-">setNeutralLossesMap(NeutralLossesMap)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Sets the map of neutral losses to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setNeutralLossesSequenceAuto-boolean-">setNeutralLossesSequenceAuto(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Sets whether neutral losses are considered only for amino acids of
 interest or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setNeutronThreshold-double-">setNeutronThreshold(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the mass after which exact neutron mass should be considered in the
 calculation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/DummyFrame.html#setNewTitle-java.lang.String-">setNewTitle(String)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/DummyFrame.html" title="class in com.compomics.util.gui">DummyFrame</a></dt>
<dd>
<div class="block">Update the frame title and return the frame.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#setnMissedCleavages-java.lang.String-int-">setnMissedCleavages(String, int)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Sets the number of allowed missed cleavages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setNoProlineRuleSeries-java.util.ArrayList-">setNoProlineRuleSeries(ArrayList&lt;Integer&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the id numbers of ion series to apply no product ions at proline
 rule at.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#setNormalizationFactors-com.compomics.util.experiment.normalization.NormalizationFactors-">setNormalizationFactors(NormalizationFactors)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">Sets the normalization factors.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#setNormalize-java.lang.Boolean-">setNormalize(Boolean)</a></span> - Method in class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Sets whether the spectrum counting index should be normalized.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html#setNovorPtmMap-java.util.HashMap-">setNovorPtmMap(HashMap&lt;String, String&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">NovorParameters</a></dt>
<dd>
<div class="block">Set the Novor to utilities PTM map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setnPeaks-int-">setnPeaks(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets the number of most intense peaks to consider.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setnPeaksInDoubleChargeWindow-int-">setnPeaksInDoubleChargeWindow(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the number of peaks in doubly charged windows.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setnPeaksInSingleChargeWindow-int-">setnPeaksInSingleChargeWindow(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the number of peaks in singly charged windows.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#setnTermModification-java.lang.String-">setnTermModification(String)</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Sets the N-term modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/tools/ProcessingParameters.html#setnThreads-int-">setnThreads(int)</a></span> - Method in class com.compomics.util.parameters.tools.<a href="com/compomics/util/parameters/tools/ProcessingParameters.html" title="class in com.compomics.util.parameters.tools">ProcessingParameters</a></dt>
<dd>
<div class="block">Sets the number or threads to use.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setNumberOfBatches-java.lang.Integer-">setNumberOfBatches(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Set the number of batches per node to strive for when using the MPI-based
 parallelization features.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setNumberOfCandidates-int-">setNumberOfCandidates(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets the number of candidates.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#setNumberOfModificationsPerPeptide-java.lang.Integer-">setNumberOfModificationsPerPeptide(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Set the maximum number of modifications per peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#setNumberOfPeptides-int-">setNumberOfPeptides(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Set the number of peptides to report per spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#setNumberOfSpectrumMarches-java.lang.Integer-">setNumberOfSpectrumMarches(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Set the maximum number of spectrum matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setNumberOfSpectrumMatches-java.lang.Integer-">setNumberOfSpectrumMatches(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the maximum number of spectrum matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setNumberOfSpectrumMatches-java.lang.Integer-">setNumberOfSpectrumMatches(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Set the maximum number of spectrum matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setNumberOfSpectrumMatches-java.lang.Integer-">setNumberOfSpectrumMatches(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the number of PSMs to report per spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#setNumberTolerableTermini-java.lang.Integer-">setNumberTolerableTermini(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Set the number of tolerable termini.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setNumChargeStates-int-">setNumChargeStates(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Sets the number of charge states.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#setNumericParameter-java.lang.String-java.lang.Double-">setNumericParameter(String, Double)</a></span> - Method in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setNumIntensityClasses-int-">setNumIntensityClasses(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Sets the number of intensity classes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setNumIntensityClasses-java.lang.Integer-">setNumIntensityClasses(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Set the number of intensity classes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setnValidatedProteins-int-">setnValidatedProteins(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the number of validated proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#setnValidatedPsms-int-">setnValidatedPsms(int)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Sets the number of validated PSMs for the currently selected peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#setnVariants-int-">setnVariants(int)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Sets the total number of variants allowed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/BoxedObject.html#setObject-K-">setObject(K)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/BoxedObject.html" title="class in com.compomics.util.general">BoxedObject</a></dt>
<dd>
<div class="block">Sets the boxed object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsDB.html#setObjectCache-com.compomics.util.db.object.ObjectsCache-">setObjectCache(ObjectsCache)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsDB.html" title="class in com.compomics.util.db.object">ObjectsDB</a></dt>
<dd>
<div class="block">Sets the object cache to be used by this database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setObservedFractionalMassesAll-java.util.HashMap-">setObservedFractionalMassesAll(HashMap&lt;String, ArrayList&lt;Double&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Set the observed molecular masses for each fraction in kDa.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#setOccurrence-com.compomics.util.experiment.biology.atoms.Atom-java.lang.Integer-java.lang.Integer-">setOccurrence(Atom, Integer, Integer)</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>
<div class="block">Sets the occurrence of a given atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/CvTerm.html#setOntology-java.lang.String-">setOntology(String)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/CvTerm.html" title="class in com.compomics.util.pride">CvTerm</a></dt>
<dd>
<div class="block">Sets the ontology.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setOutputData-boolean-">setOutputData(boolean)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Sets whether data should be copied along with the identification files in
 the SearchGUI output.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setOutputFolderName-java.lang.String-">setOutputFolderName(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the name of the output folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setOutputFormat-java.lang.String-">setOutputFormat(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Set the output format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setOutputFormat-java.lang.String-">setOutputFormat(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Set the output format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setOutputHistograms-boolean-">setOutputHistograms(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether histograms should be written in the result file</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setOutputProteins-boolean-">setOutputProteins(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether the protein bloc should be included in the export.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setOutputResults-java.lang.String-">setOutputResults(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Set the output results filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setOutputSequences-boolean-">setOutputSequences(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether the protein sequences should be included in the protein
 block of the export.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setOutputSpectra-boolean-">setOutputSpectra(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether the spectra should be exported in the result file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setOutputSuffix-java.lang.String-">setOutputSuffix(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Sets the output tags suffix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setParametersFromSearch-com.compomics.util.parameters.identification.search.SearchParameters-">setParametersFromSearch(SearchParameters)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets identification parameters based on given search parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setParametersFromSearchParameters-com.compomics.util.parameters.identification.search.SearchParameters-">setParametersFromSearchParameters(SearchParameters)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Updates the annotation parameters based on search parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html#setParentFolder-java.lang.String-">setParentFolder(String)</a></span> - Static method in class com.compomics.util.experiment.identification.identification_parameters.<a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html" title="class in com.compomics.util.experiment.identification.identification_parameters">IdentificationParametersFactory</a></dt>
<dd>
<div class="block">Set the parent folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setParentMonoisotopicMassIsotopeError-boolean-">setParentMonoisotopicMassIsotopeError(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether the parent ion mass tolerance is expanded by opening up
 multiple tolerance windows centered on the first and second 13C isotope
 peaks for a peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#setPassive-boolean-">setPassive(boolean)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Set Passive mode Transfers.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#setPathParameter-com.compomics.software.settings.UtilitiesPathParameters.UtilitiesPathKey-java.lang.String-">setPathParameter(UtilitiesPathParameters.UtilitiesPathKey, String)</a></span> - Static method in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Sets the path according to the given key and path.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#setPattern-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidPattern-">setPattern(AminoAcidPattern)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Sets the amino acid pattern targeted by this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html#setPdbaccession-java.lang.String-">setPdbaccession(String)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb">PdbParameter</a></dt>
<dd>
<div class="block">Set the PDB accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#setPeakMap-java.util.HashMap-">setPeakMap(HashMap&lt;Double, Peak&gt;)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Sets the peak map indexed by m/z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#setPeakPicked-java.lang.Boolean-">setPeakPicked(Boolean)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Set if the indexed file seems to contain only peak picked spectra or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#setPeaks-java.util.ArrayList-">setPeaks(ArrayList&lt;Peak&gt;)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Set the peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#setPeaks-java.util.HashMap-">setPeaks(HashMap)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Setter for the peaks</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#setPeaks-java.util.HashMap-">setPeaks(HashMap)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method sets the peaks on the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#setPeaksMap-java.util.HashMap-">setPeaksMap(HashMap&lt;Integer, HashMap&lt;Double, Peak&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setPeakWaterMarkColor-java.awt.Color-">setPeakWaterMarkColor(Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set the peak water mark color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setPeakWidth-float-">setPeakWidth(float)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set the peak width.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#setPepNovoPtmMap-java.util.Map-">setPepNovoPtmMap(Map&lt;String, String&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>
<div class="block">Set the PepNovo to utilities PTM map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html#setPeptide-com.compomics.util.experiment.biology.proteins.Peptide-">setPeptide(Peptide)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html" title="class in com.compomics.util.experiment.identification.matches">PeptideMatch</a></dt>
<dd>
<div class="block">Setter for the peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#setPeptide-com.compomics.util.experiment.biology.proteins.Peptide-int-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-">setPeptide(Peptide, int, ModificationParameters, SequenceProvider, SequenceMatchingParameters, SpecificAnnotationParameters)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Sets a new peptide to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html#setPeptide-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-java.util.HashMap-int-com.compomics.util.experiment.identification.spectrum_annotation.SpecificAnnotationParameters-">setPeptide(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, HashMap&lt;Integer, HashMap&lt;Integer, ArrayList&lt;Ion&gt;&gt;&gt;, int, SpecificAnnotationParameters)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/PeptideSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">PeptideSpectrumAnnotator</a></dt>
<dd>
<div class="block">Sets a new peptide to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html#setPeptide-com.compomics.util.experiment.biology.proteins.Peptide-">setPeptide(Peptide)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/PeptideAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">PeptideAssumption</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setPeptideAssumptionFilter-com.compomics.util.experiment.identification.filtering.PeptideAssumptionFilter-">setPeptideAssumptionFilter(PeptideAssumptionFilter)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the filter used when importing PSMs.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#setPeptideAssumptionMap-java.util.HashMap-">setPeptideAssumptionMap(HashMap&lt;Integer, TreeMap&lt;Double, ArrayList&lt;PeptideAssumption&gt;&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Sets the peptide assumption map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#setPeptideFilters-java.util.ArrayList-">setPeptideFilters(ArrayList&lt;Filter&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Sets the list of peptide quality filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#setPeptideHideFilters-java.util.HashMap-">setPeptideHideFilters(HashMap&lt;String, PeptideFilter&gt;)</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Sets the peptide hide filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setPeptideLengthDistribution-com.compomics.util.math.statistics.distributions.NonSymmetricalNormalDistribution-">setPeptideLengthDistribution(NonSymmetricalNormalDistribution)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the distribution of validated peptide lengths.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#setPeptideList-long:A-">setPeptideList(long[])</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Sets the current peptide list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html#setPeptideMatchesKeys-long:A-">setPeptideMatchesKeys(long[])</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ProteinMatch.html" title="class in com.compomics.util.experiment.identification.matches">ProteinMatch</a></dt>
<dd>
<div class="block">Sets the peptide keys for this protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setPeptideShakerPath-java.lang.String-">setPeptideShakerPath(String)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the path to the PeptideShaker installation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#setPeptideStarFilters-java.util.HashMap-">setPeptideStarFilters(HashMap&lt;String, PeptideFilter&gt;)</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Sets the peptide star filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setPeptideVariantsParameters-com.compomics.util.parameters.identification.advanced.PeptideVariantsParameters-">setPeptideVariantsParameters(PeptideVariantsParameters)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the peptide variant parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setPepXmlOutput-java.lang.Boolean-">setPepXmlOutput(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the pepxml output is to be created.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setPercentLength-double-">setPercentLength(double)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Set the length of the reference area in percent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#setPerformDeisotoping-boolean-">setPerformDeisotoping(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>
<div class="block">Sets if deisotoping is to be performed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setPinOutput-java.lang.Boolean-">setPinOutput(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the Percolator output is to be created.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#setPlus1-double-">setPlus1(double)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Sets the isotopic correction factor at +1 Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#setPlus2-double-">setPlus2(double)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Sets the isotopic correction factor at +2 Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Reference.html#setPmid-java.lang.String-">setPmid(String)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Reference.html" title="class in com.compomics.util.pride.prideobjects">Reference</a></dt>
<dd>
<div class="block">Set the PubMed ID of the reference.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#setPNovoPtmMap-java.util.HashMap-">setPNovoPtmMap(HashMap&lt;Character, String&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Set the pNovo to utilities PTM map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#setPNovoPtmResiduesMap-java.util.HashMap-">setPNovoPtmResiduesMap(HashMap&lt;Character, Character&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Set the pNovo to original amino acids residue map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html#setPointSize-java.lang.Integer-">setPointSize(Integer)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html" title="class in com.compomics.util.gui.spectrum">ChromatogramPanel</a></dt>
<dd>
<div class="block">This method allows the caller to set the point size for the
 profileChromatogram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#setPosition-int-">setPosition(int)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method allows the caller to set the cleavage position for the
 Enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setPotentialModificationsForFullRefinment-boolean-">setPotentialModificationsForFullRefinment(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether the refinement modifications should be used for the full
 refinement</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#setPpm-boolean-">setPpm(boolean)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#setPrecurorCharge-int-">setPrecurorCharge(int)</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#setPrecursor-com.compomics.util.experiment.mass_spectrometry.spectra.Precursor-">setPrecursor(Precursor)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Set the precursor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setPrecursorAccuracy-double-">setPrecursorAccuracy(double)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the precursor tolerance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setPrecursorAccuracyType-com.compomics.util.parameters.identification.search.SearchParameters.MassAccuracyType-">setPrecursorAccuracyType(SearchParameters.MassAccuracyType)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the precursor accuracy type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setPrecursorAdjustmentStep-double-">setPrecursorAdjustmentStep(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Set the precursor adjustment step.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html#setPrecursorChargesMissing-java.lang.Boolean-">setPrecursorChargesMissing(Boolean)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfIndex.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfIndex</a></dt>
<dd>
<div class="block">Set if at least one spectrum is missing the precursor charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#setPrecursorIntensity-double-">setPrecursorIntensity(double)</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setPrecursorIntensityAveragePerFraction-java.util.HashMap-">setPrecursorIntensityAveragePerFraction(HashMap&lt;String, Double&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Sets the precursor intensity per fraction map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setPrecursorIntensityPerFraction-java.util.HashMap-">setPrecursorIntensityPerFraction(HashMap&lt;String, ArrayList&lt;Double&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Get the precursor intensity in the given fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setPrecursorIntensitySummedPerFraction-java.util.HashMap-">setPrecursorIntensitySummedPerFraction(HashMap&lt;String, Double&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Sets the summed precursor intensity per fraction map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#setPrecursorMZ-double-">setPrecursorMZ(double)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Setter for the precursor MZ</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#setPrecursorMZ-double-">setPrecursorMZ(double)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method sets the precursor M/Z on the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#setPrecursorMz-double-">setPrecursorMz(double)</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#setPrecursorTolerance-double-">setPrecursorTolerance(double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html#setPreferencesFromSearchParameters-com.compomics.util.parameters.identification.search.SearchParameters-">setPreferencesFromSearchParameters(SearchParameters)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html" title="class in com.compomics.util.parameters.identification.advanced">GeneParameters</a></dt>
<dd>
<div class="block">Sets the preferences from the given search parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#setPrideFolder-java.lang.String-">setPrideFolder(String)</a></span> - Static method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Sets the folder where pride annotation information should be saved.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setPrimaryProgressCounter-int-">setPrimaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Sets the value of the progress bar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setPrimaryProgressCounter-int-">setPrimaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Sets the primary progress bar to the given value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#setPrimaryProgressCounter-int-">setPrimaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#setPrimaryProgressCounter-int-">setPrimaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#setPrimaryProgressCounter-int-">setPrimaryProgressCounter(int)</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Sets the primary progress counter to the given value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setPrimaryProgressCounterIndeterminate-boolean-">setPrimaryProgressCounterIndeterminate(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setPrimaryProgressCounterIndeterminate-boolean-">setPrimaryProgressCounterIndeterminate(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#setPrimaryProgressCounterIndeterminate-boolean-">setPrimaryProgressCounterIndeterminate(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#setPrimaryProgressCounterIndeterminate-boolean-">setPrimaryProgressCounterIndeterminate(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#setPrimaryProgressCounterIndeterminate-boolean-">setPrimaryProgressCounterIndeterminate(boolean)</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Sets whether the primary progress counter is indeterminate or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setPrintExpectScore-boolean-">setPrintExpectScore(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set if the export score is to be printed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setPrintPeptides-java.lang.Boolean-">setPrintPeptides(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if a list of the peptides should be printed to the output folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setPrintProgressIndicatorSize-java.lang.Integer-">setPrintProgressIndicatorSize(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the progress indicator size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#setProbabilisticScore-int-double-">setProbabilisticScore(int, double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Sets the probabilistic score at a given site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#setProbabilisticScoreCalculation-boolean-">setProbabilisticScoreCalculation(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Sets whether a probabilistic modification score is required.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#setProbabilisticScoreNeutralLosses-boolean-">setProbabilisticScoreNeutralLosses(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Sets whether the neutral losses shall be taken into account for spectrum
 annotation when calculating the probabilistic score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#setProbabilisticScoreThreshold-double-">setProbabilisticScoreThreshold(double)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Sets the probabilistic score threshold.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setProbability-double-">setProbability(double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Set the probability.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/tools/ProcessingParameters.html#setProcessingType-com.compomics.util.parameters.tools.ProcessingParameters.ProcessingType-">setProcessingType(ProcessingParameters.ProcessingType)</a></span> - Method in class com.compomics.util.parameters.tools.<a href="com/compomics/util/parameters/tools/ProcessingParameters.html" title="class in com.compomics.util.parameters.tools">ProcessingParameters</a></dt>
<dd>
<div class="block">Sets the processing type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ModificationProfile.html#setProfile-double:A:A-">setProfile(double[][])</a></span> - Method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ModificationProfile.html" title="class in com.compomics.util.gui.protein">ModificationProfile</a></dt>
<dd>
<div class="block">Sets the profile.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html#setProfileMode-boolean-">setProfileMode(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ChromatogramPanel.html" title="class in com.compomics.util.gui.spectrum">ChromatogramPanel</a></dt>
<dd>
<div class="block">Set the drawing style to profile or centroid mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#setProfileMode-boolean-">setProfileMode(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Change the drawing type of the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html#setProgress-int-java.lang.String-">setProgress(int, String)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialog</a></dt>
<dd>
<div class="block">This method allows the caller to set the progress on the progressbar to the specified value,
 along with the specified message.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html#setProgress-int-">setProgress(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialog</a></dt>
<dd>
<div class="block">This method sets the progress to the specified value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html#setProperty-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.ModificationFeature.Property-">setProperty(ModificationFeature.Property)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ModificationFeature</a></dt>
<dd>
<div class="block">Sets the feature property.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html#setProperty-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.PeptideFeature.Property-">setProperty(PeptideFeature.Property)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideFeature</a></dt>
<dd>
<div class="block">Sets the feature property.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html#setProteinConfidenceMwPlots-java.lang.Double-">setProteinConfidenceMwPlots(Double)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/FractionParameters.html" title="class in com.compomics.util.parameters.identification.advanced">FractionParameters</a></dt>
<dd>
<div class="block">Sets the protein confidence for inclusion in MW plots.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setProteinEvidence-java.lang.Integer-">setProteinEvidence(Integer)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the protein evidence level.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#setProteinFilters-java.util.ArrayList-">setProteinFilters(ArrayList&lt;Filter&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Sets the list of protein quality filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#setProteinHideFilters-java.util.HashMap-">setProteinHideFilters(HashMap&lt;String, ProteinFilter&gt;)</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Sets the protein hide filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setProteinInferenceClass-int-">setProteinInferenceClass(int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Sets the protein group class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setProteinInferenceParameters-com.compomics.util.parameters.identification.advanced.ProteinInferenceParameters-">setProteinInferenceParameters(ProteinInferenceParameters)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the protein inference parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#setProteinKeys-long:A-">setProteinKeys(long[])</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Sets the ordered protein list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setProteinKeys-long:A-">setProteinKeys(long[])</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the list of ordered protein keys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#setProteinList-long:A-">setProteinList(long[])</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Sets the protein list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#setProteinListAfterHiding-long:A-">setProteinListAfterHiding(long[])</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Sets the protein list after all hiding filters have been used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#setProteinMapping-java.util.TreeMap-">setProteinMapping(TreeMap&lt;String, int[]&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Sets the proteins mapping as a map of 0 based indexes for every protein
 accession.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setProteinPtmComplexity-double-">setProteinPtmComplexity(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Set the proteinPtmComplexity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setProteinQuickAcetyl-boolean-">setProteinQuickAcetyl(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether the protein quick acetylation option should be triggered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#setProteinStarFilters-java.util.HashMap-">setProteinStarFilters(HashMap&lt;String, ProteinFilter&gt;)</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Sets the protein star filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/GeneMaps.html#setProteinToGoMap-java.util.HashMap-">setProteinToGoMap(HashMap&lt;String, HashSet&lt;String&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/GeneMaps.html" title="class in com.compomics.util.experiment.biology.genes">GeneMaps</a></dt>
<dd>
<div class="block">Sets the protein to GO terms accession map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setProteoWizardPath-java.lang.String-">setProteoWizardPath(String)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the path to ProteoWizard.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#setProtocol-int-">setProtocol(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Set the protocol ID.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#setPsmFilters-java.util.ArrayList-">setPsmFilters(ArrayList&lt;Filter&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Sets the list of PSM quality filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#setPsmHideFilters-java.util.HashMap-">setPsmHideFilters(HashMap&lt;String, PsmFilter&gt;)</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Sets the psm hide filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#setPsmList-long:A-">setPsmList(long[])</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Sets the PSM list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setPsmScoringParameters-com.compomics.util.parameters.identification.advanced.PsmScoringParameters-">setPsmScoringParameters(PsmScoringParameters)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the scoring preferences used when scoring PSMs.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterParameters.html#setPsmStarFilters-java.util.HashMap-">setPsmStarFilters(HashMap&lt;String, PsmFilter&gt;)</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterParameters.html" title="class in com.compomics.util.gui.filtering">FilterParameters</a></dt>
<dd>
<div class="block">Sets the psm star filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setPtmIndex-java.lang.String-int-">setPtmIndex(String, int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets the index for a given modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setPtmIndex-java.lang.String-int-">setPtmIndex(String, int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the index for a given modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setPtmIndexes-com.compomics.util.parameters.identification.search.ModificationParameters-">setPtmIndexes(ModificationParameters)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Set the PTM indexes of the modifications searched.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/ModificationProfile.html#setPtmName-java.lang.String-">setPtmName(String)</a></span> - Method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/ModificationProfile.html" title="class in com.compomics.util.gui.protein">ModificationProfile</a></dt>
<dd>
<div class="block">Sets the PTM name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/PrideObjectsFactory.html#setPtmToPrideMap-com.compomics.util.pride.PtmToPrideMap-">setPtmToPrideMap(PtmToPrideMap)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/PrideObjectsFactory.html" title="class in com.compomics.util.pride">PrideObjectsFactory</a></dt>
<dd>
<div class="block">Sets a new PTM to PRIDE map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setQcFilters-java.util.HashMap-">setQcFilters(HashMap&lt;String, Boolean&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Sets the qc filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setQcResult-java.lang.String-boolean-">setQcResult(String, boolean)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Sets whether the match passed a quality control check.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setQuickPyrolidone-boolean-">setQuickPyrolidone(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether the quick pyrolidone option should be triggered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#setRank-int-">setRank(int)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Set the rank of the PeptideAssumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html#setRankScore-double-">setRankScore(double)</a></span> - Method in class com.compomics.util.experiment.refinement_parameters.<a href="com/compomics/util/experiment/refinement_parameters/PepnovoAssumptionDetails.html" title="class in com.compomics.util.experiment.refinement_parameters">PepnovoAssumptionDetails</a></dt>
<dd>
<div class="block">Sets the PepNovo rank score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setRawHeader-java.lang.String-">setRawHeader(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the entire header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#setRawScore-double-">setRawScore(double)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Sets the raw score as provided by the identification algorithm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/ObjectsCache.html#setReadOnly-boolean-">setReadOnly(boolean)</a></span> - Method in class com.compomics.util.db.object.<a href="com/compomics/util/db/object/ObjectsCache.html" title="class in com.compomics.util.db.object">ObjectsCache</a></dt>
<dd>
<div class="block">Sets the cache in read only.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#setReadOnly-boolean-">setReadOnly(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Sets the cache in read only.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setReadTweets-java.util.ArrayList-">setReadTweets(ArrayList&lt;String&gt;)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the list of read tweets.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#setRef-double-">setRef(double)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Sets the value used as reference for the correction factors.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Reference.html#setReference-java.lang.String-">setReference(String)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Reference.html" title="class in com.compomics.util.pride.prideobjects">Reference</a></dt>
<dd>
<div class="block">Set the reference.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#setReferenceMass-java.lang.Double-">setReferenceMass(Double)</a></span> - Method in class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Sets the reference total mass to use for normalization.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html#setReferences-java.util.ArrayList-">setReferences(ArrayList&lt;Reference&gt;)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/ReferenceGroup.html" title="class in com.compomics.util.pride.prideobjects">ReferenceGroup</a></dt>
<dd>
<div class="block">Set the references.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setRefine-boolean-">setRefine(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether the second pass search should be triggered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setRefinePointMutations-boolean-">setRefinePointMutations(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether point mutations should be looked for during the refinement
 process.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setRefineSemi-boolean-">setRefineSemi(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether the semi enzymatic option of the second pass search should
 be triggered.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setRefineSnaps-boolean-">setRefineSnaps(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether snAPs should be used during the refinement process.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setRefineSpectrumSynthesis-boolean-">setRefineSpectrumSynthesis(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether the spectrum synthesis option should be used during the
 refinement process.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setRefineUnanticipatedCleavages-boolean-">setRefineUnanticipatedCleavages(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether the unanticipated cleavages option should be used during the
 refinement process.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setRefMass-double-">setRefMass(double)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the reference mass used to convert ppm to Da.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setRefMass-double-">setRefMass(double)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Sets the reference mass for the conversion of the fragment ion tolerance
 from ppm to Dalton.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setRelatedIons-boolean-">setRelatedIons(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Sets whether related ions should be annotated by default.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html#setRelationship-com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.AAPropertyRelationshipFeature.Relationship-">setRelationship(AAPropertyRelationshipFeature.Relationship)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature</a></dt>
<dd>
<div class="block">Sets the relationship between the features.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setRelimsPath-java.lang.String-">setRelimsPath(String)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the path to the Relims installation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setRemoveMethionine-java.lang.Boolean-">setRemoveMethionine(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Sets if n-term methionine is to be removed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setRemovePrecursor-java.lang.Integer-">setRemovePrecursor(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set if the precursor is to be removed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setRemovePrecursor-boolean-">setRemovePrecursor(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets whether the precursor shall be removed for OMSSA.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setRemovePrecursor-java.lang.Boolean-">setRemovePrecursor(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the precursor peak is to be removed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setRemovePrecursorTolerance-java.lang.Double-">setRemovePrecursorTolerance(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the precursor removal tolerance.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setRemovePrecursorTolerance-java.lang.Double-">setRemovePrecursorTolerance(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the tolerance with which to remove the precursor peak.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setRemoveTempFolders-java.lang.Boolean-">setRemoveTempFolders(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the output and index folders are to be removed when the search has
 completed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#setRemoveUnknownModifications-boolean-">setRemoveUnknownModifications(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Set whether unknown modifications shall be removed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setRenameXTandemFile-boolean-">setRenameXTandemFile(boolean)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set if the X! Tandem file should be renamed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html#setReporterIon-com.compomics.util.experiment.biology.ions.impl.ReporterIon-">setReporterIon(ReporterIon)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/Reagent.html" title="class in com.compomics.util.experiment.quantification.reporterion">Reagent</a></dt>
<dd>
<div class="block">Sets the reporter ion to look for in the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#setReporterIons-java.util.ArrayList-">setReporterIons(ArrayList&lt;ReporterIon&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Sets the reporter ions possibly encountered with this modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setReporterIons-boolean-">setReporterIons(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Sets whether reporter ions should be annotated by default.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setReporterPath-java.lang.String-">setReporterPath(String)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the path to the PeptideShaker installation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#setReportFailed-java.lang.String-">setReportFailed(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Sets the report when the filter is not passed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#setReportPassed-java.lang.String-">setReportPassed(String)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Sets the report when the filter is passed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setRequireVariableMods-java.lang.Boolean-">setRequireVariableMods(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set if at least one variable modification is required per peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html#setResolution-java.lang.String-">setResolution(String)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb">PdbParameter</a></dt>
<dd>
<div class="block">Set the resolution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setRest-java.lang.String-">setRest(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the rest of the header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#setRestrict-char:A-">setRestrict(char[])</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method allows the caller to specify the residues that restrict
 cleavage.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#setRestrict-java.lang.String-">setRestrict(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method allows the caller to specify the residues which restrict
 cleavage.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setRewindIons-java.util.ArrayList-">setRewindIons(ArrayList&lt;Integer&gt;)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Sets the rewind ions searched as list of integers as indexed in the
 FragmentIon class.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setRunCanceled--">setRunCanceled()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setRunCanceled--">setRunCanceled()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Set the process as canceled.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#setRunCanceled--">setRunCanceled()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#setRunCanceled--">setRunCanceled()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#setRunCanceled--">setRunCanceled()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Set the process as canceled.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setRunFinished--">setRunFinished()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setRunFinished--">setRunFinished()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Set the process as finished.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#setRunFinished--">setRunFinished()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#setRunFinished--">setRunFinished()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#setRunFinished--">setRunFinished()</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Set the process as finished.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setRunNotFinished--">setRunNotFinished()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Set the run to not finished.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html#setSampleIndex-java.lang.String-java.lang.Integer-">setSampleIndex(String, Integer)</a></span> - Method in class com.compomics.util.experiment.quantification.reporterion.<a href="com/compomics/util/experiment/quantification/reporterion/ReporterIonQuantification.html" title="class in com.compomics.util.experiment.quantification.reporterion">ReporterIonQuantification</a></dt>
<dd>
<div class="block">Set the sample index for the given sample.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setScalePrecursor-boolean-">setScalePrecursor(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets whether the precursor shall be scaled for OMSSA.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#setScalingFactor-double-">setScalingFactor(double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#setScanNumber-java.lang.String-">setScanNumber(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Setter for the scan number or range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#setScanStartTime-double-">setScanStartTime(double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Sets the scan start time.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/PrideXmlValidator.html#setSchema-java.net.URL-">setSchema(URL)</a></span> - Method in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/PrideXmlValidator.html" title="class in com.compomics.util.pride.validation">PrideXmlValidator</a></dt>
<dd>
<div class="block">Set the schema.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setScientificXAxis-boolean-">setScientificXAxis(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set if the x-axis tags are to be drawn using scientific annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setScientificXAxis-java.lang.String-">setScientificXAxis(String)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set if the x-axis tags are to be drawn using scientific annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setScientificYAxis-boolean-">setScientificYAxis(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set if the y-axis tags are to be drawn using scientific annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setScientificYAxis-java.lang.String-">setScientificYAxis(String)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set if the y-axis tags are to be drawn using scientific annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setScore-double-">setScore(double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Set the peptide score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#setScore-double-">setScore(double)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>
<div class="block">Sets the score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#setSearchDecoyDatabase-boolean-">setSearchDecoyDatabase(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>
<div class="block">Set whether MS-GF+ is to create and search a decoy database.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setSearchForwardFragmentFirst-boolean-">setSearchForwardFragmentFirst(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets whether forward ions (b1) should be searched first.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSearchGuiOutputParameters-com.compomics.util.parameters.searchgui.OutputParameters-">setSearchGuiOutputParameters(OutputParameters)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Sets how SearchGUI output files should be organized.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSearchGuiPath-java.lang.String-">setSearchGuiPath(String)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the path to the SearchGUI installation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setSearchParameters-com.compomics.util.parameters.identification.search.SearchParameters-">setSearchParameters(SearchParameters)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the parameters used for the spectrum matching.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setSearchPositiveIons-boolean-">setSearchPositiveIons(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets whether positive ions are searched.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setSearchRewindFragments-boolean-">setSearchRewindFragments(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets whether C-terminal fragments should be searched.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setSecondaryProgressCounter-int-">setSecondaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Sets the value of the progress bar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setSecondaryProgressCounter-int-">setSecondaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Sets the secondary progress bar to the given value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#setSecondaryProgressCounter-int-">setSecondaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#setSecondaryProgressCounter-int-">setSecondaryProgressCounter(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#setSecondaryProgressCounter-int-">setSecondaryProgressCounter(int)</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Sets the secondary progress counter to the given value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setSecondaryProgressCounterIndeterminate-boolean-">setSecondaryProgressCounterIndeterminate(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Makes the dialog indeterminate or not indeterminate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setSecondaryProgressCounterIndeterminate-boolean-">setSecondaryProgressCounterIndeterminate(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Sets the secondary progress bar to indeterminate or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#setSecondaryProgressCounterIndeterminate-boolean-">setSecondaryProgressCounterIndeterminate(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#setSecondaryProgressCounterIndeterminate-boolean-">setSecondaryProgressCounterIndeterminate(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#setSecondaryProgressCounterIndeterminate-boolean-">setSecondaryProgressCounterIndeterminate(boolean)</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Sets the secondary progress counter to indeterminate or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setSecondaryProgressText-java.lang.String-">setSecondaryProgressText(String)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setSecondaryProgressText-java.lang.String-">setSecondaryProgressText(String)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#setSecondaryProgressText-java.lang.String-">setSecondaryProgressText(String)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#setSecondaryProgressText-java.lang.String-">setSecondaryProgressText(String)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#setSecondaryProgressText-java.lang.String-">setSecondaryProgressText(String)</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Set the secondary progress counter text.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html#setSelectedBackgroundSpecies-java.lang.Integer-">setSelectedBackgroundSpecies(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html" title="class in com.compomics.util.parameters.identification.advanced">GeneParameters</a></dt>
<dd>
<div class="block">Sets the taxon of the species selected as background species.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#setSelectedCharges-java.util.ArrayList-">setSelectedCharges(ArrayList&lt;Integer&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Sets the charges selected for annotation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramDialog.html#setSelectedExportFolder-com.compomics.util.io.file.LastSelectedFolder-">setSelectedExportFolder(LastSelectedFolder)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramDialog.html" title="class in com.compomics.util.gui">VennDiagramDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#setSelectedExportFolder-com.compomics.util.io.file.LastSelectedFolder-">setSelectedExportFolder(LastSelectedFolder)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#setSelectedIonsMap-java.util.HashMap-">setSelectedIonsMap(HashMap&lt;Ion.IonType, HashSet&lt;Integer&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Sets the map of ions to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#setSelectedMethod-com.compomics.util.experiment.quantification.spectrumcounting.SpectrumCountingMethod-">setSelectedMethod(SpectrumCountingMethod)</a></span> - Method in class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Set the current spectrum counting method.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setSelectedOutput-java.lang.String-">setSelectedOutput(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the output type, omx, csv or pepXML.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setSelectedOutputFormat-com.compomics.util.parameters.identification.tool_specific.CometParameters.CometOutputFormat-">setSelectedOutputFormat(CometParameters.CometOutputFormat)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Sets the output format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#setSelectedPositions-java.lang.Integer:A-">setSelectedPositions(Integer[])</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Set the selected positions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#setSelectedProbabilisticScore-com.compomics.util.experiment.identification.modification.ModificationLocalizationScore-">setSelectedProbabilisticScore(ModificationLocalizationScore)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Sets the selected probabilistic score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#setSelfUpdating-boolean-">setSelfUpdating(boolean)</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Sets whether the table is in self update mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#setSequence-java.lang.String-">setSequence(String)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Sets the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#setSequence-java.lang.String-">setSequence(String)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Sets for the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html#setSequence-char:A-">setSequence(char[])</a></span> - Method in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/PeptideDraft.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">PeptideDraft</a></dt>
<dd>
<div class="block">Sets the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/protein/SequenceModificationPanel.html#setSequence-java.lang.String-boolean-">setSequence(String, boolean)</a></span> - Method in class com.compomics.util.gui.protein.<a href="com/compomics/util/gui/protein/SequenceModificationPanel.html" title="class in com.compomics.util.gui.protein">SequenceModificationPanel</a></dt>
<dd>
<div class="block">Set the Sequence for the SequenceFragmentationPanel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SequenceFragmentationPanel.html#setSequence-java.lang.String-boolean-">setSequence(String, boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SequenceFragmentationPanel.html" title="class in com.compomics.util.gui.spectrum">SequenceFragmentationPanel</a></dt>
<dd>
<div class="block">Set the Sequence for the SequenceFragmentationPanel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Sequence.html#setSequence-java.lang.String-">setSequence(String)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Sequence.html" title="interface in com.compomics.util.interfaces">Sequence</a></dt>
<dd>
<div class="block">This method will set the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequence.html#setSequence-com.compomics.util.nucleotide.NucleotideSequenceImpl-">setSequence(NucleotideSequenceImpl)</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide">NucleotideSequence</a></dt>
<dd>
<div class="block">This method allows the setting of a sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html#setSequence-java.lang.String-">setSequence(String)</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html" title="class in com.compomics.util.nucleotide">NucleotideSequenceImpl</a></dt>
<dd>
<div class="block">This method will set the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/AASequenceImpl.html#setSequence-java.lang.String-">setSequence(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/AASequenceImpl.html" title="class in com.compomics.util.protein">AASequenceImpl</a></dt>
<dd>
<div class="block">Simple setter for the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#setSequence-com.compomics.util.protein.AASequenceImpl-">setSequence(AASequenceImpl)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">Simple setter for the sequence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html#setSequenceMatchingParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">setSequenceMatchingParameters(SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ModificationLocalizationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ModificationLocalizationParameters</a></dt>
<dd>
<div class="block">Sets the sequence matching preferences to use when mapping modifications on amino
 acid sequences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setSequenceMatchingParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-">setSequenceMatchingParameters(SequenceMatchingParameters)</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Sets the sequence matching preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html#setSequenceMatchingType-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters.MatchingType-">setSequenceMatchingType(SequenceMatchingParameters.MatchingType)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.html" title="class in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters</a></dt>
<dd>
<div class="block">Sets the sequence matching type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#setSerializationFile-java.lang.String-">setSerializationFile(String)</a></span> - Static method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Sets the file where to save the factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#setSerializationFolder-java.lang.String-">setSerializationFolder(String)</a></span> - Static method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Sets the folder where the factory is saved.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#setShortName-java.lang.String-">setShortName(String)</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>
<div class="block">Sets the short modification name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setShowAllPeaks-boolean-">setShowAllPeaks(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Set if all peaks or just the annotated ones are to be shown.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setShowForwardIonDeNovoTags-boolean-">setShowForwardIonDeNovoTags(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Set if the automatic forward ion de novo tags are to be shown.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#setShowLegend-boolean-">setShowLegend(boolean)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Set if the legend is to be shown.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setShowPrecursorDetails-boolean-">setShowPrecursorDetails(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set whether the precursor details are to be shown.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setShowResolution-boolean-">setShowResolution(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set whether the resolution is to be shown.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setShowRewindIonDeNovoTags-boolean-">setShowRewindIonDeNovoTags(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Set if the automatic rewind ion de novo tags are to be shown.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#setSimplifyGroups-boolean-">setSimplifyGroups(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Sets whether the protein groups should be simplified based on the
 PeptideShaker confidence.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#setSimplifyGroupsEnzymaticity-boolean-">setSimplifyGroupsEnzymaticity(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Sets whether the protein groups should be simplified based on the peptide
 enzymaticity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#setSimplifyGroupsEvidence-boolean-">setSimplifyGroupsEvidence(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Sets whether the protein groups should be simplified based on the Uniprot
 evidence level.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#setSimplifyGroupsUncharacterized-boolean-">setSimplifyGroupsUncharacterized(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Sets whether the protein groups consisting of uncharacterized proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html#setSimplifyGroupsVariants-boolean-">setSimplifyGroupsVariants(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ProteinInferenceParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ProteinInferenceParameters</a></dt>
<dd>
<div class="block">Sets whether the protein groups should be simplified based on the peptide
 variant matching.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setSingleChargeWindow-int-">setSingleChargeWindow(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets the window size for singly charged ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html#setSite-int-">setSite(int)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/ModificationMatch.html" title="class in com.compomics.util.experiment.identification.matches">ModificationMatch</a></dt>
<dd>
<div class="block">Setter for the modification site, 1 is the first amino acid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#setSiteConfidence-int-int-">setSiteConfidence(int, int)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Sets the confidence level of a modification site.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setSkylinePath-java.lang.String-">setSkylinePath(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets the skyline path.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSortPsmsOnRt-java.lang.Boolean-">setSortPsmsOnRt(Boolean)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set if the PSMs are sorted on retention time, false sorts on PSM score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/Instrument.html#setSource-com.compomics.util.pride.CvTerm-">setSource(CvTerm)</a></span> - Method in class com.compomics.util.pride.prideobjects.<a href="com/compomics/util/pride/prideobjects/Instrument.html" title="class in com.compomics.util.pride.prideobjects">Instrument</a></dt>
<dd>
<div class="block">Set the instrument source.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#setSourceFile-java.lang.String-">setSourceFile(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">This method allows to set the source for the file that is to be
 transferred.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSparklineColor-java.awt.Color-">setSparklineColor(Color)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Setter for the sparkline color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSparklineColorDoubtful-java.awt.Color-">setSparklineColorDoubtful(Color)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Setter for the doubtful sparkline color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSparklineColorFalsePositives-java.awt.Color-">setSparklineColorFalsePositives(Color)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Setter for the false positives sparkline color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSparklineColorNonValidated-java.awt.Color-">setSparklineColorNonValidated(Color)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Setter for the non-validated sparkline color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSparklineColorPossible-java.awt.Color-">setSparklineColorPossible(Color)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Setter for the possible sparkline color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.html#setSpecificity-java.lang.String-com.compomics.util.parameters.identification.search.DigestionParameters.Specificity-">setSpecificity(String, DigestionParameters.Specificity)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.html" title="class in com.compomics.util.parameters.identification.search">DigestionParameters</a></dt>
<dd>
<div class="block">Sets the expected specificity of the enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#setSpectrum-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-double-">setSpectrum(Spectrum, double)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Sets a new spectrum to annotate.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSpectrumAnnotatedMirroredPeakColor-java.awt.Color-">setSpectrumAnnotatedMirroredPeakColor(Color)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the color to use for the annotated mirrored peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSpectrumAnnotatedPeakColor-java.awt.Color-">setSpectrumAnnotatedPeakColor(Color)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the color to use for the annotated peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSpectrumAnnotatedPeakWidth-float-">setSpectrumAnnotatedPeakWidth(float)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the width of the annotated peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSpectrumBackgroundPeakColor-java.awt.Color-">setSpectrumBackgroundPeakColor(Color)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the color to use for the background peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setSpectrumBackgroundPeakWidth-float-">setSpectrumBackgroundPeakWidth(float)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the width of the background peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setSpectrumCharges-java.lang.String-">setSpectrumCharges(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the spectrum charges to search for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#setSpectrumCountingPreferences-com.compomics.util.parameters.quantification.spectrum_counting.SpectrumCountingParameters-">setSpectrumCountingPreferences(SpectrumCountingParameters)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Sets the spectrum couting preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#setSpectrumFile-com.compomics.util.interfaces.SpectrumFile-">setSpectrumFile(SpectrumFile)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This method initializes a SpectrumPanel based on the spectrum information
 in the specified SpectrumFile.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PklFile.html#setSpectrumFileId-java.lang.String-">setSpectrumFileId(String)</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PklFile.html" title="class in com.compomics.util.io">PklFile</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#setSpectrumKey-java.lang.String-">setSpectrumKey(String)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Sets the match key.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html#setSpectrumMatchesKeys-long:A-">setSpectrumMatchesKeys(long[])</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/PeptideMatch.html" title="class in com.compomics.util.experiment.identification.matches">PeptideMatch</a></dt>
<dd>
<div class="block">Sets the spectrum matches keys.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#setSpectrumNumber-int-">setSpectrumNumber(int)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Sets the spectrum number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#setSpectrumPeakColor-java.awt.Color-">setSpectrumPeakColor(Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Set the default spectrum peak color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#setSpectrumProfileModeLineColor-java.awt.Color-">setSpectrumProfileModeLineColor(Color)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">Set the default spectrum profile mode color.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#setSpectrumTitle-java.lang.String-">setSpectrumTitle(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Set the spectrum title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setSqtOutput-java.lang.Boolean-">setSqtOutput(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the sqt output is to be created.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setStarred-boolean-">setStarred(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Star/Unstar a match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#setStart-double-">setStart(double)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Set the start value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#setStartBlock-int-">setStartBlock(int)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Set the block start.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html#setStartProtein-int-">setStartProtein(int)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbBlock.html" title="class in com.compomics.util.pdbfinder.pdb">PdbBlock</a></dt>
<dd>
<div class="block">Set the protein start.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setStoreSpectraFileName-java.lang.String-">setStoreSpectraFileName(String)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the name of the file where the binary spectra are to be stored.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setStpBias-boolean-">setStpBias(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether the stP bias should be triggered</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setString-java.lang.String-">setString(String)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Sets the string to display in the progress bar.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/ProjectParameters.html#setStringParameter-java.lang.String-java.lang.String-">setStringParameter(String, String)</a></span> - Method in class com.compomics.util.experiment.<a href="com/compomics/util/experiment/ProjectParameters.html" title="class in com.compomics.util.experiment">ProjectParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setSubscriptAnnotationNumbers-boolean-">setSubscriptAnnotationNumbers(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set if the numbers in the peak annotations are to be subscripted.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html#setTag-com.compomics.util.experiment.identification.amino_acid_tags.Tag-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-">setTag(Tag, ModificationParameters, SequenceMatchingParameters, int)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">TagSpectrumAnnotator</a></dt>
<dd>
<div class="block">Sets a new tag to match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html#setTag-com.compomics.util.experiment.identification.amino_acid_tags.Tag-">setTag(Tag)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">TagAssumption</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#setTagAssumptionMap-java.util.HashMap-">setTagAssumptionMap(HashMap&lt;Integer, TreeMap&lt;Double, ArrayList&lt;TagAssumption&gt;&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Sets the tag assumption map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setTagLength-int-">setTagLength(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Sets the tag length.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#setTarget-java.lang.Integer-">setTarget(Integer)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Sets the index of the amino acid of interest in the pattern.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#setTargetDecoy-boolean-">setTargetDecoy(boolean)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Sets whether the database is concatenated target decoy.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setTargetDecoyFileNameSuffix-java.lang.String-">setTargetDecoyFileNameSuffix(String)</a></span> - Method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Set the target-decoy file name suffix.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#setTargeted-int-java.util.ArrayList-">setTargeted(int, ArrayList&lt;Character&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Sets the amino acids targeted at a given index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#setTaxonomy-java.lang.String-">setTaxonomy(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">Sets the taxonomy.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#setTerminalMassShifts-double-double-">setTerminalMassShifts(double, double)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Sets the terminal m/z shifts.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setTextOutput-java.lang.Boolean-">setTextOutput(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the text output is to be created.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setTheoreticalFragmentIonsSumOnly-java.lang.Boolean-">setTheoreticalFragmentIonsSumOnly(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set if the theoretical fragment ions are calculated on the sum in the bin
 alone.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#setTheoreticMass-double-">setTheoreticMass(double)</a></span> - Method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Sets a new theoretic mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setTicCutoffPercentage-double-">setTicCutoffPercentage(double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Set the TIC cutoff in percent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setTicCutoffPercentage-java.lang.Double-">setTicCutoffPercentage(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Set the TicCutoffPercentage.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setTiesResolution-com.compomics.util.experiment.identification.spectrum_annotation.SpectrumAnnotator.TiesResolution-">setTiesResolution(SpectrumAnnotator.TiesResolution)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Sets the ties resolution method to use when multiple peaks can be
 assigned to an ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#setTitle-java.lang.String-">setTitle(String)</a></span> - Method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Sets the title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html#setTitle-java.lang.String-">setTitle(String)</a></span> - Method in class com.compomics.util.pdbfinder.pdb.<a href="com/compomics/util/pdbfinder/pdb/PdbParameter.html" title="class in com.compomics.util.pdbfinder.pdb">PdbParameter</a></dt>
<dd>
<div class="block">Set the title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#setTitle-java.lang.String-">setTitle(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method allows the caller to change the title (name) of the Enzyme.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/ToolTipComboBoxRenderer.html#setToolTips-java.util.Vector-">setToolTips(Vector)</a></span> - Method in class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/ToolTipComboBoxRenderer.html" title="class in com.compomics.util.gui.renderers">ToolTipComboBoxRenderer</a></dt>
<dd>
<div class="block">Set the tooltips.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setTopPeaks-int-">setTopPeaks(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets the top peaks number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#setTopPeaksWindow-int-">setTopPeaksWindow(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>
<div class="block">Sets the top peaks window size.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#setTotalIntensity-java.lang.Double-">setTotalIntensity(Double)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setTotalPeptidesPerFraction-java.util.HashMap-">setTotalPeptidesPerFraction(HashMap&lt;String, Integer&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Set the total number of peptides per fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setTotalSpectrumCounting-java.lang.Double-">setTotalSpectrumCounting(Double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the total spectrum counting.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setTotalSpectrumCountingMass-double-">setTotalSpectrumCountingMass(double)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the total spectrum counting value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#setType-com.compomics.util.experiment.identification.filtering.MatchFilter.FilterType-">setType(MatchFilter.FilterType)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Sets the type of the filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/parameters/DummyParameters.html#setType--">setType()</a></span> - Method in class com.compomics.util.experiment.io.parameters.<a href="com/compomics/util/experiment/io/parameters/DummyParameters.html" title="class in com.compomics.util.experiment.io.parameters">DummyParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/parameters/MarshallableParameter.html#setType--">setType()</a></span> - Method in interface com.compomics.util.experiment.io.parameters.<a href="com/compomics/util/experiment/io/parameters/MarshallableParameter.html" title="interface in com.compomics.util.experiment.io.parameters">MarshallableParameter</a></dt>
<dd>
<div class="block">Sets the parameters type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/patient/PatientInformation.html#setType-java.lang.String-">setType(String)</a></span> - Method in class com.compomics.util.experiment.patient.<a href="com/compomics/util/experiment/patient/PatientInformation.html" title="class in com.compomics.util.experiment.patient">PatientInformation</a></dt>
<dd>
<div class="block">Sets the type of information.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.html#setType-com.compomics.util.messages.FeedBack.FeedBackType-">setType(FeedBack.FeedBackType)</a></span> - Method in class com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.html" title="class in com.compomics.util.messages">FeedBack</a></dt>
<dd>
<div class="block">Sets the feedback type.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#setType--">setType()</a></span> - Method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#setType--">setType()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/AlternateRowColoursJTable.html#setUnevenRowColor-java.awt.Color-">setUnevenRowColor(Color)</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/AlternateRowColoursJTable.html" title="class in com.compomics.util">AlternateRowColoursJTable</a></dt>
<dd>
<div class="block">Sets the colour of the uneven numbered rows; by default, the default JTable background color will be used..</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#setUnit-com.compomics.util.experiment.units.UnitOfMeasurement-">setUnit(UnitOfMeasurement)</a></span> - Method in class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Sets the unit used for normalization.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setUnstoppable-boolean-">setUnstoppable(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Set this to true of the process the progress bar is used for is not
 possible to stop, or not possible to stop nicely.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/gui/PathParametersDialog.html#setUpGUI-java.awt.Frame-java.lang.String-">setUpGUI(Frame, String)</a></span> - Method in class com.compomics.software.settings.gui.<a href="com/compomics/software/settings/gui/PathParametersDialog.html" title="class in com.compomics.software.settings.gui">PathParametersDialog</a></dt>
<dd>
<div class="block">Sets up the GUI components.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterDialog.html#setUpGUI--">setUpGUI()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterDialog.html" title="class in com.compomics.util.gui.filtering">FilterDialog</a></dt>
<dd>
<div class="block">Sets up the GUI components.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setUpperClearMzRange-java.lang.Double-">setUpperClearMzRange(Double)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block">Set the upper mass value for the clear mz range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#setUpperPrecursorMass-int-">setUpperPrecursorMass(int)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>
<div class="block">Set the upper precursor mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterDialog.html#setUpTable--">setUpTable()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterDialog.html" title="class in com.compomics.util.gui.filtering">FilterDialog</a></dt>
<dd>
<div class="block">Updates the table content according to the selected filter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/ListChooser.html#setUpTable--">setUpTable()</a></span> - Method in class com.compomics.util.gui.utils.user_choice.<a href="com/compomics/util/gui/utils/user_choice/ListChooser.html" title="class in com.compomics.util.gui.utils.user_choice">ListChooser</a></dt>
<dd>
<div class="block">Updates the table content.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/personalization/ExperimentObject.html#setUrParams-java.util.HashMap-">setUrParams(HashMap&lt;Long, UrParameter&gt;)</a></span> - Method in class com.compomics.util.experiment.personalization.<a href="com/compomics/util/experiment/personalization/ExperimentObject.html" title="class in com.compomics.util.experiment.personalization">ExperimentObject</a></dt>
<dd>
<div class="block">Sets the user parameters map.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#setUseChargeStateFromMS-boolean-">setUseChargeStateFromMS(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>
<div class="block">Sets whether the charge state from the spectrum should be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#setUseCorrelationCorrectionScore-boolean-">setUseCorrelationCorrectionScore(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>
<div class="block">Sets whether the correlation correction score should be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setUseFlankingPeaks-java.lang.Boolean-">setUseFlankingPeaks(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if flanking peaks are to be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html#setUseGeneMapping-java.lang.Boolean-">setUseGeneMapping(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/GeneParameters.html" title="class in com.compomics.util.parameters.identification.advanced">GeneParameters</a></dt>
<dd>
<div class="block">Sets whether gene mappings should be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setUseMassDeltaCombinations-boolean-">setUseMassDeltaCombinations(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">If true, pairs of delta mass annotations are used when doing de novo
 sequencing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setUseNeutralLossPeaks-java.lang.Boolean-">setUseNeutralLossPeaks(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set if the neutral loss peaks are to be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#setUseNoiseSuppression-boolean-">setUseNoiseSuppression(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Sets whether noise suppression should be used when importing spectra.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#setUserAdvocates-java.util.HashMap-">setUserAdvocates(HashMap&lt;Integer, Advocate&gt;)</a></span> - Static method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Returns the map of user advocates imported.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#setUserParametersFolder-java.lang.String-">setUserParametersFolder(String)</a></span> - Static method in class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Sets the user preferences file to be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#setUseSmartPlusThreeModel-boolean-">setUseSmartPlusThreeModel(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>
<div class="block">Sets if the UseSmartPlusThreeModel is to be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#setUseSparseMatrix-java.lang.Boolean-">setUseSparseMatrix(Boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">from Comet release 2015.02 all searches use this internal
 data representation by default</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html#setUseSpecificCount-boolean-">setUseSpecificCount(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/PeptideVariantsParameters.html" title="class in com.compomics.util.parameters.identification.advanced">PeptideVariantsParameters</a></dt>
<dd>
<div class="block">Sets whether the specific variant count limitations should be used.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportScheme.html#setValidatedOnly-java.lang.Boolean-">setValidatedOnly(Boolean)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportScheme.html" title="class in com.compomics.util.io.export">ExportScheme</a></dt>
<dd>
<div class="block">Sets whether only validated results should be exported.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setValidatedPeptidesPerFraction-java.util.HashMap-">setValidatedPeptidesPerFraction(HashMap&lt;String, Integer&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Get the number of validated peptides in the given fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html#setValidatedProteinList-long:A-">setValidatedProteinList(long[])</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesCache</a></dt>
<dd>
<div class="block">Sets the list of validated proteins.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#setValidatedSpectraPepFraction-java.util.HashMap-">setValidatedSpectraPepFraction(HashMap&lt;String, Integer&gt;)</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Get the number of validated spectra in the given fraction.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html#setValidatedTargetProteinKeys-long:A-">setValidatedTargetProteinKeys(long[])</a></span> - Method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/Metrics.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">Metrics</a></dt>
<dd>
<div class="block">Sets the keys of the validated target protein matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html#setValidationQCParameters-com.compomics.util.parameters.identification.advanced.ValidationQcParameters-">setValidationQCParameters(ValidationQcParameters)</a></span> - Method in class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/IdMatchValidationParameters.html" title="class in com.compomics.util.parameters.identification.advanced">IdMatchValidationParameters</a></dt>
<dd>
<div class="block">Sets the validation QC preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setValue-int-">setValue(int)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>
<div class="block">Sets the progress bar value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/CvTerm.html#setValue-java.lang.String-">setValue(String)</a></span> - Method in class com.compomics.util.pride.<a href="com/compomics/util/pride/CvTerm.html" title="class in com.compomics.util.pride">CvTerm</a></dt>
<dd>
<div class="block">Sets the value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#setValueAt-int-int-java.lang.Double-">setValueAt(int, int, Double)</a></span> - Method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableMap.html#setValueAt-java.lang.Object-int-int-">setValueAt(Object, int, int)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun">TableMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#setValueAt-java.lang.Object-int-int-">setValueAt(Object, int, int)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#setValueForItem-java.lang.String-java.lang.Object-">setValueForItem(String, Object)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Sets the value for a given item.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#setVariableModifications-com.compomics.util.experiment.identification.matches.ModificationMatch:A-">setVariableModifications(ModificationMatch[])</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>
<div class="block">Sets the variable modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#setVariableModifications-com.compomics.util.experiment.identification.matches.ModificationMatch:A-">setVariableModifications(ModificationMatch[])</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Sets the variable modifications.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/proteins/Peptide.html#setVariantMatches-java.util.HashMap-">setVariantMatches(HashMap&lt;String, HashMap&lt;Integer, PeptideVariantMatches&gt;&gt;)</a></span> - Method in class com.compomics.util.experiment.biology.proteins.<a href="com/compomics/util/experiment/biology/proteins/Peptide.html" title="class in com.compomics.util.experiment.biology.proteins">Peptide</a></dt>
<dd>
<div class="block">Sets the sequence variant matches of this peptide.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#setVerbosity-java.lang.Integer-">setVerbosity(Integer)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Set the verbosity type of progress output.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html#setVersion-java.lang.String-">setVersion(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaParameters.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaParameters</a></dt>
<dd>
<div class="block">Sets the database version.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setVisible-boolean-">setVisible(boolean)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VisibleTableColumnsDialogParent.html#setVisibleColumns-java.util.HashMap-">setVisibleColumns(HashMap&lt;Integer, Boolean&gt;)</a></span> - Method in interface com.compomics.util.gui.<a href="com/compomics/util/gui/VisibleTableColumnsDialogParent.html" title="interface in com.compomics.util.gui">VisibleTableColumnsDialogParent</a></dt>
<dd>
<div class="block">Sets the list of visible columns.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#setVisibleColumns-java.util.HashMap-">setVisibleColumns(HashMap&lt;Integer, Boolean&gt;)</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html#setWaitingText-java.lang.String-">setWaitingText(String)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/ProgressDialogX.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">ProgressDialogX</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#setWaitingText-java.lang.String-">setWaitingText(String)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#setWaitingText-java.lang.String-">setWaitingText(String)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#setWaitingText-java.lang.String-">setWaitingText(String)</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#setWaitingText-java.lang.String-">setWaitingText(String)</a></span> - Method in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Sets the text describing what is currently being waited for.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#setWorkbookStyle-com.compomics.util.io.export.WorkbookStyle-">setWorkbookStyle(WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>
<div class="block">Sets the workbook style.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/XYZDataPoint.html#setX-double-">setX(double)</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/XYZDataPoint.html" title="class in com.compomics.util">XYZDataPoint</a></dt>
<dd>
<div class="block">Sets the x value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setXAxisStartAtZero-boolean-">setXAxisStartAtZero(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">This method sets the start value of the x-axis to zero.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/XYZDataPoint.html#setY-double-">setY(double)</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/XYZDataPoint.html" title="class in com.compomics.util">XYZDataPoint</a></dt>
<dd>
<div class="block">Sets the y value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#setYAxisZoomExcludesBackgroundPeaks-boolean-">setYAxisZoomExcludesBackgroundPeaks(boolean)</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Set if the automatic y-axis zoom only considers the annotated peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setYAxisZoomExcludesBackgroundPeaks-boolean-">setYAxisZoomExcludesBackgroundPeaks(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set if the automatic y-axis zoom only considers the annotated peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#setYDataIsPositive-boolean-">setYDataIsPositive(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Set to true of all y data values can be assumed to be positive.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/XYZDataPoint.html#setZ-double-">setZ(double)</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/XYZDataPoint.html" title="class in com.compomics.util">XYZDataPoint</a></dt>
<dd>
<div class="block">Sets the z value.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#showAllPeaks--">showAllPeaks()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">If true, all peaks are shown, false displays the annotated peaks, and the
 non-annotated in the background.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#showAllPeaks">showAllPeaks</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">If false, only the annotated peaks will be shown.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#showAnnotatedPeaksOnly-boolean-">showAnnotatedPeaksOnly(boolean)</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">If true only the annotated peaks will be drawn.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#showFileName">showFileName</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The boolean is set to 'true' if the file name is to be shown in the
 panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#showForwardIonDeNovoTags--">showForwardIonDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns true if the automatic forward ion de novo tags are to be shown.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#showLegend--">showLegend()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Show the legend or not.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#showPrecursorDetails">showPrecursorDetails</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The boolean is set to 'true' if the precursor details is to be shown in
 the panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#showPrecursorDetails--">showPrecursorDetails()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns true of the precursor details are to be shown.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#showResolution">showResolution</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The boolean is set to 'true' if the resolution is to be shown in the
 panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#showResolution--">showResolution()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns true if the resolution is to be shown.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#showRewindIonDeNovoTags--">showRewindIonDeNovoTags()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns true if the automatic rewind ion de novo tags are to be shown.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#shuttlesort-int:A-int:A-int-int-">shuttlesort(int[], int[], int, int)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#signalStop--">signalStop()</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>
<div class="block">This method can be used to signal the monitor to halt its monitoring.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#SIMPLE_DATA_FORMAT">SIMPLE_DATA_FORMAT</a></span> - Static variable in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">Convenience date format.</div>
</dd>
<dt><a href="com/compomics/util/math/statistics/linear_regression/regressions/SimpleLinearRegression.html" title="class in com.compomics.util.math.statistics.linear_regression.regressions"><span class="typeNameLink">SimpleLinearRegression</span></a> - Class in <a href="com/compomics/util/math/statistics/linear_regression/regressions/package-summary.html">com.compomics.util.math.statistics.linear_regression.regressions</a></dt>
<dd>
<div class="block">Performs a simple linear regression.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/statistics/linear_regression/regressions/SimpleLinearRegression.html#SimpleLinearRegression--">SimpleLinearRegression()</a></span> - Constructor for class com.compomics.util.math.statistics.linear_regression.regressions.<a href="com/compomics/util/math/statistics/linear_regression/regressions/SimpleLinearRegression.html" title="class in com.compomics.util.math.statistics.linear_regression.regressions">SimpleLinearRegression</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/SimpleNoiseDistribution.html" title="class in com.compomics.util.experiment.mass_spectrometry"><span class="typeNameLink">SimpleNoiseDistribution</span></a> - Class in <a href="com/compomics/util/experiment/mass_spectrometry/package-summary.html">com.compomics.util.experiment.mass_spectrometry</a></dt>
<dd>
<div class="block">This class makes a simple modeling of the noise using a normal distribution
 on the log of the peak intensities and returns the associated binned
 cumulative probability for a given intensity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SimpleNoiseDistribution.html#SimpleNoiseDistribution-java.util.HashMap-">SimpleNoiseDistribution(HashMap&lt;Double, Peak&gt;)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SimpleNoiseDistribution.html" title="class in com.compomics.util.experiment.mass_spectrometry">SimpleNoiseDistribution</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators"><span class="typeNameLink">SimplePeptideAnnotator</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators</a></dt>
<dd>
<div class="block">A simple annotator for peptides.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.html#SimplePeptideAnnotator-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-com.compomics.util.experiment.identification.spectrum_annotation.AnnotationParameters-">SimplePeptideAnnotator(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, int, AnnotationParameters)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">SimplePeptideAnnotator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.html#SimplePeptideAnnotator-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.search.ModificationParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-int-boolean-boolean-boolean-boolean-boolean-boolean-boolean-boolean-boolean-boolean-boolean-boolean-com.compomics.util.experiment.biology.ions.impl.ReporterIon:A-">SimplePeptideAnnotator(Peptide, ModificationParameters, SequenceProvider, SequenceMatchingParameters, int, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean, ReporterIon[])</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">SimplePeptideAnnotator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.IonSeries.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators"><span class="typeNameLink">SimplePeptideAnnotator.IonSeries</span></a> - Enum in <a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators</a></dt>
<dd>
<div class="block">The type of ion series to annotate.</div>
</dd>
<dt><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml"><span class="typeNameLink">SimpleXmlWriter</span></a> - Class in <a href="com/compomics/util/io/export/xml/package-summary.html">com.compomics.util.io.export.xml</a></dt>
<dd>
<div class="block">Simple writer for xml files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#SimpleXmlWriter-java.io.BufferedWriter-">SimpleXmlWriter(BufferedWriter)</a></span> - Constructor for class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#SINGLE_AA_SUFFIX">SINGLE_AA_SUFFIX</a></span> - Static variable in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Suffix for the modification clone targeting a single amino acid instead
 of a pattern.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/SingleAAPropertyFeature.html" title="interface in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration"><span class="typeNameLink">SingleAAPropertyFeature</span></a> - Interface in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration</a></dt>
<dd>
<div class="block">Interface for the ms2pip features having a single amino acid property.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#singleBaseSubstitution">singleBaseSubstitution</a></span> - Static variable in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Substitution matrix allowing for a single base substitution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#singleLetterCode">singleLetterCode</a></span> - Variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Single letter code of the amino acid.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html" title="class in com.compomics.util.experiment.identification.protein_sequences"><span class="typeNameLink">SingleProteinSequenceProvider</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/package-summary.html">com.compomics.util.experiment.identification.protein_sequences</a></dt>
<dd>
<div class="block">Sequence provider for a single protein.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html#SingleProteinSequenceProvider-com.compomics.util.experiment.biology.proteins.Protein-">SingleProteinSequenceProvider(Protein)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.<a href="com/compomics/util/experiment/identification/protein_sequences/SingleProteinSequenceProvider.html" title="class in com.compomics.util.experiment.identification.protein_sequences">SingleProteinSequenceProvider</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#singleRangeQuery-int-int-int-">singleRangeQuery(int, int, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Returns a new left/right index range for a given character
 recursively.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#site-java.lang.String-">site(String)</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Site Command</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#size--">size()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>
<div class="block">Returns the number of atoms in this atom chain.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableFileInputStream.html#skip-long-">skip(long)</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableFileInputStream.html" title="class in com.compomics.util.io.file">MonitorableFileInputStream</a></dt>
<dd>
<div class="block">Skips over and discards <code>n</code> bytes of data from the
 input stream.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/MonitorableInputStream.html#skip-long-">skip(long)</a></span> - Method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/MonitorableInputStream.html" title="class in com.compomics.util.io.file">MonitorableInputStream</a></dt>
<dd>
<div class="block">Skips over and discards <code>n</code> bytes of data from the
 input stream.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/SnrScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores"><span class="typeNameLink">SnrScore</span></a> - Class in <a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/package-summary.html">com.compomics.util.experiment.identification.psm_scoring.psm_scores</a></dt>
<dd>
<div class="block">This score uses the intensity distribution of the peaks to evaluate an SNR
 score.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/SnrScore.html#SnrScore-com.compomics.util.experiment.identification.protein_sequences.AaOccurrence-">SnrScore(AaOccurrence)</a></span> - Constructor for class com.compomics.util.experiment.identification.psm_scoring.psm_scores.<a href="com/compomics/util/experiment/identification/psm_scoring/psm_scores/SnrScore.html" title="class in com.compomics.util.experiment.identification.psm_scoring.psm_scores">SnrScore</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/Sodium.html" title="class in com.compomics.util.experiment.biology.atoms.impl"><span class="typeNameLink">Sodium</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a></dt>
<dd>
<div class="block">The sodium atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/impl/Sodium.html#Sodium--">Sodium()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.impl.<a href="com/compomics/util/experiment/biology/atoms/impl/Sodium.html" title="class in com.compomics.util.experiment.biology.atoms.impl">Sodium</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#sonar">sonar</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Sonar search engine, integrated in radars.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html#sort-int-com.compomics.util.gui.waiting.waitinghandlers.ProgressDialogX-">sort(int, ProgressDialogX)</a></span> - Method in class com.compomics.util.gui.tablemodels.<a href="com/compomics/util/gui/tablemodels/SelfUpdatingTableModel.html" title="class in com.compomics.util.gui.tablemodels">SelfUpdatingTableModel</a></dt>
<dd>
<div class="block">Sorts the table according to a given column using the built in sorter.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#sort-java.lang.Object-">sort(Object)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#sortByColumn-int-">sortByColumn(int)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#sortByColumn-int-boolean-">sortByColumn(int, boolean)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/genes/SpeciesDialog.html" title="class in com.compomics.util.gui.genes"><span class="typeNameLink">SpeciesDialog</span></a> - Class in <a href="com/compomics/util/gui/genes/package-summary.html">com.compomics.util.gui.genes</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">use the GenePreferencesDialog instead</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/genes/SpeciesDialog.html#SpeciesDialog-java.awt.Frame-boolean-java.awt.Image-java.awt.Image-java.lang.Integer-">SpeciesDialog(Frame, boolean, Image, Image, Integer)</a></span> - Constructor for class com.compomics.util.gui.genes.<a href="com/compomics/util/gui/genes/SpeciesDialog.html" title="class in com.compomics.util.gui.genes">SpeciesDialog</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">Creates a new SpeciesDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/genes/SpeciesDialog.html#SpeciesDialog-javax.swing.JDialog-javax.swing.JFrame-boolean-java.awt.Image-java.awt.Image-java.lang.Integer-">SpeciesDialog(JDialog, JFrame, boolean, Image, Image, Integer)</a></span> - Constructor for class com.compomics.util.gui.genes.<a href="com/compomics/util/gui/genes/SpeciesDialog.html" title="class in com.compomics.util.gui.genes">SpeciesDialog</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">Creates a new SpeciesDialog.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy"><span class="typeNameLink">SpeciesFactory</span></a> - Class in <a href="com/compomics/util/experiment/biology/taxonomy/package-summary.html">com.compomics.util.experiment.biology.taxonomy</a></dt>
<dd>
<div class="block">Class related to the handling of species.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html#speciesOccurrence">speciesOccurrence</a></span> - Variable in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/FastaSummary.html" title="class in com.compomics.util.experiment.io.biology.protein">FastaSummary</a></dt>
<dd>
<div class="block">The species occurrence in the FASTA file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#speciesOption">speciesOption</a></span> - Static variable in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">The command line argument for the species.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#speciesTypeOption">speciesTypeOption</a></span> - Static variable in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">The command line argument for the species type.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation"><span class="typeNameLink">SpecificAnnotationParameters</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation</a></dt>
<dd>
<div class="block">The spectrum annotation preferences specific to a spectrum and an
 identification assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html#SpecificAnnotationParameters-java.lang.String-com.compomics.util.experiment.identification.SpectrumIdentificationAssumption-">SpecificAnnotationParameters(String, SpectrumIdentificationAssumption)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpecificAnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpecificAnnotationParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#specificAnnotationSettings">specificAnnotationSettings</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">If provided, the annotator will only look for the ions included in the
 specific annotation settings.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/SpecificSingleEnzymeCombinationIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators"><span class="typeNameLink">SpecificSingleEnzymeCombinationIterator</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion.iterators</a></dt>
<dd>
<div class="block">Iterator for enzymatic digestion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/SpecificSingleEnzymeCombinationIterator.html#SpecificSingleEnzymeCombinationIterator-com.compomics.util.experiment.identification.protein_sequences.digestion.ProteinIteratorUtils-java.lang.String-com.compomics.util.experiment.biology.enzymes.Enzyme-int-double-double-">SpecificSingleEnzymeCombinationIterator(ProteinIteratorUtils, String, Enzyme, int, double, double)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.iterators.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/SpecificSingleEnzymeCombinationIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">SpecificSingleEnzymeCombinationIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/SpecificSingleEnzymeIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators"><span class="typeNameLink">SpecificSingleEnzymeIterator</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion.iterators</a></dt>
<dd>
<div class="block">Iterator for enzymatic digestion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/SpecificSingleEnzymeIterator.html#SpecificSingleEnzymeIterator-com.compomics.util.experiment.identification.protein_sequences.digestion.ProteinIteratorUtils-java.lang.String-com.compomics.util.experiment.biology.enzymes.Enzyme-int-double-double-">SpecificSingleEnzymeIterator(ProteinIteratorUtils, String, Enzyme, int, double, double)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.iterators.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/SpecificSingleEnzymeIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">SpecificSingleEnzymeIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#spectraST">spectraST</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The SpectraST spectral library search engine (TPP).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#spectraTag">spectraTag</a></span> - Static variable in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>
<div class="block">Tag for the spectrum file.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra"><span class="typeNameLink">Spectrum</span></a> - Class in <a href="com/compomics/util/experiment/mass_spectrometry/spectra/package-summary.html">com.compomics.util.experiment.mass_spectrometry.spectra</a></dt>
<dd>
<div class="block">This class models a spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#Spectrum-int-com.compomics.util.experiment.mass_spectrometry.spectra.Precursor-java.lang.String-java.lang.String-">Spectrum(int, Precursor, String, String)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Constructor for the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#Spectrum-int-com.compomics.util.experiment.mass_spectrometry.spectra.Precursor-java.lang.String-java.util.HashMap-java.lang.String-">Spectrum(int, Precursor, String, HashMap&lt;Double, Peak&gt;, String)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Constructor for the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#Spectrum-int-com.compomics.util.experiment.mass_spectrometry.spectra.Precursor-java.lang.String-java.util.HashMap-java.lang.String-double-">Spectrum(int, Precursor, String, HashMap&lt;Double, Peak&gt;, String, double)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Constructor for the spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#SPECTRUM_KEY_SPLITTER">SPECTRUM_KEY_SPLITTER</a></span> - Static variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">The splitter in the key between spectrumFile and spectrumTitle.</div>
</dd>
<dt><a href="com/compomics/util/gui/interfaces/SpectrumAnnotation.html" title="interface in com.compomics.util.gui.interfaces"><span class="typeNameLink">SpectrumAnnotation</span></a> - Interface in <a href="com/compomics/util/gui/interfaces/package-summary.html">com.compomics.util.gui.interfaces</a></dt>
<dd>
<div class="block">This interface describes the behaviour for a spectrum annotation.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation"><span class="typeNameLink">SpectrumAnnotator</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation</a></dt>
<dd>
<div class="block">The spectrum annotator annotates peaks in a spectrum.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#SpectrumAnnotator--">SpectrumAnnotator()</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.TiesResolution.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation"><span class="typeNameLink">SpectrumAnnotator.TiesResolution</span></a> - Enum in <a href="com/compomics/util/experiment/identification/spectrum_annotation/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation</a></dt>
<dd>
<div class="block">Enum of the possibilities for ties resolution when multiple peaks can be
 annotated.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#spectrumCountingInCache-long-">spectrumCountingInCache(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Indicates whether the default spectrum counting value is in cache for a
 protein match.</div>
</dd>
<dt><a href="com/compomics/util/experiment/quantification/spectrumcounting/SpectrumCountingMethod.html" title="enum in com.compomics.util.experiment.quantification.spectrumcounting"><span class="typeNameLink">SpectrumCountingMethod</span></a> - Enum in <a href="com/compomics/util/experiment/quantification/spectrumcounting/package-summary.html">com.compomics.util.experiment.quantification.spectrumcounting</a></dt>
<dd>
<div class="block">Enum of the implemented spectrum counting methods.</div>
</dd>
<dt><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting"><span class="typeNameLink">SpectrumCountingParameters</span></a> - Class in <a href="com/compomics/util/parameters/quantification/spectrum_counting/package-summary.html">com.compomics.util.parameters.quantification.spectrum_counting</a></dt>
<dd>
<div class="block">This class contains the spectrum counting parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#SpectrumCountingParameters--">SpectrumCountingParameters()</a></span> - Constructor for class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Default constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html#SpectrumCountingParameters-com.compomics.util.parameters.quantification.spectrum_counting.SpectrumCountingParameters-">SpectrumCountingParameters(SpectrumCountingParameters)</a></span> - Constructor for class com.compomics.util.parameters.quantification.spectrum_counting.<a href="com/compomics/util/parameters/quantification/spectrum_counting/SpectrumCountingParameters.html" title="class in com.compomics.util.parameters.quantification.spectrum_counting">SpectrumCountingParameters</a></dt>
<dd>
<div class="block">Creates new preferences based on other spectrum counting preferences.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry"><span class="typeNameLink">SpectrumFactory</span></a> - Class in <a href="com/compomics/util/experiment/mass_spectrometry/package-summary.html">com.compomics.util.experiment.mass_spectrometry</a></dt>
<dd>
<div class="block">This factory will provide the spectra when needed.</div>
</dd>
<dt><a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces"><span class="typeNameLink">SpectrumFile</span></a> - Interface in <a href="com/compomics/util/interfaces/package-summary.html">com.compomics.util.interfaces</a></dt>
<dd>
<div class="block">This interface describes the behaviour for a spectrum file
 (ie., PKL file, Mascot Generic file, dat file, ...).</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification"><span class="typeNameLink">SpectrumIdentificationAssumption</span></a> - Class in <a href="com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a></dt>
<dd>
<div class="block">Spectrum identification assumption made by an identification algorithm.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html#SpectrumIdentificationAssumption--">SpectrumIdentificationAssumption()</a></span> - Constructor for class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/SpectrumIdentificationAssumption.html" title="class in com.compomics.util.experiment.identification">SpectrumIdentificationAssumption</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes"><span class="typeNameLink">SpectrumIndex</span></a> - Class in <a href="com/compomics/util/experiment/mass_spectrometry/indexes/package-summary.html">com.compomics.util.experiment.mass_spectrometry.indexes</a></dt>
<dd>
<div class="block">This map stores the fragment ions indexed by mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#SpectrumIndex--">SpectrumIndex()</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>
<div class="block">Constructor for an empty index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html#SpectrumIndex-java.util.HashMap-double-double-boolean-">SpectrumIndex(HashMap&lt;Double, Peak&gt;, double, double, boolean)</a></span> - Constructor for class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/SpectrumIndex.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">SpectrumIndex</a></dt>
<dd>
<div class="block">Builds a new index.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#spectrumLoaded-java.lang.String-java.lang.String-">spectrumLoaded(String, String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the spectrum is contained in the
 given spectrum file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#spectrumLoaded-java.lang.String-">spectrumLoaded(String)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">A boolean indicating whether the spectrum is loaded in the factory.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches"><span class="typeNameLink">SpectrumMatch</span></a> - Class in <a href="com/compomics/util/experiment/identification/matches/package-summary.html">com.compomics.util.experiment.identification.matches</a></dt>
<dd>
<div class="block">This class models a spectrum match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#SpectrumMatch--">SpectrumMatch()</a></span> - Constructor for class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Constructor for the spectrum match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html#SpectrumMatch-java.lang.String-">SpectrumMatch(String)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/SpectrumMatch.html" title="class in com.compomics.util.experiment.identification.matches">SpectrumMatch</a></dt>
<dd>
<div class="block">Constructor for the spectrum match.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/matches_iterators/SpectrumMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators"><span class="typeNameLink">SpectrumMatchesIterator</span></a> - Class in <a href="com/compomics/util/experiment/identification/matches_iterators/package-summary.html">com.compomics.util.experiment.identification.matches_iterators</a></dt>
<dd>
<div class="block">An iterator for spectrum matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/SpectrumMatchesIterator.html#SpectrumMatchesIterator-com.compomics.util.experiment.identification.Identification-com.compomics.util.waiting.WaitingHandler-boolean-">SpectrumMatchesIterator(Identification, WaitingHandler, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/SpectrumMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">SpectrumMatchesIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/SpectrumMatchesIterator.html#SpectrumMatchesIterator-long:A-com.compomics.util.experiment.identification.Identification-com.compomics.util.waiting.WaitingHandler-boolean-">SpectrumMatchesIterator(long[], Identification, WaitingHandler, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/SpectrumMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">SpectrumMatchesIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches_iterators/SpectrumMatchesIterator.html#SpectrumMatchesIterator-long:A-com.compomics.util.experiment.identification.Identification-com.compomics.util.waiting.WaitingHandler-boolean-java.lang.String-">SpectrumMatchesIterator(long[], Identification, WaitingHandler, boolean, String)</a></span> - Constructor for class com.compomics.util.experiment.identification.matches_iterators.<a href="com/compomics/util/experiment/identification/matches_iterators/SpectrumMatchesIterator.html" title="class in com.compomics.util.experiment.identification.matches_iterators">SpectrumMatchesIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#spectrumMill">spectrumMill</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Spectrum Mill search engine, Agilent.</div>
</dd>
<dt><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum"><span class="typeNameLink">SpectrumPanel</span></a> - Class in <a href="com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></dt>
<dd>
<div class="block">This class presents a JPanel that will hold and display a mass spectrum in
 centroid or profile mode.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-com.compomics.util.interfaces.SpectrumFile-">SpectrumPanel(SpectrumFile)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the spectrum
 information in the specified SpectrumFile as an interactive lines plot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-com.compomics.util.interfaces.SpectrumFile-boolean-">SpectrumPanel(SpectrumFile, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the spectrum
 information in the specified SpectrumFile as a line plot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-com.compomics.util.interfaces.SpectrumFile-com.compomics.util.gui.spectrum.GraphicsPanel.DrawingStyle-boolean-">SpectrumPanel(SpectrumFile, GraphicsPanel.DrawingStyle, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the spectrum
 information in the specified SpectrumFile with the specified drawing
 style.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-com.compomics.util.interfaces.SpectrumFile-com.compomics.util.gui.spectrum.GraphicsPanel.DrawingStyle-boolean-java.awt.Color-">SpectrumPanel(SpectrumFile, GraphicsPanel.DrawingStyle, boolean, Color)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the spectrum
 information in the specified SpectrumFile with the specified drawing
 style.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-com.compomics.util.interfaces.SpectrumFile-com.compomics.util.gui.spectrum.GraphicsPanel.DrawingStyle-boolean-java.awt.Color-int-boolean-">SpectrumPanel(SpectrumFile, GraphicsPanel.DrawingStyle, boolean, Color, int, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the spectrum
 information in the specified SpectrumFile with the specified drawing
 style.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-com.compomics.util.interfaces.SpectrumFile-com.compomics.util.gui.spectrum.GraphicsPanel.DrawingStyle-boolean-java.awt.Color-int-boolean-boolean-boolean-">SpectrumPanel(SpectrumFile, GraphicsPanel.DrawingStyle, boolean, Color, int, boolean, boolean, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the spectrum
 information in the specified SpectrumFile with the specified drawing
 style.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-com.compomics.util.interfaces.SpectrumFile-com.compomics.util.gui.spectrum.GraphicsPanel.DrawingStyle-boolean-java.awt.Color-int-boolean-boolean-boolean-int-">SpectrumPanel(SpectrumFile, GraphicsPanel.DrawingStyle, boolean, Color, int, boolean, boolean, boolean, int)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the spectrum
 information in the specified SpectrumFile with the specified drawing
 style.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-com.compomics.util.interfaces.SpectrumFile-com.compomics.util.gui.spectrum.GraphicsPanel.DrawingStyle-boolean-java.awt.Color-int-boolean-boolean-boolean-int-boolean-">SpectrumPanel(SpectrumFile, GraphicsPanel.DrawingStyle, boolean, Color, int, boolean, boolean, boolean, int, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the spectrum
 information in the specified SpectrumFile with the specified drawing
 style.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-double:A-double:A-double-java.lang.String-java.lang.String-">SpectrumPanel(double[], double[], double, String, String)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the passed parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-double:A-double:A-double-java.lang.String-java.lang.String-boolean-">SpectrumPanel(double[], double[], double, String, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the passed parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-double:A-double:A-double-java.lang.String-java.lang.String-int-boolean-">SpectrumPanel(double[], double[], double, String, String, int, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the passed parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-double:A-double:A-double-java.lang.String-java.lang.String-int-boolean-boolean-boolean-">SpectrumPanel(double[], double[], double, String, String, int, boolean, boolean, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the passed parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-double:A-double:A-double-java.lang.String-java.lang.String-int-boolean-boolean-boolean-int-">SpectrumPanel(double[], double[], double, String, String, int, boolean, boolean, boolean, int)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the passed parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/SpectrumPanel.html#SpectrumPanel-double:A-double:A-double-java.lang.String-java.lang.String-int-boolean-boolean-boolean-int-boolean-">SpectrumPanel(double[], double[], double, String, String, int, boolean, boolean, boolean, int, boolean)</a></span> - Constructor for class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/SpectrumPanel.html" title="class in com.compomics.util.gui.spectrum">SpectrumPanel</a></dt>
<dd>
<div class="block">This constructor creates a SpectrumPanel based on the passed parameters.</div>
</dd>
<dt><a href="com/compomics/util/gui/interfaces/SpectrumPanelListener.html" title="interface in com.compomics.util.gui.interfaces"><span class="typeNameLink">SpectrumPanelListener</span></a> - Interface in <a href="com/compomics/util/gui/interfaces/package-summary.html">com.compomics.util.gui.interfaces</a></dt>
<dd>
<div class="block">This interface describes the behaviour for a listener that wants to receive information
 about events that occurred on a SpectrumPanel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.PrecursorWithTitle.html#spectrumTitle">spectrumTitle</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.indexes.<a href="com/compomics/util/experiment/mass_spectrometry/indexes/PrecursorMap.PrecursorWithTitle.html" title="class in com.compomics.util.experiment.mass_spectrometry.indexes">PrecursorMap.PrecursorWithTitle</a></dt>
<dd>
<div class="block">The spectrum title</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#spectrumTitle">spectrumTitle</a></span> - Variable in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Spectrum title.</div>
</dd>
<dt><a href="com/compomics/software/SplashScreen.html" title="class in com.compomics.software"><span class="typeNameLink">SplashScreen</span></a> - Class in <a href="com/compomics/software/package-summary.html">com.compomics.software</a></dt>
<dd>
<div class="block">Class that takes care of converting an svg file to a png file using the
 command line version of apache batik.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/SplashScreen.html#SplashScreen-java.lang.String:A-">SplashScreen(String[])</a></span> - Constructor for class com.compomics.software.<a href="com/compomics/software/SplashScreen.html" title="class in com.compomics.software">SplashScreen</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#splitFile-java.io.File-int-com.compomics.util.waiting.WaitingHandler-">splitFile(File, int, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Splits an mgf file into smaller ones and returns the indexes of the
 generated files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html#splitFile-java.io.File-int-com.compomics.util.waiting.WaitingHandler-">splitFile(File, int, WaitingHandler)</a></span> - Method in class com.compomics.util.experiment.io.mass_spectrometry.msp.<a href="com/compomics/util/experiment/io/mass_spectrometry/msp/MspReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.msp">MspReader</a></dt>
<dd>
<div class="block">Splits an msp file into smaller ones and returns the indexes of the
 generated files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/cli/CommandLineUtils.html#splitInput-java.lang.String-">splitInput(String)</a></span> - Static method in class com.compomics.software.cli.<a href="com/compomics/software/cli/CommandLineUtils.html" title="class in com.compomics.software.cli">CommandLineUtils</a></dt>
<dd>
<div class="block">Splits the input of comma separated command line input and returns the
 results as an arraylist.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#spo">spo</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#SPO">SPO</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#sqid">sqid</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The SeQuence IDentfication (SQID) search engine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#standardGeneticCode">standardGeneticCode</a></span> - Variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">The standard genetic code.</div>
</dd>
<dt><a href="com/compomics/util/experiment/mass_spectrometry/utils/StandardMasses.html" title="enum in com.compomics.util.experiment.mass_spectrometry.utils"><span class="typeNameLink">StandardMasses</span></a> - Enum in <a href="com/compomics/util/experiment/mass_spectrometry/utils/package-summary.html">com.compomics.util.experiment.mass_spectrometry.utils</a></dt>
<dd>
<div class="block">Enum of standard masses.</div>
</dd>
<dt><a href="com/compomics/util/experiment/units/StandardUnit.html" title="enum in com.compomics.util.experiment.units"><span class="typeNameLink">StandardUnit</span></a> - Enum in <a href="com/compomics/util/experiment/units/package-summary.html">com.compomics.util.experiment.units</a></dt>
<dd>
<div class="block">The standard units implemented.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/StartBrowser.html#start-java.lang.String-">start(String)</a></span> - Static method in class com.compomics.util.io.<a href="com/compomics/util/io/StartBrowser.html" title="class in com.compomics.util.io">StartBrowser</a></dt>
<dd>
<div class="block">Opens the given URL in the default browser.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/SwingWorker.html#start--">start()</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/SwingWorker.html" title="class in com.compomics.util.sun">SwingWorker</a></dt>
<dd>
<div class="block">Start the worker thread.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/Duration.html#start--">start()</a></span> - Method in class com.compomics.util.waiting.<a href="com/compomics/util/waiting/Duration.html" title="class in com.compomics.util.waiting">Duration</a></dt>
<dd>
<div class="block">Sets the start time to the system current time in milliseconds.</div>
</dd>
<dt><a href="com/compomics/util/io/StartBrowser.html" title="class in com.compomics.util.io"><span class="typeNameLink">StartBrowser</span></a> - Class in <a href="com/compomics/util/io/package-summary.html">com.compomics.util.io</a></dt>
<dd>
<div class="block">This class starts a browser for the given URL.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/StartBrowser.html#StartBrowser--">StartBrowser()</a></span> - Constructor for class com.compomics.util.io.<a href="com/compomics/util/io/StartBrowser.html" title="class in com.compomics.util.io">StartBrowser</a></dt>
<dd>
<div class="block">Empty constructor.</div>
</dd>
<dt><a href="com/compomics/util/pdbfinder/das/readers/StartEndPosition.html" title="class in com.compomics.util.pdbfinder.das.readers"><span class="typeNameLink">StartEndPosition</span></a> - Class in <a href="com/compomics/util/pdbfinder/das/readers/package-summary.html">com.compomics.util.pdbfinder.das.readers</a></dt>
<dd>
<div class="block">StartEndPosition.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/StartEndPosition.html#StartEndPosition-int-int-">StartEndPosition(int, int)</a></span> - Constructor for class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/StartEndPosition.html" title="class in com.compomics.util.pdbfinder.das.readers">StartEndPosition</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBAccessorGenerator.html#startGenerator-java.lang.String-java.lang.String-java.lang.String-java.lang.String-">startGenerator(String, String, String, String)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBAccessorGenerator.html" title="class in com.compomics.util.db">DBAccessorGenerator</a></dt>
<dd>
<div class="block">This method starts the generator for the specified parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBAccessorGenerator.html#startGenerator-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.lang.String-boolean-">startGenerator(String, String, String, String, String, String, boolean)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBAccessorGenerator.html" title="class in com.compomics.util.db">DBAccessorGenerator</a></dt>
<dd>
<div class="block">This method starts the generator for the specified parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#startNewSection--">startNewSection()</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Starts a new section.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#startNewSection-java.lang.String-">startNewSection(String)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Starts a new section.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#startNewSection-java.lang.String-com.compomics.util.io.export.WorkbookStyle-">startNewSection(String, WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Starts a new section.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#startNewSection-java.lang.String-com.compomics.util.io.export.WorkbookStyle-">startNewSection(String, WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/TextWriter.html#startNewSection-java.lang.String-com.compomics.util.io.export.WorkbookStyle-">startNewSection(String, WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/TextWriter.html" title="class in com.compomics.util.io.export.writers">TextWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#startPeptideShaker-javax.swing.JFrame-">startPeptideShaker(JFrame)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">Starts PeptideShaker from the location of utilities preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#startPeptideShaker-javax.swing.JFrame-java.io.File-">startPeptideShaker(JFrame, File)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">Starts PeptideShaker from the location of utilities preferences and opens
 the file given as argument.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#startPeptideShakerFromURL-javax.swing.JFrame-java.lang.String-java.lang.String-">startPeptideShakerFromURL(JFrame, String, String)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">Starts PeptideShaker from the location of utilities preferences and opens
 the file given as argument.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#startReporter-javax.swing.JFrame-">startReporter(JFrame)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">Starts Reporter from the location of utilities preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#startReshake-javax.swing.JFrame-java.lang.String-">startReshake(JFrame, String)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">Starts PeptideShaker from the location of utilities preferences in the
 Reshake mode and attempts at selecting the given project.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#startSearchGUI-javax.swing.JFrame-">startSearchGUI(JFrame)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">Starts SearchGUI from the location of utilities preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#startSearchGUI-javax.swing.JFrame-java.util.ArrayList-java.util.ArrayList-java.io.File-java.io.File-java.lang.String-java.lang.String-java.lang.String-">startSearchGUI(JFrame, ArrayList&lt;File&gt;, ArrayList&lt;File&gt;, File, File, String, String, String)</a></span> - Static method in class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>
<div class="block">Starts SearchGUI from the location of utilities preferences.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#startShake--">startShake()</a></span> - Method in class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Make the dialog shake when the process has completed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#std-java.util.ArrayList-">std(ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Convenience method returning the standard deviation of a list of doubles.</div>
</dd>
<dt><a href="com/compomics/util/io/StreamGobbler.html" title="class in com.compomics.util.io"><span class="typeNameLink">StreamGobbler</span></a> - Class in <a href="com/compomics/util/io/package-summary.html">com.compomics.util.io</a></dt>
<dd>
<div class="block">A stream gobbler.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/StreamGobbler.html#StreamGobbler-java.io.InputStream-">StreamGobbler(InputStream)</a></span> - Constructor for class com.compomics.util.io.<a href="com/compomics/util/io/StreamGobbler.html" title="class in com.compomics.util.io">StreamGobbler</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#strFileNameAndPath">strFileNameAndPath</a></span> - Variable in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">String to hold the file we are up/downloading</div>
</dd>
<dt><a href="com/compomics/util/gui/utils/user_choice/list_choosers/StringListChooser.html" title="class in com.compomics.util.gui.utils.user_choice.list_choosers"><span class="typeNameLink">StringListChooser</span></a> - Class in <a href="com/compomics/util/gui/utils/user_choice/list_choosers/package-summary.html">com.compomics.util.gui.utils.user_choice.list_choosers</a></dt>
<dd>
<div class="block">Dialog for choosing an item in a list of String.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/list_choosers/StringListChooser.html#StringListChooser-java.awt.Frame-java.util.ArrayList-java.lang.String-java.lang.String-java.lang.String-boolean-">StringListChooser(Frame, ArrayList&lt;String&gt;, String, String, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.utils.user_choice.list_choosers.<a href="com/compomics/util/gui/utils/user_choice/list_choosers/StringListChooser.html" title="class in com.compomics.util.gui.utils.user_choice.list_choosers">StringListChooser</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/user_choice/list_choosers/StringListChooser.html#StringListChooser-javax.swing.JDialog-java.util.ArrayList-java.lang.String-java.lang.String-java.lang.String-boolean-">StringListChooser(JDialog, ArrayList&lt;String&gt;, String, String, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.utils.user_choice.list_choosers.<a href="com/compomics/util/gui/utils/user_choice/list_choosers/StringListChooser.html" title="class in com.compomics.util.gui.utils.user_choice.list_choosers">StringListChooser</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#subAminoAcidsWithCombination">subAminoAcidsWithCombination</a></span> - Variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">The sub amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#subAminoAcidsWithoutCombination">subAminoAcidsWithoutCombination</a></span> - Variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">The sub amino acids.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/variants/amino_acids/Substitution.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids"><span class="typeNameLink">Substitution</span></a> - Class in <a href="com/compomics/util/experiment/biology/variants/amino_acids/package-summary.html">com.compomics.util.experiment.biology.variants.amino_acids</a></dt>
<dd>
<div class="block">Class representing an amino acid substitution.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/amino_acids/Substitution.html#Substitution-char-char-">Substitution(char, char)</a></span> - Constructor for class com.compomics.util.experiment.biology.variants.amino_acids.<a href="com/compomics/util/experiment/biology/variants/amino_acids/Substitution.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids">Substitution</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#subtractIsotope">subtractIsotope</a></span> - Static variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Minimal isotopic correction when matching an ion.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/atoms/impl/Sulfur.html" title="class in com.compomics.util.experiment.biology.atoms.impl"><span class="typeNameLink">Sulfur</span></a> - Class in <a href="com/compomics/util/experiment/biology/atoms/impl/package-summary.html">com.compomics.util.experiment.biology.atoms.impl</a></dt>
<dd>
<div class="block">The sulfur atom.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/impl/Sulfur.html#Sulfur--">Sulfur()</a></span> - Constructor for class com.compomics.util.experiment.biology.atoms.impl.<a href="com/compomics/util/experiment/biology/atoms/impl/Sulfur.html" title="class in com.compomics.util.experiment.biology.atoms.impl">Sulfur</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BasicMathFunctions.html#sum-java.util.ArrayList-">sum(ArrayList&lt;Double&gt;)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/BasicMathFunctions.html" title="class in com.compomics.util.math">BasicMathFunctions</a></dt>
<dd>
<div class="block">Convenience method returning the sum of a list of doubles.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#supportedVersion-java.io.File-">supportedVersion(File)</a></span> - Static method in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Returns a boolean indicating whether the version of the parameters file
 is supported.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#svg">svg</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#SVG">SVG</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/SvgFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">SvgFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.svg files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/SvgFileFilter.html#SvgFileFilter--">SvgFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/SvgFileFilter.html" title="class in com.compomics.util.io.file.filefilters">SvgFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/variants/amino_acids/Swap.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids"><span class="typeNameLink">Swap</span></a> - Class in <a href="com/compomics/util/experiment/biology/variants/amino_acids/package-summary.html">com.compomics.util.experiment.biology.variants.amino_acids</a></dt>
<dd>
<div class="block">Class representing an amino acid swap.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/amino_acids/Swap.html#Swap-char-char-">Swap(char, char)</a></span> - Constructor for class com.compomics.util.experiment.biology.variants.amino_acids.<a href="com/compomics/util/experiment/biology/variants/amino_acids/Swap.html" title="class in com.compomics.util.experiment.biology.variants.amino_acids">Swap</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#swap-int-int-">swap(int, int)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#swapRows-int-int-">swapRows(int, int)</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>
<div class="block">Swap two rows in the pattern.</div>
</dd>
<dt><a href="com/compomics/util/gui/utils/SwingUtils.html" title="class in com.compomics.util.gui.utils"><span class="typeNameLink">SwingUtils</span></a> - Class in <a href="com/compomics/util/gui/utils/package-summary.html">com.compomics.util.gui.utils</a></dt>
<dd>
<div class="block">Swing utilities methods.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/utils/SwingUtils.html#SwingUtils--">SwingUtils()</a></span> - Constructor for class com.compomics.util.gui.utils.<a href="com/compomics/util/gui/utils/SwingUtils.html" title="class in com.compomics.util.gui.utils">SwingUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/sun/SwingWorker.html" title="class in com.compomics.util.sun"><span class="typeNameLink">SwingWorker</span></a> - Class in <a href="com/compomics/util/sun/package-summary.html">com.compomics.util.sun</a></dt>
<dd>
<div class="block">This is the 3rd version of SwingWorker (also known as
 SwingWorker 3), an abstract class that you subclass to
 perform GUI-related work in a dedicated thread.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/SwingWorker.html#SwingWorker--">SwingWorker()</a></span> - Constructor for class com.compomics.util.sun.<a href="com/compomics/util/sun/SwingWorker.html" title="class in com.compomics.util.sun">SwingWorker</a></dt>
<dd>
<div class="block">Start a thread that will call the <code>construct</code> method
 and then exit.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/MetricsPrefix.html#SYMBOL">SYMBOL</a></span> - Variable in enum com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/MetricsPrefix.html" title="enum in com.compomics.util.experiment.units">MetricsPrefix</a></dt>
<dd>
<div class="block">The prefix symbol.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#synonymousVariant">synonymousVariant</a></span> - Static variable in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Substitution matrix grouping synonymous amino acids.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#syst--">syst()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Retrieve the system type from the remote server.</div>
</dd>
</dl>
<a name="I:T">
<!--   -->
</a>
<h2 class="title">T</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#T">T</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#TAB_HTML">TAB_HTML</a></span> - Static variable in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">The tab space to add when using HTML.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/WaitingHandler.html#TAB_NON_HTML">TAB_NON_HTML</a></span> - Static variable in interface com.compomics.util.waiting.<a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting">WaitingHandler</a></dt>
<dd>
<div class="block">The tab space to add when not using HTML.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableMap.html#tableChanged-javax.swing.event.TableModelEvent-">tableChanged(TableModelEvent)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun">TableMap</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#tableChanged-javax.swing.event.TableModelEvent-">tableChanged(TableModelEvent)</a></span> - Method in class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html#tableHeader">tableHeader</a></span> - Static variable in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/PepNovoIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">PepNovoIdfileReader</a></dt>
<dd>
<div class="block">The standard format.</div>
</dd>
<dt><a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun"><span class="typeNameLink">TableMap</span></a> - Class in <a href="com/compomics/util/sun/package-summary.html">com.compomics.util.sun</a></dt>
<dd>
<div class="block">In a chain of data manipulators some behaviour is common.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableMap.html#TableMap--">TableMap()</a></span> - Constructor for class com.compomics.util.sun.<a href="com/compomics/util/sun/TableMap.html" title="class in com.compomics.util.sun">TableMap</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/TableMouseWheelListener.html" title="class in com.compomics.util.gui"><span class="typeNameLink">TableMouseWheelListener</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">MouseWheelListener for SelfUpdatingTableModels that supports scrolling in a
 more efficient way.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/TableMouseWheelListener.html#TableMouseWheelListener-javax.swing.JTable-">TableMouseWheelListener(JTable)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/TableMouseWheelListener.html" title="class in com.compomics.util.gui">TableMouseWheelListener</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/TableProperties.html" title="class in com.compomics.util.gui"><span class="typeNameLink">TableProperties</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">This class groups properties for the tables</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/TableProperties.html#TableProperties--">TableProperties()</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/TableProperties.html" title="class in com.compomics.util.gui">TableProperties</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/TableScrollBarListener.html" title="class in com.compomics.util.gui"><span class="typeNameLink">TableScrollBarListener</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">AdjustmentListener for SelfUpdatingTableModels supporting scrolling in a more
 efficient way.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/TableScrollBarListener.html#TableScrollBarListener-javax.swing.JTable-">TableScrollBarListener(JTable)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/TableScrollBarListener.html" title="class in com.compomics.util.gui">TableScrollBarListener</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun"><span class="typeNameLink">TableSorter</span></a> - Class in <a href="com/compomics/util/sun/package-summary.html">com.compomics.util.sun</a></dt>
<dd>
<div class="block">A sorter for TableModels.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#TableSorter--">TableSorter()</a></span> - Constructor for class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/sun/TableSorter.html#TableSorter-javax.swing.table.TableModel-">TableSorter(TableModel)</a></span> - Constructor for class com.compomics.util.sun.<a href="com/compomics/util/sun/TableSorter.html" title="class in com.compomics.util.sun">TableSorter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#tableToFile-javax.swing.JTable-java.lang.String-com.compomics.util.gui.waiting.waitinghandlers.ProgressDialogX-boolean-java.io.BufferedWriter-">tableToFile(JTable, String, ProgressDialogX, boolean, BufferedWriter)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Writes the table to a file as separated text.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#tableToText-javax.swing.JTable-java.lang.String-com.compomics.util.gui.waiting.waitinghandlers.ProgressDialogX-boolean-">tableToText(JTable, String, ProgressDialogX, boolean)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Returns the table as a separated text file.</div>
</dd>
<dt><a href="com/compomics/util/interfaces/TableValueWrapper.html" title="interface in com.compomics.util.interfaces"><span class="typeNameLink">TableValueWrapper</span></a> - Interface in <a href="com/compomics/util/interfaces/package-summary.html">com.compomics.util.interfaces</a></dt>
<dd>
<div class="block">This interface describes the behaviour for a class that wraps a table element.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags"><span class="typeNameLink">Tag</span></a> - Class in <a href="com/compomics/util/experiment/identification/amino_acid_tags/package-summary.html">com.compomics.util.experiment.identification.amino_acid_tags</a></dt>
<dd>
<div class="block">This class represents a sequence mass tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#Tag--">Tag()</a></span> - Constructor for class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Constructor for an empty tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#Tag-com.compomics.util.experiment.identification.amino_acid_tags.Tag-">Tag(Tag)</a></span> - Constructor for class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Creates a new tag instance based on the given one.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#Tag-double-com.compomics.util.experiment.biology.aminoacids.sequence.AminoAcidSequence-double-">Tag(double, AminoAcidSequence, double)</a></span> - Constructor for class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>
<div class="block">Constructor for a tag consisting of a sequence tag between two mass tags.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions"><span class="typeNameLink">TagAssumption</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_assumptions/package-summary.html">com.compomics.util.experiment.identification.spectrum_assumptions</a></dt>
<dd>
<div class="block">This class represent a tag assumption made by an identification algorithm
 based on a sequence tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html#TagAssumption--">TagAssumption()</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">TagAssumption</a></dt>
<dd>
<div class="block">Constructor for en empty assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html#TagAssumption-int-int-com.compomics.util.experiment.identification.amino_acid_tags.Tag-int-double-">TagAssumption(int, int, Tag, int, double)</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">TagAssumption</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/amino_acid_tags/TagComponent.html" title="interface in com.compomics.util.experiment.identification.amino_acid_tags"><span class="typeNameLink">TagComponent</span></a> - Interface in <a href="com/compomics/util/experiment/identification/amino_acid_tags/package-summary.html">com.compomics.util.experiment.identification.amino_acid_tags</a></dt>
<dd>
<div class="block">Interface for a sequence tag component.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#tagComponent">tagComponent</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Information about the component where the X belongs to.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/TagFactory.html" title="class in com.compomics.util.experiment.identification"><span class="typeNameLink">TagFactory</span></a> - Class in <a href="com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a></dt>
<dd>
<div class="block">Convenience class for sequence tag generation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/TagFactory.html#TagFactory--">TagFactory()</a></span> - Constructor for class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/TagFactory.html" title="class in com.compomics.util.experiment.identification">TagFactory</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl"><span class="typeNameLink">TagFragmentIon</span></a> - Class in <a href="com/compomics/util/experiment/biology/ions/impl/package-summary.html">com.compomics.util.experiment.biology.ions.impl</a></dt>
<dd>
<div class="block">A fragment ion obtained from a tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#TagFragmentIon-int-int-int-double-com.compomics.util.experiment.biology.ions.NeutralLoss:A-double-">TagFragmentIon(int, int, int, double, NeutralLoss[], double)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#TagFragmentIon-int-com.compomics.util.experiment.biology.ions.NeutralLoss:A-">TagFragmentIon(int, NeutralLoss[])</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Constructor for a generic ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#TagFragmentIon-int-">TagFragmentIon(int)</a></span> - Constructor for class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Constructor for a generic ion without neutral losses.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators"><span class="typeNameLink">TagSpectrumAnnotator</span></a> - Class in <a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators</a></dt>
<dd>
<div class="block">Annotates a spectrum with information from a tag.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html#TagSpectrumAnnotator--">TagSpectrumAnnotator()</a></span> - Constructor for class com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/TagSpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">TagSpectrumAnnotator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/identification/utils/TagUtils.html" title="class in com.compomics.util.experiment.identification.utils"><span class="typeNameLink">TagUtils</span></a> - Class in <a href="com/compomics/util/experiment/identification/utils/package-summary.html">com.compomics.util.experiment.identification.utils</a></dt>
<dd>
<div class="block">This class groups functions that can be used to work with peptides.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/TagUtils.html#TagUtils--">TagUtils()</a></span> - Constructor for class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/TagUtils.html" title="class in com.compomics.util.experiment.identification.utils">TagUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/TarUtils.html#tarFolder-java.io.File-java.io.File-com.compomics.util.waiting.WaitingHandler-">tarFolder(File, File, WaitingHandler)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/TarUtils.html" title="class in com.compomics.util.io.compression">TarUtils</a></dt>
<dd>
<div class="block">Tar a given folder to a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/TarUtils.html#tarFolderContent-java.io.File-java.io.File-java.util.HashSet-com.compomics.util.waiting.WaitingHandler-">tarFolderContent(File, File, HashSet&lt;String&gt;, WaitingHandler)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/TarUtils.html" title="class in com.compomics.util.io.compression">TarUtils</a></dt>
<dd>
<div class="block">Tar the content of a given folder to a file.</div>
</dd>
<dt><a href="com/compomics/util/io/compression/TarUtils.html" title="class in com.compomics.util.io.compression"><span class="typeNameLink">TarUtils</span></a> - Class in <a href="com/compomics/util/io/compression/package-summary.html">com.compomics.util.io.compression</a></dt>
<dd>
<div class="block">This class contains convenience methods for taring files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/TarUtils.html#TarUtils--">TarUtils()</a></span> - Constructor for class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/TarUtils.html" title="class in com.compomics.util.io.compression">TarUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/file_handling/TempFilesManager.html" title="class in com.compomics.util.gui.file_handling"><span class="typeNameLink">TempFilesManager</span></a> - Class in <a href="com/compomics/util/gui/file_handling/package-summary.html">com.compomics.util.gui.file_handling</a></dt>
<dd>
<div class="block">This class can be used to reference temp files and manage them.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/file_handling/TempFilesManager.html#TempFilesManager--">TempFilesManager()</a></span> - Constructor for class com.compomics.util.gui.file_handling.<a href="com/compomics/util/gui/file_handling/TempFilesManager.html" title="class in com.compomics.util.gui.file_handling">TempFilesManager</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/junit/TestCaseLM.html" title="class in com.compomics.util.junit"><span class="typeNameLink">TestCaseLM</span></a> - Class in <a href="com/compomics/util/junit/package-summary.html">com.compomics.util.junit</a></dt>
<dd>
<div class="block">This TestCase extension contains some useful methods for loading Properties files
 from the classpath and getting general files' full name or an InputStream from
 the classpath.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTPClient.html#testFTPConnection--">testFTPConnection()</a></span> - Method in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTPClient.html" title="class in com.compomics.util.io.ftp">FTPClient</a></dt>
<dd>
<div class="block">This method can be used to test the connection with the FTP server.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#testPath-java.lang.String-">testPath(String)</a></span> - Static method in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Tests whether it is possible to write in a destination folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#TEXTMODE">TEXTMODE</a></span> - Static variable in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/export/writers/TextWriter.html" title="class in com.compomics.util.io.export.writers"><span class="typeNameLink">TextWriter</span></a> - Class in <a href="com/compomics/util/io/export/writers/package-summary.html">com.compomics.util.io.export.writers</a></dt>
<dd>
<div class="block">ExportWriter for the export to text files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/TextWriter.html#TextWriter-java.io.File-java.lang.String-int-boolean-">TextWriter(File, String, int, boolean)</a></span> - Constructor for class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/TextWriter.html" title="class in com.compomics.util.io.export.writers">TextWriter</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#theoreticalFragmentIons">theoreticalFragmentIons</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">The theoretic fragment ions.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#theoreticMass">theoreticMass</a></span> - Variable in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="block"><span class="deprecationComment">use the double value instead.</span></div>
</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#theoreticMass1">theoreticMass1</a></span> - Variable in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">The theoretic mass.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigMathUtils.html#thousand">thousand</a></span> - Static variable in class com.compomics.util.math.<a href="com/compomics/util/math/BigMathUtils.html" title="class in com.compomics.util.math">BigMathUtils</a></dt>
<dd>
<div class="block">Big decimal value of 1000.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#threeLetterCode">threeLetterCode</a></span> - Variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>
<div class="block">Three letter code of the amino acid.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Threonine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Threonine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html#Threonine--">Threonine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Threonine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Threonine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#THREONINE">THREONINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Threonine immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#tide">tide</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The Tide search engine.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">TideIdfileReader</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">This IdfileReader reads identifications from an Tide tsv results file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html#TideIdfileReader--">TideIdfileReader()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">TideIdfileReader</a></dt>
<dd>
<div class="block">Default constructor for the purpose of instantiation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html#TideIdfileReader-java.io.File-">TideIdfileReader(File)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">TideIdfileReader</a></dt>
<dd>
<div class="block">Constructor for a Tide tsv results file reader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html#TideIdfileReader-java.io.File-com.compomics.util.waiting.WaitingHandler-">TideIdfileReader(File, WaitingHandler)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/TideIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">TideIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an Tide tsv result file reader.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">TideParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">The Tide specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#TideParameters--">TideParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm"><span class="typeNameLink">TideParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a></dt>
<dd>
<div class="block">Dialog for the Tide specific settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html#TideParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.TideParameters-boolean-">TideParametersDialog(Frame, TideParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">TideParametersDialog</a></dt>
<dd>
<div class="block">Creates a new TideSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html#TideParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.TideParameters-boolean-">TideParametersDialog(Dialog, Frame, TideParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">TideParametersDialog</a></dt>
<dd>
<div class="block">Creates a new TideSettingsDialog with a dialog as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#tiesResolution">tiesResolution</a></span> - Variable in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">The methods to use to select the best peak when multiple are possible.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#tiff">tiff</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#TIFF">TIFF</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/TiffFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">TiffFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.tiff files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/TiffFileFilter.html#TiffFileFilter--">TiffFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/TiffFileFilter.html" title="class in com.compomics.util.io.file.filefilters">TiffFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#timeOut">timeOut</a></span> - Static variable in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">The time out in milliseconds when querying the file.</div>
</dd>
<dt><a href="com/compomics/util/gui/renderers/TimestampRenderer.html" title="class in com.compomics.util.gui.renderers"><span class="typeNameLink">TimestampRenderer</span></a> - Class in <a href="com/compomics/util/gui/renderers/package-summary.html">com.compomics.util.gui.renderers</a></dt>
<dd>
<div class="block">TimestampRenderer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/TimestampRenderer.html#TimestampRenderer--">TimestampRenderer()</a></span> - Constructor for class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/TimestampRenderer.html" title="class in com.compomics.util.gui.renderers">TimestampRenderer</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html#TITLED_BORDER_HORIZONTAL_PADDING">TITLED_BORDER_HORIZONTAL_PADDING</a></span> - Static variable in class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SearchParametersDialog</a></dt>
<dd>
<div class="block">The horizontal padding used before and after the text in the titled
 borders.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_126">TMT_126</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 126.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_126_ETD">TMT_126_ETD</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 126 with ETD fragmentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_127C">TMT_127C</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 127C.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_127C_ETD">TMT_127C_ETD</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 127C with ETD fragmentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_127N">TMT_127N</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 127N.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_127N_ETD">TMT_127N_ETD</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 127N with ETD fragmentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_128C">TMT_128C</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 128C.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_128C_ETD">TMT_128C_ETD</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 128C with ETD fragmentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_128N">TMT_128N</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 128N.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_128N_ETD">TMT_128N_ETD</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 128N with ETD fragmentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_129C">TMT_129C</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 129C.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_129C_ETD">TMT_129C_ETD</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 129C with ETD fragmentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_129N">TMT_129N</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 129N.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_129N_ETD">TMT_129N_ETD</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 129N with ETD fragmentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_130C">TMT_130C</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 130C.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_130C_ETD">TMT_130C_ETD</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 130C with ETD fragmentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_130N">TMT_130N</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 130N.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_130N_ETD">TMT_130N_ETD</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 130N with ETD fragmentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_131">TMT_131</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 131.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_131_ETD">TMT_131_ETD</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT 131 with ETD fragmentation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_226">TMT_226</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT (reporter + balancer).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html#TMT_230">TMT_230</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ReporterIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ReporterIon</a></dt>
<dd>
<div class="block">Standard reporter ion TMT (reporter + balancer).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/JsonMarshaller.html#toJson-java.lang.Object-">toJson(Object)</a></span> - Method in class com.compomics.util.io.json.<a href="com/compomics/util/io/json/JsonMarshaller.html" title="class in com.compomics.util.io.json">JsonMarshaller</a></dt>
<dd>
<div class="block">Convert an object to JSON.</div>
</dd>
<dt><a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software"><span class="typeNameLink">ToolFactory</span></a> - Class in <a href="com/compomics/software/package-summary.html">com.compomics.software</a></dt>
<dd>
<div class="block">This class can be used to start compomics tools.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/ToolFactory.html#ToolFactory--">ToolFactory()</a></span> - Constructor for class com.compomics.software.<a href="com/compomics/software/ToolFactory.html" title="class in com.compomics.software">ToolFactory</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/renderers/ToolTipComboBoxRenderer.html" title="class in com.compomics.util.gui.renderers"><span class="typeNameLink">ToolTipComboBoxRenderer</span></a> - Class in <a href="com/compomics/util/gui/renderers/package-summary.html">com.compomics.util.gui.renderers</a></dt>
<dd>
<div class="block">A combo box renderer that allows tooltip for each element in the combo box
 list.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/ToolTipComboBoxRenderer.html#ToolTipComboBoxRenderer-java.util.Vector-int-">ToolTipComboBoxRenderer(Vector, int)</a></span> - Constructor for class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/ToolTipComboBoxRenderer.html" title="class in com.compomics.util.gui.renderers">ToolTipComboBoxRenderer</a></dt>
<dd>
<div class="block">Creates a new instance of the MyComboBoxRenderer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/renderers/ToolTipComboBoxRenderer.html#ToolTipComboBoxRenderer-java.util.Vector-int-java.awt.Color-">ToolTipComboBoxRenderer(Vector, int, Color)</a></span> - Constructor for class com.compomics.util.gui.renderers.<a href="com/compomics/util/gui/renderers/ToolTipComboBoxRenderer.html" title="class in com.compomics.util.gui.renderers">ToolTipComboBoxRenderer</a></dt>
<dd>
<div class="block">Creates a new instance of the MyComboBoxRenderer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLI.html#toString--">toString()</a></span> - Method in class com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLI.html" title="class in com.compomics.cli.enzymes">EnzymesCLI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLI.html#toString--">toString()</a></span> - Method in class com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLI.html" title="class in com.compomics.cli.modifications">ModificationsCLI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/BinaryArrayImpl.html#toString--">toString()</a></span> - Method in class com.compomics.util.<a href="com/compomics/util/BinaryArrayImpl.html" title="class in com.compomics.util">BinaryArrayImpl</a></dt>
<dd>
<div class="block">Returns a useful String representation of this Imlplementation instance
 that includes details of all fields.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/Constructors.html#toString--">toString()</a></span> - Method in class com.compomics.util.db.components.<a href="com/compomics/util/db/components/Constructors.html" title="class in com.compomics.util.db.components">Constructors</a></dt>
<dd>
<div class="block">This method outputs the generated code in String format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/DeleteableCode.html#toString--">toString()</a></span> - Method in class com.compomics.util.db.components.<a href="com/compomics/util/db/components/DeleteableCode.html" title="class in com.compomics.util.db.components">DeleteableCode</a></dt>
<dd>
<div class="block">This method will report on the generated code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/GettersAndSetters.html#toString--">toString()</a></span> - Method in class com.compomics.util.db.components.<a href="com/compomics/util/db/components/GettersAndSetters.html" title="class in com.compomics.util.db.components">GettersAndSetters</a></dt>
<dd>
<div class="block">This method generates the code for the getters and setters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/InstanceVariables.html#toString--">toString()</a></span> - Method in class com.compomics.util.db.components.<a href="com/compomics/util/db/components/InstanceVariables.html" title="class in com.compomics.util.db.components">InstanceVariables</a></dt>
<dd>
<div class="block">This method generates the code for the instance variables.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/PersistableCode.html#toString--">toString()</a></span> - Method in class com.compomics.util.db.components.<a href="com/compomics/util/db/components/PersistableCode.html" title="class in com.compomics.util.db.components">PersistableCode</a></dt>
<dd>
<div class="block">This method will report on the generated code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/RetrievableCode.html#toString--">toString()</a></span> - Method in class com.compomics.util.db.components.<a href="com/compomics/util/db/components/RetrievableCode.html" title="class in com.compomics.util.db.components">RetrievableCode</a></dt>
<dd>
<div class="block">This method will report on the generated code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/UpdateableCode.html#toString--">toString()</a></span> - Method in class com.compomics.util.db.components.<a href="com/compomics/util/db/components/UpdateableCode.html" title="class in com.compomics.util.db.components">UpdateableCode</a></dt>
<dd>
<div class="block">This method will report on the generated code.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBAccessor.html#toString--">toString()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBAccessor.html" title="class in com.compomics.util.db">DBAccessor</a></dt>
<dd>
<div class="block">This method returns the generated code as a String.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBMetaData.html#toString--">toString()</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBMetaData.html" title="class in com.compomics.util.db">DBMetaData</a></dt>
<dd>
<div class="block">This method presents a String representation for this object.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/GeneratorException.html#toString--">toString()</a></span> - Method in exception com.compomics.util.db.<a href="com/compomics/util/db/GeneratorException.html" title="class in com.compomics.util.db">GeneratorException</a></dt>
<dd>
<div class="block">This method returns a String with the description for this exception.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidPattern.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidPattern</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.biology.aminoacids.sequence.<a href="com/compomics/util/experiment/biology/aminoacids/sequence/AminoAcidSequence.html" title="class in com.compomics.util.experiment.biology.aminoacids.sequence">AminoAcidSequence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/Atom.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/Atom.html" title="class in com.compomics.util.experiment.biology.atoms">Atom</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomChain.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomChain.html" title="class in com.compomics.util.experiment.biology.atoms">AtomChain</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.biology.atoms.<a href="com/compomics/util/experiment/biology/atoms/AtomImpl.html" title="class in com.compomics.util.experiment.biology.atoms">AtomImpl</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Charge.html#toString-int-">toString(int)</a></span> - Static method in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Charge.html" title="class in com.compomics.util.experiment.biology.ions">Charge</a></dt>
<dd>
<div class="block">Returns a string representing the charge.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/Modification.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/Modification.html" title="class in com.compomics.util.experiment.biology.modifications">Modification</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItemComparator.html#toString--">toString()</a></span> - Method in enum com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItemComparator.html" title="enum in com.compomics.util.experiment.filtering">FilterItemComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.identification.amino_acid_tags.<a href="com/compomics/util/experiment/identification/amino_acid_tags/Tag.html" title="class in com.compomics.util.experiment.identification.amino_acid_tags">Tag</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html#toString--">toString()</a></span> - Method in enum com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html" title="enum in com.compomics.util.experiment.identification.modification">ModificationLocalizationScore</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_assumptions.<a href="com/compomics/util/experiment/identification/spectrum_assumptions/TagAssumption.html" title="class in com.compomics.util.experiment.identification.spectrum_assumptions">TagAssumption</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#toString--">toString()</a></span> - Method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method reports on the entire processed(!) header.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/Header.html#toString-java.lang.String-">toString(String)</a></span> - Method in class com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/Header.html" title="class in com.compomics.util.experiment.io.biology.protein">Header</a></dt>
<dd>
<div class="block">This method reports on the entire processed(!) header, with the given
 decoy tag added.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html#toString--">toString()</a></span> - Method in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsFormat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Peak.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Peak</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/UnitOfMeasurement.html#toString--">toString()</a></span> - Method in class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/UnitOfMeasurement.html" title="class in com.compomics.util.experiment.units">UnitOfMeasurement</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicElement.html#toString--">toString()</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicElement.html" title="class in com.compomics.util.general">IsotopicElement</a></dt>
<dd>
<div class="block">To string method</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationAlgorithmParameter.html#toString-boolean-">toString(boolean)</a></span> - Method in interface com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationAlgorithmParameter.html" title="interface in com.compomics.util.gui.parameters.identification">IdentificationAlgorithmParameter</a></dt>
<dd>
<div class="block">Returns the parameters as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/Modification.html#toString--">toString()</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/Modification.html" title="interface in com.compomics.util.interfaces">Modification</a></dt>
<dd>
<div class="block">This method returns a String representation of the Modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html#toString--">toString()</a></span> - Method in enum com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html" title="enum in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters.MatchingType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html#toString--">toString()</a></span> - Method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.CleavageParameter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html#toString--">toString()</a></span> - Method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.Specificity</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.MassAccuracyType.html#toString--">toString()</a></span> - Method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.MassAccuracyType.html" title="enum in com.compomics.util.parameters.identification.search">SearchParameters.MassAccuracyType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#toString--">toString()</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Returns the search parameters as a string.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">CometParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/DirecTagParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">DirecTagParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsAmandaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsAmandaParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MsgfParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MsgfParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/MyriMatchParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">MyriMatchParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/NovorParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">NovorParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/OmssaParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">OmssaParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PepnovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PepnovoParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/PNovoParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">PNovoParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/TideParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">TideParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#toString-boolean-">toString(boolean)</a></span> - Method in class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html#toString--">toString()</a></span> - Method in class com.compomics.util.pdbfinder.das.readers.<a href="com/compomics/util/pdbfinder/das/readers/DasFeature.html" title="class in com.compomics.util.pdbfinder.das.readers">DasFeature</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#toString--">toString()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">This method generates a String representation of the DualEnzyme,
 which is useful for displaying as useful information for the user or
 during testing/debugging.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/DualEnzyme.html#toString-java.lang.String-">toString(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/DualEnzyme.html" title="class in com.compomics.util.protein">DualEnzyme</a></dt>
<dd>
<div class="block">This method generates a String representation of the DualEnzyme,
 which is useful for displaying as useful information for the user or
 during testing/debugging.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#toString--">toString()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method generates a String representation of the Enzyme, which is
 useful for displaying as useful information for the user or during
 testing/debugging.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Enzyme.html#toString-java.lang.String-">toString(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Enzyme.html" title="class in com.compomics.util.protein">Enzyme</a></dt>
<dd>
<div class="block">This method generates a String representation of the Enzyme, which is
 useful for displaying as useful information for the user or during
 testing/debugging.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationImplementation.html#toString--">toString()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationImplementation.html" title="class in com.compomics.util.protein">ModificationImplementation</a></dt>
<dd>
<div class="block">This method returns a String representation of the Modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/ModificationTemplate.html#toString--">toString()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/ModificationTemplate.html" title="class in com.compomics.util.protein">ModificationTemplate</a></dt>
<dd>
<div class="block">This method returns a String representation of the Modification.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/MolecularFormula.html#toString--">toString()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/MolecularFormula.html" title="class in com.compomics.util.protein">MolecularFormula</a></dt>
<dd>
<div class="block">To string method</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/RegExEnzyme.html#toString--">toString()</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/RegExEnzyme.html" title="class in com.compomics.util.protein">RegExEnzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/RegExEnzyme.html#toString-java.lang.String-">toString(String)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/RegExEnzyme.html" title="class in com.compomics.util.protein">RegExEnzyme</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#toString-java.lang.String-java.lang.String-">toString(String, String)</a></span> - Static method in class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>
<div class="block">Returns a string in the form value(attribute).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/waiting/Duration.html#toString--">toString()</a></span> - Method in class com.compomics.util.waiting.<a href="com/compomics/util/waiting/Duration.html" title="class in com.compomics.util.waiting">Duration</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#transformScore-double-">transformScore(double)</a></span> - Static method in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Returns a score from a raw score where the score = -10*log(rawScore).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#transitionsSingleBaseSubstitution">transitionsSingleBaseSubstitution</a></span> - Static variable in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Substitution matrix allowing for a single base transition variant.</div>
</dd>
<dt><a href="com/compomics/util/general/Translate.html" title="class in com.compomics.util.general"><span class="typeNameLink">Translate</span></a> - Class in <a href="com/compomics/util/general/package-summary.html">com.compomics.util.general</a></dt>
<dd>
<div class="block">This class allows the user to translate a DNA sequence into 6 reading frames.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/Translate.html#Translate--">Translate()</a></span> - Constructor for class com.compomics.util.general.<a href="com/compomics/util/general/Translate.html" title="class in com.compomics.util.general">Translate</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequence.html#translate--">translate()</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide">NucleotideSequence</a></dt>
<dd>
<div class="block">This method translates the nucleotide sequence in six reading frames.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html#translate--">translate()</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html" title="class in com.compomics.util.nucleotide">NucleotideSequenceImpl</a></dt>
<dd>
<div class="block">This method translates the specified nucleotidesequence into
 the six reading frames.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html#translateToStopCodonSeparatedEntries-java.lang.String-java.lang.String-">translateToStopCodonSeparatedEntries(String, String)</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequenceImpl.html" title="class in com.compomics.util.nucleotide">NucleotideSequenceImpl</a></dt>
<dd>
<div class="block">This method translates the specified nucleotidesequence into
 the six reading frames.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/matrix/DoubleMatrix.html#transpose-com.compomics.util.math.matrix.DoubleMatrix-">transpose(DoubleMatrix)</a></span> - Static method in class com.compomics.util.math.matrix.<a href="com/compomics/util/math/matrix/DoubleMatrix.html" title="class in com.compomics.util.math.matrix">DoubleMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html#transversalSingleBaseSubstitution">transversalSingleBaseSubstitution</a></span> - Static variable in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AaSubstitutionMatrix.html" title="class in com.compomics.util.experiment.biology.variants">AaSubstitutionMatrix</a></dt>
<dd>
<div class="block">Substitution matrix allowing for a single base transversion variant.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItemComparator.html#trueFalse">trueFalse</a></span> - Static variable in enum com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItemComparator.html" title="enum in com.compomics.util.experiment.filtering">FilterItemComparator</a></dt>
<dd>
<div class="block">Array of possibilities used for a boolean value.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Tryptophan</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Tryptophan.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html#Tryptophan--">Tryptophan()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tryptophan.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tryptophan</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#TRYPTOPHAN">TRYPTOPHAN</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Tryptophan immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/BigMathUtils.html#two">two</a></span> - Static variable in class com.compomics.util.math.<a href="com/compomics/util/math/BigMathUtils.html" title="class in com.compomics.util.math">BigMathUtils</a></dt>
<dd>
<div class="block">Big decimal value of 2.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#txt">txt</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#TXT">TXT</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.html#type">type</a></span> - Variable in class com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.html" title="class in com.compomics.util.experiment.biology.ions">Ion</a></dt>
<dd>
<div class="block">Type of ion.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Tyrosine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Tyrosine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html#Tyrosine--">Tyrosine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Tyrosine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Tyrosine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#TYROSINE">TYROSINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Tyrosine immonium ion.</div>
</dd>
</dl>
<a name="I:U">
<!--   -->
</a>
<h2 class="title">U</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#U">U</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/object/objects/BlobObject.html#unBlob--">unBlob()</a></span> - Method in class com.compomics.util.db.object.objects.<a href="com/compomics/util/db/object/objects/BlobObject.html" title="class in com.compomics.util.db.object.objects">BlobObject</a></dt>
<dd>
<div class="block">Returns the object represented by this blob.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html#UNDERIVATED_MASS">UNDERIVATED_MASS</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/Glycan.html" title="class in com.compomics.util.experiment.biology.ions.impl">Glycan</a></dt>
<dd>
<div class="block">Final index for underivated masses.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/FileDAO.html#unGzipAndUntarFile-java.util.zip.GZIPInputStream-java.io.File-com.compomics.util.waiting.WaitingHandler-">unGzipAndUntarFile(GZIPInputStream, File, WaitingHandler)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/FileDAO.html" title="class in com.compomics.software.autoupdater">FileDAO</a></dt>
<dd>
<div class="block">Untars and ungzips a .tar.gz file.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/UniformFragmentation.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models"><span class="typeNameLink">UniformFragmentation</span></a> - Class in <a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models</a></dt>
<dd>
<div class="block">Dummy intensity predictor that returns 1.0 for every ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/UniformFragmentation.html#UniformFragmentation--">UniformFragmentation()</a></span> - Constructor for class com.compomics.util.experiment.identification.peptide_fragmentation.models.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/UniformFragmentation.html" title="class in com.compomics.util.experiment.identification.peptide_fragmentation.models">UniformFragmentation</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#UNIPROT_TAXONOMY_FILENAME">UNIPROT_TAXONOMY_FILENAME</a></span> - Static variable in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">The name of the UniProt taxonomy file.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings"><span class="typeNameLink">UniprotTaxonomy</span></a> - Class in <a href="com/compomics/util/experiment/biology/taxonomy/mappings/package-summary.html">com.compomics.util.experiment.biology.taxonomy.mappings</a></dt>
<dd>
<div class="block">Mapping of the UniProt taxonomy.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html#UniprotTaxonomy--">UniprotTaxonomy()</a></span> - Constructor for class com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/UniprotTaxonomy.html" title="class in com.compomics.util.experiment.biology.taxonomy.mappings">UniprotTaxonomy</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html#unit">unit</a></span> - Variable in enum com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html" title="enum in com.compomics.util.experiment.identification.matches">IonMatch.MzErrorType</a></dt>
<dd>
<div class="block">The unit to use</div>
</dd>
<dt><a href="com/compomics/util/experiment/units/UnitOfMeasurement.html" title="class in com.compomics.util.experiment.units"><span class="typeNameLink">UnitOfMeasurement</span></a> - Class in <a href="com/compomics/util/experiment/units/package-summary.html">com.compomics.util.experiment.units</a></dt>
<dd>
<div class="block">The unit to use for a measure.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/UnitOfMeasurement.html#UnitOfMeasurement-java.lang.String-java.lang.String-com.compomics.util.experiment.units.MetricsPrefix-">UnitOfMeasurement(String, String, MetricsPrefix)</a></span> - Constructor for class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/UnitOfMeasurement.html" title="class in com.compomics.util.experiment.units">UnitOfMeasurement</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/UnitOfMeasurement.html#UnitOfMeasurement-com.compomics.util.experiment.units.StandardUnit-com.compomics.util.experiment.units.MetricsPrefix-">UnitOfMeasurement(StandardUnit, MetricsPrefix)</a></span> - Constructor for class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/UnitOfMeasurement.html" title="class in com.compomics.util.experiment.units">UnitOfMeasurement</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/UnitOfMeasurement.html#UnitOfMeasurement-com.compomics.util.experiment.units.StandardUnit-">UnitOfMeasurement(StandardUnit)</a></span> - Constructor for class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/UnitOfMeasurement.html" title="class in com.compomics.util.experiment.units">UnitOfMeasurement</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/units/Units.html" title="class in com.compomics.util.experiment.units"><span class="typeNameLink">Units</span></a> - Class in <a href="com/compomics/util/experiment/units/package-summary.html">com.compomics.util.experiment.units</a></dt>
<dd>
<div class="block">Convenience class providing units.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/Units.html#Units--">Units()</a></span> - Constructor for class com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/Units.html" title="class in com.compomics.util.experiment.units">Units</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html#UNKNOWN">UNKNOWN</a></span> - Static variable in class com.compomics.util.experiment.biology.taxonomy.<a href="com/compomics/util/experiment/biology/taxonomy/SpeciesFactory.html" title="class in com.compomics.util.experiment.biology.taxonomy">SpeciesFactory</a></dt>
<dd>
<div class="block">Tag for unknown species.</div>
</dd>
<dt><a href="com/compomics/util/general/UnknownElementMassException.html" title="class in com.compomics.util.general"><span class="typeNameLink">UnknownElementMassException</span></a> - Exception in <a href="com/compomics/util/general/package-summary.html">com.compomics.util.general</a></dt>
<dd>
<div class="block">This class implements an Exception, thrown when a MassCalc instance
 is confronted with an element symbol it cannot retrieve in its 
 element lists.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/UnknownElementMassException.html#UnknownElementMassException-java.lang.String-">UnknownElementMassException(String)</a></span> - Constructor for exception com.compomics.util.general.<a href="com/compomics/util/general/UnknownElementMassException.html" title="class in com.compomics.util.general">UnknownElementMassException</a></dt>
<dd>
<div class="block">The constructor requires the caller to specify the element
 which was not recognized.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PushBackStringReader.html#unreadLine--">unreadLine()</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PushBackStringReader.html" title="class in com.compomics.util.io">PushBackStringReader</a></dt>
<dd>
<div class="block">This method 'unreads' a line of text from the Buffer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html#UNRELATED">UNRELATED</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/PSParameter.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">PSParameter</a></dt>
<dd>
<div class="block">Static index for a protein group: 3 - unrelated proteins proteins or
 peptide shared by unrelated proteins.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/UnspecificCombinationIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators"><span class="typeNameLink">UnspecificCombinationIterator</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion.iterators</a></dt>
<dd>
<div class="block">Iterator for unspecific cleavage.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/UnspecificCombinationIterator.html#UnspecificCombinationIterator-com.compomics.util.experiment.identification.protein_sequences.digestion.ProteinIteratorUtils-java.lang.String-java.lang.Double-java.lang.Double-">UnspecificCombinationIterator(ProteinIteratorUtils, String, Double, Double)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.iterators.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/UnspecificCombinationIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">UnspecificCombinationIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/UnspecificIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators"><span class="typeNameLink">UnspecificIterator</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion.iterators</a></dt>
<dd>
<div class="block">Iterator for unspecific cleavage.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/UnspecificIterator.html#UnspecificIterator-com.compomics.util.experiment.identification.protein_sequences.digestion.ProteinIteratorUtils-java.lang.String-java.lang.Double-java.lang.Double-">UnspecificIterator(ProteinIteratorUtils, String, Double, Double)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_sequences.digestion.iterators.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/UnspecificIterator.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">UnspecificIterator</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/ZipUtils.html#unzip-java.io.File-java.io.File-com.compomics.util.waiting.WaitingHandler-">unzip(File, File, WaitingHandler)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression">ZipUtils</a></dt>
<dd>
<div class="block">Unzips the content of an archive into a given folder.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/FileDAO.html#unzipFile-java.util.zip.ZipFile-java.io.File-">unzipFile(ZipFile, File)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/FileDAO.html" title="class in com.compomics.software.autoupdater">FileDAO</a></dt>
<dd>
<div class="block">Unzips a zip archive.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/interfaces/Updateable.html#update-java.sql.Connection-">update(Connection)</a></span> - Method in interface com.compomics.util.db.interfaces.<a href="com/compomics/util/db/interfaces/Updateable.html" title="interface in com.compomics.util.db.interfaces">Updateable</a></dt>
<dd>
<div class="block">This method allows the implemented object to store its altered state in the
 persistent store.</div>
</dd>
<dt><a href="com/compomics/util/db/interfaces/Updateable.html" title="interface in com.compomics.util.db.interfaces"><span class="typeNameLink">Updateable</span></a> - Interface in <a href="com/compomics/util/db/interfaces/package-summary.html">com.compomics.util.db.interfaces</a></dt>
<dd>
<div class="block">This interface indicates that the implementing class can be updated in permanent storage.</div>
</dd>
<dt><a href="com/compomics/util/db/components/UpdateableCode.html" title="class in com.compomics.util.db.components"><span class="typeNameLink">UpdateableCode</span></a> - Class in <a href="com/compomics/util/db/components/package-summary.html">com.compomics.util.db.components</a></dt>
<dd>
<div class="block">This class generates the code to make a DBAccessor an implementation of the Updateable
 interface.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/components/UpdateableCode.html#UpdateableCode-com.compomics.util.db.DBMetaData-">UpdateableCode(DBMetaData)</a></span> - Constructor for class com.compomics.util.db.components.<a href="com/compomics/util/db/components/UpdateableCode.html" title="class in com.compomics.util.db.components">UpdateableCode</a></dt>
<dd>
<div class="block">This constructor allows the generation of the code for the implementation of
 the Updateable interface for a DBAccessor class, based on the specified metadata.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/atoms/AtomChainDialog.html#updateAtomComposition--">updateAtomComposition()</a></span> - Method in class com.compomics.util.gui.atoms.<a href="com/compomics/util/gui/atoms/AtomChainDialog.html" title="class in com.compomics.util.gui.atoms">AtomChainDialog</a></dt>
<dd>
<div class="block">Updates the atom composition panel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#updateCoverableAA-long-">updateCoverableAA(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Updates the array of booleans indicating whether the amino acids of given
 peptides can generate peptides.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html#updateEnsemblVersion-java.lang.String-java.lang.String-">updateEnsemblVersion(String, String)</a></span> - Method in class com.compomics.util.experiment.biology.genes.<a href="com/compomics/util/experiment/biology/genes/ProteinGeneDetailsProvider.html" title="class in com.compomics.util.experiment.biology.genes">ProteinGeneDetailsProvider</a></dt>
<dd>
<div class="block">Update the Ensembl version for the given species in the local map and in
 the Ensembl versions file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html#updateGUI--">updateGUI()</a></span> - Method in class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersEditionDialog</a></dt>
<dd>
<div class="block">Populates the GUI using the given identification parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html#updateIdentificationParameters--">updateIdentificationParameters()</a></span> - Method in class com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersInputBean.html" title="class in com.compomics.cli.identification_parameters">IdentificationParametersInputBean</a></dt>
<dd>
<div class="block">Updates the identification parameters according to the command line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html#updateIdentificationParameters-com.compomics.util.parameters.identification.IdentificationParameters-com.compomics.util.parameters.identification.IdentificationParameters-">updateIdentificationParameters(IdentificationParameters, IdentificationParameters)</a></span> - Method in class com.compomics.util.experiment.identification.identification_parameters.<a href="com/compomics/util/experiment/identification/identification_parameters/IdentificationParametersFactory.html" title="class in com.compomics.util.experiment.identification.identification_parameters">IdentificationParametersFactory</a></dt>
<dd>
<div class="block">Replaces old parameters by new.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PropertiesManager.html#updateLog4jConfiguration-org.apache.log4j.Logger-com.compomics.util.enumeration.CompomicsTools-">updateLog4jConfiguration(Logger, CompomicsTools)</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PropertiesManager.html" title="class in com.compomics.util.io">PropertiesManager</a></dt>
<dd>
<div class="block">This method will delete the log4j log file in the folder of the package and will create a log file in the
 CompomicsTools specific .compomics folder</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html#updateMassShifts--">updateMassShifts()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator</a></dt>
<dd>
<div class="block">Updates the mass shifts.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/modification/ModificationsDialog.html#updateModifications--">updateModifications()</a></span> - Method in class com.compomics.util.gui.modification.<a href="com/compomics/util/gui/modification/ModificationsDialog.html" title="class in com.compomics.util.gui.modification">ModificationsDialog</a></dt>
<dd>
<div class="block">Updates the modification lists and tables.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#updateNConfidentPeptides-long-">updateNConfidentPeptides(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Updates the number of confident peptides for a given protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#updateNConfidentSpectra-long-">updateNConfidentSpectra(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Updates the number of confident spectra for a given protein match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#updateNConfidentSpectraForPeptide-long-">updateNConfidentSpectraForPeptide(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Updates the number of confident spectra for a given peptide match.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#updateObservableCoverage-long-">updateObservableCoverage(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Updates the best protein coverage possible according to the given
 cleavage settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersNameDialog.html#updateParameters-com.compomics.util.parameters.identification.IdentificationParameters-">updateParameters(IdentificationParameters)</a></span> - Method in class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersNameDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersNameDialog</a></dt>
<dd>
<div class="block">Updates the given identification parameters attributes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#updatePlot--">updatePlot()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Update the plot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#updatePlot--">updatePlot()</a></span> - Method in class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>
<div class="block">Update the plot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/PropertiesManager.html#updateProperties-com.compomics.util.enumeration.CompomicsTools-java.lang.String-java.util.Properties-">updateProperties(CompomicsTools, String, Properties)</a></span> - Method in class com.compomics.util.io.<a href="com/compomics/util/io/PropertiesManager.html" title="class in com.compomics.util.io">PropertiesManager</a></dt>
<dd>
<div class="block">Update the content of a user properties instance to the ms_lims properties directory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/modification/ModificationsDialog.html#updatePtmSelection--">updatePtmSelection()</a></span> - Method in class com.compomics.util.gui.modification.<a href="com/compomics/util/gui/modification/ModificationsDialog.html" title="class in com.compomics.util.gui.modification">ModificationsDialog</a></dt>
<dd>
<div class="block">Updates the Modification selection in the default table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#updateSequenceCoverage-long-">updateSequenceCoverage(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Updates the sequence coverage of the protein of interest.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterDialog.html#updateTable--">updateTable()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterDialog.html" title="class in com.compomics.util.gui.filtering">FilterDialog</a></dt>
<dd>
<div class="block">Updates the table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersOverviewDialog.html#updateTable--">updateTable()</a></span> - Method in class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersOverviewDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersOverviewDialog</a></dt>
<dd>
<div class="block">Updates the parameters table.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pdbfinder/FindPdbForUniprotAccessions.html#urlWasRead--">urlWasRead()</a></span> - Method in class com.compomics.util.pdbfinder.<a href="com/compomics/util/pdbfinder/FindPdbForUniprotAccessions.html" title="class in com.compomics.util.pdbfinder">FindPdbForUniprotAccessions</a></dt>
<dd>
<div class="block">Returns true if the PDB URL was read, false otherwise.</div>
</dd>
<dt><a href="com/compomics/util/experiment/personalization/UrParameter.html" title="interface in com.compomics.util.experiment.personalization"><span class="typeNameLink">UrParameter</span></a> - Interface in <a href="com/compomics/util/experiment/personalization/package-summary.html">com.compomics.util.experiment.personalization</a></dt>
<dd>
<div class="block">This interface is used to reference refinement parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/ReferenceArea.html#useBoldFont--">useBoldFont()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/ReferenceArea.html" title="class in com.compomics.util.gui.spectrum">ReferenceArea</a></dt>
<dd>
<div class="block">Returns true of the label should be in bold.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#useMassDeltaCombinations--">useMassDeltaCombinations()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">If true, pairs of delta mass annotations are used when doing de novo
 sequencing.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/FolderMonitor.html#USER">USER</a></span> - Static variable in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/FolderMonitor.html" title="class in com.compomics.util.io.file">FolderMonitor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html#usersModsSorted">usersModsSorted</a></span> - Variable in class com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationFactory.html" title="class in com.compomics.util.experiment.biology.modifications">ModificationFactory</a></dt>
<dd>
<div class="block">Set to true if the users mods are sorted alphabetically.</div>
</dd>
<dt><a href="com/compomics/util/Util.html" title="class in com.compomics.util"><span class="typeNameLink">Util</span></a> - Class in <a href="com/compomics/util/package-summary.html">com.compomics.util</a></dt>
<dd>
<div class="block">Includes general help methods that are used by the other classes.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/Util.html#Util--">Util()</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/Util.html" title="class in com.compomics.util">Util</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/examples/UtilitiesDemo.html" title="class in com.compomics.util.examples"><span class="typeNameLink">UtilitiesDemo</span></a> - Class in <a href="com/compomics/util/examples/package-summary.html">com.compomics.util.examples</a></dt>
<dd>
<div class="block">This class serves as a demo of how the compomics-utilities library can be
 used in other projects.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/examples/UtilitiesDemo.html#UtilitiesDemo--">UtilitiesDemo()</a></span> - Constructor for class com.compomics.util.examples.<a href="com/compomics/util/examples/UtilitiesDemo.html" title="class in com.compomics.util.examples">UtilitiesDemo</a></dt>
<dd>
<div class="block">Creates a new UtilitiesDemo frame and makes it visible.</div>
</dd>
<dt><a href="com/compomics/util/gui/UtilitiesGUIDefaults.html" title="class in com.compomics.util.gui"><span class="typeNameLink">UtilitiesGUIDefaults</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">This class contains a list of GUI defaults to be used across the tools 
 using the utilities library to make sure that the tools have the same 
 look and feel.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/UtilitiesGUIDefaults.html#UtilitiesGUIDefaults--">UtilitiesGUIDefaults()</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/UtilitiesGUIDefaults.html" title="class in com.compomics.util.gui">UtilitiesGUIDefaults</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings"><span class="typeNameLink">UtilitiesPathParameters</span></a> - Class in <a href="com/compomics/software/settings/package-summary.html">com.compomics.software.settings</a></dt>
<dd>
<div class="block">This class sets the path preferences for the files to read/write.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#UtilitiesPathParameters--">UtilitiesPathParameters()</a></span> - Constructor for class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/software/settings/UtilitiesPathParameters.UtilitiesPathKey.html" title="enum in com.compomics.software.settings"><span class="typeNameLink">UtilitiesPathParameters.UtilitiesPathKey</span></a> - Enum in <a href="com/compomics/software/settings/package-summary.html">com.compomics.software.settings</a></dt>
<dd>
<div class="block">Enum of the paths which can be set in utilities.</div>
</dd>
<dt><a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters"><span class="typeNameLink">UtilitiesUserParameters</span></a> - Class in <a href="com/compomics/util/parameters/package-summary.html">com.compomics.util.parameters</a></dt>
<dd>
<div class="block">Utilities user parameters can be used to store user parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/UtilitiesUserParameters.html#UtilitiesUserParameters--">UtilitiesUserParameters()</a></span> - Constructor for class com.compomics.util.parameters.<a href="com/compomics/util/parameters/UtilitiesUserParameters.html" title="class in com.compomics.util.parameters">UtilitiesUserParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
</dl>
<a name="I:V">
<!--   -->
</a>
<h2 class="title">V</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#V">V</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/PrideXmlValidator.html#validate-java.io.File-">validate(File)</a></span> - Method in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/PrideXmlValidator.html" title="class in com.compomics.util.pride.validation">PrideXmlValidator</a></dt>
<dd>
<div class="block">Returns true of the given PRIDE XML file is valid.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/GuiUtilities.html#validateDoubleInput-java.awt.Component-javax.swing.JLabel-javax.swing.JTextField-java.lang.String-java.lang.String-boolean-boolean-boolean-">validateDoubleInput(Component, JLabel, JTextField, String, String, boolean, boolean, boolean)</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/GuiUtilities.html" title="class in com.compomics.util.gui">GuiUtilities</a></dt>
<dd>
<div class="block">Validate double input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html#validatedSequenceCoverageInCache-long-">validatedSequenceCoverageInCache(long)</a></span> - Method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesGenerator.html" title="class in com.compomics.util.experiment.identification">IdentificationFeaturesGenerator</a></dt>
<dd>
<div class="block">Indicates whether the sequence coverage is in cache.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/filtering/FilterDialog.html#validateInput--">validateInput()</a></span> - Method in class com.compomics.util.gui.filtering.<a href="com/compomics/util/gui/filtering/FilterDialog.html" title="class in com.compomics.util.gui.filtering">FilterDialog</a></dt>
<dd>
<div class="block">Validates the input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ModificationLocalizationParametersDialog.html#validateInput--">validateInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ModificationLocalizationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ModificationLocalizationParametersDialog</a></dt>
<dd>
<div class="block">Validates the user input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/SequenceMatchingParametersDialog.html#validateInput--">validateInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/SequenceMatchingParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">SequenceMatchingParametersDialog</a></dt>
<dd>
<div class="block">Validates the user input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationParametersDialog.html#validateInput--">validateInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationParametersDialog</a></dt>
<dd>
<div class="block">Validates the user input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html#validateInput-boolean-">validateInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/AndromedaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">AndromedaParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameter validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html#validateInput-boolean-">validateInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/CometParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">CometParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameter validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html#validateInput-boolean-">validateInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MsAmandaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MsAmandaParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameters validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html#validateInput-boolean-">validateInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MsgfParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MsgfParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameters validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html#validateInput-boolean-">validateInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/MyriMatchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">MyriMatchParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameters validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html#validateInput-boolean-">validateInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/OmssaParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">OmssaParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameters validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html#validateInput-boolean-">validateInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/TideParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">TideParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameter validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html#validateInput-boolean-">validateInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">XTandemParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameters validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html#validateInput--">validateInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersEditionDialog</a></dt>
<dd>
<div class="block">Validates the user input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersNameDialog.html#validateInput--">validateInput()</a></span> - Method in class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersNameDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersNameDialog</a></dt>
<dd>
<div class="block">Validates the user input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/tools/ProcessingParametersDialog.html#validateInput--">validateInput()</a></span> - Method in class com.compomics.util.gui.parameters.tools.<a href="com/compomics/util/gui/parameters/tools/ProcessingParametersDialog.html" title="class in com.compomics.util.gui.parameters.tools">ProcessingParametersDialog</a></dt>
<dd>
<div class="block">Validates the user input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/GuiUtilities.html#validateIntegerInput-java.awt.Component-javax.swing.JLabel-javax.swing.JTextField-java.lang.String-java.lang.String-boolean-boolean-boolean-">validateIntegerInput(Component, JLabel, JTextField, String, String, boolean, boolean, boolean)</a></span> - Static method in class com.compomics.util.gui.<a href="com/compomics/util/gui/GuiUtilities.html" title="class in com.compomics.util.gui">GuiUtilities</a></dt>
<dd>
<div class="block">Validate integer input.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#validateModifications-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.parameters.identification.search.ModificationParameters-">validateModifications(Peptide, SequenceMatchingParameters, SequenceMatchingParameters, ModificationParameters)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Verifies that the definition of every modification name is available.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html#validateParametersInput-boolean-">validateParametersInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/DirecTagParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">DirecTagParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameters validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html#validateParametersInput-boolean-">validateParametersInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/NovorParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">NovorParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameters validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html#validateParametersInput-boolean-">validateParametersInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/PepNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">PepNovoParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameters validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html#validateParametersInput-boolean-">validateParametersInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/PNovoParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">PNovoParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameters validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html#validateParametersInput-boolean-">validateParametersInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.<a href="com/compomics/util/gui/parameters/identification/IdentificationParametersEditionDialog.html" title="class in com.compomics.util.gui.parameters.identification">IdentificationParametersEditionDialog</a></dt>
<dd>
<div class="block">Inspects the search parameter validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html#validateParametersInput-boolean-">validateParametersInput(boolean)</a></span> - Method in class com.compomics.util.gui.parameters.identification.search.<a href="com/compomics/util/gui/parameters/identification/search/SearchParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.search">SearchParametersDialog</a></dt>
<dd>
<div class="block">Inspects the parameters validity.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#validatePeptide-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.parameters.identification.search.DigestionParameters-">validatePeptide(Peptide, SequenceMatchingParameters, DigestionParameters)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Validates the peptide based on the peptide length, the share of X's in
 the sequence and the allowed number of missed cleavages.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#validatePrecursor-com.compomics.util.experiment.identification.spectrum_assumptions.PeptideAssumption-java.lang.String-com.compomics.util.experiment.mass_spectrometry.SpectrumFactory-com.compomics.util.parameters.identification.search.SearchParameters-">validatePrecursor(PeptideAssumption, String, SpectrumFactory, SearchParameters)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Validates the mass deviation of a peptide assumption.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html#validateProteins-com.compomics.util.experiment.biology.proteins.Peptide-com.compomics.util.parameters.identification.advanced.SequenceMatchingParameters-com.compomics.util.experiment.io.biology.protein.SequenceProvider-">validateProteins(Peptide, SequenceMatchingParameters, SequenceProvider)</a></span> - Method in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/PeptideAssumptionFilter.html" title="class in com.compomics.util.experiment.identification.filtering">PeptideAssumptionFilter</a></dt>
<dd>
<div class="block">Validates a peptide depending on its protein inference status.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced"><span class="typeNameLink">ValidationParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/advanced/package-summary.html">com.compomics.util.gui.parameters.identification.advanced</a></dt>
<dd>
<div class="block">Dialog for the edition of the sequence matching settings</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationParametersDialog.html#ValidationParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.IdMatchValidationParameters-boolean-">ValidationParametersDialog(Frame, IdMatchValidationParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationParametersDialog</a></dt>
<dd>
<div class="block">Creates a new ValidationSettingsDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationParametersDialog.html#ValidationParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.IdMatchValidationParameters-boolean-">ValidationParametersDialog(Dialog, Frame, IdMatchValidationParameters, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationParametersDialog</a></dt>
<dd>
<div class="block">Creates a new ValidationSettingsDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced"><span class="typeNameLink">ValidationQcParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/advanced/package-summary.html">com.compomics.util.parameters.identification.advanced</a></dt>
<dd>
<div class="block">This class lists the criteria used for quality control of the validated
 matches.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#ValidationQcParameters--">ValidationQcParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Creates a validation quality control preferences object with default
 settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html#ValidationQcParameters-com.compomics.util.parameters.identification.advanced.ValidationQcParameters-">ValidationQcParameters(ValidationQcParameters)</a></span> - Constructor for class com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/ValidationQcParameters.html" title="class in com.compomics.util.parameters.identification.advanced">ValidationQcParameters</a></dt>
<dd>
<div class="block">Creates a validation quality control preferences object based on an other
 ValidationQCPreferences.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced"><span class="typeNameLink">ValidationQCParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/advanced/package-summary.html">com.compomics.util.gui.parameters.identification.advanced</a></dt>
<dd>
<div class="block">Dialog for the edition of validation QC filters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html#ValidationQCParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.ValidationQcParameters-java.util.ArrayList-boolean-">ValidationQCParametersDialog(Frame, ValidationQcParameters, ArrayList&lt;String&gt;, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationQCParametersDialog</a></dt>
<dd>
<div class="block">Creates a new ValidationQCPreferencesDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html#ValidationQCParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.advanced.ValidationQcParameters-java.util.ArrayList-boolean-">ValidationQCParametersDialog(Dialog, Frame, ValidationQcParameters, ArrayList&lt;String&gt;, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.advanced.<a href="com/compomics/util/gui/parameters/identification/advanced/ValidationQCParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.advanced">ValidationQCParametersDialog</a></dt>
<dd>
<div class="block">Creates a new ValidationQCPreferencesDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Valine</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Valine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html#Valine--">Valine()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Valine.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Valine</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html#VALINE">VALINE</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/ImmoniumIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">ImmoniumIon</a></dt>
<dd>
<div class="block">Valine immonium ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIParams.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIParams.html" title="enum in com.compomics.cli.enzymes">EnzymesCLIParams</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/fasta/FastaParametersCLIParams.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.cli.fasta.<a href="com/compomics/cli/fasta/FastaParametersCLIParams.html" title="enum in com.compomics.cli.fasta">FastaParametersCLIParams</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html" title="enum in com.compomics.cli.identification_parameters">IdentificationParametersCLIParams</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIParams.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIParams.html" title="enum in com.compomics.cli.modifications">ModificationsCLIParams</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.UtilitiesPathKey.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.UtilitiesPathKey.html" title="enum in com.compomics.software.settings">UtilitiesPathParameters.UtilitiesPathKey</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/enumeration/CompomicsTools.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.enumeration.<a href="com/compomics/util/enumeration/CompomicsTools.html" title="enum in com.compomics.util.enumeration">CompomicsTools</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/enumeration/ImageType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.enumeration.<a href="com/compomics/util/enumeration/ImageType.html" title="enum in com.compomics.util.enumeration">ImageType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/enumeration/MolecularElement.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.enumeration.<a href="com/compomics/util/enumeration/MolecularElement.html" title="enum in com.compomics.util.enumeration">MolecularElement</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/enumeration/OperatingSystemEnum.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.enumeration.<a href="com/compomics/util/enumeration/OperatingSystemEnum.html" title="enum in com.compomics.util.enumeration">OperatingSystemEnum</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.Property.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.Property.html" title="enum in com.compomics.util.experiment.biology.aminoacids">AminoAcid.Property</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.IonType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.IonType.html" title="enum in com.compomics.util.experiment.biology.ions">Ion.IonType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationType.html" title="enum in com.compomics.util.experiment.biology.modifications">ModificationType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.EnsemblGenomeDivision.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.EnsemblGenomeDivision.html" title="enum in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblGenomesSpecies.EnsemblGenomeDivision</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItemComparator.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItemComparator.html" title="enum in com.compomics.util.experiment.filtering">FilterItemComparator</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.AdvocateType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.AdvocateType.html" title="enum in com.compomics.util.experiment.identification">Advocate.AdvocateType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">AssumptionFilterItem</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PeptideFilterItem</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">ProteinFilterItem</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PsmFilterItem</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.FilterType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.FilterType.html" title="enum in com.compomics.util.experiment.identification.filtering">MatchFilter.FilterType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.ObjectType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.ObjectType.html" title="enum in com.compomics.util.experiment.identification">IdentificationFeaturesCache.ObjectType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationMatch.MatchType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationMatch.MatchType.html" title="enum in com.compomics.util.experiment.identification">IdentificationMatch.MatchType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html" title="enum in com.compomics.util.experiment.identification.matches">IonMatch.MzErrorType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html" title="enum in com.compomics.util.experiment.identification.modification">ModificationLocalizationScore</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.Relationship.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.Relationship.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature.Relationship</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.Function.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.Function.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AASequenceFeature.Function</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.Property.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">IonFeature.Property</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.Property.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ModificationFeature.Property</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.Property.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideFeature.Property</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/PeptideFragmentationModel.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.peptide_fragmentation.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/PeptideFragmentationModel.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation">PeptideFragmentationModel</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html" title="enum in com.compomics.util.experiment.identification.psm_scoring">PsmScore</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.IntensityThresholdType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.IntensityThresholdType.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters.IntensityThresholdType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.IonSeries.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.IonSeries.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">SimplePeptideAnnotator.IonSeries</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.TiesResolution.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.TiesResolution.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator.TiesResolution</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html" title="enum in com.compomics.util.experiment.io.biology.protein">ProteinDatabase</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.RawValueConversionType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.RawValueConversionType.html" title="enum in com.compomics.util.experiment.io.identification.idfilereaders">MzIdentMLIdfileReader.RawValueConversionType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html" title="enum in com.compomics.util.experiment.io.identification">MzIdentMLVersion</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/parameters/MarshallableParameter.Type.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.io.parameters.<a href="com/compomics/util/experiment/io/parameters/MarshallableParameter.Type.html" title="enum in com.compomics.util.experiment.io.parameters">MarshallableParameter.Type</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/FragmentationMethod.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/FragmentationMethod.html" title="enum in com.compomics.util.experiment.mass_spectrometry">FragmentationMethod</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsFormat</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">ProteoWizardFilter</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/utils/StandardMasses.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.mass_spectrometry.utils.<a href="com/compomics/util/experiment/mass_spectrometry/utils/StandardMasses.html" title="enum in com.compomics.util.experiment.mass_spectrometry.utils">StandardMasses</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/Quantification.QuantificationMethod.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.quantification.<a href="com/compomics/util/experiment/quantification/Quantification.QuantificationMethod.html" title="enum in com.compomics.util.experiment.quantification">Quantification.QuantificationMethod</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/spectrumcounting/SpectrumCountingMethod.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.quantification.spectrumcounting.<a href="com/compomics/util/experiment/quantification/spectrumcounting/SpectrumCountingMethod.html" title="enum in com.compomics.util.experiment.quantification.spectrumcounting">SpectrumCountingMethod</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/MetricsPrefix.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/MetricsPrefix.html" title="enum in com.compomics.util.experiment.units">MetricsPrefix</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/StandardUnit.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/StandardUnit.html" title="enum in com.compomics.util.experiment.units">StandardUnit</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.DrawingStyle.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.DrawingStyle.html" title="enum in com.compomics.util.gui.spectrum">GraphicsPanel.DrawingStyle</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.GraphicsPanelType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.GraphicsPanelType.html" title="enum in com.compomics.util.gui.spectrum">GraphicsPanel.GraphicsPanelType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.VennDiagramType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.VennDiagramType.html" title="enum in com.compomics.util.gui">VennDiagramPanel.VennDiagramType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.PlottingDialogPlotType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.PlottingDialogPlotType.html" title="enum in com.compomics.util.gui">XYPlottingDialog.PlottingDialogPlotType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFormat.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFormat.html" title="enum in com.compomics.util.io.export">ExportFormat</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/roc/DataRoc.RocInterpolation.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.math.roc.<a href="com/compomics/util/math/roc/DataRoc.RocInterpolation.html" title="enum in com.compomics.util.math.roc">DataRoc.RocInterpolation</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.FeedBackType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.FeedBackType.html" title="enum in com.compomics.util.messages">FeedBack.FeedBackType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html" title="enum in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters.MatchingType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.CleavageParameter</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.Specificity</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.MassAccuracyType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.MassAccuracyType.html" title="enum in com.compomics.util.parameters.identification.search">SearchParameters.MassAccuracyType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.AndromedaDecoyMode.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.AndromedaDecoyMode.html" title="enum in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters.AndromedaDecoyMode</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.CometOutputFormat.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.CometOutputFormat.html" title="enum in com.compomics.util.parameters.identification.tool_specific">CometParameters.CometOutputFormat</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/peptide_shaker/ProjectType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.parameters.peptide_shaker.<a href="com/compomics/util/parameters/peptide_shaker/ProjectType.html" title="enum in com.compomics.util.parameters.peptide_shaker">ProjectType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/searchgui/OutputParameters.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.parameters.searchgui.<a href="com/compomics/util/parameters/searchgui/OutputParameters.html" title="enum in com.compomics.util.parameters.searchgui">OutputParameters</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/tools/ProcessingParameters.ProcessingType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.parameters.tools.<a href="com/compomics/util/parameters/tools/ProcessingParameters.ProcessingType.html" title="enum in com.compomics.util.parameters.tools">ProcessingParameters.ProcessingType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/webservice/file/FileType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.pride.prideobjects.webservice.file.<a href="com/compomics/util/pride/prideobjects/webservice/file/FileType.html" title="enum in com.compomics.util.pride.prideobjects.webservice.file">FileType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/webservice/PrideQuery.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.pride.prideobjects.webservice.<a href="com/compomics/util/pride/prideobjects/webservice/PrideQuery.html" title="enum in com.compomics.util.pride.prideobjects.webservice">PrideQuery</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/webservice/query/PrideFilterType.html#valueOf-java.lang.String-">valueOf(String)</a></span> - Static method in enum com.compomics.util.pride.prideobjects.webservice.query.<a href="com/compomics/util/pride/prideobjects/webservice/query/PrideFilterType.html" title="enum in com.compomics.util.pride.prideobjects.webservice.query">PrideFilterType</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/enzymes/EnzymesCLIParams.html#values--">values()</a></span> - Static method in enum com.compomics.cli.enzymes.<a href="com/compomics/cli/enzymes/EnzymesCLIParams.html" title="enum in com.compomics.cli.enzymes">EnzymesCLIParams</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/fasta/FastaParametersCLIParams.html#values--">values()</a></span> - Static method in enum com.compomics.cli.fasta.<a href="com/compomics/cli/fasta/FastaParametersCLIParams.html" title="enum in com.compomics.cli.fasta">FastaParametersCLIParams</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html#values--">values()</a></span> - Static method in enum com.compomics.cli.identification_parameters.<a href="com/compomics/cli/identification_parameters/IdentificationParametersCLIParams.html" title="enum in com.compomics.cli.identification_parameters">IdentificationParametersCLIParams</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/cli/modifications/ModificationsCLIParams.html#values--">values()</a></span> - Static method in enum com.compomics.cli.modifications.<a href="com/compomics/cli/modifications/ModificationsCLIParams.html" title="enum in com.compomics.cli.modifications">ModificationsCLIParams</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.UtilitiesPathKey.html#values--">values()</a></span> - Static method in enum com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.UtilitiesPathKey.html" title="enum in com.compomics.software.settings">UtilitiesPathParameters.UtilitiesPathKey</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/enumeration/CompomicsTools.html#values--">values()</a></span> - Static method in enum com.compomics.util.enumeration.<a href="com/compomics/util/enumeration/CompomicsTools.html" title="enum in com.compomics.util.enumeration">CompomicsTools</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/enumeration/ImageType.html#values--">values()</a></span> - Static method in enum com.compomics.util.enumeration.<a href="com/compomics/util/enumeration/ImageType.html" title="enum in com.compomics.util.enumeration">ImageType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/enumeration/MolecularElement.html#values--">values()</a></span> - Static method in enum com.compomics.util.enumeration.<a href="com/compomics/util/enumeration/MolecularElement.html" title="enum in com.compomics.util.enumeration">MolecularElement</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/enumeration/OperatingSystemEnum.html#values--">values()</a></span> - Static method in enum com.compomics.util.enumeration.<a href="com/compomics/util/enumeration/OperatingSystemEnum.html" title="enum in com.compomics.util.enumeration">OperatingSystemEnum</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.Property.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.Property.html" title="enum in com.compomics.util.experiment.biology.aminoacids">AminoAcid.Property</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/Ion.IonType.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.biology.ions.<a href="com/compomics/util/experiment/biology/ions/Ion.IonType.html" title="enum in com.compomics.util.experiment.biology.ions">Ion.IonType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/modifications/ModificationType.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.biology.modifications.<a href="com/compomics/util/experiment/biology/modifications/ModificationType.html" title="enum in com.compomics.util.experiment.biology.modifications">ModificationType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.EnsemblGenomeDivision.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.biology.taxonomy.mappings.<a href="com/compomics/util/experiment/biology/taxonomy/mappings/EnsemblGenomesSpecies.EnsemblGenomeDivision.html" title="enum in com.compomics.util.experiment.biology.taxonomy.mappings">EnsemblGenomesSpecies.EnsemblGenomeDivision</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/filtering/FilterItemComparator.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.filtering.<a href="com/compomics/util/experiment/filtering/FilterItemComparator.html" title="enum in com.compomics.util.experiment.filtering">FilterItemComparator</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.AdvocateType.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.AdvocateType.html" title="enum in com.compomics.util.experiment.identification">Advocate.AdvocateType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#values--">values()</a></span> - Static method in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">Returns the implemented advocates in an array.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/AssumptionFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">AssumptionFilterItem</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PeptideFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PeptideFilterItem</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/ProteinFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">ProteinFilterItem</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.items.<a href="com/compomics/util/experiment/identification/filtering/items/PsmFilterItem.html" title="enum in com.compomics.util.experiment.identification.filtering.items">PsmFilterItem</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.FilterType.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.FilterType.html" title="enum in com.compomics.util.experiment.identification.filtering">MatchFilter.FilterType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.ObjectType.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationFeaturesCache.ObjectType.html" title="enum in com.compomics.util.experiment.identification">IdentificationFeaturesCache.ObjectType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/IdentificationMatch.MatchType.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/IdentificationMatch.MatchType.html" title="enum in com.compomics.util.experiment.identification">IdentificationMatch.MatchType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.matches.<a href="com/compomics/util/experiment/identification/matches/IonMatch.MzErrorType.html" title="enum in com.compomics.util.experiment.identification.matches">IonMatch.MzErrorType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.modification.<a href="com/compomics/util/experiment/identification/modification/ModificationLocalizationScore.html" title="enum in com.compomics.util.experiment.identification.modification">ModificationLocalizationScore</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.Relationship.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/AAPropertyRelationshipFeature.Relationship.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AAPropertyRelationshipFeature.Relationship</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.Function.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/AASequenceFeature.Function.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AASequenceFeature.Function</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.Property.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/IonFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">IonFeature.Property</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.Property.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/ModificationFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">ModificationFeature.Property</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.Property.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/PeptideFeature.Property.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">PeptideFeature.Property</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_fragmentation/PeptideFragmentationModel.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.peptide_fragmentation.<a href="com/compomics/util/experiment/identification/peptide_fragmentation/PeptideFragmentationModel.html" title="enum in com.compomics.util.experiment.identification.peptide_fragmentation">PeptideFragmentationModel</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.psm_scoring.<a href="com/compomics/util/experiment/identification/psm_scoring/PsmScore.html" title="enum in com.compomics.util.experiment.identification.psm_scoring">PsmScore</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.IntensityThresholdType.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.IntensityThresholdType.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters.IntensityThresholdType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.IonSeries.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators.<a href="com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/SimplePeptideAnnotator.IonSeries.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">SimplePeptideAnnotator.IonSeries</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.TiesResolution.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/SpectrumAnnotator.TiesResolution.html" title="enum in com.compomics.util.experiment.identification.spectrum_annotation">SpectrumAnnotator.TiesResolution</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.identification.validation.<a href="com/compomics/util/experiment/identification/validation/MatchValidationLevel.html" title="enum in com.compomics.util.experiment.identification.validation">MatchValidationLevel</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.io.biology.protein.<a href="com/compomics/util/experiment/io/biology/protein/ProteinDatabase.html" title="enum in com.compomics.util.experiment.io.biology.protein">ProteinDatabase</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.RawValueConversionType.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/MzIdentMLIdfileReader.RawValueConversionType.html" title="enum in com.compomics.util.experiment.io.identification.idfilereaders">MzIdentMLIdfileReader.RawValueConversionType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.io.identification.<a href="com/compomics/util/experiment/io/identification/MzIdentMLVersion.html" title="enum in com.compomics.util.experiment.io.identification">MzIdentMLVersion</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/parameters/MarshallableParameter.Type.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.io.parameters.<a href="com/compomics/util/experiment/io/parameters/MarshallableParameter.Type.html" title="enum in com.compomics.util.experiment.io.parameters">MarshallableParameter.Type</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/FragmentationMethod.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/FragmentationMethod.html" title="enum in com.compomics.util.experiment.mass_spectrometry">FragmentationMethod</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/MsFormat.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">MsFormat</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.mass_spectrometry.proteowizard.<a href="com/compomics/util/experiment/mass_spectrometry/proteowizard/ProteoWizardFilter.html" title="enum in com.compomics.util.experiment.mass_spectrometry.proteowizard">ProteoWizardFilter</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/utils/StandardMasses.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.mass_spectrometry.utils.<a href="com/compomics/util/experiment/mass_spectrometry/utils/StandardMasses.html" title="enum in com.compomics.util.experiment.mass_spectrometry.utils">StandardMasses</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/Quantification.QuantificationMethod.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.quantification.<a href="com/compomics/util/experiment/quantification/Quantification.QuantificationMethod.html" title="enum in com.compomics.util.experiment.quantification">Quantification.QuantificationMethod</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/quantification/spectrumcounting/SpectrumCountingMethod.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.quantification.spectrumcounting.<a href="com/compomics/util/experiment/quantification/spectrumcounting/SpectrumCountingMethod.html" title="enum in com.compomics.util.experiment.quantification.spectrumcounting">SpectrumCountingMethod</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/MetricsPrefix.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/MetricsPrefix.html" title="enum in com.compomics.util.experiment.units">MetricsPrefix</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/units/StandardUnit.html#values--">values()</a></span> - Static method in enum com.compomics.util.experiment.units.<a href="com/compomics/util/experiment/units/StandardUnit.html" title="enum in com.compomics.util.experiment.units">StandardUnit</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.DrawingStyle.html#values--">values()</a></span> - Static method in enum com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.DrawingStyle.html" title="enum in com.compomics.util.gui.spectrum">GraphicsPanel.DrawingStyle</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.GraphicsPanelType.html#values--">values()</a></span> - Static method in enum com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.GraphicsPanelType.html" title="enum in com.compomics.util.gui.spectrum">GraphicsPanel.GraphicsPanelType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.VennDiagramType.html#values--">values()</a></span> - Static method in enum com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.VennDiagramType.html" title="enum in com.compomics.util.gui">VennDiagramPanel.VennDiagramType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.PlottingDialogPlotType.html#values--">values()</a></span> - Static method in enum com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.PlottingDialogPlotType.html" title="enum in com.compomics.util.gui">XYPlottingDialog.PlottingDialogPlotType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportFormat.html#values--">values()</a></span> - Static method in enum com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportFormat.html" title="enum in com.compomics.util.io.export">ExportFormat</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/roc/DataRoc.RocInterpolation.html#values--">values()</a></span> - Static method in enum com.compomics.util.math.roc.<a href="com/compomics/util/math/roc/DataRoc.RocInterpolation.html" title="enum in com.compomics.util.math.roc">DataRoc.RocInterpolation</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/messages/FeedBack.FeedBackType.html#values--">values()</a></span> - Static method in enum com.compomics.util.messages.<a href="com/compomics/util/messages/FeedBack.FeedBackType.html" title="enum in com.compomics.util.messages">FeedBack.FeedBackType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html#values--">values()</a></span> - Static method in enum com.compomics.util.parameters.identification.advanced.<a href="com/compomics/util/parameters/identification/advanced/SequenceMatchingParameters.MatchingType.html" title="enum in com.compomics.util.parameters.identification.advanced">SequenceMatchingParameters.MatchingType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html#values--">values()</a></span> - Static method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.CleavageParameter.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.CleavageParameter</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html#values--">values()</a></span> - Static method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/DigestionParameters.Specificity.html" title="enum in com.compomics.util.parameters.identification.search">DigestionParameters.Specificity</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.MassAccuracyType.html#values--">values()</a></span> - Static method in enum com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.MassAccuracyType.html" title="enum in com.compomics.util.parameters.identification.search">SearchParameters.MassAccuracyType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.AndromedaDecoyMode.html#values--">values()</a></span> - Static method in enum com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/AndromedaParameters.AndromedaDecoyMode.html" title="enum in com.compomics.util.parameters.identification.tool_specific">AndromedaParameters.AndromedaDecoyMode</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.CometOutputFormat.html#values--">values()</a></span> - Static method in enum com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/CometParameters.CometOutputFormat.html" title="enum in com.compomics.util.parameters.identification.tool_specific">CometParameters.CometOutputFormat</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/peptide_shaker/ProjectType.html#values--">values()</a></span> - Static method in enum com.compomics.util.parameters.peptide_shaker.<a href="com/compomics/util/parameters/peptide_shaker/ProjectType.html" title="enum in com.compomics.util.parameters.peptide_shaker">ProjectType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/searchgui/OutputParameters.html#values--">values()</a></span> - Static method in enum com.compomics.util.parameters.searchgui.<a href="com/compomics/util/parameters/searchgui/OutputParameters.html" title="enum in com.compomics.util.parameters.searchgui">OutputParameters</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/tools/ProcessingParameters.ProcessingType.html#values--">values()</a></span> - Static method in enum com.compomics.util.parameters.tools.<a href="com/compomics/util/parameters/tools/ProcessingParameters.ProcessingType.html" title="enum in com.compomics.util.parameters.tools">ProcessingParameters.ProcessingType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/webservice/file/FileType.html#values--">values()</a></span> - Static method in enum com.compomics.util.pride.prideobjects.webservice.file.<a href="com/compomics/util/pride/prideobjects/webservice/file/FileType.html" title="enum in com.compomics.util.pride.prideobjects.webservice.file">FileType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/webservice/PrideQuery.html#values--">values()</a></span> - Static method in enum com.compomics.util.pride.prideobjects.webservice.<a href="com/compomics/util/pride/prideobjects/webservice/PrideQuery.html" title="enum in com.compomics.util.pride.prideobjects.webservice">PrideQuery</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/prideobjects/webservice/query/PrideFilterType.html#values--">values()</a></span> - Static method in enum com.compomics.util.pride.prideobjects.webservice.query.<a href="com/compomics/util/pride/prideobjects/webservice/query/PrideFilterType.html" title="enum in com.compomics.util.pride.prideobjects.webservice.query">PrideFilterType</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html#valuesMap">valuesMap</a></span> - Variable in class com.compomics.util.experiment.identification.filtering.<a href="com/compomics/util/experiment/identification/filtering/MatchFilter.html" title="class in com.compomics.util.experiment.identification.filtering">MatchFilter</a></dt>
<dd>
<div class="block">Map of the values to filter on.</div>
</dd>
<dt><a href="com/compomics/util/experiment/biology/variants/Variant.html" title="interface in com.compomics.util.experiment.biology.variants"><span class="typeNameLink">Variant</span></a> - Interface in <a href="com/compomics/util/experiment/biology/variants/package-summary.html">com.compomics.util.experiment.biology.variants</a></dt>
<dd>
<div class="block">Interface for a variant.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#variant">variant</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">Type of edit operation, either deletion 'd', substitution 's' or
 insertion 'i'.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/variants/AminoAcidSubstitution.html#variantAminoAcid">variantAminoAcid</a></span> - Variable in class com.compomics.util.experiment.biology.variants.<a href="com/compomics/util/experiment/biology/variants/AminoAcidSubstitution.html" title="class in com.compomics.util.experiment.biology.variants">AminoAcidSubstitution</a></dt>
<dd>
<div class="block">The variant amino acid represented by its single letter code.</div>
</dd>
<dt><a href="com/compomics/util/math/VennDiagram.html" title="class in com.compomics.util.math"><span class="typeNameLink">VennDiagram</span></a> - Class in <a href="com/compomics/util/math/package-summary.html">com.compomics.util.math</a></dt>
<dd>
<div class="block">Class for calculating the groups for Venn diagrams.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/VennDiagram.html#VennDiagram--">VennDiagram()</a></span> - Constructor for class com.compomics.util.math.<a href="com/compomics/util/math/VennDiagram.html" title="class in com.compomics.util.math">VennDiagram</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/VennDiagramDialog.html" title="class in com.compomics.util.gui"><span class="typeNameLink">VennDiagramDialog</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">A dialog for creating and displaying Venn diagrams.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramDialog.html#VennDiagramDialog-java.awt.Frame-java.awt.Image-java.awt.Image-boolean-">VennDiagramDialog(Frame, Image, Image, boolean)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramDialog.html" title="class in com.compomics.util.gui">VennDiagramDialog</a></dt>
<dd>
<div class="block">Creates a new XYPlottingDialog.</div>
</dd>
<dt><a href="com/compomics/util/gui/VennDiagramLegendLocationDialog.html" title="class in com.compomics.util.gui"><span class="typeNameLink">VennDiagramLegendLocationDialog</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">Dialog for setting the location of the legend items in a Venn diagram.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramLegendLocationDialog.html#VennDiagramLegendLocationDialog-com.compomics.util.gui.VennDiagramDialog-boolean-">VennDiagramLegendLocationDialog(VennDiagramDialog, boolean)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramLegendLocationDialog.html" title="class in com.compomics.util.gui">VennDiagramLegendLocationDialog</a></dt>
<dd>
<div class="block">Creates a new VennDiagramLegendLocationDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/math/VennDiagram.html#vennDiagramMaker-java.util.ArrayList-java.util.ArrayList-java.util.ArrayList-java.util.ArrayList-">vennDiagramMaker(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, ArrayList&lt;String&gt;)</a></span> - Static method in class com.compomics.util.math.<a href="com/compomics/util/math/VennDiagram.html" title="class in com.compomics.util.math">VennDiagram</a></dt>
<dd>
<div class="block">Create the Venn diagram groupings based on the provided data.</div>
</dd>
<dt><a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui"><span class="typeNameLink">VennDiagramPanel</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">A panel displaying a Venn diagram based on the provided data.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VennDiagramPanel.html#VennDiagramPanel-java.util.ArrayList-java.util.ArrayList-java.util.ArrayList-java.util.ArrayList-java.lang.String-java.lang.String-java.lang.String-java.lang.String-java.awt.Color-java.awt.Color-java.awt.Color-java.awt.Color-">VennDiagramPanel(ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, ArrayList&lt;String&gt;, String, String, String, String, Color, Color, Color, Color)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/VennDiagramPanel.html" title="class in com.compomics.util.gui">VennDiagramPanel</a></dt>
<dd>
<div class="block">Creates a new VennDiagramPanel.</div>
</dd>
<dt><a href="com/compomics/util/gui/VennDiagramPanel.VennDiagramType.html" title="enum in com.compomics.util.gui"><span class="typeNameLink">VennDiagramPanel.VennDiagramType</span></a> - Enum in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">The supported Venn diagram types.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/PrideXmlValidator.html#VERIFIER_FACTORY">VERIFIER_FACTORY</a></span> - Static variable in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/PrideXmlValidator.html" title="class in com.compomics.util.pride.validation">PrideXmlValidator</a></dt>
<dd>
<div class="block">The verifier factory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/parameters/DummyParameters.html#version">version</a></span> - Variable in class com.compomics.util.experiment.io.parameters.<a href="com/compomics/util/experiment/io/parameters/DummyParameters.html" title="class in com.compomics.util.experiment.io.parameters">DummyParameters</a></dt>
<dd>
<div class="block">Version number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/IdentificationParameters.html#version">version</a></span> - Variable in class com.compomics.util.parameters.identification.<a href="com/compomics/util/parameters/identification/IdentificationParameters.html" title="class in com.compomics.util.parameters.identification">IdentificationParameters</a></dt>
<dd>
<div class="block">Version number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/search/SearchParameters.html#version">version</a></span> - Variable in class com.compomics.util.parameters.identification.search.<a href="com/compomics/util/parameters/identification/search/SearchParameters.html" title="class in com.compomics.util.parameters.identification.search">SearchParameters</a></dt>
<dd>
<div class="block">Version number.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html#versionTag">versionTag</a></span> - Static variable in class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/OnyaseIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">OnyaseIdfileReader</a></dt>
<dd>
<div class="block">Tag for the version.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html#VERY_CONFIDENT">VERY_CONFIDENT</a></span> - Static variable in class com.compomics.util.experiment.identification.peptide_shaker.<a href="com/compomics/util/experiment/identification/peptide_shaker/ModificationScoring.html" title="class in com.compomics.util.experiment.identification.peptide_shaker">ModificationScoring</a></dt>
<dd>
<div class="block">Index for a very confident assignment.</div>
</dd>
<dt><a href="com/compomics/util/gui/VisibleTableColumnsDialog.html" title="class in com.compomics.util.gui"><span class="typeNameLink">VisibleTableColumnsDialog</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">A dialog that lets the user decide which columns to show/hide in a JTable.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VisibleTableColumnsDialog.html#VisibleTableColumnsDialog-javax.swing.JDialog-com.compomics.util.gui.VisibleTableColumnsDialogParent-boolean-">VisibleTableColumnsDialog(JDialog, VisibleTableColumnsDialogParent, boolean)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/VisibleTableColumnsDialog.html" title="class in com.compomics.util.gui">VisibleTableColumnsDialog</a></dt>
<dd>
<div class="block">Creates a new VisibleTableColumnsDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/VisibleTableColumnsDialog.html#VisibleTableColumnsDialog-javax.swing.JFrame-com.compomics.util.gui.VisibleTableColumnsDialogParent-boolean-">VisibleTableColumnsDialog(JFrame, VisibleTableColumnsDialogParent, boolean)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/VisibleTableColumnsDialog.html" title="class in com.compomics.util.gui">VisibleTableColumnsDialog</a></dt>
<dd>
<div class="block">Creates a new VisibleTableColumnsDialog.</div>
</dd>
<dt><a href="com/compomics/util/gui/VisibleTableColumnsDialogParent.html" title="interface in com.compomics.util.gui"><span class="typeNameLink">VisibleTableColumnsDialogParent</span></a> - Interface in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">Interface for parents of VisibleTableColumnsDialog.</div>
</dd>
</dl>
<a name="I:W">
<!--   -->
</a>
<h2 class="title">W</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#W">W</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/waiting/WaitingActionListener.html" title="interface in com.compomics.util.waiting"><span class="typeNameLink">WaitingActionListener</span></a> - Interface in <a href="com/compomics/util/waiting/package-summary.html">com.compomics.util.waiting</a></dt>
<dd>
<div class="block">This class is used to pass actions like cancel while waiting.</div>
</dd>
<dt><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers"><span class="typeNameLink">WaitingDialog</span></a> - Class in <a href="com/compomics/util/gui/waiting/waitinghandlers/package-summary.html">com.compomics.util.gui.waiting.waitinghandlers</a></dt>
<dd>
<div class="block">A dialog displaying progress details.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#WaitingDialog-java.awt.Frame-java.awt.Image-java.awt.Image-boolean-java.lang.String-java.lang.String-java.lang.String-boolean-">WaitingDialog(Frame, Image, Image, boolean, String, String, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Creates a new WaitingDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html#WaitingDialog-java.awt.Frame-java.awt.Image-java.awt.Image-boolean-java.util.ArrayList-java.lang.String-java.lang.String-java.lang.String-boolean-">WaitingDialog(Frame, Image, Image, boolean, ArrayList&lt;String&gt;, String, String, String, boolean)</a></span> - Constructor for class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingDialog.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingDialog</a></dt>
<dd>
<div class="block">Creates a new WaitingDialog.</div>
</dd>
<dt><a href="com/compomics/util/exceptions/exception_handlers/WaitingDialogExceptionHandler.html" title="class in com.compomics.util.exceptions.exception_handlers"><span class="typeNameLink">WaitingDialogExceptionHandler</span></a> - Class in <a href="com/compomics/util/exceptions/exception_handlers/package-summary.html">com.compomics.util.exceptions.exception_handlers</a></dt>
<dd>
<div class="block">Exception handler for processes making use of a waiting dialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/exception_handlers/WaitingDialogExceptionHandler.html#WaitingDialogExceptionHandler-com.compomics.util.gui.waiting.waitinghandlers.WaitingDialog-java.lang.String-">WaitingDialogExceptionHandler(WaitingDialog, String)</a></span> - Constructor for class com.compomics.util.exceptions.exception_handlers.<a href="com/compomics/util/exceptions/exception_handlers/WaitingDialogExceptionHandler.html" title="class in com.compomics.util.exceptions.exception_handlers">WaitingDialogExceptionHandler</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/exceptions/exception_handlers/WaitingDialogExceptionHandler.html#WaitingDialogExceptionHandler-com.compomics.util.gui.waiting.waitinghandlers.WaitingDialog-">WaitingDialogExceptionHandler(WaitingDialog)</a></span> - Constructor for class com.compomics.util.exceptions.exception_handlers.<a href="com/compomics/util/exceptions/exception_handlers/WaitingDialogExceptionHandler.html" title="class in com.compomics.util.exceptions.exception_handlers">WaitingDialogExceptionHandler</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/waiting/WaitingHandler.html" title="interface in com.compomics.util.waiting"><span class="typeNameLink">WaitingHandler</span></a> - Interface in <a href="com/compomics/util/waiting/package-summary.html">com.compomics.util.waiting</a></dt>
<dd>
<div class="block">An interface for code dealing with how to handle information that is
 displayed to the user during loading or processing.</div>
</dd>
<dt><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers"><span class="typeNameLink">WaitingHandlerCLIImpl</span></a> - Class in <a href="com/compomics/util/gui/waiting/waitinghandlers/package-summary.html">com.compomics.util.gui.waiting.waitinghandlers</a></dt>
<dd>
<div class="block">This class is an implementation of the WaitingHandler interface to be used
 when operating through the Command Line Interface.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html#WaitingHandlerCLIImpl--">WaitingHandlerCLIImpl()</a></span> - Constructor for class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerCLIImpl.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerCLIImpl</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers"><span class="typeNameLink">WaitingHandlerDummy</span></a> - Class in <a href="com/compomics/util/gui/waiting/waitinghandlers/package-summary.html">com.compomics.util.gui.waiting.waitinghandlers</a></dt>
<dd>
<div class="block">This class is an implementation of the WaitingHandler interface to be used
 when a process can be canceled, but no monitoring is needed.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html#WaitingHandlerDummy--">WaitingHandlerDummy()</a></span> - Constructor for class com.compomics.util.gui.waiting.waitinghandlers.<a href="com/compomics/util/gui/waiting/waitinghandlers/WaitingHandlerDummy.html" title="class in com.compomics.util.gui.waiting.waitinghandlers">WaitingHandlerDummy</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html#warning-org.xml.sax.SAXParseException-">warning(SAXParseException)</a></span> - Method in class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html" title="class in com.compomics.util.pride.validation">XMLValidationErrorHandler</a></dt>
<dd>
<div class="block">Receive notification of a warning.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html#WATER_MASS">WATER_MASS</a></span> - Static variable in class com.compomics.util.experiment.identification.protein_sequences.digestion.<a href="com/compomics/util/experiment/identification/protein_sequences/digestion/ProteinIteratorUtils.html" title="class in com.compomics.util.experiment.identification.protein_sequences.digestion">ProteinIteratorUtils</a></dt>
<dd>
<div class="block">The mass of water (H2O).</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index"><span class="typeNameLink">WaveletTree</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_inference/fm_index/package-summary.html">com.compomics.util.experiment.identification.protein_inference.fm_index</a></dt>
<dd>
<div class="block">Wavelet tree.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#WaveletTree-byte:A-long:A-com.compomics.util.waiting.WaitingHandler-int-boolean-">WaveletTree(byte[], long[], WaitingHandler, int, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#WaveletTree-byte:A-long:A-com.compomics.util.waiting.WaitingHandler-int-">WaveletTree(byte[], long[], WaitingHandler, int)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html#WaveletTree-byte:A-com.compomics.util.waiting.WaitingHandler-com.compomics.util.experiment.identification.protein_inference.fm_index.WaveletTree.HuffmanNode-int-boolean-">WaveletTree(byte[], WaitingHandler, WaveletTree.HuffmanNode, int, boolean)</a></span> - Constructor for class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">WaveletTree</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/WaveletTree.HuffmanNode.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index"><span class="typeNameLink">WaveletTree.HuffmanNode</span></a> - Class in <a href="com/compomics/util/experiment/identification/protein_inference/fm_index/package-summary.html">com.compomics.util.experiment.identification.protein_inference.fm_index</a></dt>
<dd>
<div class="block">Class for huffman nodes.</div>
</dd>
<dt><a href="com/compomics/software/autoupdater/WebDAO.html" title="class in com.compomics.software.autoupdater"><span class="typeNameLink">WebDAO</span></a> - Class in <a href="com/compomics/software/autoupdater/package-summary.html">com.compomics.software.autoupdater</a></dt>
<dd>
<div class="block">WebDAO.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/WebDAO.html#WebDAO--">WebDAO()</a></span> - Constructor for class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/WebDAO.html" title="class in com.compomics.software.autoupdater">WebDAO</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/ftp/FTP.html#welcomeMsg">welcomeMsg</a></span> - Variable in class com.compomics.util.io.ftp.<a href="com/compomics/util/io/ftp/FTP.html" title="class in com.compomics.util.io.ftp">FTP</a></dt>
<dd>
<div class="block">Welcome message from the server, if any.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html#WINDOW_SIZE">WINDOW_SIZE</a></span> - Static variable in class com.compomics.util.experiment.identification.modification.scores.<a href="com/compomics/util/experiment/identification/modification/scores/PhosphoRS.html" title="class in com.compomics.util.experiment.identification.modification.scores">PhosphoRS</a></dt>
<dd>
<div class="block">The window size in m/z.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html#withinMassTolerance-double-int-">withinMassTolerance(double, int)</a></span> - Method in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/FMIndex.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">FMIndex</a></dt>
<dd>
<div class="block">Lookup, if mass can be described a combination of numX different amino
 acids</div>
</dd>
<dt><a href="com/compomics/util/io/export/WorkbookStyle.html" title="interface in com.compomics.util.io.export"><span class="typeNameLink">WorkbookStyle</span></a> - Interface in <a href="com/compomics/util/io/export/package-summary.html">com.compomics.util.io.export</a></dt>
<dd>
<div class="block">This interface sets the style of a workbook export.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#write-java.lang.String-">write(String)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Writes text to the export.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#write-java.lang.String-com.compomics.util.io.export.WorkbookStyle-">write(String, WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Writes text to the export.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#write-java.lang.String-com.compomics.util.io.export.WorkbookStyle-">write(String, WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/TextWriter.html#write-java.lang.String-com.compomics.util.io.export.WorkbookStyle-">write(String, WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/TextWriter.html" title="class in com.compomics.util.io.export.writers">TextWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#write-java.lang.String-">write(String)</a></span> - Method in class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Writes the content to the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/json/adapter/FileAdapter.html#write-com.google.gson.stream.JsonWriter-java.lang.Object-">write(JsonWriter, Object)</a></span> - Method in class com.compomics.util.io.json.adapter.<a href="com/compomics/util/io/json/adapter/FileAdapter.html" title="class in com.compomics.util.io.json.adapter">FileAdapter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html#writeAplFile-java.io.File-java.io.File-java.lang.String-">writeAplFile(File, File, String)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.mgf.<a href="com/compomics/util/experiment/io/mass_spectrometry/mgf/MgfReader.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.mgf">MgfReader</a></dt>
<dd>
<div class="block">Writes an apl file from an MGF file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#writeConfigurationToFile-java.io.File-">writeConfigurationToFile(File)</a></span> - Static method in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Writes all path configurations to the given file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#writeConfigurationToFile-java.io.BufferedWriter-">writeConfigurationToFile(BufferedWriter)</a></span> - Static method in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Writes the configuration file using the provided buffered writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/export/Ms2Exporter.html#writeHeader-java.io.BufferedWriter-">writeHeader(BufferedWriter)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.export.<a href="com/compomics/util/experiment/io/mass_spectrometry/export/Ms2Exporter.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.export">Ms2Exporter</a></dt>
<dd>
<div class="block">Writes the header of the file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#writeHeaderText-java.lang.String-">writeHeaderText(String)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Writes header text to the export.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#writeHeaderText-java.lang.String-com.compomics.util.io.export.WorkbookStyle-">writeHeaderText(String, WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Writes header text to the export.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#writeHeaderText-java.lang.String-com.compomics.util.io.export.WorkbookStyle-">writeHeaderText(String, WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/TextWriter.html#writeHeaderText-java.lang.String-com.compomics.util.io.export.WorkbookStyle-">writeHeaderText(String, WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/TextWriter.html" title="class in com.compomics.util.io.export.writers">TextWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#writeIndent--">writeIndent()</a></span> - Method in class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Writes the current indent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html#writeIndex-com.compomics.util.experiment.io.mass_spectrometry.mgf.MgfIndex-java.io.File-">writeIndex(MgfIndex, File)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.<a href="com/compomics/util/experiment/mass_spectrometry/SpectrumFactory.html" title="class in com.compomics.util.experiment.mass_spectrometry">SpectrumFactory</a></dt>
<dd>
<div class="block">Writes the given mgf file index in the given directory.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#writeLine-java.lang.String-">writeLine(String)</a></span> - Method in class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Writes the given line.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#writeLineDecreasedIndent-java.lang.String-">writeLineDecreasedIndent(String)</a></span> - Method in class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Writes the given line with a decreased indent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html#writeLineIncreasedIndent-java.lang.String-">writeLineIncreasedIndent(String)</a></span> - Method in class com.compomics.util.io.export.xml.<a href="com/compomics/util/io/export/xml/SimpleXmlWriter.html" title="class in com.compomics.util.io.export.xml">SimpleXmlWriter</a></dt>
<dd>
<div class="block">Writes the given line with an increased indent.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#writeMainTitle-java.lang.String-">writeMainTitle(String)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Writes the main title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/ExportWriter.html#writeMainTitle-java.lang.String-com.compomics.util.io.export.WorkbookStyle-">writeMainTitle(String, WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.<a href="com/compomics/util/io/export/ExportWriter.html" title="class in com.compomics.util.io.export">ExportWriter</a></dt>
<dd>
<div class="block">Writes the main title.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/ExcelWriter.html#writeMainTitle-java.lang.String-com.compomics.util.io.export.WorkbookStyle-">writeMainTitle(String, WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/ExcelWriter.html" title="class in com.compomics.util.io.export.writers">ExcelWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/export/writers/TextWriter.html#writeMainTitle-java.lang.String-com.compomics.util.io.export.WorkbookStyle-">writeMainTitle(String, WorkbookStyle)</a></span> - Method in class com.compomics.util.io.export.writers.<a href="com/compomics/util/io/export/writers/TextWriter.html" title="class in com.compomics.util.io.export.writers">TextWriter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#writeMgf-java.io.BufferedWriter-">writeMgf(BufferedWriter)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Writes the spectrum in the mgf format using the given writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html#writeMgf-java.io.BufferedWriter-java.util.HashMap-">writeMgf(BufferedWriter, HashMap&lt;String, String&gt;)</a></span> - Method in class com.compomics.util.experiment.mass_spectrometry.spectra.<a href="com/compomics/util/experiment/mass_spectrometry/spectra/Spectrum.html" title="class in com.compomics.util.experiment.mass_spectrometry.spectra">Spectrum</a></dt>
<dd>
<div class="block">Writes the spectrum in the mgf format using the given writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html#writeMsAmandaEnzymeFile-java.io.File-">writeMsAmandaEnzymeFile(File)</a></span> - Method in class com.compomics.util.experiment.biology.enzymes.<a href="com/compomics/util/experiment/biology/enzymes/EnzymeFactory.html" title="class in com.compomics.util.experiment.biology.enzymes">EnzymeFactory</a></dt>
<dd>
<div class="block">Creates the MS Amanda enzyme settings file corresponding to the enzymes
 loaded in the factory to the given file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/SerializationUtils.html#writeObject-java.lang.Object-java.io.File-">writeObject(Object, File)</a></span> - Static method in class com.compomics.util.io.file.<a href="com/compomics/util/io/file/SerializationUtils.html" title="class in com.compomics.util.io.file">SerializationUtils</a></dt>
<dd>
<div class="block">Writes an object to the destination file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/settings/UtilitiesPathParameters.html#writePathToFile-java.io.BufferedWriter-com.compomics.software.settings.UtilitiesPathParameters.UtilitiesPathKey-">writePathToFile(BufferedWriter, UtilitiesPathParameters.UtilitiesPathKey)</a></span> - Static method in class com.compomics.software.settings.<a href="com/compomics/software/settings/UtilitiesPathParameters.html" title="class in com.compomics.software.settings">UtilitiesPathParameters</a></dt>
<dd>
<div class="block">Writes the path of interest using the provided buffered writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/export/AplExporter.html#writeSpectrum-java.io.BufferedWriter-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-com.compomics.util.experiment.mass_spectrometry.FragmentationMethod-int-">writeSpectrum(BufferedWriter, Spectrum, FragmentationMethod, int)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.export.<a href="com/compomics/util/experiment/io/mass_spectrometry/export/AplExporter.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.export">AplExporter</a></dt>
<dd>
<div class="block">Writes the given spectrum in ms2 format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/mass_spectrometry/export/Ms2Exporter.html#writeSpectrum-java.io.BufferedWriter-com.compomics.util.experiment.mass_spectrometry.spectra.Spectrum-java.lang.Integer-">writeSpectrum(BufferedWriter, Spectrum, Integer)</a></span> - Static method in class com.compomics.util.experiment.io.mass_spectrometry.export.<a href="com/compomics/util/experiment/io/mass_spectrometry/export/Ms2Exporter.html" title="class in com.compomics.util.experiment.io.mass_spectrometry.export">Ms2Exporter</a></dt>
<dd>
<div class="block">Writes the given spectrum in ms2 format.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/software/autoupdater/FileDAO.html#writeStreamToDisk-java.io.InputStream-java.lang.String-java.io.File-">writeStreamToDisk(InputStream, String, File)</a></span> - Method in class com.compomics.software.autoupdater.<a href="com/compomics/software/autoupdater/FileDAO.html" title="class in com.compomics.software.autoupdater">FileDAO</a></dt>
<dd>
<div class="block">Writes a stream to disk.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBResultSet.html#writeToCSVFile-java.io.Writer-java.lang.String-">writeToCSVFile(Writer, String)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db">DBResultSet</a></dt>
<dd>
<div class="block">This method allows the caller to write the current dataset to the
 specified Writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/nucleotide/NucleotideSequence.html#writeToFASTAFile-java.io.PrintWriter-">writeToFASTAFile(PrintWriter)</a></span> - Method in class com.compomics.util.nucleotide.<a href="com/compomics/util/nucleotide/NucleotideSequence.html" title="class in com.compomics.util.nucleotide">NucleotideSequence</a></dt>
<dd>
<div class="block">This method can be used to append this nucleotide sequence to the
 FASTA DB flatfile the PrintWriter points to.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/protein/Protein.html#writeToFASTAFile-java.io.PrintWriter-">writeToFASTAFile(PrintWriter)</a></span> - Method in class com.compomics.util.protein.<a href="com/compomics/util/protein/Protein.html" title="class in com.compomics.util.protein">Protein</a></dt>
<dd>
<div class="block">This method can be used to append this protein to the
 FASTA DB flatfile the PrintWriter points to.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#writeToFile-java.io.File-">writeToFile(File)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Method to write to a given file</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#writeToFile-java.io.File-">writeToFile(File)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method allows the caller to write the spectrum file to the specified folder
 using its current filename.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/db/DBResultSet.html#writeToHTMLTable-java.io.Writer-int-">writeToHTMLTable(Writer, int)</a></span> - Method in class com.compomics.util.db.<a href="com/compomics/util/db/DBResultSet.html" title="class in com.compomics.util.db">DBResultSet</a></dt>
<dd>
<div class="block">This method allows the caller to write the current dataset to the
 specified Writer.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/general/IsotopicDistributionSpectrum.html#writeToStream-java.io.OutputStream-">writeToStream(OutputStream)</a></span> - Method in class com.compomics.util.general.<a href="com/compomics/util/general/IsotopicDistributionSpectrum.html" title="class in com.compomics.util.general">IsotopicDistributionSpectrum</a></dt>
<dd>
<div class="block">Method to write to a givern stream</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/interfaces/SpectrumFile.html#writeToStream-java.io.OutputStream-">writeToStream(OutputStream)</a></span> - Method in interface com.compomics.util.interfaces.<a href="com/compomics/util/interfaces/SpectrumFile.html" title="interface in com.compomics.util.interfaces">SpectrumFile</a></dt>
<dd>
<div class="block">This method allows to write the spectrum file to the specified OutputStream.</div>
</dd>
</dl>
<a name="I:X">
<!--   -->
</a>
<h2 class="title">X</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#X">X</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">X</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Unknown amino acid (Mascot).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html#X--">X()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/X.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">X</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#X_ION">X_ION</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Identifier for an x ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#X_ION">X_ION</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Identifier for an x ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#xAxisZoomRangeLowerValue">xAxisZoomRangeLowerValue</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The lower range for the current zoom range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#xAxisZoomRangeUpperValue">xAxisZoomRangeUpperValue</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The upper range for the current zoom range.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html#XMassDiff">XMassDiff</a></span> - Variable in class com.compomics.util.experiment.identification.protein_inference.fm_index.<a href="com/compomics/util/experiment/identification/protein_inference/fm_index/MatrixContent.html" title="class in com.compomics.util.experiment.identification.protein_inference.fm_index">MatrixContent</a></dt>
<dd>
<div class="block">The X mass difference,</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#xml">xml</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html#XML">XML</a></span> - Static variable in class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/FileFilterUtils.html" title="class in com.compomics.util.io.file.filefilters">FileFilterUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/io/file/filefilters/XmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters"><span class="typeNameLink">XmlFileFilter</span></a> - Class in <a href="com/compomics/util/io/file/filefilters/package-summary.html">com.compomics.util.io.file.filefilters</a></dt>
<dd>
<div class="block">File filter for *.xml files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/file/filefilters/XmlFileFilter.html#XmlFileFilter--">XmlFileFilter()</a></span> - Constructor for class com.compomics.util.io.file.filefilters.<a href="com/compomics/util/io/file/filefilters/XmlFileFilter.html" title="class in com.compomics.util.io.file.filefilters">XmlFileFilter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html" title="class in com.compomics.util.pride.validation"><span class="typeNameLink">XMLValidationErrorHandler</span></a> - Class in <a href="com/compomics/util/pride/validation/package-summary.html">com.compomics.util.pride.validation</a></dt>
<dd>
<div class="block">Edited using IntelliJ IDEA Date: 16-Sep-2005 Time: 15:26:12</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html#XMLValidationErrorHandler--">XMLValidationErrorHandler()</a></span> - Constructor for class com.compomics.util.pride.validation.<a href="com/compomics/util/pride/validation/XMLValidationErrorHandler.html" title="class in com.compomics.util.pride.validation">XMLValidationErrorHandler</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#xTagCount">xTagCount</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The number of X-axis tags.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#xtandem">xtandem</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The X! Tandem search engine.</div>
</dd>
<dt><a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders"><span class="typeNameLink">XTandemIdfileReader</span></a> - Class in <a href="com/compomics/util/experiment/io/identification/idfilereaders/package-summary.html">com.compomics.util.experiment.io.identification.idfilereaders</a></dt>
<dd>
<div class="block">This IdfileReader reads identifications from an X! Tandem xml result file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html#XTandemIdfileReader--">XTandemIdfileReader()</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">XTandemIdfileReader</a></dt>
<dd>
<div class="block">Default constructor for the purpose of instantiation.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html#XTandemIdfileReader-java.io.File-">XTandemIdfileReader(File)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">XTandemIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an X!Tandem xml result file reader.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html#XTandemIdfileReader-java.io.File-com.compomics.util.waiting.WaitingHandler-">XTandemIdfileReader(File, WaitingHandler)</a></span> - Constructor for class com.compomics.util.experiment.io.identification.idfilereaders.<a href="com/compomics/util/experiment/io/identification/idfilereaders/XTandemIdfileReader.html" title="class in com.compomics.util.experiment.io.identification.idfilereaders">XTandemIdfileReader</a></dt>
<dd>
<div class="block">Constructor for an X!Tandem xml result file reader.</div>
</dd>
<dt><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific"><span class="typeNameLink">XtandemParameters</span></a> - Class in <a href="com/compomics/util/parameters/identification/tool_specific/package-summary.html">com.compomics.util.parameters.identification.tool_specific</a></dt>
<dd>
<div class="block">The X!Tandem specific parameters.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html#XtandemParameters--">XtandemParameters()</a></span> - Constructor for class com.compomics.util.parameters.identification.tool_specific.<a href="com/compomics/util/parameters/identification/tool_specific/XtandemParameters.html" title="class in com.compomics.util.parameters.identification.tool_specific">XtandemParameters</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm"><span class="typeNameLink">XTandemParametersDialog</span></a> - Class in <a href="com/compomics/util/gui/parameters/identification/algorithm/package-summary.html">com.compomics.util.gui.parameters.identification.algorithm</a></dt>
<dd>
<div class="block">Dialog for the X!Tandem specific settings.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html#XTandemParametersDialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.XtandemParameters-com.compomics.util.parameters.identification.search.ModificationParameters-double-boolean-">XTandemParametersDialog(Frame, XtandemParameters, ModificationParameters, double, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">XTandemParametersDialog</a></dt>
<dd>
<div class="block">Creates new form XtandemParametersDialog with a frame as owner.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html#XTandemParametersDialog-java.awt.Dialog-java.awt.Frame-com.compomics.util.parameters.identification.tool_specific.XtandemParameters-com.compomics.util.parameters.identification.search.ModificationParameters-double-boolean-">XTandemParametersDialog(Dialog, Frame, XtandemParameters, ModificationParameters, double, boolean)</a></span> - Constructor for class com.compomics.util.gui.parameters.identification.algorithm.<a href="com/compomics/util/gui/parameters/identification/algorithm/XTandemParametersDialog.html" title="class in com.compomics.util.gui.parameters.identification.algorithm">XTandemParametersDialog</a></dt>
<dd>
<div class="block">Creates new form XtandemParametersDialog with a dialog as owner.</div>
</dd>
<dt><a href="com/compomics/util/gui/XYPlotFiltersDialog.html" title="class in com.compomics.util.gui"><span class="typeNameLink">XYPlotFiltersDialog</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">Dialog for editing the data filters for an XYPlottingDialog.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlotFiltersDialog.html#XYPlotFiltersDialog-com.compomics.util.gui.XYPlottingDialog-boolean-">XYPlotFiltersDialog(XYPlottingDialog, boolean)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlotFiltersDialog.html" title="class in com.compomics.util.gui">XYPlotFiltersDialog</a></dt>
<dd>
<div class="block">Creates a new XYPlotFiltersDialog.</div>
</dd>
<dt><a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui"><span class="typeNameLink">XYPlottingDialog</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">A dialog that makes it straightforward to inspect compare the values of two
 columns in a table in a XY plot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/XYPlottingDialog.html#XYPlottingDialog-java.awt.Frame-javax.swing.JTable-java.lang.String-com.compomics.util.gui.XYPlottingDialog.PlottingDialogPlotType-java.util.ArrayList-java.awt.Image-java.awt.Image-boolean-">XYPlottingDialog(Frame, JTable, String, XYPlottingDialog.PlottingDialogPlotType, ArrayList&lt;String&gt;, Image, Image, boolean)</a></span> - Constructor for class com.compomics.util.gui.<a href="com/compomics/util/gui/XYPlottingDialog.html" title="class in com.compomics.util.gui">XYPlottingDialog</a></dt>
<dd>
<div class="block">Creates a new XYPlottingDialog.</div>
</dd>
<dt><a href="com/compomics/util/gui/XYPlottingDialog.PlottingDialogPlotType.html" title="enum in com.compomics.util.gui"><span class="typeNameLink">XYPlottingDialog.PlottingDialogPlotType</span></a> - Enum in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">The plotting dialog types.</div>
</dd>
<dt><a href="com/compomics/util/gui/XYPlottingDialog.SelectedValuesTableFilter.html" title="class in com.compomics.util.gui"><span class="typeNameLink">XYPlottingDialog.SelectedValuesTableFilter</span></a> - Class in <a href="com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></dt>
<dd>
<div class="block">A filter that filters the table based on if the data point is selected in
 the plot or not.</div>
</dd>
<dt><a href="com/compomics/util/XYZDataPoint.html" title="class in com.compomics.util"><span class="typeNameLink">XYZDataPoint</span></a> - Class in <a href="com/compomics/util/package-summary.html">com.compomics.util</a></dt>
<dd>
<div class="block">Object that stores data about one data point in an XYZ plot.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/XYZDataPoint.html#XYZDataPoint-double-double-double-">XYZDataPoint(double, double, double)</a></span> - Constructor for class com.compomics.util.<a href="com/compomics/util/XYZDataPoint.html" title="class in com.compomics.util">XYZDataPoint</a></dt>
<dd>
<div class="block">Create a new XYZDataPoint.</div>
</dd>
</dl>
<a name="I:Y">
<!--   -->
</a>
<h2 class="title">Y</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#Y">Y</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#Y_ION">Y_ION</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Identifier for a y ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#Y_ION">Y_ION</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Identifier for a y ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html#yAxisZoomExcludesBackgroundPeaks--">yAxisZoomExcludesBackgroundPeaks()</a></span> - Method in class com.compomics.util.experiment.identification.spectrum_annotation.<a href="com/compomics/util/experiment/identification/spectrum_annotation/AnnotationParameters.html" title="class in com.compomics.util.experiment.identification.spectrum_annotation">AnnotationParameters</a></dt>
<dd>
<div class="block">Returns true if the automatic y-axis zoom excludes background peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#yAxisZoomExcludesBackgroundPeaks">yAxisZoomExcludesBackgroundPeaks</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">If true, the automatic y-axis zoom excludes the background peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#yAxisZoomOnlyExcludesBackgroundPeaks--">yAxisZoomOnlyExcludesBackgroundPeaks()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns true if the automatic y-axis zoom excludes background peaks.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#yDataIsPositive">yDataIsPositive</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">If set to true, all y data is assumed to be positive.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#yDataIsPositive--">yDataIsPositive()</a></span> - Method in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">Returns true of all the y-data is to be assumed as positive.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/gui/spectrum/GraphicsPanel.html#yTagCount">yTagCount</a></span> - Variable in class com.compomics.util.gui.spectrum.<a href="com/compomics/util/gui/spectrum/GraphicsPanel.html" title="class in com.compomics.util.gui.spectrum">GraphicsPanel</a></dt>
<dd>
<div class="block">The number of Y-axis tags.</div>
</dd>
</dl>
<a name="I:Z">
<!--   -->
</a>
<h2 class="title">Z</h2>
<dl>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html#Z">Z</a></span> - Static variable in class com.compomics.util.experiment.biology.aminoacids.<a href="com/compomics/util/experiment/biology/aminoacids/AminoAcid.html" title="class in com.compomics.util.experiment.biology.aminoacids">AminoAcid</a></dt>
<dd>&nbsp;</dd>
<dt><a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl"><span class="typeNameLink">Z</span></a> - Class in <a href="com/compomics/util/experiment/biology/aminoacids/impl/package-summary.html">com.compomics.util.experiment.biology.aminoacids.impl</a></dt>
<dd>
<div class="block">Glu or Gln: Glx (Mascot).</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html#Z--">Z()</a></span> - Constructor for class com.compomics.util.experiment.biology.aminoacids.impl.<a href="com/compomics/util/experiment/biology/aminoacids/impl/Z.html" title="class in com.compomics.util.experiment.biology.aminoacids.impl">Z</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html#Z_ION">Z_ION</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/PeptideFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">PeptideFragmentIon</a></dt>
<dd>
<div class="block">Identifier for a z ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html#Z_ION">Z_ION</a></span> - Static variable in class com.compomics.util.experiment.biology.ions.impl.<a href="com/compomics/util/experiment/biology/ions/impl/TagFragmentIon.html" title="class in com.compomics.util.experiment.biology.ions.impl">TagFragmentIon</a></dt>
<dd>
<div class="block">Identifier for a z ion.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/Advocate.html#zCore">zCore</a></span> - Static variable in class com.compomics.util.experiment.identification.<a href="com/compomics/util/experiment/identification/Advocate.html" title="class in com.compomics.util.experiment.identification">Advocate</a></dt>
<dd>
<div class="block">The ZCore search engine, ETD search engine.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html#zero">zero</a></span> - Static variable in class com.compomics.util.experiment.identification.utils.<a href="com/compomics/util/experiment/identification/utils/ModificationUtils.html" title="class in com.compomics.util.experiment.identification.utils">ModificationUtils</a></dt>
<dd>
<div class="block">Array for n-term.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/ZipUtils.html#zip-java.io.File-java.io.File-">zip(File, File)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression">ZipUtils</a></dt>
<dd>
<div class="block">Zips a file.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/ZipUtils.html#zip-java.io.File-java.io.File-com.compomics.util.waiting.WaitingHandler-long-">zip(File, File, WaitingHandler, long)</a></span> - Static method in class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression">ZipUtils</a></dt>
<dd>
<div class="block">Zips a file.</div>
</dd>
<dt><a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression"><span class="typeNameLink">ZipUtils</span></a> - Class in <a href="com/compomics/util/io/compression/package-summary.html">com.compomics.util.io.compression</a></dt>
<dd>
<div class="block">Convenience class for the handling of zip files.</div>
</dd>
<dt><span class="memberNameLink"><a href="com/compomics/util/io/compression/ZipUtils.html#ZipUtils--">ZipUtils()</a></span> - Constructor for class com.compomics.util.io.compression.<a href="com/compomics/util/io/compression/ZipUtils.html" title="class in com.compomics.util.io.compression">ZipUtils</a></dt>
<dd>&nbsp;</dd>
</dl>
<a href="#I:Z:Z:D">$</a>&nbsp;<a href="#I:A">A</a>&nbsp;<a href="#I:B">B</a>&nbsp;<a href="#I:C">C</a>&nbsp;<a href="#I:D">D</a>&nbsp;<a href="#I:E">E</a>&nbsp;<a href="#I:F">F</a>&nbsp;<a href="#I:G">G</a>&nbsp;<a href="#I:H">H</a>&nbsp;<a href="#I:I">I</a>&nbsp;<a href="#I:J">J</a>&nbsp;<a href="#I:K">K</a>&nbsp;<a href="#I:L">L</a>&nbsp;<a href="#I:M">M</a>&nbsp;<a href="#I:N">N</a>&nbsp;<a href="#I:O">O</a>&nbsp;<a href="#I:P">P</a>&nbsp;<a href="#I:Q">Q</a>&nbsp;<a href="#I:R">R</a>&nbsp;<a href="#I:S">S</a>&nbsp;<a href="#I:T">T</a>&nbsp;<a href="#I:U">U</a>&nbsp;<a href="#I:V">V</a>&nbsp;<a href="#I:W">W</a>&nbsp;<a href="#I:X">X</a>&nbsp;<a href="#I:Y">Y</a>&nbsp;<a href="#I:Z">Z</a>&nbsp;</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="overview-summary.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="index.html?index-all.html" target="_top">Frames</a></li>
<li><a href="index-all.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2018. All rights reserved.</small></p>
</body>
</html>
