<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Package com.compomics.util.experiment.biology (computational omics utilities 4.12.16 API)</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Package com.compomics.util.experiment.biology (computational omics utilities 4.12.16 API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/compomics/util/experiment/biology/package-use.html" target="_top">Frames</a></li>
<li><a href="package-use.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h1 title="Uses of Package com.compomics.util.experiment.biology" class="title">Uses of Package<br>com.compomics.util.experiment.biology</h1>
</div>
<div class="contentContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.cli.enzymes">com.compomics.cli.enzymes</a></td>
<td class="colLast">
<div class="block">Command line for the handling of enzymes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.cli.modifications">com.compomics.cli.modifications</a></td>
<td class="colLast">
<div class="block">Command line for the handling of modifications.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment">com.compomics.util.experiment</a></td>
<td class="colLast">
<div class="block">Gene factory classes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.biology">com.compomics.util.experiment.biology</a></td>
<td class="colLast">
<div class="block">Experiment classes related to biological entities.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.biology.aminoacids">com.compomics.util.experiment.biology.aminoacids</a></td>
<td class="colLast">
<div class="block">The amino acid classes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.biology.atoms">com.compomics.util.experiment.biology.atoms</a></td>
<td class="colLast">
<div class="block">The atom classes.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.biology.ions">com.compomics.util.experiment.biology.ions</a></td>
<td class="colLast">
<div class="block">Experiment classes related to ions.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.biology.mutations">com.compomics.util.experiment.biology.mutations</a></td>
<td class="colLast">
<div class="block">Experiment classes related to mutations.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification">com.compomics.util.experiment.identification</a></td>
<td class="colLast">
<div class="block">Experiment classes related to identifications.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.amino_acid_tags">com.compomics.util.experiment.identification.amino_acid_tags</a></td>
<td class="colLast">
<div class="block">Amino Acid Tag classes.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.amino_acid_tags.matchers">com.compomics.util.experiment.identification.amino_acid_tags.matchers</a></td>
<td class="colLast">
<div class="block">Tag matching classes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.filtering">com.compomics.util.experiment.identification.filtering</a></td>
<td class="colLast">
<div class="block">Classes for the filtering of identification objects.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.identification_parameters">com.compomics.util.experiment.identification.identification_parameters</a></td>
<td class="colLast">
<div class="block">Parameters settings for the spectrum identification algorithms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.matches">com.compomics.util.experiment.identification.matches</a></td>
<td class="colLast">
<div class="block">Experiment classes related to matches.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.peptide_fragmentation">com.compomics.util.experiment.identification.peptide_fragmentation</a></td>
<td class="colLast">
<div class="block">Peptide fragment intensities.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.peptide_fragmentation.models">com.compomics.util.experiment.identification.peptide_fragmentation.models</a></td>
<td class="colLast">
<div class="block">Predictors for peptide fragment intensities.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration</a></td>
<td class="colLast">
<div class="block">This package contains the classes needed to configure the ms2pip features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></td>
<td class="colLast">
<div class="block">This package contains features implemented for ms2pip.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic</a></td>
<td class="colLast">
<div class="block">This package contains generic features implemented for ms2pip.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation</a></td>
<td class="colLast">
<div class="block">This package contains the classes needed to generate ms2pip features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.protein_inference">com.compomics.util.experiment.identification.protein_inference</a></td>
<td class="colLast">
<div class="block">Classes related to protein inference.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.protein_inference.fm_index">com.compomics.util.experiment.identification.protein_inference.fm_index</a></td>
<td class="colLast">
<div class="block">Protein sequence database indexing classes based on Burrows-Wheeler transform.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.protein_inference.proteintree">com.compomics.util.experiment.identification.protein_inference.proteintree</a></td>
<td class="colLast">
<div class="block">Protein sequence database indexing classes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.protein_sequences">com.compomics.util.experiment.identification.protein_sequences</a></td>
<td class="colLast">
<div class="block">Protein sequence handling related classes.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.protein_sequences.digestion">com.compomics.util.experiment.identification.protein_sequences.digestion</a></td>
<td class="colLast">
<div class="block">This package contains classes used to digest protein sequences into peptides.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">com.compomics.util.experiment.identification.protein_sequences.digestion.iterators</a></td>
<td class="colLast">
<div class="block">This package contains implementations of the SequenceIterator interface.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.psm_scoring">com.compomics.util.experiment.identification.psm_scoring</a></td>
<td class="colLast">
<div class="block">Main PSM scoring class.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.psm_scoring.psm_scores">com.compomics.util.experiment.identification.psm_scoring.psm_scores</a></td>
<td class="colLast">
<div class="block">Main PTM scoring classes.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.ptm">com.compomics.util.experiment.identification.ptm</a></td>
<td class="colLast">
<div class="block">Additional PSM scoring classes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.ptm.ptmscores">com.compomics.util.experiment.identification.ptm.ptmscores</a></td>
<td class="colLast">
<div class="block">PTM scores.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.spectrum_annotation">com.compomics.util.experiment.identification.spectrum_annotation</a></td>
<td class="colLast">
<div class="block">Classes used for ms2 spectrum annotation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators</a></td>
<td class="colLast">
<div class="block">This package contains implementations of the SimpleAnnotator allowing the annotation of specific sets of ions.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators</a></td>
<td class="colLast">
<div class="block">This package contains implementations of the SpectrumAnnotator allowing the annotation of all ions.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.identification.spectrum_assumptions">com.compomics.util.experiment.identification.spectrum_assumptions</a></td>
<td class="colLast">
<div class="block">Implementations of the SpectrumIdentificationAssumption.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.experiment.quantification.reporterion">com.compomics.util.experiment.quantification.reporterion</a></td>
<td class="colLast">
<div class="block">Experiment classes related to reporter ions.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.gui">com.compomics.util.gui</a></td>
<td class="colLast">
<div class="block">Basic GUI classes.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.gui.atoms">com.compomics.util.gui.atoms</a></td>
<td class="colLast">
<div class="block">Atoms GUI dialogs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.gui.parameters.identification_parameters">com.compomics.util.gui.parameters.identification_parameters</a></td>
<td class="colLast">
<div class="block">GUI classes for parameters, settings and preferences used for identification.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.gui.ptm">com.compomics.util.gui.ptm</a></td>
<td class="colLast">
<div class="block">PTM GUI dialogs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.compomics.util.gui.spectrum">com.compomics.util.gui.spectrum</a></td>
<td class="colLast">
<div class="block">Spectrum and Chromatogram visualization GUI classes.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.compomics.util.preferences">com.compomics.util.preferences</a></td>
<td class="colLast">
<div class="block">Utilities preferences classes.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.cli.enzymes">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/cli/enzymes/package-summary.html">com.compomics.cli.enzymes</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Enzyme.html#com.compomics.cli.enzymes">Enzyme</a>
<div class="block">This class models an enzyme.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.cli.modifications">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/cli/modifications/package-summary.html">com.compomics.cli.modifications</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/PTM.html#com.compomics.cli.modifications">PTM</a>
<div class="block">This class models a post-translational modification.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/package-summary.html">com.compomics.util.experiment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Sample.html#com.compomics.util.experiment">Sample</a>
<div class="block">This class models a proteomic sample.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.biology">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcid.html#com.compomics.util.experiment.biology">AminoAcid</a>
<div class="block">Class representing amino acids.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcid.Property.html#com.compomics.util.experiment.biology">AminoAcid.Property</a>
<div class="block">Properties of the amino acids.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcidPattern.html#com.compomics.util.experiment.biology">AminoAcidPattern</a>
<div class="block">An amino acid pattern is a sequence of amino acids.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcidSequence.html#com.compomics.util.experiment.biology">AminoAcidSequence</a>
<div class="block">This class represents a series of amino acids with associated modifications.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Atom.html#com.compomics.util.experiment.biology">Atom</a>
<div class="block">This interface contains information about atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AtomChain.html#com.compomics.util.experiment.biology">AtomChain</a>
<div class="block">A chain of atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AtomImpl.html#com.compomics.util.experiment.biology">AtomImpl</a>
<div class="block">Class for a specific atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/ElementaryElement.html#com.compomics.util.experiment.biology">ElementaryElement</a>
<div class="block">This class can be used to retrieve elementary elements like a neutron.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Enzyme.html#com.compomics.util.experiment.biology">Enzyme</a>
<div class="block">This class models an enzyme.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/EnzymeFactory.html#com.compomics.util.experiment.biology">EnzymeFactory</a>
<div class="block">This factory will provide the implemented enzymes.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Ion.html#com.compomics.util.experiment.biology">Ion</a>
<div class="block">This class models an ion.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Ion.IonType.html#com.compomics.util.experiment.biology">Ion.IonType</a>
<div class="block">An enumerator of the supported ion types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/IonFactory.html#com.compomics.util.experiment.biology">IonFactory</a>
<div class="block">This factory generates the expected ions from a peptide.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/NeutralLoss.html#com.compomics.util.experiment.biology">NeutralLoss</a>
<div class="block">This class represents a neutral loss.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/NeutralLossCombination.html#com.compomics.util.experiment.biology">NeutralLossCombination</a>
<div class="block">This class represents a combination of multiple neutral losses.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.biology">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Protein.html#com.compomics.util.experiment.biology">Protein</a>
<div class="block">This class models a protein.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/PTM.html#com.compomics.util.experiment.biology">PTM</a>
<div class="block">This class models a post-translational modification.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/PTMFactory.html#com.compomics.util.experiment.biology">PTMFactory</a>
<div class="block">This factory will load PTM from an XML file and provide them on demand as
 standard class.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Sample.html#com.compomics.util.experiment.biology">Sample</a>
<div class="block">This class models a proteomic sample.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.biology.aminoacids">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/biology/aminoacids/package-summary.html">com.compomics.util.experiment.biology.aminoacids</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcid.html#com.compomics.util.experiment.biology.aminoacids">AminoAcid</a>
<div class="block">Class representing amino acids.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.biology.atoms">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/biology/atoms/package-summary.html">com.compomics.util.experiment.biology.atoms</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Atom.html#com.compomics.util.experiment.biology.atoms">Atom</a>
<div class="block">This interface contains information about atoms.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.biology.ions">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/biology/ions/package-summary.html">com.compomics.util.experiment.biology.ions</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcid.html#com.compomics.util.experiment.biology.ions">AminoAcid</a>
<div class="block">Class representing amino acids.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AtomChain.html#com.compomics.util.experiment.biology.ions">AtomChain</a>
<div class="block">A chain of atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Ion.html#com.compomics.util.experiment.biology.ions">Ion</a>
<div class="block">This class models an ion.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/NeutralLoss.html#com.compomics.util.experiment.biology.ions">NeutralLoss</a>
<div class="block">This class represents a neutral loss.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.biology.mutations">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/biology/mutations/package-summary.html">com.compomics.util.experiment.biology.mutations</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcid.html#com.compomics.util.experiment.biology.mutations">AminoAcid</a>
<div class="block">Class representing amino acids.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/package-summary.html">com.compomics.util.experiment.identification</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.amino_acid_tags">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/amino_acid_tags/package-summary.html">com.compomics.util.experiment.identification.amino_acid_tags</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcid.html#com.compomics.util.experiment.identification.amino_acid_tags">AminoAcid</a>
<div class="block">Class representing amino acids.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcidPattern.html#com.compomics.util.experiment.identification.amino_acid_tags">AminoAcidPattern</a>
<div class="block">An amino acid pattern is a sequence of amino acids.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcidSequence.html#com.compomics.util.experiment.identification.amino_acid_tags">AminoAcidSequence</a>
<div class="block">This class represents a series of amino acids with associated modifications.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/PTM.html#com.compomics.util.experiment.identification.amino_acid_tags">PTM</a>
<div class="block">This class models a post-translational modification.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.amino_acid_tags.matchers">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/amino_acid_tags/matchers/package-summary.html">com.compomics.util.experiment.identification.amino_acid_tags.matchers</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcid.html#com.compomics.util.experiment.identification.amino_acid_tags.matchers">AminoAcid</a>
<div class="block">Class representing amino acids.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.filtering">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/filtering/package-summary.html">com.compomics.util.experiment.identification.filtering</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification.filtering">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.identification_parameters">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/identification_parameters/package-summary.html">com.compomics.util.experiment.identification.identification_parameters</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Enzyme.html#com.compomics.util.experiment.identification.identification_parameters">Enzyme</a>
<div class="block">This class models an enzyme.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/PTM.html#com.compomics.util.experiment.identification.identification_parameters">PTM</a>
<div class="block">This class models a post-translational modification.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.matches">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/matches/package-summary.html">com.compomics.util.experiment.identification.matches</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Enzyme.html#com.compomics.util.experiment.identification.matches">Enzyme</a>
<div class="block">This class models an enzyme.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Ion.html#com.compomics.util.experiment.identification.matches">Ion</a>
<div class="block">This class models an ion.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification.matches">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.peptide_fragmentation">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/peptide_fragmentation/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Ion.html#com.compomics.util.experiment.identification.peptide_fragmentation">Ion</a>
<div class="block">This class models an ion.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.peptide_fragmentation.models">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/peptide_fragmentation/models/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Ion.html#com.compomics.util.experiment.identification.peptide_fragmentation.models">Ion</a>
<div class="block">This class models an ion.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcid.Property.html#com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration">AminoAcid.Property</a>
<div class="block">Properties of the amino acids.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcid.Property.html#com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features">AminoAcid.Property</a>
<div class="block">Properties of the amino acids.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_configuration/features/generic/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcid.Property.html#com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_configuration.features.generic">AminoAcid.Property</a>
<div class="block">Properties of the amino acids.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/peptide_fragmentation/models/ms2pip/features_generation/package-summary.html">com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification.peptide_fragmentation.models.ms2pip.features_generation">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.protein_inference">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/protein_inference/package-summary.html">com.compomics.util.experiment.identification.protein_inference</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification.protein_inference">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.protein_inference.fm_index">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/protein_inference/fm_index/package-summary.html">com.compomics.util.experiment.identification.protein_inference.fm_index</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/PTM.html#com.compomics.util.experiment.identification.protein_inference.fm_index">PTM</a>
<div class="block">This class models a post-translational modification.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.protein_inference.proteintree">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/protein_inference/proteintree/package-summary.html">com.compomics.util.experiment.identification.protein_inference.proteintree</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcidSequence.html#com.compomics.util.experiment.identification.protein_inference.proteintree">AminoAcidSequence</a>
<div class="block">This class represents a series of amino acids with associated modifications.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Enzyme.html#com.compomics.util.experiment.identification.protein_inference.proteintree">Enzyme</a>
<div class="block">This class models an enzyme.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.protein_sequences">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/protein_sequences/package-summary.html">com.compomics.util.experiment.identification.protein_sequences</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Protein.html#com.compomics.util.experiment.identification.protein_sequences">Protein</a>
<div class="block">This class models a protein.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.protein_sequences.digestion">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/protein_sequences/digestion/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcidPattern.html#com.compomics.util.experiment.identification.protein_sequences.digestion">AminoAcidPattern</a>
<div class="block">An amino acid pattern is a sequence of amino acids.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification.protein_sequences.digestion">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/protein_sequences/digestion/iterators/package-summary.html">com.compomics.util.experiment.identification.protein_sequences.digestion.iterators</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Enzyme.html#com.compomics.util.experiment.identification.protein_sequences.digestion.iterators">Enzyme</a>
<div class="block">This class models an enzyme.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.psm_scoring">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/psm_scoring/package-summary.html">com.compomics.util.experiment.identification.psm_scoring</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification.psm_scoring">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.psm_scoring.psm_scores">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/psm_scoring/psm_scores/package-summary.html">com.compomics.util.experiment.identification.psm_scoring.psm_scores</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification.psm_scoring.psm_scores">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.ptm">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/ptm/package-summary.html">com.compomics.util.experiment.identification.ptm</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification.ptm">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/PTM.html#com.compomics.util.experiment.identification.ptm">PTM</a>
<div class="block">This class models a post-translational modification.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.ptm.ptmscores">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/ptm/ptmscores/package-summary.html">com.compomics.util.experiment.identification.ptm.ptmscores</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification.ptm.ptmscores">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/PTM.html#com.compomics.util.experiment.identification.ptm.ptmscores">PTM</a>
<div class="block">This class models a post-translational modification.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.spectrum_annotation">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/spectrum_annotation/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Ion.html#com.compomics.util.experiment.identification.spectrum_annotation">Ion</a>
<div class="block">This class models an ion.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Ion.IonType.html#com.compomics.util.experiment.identification.spectrum_annotation">Ion.IonType</a>
<div class="block">An enumerator of the supported ion types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/IonFactory.html#com.compomics.util.experiment.identification.spectrum_annotation">IonFactory</a>
<div class="block">This factory generates the expected ions from a peptide.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/NeutralLoss.html#com.compomics.util.experiment.identification.spectrum_annotation">NeutralLoss</a>
<div class="block">This class represents a neutral loss.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/spectrum_annotation/simple_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification.spectrum_annotation.simple_annotators">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/spectrum_annotation/spectrum_annotators/package-summary.html">com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Ion.html#com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">Ion</a>
<div class="block">This class models an ion.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification.spectrum_annotation.spectrum_annotators">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.identification.spectrum_assumptions">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/identification/spectrum_assumptions/package-summary.html">com.compomics.util.experiment.identification.spectrum_assumptions</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.experiment.identification.spectrum_assumptions">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.experiment.quantification.reporterion">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/experiment/quantification/reporterion/package-summary.html">com.compomics.util.experiment.quantification.reporterion</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Sample.html#com.compomics.util.experiment.quantification.reporterion">Sample</a>
<div class="block">This class models a proteomic sample.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.gui">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/gui/package-summary.html">com.compomics.util.gui</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AminoAcidPattern.html#com.compomics.util.gui">AminoAcidPattern</a>
<div class="block">An amino acid pattern is a sequence of amino acids.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.gui.atoms">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/gui/atoms/package-summary.html">com.compomics.util.gui.atoms</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Atom.html#com.compomics.util.gui.atoms">Atom</a>
<div class="block">This interface contains information about atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/AtomChain.html#com.compomics.util.gui.atoms">AtomChain</a>
<div class="block">A chain of atoms.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.gui.parameters.identification_parameters">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/gui/parameters/identification_parameters/package-summary.html">com.compomics.util.gui.parameters.identification_parameters</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Enzyme.html#com.compomics.util.gui.parameters.identification_parameters">Enzyme</a>
<div class="block">This class models an enzyme.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/NeutralLoss.html#com.compomics.util.gui.parameters.identification_parameters">NeutralLoss</a>
<div class="block">This class represents a neutral loss.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.gui.ptm">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/gui/ptm/package-summary.html">com.compomics.util.gui.ptm</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/PTM.html#com.compomics.util.gui.ptm">PTM</a>
<div class="block">This class models a post-translational modification.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.gui.spectrum">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/gui/spectrum/package-summary.html">com.compomics.util.gui.spectrum</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Ion.html#com.compomics.util.gui.spectrum">Ion</a>
<div class="block">This class models an ion.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Ion.IonType.html#com.compomics.util.gui.spectrum">Ion.IonType</a>
<div class="block">An enumerator of the supported ion types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/NeutralLoss.html#com.compomics.util.gui.spectrum">NeutralLoss</a>
<div class="block">This class represents a neutral loss.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Peptide.html#com.compomics.util.gui.spectrum">Peptide</a>
<div class="block">This class models a peptide.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.compomics.util.preferences">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../com/compomics/util/experiment/biology/package-summary.html">com.compomics.util.experiment.biology</a> used by <a href="../../../../../com/compomics/util/preferences/package-summary.html">com.compomics.util.preferences</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../com/compomics/util/experiment/biology/class-use/Enzyme.html#com.compomics.util.preferences">Enzyme</a>
<div class="block">This class models an enzyme.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?com/compomics/util/experiment/biology/package-use.html" target="_top">Frames</a></li>
<li><a href="package-use.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2020. All rights reserved.</small></p>
</body>
</html>
